// primeng/components/tree/tree.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from '@angular/core/src/linker/ng_module_factory';
import * as import1 from 'primeng/components/tree/tree';
import * as import2 from '@angular/common/src/common_module';
import * as import3 from 'primeng/components/common/shared';
import * as import4 from '@angular/common/src/localization';
import * as import5 from '@angular/core/src/di/injector';
import * as import6 from '@angular/core/src/i18n/tokens';
import * as import7 from '@angular/core/src/change_detection/change_detection_util';
import * as import8 from '@angular/core/src/linker/view';
import * as import9 from '@angular/core/src/linker/view_utils';
import * as import10 from '@angular/core/src/render/api';
import * as import11 from '@angular/core/src/metadata/view';
import * as import12 from '@angular/core/src/linker/query_list';
import * as import13 from '@angular/core/src/linker/view_type';
import * as import14 from '@angular/core/src/change_detection/constants';
import * as import15 from '@angular/core/src/linker/component_factory';
import * as import16 from '@angular/core/src/linker/view_container';
import * as import17 from '../../../@angular/common/src/directives/ng_class.ngfactory';
import * as import18 from '../../../@angular/common/src/directives/ng_style.ngfactory';
import * as import19 from '../../../@angular/common/src/directives/ng_for.ngfactory';
import * as import20 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import21 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import22 from '@angular/core/src/linker/element_ref';
import * as import23 from '@angular/core/src/linker/template_ref';
import * as import24 from '@angular/common/src/directives/ng_for';
import * as import25 from '@angular/common/src/directives/ng_class';
import * as import26 from '@angular/common/src/directives/ng_style';
import * as import27 from '../../../@angular/common/src/directives/ng_if.ngfactory';
import * as import28 from '@angular/common/src/directives/ng_if';
import * as import29 from '@angular/core/src/security';
class TreeModuleInjector extends import0.NgModuleInjector<import1.TreeModule> {
  _CommonModule_0:import2.CommonModule;
  _SharedModule_1:import3.SharedModule;
  _TreeModule_2:import1.TreeModule;
  __NgLocalization_3:import4.NgLocaleLocalization;
  constructor(parent:import5.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_3():import4.NgLocaleLocalization {
    if ((this.__NgLocalization_3 == null)) { (this.__NgLocalization_3 = new import4.NgLocaleLocalization(this.parent.get(import6.LOCALE_ID))); }
    return this.__NgLocalization_3;
  }
  createInternal():import1.TreeModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._SharedModule_1 = new import3.SharedModule();
    this._TreeModule_2 = new import1.TreeModule();
    return this._TreeModule_2;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.SharedModule)) { return this._SharedModule_1; }
    if ((token === import1.TreeModule)) { return this._TreeModule_2; }
    if ((token === import4.NgLocalization)) { return this._NgLocalization_3; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const TreeModuleNgFactory:import0.NgModuleFactory<import1.TreeModule> = new import0.NgModuleFactory(TreeModuleInjector,import1.TreeModule);
export class Wrapper_Tree {
  /*private*/ _eventHandler:Function;
  context:import1.Tree;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  /*private*/ _expr_5:any;
  /*private*/ _expr_6:any;
  subscription0:any;
  subscription1:any;
  subscription2:any;
  subscription3:any;
  subscription4:any;
  subscription5:any;
  constructor() {
    this._changed = false;
    this.context = new import1.Tree();
    this._expr_0 = import7.UNINITIALIZED;
    this._expr_1 = import7.UNINITIALIZED;
    this._expr_2 = import7.UNINITIALIZED;
    this._expr_3 = import7.UNINITIALIZED;
    this._expr_4 = import7.UNINITIALIZED;
    this._expr_5 = import7.UNINITIALIZED;
    this._expr_6 = import7.UNINITIALIZED;
  }
  ngOnDetach(view:import8.AppView<any>,componentView:import8.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
    (this.subscription0 && this.subscription0.unsubscribe());
    (this.subscription1 && this.subscription1.unsubscribe());
    (this.subscription2 && this.subscription2.unsubscribe());
    (this.subscription3 && this.subscription3.unsubscribe());
    (this.subscription4 && this.subscription4.unsubscribe());
    (this.subscription5 && this.subscription5.unsubscribe());
  }
  check_value(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.value = currValue;
      this._expr_0 = currValue;
    }
  }
  check_selectionMode(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.selectionMode = currValue;
      this._expr_1 = currValue;
    }
  }
  check_selection(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_2,currValue))) {
      this._changed = true;
      this.context.selection = currValue;
      this._expr_2 = currValue;
    }
  }
  check_style(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_3,currValue))) {
      this._changed = true;
      this.context.style = currValue;
      this._expr_3 = currValue;
    }
  }
  check_styleClass(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_4,currValue))) {
      this._changed = true;
      this.context.styleClass = currValue;
      this._expr_4 = currValue;
    }
  }
  check_contextMenu(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_5,currValue))) {
      this._changed = true;
      this.context.contextMenu = currValue;
      this._expr_5 = currValue;
    }
  }
  check_layout(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_6,currValue))) {
      this._changed = true;
      this.context.layout = currValue;
      this._expr_6 = currValue;
    }
  }
  ngDoCheck(view:import8.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    return changed;
  }
  checkHost(view:import8.AppView<any>,componentView:import8.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import8.AppView<any>,_eventHandler:any,emit0:boolean,emit1:boolean,emit2:boolean,emit3:boolean,emit4:boolean,emit5:boolean):void {
    this._eventHandler = _eventHandler;
    if (emit0) { (this.subscription0 = this.context.selectionChange.subscribe(_eventHandler.bind(view,'selectionChange'))); }
    if (emit1) { (this.subscription1 = this.context.onNodeSelect.subscribe(_eventHandler.bind(view,'onNodeSelect'))); }
    if (emit2) { (this.subscription2 = this.context.onNodeUnselect.subscribe(_eventHandler.bind(view,'onNodeUnselect'))); }
    if (emit3) { (this.subscription3 = this.context.onNodeExpand.subscribe(_eventHandler.bind(view,'onNodeExpand'))); }
    if (emit4) { (this.subscription4 = this.context.onNodeCollapse.subscribe(_eventHandler.bind(view,'onNodeCollapse'))); }
    if (emit5) { (this.subscription5 = this.context.onNodeContextMenuSelect.subscribe(_eventHandler.bind(view,'onNodeContextMenuSelect'))); }
  }
}
export class Wrapper_UITreeNode {
  /*private*/ _eventHandler:Function;
  context:import1.UITreeNode;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  constructor(p0:any) {
    this._changed = false;
    this.context = new import1.UITreeNode(p0);
    this._expr_0 = import7.UNINITIALIZED;
    this._expr_1 = import7.UNINITIALIZED;
    this._expr_2 = import7.UNINITIALIZED;
    this._expr_3 = import7.UNINITIALIZED;
    this._expr_4 = import7.UNINITIALIZED;
  }
  ngOnDetach(view:import8.AppView<any>,componentView:import8.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
  }
  check_node(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.node = currValue;
      this._expr_0 = currValue;
    }
  }
  check_parentNode(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.parentNode = currValue;
      this._expr_1 = currValue;
    }
  }
  check_root(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_2,currValue))) {
      this._changed = true;
      this.context.root = currValue;
      this._expr_2 = currValue;
    }
  }
  check_firstChild(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_3,currValue))) {
      this._changed = true;
      this.context.firstChild = currValue;
      this._expr_3 = currValue;
    }
  }
  check_lastChild(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_4,currValue))) {
      this._changed = true;
      this.context.lastChild = currValue;
      this._expr_4 = currValue;
    }
  }
  ngDoCheck(view:import8.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    if (!throwOnChange) { if ((view.numberOfChecks === 0)) { this.context.ngOnInit(); } }
    return changed;
  }
  checkHost(view:import8.AppView<any>,componentView:import8.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import8.AppView<any>,_eventHandler:any):void {
    this._eventHandler = _eventHandler;
  }
}
export class Wrapper_TreeNodeTemplateLoader {
  /*private*/ _eventHandler:Function;
  context:import1.TreeNodeTemplateLoader;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  constructor(p0:any) {
    this._changed = false;
    this.context = new import1.TreeNodeTemplateLoader(p0);
    this._expr_0 = import7.UNINITIALIZED;
    this._expr_1 = import7.UNINITIALIZED;
  }
  ngOnDetach(view:import8.AppView<any>,componentView:import8.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
  }
  check_node(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.node = currValue;
      this._expr_0 = currValue;
    }
  }
  check_template(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.template = currValue;
      this._expr_1 = currValue;
    }
  }
  ngDoCheck(view:import8.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    if (!throwOnChange) { if ((view.numberOfChecks === 0)) { this.context.ngOnInit(); } }
    return changed;
  }
  checkHost(view:import8.AppView<any>,componentView:import8.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import8.AppView<any>,_eventHandler:any):void {
    this._eventHandler = _eventHandler;
  }
}
var renderType_Tree_Host:import10.RenderComponentType = import9.createRenderComponentType('',0,import11.ViewEncapsulation.None,([] as any[]),{});
class View_Tree_Host0 extends import8.AppView<any> {
  _el_0:any;
  compView_0:import8.AppView<import1.Tree>;
  _Tree_0_3:Wrapper_Tree;
  _query_PrimeTemplate_0_0:import12.QueryList<any>;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_Tree_Host0,renderType_Tree_Host,import13.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.selectOrCreateRenderHostElement(this.renderer,'p-tree',import9.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this.compView_0 = new View_Tree0(this.viewUtils,this,0,this._el_0);
    this._Tree_0_3 = new Wrapper_Tree();
    this._query_PrimeTemplate_0_0 = new import12.QueryList<any>();
    this.compView_0.create(this._Tree_0_3.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import15.ComponentRef_<any>(0,this,this._el_0,this._Tree_0_3.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import1.Tree) && (0 === requestNodeIndex))) { return this._Tree_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._Tree_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    if (!throwOnChange) {
      if (this._query_PrimeTemplate_0_0.dirty) {
        this._query_PrimeTemplate_0_0.reset(([] as any[]));
        this._Tree_0_3.context.templates = this._query_PrimeTemplate_0_0;
        this._query_PrimeTemplate_0_0.notifyOnChanges();
      }
      if ((this.numberOfChecks === 0)) { this._Tree_0_3.context.ngAfterContentInit(); }
    }
    this.compView_0.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
    this._Tree_0_3.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
export const TreeNgFactory:import15.ComponentFactory<import1.Tree> = new import15.ComponentFactory<import1.Tree>('p-tree',View_Tree_Host0,import1.Tree);
const styles_Tree:any[] = ([] as any[]);
class View_Tree2 extends import8.AppView<any> {
  _el_0:any;
  compView_0:import8.AppView<import1.UITreeNode>;
  _UITreeNode_0_3:Wrapper_UITreeNode;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_Tree2,renderType_Tree,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'p-treeNode',import9.EMPTY_INLINE_ARRAY,(null as any));
    this.compView_0 = new View_UITreeNode0(this.viewUtils,this,0,this._el_0);
    this._UITreeNode_0_3 = new Wrapper_UITreeNode(this.parentView.parentView.injectorGet(import1.Tree,this.parentView.parentIndex));
    this.compView_0.create(this._UITreeNode_0_3.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import1.UITreeNode) && (0 === requestNodeIndex))) { return this._UITreeNode_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this.context.$implicit;
    this._UITreeNode_0_3.check_node(currVal_0_0_0,throwOnChange,false);
    this._UITreeNode_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_Tree1 extends import8.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import17.Wrapper_NgClass;
  _NgStyle_0_4:import18.Wrapper_NgStyle;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _anchor_4:any;
  /*private*/ _vc_4:import16.ViewContainer;
  _TemplateRef_4_5:any;
  _NgFor_4_6:import19.Wrapper_NgFor;
  _text_5:any;
  _text_6:any;
  _map_12:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_Tree1,renderType_Tree,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_12 = import9.pureProxy2((p0:any,p1:any):{[key: string]:any} => {
      return {
        'ui-tree ui-widget ui-widget-content ui-corner-all': p0,
        'ui-tree-selectable': p1
      }
      ;
    });
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'div',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_0_3 = new import17.Wrapper_NgClass(this.parentView.injectorGet(import20.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import21.KeyValueDiffers,this.parentIndex),new import22.ElementRef(this._el_0),this.renderer);
    this._NgStyle_0_4 = new import18.Wrapper_NgStyle(this.parentView.injectorGet(import21.KeyValueDiffers,this.parentIndex),new import22.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n            ',(null as any));
    this._el_2 = import9.createRenderElement(this.renderer,this._el_0,'ul',new import9.InlineArray2(2,'class','ui-tree-container'),(null as any));
    this._text_3 = this.renderer.createText(this._el_2,'\n                ',(null as any));
    this._anchor_4 = this.renderer.createTemplateAnchor(this._el_2,(null as any));
    this._vc_4 = new import16.ViewContainer(4,2,this,this._anchor_4);
    this._TemplateRef_4_5 = new import23.TemplateRef_(this,4,this._anchor_4);
    this._NgFor_4_6 = new import19.Wrapper_NgFor(this._vc_4.vcRef,this._TemplateRef_4_5,this.parentView.injectorGet(import20.IterableDiffers,this.parentIndex),this.parentView.ref);
    this._text_5 = this.renderer.createText(this._el_2,'\n            ',(null as any));
    this._text_6 = this.renderer.createText(this._el_0,'\n        ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._anchor_4,
      this._text_5,
      this._text_6
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import23.TemplateRef) && (4 === requestNodeIndex))) { return this._TemplateRef_4_5; }
    if (((token === import24.NgFor) && (4 === requestNodeIndex))) { return this._NgFor_4_6.context; }
    if (((token === import25.NgClass) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 6)))) { return this._NgClass_0_3.context; }
    if (((token === import26.NgStyle) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 6)))) { return this._NgStyle_0_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this.parentView.context.styleClass;
    this._NgClass_0_3.check_klass(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = this._map_12(true,this.parentView.context.selectionMode);
    this._NgClass_0_3.check_ngClass(currVal_0_0_1,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_0_1_0:any = this.parentView.context.style;
    this._NgStyle_0_4.check_ngStyle(currVal_0_1_0,throwOnChange,false);
    this._NgStyle_0_4.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_4_0_0:any = this.parentView.context.value;
    this._NgFor_4_6.check_ngForOf(currVal_4_0_0,throwOnChange,false);
    this._NgFor_4_6.ngDoCheck(this,this._anchor_4,throwOnChange);
    this._vc_4.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_4.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import8.AppView<any> {
    if ((nodeIndex == 4)) { return new View_Tree2(this.viewUtils,this,4,this._anchor_4,this._vc_4); }
    return (null as any);
  }
}
class View_Tree4 extends import8.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  compView_2:import8.AppView<import1.UITreeNode>;
  _UITreeNode_2_3:Wrapper_UITreeNode;
  _text_3:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_Tree4,renderType_Tree,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'table',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    this._el_2 = import9.createRenderElement(this.renderer,this._el_0,'p-treeNode',import9.EMPTY_INLINE_ARRAY,(null as any));
    this.compView_2 = new View_UITreeNode0(this.viewUtils,this,2,this._el_2);
    this._UITreeNode_2_3 = new Wrapper_UITreeNode(this.parentView.parentView.injectorGet(import1.Tree,this.parentView.parentIndex));
    this.compView_2.create(this._UITreeNode_2_3.context);
    this._text_3 = this.renderer.createText(this._el_0,'\n            ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import1.UITreeNode) && (2 === requestNodeIndex))) { return this._UITreeNode_2_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2_0_0:any = this.parentView.parentView.context.value[0];
    this._UITreeNode_2_3.check_node(currVal_2_0_0,throwOnChange,false);
    const currVal_2_0_1:any = true;
    this._UITreeNode_2_3.check_root(currVal_2_0_1,throwOnChange,false);
    this._UITreeNode_2_3.ngDoCheck(this,this._el_2,throwOnChange);
    this.compView_2.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_2.destroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_Tree3 extends import8.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import17.Wrapper_NgClass;
  _NgStyle_0_4:import18.Wrapper_NgStyle;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import16.ViewContainer;
  _TemplateRef_2_5:any;
  _NgIf_2_6:import27.Wrapper_NgIf;
  _text_3:any;
  _map_9:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_Tree3,renderType_Tree,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_9 = import9.pureProxy2((p0:any,p1:any):{[key: string]:any} => {
      return {
        'ui-tree ui-tree-horizontal ui-widget ui-widget-content ui-corner-all': p0,
        'ui-tree-selectable': p1
      }
      ;
    });
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'div',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_0_3 = new import17.Wrapper_NgClass(this.parentView.injectorGet(import20.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import21.KeyValueDiffers,this.parentIndex),new import22.ElementRef(this._el_0),this.renderer);
    this._NgStyle_0_4 = new import18.Wrapper_NgStyle(this.parentView.injectorGet(import21.KeyValueDiffers,this.parentIndex),new import22.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n            ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import16.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import23.TemplateRef_(this,2,this._anchor_2);
    this._NgIf_2_6 = new import27.Wrapper_NgIf(this._vc_2.vcRef,this._TemplateRef_2_5);
    this._text_3 = this.renderer.createText(this._el_0,'\n        ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import23.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import28.NgIf) && (2 === requestNodeIndex))) { return this._NgIf_2_6.context; }
    if (((token === import25.NgClass) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 3)))) { return this._NgClass_0_3.context; }
    if (((token === import26.NgStyle) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 3)))) { return this._NgStyle_0_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this.parentView.context.styleClass;
    this._NgClass_0_3.check_klass(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = this._map_9(true,this.parentView.context.selectionMode);
    this._NgClass_0_3.check_ngClass(currVal_0_0_1,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_0_1_0:any = this.parentView.context.style;
    this._NgStyle_0_4.check_ngStyle(currVal_0_1_0,throwOnChange,false);
    this._NgStyle_0_4.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_2_0_0:any = (this.parentView.context.value && this.parentView.context.value[0]);
    this._NgIf_2_6.check_ngIf(currVal_2_0_0,throwOnChange,false);
    this._NgIf_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import8.AppView<any> {
    if ((nodeIndex == 2)) { return new View_Tree4(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    return (null as any);
  }
}
var renderType_Tree:import10.RenderComponentType = import9.createRenderComponentType('',0,import11.ViewEncapsulation.None,styles_Tree,{});
export class View_Tree0 extends import8.AppView<import1.Tree> {
  _text_0:any;
  _anchor_1:any;
  /*private*/ _vc_1:import16.ViewContainer;
  _TemplateRef_1_5:any;
  _NgIf_1_6:import27.Wrapper_NgIf;
  _text_2:any;
  _anchor_3:any;
  /*private*/ _vc_3:import16.ViewContainer;
  _TemplateRef_3_5:any;
  _NgIf_3_6:import27.Wrapper_NgIf;
  _text_4:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_Tree0,renderType_Tree,import13.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n        ',(null as any));
    this._anchor_1 = this.renderer.createTemplateAnchor(parentRenderNode,(null as any));
    this._vc_1 = new import16.ViewContainer(1,(null as any),this,this._anchor_1);
    this._TemplateRef_1_5 = new import23.TemplateRef_(this,1,this._anchor_1);
    this._NgIf_1_6 = new import27.Wrapper_NgIf(this._vc_1.vcRef,this._TemplateRef_1_5);
    this._text_2 = this.renderer.createText(parentRenderNode,'\n        ',(null as any));
    this._anchor_3 = this.renderer.createTemplateAnchor(parentRenderNode,(null as any));
    this._vc_3 = new import16.ViewContainer(3,(null as any),this,this._anchor_3);
    this._TemplateRef_3_5 = new import23.TemplateRef_(this,3,this._anchor_3);
    this._NgIf_3_6 = new import27.Wrapper_NgIf(this._vc_3.vcRef,this._TemplateRef_3_5);
    this._text_4 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._anchor_1,
      this._text_2,
      this._anchor_3,
      this._text_4
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import23.TemplateRef) && (1 === requestNodeIndex))) { return this._TemplateRef_1_5; }
    if (((token === import28.NgIf) && (1 === requestNodeIndex))) { return this._NgIf_1_6.context; }
    if (((token === import23.TemplateRef) && (3 === requestNodeIndex))) { return this._TemplateRef_3_5; }
    if (((token === import28.NgIf) && (3 === requestNodeIndex))) { return this._NgIf_3_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1_0_0:boolean = !this.context.horizontal;
    this._NgIf_1_6.check_ngIf(currVal_1_0_0,throwOnChange,false);
    this._NgIf_1_6.ngDoCheck(this,this._anchor_1,throwOnChange);
    const currVal_3_0_0:any = this.context.horizontal;
    this._NgIf_3_6.check_ngIf(currVal_3_0_0,throwOnChange,false);
    this._NgIf_3_6.ngDoCheck(this,this._anchor_3,throwOnChange);
    this._vc_1.detectChangesInNestedViews(throwOnChange);
    this._vc_3.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_1.destroyNestedViews();
    this._vc_3.destroyNestedViews();
  }
  createEmbeddedViewInternal(nodeIndex:number):import8.AppView<any> {
    if ((nodeIndex == 1)) { return new View_Tree1(this.viewUtils,this,1,this._anchor_1,this._vc_1); }
    if ((nodeIndex == 3)) { return new View_Tree3(this.viewUtils,this,3,this._anchor_3,this._vc_3); }
    return (null as any);
  }
}
var renderType_UITreeNode_Host:import10.RenderComponentType = import9.createRenderComponentType('',0,import11.ViewEncapsulation.None,([] as any[]),{});
class View_UITreeNode_Host0 extends import8.AppView<any> {
  _el_0:any;
  compView_0:import8.AppView<import1.UITreeNode>;
  _UITreeNode_0_3:Wrapper_UITreeNode;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_UITreeNode_Host0,renderType_UITreeNode_Host,import13.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.selectOrCreateRenderHostElement(this.renderer,'p-treeNode',import9.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this.compView_0 = new View_UITreeNode0(this.viewUtils,this,0,this._el_0);
    this._UITreeNode_0_3 = new Wrapper_UITreeNode(this.injectorGet(import1.Tree,this.parentIndex));
    this.compView_0.create(this._UITreeNode_0_3.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import15.ComponentRef_<any>(0,this,this._el_0,this._UITreeNode_0_3.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import1.UITreeNode) && (0 === requestNodeIndex))) { return this._UITreeNode_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._UITreeNode_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
export const UITreeNodeNgFactory:import15.ComponentFactory<import1.UITreeNode> = new import15.ComponentFactory<import1.UITreeNode>('p-treeNode',View_UITreeNode_Host0,import1.UITreeNode);
const styles_UITreeNode:any[] = ([] as any[]);
class View_UITreeNode3 extends import8.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import17.Wrapper_NgClass;
  _map_2:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_UITreeNode3,renderType_UITreeNode,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_2 = import9.pureProxy2((p0:any,p1:any):{[key: string]:any} => {
      return {
        'fa-caret-right': p0,
        'fa-caret-down': p1
      }
      ;
    });
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'span',new import9.InlineArray2(2,'class','ui-tree-toggler fa fa-fw'),(null as any));
    this._NgClass_0_3 = new import17.Wrapper_NgClass(this.parentView.parentView.injectorGet(import20.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import21.KeyValueDiffers,this.parentView.parentIndex),new import22.ElementRef(this._el_0),this.renderer);
    var disposable_0:Function = import9.subscribeToRenderElement(this,this._el_0,new import9.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import25.NgClass) && (0 === requestNodeIndex))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = 'ui-tree-toggler fa fa-fw';
    this._NgClass_0_3.check_klass(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = this._map_2(!this.parentView.parentView.parentView.context.node.expanded,this.parentView.parentView.parentView.context.node.expanded);
    this._NgClass_0_3.check_ngClass(currVal_0_0_1,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.parentView.parentView.context.toggle($event)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
class View_UITreeNode4 extends import8.AppView<any> {
  _el_0:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_UITreeNode4,renderType_UITreeNode,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'span',new import9.InlineArray2(2,'class','ui-treenode-leaf-icon'),(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return (null as any);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_UITreeNode5 extends import8.AppView<any> {
  _el_0:any;
  _el_1:any;
  _text_2:any;
  _el_3:any;
  _NgClass_3_3:import17.Wrapper_NgClass;
  _map_5:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_UITreeNode5,renderType_UITreeNode,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_5 = import9.pureProxy2((p0:any,p1:any):{[key: string]:any} => {
      return {
        'fa-check': p0,
        'fa-minus': p1
      }
      ;
    });
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'div',new import9.InlineArray2(2,'class','ui-chkbox'),(null as any));
    this._el_1 = import9.createRenderElement(this.renderer,this._el_0,'div',new import9.InlineArray2(2,'class','ui-chkbox-box ui-widget ui-corner-all ui-state-default'),(null as any));
    this._text_2 = this.renderer.createText(this._el_1,'\n                        ',(null as any));
    this._el_3 = import9.createRenderElement(this.renderer,this._el_1,'span',new import9.InlineArray2(2,'class','ui-chkbox-icon ui-c fa'),(null as any));
    this._NgClass_3_3 = new import17.Wrapper_NgClass(this.parentView.parentView.injectorGet(import20.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import21.KeyValueDiffers,this.parentView.parentIndex),new import22.ElementRef(this._el_3),this.renderer);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._el_1,
      this._text_2,
      this._el_3
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import25.NgClass) && (3 === requestNodeIndex))) { return this._NgClass_3_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_3_0_0:any = 'ui-chkbox-icon ui-c fa';
    this._NgClass_3_3.check_klass(currVal_3_0_0,throwOnChange,false);
    const currVal_3_0_1:any = this._map_5(this.parentView.parentView.parentView.context.isSelected(),this.parentView.parentView.parentView.context.node.partialSelected);
    this._NgClass_3_3.check_ngClass(currVal_3_0_1,throwOnChange,false);
    this._NgClass_3_3.ngDoCheck(this,this._el_3,throwOnChange);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_UITreeNode6 extends import8.AppView<any> {
  _el_0:any;
  /*private*/ _expr_1:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_UITreeNode6,renderType_UITreeNode,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_1 = import7.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'span',import9.EMPTY_INLINE_ARRAY,(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1:any = this.parentView.parentView.parentView.context.getIcon();
    if (import9.checkBinding(throwOnChange,this._expr_1,currVal_1)) {
      this.renderer.setElementProperty(this._el_0,'className',currVal_1);
      this._expr_1 = currVal_1;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_UITreeNode7 extends import8.AppView<any> {
  _el_0:any;
  _text_1:any;
  /*private*/ _expr_2:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_UITreeNode7,renderType_UITreeNode,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_2 = import7.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'span',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1
    ]
    ),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2:any = import9.inlineInterpolate(1,'',this.parentView.parentView.parentView.context.node.label,'');
    if (import9.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setText(this._text_1,currVal_2);
      this._expr_2 = currVal_2;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_UITreeNode8 extends import8.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  /*private*/ _vc_2:import16.ViewContainer;
  compView_2:import8.AppView<import1.TreeNodeTemplateLoader>;
  _TreeNodeTemplateLoader_2_5:Wrapper_TreeNodeTemplateLoader;
  _text_3:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_UITreeNode8,renderType_UITreeNode,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'span',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                                ',(null as any));
    this._el_2 = import9.createRenderElement(this.renderer,this._el_0,'p-treeNodeTemplateLoader',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._vc_2 = new import16.ViewContainer(2,0,this,this._el_2);
    this.compView_2 = new View_TreeNodeTemplateLoader0(this.viewUtils,this,2,this._el_2);
    this._TreeNodeTemplateLoader_2_5 = new Wrapper_TreeNodeTemplateLoader(this._vc_2.vcRef);
    this.compView_2.create(this._TreeNodeTemplateLoader_2_5.context);
    this._text_3 = this.renderer.createText(this._el_0,'\n                            ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import1.TreeNodeTemplateLoader) && (2 === requestNodeIndex))) { return this._TreeNodeTemplateLoader_2_5.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2_0_0:any = this.parentView.parentView.parentView.context.node;
    this._TreeNodeTemplateLoader_2_5.check_node(currVal_2_0_0,throwOnChange,false);
    const currVal_2_0_1:any = this.parentView.parentView.parentView.context.tree.getTemplateForNode(this.parentView.parentView.parentView.context.node);
    this._TreeNodeTemplateLoader_2_5.check_template(currVal_2_0_1,throwOnChange,false);
    this._TreeNodeTemplateLoader_2_5.ngDoCheck(this,this._el_2,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
    this.compView_2.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
    this.compView_2.destroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_UITreeNode10 extends import8.AppView<any> {
  _el_0:any;
  compView_0:import8.AppView<import1.UITreeNode>;
  _UITreeNode_0_3:Wrapper_UITreeNode;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_UITreeNode10,renderType_UITreeNode,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'p-treeNode',import9.EMPTY_INLINE_ARRAY,(null as any));
    this.compView_0 = new View_UITreeNode0(this.viewUtils,this,0,this._el_0);
    this._UITreeNode_0_3 = new Wrapper_UITreeNode(this.parentView.parentView.parentView.injectorGet(import1.Tree,this.parentView.parentView.parentIndex));
    this.compView_0.create(this._UITreeNode_0_3.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import1.UITreeNode) && (0 === requestNodeIndex))) { return this._UITreeNode_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this.context.$implicit;
    this._UITreeNode_0_3.check_node(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = this.parentView.parentView.parentView.parentView.context.node;
    this._UITreeNode_0_3.check_parentNode(currVal_0_0_1,throwOnChange,false);
    this._UITreeNode_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_UITreeNode9 extends import8.AppView<any> {
  _el_0:any;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import16.ViewContainer;
  _TemplateRef_2_5:any;
  _NgFor_2_6:import19.Wrapper_NgFor;
  _text_3:any;
  /*private*/ _expr_7:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_UITreeNode9,renderType_UITreeNode,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_7 = import7.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'ul',new import9.InlineArray4(4,'class','ui-treenode-children','style','display: none;'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import16.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import23.TemplateRef_(this,2,this._anchor_2);
    this._NgFor_2_6 = new import19.Wrapper_NgFor(this._vc_2.vcRef,this._TemplateRef_2_5,this.parentView.parentView.injectorGet(import20.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.parentView.ref);
    this._text_3 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import23.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import24.NgFor) && (2 === requestNodeIndex))) { return this._NgFor_2_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2_0_0:any = this.parentView.parentView.parentView.context.node.children;
    this._NgFor_2_6.check_ngForOf(currVal_2_0_0,throwOnChange,false);
    this._NgFor_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
    const currVal_7:any = (this.parentView.parentView.parentView.context.node.expanded? 'block': 'none');
    if (import9.checkBinding(throwOnChange,this._expr_7,currVal_7)) {
      this.renderer.setElementStyle(this._el_0,'display',((this.viewUtils.sanitizer.sanitize(import29.SecurityContext.STYLE,currVal_7) == null)? (null as any): this.viewUtils.sanitizer.sanitize(import29.SecurityContext.STYLE,currVal_7).toString()));
      this._expr_7 = currVal_7;
    }
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import8.AppView<any> {
    if ((nodeIndex == 2)) { return new View_UITreeNode10(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    return (null as any);
  }
}
class View_UITreeNode2 extends import8.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _anchor_4:any;
  /*private*/ _vc_4:import16.ViewContainer;
  _TemplateRef_4_5:any;
  _NgIf_4_6:import27.Wrapper_NgIf;
  _anchor_5:any;
  /*private*/ _vc_5:import16.ViewContainer;
  _TemplateRef_5_5:any;
  _NgIf_5_6:import27.Wrapper_NgIf;
  _anchor_6:any;
  /*private*/ _vc_6:import16.ViewContainer;
  _TemplateRef_6_5:any;
  _NgIf_6_6:import27.Wrapper_NgIf;
  _anchor_7:any;
  /*private*/ _vc_7:import16.ViewContainer;
  _TemplateRef_7_5:any;
  _NgIf_7_6:import27.Wrapper_NgIf;
  _el_8:any;
  _NgClass_8_3:import17.Wrapper_NgClass;
  _text_9:any;
  _anchor_10:any;
  /*private*/ _vc_10:import16.ViewContainer;
  _TemplateRef_10_5:any;
  _NgIf_10_6:import27.Wrapper_NgIf;
  _text_11:any;
  _anchor_12:any;
  /*private*/ _vc_12:import16.ViewContainer;
  _TemplateRef_12_5:any;
  _NgIf_12_6:import27.Wrapper_NgIf;
  _text_13:any;
  _text_14:any;
  _text_15:any;
  _anchor_16:any;
  /*private*/ _vc_16:import16.ViewContainer;
  _TemplateRef_16_5:any;
  _NgIf_16_6:import27.Wrapper_NgIf;
  _text_17:any;
  _map_40:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_UITreeNode2,renderType_UITreeNode,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_40 = import9.pureProxy2((p0:any,p1:any):{[key: string]:any} => {
      return {
        'ui-state-hover': p0,
        'ui-state-highlight': p1
      }
      ;
    });
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'li',new import9.InlineArray2(2,'class','ui-treenode'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    this._el_2 = import9.createRenderElement(this.renderer,this._el_0,'div',new import9.InlineArray2(2,'class','ui-treenode-content'),(null as any));
    this._text_3 = this.renderer.createText(this._el_2,'\n                    ',(null as any));
    this._anchor_4 = this.renderer.createTemplateAnchor(this._el_2,(null as any));
    this._vc_4 = new import16.ViewContainer(4,2,this,this._anchor_4);
    this._TemplateRef_4_5 = new import23.TemplateRef_(this,4,this._anchor_4);
    this._NgIf_4_6 = new import27.Wrapper_NgIf(this._vc_4.vcRef,this._TemplateRef_4_5);
    this._anchor_5 = this.renderer.createTemplateAnchor(this._el_2,(null as any));
    this._vc_5 = new import16.ViewContainer(5,2,this,this._anchor_5);
    this._TemplateRef_5_5 = new import23.TemplateRef_(this,5,this._anchor_5);
    this._NgIf_5_6 = new import27.Wrapper_NgIf(this._vc_5.vcRef,this._TemplateRef_5_5);
    this._anchor_6 = this.renderer.createTemplateAnchor(this._el_2,(null as any));
    this._vc_6 = new import16.ViewContainer(6,2,this,this._anchor_6);
    this._TemplateRef_6_5 = new import23.TemplateRef_(this,6,this._anchor_6);
    this._NgIf_6_6 = new import27.Wrapper_NgIf(this._vc_6.vcRef,this._TemplateRef_6_5);
    this._anchor_7 = this.renderer.createTemplateAnchor(this._el_2,(null as any));
    this._vc_7 = new import16.ViewContainer(7,2,this,this._anchor_7);
    this._TemplateRef_7_5 = new import23.TemplateRef_(this,7,this._anchor_7);
    this._NgIf_7_6 = new import27.Wrapper_NgIf(this._vc_7.vcRef,this._TemplateRef_7_5);
    this._el_8 = import9.createRenderElement(this.renderer,this._el_2,'span',new import9.InlineArray2(2,'class','ui-treenode-label ui-corner-all'),(null as any));
    this._NgClass_8_3 = new import17.Wrapper_NgClass(this.parentView.injectorGet(import20.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import21.KeyValueDiffers,this.parentIndex),new import22.ElementRef(this._el_8),this.renderer);
    this._text_9 = this.renderer.createText(this._el_8,'\n                            ',(null as any));
    this._anchor_10 = this.renderer.createTemplateAnchor(this._el_8,(null as any));
    this._vc_10 = new import16.ViewContainer(10,8,this,this._anchor_10);
    this._TemplateRef_10_5 = new import23.TemplateRef_(this,10,this._anchor_10);
    this._NgIf_10_6 = new import27.Wrapper_NgIf(this._vc_10.vcRef,this._TemplateRef_10_5);
    this._text_11 = this.renderer.createText(this._el_8,'\n                            ',(null as any));
    this._anchor_12 = this.renderer.createTemplateAnchor(this._el_8,(null as any));
    this._vc_12 = new import16.ViewContainer(12,8,this,this._anchor_12);
    this._TemplateRef_12_5 = new import23.TemplateRef_(this,12,this._anchor_12);
    this._NgIf_12_6 = new import27.Wrapper_NgIf(this._vc_12.vcRef,this._TemplateRef_12_5);
    this._text_13 = this.renderer.createText(this._el_8,'\n                    ',(null as any));
    this._text_14 = this.renderer.createText(this._el_2,'\n                ',(null as any));
    this._text_15 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    this._anchor_16 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_16 = new import16.ViewContainer(16,0,this,this._anchor_16);
    this._TemplateRef_16_5 = new import23.TemplateRef_(this,16,this._anchor_16);
    this._NgIf_16_6 = new import27.Wrapper_NgIf(this._vc_16.vcRef,this._TemplateRef_16_5);
    this._text_17 = this.renderer.createText(this._el_0,'\n            ',(null as any));
    var disposable_0:Function = import9.subscribeToRenderElement(this,this._el_2,new import9.InlineArray8(8,'mouseenter',(null as any),'mouseleave',(null as any),'click',(null as any),'contextmenu',(null as any)),this.eventHandler(this.handleEvent_2));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._anchor_4,
      this._anchor_5,
      this._anchor_6,
      this._anchor_7,
      this._el_8,
      this._text_9,
      this._anchor_10,
      this._text_11,
      this._anchor_12,
      this._text_13,
      this._text_14,
      this._text_15,
      this._anchor_16,
      this._text_17
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import23.TemplateRef) && (4 === requestNodeIndex))) { return this._TemplateRef_4_5; }
    if (((token === import28.NgIf) && (4 === requestNodeIndex))) { return this._NgIf_4_6.context; }
    if (((token === import23.TemplateRef) && (5 === requestNodeIndex))) { return this._TemplateRef_5_5; }
    if (((token === import28.NgIf) && (5 === requestNodeIndex))) { return this._NgIf_5_6.context; }
    if (((token === import23.TemplateRef) && (6 === requestNodeIndex))) { return this._TemplateRef_6_5; }
    if (((token === import28.NgIf) && (6 === requestNodeIndex))) { return this._NgIf_6_6.context; }
    if (((token === import23.TemplateRef) && (7 === requestNodeIndex))) { return this._TemplateRef_7_5; }
    if (((token === import28.NgIf) && (7 === requestNodeIndex))) { return this._NgIf_7_6.context; }
    if (((token === import23.TemplateRef) && (10 === requestNodeIndex))) { return this._TemplateRef_10_5; }
    if (((token === import28.NgIf) && (10 === requestNodeIndex))) { return this._NgIf_10_6.context; }
    if (((token === import23.TemplateRef) && (12 === requestNodeIndex))) { return this._TemplateRef_12_5; }
    if (((token === import28.NgIf) && (12 === requestNodeIndex))) { return this._NgIf_12_6.context; }
    if (((token === import25.NgClass) && ((8 <= requestNodeIndex) && (requestNodeIndex <= 13)))) { return this._NgClass_8_3.context; }
    if (((token === import23.TemplateRef) && (16 === requestNodeIndex))) { return this._TemplateRef_16_5; }
    if (((token === import28.NgIf) && (16 === requestNodeIndex))) { return this._NgIf_16_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_4_0_0:boolean = !this.parentView.parentView.context.isLeaf();
    this._NgIf_4_6.check_ngIf(currVal_4_0_0,throwOnChange,false);
    this._NgIf_4_6.ngDoCheck(this,this._anchor_4,throwOnChange);
    const currVal_5_0_0:any = this.parentView.parentView.context.isLeaf();
    this._NgIf_5_6.check_ngIf(currVal_5_0_0,throwOnChange,false);
    this._NgIf_5_6.ngDoCheck(this,this._anchor_5,throwOnChange);
    const currVal_6_0_0:any = (this.parentView.parentView.context.tree.selectionMode == 'checkbox');
    this._NgIf_6_6.check_ngIf(currVal_6_0_0,throwOnChange,false);
    this._NgIf_6_6.ngDoCheck(this,this._anchor_6,throwOnChange);
    const currVal_7_0_0:any = ((this.parentView.parentView.context.node.icon || this.parentView.parentView.context.node.expandedIcon) || this.parentView.parentView.context.node.collapsedIcon);
    this._NgIf_7_6.check_ngIf(currVal_7_0_0,throwOnChange,false);
    this._NgIf_7_6.ngDoCheck(this,this._anchor_7,throwOnChange);
    const currVal_8_0_0:any = 'ui-treenode-label ui-corner-all';
    this._NgClass_8_3.check_klass(currVal_8_0_0,throwOnChange,false);
    const currVal_8_0_1:any = this._map_40((this.parentView.parentView.context.hover && this.parentView.parentView.context.tree.selectionMode),this.parentView.parentView.context.isSelected());
    this._NgClass_8_3.check_ngClass(currVal_8_0_1,throwOnChange,false);
    this._NgClass_8_3.ngDoCheck(this,this._el_8,throwOnChange);
    const currVal_10_0_0:boolean = !this.parentView.parentView.context.tree.getTemplateForNode(this.parentView.parentView.context.node);
    this._NgIf_10_6.check_ngIf(currVal_10_0_0,throwOnChange,false);
    this._NgIf_10_6.ngDoCheck(this,this._anchor_10,throwOnChange);
    const currVal_12_0_0:any = this.parentView.parentView.context.tree.getTemplateForNode(this.parentView.parentView.context.node);
    this._NgIf_12_6.check_ngIf(currVal_12_0_0,throwOnChange,false);
    this._NgIf_12_6.ngDoCheck(this,this._anchor_12,throwOnChange);
    const currVal_16_0_0:any = (this.parentView.parentView.context.node.children && this.parentView.parentView.context.node.expanded);
    this._NgIf_16_6.check_ngIf(currVal_16_0_0,throwOnChange,false);
    this._NgIf_16_6.ngDoCheck(this,this._anchor_16,throwOnChange);
    this._vc_4.detectChangesInNestedViews(throwOnChange);
    this._vc_5.detectChangesInNestedViews(throwOnChange);
    this._vc_6.detectChangesInNestedViews(throwOnChange);
    this._vc_7.detectChangesInNestedViews(throwOnChange);
    this._vc_10.detectChangesInNestedViews(throwOnChange);
    this._vc_12.detectChangesInNestedViews(throwOnChange);
    this._vc_16.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_4.destroyNestedViews();
    this._vc_5.destroyNestedViews();
    this._vc_6.destroyNestedViews();
    this._vc_7.destroyNestedViews();
    this._vc_10.destroyNestedViews();
    this._vc_12.destroyNestedViews();
    this._vc_16.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import8.AppView<any> {
    if ((nodeIndex == 4)) { return new View_UITreeNode3(this.viewUtils,this,4,this._anchor_4,this._vc_4); }
    if ((nodeIndex == 5)) { return new View_UITreeNode4(this.viewUtils,this,5,this._anchor_5,this._vc_5); }
    if ((nodeIndex == 6)) { return new View_UITreeNode5(this.viewUtils,this,6,this._anchor_6,this._vc_6); }
    if ((nodeIndex == 7)) { return new View_UITreeNode6(this.viewUtils,this,7,this._anchor_7,this._vc_7); }
    if ((nodeIndex == 10)) { return new View_UITreeNode7(this.viewUtils,this,10,this._anchor_10,this._vc_10); }
    if ((nodeIndex == 12)) { return new View_UITreeNode8(this.viewUtils,this,12,this._anchor_12,this._vc_12); }
    if ((nodeIndex == 16)) { return new View_UITreeNode9(this.viewUtils,this,16,this._anchor_16,this._vc_16); }
    return (null as any);
  }
  handleEvent_2(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'mouseenter')) {
      const pd_sub_0:any = ((<any>(this.parentView.parentView.context.hover = true)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'mouseleave')) {
      const pd_sub_1:any = ((<any>(this.parentView.parentView.context.hover = false)) !== false);
      result = (pd_sub_1 && result);
    }
    if ((eventName == 'click')) {
      const pd_sub_2:any = ((<any>this.parentView.parentView.context.onNodeClick($event)) !== false);
      result = (pd_sub_2 && result);
    }
    if ((eventName == 'contextmenu')) {
      const pd_sub_3:any = ((<any>this.parentView.parentView.context.onNodeRightClick($event)) !== false);
      result = (pd_sub_3 && result);
    }
    return result;
  }
}
class View_UITreeNode12 extends import8.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _el_4:any;
  _text_5:any;
  _el_6:any;
  _text_7:any;
  _el_8:any;
  _NgClass_8_3:import17.Wrapper_NgClass;
  _text_9:any;
  _text_10:any;
  _el_11:any;
  _text_12:any;
  _el_13:any;
  _NgClass_13_3:import17.Wrapper_NgClass;
  _text_14:any;
  _text_15:any;
  _text_16:any;
  _text_17:any;
  _map_20:any;
  _map_21:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_UITreeNode12,renderType_UITreeNode,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_20 = import9.pureProxy1((p0:any):{[key: string]:any} => {
      return {'ui-treenode-connector-line': p0};
    });
    this._map_21 = import9.pureProxy1((p0:any):{[key: string]:any} => {
      return {'ui-treenode-connector-line': p0};
    });
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'td',new import9.InlineArray2(2,'class','ui-treenode-connector'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                            ',(null as any));
    this._el_2 = import9.createRenderElement(this.renderer,this._el_0,'table',new import9.InlineArray2(2,'class','ui-treenode-connector-table'),(null as any));
    this._text_3 = this.renderer.createText(this._el_2,'\n                                ',(null as any));
    this._el_4 = import9.createRenderElement(this.renderer,this._el_2,'tbody',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._text_5 = this.renderer.createText(this._el_4,'\n                                    ',(null as any));
    this._el_6 = import9.createRenderElement(this.renderer,this._el_4,'tr',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._text_7 = this.renderer.createText(this._el_6,'\n                                        ',(null as any));
    this._el_8 = import9.createRenderElement(this.renderer,this._el_6,'td',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_8_3 = new import17.Wrapper_NgClass(this.parentView.parentView.injectorGet(import20.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import21.KeyValueDiffers,this.parentView.parentIndex),new import22.ElementRef(this._el_8),this.renderer);
    this._text_9 = this.renderer.createText(this._el_6,'\n                                    ',(null as any));
    this._text_10 = this.renderer.createText(this._el_4,'\n                                    ',(null as any));
    this._el_11 = import9.createRenderElement(this.renderer,this._el_4,'tr',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._text_12 = this.renderer.createText(this._el_11,'\n                                        ',(null as any));
    this._el_13 = import9.createRenderElement(this.renderer,this._el_11,'td',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_13_3 = new import17.Wrapper_NgClass(this.parentView.parentView.injectorGet(import20.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import21.KeyValueDiffers,this.parentView.parentIndex),new import22.ElementRef(this._el_13),this.renderer);
    this._text_14 = this.renderer.createText(this._el_11,'\n                                    ',(null as any));
    this._text_15 = this.renderer.createText(this._el_4,'\n                                ',(null as any));
    this._text_16 = this.renderer.createText(this._el_2,'\n                            ',(null as any));
    this._text_17 = this.renderer.createText(this._el_0,'\n                        ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._el_4,
      this._text_5,
      this._el_6,
      this._text_7,
      this._el_8,
      this._text_9,
      this._text_10,
      this._el_11,
      this._text_12,
      this._el_13,
      this._text_14,
      this._text_15,
      this._text_16,
      this._text_17
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import25.NgClass) && (8 === requestNodeIndex))) { return this._NgClass_8_3.context; }
    if (((token === import25.NgClass) && (13 === requestNodeIndex))) { return this._NgClass_13_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_8_0_0:any = this._map_20(!this.parentView.parentView.parentView.context.firstChild);
    this._NgClass_8_3.check_ngClass(currVal_8_0_0,throwOnChange,false);
    this._NgClass_8_3.ngDoCheck(this,this._el_8,throwOnChange);
    const currVal_13_0_0:any = this._map_21(!this.parentView.parentView.parentView.context.lastChild);
    this._NgClass_13_3.check_ngClass(currVal_13_0_0,throwOnChange,false);
    this._NgClass_13_3.ngDoCheck(this,this._el_13,throwOnChange);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_UITreeNode13 extends import8.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import17.Wrapper_NgClass;
  _map_2:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_UITreeNode13,renderType_UITreeNode,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_2 = import9.pureProxy2((p0:any,p1:any):{[key: string]:any} => {
      return {
        'fa-plus': p0,
        'fa-minus': p1
      }
      ;
    });
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'span',new import9.InlineArray2(2,'class','ui-tree-toggler fa fa-fw'),(null as any));
    this._NgClass_0_3 = new import17.Wrapper_NgClass(this.parentView.parentView.injectorGet(import20.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import21.KeyValueDiffers,this.parentView.parentIndex),new import22.ElementRef(this._el_0),this.renderer);
    var disposable_0:Function = import9.subscribeToRenderElement(this,this._el_0,new import9.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import25.NgClass) && (0 === requestNodeIndex))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = 'ui-tree-toggler fa fa-fw';
    this._NgClass_0_3.check_klass(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = this._map_2(!this.parentView.parentView.parentView.context.node.expanded,this.parentView.parentView.parentView.context.node.expanded);
    this._NgClass_0_3.check_ngClass(currVal_0_0_1,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.parentView.parentView.context.toggle($event)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
class View_UITreeNode14 extends import8.AppView<any> {
  _el_0:any;
  /*private*/ _expr_1:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_UITreeNode14,renderType_UITreeNode,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_1 = import7.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'span',import9.EMPTY_INLINE_ARRAY,(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1:any = this.parentView.parentView.parentView.context.getIcon();
    if (import9.checkBinding(throwOnChange,this._expr_1,currVal_1)) {
      this.renderer.setElementProperty(this._el_0,'className',currVal_1);
      this._expr_1 = currVal_1;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_UITreeNode15 extends import8.AppView<any> {
  _el_0:any;
  _text_1:any;
  /*private*/ _expr_2:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_UITreeNode15,renderType_UITreeNode,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_2 = import7.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'span',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1
    ]
    ),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2:any = import9.inlineInterpolate(1,'',this.parentView.parentView.parentView.context.node.label,'');
    if (import9.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setText(this._text_1,currVal_2);
      this._expr_2 = currVal_2;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_UITreeNode16 extends import8.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  /*private*/ _vc_2:import16.ViewContainer;
  compView_2:import8.AppView<import1.TreeNodeTemplateLoader>;
  _TreeNodeTemplateLoader_2_5:Wrapper_TreeNodeTemplateLoader;
  _text_3:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_UITreeNode16,renderType_UITreeNode,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'span',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                                            ',(null as any));
    this._el_2 = import9.createRenderElement(this.renderer,this._el_0,'p-treeNodeTemplateLoader',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._vc_2 = new import16.ViewContainer(2,0,this,this._el_2);
    this.compView_2 = new View_TreeNodeTemplateLoader0(this.viewUtils,this,2,this._el_2);
    this._TreeNodeTemplateLoader_2_5 = new Wrapper_TreeNodeTemplateLoader(this._vc_2.vcRef);
    this.compView_2.create(this._TreeNodeTemplateLoader_2_5.context);
    this._text_3 = this.renderer.createText(this._el_0,'\n                                        ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import1.TreeNodeTemplateLoader) && (2 === requestNodeIndex))) { return this._TreeNodeTemplateLoader_2_5.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2_0_0:any = this.parentView.parentView.parentView.context.node;
    this._TreeNodeTemplateLoader_2_5.check_node(currVal_2_0_0,throwOnChange,false);
    const currVal_2_0_1:any = this.parentView.parentView.parentView.context.tree.getTemplateForNode(this.parentView.parentView.parentView.context.node);
    this._TreeNodeTemplateLoader_2_5.check_template(currVal_2_0_1,throwOnChange,false);
    this._TreeNodeTemplateLoader_2_5.ngDoCheck(this,this._el_2,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
    this.compView_2.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
    this.compView_2.destroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_UITreeNode18 extends import8.AppView<any> {
  _el_0:any;
  compView_0:import8.AppView<import1.UITreeNode>;
  _UITreeNode_0_3:Wrapper_UITreeNode;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_UITreeNode18,renderType_UITreeNode,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'p-treeNode',import9.EMPTY_INLINE_ARRAY,(null as any));
    this.compView_0 = new View_UITreeNode0(this.viewUtils,this,0,this._el_0);
    this._UITreeNode_0_3 = new Wrapper_UITreeNode(this.parentView.parentView.parentView.injectorGet(import1.Tree,this.parentView.parentView.parentIndex));
    this.compView_0.create(this._UITreeNode_0_3.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import1.UITreeNode) && (0 === requestNodeIndex))) { return this._UITreeNode_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this.context.$implicit;
    this._UITreeNode_0_3.check_node(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = this.context.first;
    this._UITreeNode_0_3.check_firstChild(currVal_0_0_1,throwOnChange,false);
    const currVal_0_0_2:any = this.context.last;
    this._UITreeNode_0_3.check_lastChild(currVal_0_0_2,throwOnChange,false);
    this._UITreeNode_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_UITreeNode17 extends import8.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _anchor_4:any;
  /*private*/ _vc_4:import16.ViewContainer;
  _TemplateRef_4_5:any;
  _NgFor_4_6:import19.Wrapper_NgFor;
  _text_5:any;
  _text_6:any;
  /*private*/ _expr_10:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_UITreeNode17,renderType_UITreeNode,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_10 = import7.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'td',new import9.InlineArray2(2,'class','ui-treenode-children-container'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                            ',(null as any));
    this._el_2 = import9.createRenderElement(this.renderer,this._el_0,'div',new import9.InlineArray2(2,'class','ui-treenode-children'),(null as any));
    this._text_3 = this.renderer.createText(this._el_2,'\n                                ',(null as any));
    this._anchor_4 = this.renderer.createTemplateAnchor(this._el_2,(null as any));
    this._vc_4 = new import16.ViewContainer(4,2,this,this._anchor_4);
    this._TemplateRef_4_5 = new import23.TemplateRef_(this,4,this._anchor_4);
    this._NgFor_4_6 = new import19.Wrapper_NgFor(this._vc_4.vcRef,this._TemplateRef_4_5,this.parentView.parentView.injectorGet(import20.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.parentView.ref);
    this._text_5 = this.renderer.createText(this._el_2,'\n                            ',(null as any));
    this._text_6 = this.renderer.createText(this._el_0,'\n                        ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._anchor_4,
      this._text_5,
      this._text_6
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import23.TemplateRef) && (4 === requestNodeIndex))) { return this._TemplateRef_4_5; }
    if (((token === import24.NgFor) && (4 === requestNodeIndex))) { return this._NgFor_4_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_4_0_0:any = this.parentView.parentView.parentView.context.node.children;
    this._NgFor_4_6.check_ngForOf(currVal_4_0_0,throwOnChange,false);
    this._NgFor_4_6.ngDoCheck(this,this._anchor_4,throwOnChange);
    this._vc_4.detectChangesInNestedViews(throwOnChange);
    const currVal_10:any = (this.parentView.parentView.parentView.context.node.expanded? 'table-cell': 'none');
    if (import9.checkBinding(throwOnChange,this._expr_10,currVal_10)) {
      this.renderer.setElementStyle(this._el_0,'display',((this.viewUtils.sanitizer.sanitize(import29.SecurityContext.STYLE,currVal_10) == null)? (null as any): this.viewUtils.sanitizer.sanitize(import29.SecurityContext.STYLE,currVal_10).toString()));
      this._expr_10 = currVal_10;
    }
  }
  destroyInternal():void {
    this._vc_4.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import8.AppView<any> {
    if ((nodeIndex == 4)) { return new View_UITreeNode18(this.viewUtils,this,4,this._anchor_4,this._vc_4); }
    return (null as any);
  }
}
class View_UITreeNode11 extends import8.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _el_4:any;
  _text_5:any;
  _anchor_6:any;
  /*private*/ _vc_6:import16.ViewContainer;
  _TemplateRef_6_5:any;
  _NgIf_6_6:import27.Wrapper_NgIf;
  _text_7:any;
  _el_8:any;
  _NgClass_8_3:import17.Wrapper_NgClass;
  _text_9:any;
  _el_10:any;
  _NgClass_10_3:import17.Wrapper_NgClass;
  _text_11:any;
  _anchor_12:any;
  /*private*/ _vc_12:import16.ViewContainer;
  _TemplateRef_12_5:any;
  _NgIf_12_6:import27.Wrapper_NgIf;
  _anchor_13:any;
  /*private*/ _vc_13:import16.ViewContainer;
  _TemplateRef_13_5:any;
  _NgIf_13_6:import27.Wrapper_NgIf;
  _el_14:any;
  _text_15:any;
  _anchor_16:any;
  /*private*/ _vc_16:import16.ViewContainer;
  _TemplateRef_16_5:any;
  _NgIf_16_6:import27.Wrapper_NgIf;
  _text_17:any;
  _anchor_18:any;
  /*private*/ _vc_18:import16.ViewContainer;
  _TemplateRef_18_5:any;
  _NgIf_18_6:import27.Wrapper_NgIf;
  _text_19:any;
  _text_20:any;
  _text_21:any;
  _text_22:any;
  _anchor_23:any;
  /*private*/ _vc_23:import16.ViewContainer;
  _TemplateRef_23_5:any;
  _NgIf_23_6:import27.Wrapper_NgIf;
  _text_24:any;
  _text_25:any;
  _text_26:any;
  _map_47:any;
  _map_48:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_UITreeNode11,renderType_UITreeNode,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_47 = import9.pureProxy1((p0:any):{[key: string]:any} => {
      return {'ui-treenode-collapsed': p0};
    });
    this._map_48 = import9.pureProxy2((p0:any,p1:any):{[key: string]:any} => {
      return {
        'ui-state-hover': p0,
        'ui-state-highlight': p1
      }
      ;
    });
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'table',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    this._el_2 = import9.createRenderElement(this.renderer,this._el_0,'tbody',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._text_3 = this.renderer.createText(this._el_2,'\n                    ',(null as any));
    this._el_4 = import9.createRenderElement(this.renderer,this._el_2,'tr',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._text_5 = this.renderer.createText(this._el_4,'\n                        ',(null as any));
    this._anchor_6 = this.renderer.createTemplateAnchor(this._el_4,(null as any));
    this._vc_6 = new import16.ViewContainer(6,4,this,this._anchor_6);
    this._TemplateRef_6_5 = new import23.TemplateRef_(this,6,this._anchor_6);
    this._NgIf_6_6 = new import27.Wrapper_NgIf(this._vc_6.vcRef,this._TemplateRef_6_5);
    this._text_7 = this.renderer.createText(this._el_4,'\n                        ',(null as any));
    this._el_8 = import9.createRenderElement(this.renderer,this._el_4,'td',new import9.InlineArray2(2,'class','ui-treenode'),(null as any));
    this._NgClass_8_3 = new import17.Wrapper_NgClass(this.parentView.injectorGet(import20.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import21.KeyValueDiffers,this.parentIndex),new import22.ElementRef(this._el_8),this.renderer);
    this._text_9 = this.renderer.createText(this._el_8,'\n                            ',(null as any));
    this._el_10 = import9.createRenderElement(this.renderer,this._el_8,'div',new import9.InlineArray2(2,'class','ui-treenode-content ui-state-default ui-corner-all'),(null as any));
    this._NgClass_10_3 = new import17.Wrapper_NgClass(this.parentView.injectorGet(import20.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import21.KeyValueDiffers,this.parentIndex),new import22.ElementRef(this._el_10),this.renderer);
    this._text_11 = this.renderer.createText(this._el_10,'\n                                ',(null as any));
    this._anchor_12 = this.renderer.createTemplateAnchor(this._el_10,(null as any));
    this._vc_12 = new import16.ViewContainer(12,10,this,this._anchor_12);
    this._TemplateRef_12_5 = new import23.TemplateRef_(this,12,this._anchor_12);
    this._NgIf_12_6 = new import27.Wrapper_NgIf(this._vc_12.vcRef,this._TemplateRef_12_5);
    this._anchor_13 = this.renderer.createTemplateAnchor(this._el_10,(null as any));
    this._vc_13 = new import16.ViewContainer(13,10,this,this._anchor_13);
    this._TemplateRef_13_5 = new import23.TemplateRef_(this,13,this._anchor_13);
    this._NgIf_13_6 = new import27.Wrapper_NgIf(this._vc_13.vcRef,this._TemplateRef_13_5);
    this._el_14 = import9.createRenderElement(this.renderer,this._el_10,'span',new import9.InlineArray2(2,'class','ui-treenode-label ui-corner-all'),(null as any));
    this._text_15 = this.renderer.createText(this._el_14,'\n                                        ',(null as any));
    this._anchor_16 = this.renderer.createTemplateAnchor(this._el_14,(null as any));
    this._vc_16 = new import16.ViewContainer(16,14,this,this._anchor_16);
    this._TemplateRef_16_5 = new import23.TemplateRef_(this,16,this._anchor_16);
    this._NgIf_16_6 = new import27.Wrapper_NgIf(this._vc_16.vcRef,this._TemplateRef_16_5);
    this._text_17 = this.renderer.createText(this._el_14,'\n                                        ',(null as any));
    this._anchor_18 = this.renderer.createTemplateAnchor(this._el_14,(null as any));
    this._vc_18 = new import16.ViewContainer(18,14,this,this._anchor_18);
    this._TemplateRef_18_5 = new import23.TemplateRef_(this,18,this._anchor_18);
    this._NgIf_18_6 = new import27.Wrapper_NgIf(this._vc_18.vcRef,this._TemplateRef_18_5);
    this._text_19 = this.renderer.createText(this._el_14,'\n                                ',(null as any));
    this._text_20 = this.renderer.createText(this._el_10,'\n                            ',(null as any));
    this._text_21 = this.renderer.createText(this._el_8,'\n                        ',(null as any));
    this._text_22 = this.renderer.createText(this._el_4,'\n                        ',(null as any));
    this._anchor_23 = this.renderer.createTemplateAnchor(this._el_4,(null as any));
    this._vc_23 = new import16.ViewContainer(23,4,this,this._anchor_23);
    this._TemplateRef_23_5 = new import23.TemplateRef_(this,23,this._anchor_23);
    this._NgIf_23_6 = new import27.Wrapper_NgIf(this._vc_23.vcRef,this._TemplateRef_23_5);
    this._text_24 = this.renderer.createText(this._el_4,'\n                    ',(null as any));
    this._text_25 = this.renderer.createText(this._el_2,'\n                ',(null as any));
    this._text_26 = this.renderer.createText(this._el_0,'\n            ',(null as any));
    var disposable_0:Function = import9.subscribeToRenderElement(this,this._el_10,new import9.InlineArray8(8,'mouseenter',(null as any),'mouseleave',(null as any),'click',(null as any),'contextmenu',(null as any)),this.eventHandler(this.handleEvent_10));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._el_4,
      this._text_5,
      this._anchor_6,
      this._text_7,
      this._el_8,
      this._text_9,
      this._el_10,
      this._text_11,
      this._anchor_12,
      this._anchor_13,
      this._el_14,
      this._text_15,
      this._anchor_16,
      this._text_17,
      this._anchor_18,
      this._text_19,
      this._text_20,
      this._text_21,
      this._text_22,
      this._anchor_23,
      this._text_24,
      this._text_25,
      this._text_26
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import23.TemplateRef) && (6 === requestNodeIndex))) { return this._TemplateRef_6_5; }
    if (((token === import28.NgIf) && (6 === requestNodeIndex))) { return this._NgIf_6_6.context; }
    if (((token === import23.TemplateRef) && (12 === requestNodeIndex))) { return this._TemplateRef_12_5; }
    if (((token === import28.NgIf) && (12 === requestNodeIndex))) { return this._NgIf_12_6.context; }
    if (((token === import23.TemplateRef) && (13 === requestNodeIndex))) { return this._TemplateRef_13_5; }
    if (((token === import28.NgIf) && (13 === requestNodeIndex))) { return this._NgIf_13_6.context; }
    if (((token === import23.TemplateRef) && (16 === requestNodeIndex))) { return this._TemplateRef_16_5; }
    if (((token === import28.NgIf) && (16 === requestNodeIndex))) { return this._NgIf_16_6.context; }
    if (((token === import23.TemplateRef) && (18 === requestNodeIndex))) { return this._TemplateRef_18_5; }
    if (((token === import28.NgIf) && (18 === requestNodeIndex))) { return this._NgIf_18_6.context; }
    if (((token === import25.NgClass) && ((10 <= requestNodeIndex) && (requestNodeIndex <= 20)))) { return this._NgClass_10_3.context; }
    if (((token === import25.NgClass) && ((8 <= requestNodeIndex) && (requestNodeIndex <= 21)))) { return this._NgClass_8_3.context; }
    if (((token === import23.TemplateRef) && (23 === requestNodeIndex))) { return this._TemplateRef_23_5; }
    if (((token === import28.NgIf) && (23 === requestNodeIndex))) { return this._NgIf_23_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_6_0_0:boolean = !this.parentView.parentView.context.root;
    this._NgIf_6_6.check_ngIf(currVal_6_0_0,throwOnChange,false);
    this._NgIf_6_6.ngDoCheck(this,this._anchor_6,throwOnChange);
    const currVal_8_0_0:any = 'ui-treenode';
    this._NgClass_8_3.check_klass(currVal_8_0_0,throwOnChange,false);
    const currVal_8_0_1:any = this._map_47(!this.parentView.parentView.context.node.expanded);
    this._NgClass_8_3.check_ngClass(currVal_8_0_1,throwOnChange,false);
    this._NgClass_8_3.ngDoCheck(this,this._el_8,throwOnChange);
    const currVal_10_0_0:any = 'ui-treenode-content ui-state-default ui-corner-all';
    this._NgClass_10_3.check_klass(currVal_10_0_0,throwOnChange,false);
    const currVal_10_0_1:any = this._map_48((this.parentView.parentView.context.hover && this.parentView.parentView.context.tree.selectionMode),this.parentView.parentView.context.isSelected());
    this._NgClass_10_3.check_ngClass(currVal_10_0_1,throwOnChange,false);
    this._NgClass_10_3.ngDoCheck(this,this._el_10,throwOnChange);
    const currVal_12_0_0:boolean = !this.parentView.parentView.context.isLeaf();
    this._NgIf_12_6.check_ngIf(currVal_12_0_0,throwOnChange,false);
    this._NgIf_12_6.ngDoCheck(this,this._anchor_12,throwOnChange);
    const currVal_13_0_0:any = ((this.parentView.parentView.context.node.icon || this.parentView.parentView.context.node.expandedIcon) || this.parentView.parentView.context.node.collapsedIcon);
    this._NgIf_13_6.check_ngIf(currVal_13_0_0,throwOnChange,false);
    this._NgIf_13_6.ngDoCheck(this,this._anchor_13,throwOnChange);
    const currVal_16_0_0:boolean = !this.parentView.parentView.context.tree.getTemplateForNode(this.parentView.parentView.context.node);
    this._NgIf_16_6.check_ngIf(currVal_16_0_0,throwOnChange,false);
    this._NgIf_16_6.ngDoCheck(this,this._anchor_16,throwOnChange);
    const currVal_18_0_0:any = this.parentView.parentView.context.tree.getTemplateForNode(this.parentView.parentView.context.node);
    this._NgIf_18_6.check_ngIf(currVal_18_0_0,throwOnChange,false);
    this._NgIf_18_6.ngDoCheck(this,this._anchor_18,throwOnChange);
    const currVal_23_0_0:any = (this.parentView.parentView.context.node.children && this.parentView.parentView.context.node.expanded);
    this._NgIf_23_6.check_ngIf(currVal_23_0_0,throwOnChange,false);
    this._NgIf_23_6.ngDoCheck(this,this._anchor_23,throwOnChange);
    this._vc_6.detectChangesInNestedViews(throwOnChange);
    this._vc_12.detectChangesInNestedViews(throwOnChange);
    this._vc_13.detectChangesInNestedViews(throwOnChange);
    this._vc_16.detectChangesInNestedViews(throwOnChange);
    this._vc_18.detectChangesInNestedViews(throwOnChange);
    this._vc_23.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_6.destroyNestedViews();
    this._vc_12.destroyNestedViews();
    this._vc_13.destroyNestedViews();
    this._vc_16.destroyNestedViews();
    this._vc_18.destroyNestedViews();
    this._vc_23.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import8.AppView<any> {
    if ((nodeIndex == 6)) { return new View_UITreeNode12(this.viewUtils,this,6,this._anchor_6,this._vc_6); }
    if ((nodeIndex == 12)) { return new View_UITreeNode13(this.viewUtils,this,12,this._anchor_12,this._vc_12); }
    if ((nodeIndex == 13)) { return new View_UITreeNode14(this.viewUtils,this,13,this._anchor_13,this._vc_13); }
    if ((nodeIndex == 16)) { return new View_UITreeNode15(this.viewUtils,this,16,this._anchor_16,this._vc_16); }
    if ((nodeIndex == 18)) { return new View_UITreeNode16(this.viewUtils,this,18,this._anchor_18,this._vc_18); }
    if ((nodeIndex == 23)) { return new View_UITreeNode17(this.viewUtils,this,23,this._anchor_23,this._vc_23); }
    return (null as any);
  }
  handleEvent_10(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'mouseenter')) {
      const pd_sub_0:any = ((<any>(this.parentView.parentView.context.hover = true)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'mouseleave')) {
      const pd_sub_1:any = ((<any>(this.parentView.parentView.context.hover = false)) !== false);
      result = (pd_sub_1 && result);
    }
    if ((eventName == 'click')) {
      const pd_sub_2:any = ((<any>this.parentView.parentView.context.onNodeClick($event)) !== false);
      result = (pd_sub_2 && result);
    }
    if ((eventName == 'contextmenu')) {
      const pd_sub_3:any = ((<any>this.parentView.parentView.context.onNodeRightClick($event)) !== false);
      result = (pd_sub_3 && result);
    }
    return result;
  }
}
class View_UITreeNode1 extends import8.AppView<any> {
  _text_0:any;
  _anchor_1:any;
  /*private*/ _vc_1:import16.ViewContainer;
  _TemplateRef_1_5:any;
  _NgIf_1_6:import27.Wrapper_NgIf;
  _text_2:any;
  _anchor_3:any;
  /*private*/ _vc_3:import16.ViewContainer;
  _TemplateRef_3_5:any;
  _NgIf_3_6:import27.Wrapper_NgIf;
  _text_4:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_UITreeNode1,renderType_UITreeNode,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._text_0 = this.renderer.createText((null as any),'\n            ',(null as any));
    this._anchor_1 = this.renderer.createTemplateAnchor((null as any),(null as any));
    this._vc_1 = new import16.ViewContainer(1,(null as any),this,this._anchor_1);
    this._TemplateRef_1_5 = new import23.TemplateRef_(this,1,this._anchor_1);
    this._NgIf_1_6 = new import27.Wrapper_NgIf(this._vc_1.vcRef,this._TemplateRef_1_5);
    this._text_2 = this.renderer.createText((null as any),'\n            ',(null as any));
    this._anchor_3 = this.renderer.createTemplateAnchor((null as any),(null as any));
    this._vc_3 = new import16.ViewContainer(3,(null as any),this,this._anchor_3);
    this._TemplateRef_3_5 = new import23.TemplateRef_(this,3,this._anchor_3);
    this._NgIf_3_6 = new import27.Wrapper_NgIf(this._vc_3.vcRef,this._TemplateRef_3_5);
    this._text_4 = this.renderer.createText((null as any),'\n        ',(null as any));
    this.init(this._text_4,((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._anchor_1,
      this._text_2,
      this._anchor_3,
      this._text_4
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import23.TemplateRef) && (1 === requestNodeIndex))) { return this._TemplateRef_1_5; }
    if (((token === import28.NgIf) && (1 === requestNodeIndex))) { return this._NgIf_1_6.context; }
    if (((token === import23.TemplateRef) && (3 === requestNodeIndex))) { return this._TemplateRef_3_5; }
    if (((token === import28.NgIf) && (3 === requestNodeIndex))) { return this._NgIf_3_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1_0_0:boolean = !this.parentView.context.tree.horizontal;
    this._NgIf_1_6.check_ngIf(currVal_1_0_0,throwOnChange,false);
    this._NgIf_1_6.ngDoCheck(this,this._anchor_1,throwOnChange);
    const currVal_3_0_0:any = this.parentView.context.tree.horizontal;
    this._NgIf_3_6.check_ngIf(currVal_3_0_0,throwOnChange,false);
    this._NgIf_3_6.ngDoCheck(this,this._anchor_3,throwOnChange);
    this._vc_1.detectChangesInNestedViews(throwOnChange);
    this._vc_3.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_1.destroyNestedViews();
    this._vc_3.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._text_0,ctx);
    cb(this._vc_1.nativeElement,ctx);
    this._vc_1.visitNestedViewRootNodes(cb,ctx);
    cb(this._text_2,ctx);
    cb(this._vc_3.nativeElement,ctx);
    this._vc_3.visitNestedViewRootNodes(cb,ctx);
    cb(this._text_4,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import8.AppView<any> {
    if ((nodeIndex == 1)) { return new View_UITreeNode2(this.viewUtils,this,1,this._anchor_1,this._vc_1); }
    if ((nodeIndex == 3)) { return new View_UITreeNode11(this.viewUtils,this,3,this._anchor_3,this._vc_3); }
    return (null as any);
  }
}
var renderType_UITreeNode:import10.RenderComponentType = import9.createRenderComponentType('',0,import11.ViewEncapsulation.None,styles_UITreeNode,{});
export class View_UITreeNode0 extends import8.AppView<import1.UITreeNode> {
  _text_0:any;
  _anchor_1:any;
  /*private*/ _vc_1:import16.ViewContainer;
  _TemplateRef_1_5:any;
  _NgIf_1_6:import27.Wrapper_NgIf;
  _text_2:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_UITreeNode0,renderType_UITreeNode,import13.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n        ',(null as any));
    this._anchor_1 = this.renderer.createTemplateAnchor(parentRenderNode,(null as any));
    this._vc_1 = new import16.ViewContainer(1,(null as any),this,this._anchor_1);
    this._TemplateRef_1_5 = new import23.TemplateRef_(this,1,this._anchor_1);
    this._NgIf_1_6 = new import27.Wrapper_NgIf(this._vc_1.vcRef,this._TemplateRef_1_5);
    this._text_2 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._anchor_1,
      this._text_2
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import23.TemplateRef) && (1 === requestNodeIndex))) { return this._TemplateRef_1_5; }
    if (((token === import28.NgIf) && (1 === requestNodeIndex))) { return this._NgIf_1_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1_0_0:any = this.context.node;
    this._NgIf_1_6.check_ngIf(currVal_1_0_0,throwOnChange,false);
    this._NgIf_1_6.ngDoCheck(this,this._anchor_1,throwOnChange);
    this._vc_1.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_1.destroyNestedViews();
  }
  createEmbeddedViewInternal(nodeIndex:number):import8.AppView<any> {
    if ((nodeIndex == 1)) { return new View_UITreeNode1(this.viewUtils,this,1,this._anchor_1,this._vc_1); }
    return (null as any);
  }
}
var renderType_TreeNodeTemplateLoader_Host:import10.RenderComponentType = import9.createRenderComponentType('',0,import11.ViewEncapsulation.None,([] as any[]),{});
class View_TreeNodeTemplateLoader_Host0 extends import8.AppView<any> {
  _el_0:any;
  /*private*/ _vc_0:import16.ViewContainer;
  compView_0:import8.AppView<import1.TreeNodeTemplateLoader>;
  _TreeNodeTemplateLoader_0_5:Wrapper_TreeNodeTemplateLoader;
  _el_1:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_TreeNodeTemplateLoader_Host0,renderType_TreeNodeTemplateLoader_Host,import13.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.selectOrCreateRenderHostElement(this.renderer,'p-treeNodeTemplateLoader',import9.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this._vc_0 = new import16.ViewContainer(0,(null as any),this,this._el_0);
    this.compView_0 = new View_TreeNodeTemplateLoader0(this.viewUtils,this,0,this._el_0);
    this._TreeNodeTemplateLoader_0_5 = new Wrapper_TreeNodeTemplateLoader(this._vc_0.vcRef);
    this.compView_0.create(this._TreeNodeTemplateLoader_0_5.context);
    this._el_1 = this.renderer.createTemplateAnchor((null as any),(null as any));
    this.init(this._el_1,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import15.ComponentRef_<any>(0,this,this._el_0,this._TreeNodeTemplateLoader_0_5.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import1.TreeNodeTemplateLoader) && (0 === requestNodeIndex))) { return this._TreeNodeTemplateLoader_0_5.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._TreeNodeTemplateLoader_0_5.ngDoCheck(this,this._el_0,throwOnChange);
    this._vc_0.detectChangesInNestedViews(throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this._vc_0.destroyNestedViews();
    this.compView_0.destroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._vc_0.nativeElement,ctx);
    this._vc_0.visitNestedViewRootNodes(cb,ctx);
    cb(this._el_1,ctx);
  }
}
export const TreeNodeTemplateLoaderNgFactory:import15.ComponentFactory<import1.TreeNodeTemplateLoader> = new import15.ComponentFactory<import1.TreeNodeTemplateLoader>('p-treeNodeTemplateLoader',View_TreeNodeTemplateLoader_Host0,import1.TreeNodeTemplateLoader);
const styles_TreeNodeTemplateLoader:any[] = ([] as any[]);
var renderType_TreeNodeTemplateLoader:import10.RenderComponentType = import9.createRenderComponentType('',0,import11.ViewEncapsulation.None,styles_TreeNodeTemplateLoader,{});
export class View_TreeNodeTemplateLoader0 extends import8.AppView<import1.TreeNodeTemplateLoader> {
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_TreeNodeTemplateLoader0,renderType_TreeNodeTemplateLoader,import13.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): ([] as any[])),(null as any));
    return (null as any);
  }
}
