// primeng/components/orderlist/orderlist.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from '@angular/core/src/linker/ng_module_factory';
import * as import1 from 'primeng/components/orderlist/orderlist';
import * as import2 from '@angular/common/src/common_module';
import * as import3 from 'primeng/components/button/button';
import * as import4 from 'primeng/components/common/shared';
import * as import5 from '@angular/common/src/localization';
import * as import6 from '@angular/core/src/di/injector';
import * as import7 from '@angular/core/src/i18n/tokens';
import * as import8 from '@angular/core/src/change_detection/change_detection_util';
import * as import9 from '@angular/core/src/linker/view';
import * as import10 from '@angular/core/src/linker/view_utils';
import * as import11 from '@angular/core/src/render/api';
import * as import12 from '@angular/core/src/metadata/view';
import * as import13 from 'primeng/components/dom/domhandler';
import * as import14 from '@angular/core/src/linker/query_list';
import * as import15 from '@angular/core/src/linker/view_type';
import * as import16 from '@angular/core/src/change_detection/constants';
import * as import17 from '@angular/core/src/linker/component_factory';
import * as import18 from '@angular/core/src/linker/element_ref';
import * as import19 from '@angular/core/src/linker/view_container';
import * as import20 from '../../../@angular/common/src/directives/ng_class.ngfactory';
import * as import21 from '../common/shared.ngfactory';
import * as import22 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import23 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import24 from '@angular/core/src/linker/template_ref';
import * as import25 from '@angular/common/src/directives/ng_class';
import * as import26 from '../../../@angular/common/src/directives/ng_style.ngfactory';
import * as import27 from '../button/button.ngfactory';
import * as import28 from '../../../@angular/common/src/directives/ng_if.ngfactory';
import * as import29 from '../../../@angular/common/src/directives/ng_for.ngfactory';
import * as import30 from '@angular/common/src/directives/ng_if';
import * as import31 from '@angular/common/src/directives/ng_for';
import * as import32 from '@angular/common/src/directives/ng_style';
class OrderListModuleInjector extends import0.NgModuleInjector<import1.OrderListModule> {
  _CommonModule_0:import2.CommonModule;
  _ButtonModule_1:import3.ButtonModule;
  _SharedModule_2:import4.SharedModule;
  _OrderListModule_3:import1.OrderListModule;
  __NgLocalization_4:import5.NgLocaleLocalization;
  constructor(parent:import6.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_4():import5.NgLocaleLocalization {
    if ((this.__NgLocalization_4 == null)) { (this.__NgLocalization_4 = new import5.NgLocaleLocalization(this.parent.get(import7.LOCALE_ID))); }
    return this.__NgLocalization_4;
  }
  createInternal():import1.OrderListModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._ButtonModule_1 = new import3.ButtonModule();
    this._SharedModule_2 = new import4.SharedModule();
    this._OrderListModule_3 = new import1.OrderListModule();
    return this._OrderListModule_3;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.ButtonModule)) { return this._ButtonModule_1; }
    if ((token === import4.SharedModule)) { return this._SharedModule_2; }
    if ((token === import1.OrderListModule)) { return this._OrderListModule_3; }
    if ((token === import5.NgLocalization)) { return this._NgLocalization_4; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const OrderListModuleNgFactory:import0.NgModuleFactory<import1.OrderListModule> = new import0.NgModuleFactory(OrderListModuleInjector,import1.OrderListModule);
export class Wrapper_OrderList {
  /*private*/ _eventHandler:Function;
  context:import1.OrderList;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  /*private*/ _expr_5:any;
  subscription0:any;
  constructor(p0:any,p1:any) {
    this._changed = false;
    this.context = new import1.OrderList(p0,p1);
    this._expr_0 = import8.UNINITIALIZED;
    this._expr_1 = import8.UNINITIALIZED;
    this._expr_2 = import8.UNINITIALIZED;
    this._expr_3 = import8.UNINITIALIZED;
    this._expr_4 = import8.UNINITIALIZED;
    this._expr_5 = import8.UNINITIALIZED;
  }
  ngOnDetach(view:import9.AppView<any>,componentView:import9.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
    (this.subscription0 && this.subscription0.unsubscribe());
  }
  check_value(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.value = currValue;
      this._expr_0 = currValue;
    }
  }
  check_header(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.header = currValue;
      this._expr_1 = currValue;
    }
  }
  check_style(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_2,currValue))) {
      this._changed = true;
      this.context.style = currValue;
      this._expr_2 = currValue;
    }
  }
  check_styleClass(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_3,currValue))) {
      this._changed = true;
      this.context.styleClass = currValue;
      this._expr_3 = currValue;
    }
  }
  check_listStyle(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_4,currValue))) {
      this._changed = true;
      this.context.listStyle = currValue;
      this._expr_4 = currValue;
    }
  }
  check_responsive(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_5,currValue))) {
      this._changed = true;
      this.context.responsive = currValue;
      this._expr_5 = currValue;
    }
  }
  ngDoCheck(view:import9.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    return changed;
  }
  checkHost(view:import9.AppView<any>,componentView:import9.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import9.AppView<any>,_eventHandler:any,emit0:boolean):void {
    this._eventHandler = _eventHandler;
    if (emit0) { (this.subscription0 = this.context.onReorder.subscribe(_eventHandler.bind(view,'onReorder'))); }
  }
}
var renderType_OrderList_Host:import11.RenderComponentType = import10.createRenderComponentType('',0,import12.ViewEncapsulation.None,([] as any[]),{});
class View_OrderList_Host0 extends import9.AppView<any> {
  _el_0:any;
  compView_0:import9.AppView<import1.OrderList>;
  _DomHandler_0_3:import13.DomHandler;
  _OrderList_0_4:Wrapper_OrderList;
  _query_TemplateRef_0_0:import14.QueryList<any>;
  constructor(viewUtils:import10.ViewUtils,parentView:import9.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_OrderList_Host0,renderType_OrderList_Host,import15.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import16.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import17.ComponentRef<any> {
    this._el_0 = import10.selectOrCreateRenderHostElement(this.renderer,'p-orderList',import10.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this.compView_0 = new View_OrderList0(this.viewUtils,this,0,this._el_0);
    this._DomHandler_0_3 = new import13.DomHandler();
    this._OrderList_0_4 = new Wrapper_OrderList(new import18.ElementRef(this._el_0),this._DomHandler_0_3);
    this._query_TemplateRef_0_0 = new import14.QueryList<any>();
    this.compView_0.create(this._OrderList_0_4.context);
    this._query_TemplateRef_0_0.reset(([] as any[]));
    this._OrderList_0_4.context.itemTemplate = this._query_TemplateRef_0_0.first;
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import17.ComponentRef_<any>(0,this,this._el_0,this._OrderList_0_4.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import13.DomHandler) && (0 === requestNodeIndex))) { return this._DomHandler_0_3; }
    if (((token === import1.OrderList) && (0 === requestNodeIndex))) { return this._OrderList_0_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._OrderList_0_4.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
    if (!throwOnChange) {
      if ((this.numberOfChecks === 0)) { this._OrderList_0_4.context.ngAfterViewInit(); }
      this._OrderList_0_4.context.ngAfterViewChecked();
    }
  }
  destroyInternal():void {
    this.compView_0.destroy();
    this._OrderList_0_4.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
export const OrderListNgFactory:import17.ComponentFactory<import1.OrderList> = new import17.ComponentFactory<import1.OrderList>('p-orderList',View_OrderList_Host0,import1.OrderList);
const styles_OrderList:any[] = ([] as any[]);
class View_OrderList1 extends import9.AppView<any> {
  _el_0:any;
  _text_1:any;
  /*private*/ _expr_2:any;
  constructor(viewUtils:import10.ViewUtils,parentView:import9.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import19.ViewContainer) {
    super(View_OrderList1,renderType_OrderList,import15.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import16.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_2 = import8.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import17.ComponentRef<any> {
    this._el_0 = import10.createRenderElement(this.renderer,(null as any),'div',new import10.InlineArray2(2,'class','ui-orderlist-caption ui-widget-header ui-corner-top'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1
    ]
    ),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2:any = import10.inlineInterpolate(1,'',this.parentView.context.header,'');
    if (import10.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setText(this._text_1,currVal_2);
      this._expr_2 = currVal_2;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_OrderList3 extends import9.AppView<any> {
  _el_0:any;
  constructor(viewUtils:import10.ViewUtils,parentView:import9.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import19.ViewContainer) {
    super(View_OrderList3,renderType_OrderList,import15.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import16.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import17.ComponentRef<any> {
    this._el_0 = this.renderer.createTemplateAnchor((null as any),(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): ([] as any[])),(null as any));
    return (null as any);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_OrderList2 extends import9.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import20.Wrapper_NgClass;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import19.ViewContainer;
  _TemplateRef_2_5:any;
  _TemplateWrapper_2_6:import21.Wrapper_TemplateWrapper;
  _text_3:any;
  _map_8:any;
  constructor(viewUtils:import10.ViewUtils,parentView:import9.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import19.ViewContainer) {
    super(View_OrderList2,renderType_OrderList,import15.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import16.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_8 = import10.pureProxy3((p0:any,p1:any,p2:any):{[key: string]:any} => {
      return {
        'ui-orderlist-item': p0,
        'ui-state-hover': p1,
        'ui-state-highlight': p2
      }
      ;
    });
  }
  createInternal(rootSelector:string):import17.ComponentRef<any> {
    this._el_0 = import10.createRenderElement(this.renderer,(null as any),'li',import10.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_0_3 = new import20.Wrapper_NgClass(this.parentView.parentView.injectorGet(import22.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import23.KeyValueDiffers,this.parentView.parentIndex),new import18.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n                            ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import19.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import24.TemplateRef_(this,2,this._anchor_2);
    this._TemplateWrapper_2_6 = new import21.Wrapper_TemplateWrapper(this._vc_2.vcRef);
    this._text_3 = this.renderer.createText(this._el_0,'\n                        ',(null as any));
    var disposable_0:Function = import10.subscribeToRenderElement(this,this._el_0,new import10.InlineArray8(6,'mouseenter',(null as any),'mouseleave',(null as any),'click',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import24.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import4.TemplateWrapper) && (2 === requestNodeIndex))) { return this._TemplateWrapper_2_6.context; }
    if (((token === import25.NgClass) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 3)))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this._map_8(true,(this.parentView.context.hoveredItem == this.context.$implicit),this.parentView.context.isSelected(this.context.$implicit));
    this._NgClass_0_3.check_ngClass(currVal_0_0_0,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_2_0_0:any = this.context.$implicit;
    this._TemplateWrapper_2_6.check_item(currVal_2_0_0,throwOnChange,false);
    const currVal_2_0_1:any = this.parentView.context.itemTemplate;
    this._TemplateWrapper_2_6.check_templateRef(currVal_2_0_1,throwOnChange,false);
    this._TemplateWrapper_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import9.AppView<any> {
    if ((nodeIndex == 2)) { return new View_OrderList3(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    return (null as any);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'mouseenter')) {
      const pd_sub_0:any = ((<any>(this.parentView.context.hoveredItem = this.context.$implicit)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'mouseleave')) {
      const pd_sub_1:any = ((<any>(this.parentView.context.hoveredItem = (null as any))) !== false);
      result = (pd_sub_1 && result);
    }
    if ((eventName == 'click')) {
      const pd_sub_2:any = ((<any>this.parentView.context.onItemClick($event,this.context.$implicit)) !== false);
      result = (pd_sub_2 && result);
    }
    return result;
  }
}
var renderType_OrderList:import11.RenderComponentType = import10.createRenderComponentType('',0,import12.ViewEncapsulation.None,styles_OrderList,{});
export class View_OrderList0 extends import9.AppView<import1.OrderList> {
  _text_0:any;
  _el_1:any;
  _NgClass_1_3:import20.Wrapper_NgClass;
  _NgStyle_1_4:import26.Wrapper_NgStyle;
  _text_2:any;
  _el_3:any;
  _text_4:any;
  _el_5:any;
  _text_6:any;
  _el_7:any;
  _DomHandler_7_3:import13.DomHandler;
  _Button_7_4:import27.Wrapper_Button;
  _text_8:any;
  _el_9:any;
  _DomHandler_9_3:import13.DomHandler;
  _Button_9_4:import27.Wrapper_Button;
  _text_10:any;
  _el_11:any;
  _DomHandler_11_3:import13.DomHandler;
  _Button_11_4:import27.Wrapper_Button;
  _text_12:any;
  _el_13:any;
  _DomHandler_13_3:import13.DomHandler;
  _Button_13_4:import27.Wrapper_Button;
  _text_14:any;
  _text_15:any;
  _el_16:any;
  _text_17:any;
  _anchor_18:any;
  /*private*/ _vc_18:import19.ViewContainer;
  _TemplateRef_18_5:any;
  _NgIf_18_6:import28.Wrapper_NgIf;
  _text_19:any;
  _el_20:any;
  _NgStyle_20_3:import26.Wrapper_NgStyle;
  _text_21:any;
  _anchor_22:any;
  /*private*/ _vc_22:import19.ViewContainer;
  _TemplateRef_22_5:any;
  _NgFor_22_6:import29.Wrapper_NgFor;
  _text_23:any;
  _text_24:any;
  _text_25:any;
  _text_26:any;
  _text_27:any;
  _map_45:any;
  constructor(viewUtils:import10.ViewUtils,parentView:import9.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_OrderList0,renderType_OrderList,import15.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import16.ChangeDetectorStatus.CheckAlways);
    this._map_45 = import10.pureProxy2((p0:any,p1:any):{[key: string]:any} => {
      return {
        'ui-orderlist ui-grid ui-widget': p0,
        'ui-grid-responsive': p1
      }
      ;
    });
  }
  createInternal(rootSelector:string):import17.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n        ',(null as any));
    this._el_1 = import10.createRenderElement(this.renderer,parentRenderNode,'div',import10.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_1_3 = new import20.Wrapper_NgClass(this.parentView.injectorGet(import22.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import23.KeyValueDiffers,this.parentIndex),new import18.ElementRef(this._el_1),this.renderer);
    this._NgStyle_1_4 = new import26.Wrapper_NgStyle(this.parentView.injectorGet(import23.KeyValueDiffers,this.parentIndex),new import18.ElementRef(this._el_1),this.renderer);
    this._text_2 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_3 = import10.createRenderElement(this.renderer,this._el_1,'div',new import10.InlineArray2(2,'class','ui-grid-row'),(null as any));
    this._text_4 = this.renderer.createText(this._el_3,'\n                ',(null as any));
    this._el_5 = import10.createRenderElement(this.renderer,this._el_3,'div',new import10.InlineArray2(2,'class','ui-orderlist-controls ui-grid-col-2'),(null as any));
    this._text_6 = this.renderer.createText(this._el_5,'\n                    ',(null as any));
    this._el_7 = import10.createRenderElement(this.renderer,this._el_5,'button',new import10.InlineArray8(6,'icon','fa-angle-up','pButton','','type','button'),(null as any));
    this._DomHandler_7_3 = new import13.DomHandler();
    this._Button_7_4 = new import27.Wrapper_Button(new import18.ElementRef(this._el_7),this._DomHandler_7_3);
    this._text_8 = this.renderer.createText(this._el_5,'\n                    ',(null as any));
    this._el_9 = import10.createRenderElement(this.renderer,this._el_5,'button',new import10.InlineArray8(6,'icon','fa-angle-double-up','pButton','','type','button'),(null as any));
    this._DomHandler_9_3 = new import13.DomHandler();
    this._Button_9_4 = new import27.Wrapper_Button(new import18.ElementRef(this._el_9),this._DomHandler_9_3);
    this._text_10 = this.renderer.createText(this._el_5,'\n                    ',(null as any));
    this._el_11 = import10.createRenderElement(this.renderer,this._el_5,'button',new import10.InlineArray8(6,'icon','fa-angle-down','pButton','','type','button'),(null as any));
    this._DomHandler_11_3 = new import13.DomHandler();
    this._Button_11_4 = new import27.Wrapper_Button(new import18.ElementRef(this._el_11),this._DomHandler_11_3);
    this._text_12 = this.renderer.createText(this._el_5,'\n                    ',(null as any));
    this._el_13 = import10.createRenderElement(this.renderer,this._el_5,'button',new import10.InlineArray8(6,'icon','fa-angle-double-down','pButton','','type','button'),(null as any));
    this._DomHandler_13_3 = new import13.DomHandler();
    this._Button_13_4 = new import27.Wrapper_Button(new import18.ElementRef(this._el_13),this._DomHandler_13_3);
    this._text_14 = this.renderer.createText(this._el_5,'\n                ',(null as any));
    this._text_15 = this.renderer.createText(this._el_3,'\n                ',(null as any));
    this._el_16 = import10.createRenderElement(this.renderer,this._el_3,'div',new import10.InlineArray2(2,'class','ui-grid-col-10'),(null as any));
    this._text_17 = this.renderer.createText(this._el_16,'\n                    ',(null as any));
    this._anchor_18 = this.renderer.createTemplateAnchor(this._el_16,(null as any));
    this._vc_18 = new import19.ViewContainer(18,16,this,this._anchor_18);
    this._TemplateRef_18_5 = new import24.TemplateRef_(this,18,this._anchor_18);
    this._NgIf_18_6 = new import28.Wrapper_NgIf(this._vc_18.vcRef,this._TemplateRef_18_5);
    this._text_19 = this.renderer.createText(this._el_16,'\n                    ',(null as any));
    this._el_20 = import10.createRenderElement(this.renderer,this._el_16,'ul',new import10.InlineArray2(2,'class','ui-widget-content ui-orderlist-list ui-corner-bottom'),(null as any));
    this._NgStyle_20_3 = new import26.Wrapper_NgStyle(this.parentView.injectorGet(import23.KeyValueDiffers,this.parentIndex),new import18.ElementRef(this._el_20),this.renderer);
    this._text_21 = this.renderer.createText(this._el_20,'\n                        ',(null as any));
    this._anchor_22 = this.renderer.createTemplateAnchor(this._el_20,(null as any));
    this._vc_22 = new import19.ViewContainer(22,20,this,this._anchor_22);
    this._TemplateRef_22_5 = new import24.TemplateRef_(this,22,this._anchor_22);
    this._NgFor_22_6 = new import29.Wrapper_NgFor(this._vc_22.vcRef,this._TemplateRef_22_5,this.parentView.injectorGet(import22.IterableDiffers,this.parentIndex),this.ref);
    this._text_23 = this.renderer.createText(this._el_20,'\n                    ',(null as any));
    this._text_24 = this.renderer.createText(this._el_16,'\n                ',(null as any));
    this._text_25 = this.renderer.createText(this._el_3,'\n            ',(null as any));
    this._text_26 = this.renderer.createText(this._el_1,'\n        ',(null as any));
    this._text_27 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    var disposable_0:Function = import10.subscribeToRenderElement(this,this._el_7,new import10.InlineArray16(14,'click',(null as any),'mouseenter',(null as any),'mouseleave',(null as any),'mousedown',(null as any),'mouseup',(null as any),'focus',(null as any),'blur',(null as any)),this.eventHandler(this.handleEvent_7));
    var disposable_1:Function = import10.subscribeToRenderElement(this,this._el_9,new import10.InlineArray16(14,'click',(null as any),'mouseenter',(null as any),'mouseleave',(null as any),'mousedown',(null as any),'mouseup',(null as any),'focus',(null as any),'blur',(null as any)),this.eventHandler(this.handleEvent_9));
    var disposable_2:Function = import10.subscribeToRenderElement(this,this._el_11,new import10.InlineArray16(14,'click',(null as any),'mouseenter',(null as any),'mouseleave',(null as any),'mousedown',(null as any),'mouseup',(null as any),'focus',(null as any),'blur',(null as any)),this.eventHandler(this.handleEvent_11));
    var disposable_3:Function = import10.subscribeToRenderElement(this,this._el_13,new import10.InlineArray16(14,'click',(null as any),'mouseenter',(null as any),'mouseleave',(null as any),'mousedown',(null as any),'mouseup',(null as any),'focus',(null as any),'blur',(null as any)),this.eventHandler(this.handleEvent_13));
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2,
      this._el_3,
      this._text_4,
      this._el_5,
      this._text_6,
      this._el_7,
      this._text_8,
      this._el_9,
      this._text_10,
      this._el_11,
      this._text_12,
      this._el_13,
      this._text_14,
      this._text_15,
      this._el_16,
      this._text_17,
      this._anchor_18,
      this._text_19,
      this._el_20,
      this._text_21,
      this._anchor_22,
      this._text_23,
      this._text_24,
      this._text_25,
      this._text_26,
      this._text_27
    ]
    ),[
      disposable_0,
      disposable_1,
      disposable_2,
      disposable_3
    ]
    );
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import13.DomHandler) && (7 === requestNodeIndex))) { return this._DomHandler_7_3; }
    if (((token === import3.Button) && (7 === requestNodeIndex))) { return this._Button_7_4.context; }
    if (((token === import13.DomHandler) && (9 === requestNodeIndex))) { return this._DomHandler_9_3; }
    if (((token === import3.Button) && (9 === requestNodeIndex))) { return this._Button_9_4.context; }
    if (((token === import13.DomHandler) && (11 === requestNodeIndex))) { return this._DomHandler_11_3; }
    if (((token === import3.Button) && (11 === requestNodeIndex))) { return this._Button_11_4.context; }
    if (((token === import13.DomHandler) && (13 === requestNodeIndex))) { return this._DomHandler_13_3; }
    if (((token === import3.Button) && (13 === requestNodeIndex))) { return this._Button_13_4.context; }
    if (((token === import24.TemplateRef) && (18 === requestNodeIndex))) { return this._TemplateRef_18_5; }
    if (((token === import30.NgIf) && (18 === requestNodeIndex))) { return this._NgIf_18_6.context; }
    if (((token === import24.TemplateRef) && (22 === requestNodeIndex))) { return this._TemplateRef_22_5; }
    if (((token === import31.NgFor) && (22 === requestNodeIndex))) { return this._NgFor_22_6.context; }
    if (((token === import32.NgStyle) && ((20 <= requestNodeIndex) && (requestNodeIndex <= 23)))) { return this._NgStyle_20_3.context; }
    if (((token === import25.NgClass) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 26)))) { return this._NgClass_1_3.context; }
    if (((token === import32.NgStyle) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 26)))) { return this._NgStyle_1_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1_0_0:any = this.context.styleClass;
    this._NgClass_1_3.check_klass(currVal_1_0_0,throwOnChange,false);
    const currVal_1_0_1:any = this._map_45(true,this.context.responsive);
    this._NgClass_1_3.check_ngClass(currVal_1_0_1,throwOnChange,false);
    this._NgClass_1_3.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_1_1_0:any = this.context.style;
    this._NgStyle_1_4.check_ngStyle(currVal_1_1_0,throwOnChange,false);
    this._NgStyle_1_4.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_7_0_0:any = 'fa-angle-up';
    this._Button_7_4.check_icon(currVal_7_0_0,throwOnChange,false);
    this._Button_7_4.ngDoCheck(this,this._el_7,throwOnChange);
    const currVal_9_0_0:any = 'fa-angle-double-up';
    this._Button_9_4.check_icon(currVal_9_0_0,throwOnChange,false);
    this._Button_9_4.ngDoCheck(this,this._el_9,throwOnChange);
    const currVal_11_0_0:any = 'fa-angle-down';
    this._Button_11_4.check_icon(currVal_11_0_0,throwOnChange,false);
    this._Button_11_4.ngDoCheck(this,this._el_11,throwOnChange);
    const currVal_13_0_0:any = 'fa-angle-double-down';
    this._Button_13_4.check_icon(currVal_13_0_0,throwOnChange,false);
    this._Button_13_4.ngDoCheck(this,this._el_13,throwOnChange);
    const currVal_18_0_0:any = this.context.header;
    this._NgIf_18_6.check_ngIf(currVal_18_0_0,throwOnChange,false);
    this._NgIf_18_6.ngDoCheck(this,this._anchor_18,throwOnChange);
    const currVal_20_0_0:any = this.context.listStyle;
    this._NgStyle_20_3.check_ngStyle(currVal_20_0_0,throwOnChange,false);
    this._NgStyle_20_3.ngDoCheck(this,this._el_20,throwOnChange);
    const currVal_22_0_0:any = this.context.value;
    this._NgFor_22_6.check_ngForOf(currVal_22_0_0,throwOnChange,false);
    this._NgFor_22_6.ngDoCheck(this,this._anchor_22,throwOnChange);
    this._vc_18.detectChangesInNestedViews(throwOnChange);
    this._vc_22.detectChangesInNestedViews(throwOnChange);
    this._Button_7_4.checkHost(this,this,this._el_7,throwOnChange);
    this._Button_9_4.checkHost(this,this,this._el_9,throwOnChange);
    this._Button_11_4.checkHost(this,this,this._el_11,throwOnChange);
    this._Button_13_4.checkHost(this,this,this._el_13,throwOnChange);
    if (!throwOnChange) {
      if ((this.numberOfChecks === 0)) { this._Button_7_4.context.ngAfterViewInit(); }
      if ((this.numberOfChecks === 0)) { this._Button_9_4.context.ngAfterViewInit(); }
      if ((this.numberOfChecks === 0)) { this._Button_11_4.context.ngAfterViewInit(); }
      if ((this.numberOfChecks === 0)) { this._Button_13_4.context.ngAfterViewInit(); }
    }
  }
  destroyInternal():void {
    this._vc_18.destroyNestedViews();
    this._vc_22.destroyNestedViews();
    this._Button_7_4.ngOnDestroy();
    this._Button_9_4.ngOnDestroy();
    this._Button_11_4.ngOnDestroy();
    this._Button_13_4.ngOnDestroy();
  }
  createEmbeddedViewInternal(nodeIndex:number):import9.AppView<any> {
    if ((nodeIndex == 18)) { return new View_OrderList1(this.viewUtils,this,18,this._anchor_18,this._vc_18); }
    if ((nodeIndex == 22)) { return new View_OrderList2(this.viewUtils,this,22,this._anchor_22,this._vc_22); }
    return (null as any);
  }
  handleEvent_7(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    result = (this._Button_7_4.handleEvent(eventName,$event) && result);
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.context.moveUp($event,this._el_20)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
  handleEvent_9(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    result = (this._Button_9_4.handleEvent(eventName,$event) && result);
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.context.moveTop($event,this._el_20)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
  handleEvent_11(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    result = (this._Button_11_4.handleEvent(eventName,$event) && result);
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.context.moveDown($event,this._el_20)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
  handleEvent_13(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    result = (this._Button_13_4.handleEvent(eventName,$event) && result);
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.context.moveBottom($event,this._el_20)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
