// primeng/components/treetable/treetable.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from '@angular/core/src/linker/ng_module_factory';
import * as import1 from 'primeng/components/treetable/treetable';
import * as import2 from '@angular/common/src/common_module';
import * as import3 from 'primeng/components/common/shared';
import * as import4 from '@angular/common/src/localization';
import * as import5 from '@angular/core/src/di/injector';
import * as import6 from '@angular/core/src/i18n/tokens';
import * as import7 from '@angular/core/src/change_detection/change_detection_util';
import * as import8 from '@angular/core/src/linker/view';
import * as import9 from '@angular/core/src/linker/view_utils';
import * as import10 from '@angular/core/src/render/api';
import * as import11 from '@angular/core/src/metadata/view';
import * as import12 from '@angular/core/src/linker/query_list';
import * as import13 from '@angular/core/src/linker/view_type';
import * as import14 from '@angular/core/src/change_detection/constants';
import * as import15 from '@angular/core/src/linker/component_factory';
import * as import16 from '@angular/core/src/linker/view_container';
import * as import17 from '../common/shared.ngfactory';
import * as import18 from '../../../@angular/common/src/directives/ng_class.ngfactory';
import * as import19 from '../../../@angular/common/src/directives/ng_style.ngfactory';
import * as import20 from '../../../@angular/common/src/directives/ng_if.ngfactory';
import * as import21 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import22 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import23 from '@angular/core/src/linker/element_ref';
import * as import24 from '@angular/core/src/linker/template_ref';
import * as import25 from '@angular/common/src/directives/ng_if';
import * as import26 from '@angular/common/src/directives/ng_class';
import * as import27 from '@angular/common/src/directives/ng_style';
import * as import28 from '../../../@angular/common/src/directives/ng_for.ngfactory';
import * as import29 from '@angular/common/src/directives/ng_for';
class TreeTableModuleInjector extends import0.NgModuleInjector<import1.TreeTableModule> {
  _CommonModule_0:import2.CommonModule;
  _SharedModule_1:import3.SharedModule;
  _TreeTableModule_2:import1.TreeTableModule;
  __NgLocalization_3:import4.NgLocaleLocalization;
  constructor(parent:import5.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_3():import4.NgLocaleLocalization {
    if ((this.__NgLocalization_3 == null)) { (this.__NgLocalization_3 = new import4.NgLocaleLocalization(this.parent.get(import6.LOCALE_ID))); }
    return this.__NgLocalization_3;
  }
  createInternal():import1.TreeTableModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._SharedModule_1 = new import3.SharedModule();
    this._TreeTableModule_2 = new import1.TreeTableModule();
    return this._TreeTableModule_2;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.SharedModule)) { return this._SharedModule_1; }
    if ((token === import1.TreeTableModule)) { return this._TreeTableModule_2; }
    if ((token === import4.NgLocalization)) { return this._NgLocalization_3; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const TreeTableModuleNgFactory:import0.NgModuleFactory<import1.TreeTableModule> = new import0.NgModuleFactory(TreeTableModuleInjector,import1.TreeTableModule);
export class Wrapper_TreeTable {
  /*private*/ _eventHandler:Function;
  context:import1.TreeTable;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  subscription0:any;
  subscription1:any;
  subscription2:any;
  subscription3:any;
  subscription4:any;
  constructor() {
    this._changed = false;
    this.context = new import1.TreeTable();
    this._expr_0 = import7.UNINITIALIZED;
    this._expr_1 = import7.UNINITIALIZED;
    this._expr_2 = import7.UNINITIALIZED;
    this._expr_3 = import7.UNINITIALIZED;
    this._expr_4 = import7.UNINITIALIZED;
  }
  ngOnDetach(view:import8.AppView<any>,componentView:import8.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
    (this.subscription0 && this.subscription0.unsubscribe());
    (this.subscription1 && this.subscription1.unsubscribe());
    (this.subscription2 && this.subscription2.unsubscribe());
    (this.subscription3 && this.subscription3.unsubscribe());
    (this.subscription4 && this.subscription4.unsubscribe());
  }
  check_value(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.value = currValue;
      this._expr_0 = currValue;
    }
  }
  check_selectionMode(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.selectionMode = currValue;
      this._expr_1 = currValue;
    }
  }
  check_selection(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_2,currValue))) {
      this._changed = true;
      this.context.selection = currValue;
      this._expr_2 = currValue;
    }
  }
  check_style(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_3,currValue))) {
      this._changed = true;
      this.context.style = currValue;
      this._expr_3 = currValue;
    }
  }
  check_styleClass(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_4,currValue))) {
      this._changed = true;
      this.context.styleClass = currValue;
      this._expr_4 = currValue;
    }
  }
  ngDoCheck(view:import8.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    return changed;
  }
  checkHost(view:import8.AppView<any>,componentView:import8.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import8.AppView<any>,_eventHandler:any,emit0:boolean,emit1:boolean,emit2:boolean,emit3:boolean,emit4:boolean):void {
    this._eventHandler = _eventHandler;
    if (emit0) { (this.subscription0 = this.context.selectionChange.subscribe(_eventHandler.bind(view,'selectionChange'))); }
    if (emit1) { (this.subscription1 = this.context.onNodeSelect.subscribe(_eventHandler.bind(view,'onNodeSelect'))); }
    if (emit2) { (this.subscription2 = this.context.onNodeUnselect.subscribe(_eventHandler.bind(view,'onNodeUnselect'))); }
    if (emit3) { (this.subscription3 = this.context.onNodeExpand.subscribe(_eventHandler.bind(view,'onNodeExpand'))); }
    if (emit4) { (this.subscription4 = this.context.onNodeCollapse.subscribe(_eventHandler.bind(view,'onNodeCollapse'))); }
  }
}
export class Wrapper_UITreeRow {
  /*private*/ _eventHandler:Function;
  context:import1.UITreeRow;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  constructor(p0:any) {
    this._changed = false;
    this.context = new import1.UITreeRow(p0);
    this._expr_0 = import7.UNINITIALIZED;
    this._expr_1 = import7.UNINITIALIZED;
  }
  ngOnDetach(view:import8.AppView<any>,componentView:import8.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
  }
  check_node(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.node = currValue;
      this._expr_0 = currValue;
    }
  }
  check_level(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.level = currValue;
      this._expr_1 = currValue;
    }
  }
  ngDoCheck(view:import8.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    return changed;
  }
  checkHost(view:import8.AppView<any>,componentView:import8.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import8.AppView<any>,_eventHandler:any):void {
    this._eventHandler = _eventHandler;
  }
}
var renderType_TreeTable_Host:import10.RenderComponentType = import9.createRenderComponentType('',0,import11.ViewEncapsulation.None,([] as any[]),{});
class View_TreeTable_Host0 extends import8.AppView<any> {
  _el_0:any;
  compView_0:import8.AppView<import1.TreeTable>;
  _TreeTable_0_3:Wrapper_TreeTable;
  _query_Header_0_0:import12.QueryList<any>;
  _query_Footer_0_1:import12.QueryList<any>;
  _query_Column_0_2:import12.QueryList<any>;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_TreeTable_Host0,renderType_TreeTable_Host,import13.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.selectOrCreateRenderHostElement(this.renderer,'p-treeTable',import9.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this.compView_0 = new View_TreeTable0(this.viewUtils,this,0,this._el_0);
    this._TreeTable_0_3 = new Wrapper_TreeTable();
    this._query_Header_0_0 = new import12.QueryList<any>();
    this._query_Footer_0_1 = new import12.QueryList<any>();
    this._query_Column_0_2 = new import12.QueryList<any>();
    this.compView_0.create(this._TreeTable_0_3.context);
    this._query_Header_0_0.reset(([] as any[]));
    this._TreeTable_0_3.context.header = this._query_Header_0_0.first;
    this._query_Footer_0_1.reset(([] as any[]));
    this._TreeTable_0_3.context.footer = this._query_Footer_0_1.first;
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import15.ComponentRef_<any>(0,this,this._el_0,this._TreeTable_0_3.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import1.TreeTable) && (0 === requestNodeIndex))) { return this._TreeTable_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._TreeTable_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    if (!throwOnChange) { if (this._query_Column_0_2.dirty) {
      this._query_Column_0_2.reset(([] as any[]));
      this._TreeTable_0_3.context.columns = this._query_Column_0_2;
      this._query_Column_0_2.notifyOnChanges();
    } }
    this.compView_0.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
    this._TreeTable_0_3.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  visitProjectableNodesInternal(nodeIndex:number,ngContentIndex:number,cb:any,ctx:any):void {
    if (((nodeIndex == 0) && (ngContentIndex == 0))) {  }
    if (((nodeIndex == 0) && (ngContentIndex == 1))) {  }
  }
}
export const TreeTableNgFactory:import15.ComponentFactory<import1.TreeTable> = new import15.ComponentFactory<import1.TreeTable>('p-treeTable',View_TreeTable_Host0,import1.TreeTable);
const styles_TreeTable:any[] = ([] as any[]);
class View_TreeTable1 extends import8.AppView<any> {
  _el_0:any;
  _text_1:any;
  _text_2:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_TreeTable1,renderType_TreeTable,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'div',new import9.InlineArray2(2,'class','ui-treetable-header ui-widget-header'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    this.projectNodes(this._el_0,0);
    this._text_2 = this.renderer.createText(this._el_0,'\n            ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._text_2
    ]
    ),(null as any));
    return (null as any);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_TreeTable3 extends import8.AppView<any> {
  _el_0:any;
  _text_1:any;
  /*private*/ _expr_2:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_TreeTable3,renderType_TreeTable,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_2 = import7.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'span',new import9.InlineArray2(2,'class','ui-column-title'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1
    ]
    ),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2:any = import9.inlineInterpolate(1,'',this.parentView.context.$implicit.header,'');
    if (import9.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setText(this._text_1,currVal_2);
      this._expr_2 = currVal_2;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_TreeTable4 extends import8.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  /*private*/ _vc_2:import16.ViewContainer;
  compView_2:import8.AppView<import3.ColumnHeaderTemplateLoader>;
  _ColumnHeaderTemplateLoader_2_5:import17.Wrapper_ColumnHeaderTemplateLoader;
  _text_3:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_TreeTable4,renderType_TreeTable,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'span',new import9.InlineArray2(2,'class','ui-column-title'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                                    ',(null as any));
    this._el_2 = import9.createRenderElement(this.renderer,this._el_0,'p-columnHeaderTemplateLoader',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._vc_2 = new import16.ViewContainer(2,0,this,this._el_2);
    this.compView_2 = new import17.View_ColumnHeaderTemplateLoader0(this.viewUtils,this,2,this._el_2);
    this._ColumnHeaderTemplateLoader_2_5 = new import17.Wrapper_ColumnHeaderTemplateLoader(this._vc_2.vcRef);
    this.compView_2.create(this._ColumnHeaderTemplateLoader_2_5.context);
    this._text_3 = this.renderer.createText(this._el_0,'\n                                ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import3.ColumnHeaderTemplateLoader) && (2 === requestNodeIndex))) { return this._ColumnHeaderTemplateLoader_2_5.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2_0_0:any = this.parentView.context.$implicit;
    this._ColumnHeaderTemplateLoader_2_5.check_column(currVal_2_0_0,throwOnChange,false);
    this._ColumnHeaderTemplateLoader_2_5.ngDoCheck(this,this._el_2,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
    this.compView_2.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
    this.compView_2.destroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_TreeTable2 extends import8.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import18.Wrapper_NgClass;
  _NgStyle_0_4:import19.Wrapper_NgStyle;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import16.ViewContainer;
  _TemplateRef_2_5:any;
  _NgIf_2_6:import20.Wrapper_NgIf;
  _text_3:any;
  _anchor_4:any;
  /*private*/ _vc_4:import16.ViewContainer;
  _TemplateRef_4_5:any;
  _NgIf_4_6:import20.Wrapper_NgIf;
  _text_5:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_TreeTable2,renderType_TreeTable,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'th',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_0_3 = new import18.Wrapper_NgClass(this.parentView.parentView.injectorGet(import21.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import22.KeyValueDiffers,this.parentView.parentIndex),new import23.ElementRef(this._el_0),this.renderer);
    this._NgStyle_0_4 = new import19.Wrapper_NgStyle(this.parentView.parentView.injectorGet(import22.KeyValueDiffers,this.parentView.parentIndex),new import23.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n                                ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import16.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import24.TemplateRef_(this,2,this._anchor_2);
    this._NgIf_2_6 = new import20.Wrapper_NgIf(this._vc_2.vcRef,this._TemplateRef_2_5);
    this._text_3 = this.renderer.createText(this._el_0,'\n                                ',(null as any));
    this._anchor_4 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_4 = new import16.ViewContainer(4,0,this,this._anchor_4);
    this._TemplateRef_4_5 = new import24.TemplateRef_(this,4,this._anchor_4);
    this._NgIf_4_6 = new import20.Wrapper_NgIf(this._vc_4.vcRef,this._TemplateRef_4_5);
    this._text_5 = this.renderer.createText(this._el_0,'\n                            ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3,
      this._anchor_4,
      this._text_5
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import24.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import25.NgIf) && (2 === requestNodeIndex))) { return this._NgIf_2_6.context; }
    if (((token === import24.TemplateRef) && (4 === requestNodeIndex))) { return this._TemplateRef_4_5; }
    if (((token === import25.NgIf) && (4 === requestNodeIndex))) { return this._NgIf_4_6.context; }
    if (((token === import26.NgClass) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 5)))) { return this._NgClass_0_3.context; }
    if (((token === import27.NgStyle) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 5)))) { return this._NgStyle_0_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this.context.$implicit.styleClass;
    this._NgClass_0_3.check_klass(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = 'ui-state-default ui-unselectable-text';
    this._NgClass_0_3.check_ngClass(currVal_0_0_1,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_0_1_0:any = this.context.$implicit.style;
    this._NgStyle_0_4.check_ngStyle(currVal_0_1_0,throwOnChange,false);
    this._NgStyle_0_4.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_2_0_0:boolean = !this.context.$implicit.headerTemplate;
    this._NgIf_2_6.check_ngIf(currVal_2_0_0,throwOnChange,false);
    this._NgIf_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    const currVal_4_0_0:any = this.context.$implicit.headerTemplate;
    this._NgIf_4_6.check_ngIf(currVal_4_0_0,throwOnChange,false);
    this._NgIf_4_6.ngDoCheck(this,this._anchor_4,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
    this._vc_4.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
    this._vc_4.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import8.AppView<any> {
    if ((nodeIndex == 2)) { return new View_TreeTable3(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    if ((nodeIndex == 4)) { return new View_TreeTable4(this.viewUtils,this,4,this._anchor_4,this._vc_4); }
    return (null as any);
  }
}
class View_TreeTable7 extends import8.AppView<any> {
  _el_0:any;
  _text_1:any;
  /*private*/ _expr_2:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_TreeTable7,renderType_TreeTable,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_2 = import7.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'span',new import9.InlineArray2(2,'class','ui-column-footer'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1
    ]
    ),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2:any = import9.inlineInterpolate(1,'',this.parentView.context.$implicit.footer,'');
    if (import9.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setText(this._text_1,currVal_2);
      this._expr_2 = currVal_2;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_TreeTable8 extends import8.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  /*private*/ _vc_2:import16.ViewContainer;
  compView_2:import8.AppView<import3.ColumnFooterTemplateLoader>;
  _ColumnFooterTemplateLoader_2_5:import17.Wrapper_ColumnFooterTemplateLoader;
  _text_3:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_TreeTable8,renderType_TreeTable,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'span',new import9.InlineArray2(2,'class','ui-column-footer'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                                    ',(null as any));
    this._el_2 = import9.createRenderElement(this.renderer,this._el_0,'p-columnFooterTemplateLoader',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._vc_2 = new import16.ViewContainer(2,0,this,this._el_2);
    this.compView_2 = new import17.View_ColumnFooterTemplateLoader0(this.viewUtils,this,2,this._el_2);
    this._ColumnFooterTemplateLoader_2_5 = new import17.Wrapper_ColumnFooterTemplateLoader(this._vc_2.vcRef);
    this.compView_2.create(this._ColumnFooterTemplateLoader_2_5.context);
    this._text_3 = this.renderer.createText(this._el_0,'\n                                ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import3.ColumnFooterTemplateLoader) && (2 === requestNodeIndex))) { return this._ColumnFooterTemplateLoader_2_5.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2_0_0:any = this.parentView.context.$implicit;
    this._ColumnFooterTemplateLoader_2_5.check_column(currVal_2_0_0,throwOnChange,false);
    this._ColumnFooterTemplateLoader_2_5.ngDoCheck(this,this._el_2,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
    this.compView_2.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
    this.compView_2.destroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_TreeTable6 extends import8.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import18.Wrapper_NgClass;
  _NgStyle_0_4:import19.Wrapper_NgStyle;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import16.ViewContainer;
  _TemplateRef_2_5:any;
  _NgIf_2_6:import20.Wrapper_NgIf;
  _text_3:any;
  _anchor_4:any;
  /*private*/ _vc_4:import16.ViewContainer;
  _TemplateRef_4_5:any;
  _NgIf_4_6:import20.Wrapper_NgIf;
  _text_5:any;
  _map_14:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_TreeTable6,renderType_TreeTable,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_14 = import9.pureProxy1((p0:any):{[key: string]:any} => {
      return {'ui-state-default': p0};
    });
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'td',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_0_3 = new import18.Wrapper_NgClass(this.parentView.parentView.parentView.injectorGet(import21.IterableDiffers,this.parentView.parentView.parentIndex),this.parentView.parentView.parentView.injectorGet(import22.KeyValueDiffers,this.parentView.parentView.parentIndex),new import23.ElementRef(this._el_0),this.renderer);
    this._NgStyle_0_4 = new import19.Wrapper_NgStyle(this.parentView.parentView.parentView.injectorGet(import22.KeyValueDiffers,this.parentView.parentView.parentIndex),new import23.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n                                ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import16.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import24.TemplateRef_(this,2,this._anchor_2);
    this._NgIf_2_6 = new import20.Wrapper_NgIf(this._vc_2.vcRef,this._TemplateRef_2_5);
    this._text_3 = this.renderer.createText(this._el_0,'\n                                ',(null as any));
    this._anchor_4 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_4 = new import16.ViewContainer(4,0,this,this._anchor_4);
    this._TemplateRef_4_5 = new import24.TemplateRef_(this,4,this._anchor_4);
    this._NgIf_4_6 = new import20.Wrapper_NgIf(this._vc_4.vcRef,this._TemplateRef_4_5);
    this._text_5 = this.renderer.createText(this._el_0,'\n                            ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3,
      this._anchor_4,
      this._text_5
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import24.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import25.NgIf) && (2 === requestNodeIndex))) { return this._NgIf_2_6.context; }
    if (((token === import24.TemplateRef) && (4 === requestNodeIndex))) { return this._TemplateRef_4_5; }
    if (((token === import25.NgIf) && (4 === requestNodeIndex))) { return this._NgIf_4_6.context; }
    if (((token === import26.NgClass) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 5)))) { return this._NgClass_0_3.context; }
    if (((token === import27.NgStyle) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 5)))) { return this._NgStyle_0_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this.context.$implicit.styleClass;
    this._NgClass_0_3.check_klass(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = this._map_14(true);
    this._NgClass_0_3.check_ngClass(currVal_0_0_1,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_0_1_0:any = this.context.$implicit.style;
    this._NgStyle_0_4.check_ngStyle(currVal_0_1_0,throwOnChange,false);
    this._NgStyle_0_4.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_2_0_0:boolean = !this.context.$implicit.footerTemplate;
    this._NgIf_2_6.check_ngIf(currVal_2_0_0,throwOnChange,false);
    this._NgIf_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    const currVal_4_0_0:any = this.context.$implicit.footerTemplate;
    this._NgIf_4_6.check_ngIf(currVal_4_0_0,throwOnChange,false);
    this._NgIf_4_6.ngDoCheck(this,this._anchor_4,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
    this._vc_4.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
    this._vc_4.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import8.AppView<any> {
    if ((nodeIndex == 2)) { return new View_TreeTable7(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    if ((nodeIndex == 4)) { return new View_TreeTable8(this.viewUtils,this,4,this._anchor_4,this._vc_4); }
    return (null as any);
  }
}
class View_TreeTable5 extends import8.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _anchor_4:any;
  /*private*/ _vc_4:import16.ViewContainer;
  _TemplateRef_4_5:any;
  _NgFor_4_6:import28.Wrapper_NgFor;
  _text_5:any;
  _text_6:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_TreeTable5,renderType_TreeTable,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'tfoot',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                        ',(null as any));
    this._el_2 = import9.createRenderElement(this.renderer,this._el_0,'tr',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._text_3 = this.renderer.createText(this._el_2,'\n                            ',(null as any));
    this._anchor_4 = this.renderer.createTemplateAnchor(this._el_2,(null as any));
    this._vc_4 = new import16.ViewContainer(4,2,this,this._anchor_4);
    this._TemplateRef_4_5 = new import24.TemplateRef_(this,4,this._anchor_4);
    this._NgFor_4_6 = new import28.Wrapper_NgFor(this._vc_4.vcRef,this._TemplateRef_4_5,this.parentView.parentView.injectorGet(import21.IterableDiffers,this.parentView.parentIndex),this.parentView.ref);
    this._text_5 = this.renderer.createText(this._el_2,'\n                        ',(null as any));
    this._text_6 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._anchor_4,
      this._text_5,
      this._text_6
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import24.TemplateRef) && (4 === requestNodeIndex))) { return this._TemplateRef_4_5; }
    if (((token === import29.NgFor) && (4 === requestNodeIndex))) { return this._NgFor_4_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_4_0_0:any = this.parentView.context.columns;
    this._NgFor_4_6.check_ngForOf(currVal_4_0_0,throwOnChange,false);
    this._NgFor_4_6.ngDoCheck(this,this._anchor_4,throwOnChange);
    this._vc_4.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_4.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import8.AppView<any> {
    if ((nodeIndex == 4)) { return new View_TreeTable6(this.viewUtils,this,4,this._anchor_4,this._vc_4); }
    return (null as any);
  }
}
class View_TreeTable9 extends import8.AppView<any> {
  _el_0:any;
  compView_0:import8.AppView<import1.UITreeRow>;
  _UITreeRow_0_3:Wrapper_UITreeRow;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_TreeTable9,renderType_TreeTable,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'tbody',new import9.InlineArray2(2,'pTreeRow',''),(null as any));
    this.compView_0 = new View_UITreeRow0(this.viewUtils,this,0,this._el_0);
    this._UITreeRow_0_3 = new Wrapper_UITreeRow(this.parentView.parentView.injectorGet(import1.TreeTable,this.parentView.parentIndex));
    this.compView_0.create(this._UITreeRow_0_3.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import1.UITreeRow) && (0 === requestNodeIndex))) { return this._UITreeRow_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this.context.$implicit;
    this._UITreeRow_0_3.check_node(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = 0;
    this._UITreeRow_0_3.check_level(currVal_0_0_1,throwOnChange,false);
    this._UITreeRow_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_TreeTable10 extends import8.AppView<any> {
  _el_0:any;
  _text_1:any;
  _text_2:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_TreeTable10,renderType_TreeTable,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'div',new import9.InlineArray2(2,'class','ui-treetable-footer ui-widget-header'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    this.projectNodes(this._el_0,1);
    this._text_2 = this.renderer.createText(this._el_0,'\n            ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._text_2
    ]
    ),(null as any));
    return (null as any);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
var renderType_TreeTable:import10.RenderComponentType = import9.createRenderComponentType('',2,import11.ViewEncapsulation.None,styles_TreeTable,{});
export class View_TreeTable0 extends import8.AppView<import1.TreeTable> {
  _text_0:any;
  _el_1:any;
  _NgClass_1_3:import18.Wrapper_NgClass;
  _NgStyle_1_4:import19.Wrapper_NgStyle;
  _text_2:any;
  _anchor_3:any;
  /*private*/ _vc_3:import16.ViewContainer;
  _TemplateRef_3_5:any;
  _NgIf_3_6:import20.Wrapper_NgIf;
  _text_4:any;
  _el_5:any;
  _text_6:any;
  _el_7:any;
  _text_8:any;
  _el_9:any;
  _text_10:any;
  _el_11:any;
  _text_12:any;
  _anchor_13:any;
  /*private*/ _vc_13:import16.ViewContainer;
  _TemplateRef_13_5:any;
  _NgFor_13_6:import28.Wrapper_NgFor;
  _text_14:any;
  _text_15:any;
  _text_16:any;
  _anchor_17:any;
  /*private*/ _vc_17:import16.ViewContainer;
  _TemplateRef_17_5:any;
  _NgIf_17_6:import20.Wrapper_NgIf;
  _text_18:any;
  _anchor_19:any;
  /*private*/ _vc_19:import16.ViewContainer;
  _TemplateRef_19_5:any;
  _NgFor_19_6:import28.Wrapper_NgFor;
  _text_20:any;
  _text_21:any;
  _text_22:any;
  _anchor_23:any;
  /*private*/ _vc_23:import16.ViewContainer;
  _TemplateRef_23_5:any;
  _NgIf_23_6:import20.Wrapper_NgIf;
  _text_24:any;
  _text_25:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_TreeTable0,renderType_TreeTable,import13.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n        ',(null as any));
    this._el_1 = import9.createRenderElement(this.renderer,parentRenderNode,'div',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_1_3 = new import18.Wrapper_NgClass(this.parentView.injectorGet(import21.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import22.KeyValueDiffers,this.parentIndex),new import23.ElementRef(this._el_1),this.renderer);
    this._NgStyle_1_4 = new import19.Wrapper_NgStyle(this.parentView.injectorGet(import22.KeyValueDiffers,this.parentIndex),new import23.ElementRef(this._el_1),this.renderer);
    this._text_2 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._anchor_3 = this.renderer.createTemplateAnchor(this._el_1,(null as any));
    this._vc_3 = new import16.ViewContainer(3,1,this,this._anchor_3);
    this._TemplateRef_3_5 = new import24.TemplateRef_(this,3,this._anchor_3);
    this._NgIf_3_6 = new import20.Wrapper_NgIf(this._vc_3.vcRef,this._TemplateRef_3_5);
    this._text_4 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_5 = import9.createRenderElement(this.renderer,this._el_1,'div',new import9.InlineArray2(2,'class','ui-treetable-tablewrapper'),(null as any));
    this._text_6 = this.renderer.createText(this._el_5,'\n                ',(null as any));
    this._el_7 = import9.createRenderElement(this.renderer,this._el_5,'table',new import9.InlineArray2(2,'class','ui-widget-content'),(null as any));
    this._text_8 = this.renderer.createText(this._el_7,'\n                    ',(null as any));
    this._el_9 = import9.createRenderElement(this.renderer,this._el_7,'thead',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._text_10 = this.renderer.createText(this._el_9,'\n                        ',(null as any));
    this._el_11 = import9.createRenderElement(this.renderer,this._el_9,'tr',new import9.InlineArray2(2,'class','ui-state-default'),(null as any));
    this._text_12 = this.renderer.createText(this._el_11,'\n                            ',(null as any));
    this._anchor_13 = this.renderer.createTemplateAnchor(this._el_11,(null as any));
    this._vc_13 = new import16.ViewContainer(13,11,this,this._anchor_13);
    this._TemplateRef_13_5 = new import24.TemplateRef_(this,13,this._anchor_13);
    this._NgFor_13_6 = new import28.Wrapper_NgFor(this._vc_13.vcRef,this._TemplateRef_13_5,this.parentView.injectorGet(import21.IterableDiffers,this.parentIndex),this.ref);
    this._text_14 = this.renderer.createText(this._el_11,'\n                        ',(null as any));
    this._text_15 = this.renderer.createText(this._el_9,'\n                    ',(null as any));
    this._text_16 = this.renderer.createText(this._el_7,'\n                    ',(null as any));
    this._anchor_17 = this.renderer.createTemplateAnchor(this._el_7,(null as any));
    this._vc_17 = new import16.ViewContainer(17,7,this,this._anchor_17);
    this._TemplateRef_17_5 = new import24.TemplateRef_(this,17,this._anchor_17);
    this._NgIf_17_6 = new import20.Wrapper_NgIf(this._vc_17.vcRef,this._TemplateRef_17_5);
    this._text_18 = this.renderer.createText(this._el_7,'\n                    ',(null as any));
    this._anchor_19 = this.renderer.createTemplateAnchor(this._el_7,(null as any));
    this._vc_19 = new import16.ViewContainer(19,7,this,this._anchor_19);
    this._TemplateRef_19_5 = new import24.TemplateRef_(this,19,this._anchor_19);
    this._NgFor_19_6 = new import28.Wrapper_NgFor(this._vc_19.vcRef,this._TemplateRef_19_5,this.parentView.injectorGet(import21.IterableDiffers,this.parentIndex),this.ref);
    this._text_20 = this.renderer.createText(this._el_7,'\n                ',(null as any));
    this._text_21 = this.renderer.createText(this._el_5,'\n            ',(null as any));
    this._text_22 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._anchor_23 = this.renderer.createTemplateAnchor(this._el_1,(null as any));
    this._vc_23 = new import16.ViewContainer(23,1,this,this._anchor_23);
    this._TemplateRef_23_5 = new import24.TemplateRef_(this,23,this._anchor_23);
    this._NgIf_23_6 = new import20.Wrapper_NgIf(this._vc_23.vcRef,this._TemplateRef_23_5);
    this._text_24 = this.renderer.createText(this._el_1,'\n        ',(null as any));
    this._text_25 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2,
      this._anchor_3,
      this._text_4,
      this._el_5,
      this._text_6,
      this._el_7,
      this._text_8,
      this._el_9,
      this._text_10,
      this._el_11,
      this._text_12,
      this._anchor_13,
      this._text_14,
      this._text_15,
      this._text_16,
      this._anchor_17,
      this._text_18,
      this._anchor_19,
      this._text_20,
      this._text_21,
      this._text_22,
      this._anchor_23,
      this._text_24,
      this._text_25
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import24.TemplateRef) && (3 === requestNodeIndex))) { return this._TemplateRef_3_5; }
    if (((token === import25.NgIf) && (3 === requestNodeIndex))) { return this._NgIf_3_6.context; }
    if (((token === import24.TemplateRef) && (13 === requestNodeIndex))) { return this._TemplateRef_13_5; }
    if (((token === import29.NgFor) && (13 === requestNodeIndex))) { return this._NgFor_13_6.context; }
    if (((token === import24.TemplateRef) && (17 === requestNodeIndex))) { return this._TemplateRef_17_5; }
    if (((token === import25.NgIf) && (17 === requestNodeIndex))) { return this._NgIf_17_6.context; }
    if (((token === import24.TemplateRef) && (19 === requestNodeIndex))) { return this._TemplateRef_19_5; }
    if (((token === import29.NgFor) && (19 === requestNodeIndex))) { return this._NgFor_19_6.context; }
    if (((token === import24.TemplateRef) && (23 === requestNodeIndex))) { return this._TemplateRef_23_5; }
    if (((token === import25.NgIf) && (23 === requestNodeIndex))) { return this._NgIf_23_6.context; }
    if (((token === import26.NgClass) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 24)))) { return this._NgClass_1_3.context; }
    if (((token === import27.NgStyle) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 24)))) { return this._NgStyle_1_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1_0_0:any = this.context.styleClass;
    this._NgClass_1_3.check_klass(currVal_1_0_0,throwOnChange,false);
    const currVal_1_0_1:any = 'ui-treetable ui-widget';
    this._NgClass_1_3.check_ngClass(currVal_1_0_1,throwOnChange,false);
    this._NgClass_1_3.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_1_1_0:any = this.context.style;
    this._NgStyle_1_4.check_ngStyle(currVal_1_1_0,throwOnChange,false);
    this._NgStyle_1_4.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_3_0_0:any = this.context.header;
    this._NgIf_3_6.check_ngIf(currVal_3_0_0,throwOnChange,false);
    this._NgIf_3_6.ngDoCheck(this,this._anchor_3,throwOnChange);
    const currVal_13_0_0:any = this.context.columns;
    this._NgFor_13_6.check_ngForOf(currVal_13_0_0,throwOnChange,false);
    this._NgFor_13_6.ngDoCheck(this,this._anchor_13,throwOnChange);
    const currVal_17_0_0:any = this.context.hasFooter();
    this._NgIf_17_6.check_ngIf(currVal_17_0_0,throwOnChange,false);
    this._NgIf_17_6.ngDoCheck(this,this._anchor_17,throwOnChange);
    const currVal_19_0_0:any = this.context.value;
    this._NgFor_19_6.check_ngForOf(currVal_19_0_0,throwOnChange,false);
    this._NgFor_19_6.ngDoCheck(this,this._anchor_19,throwOnChange);
    const currVal_23_0_0:any = this.context.footer;
    this._NgIf_23_6.check_ngIf(currVal_23_0_0,throwOnChange,false);
    this._NgIf_23_6.ngDoCheck(this,this._anchor_23,throwOnChange);
    this._vc_3.detectChangesInNestedViews(throwOnChange);
    this._vc_13.detectChangesInNestedViews(throwOnChange);
    this._vc_17.detectChangesInNestedViews(throwOnChange);
    this._vc_19.detectChangesInNestedViews(throwOnChange);
    this._vc_23.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_3.destroyNestedViews();
    this._vc_13.destroyNestedViews();
    this._vc_17.destroyNestedViews();
    this._vc_19.destroyNestedViews();
    this._vc_23.destroyNestedViews();
  }
  createEmbeddedViewInternal(nodeIndex:number):import8.AppView<any> {
    if ((nodeIndex == 3)) { return new View_TreeTable1(this.viewUtils,this,3,this._anchor_3,this._vc_3); }
    if ((nodeIndex == 13)) { return new View_TreeTable2(this.viewUtils,this,13,this._anchor_13,this._vc_13); }
    if ((nodeIndex == 17)) { return new View_TreeTable5(this.viewUtils,this,17,this._anchor_17,this._vc_17); }
    if ((nodeIndex == 19)) { return new View_TreeTable9(this.viewUtils,this,19,this._anchor_19,this._vc_19); }
    if ((nodeIndex == 23)) { return new View_TreeTable10(this.viewUtils,this,23,this._anchor_23,this._vc_23); }
    return (null as any);
  }
}
var renderType_UITreeRow_Host:import10.RenderComponentType = import9.createRenderComponentType('',0,import11.ViewEncapsulation.None,([] as any[]),{});
class View_UITreeRow_Host0 extends import8.AppView<any> {
  _el_0:any;
  compView_0:import8.AppView<import1.UITreeRow>;
  _UITreeRow_0_3:Wrapper_UITreeRow;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_UITreeRow_Host0,renderType_UITreeRow_Host,import13.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.selectOrCreateRenderHostElement(this.renderer,'div',new import9.InlineArray2(2,'pTreeRow',''),rootSelector,(null as any));
    this.compView_0 = new View_UITreeRow0(this.viewUtils,this,0,this._el_0);
    this._UITreeRow_0_3 = new Wrapper_UITreeRow(this.injectorGet(import1.TreeTable,this.parentIndex));
    this.compView_0.create(this._UITreeRow_0_3.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import15.ComponentRef_<any>(0,this,this._el_0,this._UITreeRow_0_3.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import1.UITreeRow) && (0 === requestNodeIndex))) { return this._UITreeRow_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._UITreeRow_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
export const UITreeRowNgFactory:import15.ComponentFactory<import1.UITreeRow> = new import15.ComponentFactory<import1.UITreeRow>('[pTreeRow]',View_UITreeRow_Host0,import1.UITreeRow);
const styles_UITreeRow:any[] = ([] as any[]);
class View_UITreeRow2 extends import8.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import18.Wrapper_NgClass;
  _NgStyle_0_4:import19.Wrapper_NgStyle;
  _map_3:any;
  _map_4:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_UITreeRow2,renderType_UITreeRow,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_3 = import9.pureProxy2((p0:any,p1:any):{[key: string]:any} => {
      return {
        'fa-caret-down': p0,
        'fa-caret-right': p1
      }
      ;
    });
    this._map_4 = import9.pureProxy2((p0:any,p1:any):{[key: string]:any} => {
      return {
        'margin-left': p0,
        visibility: p1
      }
      ;
    });
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'span',new import9.InlineArray2(2,'class','ui-treetable-toggler fa fa-fw ui-c'),(null as any));
    this._NgClass_0_3 = new import18.Wrapper_NgClass(this.parentView.parentView.parentView.injectorGet(import21.IterableDiffers,this.parentView.parentView.parentIndex),this.parentView.parentView.parentView.injectorGet(import22.KeyValueDiffers,this.parentView.parentView.parentIndex),new import23.ElementRef(this._el_0),this.renderer);
    this._NgStyle_0_4 = new import19.Wrapper_NgStyle(this.parentView.parentView.parentView.injectorGet(import22.KeyValueDiffers,this.parentView.parentView.parentIndex),new import23.ElementRef(this._el_0),this.renderer);
    var disposable_0:Function = import9.subscribeToRenderElement(this,this._el_0,new import9.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import26.NgClass) && (0 === requestNodeIndex))) { return this._NgClass_0_3.context; }
    if (((token === import27.NgStyle) && (0 === requestNodeIndex))) { return this._NgStyle_0_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = 'ui-treetable-toggler fa fa-fw ui-c';
    this._NgClass_0_3.check_klass(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = this._map_3(this.parentView.parentView.context.node.expanded,!this.parentView.parentView.context.node.expanded);
    this._NgClass_0_3.check_ngClass(currVal_0_0_1,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_0_1_0:any = this._map_4(((this.parentView.parentView.context.level * 16) + 'px'),(this.parentView.parentView.context.isLeaf()? 'hidden': 'visible'));
    this._NgStyle_0_4.check_ngStyle(currVal_0_1_0,throwOnChange,false);
    this._NgStyle_0_4.ngDoCheck(this,this._el_0,throwOnChange);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.parentView.context.toggle($event)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
class View_UITreeRow3 extends import8.AppView<any> {
  _el_0:any;
  _text_1:any;
  /*private*/ _expr_2:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_UITreeRow3,renderType_UITreeRow,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_2 = import7.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'span',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1
    ]
    ),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2:any = import9.inlineInterpolate(1,'',this.parentView.parentView.context.resolveFieldData(this.parentView.parentView.context.node.data,this.parentView.context.$implicit.field),'');
    if (import9.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setText(this._text_1,currVal_2);
      this._expr_2 = currVal_2;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_UITreeRow4 extends import8.AppView<any> {
  _el_0:any;
  /*private*/ _vc_0:import16.ViewContainer;
  compView_0:import8.AppView<import3.ColumnBodyTemplateLoader>;
  _ColumnBodyTemplateLoader_0_5:import17.Wrapper_ColumnBodyTemplateLoader;
  _el_1:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_UITreeRow4,renderType_UITreeRow,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'p-columnBodyTemplateLoader',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._vc_0 = new import16.ViewContainer(0,(null as any),this,this._el_0);
    this.compView_0 = new import17.View_ColumnBodyTemplateLoader0(this.viewUtils,this,0,this._el_0);
    this._ColumnBodyTemplateLoader_0_5 = new import17.Wrapper_ColumnBodyTemplateLoader(this._vc_0.vcRef);
    this.compView_0.create(this._ColumnBodyTemplateLoader_0_5.context);
    this._el_1 = this.renderer.createTemplateAnchor((null as any),(null as any));
    this.init(this._el_1,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import3.ColumnBodyTemplateLoader) && (0 === requestNodeIndex))) { return this._ColumnBodyTemplateLoader_0_5.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this.parentView.context.$implicit;
    this._ColumnBodyTemplateLoader_0_5.check_column(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = this.parentView.parentView.context.node;
    this._ColumnBodyTemplateLoader_0_5.check_rowData(currVal_0_0_1,throwOnChange,false);
    this._ColumnBodyTemplateLoader_0_5.ngDoCheck(this,this._el_0,throwOnChange);
    this._vc_0.detectChangesInNestedViews(throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this._vc_0.destroyNestedViews();
    this.compView_0.destroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._vc_0.nativeElement,ctx);
    this._vc_0.visitNestedViewRootNodes(cb,ctx);
    cb(this._el_1,ctx);
  }
}
class View_UITreeRow1 extends import8.AppView<any> {
  _el_0:any;
  _NgStyle_0_3:import19.Wrapper_NgStyle;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import16.ViewContainer;
  _TemplateRef_2_5:any;
  _NgIf_2_6:import20.Wrapper_NgIf;
  _text_3:any;
  _anchor_4:any;
  /*private*/ _vc_4:import16.ViewContainer;
  _TemplateRef_4_5:any;
  _NgIf_4_6:import20.Wrapper_NgIf;
  _text_5:any;
  _anchor_6:any;
  /*private*/ _vc_6:import16.ViewContainer;
  _TemplateRef_6_5:any;
  _NgIf_6_6:import20.Wrapper_NgIf;
  _text_7:any;
  /*private*/ _expr_18:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_UITreeRow1,renderType_UITreeRow,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_18 = import7.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'td',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._NgStyle_0_3 = new import19.Wrapper_NgStyle(this.parentView.parentView.injectorGet(import22.KeyValueDiffers,this.parentView.parentIndex),new import23.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import16.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import24.TemplateRef_(this,2,this._anchor_2);
    this._NgIf_2_6 = new import20.Wrapper_NgIf(this._vc_2.vcRef,this._TemplateRef_2_5);
    this._text_3 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    this._anchor_4 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_4 = new import16.ViewContainer(4,0,this,this._anchor_4);
    this._TemplateRef_4_5 = new import24.TemplateRef_(this,4,this._anchor_4);
    this._NgIf_4_6 = new import20.Wrapper_NgIf(this._vc_4.vcRef,this._TemplateRef_4_5);
    this._text_5 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    this._anchor_6 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_6 = new import16.ViewContainer(6,0,this,this._anchor_6);
    this._TemplateRef_6_5 = new import24.TemplateRef_(this,6,this._anchor_6);
    this._NgIf_6_6 = new import20.Wrapper_NgIf(this._vc_6.vcRef,this._TemplateRef_6_5);
    this._text_7 = this.renderer.createText(this._el_0,'\n            ',(null as any));
    var disposable_0:Function = import9.subscribeToRenderElement(this,this._el_0,new import9.InlineArray8(6,'mouseenter',(null as any),'mouseleave',(null as any),'click',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3,
      this._anchor_4,
      this._text_5,
      this._anchor_6,
      this._text_7
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import24.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import25.NgIf) && (2 === requestNodeIndex))) { return this._NgIf_2_6.context; }
    if (((token === import24.TemplateRef) && (4 === requestNodeIndex))) { return this._TemplateRef_4_5; }
    if (((token === import25.NgIf) && (4 === requestNodeIndex))) { return this._NgIf_4_6.context; }
    if (((token === import24.TemplateRef) && (6 === requestNodeIndex))) { return this._TemplateRef_6_5; }
    if (((token === import25.NgIf) && (6 === requestNodeIndex))) { return this._NgIf_6_6.context; }
    if (((token === import27.NgStyle) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 7)))) { return this._NgStyle_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this.context.$implicit.style;
    this._NgStyle_0_3.check_ngStyle(currVal_0_0_0,throwOnChange,false);
    this._NgStyle_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_2_0_0:any = (this.context.index == 0);
    this._NgIf_2_6.check_ngIf(currVal_2_0_0,throwOnChange,false);
    this._NgIf_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    const currVal_4_0_0:boolean = !this.context.$implicit.template;
    this._NgIf_4_6.check_ngIf(currVal_4_0_0,throwOnChange,false);
    this._NgIf_4_6.ngDoCheck(this,this._anchor_4,throwOnChange);
    const currVal_6_0_0:any = this.context.$implicit.template;
    this._NgIf_6_6.check_ngIf(currVal_6_0_0,throwOnChange,false);
    this._NgIf_6_6.ngDoCheck(this,this._anchor_6,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
    this._vc_4.detectChangesInNestedViews(throwOnChange);
    this._vc_6.detectChangesInNestedViews(throwOnChange);
    const currVal_18:any = this.context.$implicit.styleClass;
    if (import9.checkBinding(throwOnChange,this._expr_18,currVal_18)) {
      this.renderer.setElementProperty(this._el_0,'className',currVal_18);
      this._expr_18 = currVal_18;
    }
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
    this._vc_4.destroyNestedViews();
    this._vc_6.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import8.AppView<any> {
    if ((nodeIndex == 2)) { return new View_UITreeRow2(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    if ((nodeIndex == 4)) { return new View_UITreeRow3(this.viewUtils,this,4,this._anchor_4,this._vc_4); }
    if ((nodeIndex == 6)) { return new View_UITreeRow4(this.viewUtils,this,6,this._anchor_6,this._vc_6); }
    return (null as any);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'mouseenter')) {
      const pd_sub_0:any = ((<any>(this.parentView.context.hover = true)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'mouseleave')) {
      const pd_sub_1:any = ((<any>(this.parentView.context.hover = false)) !== false);
      result = (pd_sub_1 && result);
    }
    if ((eventName == 'click')) {
      const pd_sub_2:any = ((<any>this.parentView.context.onRowClick($event)) !== false);
      result = (pd_sub_2 && result);
    }
    return result;
  }
}
class View_UITreeRow6 extends import8.AppView<any> {
  _el_0:any;
  compView_0:import8.AppView<import1.UITreeRow>;
  _UITreeRow_0_3:Wrapper_UITreeRow;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_UITreeRow6,renderType_UITreeRow,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'tbody',new import9.InlineArray2(2,'pTreeRow',''),(null as any));
    this.compView_0 = new View_UITreeRow0(this.viewUtils,this,0,this._el_0);
    this._UITreeRow_0_3 = new Wrapper_UITreeRow(this.parentView.parentView.injectorGet(import1.TreeTable,this.parentView.parentIndex));
    this.compView_0.create(this._UITreeRow_0_3.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import1.UITreeRow) && (0 === requestNodeIndex))) { return this._UITreeRow_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this.context.$implicit;
    this._UITreeRow_0_3.check_node(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = (this.parentView.parentView.context.level + 1);
    this._UITreeRow_0_3.check_level(currVal_0_0_1,throwOnChange,false);
    this._UITreeRow_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_UITreeRow5 extends import8.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _el_4:any;
  _text_5:any;
  _anchor_6:any;
  /*private*/ _vc_6:import16.ViewContainer;
  _TemplateRef_6_5:any;
  _NgFor_6_6:import28.Wrapper_NgFor;
  _text_7:any;
  _text_8:any;
  _text_9:any;
  /*private*/ _expr_13:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_UITreeRow5,renderType_UITreeRow,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_13 = import7.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'div',new import9.InlineArray4(4,'class','ui-treetable-row','style','display:table-row'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n            ',(null as any));
    this._el_2 = import9.createRenderElement(this.renderer,this._el_0,'td',new import9.InlineArray2(2,'class','ui-treetable-child-table-container'),(null as any));
    this._text_3 = this.renderer.createText(this._el_2,'\n                ',(null as any));
    this._el_4 = import9.createRenderElement(this.renderer,this._el_2,'table',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._text_5 = this.renderer.createText(this._el_4,'\n                    ',(null as any));
    this._anchor_6 = this.renderer.createTemplateAnchor(this._el_4,(null as any));
    this._vc_6 = new import16.ViewContainer(6,4,this,this._anchor_6);
    this._TemplateRef_6_5 = new import24.TemplateRef_(this,6,this._anchor_6);
    this._NgFor_6_6 = new import28.Wrapper_NgFor(this._vc_6.vcRef,this._TemplateRef_6_5,this.parentView.injectorGet(import21.IterableDiffers,this.parentIndex),this.parentView.ref);
    this._text_7 = this.renderer.createText(this._el_4,'\n                ',(null as any));
    this._text_8 = this.renderer.createText(this._el_2,'\n            ',(null as any));
    this._text_9 = this.renderer.createText(this._el_0,'\n        ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._el_4,
      this._text_5,
      this._anchor_6,
      this._text_7,
      this._text_8,
      this._text_9
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import24.TemplateRef) && (6 === requestNodeIndex))) { return this._TemplateRef_6_5; }
    if (((token === import29.NgFor) && (6 === requestNodeIndex))) { return this._NgFor_6_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_6_0_0:any = this.parentView.context.node.children;
    this._NgFor_6_6.check_ngForOf(currVal_6_0_0,throwOnChange,false);
    this._NgFor_6_6.ngDoCheck(this,this._anchor_6,throwOnChange);
    this._vc_6.detectChangesInNestedViews(throwOnChange);
    const currVal_13:any = this.parentView.context.treeTable.columns.length;
    if (import9.checkBinding(throwOnChange,this._expr_13,currVal_13)) {
      this.renderer.setElementAttribute(this._el_2,'colspan',((currVal_13 == null)? (null as any): currVal_13.toString()));
      this._expr_13 = currVal_13;
    }
  }
  destroyInternal():void {
    this._vc_6.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import8.AppView<any> {
    if ((nodeIndex == 6)) { return new View_UITreeRow6(this.viewUtils,this,6,this._anchor_6,this._vc_6); }
    return (null as any);
  }
}
var renderType_UITreeRow:import10.RenderComponentType = import9.createRenderComponentType('',0,import11.ViewEncapsulation.None,styles_UITreeRow,{});
export class View_UITreeRow0 extends import8.AppView<import1.UITreeRow> {
  _text_0:any;
  _el_1:any;
  _NgClass_1_3:import18.Wrapper_NgClass;
  _text_2:any;
  _anchor_3:any;
  /*private*/ _vc_3:import16.ViewContainer;
  _TemplateRef_3_5:any;
  _NgFor_3_6:import28.Wrapper_NgFor;
  _text_4:any;
  _text_5:any;
  _anchor_6:any;
  /*private*/ _vc_6:import16.ViewContainer;
  _TemplateRef_6_5:any;
  _NgIf_6_6:import20.Wrapper_NgIf;
  _text_7:any;
  _map_15:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_UITreeRow0,renderType_UITreeRow,import13.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways);
    this._map_15 = import9.pureProxy2((p0:any,p1:any):{[key: string]:any} => {
      return {
        'ui-state-hover': p0,
        'ui-state-highlight': p1
      }
      ;
    });
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n        ',(null as any));
    this._el_1 = import9.createRenderElement(this.renderer,parentRenderNode,'div',new import9.InlineArray2(2,'class','ui-treetable-row'),(null as any));
    this._NgClass_1_3 = new import18.Wrapper_NgClass(this.parentView.injectorGet(import21.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import22.KeyValueDiffers,this.parentIndex),new import23.ElementRef(this._el_1),this.renderer);
    this._text_2 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._anchor_3 = this.renderer.createTemplateAnchor(this._el_1,(null as any));
    this._vc_3 = new import16.ViewContainer(3,1,this,this._anchor_3);
    this._TemplateRef_3_5 = new import24.TemplateRef_(this,3,this._anchor_3);
    this._NgFor_3_6 = new import28.Wrapper_NgFor(this._vc_3.vcRef,this._TemplateRef_3_5,this.parentView.injectorGet(import21.IterableDiffers,this.parentIndex),this.ref);
    this._text_4 = this.renderer.createText(this._el_1,'\n        ',(null as any));
    this._text_5 = this.renderer.createText(parentRenderNode,'\n        ',(null as any));
    this._anchor_6 = this.renderer.createTemplateAnchor(parentRenderNode,(null as any));
    this._vc_6 = new import16.ViewContainer(6,(null as any),this,this._anchor_6);
    this._TemplateRef_6_5 = new import24.TemplateRef_(this,6,this._anchor_6);
    this._NgIf_6_6 = new import20.Wrapper_NgIf(this._vc_6.vcRef,this._TemplateRef_6_5);
    this._text_7 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2,
      this._anchor_3,
      this._text_4,
      this._text_5,
      this._anchor_6,
      this._text_7
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import24.TemplateRef) && (3 === requestNodeIndex))) { return this._TemplateRef_3_5; }
    if (((token === import29.NgFor) && (3 === requestNodeIndex))) { return this._NgFor_3_6.context; }
    if (((token === import26.NgClass) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 4)))) { return this._NgClass_1_3.context; }
    if (((token === import24.TemplateRef) && (6 === requestNodeIndex))) { return this._TemplateRef_6_5; }
    if (((token === import25.NgIf) && (6 === requestNodeIndex))) { return this._NgIf_6_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1_0_0:any = 'ui-treetable-row';
    this._NgClass_1_3.check_klass(currVal_1_0_0,throwOnChange,false);
    const currVal_1_0_1:any = this._map_15((this.context.hover && this.context.treeTable.selectionMode),this.context.isSelected());
    this._NgClass_1_3.check_ngClass(currVal_1_0_1,throwOnChange,false);
    this._NgClass_1_3.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_3_0_0:any = this.context.treeTable.columns;
    this._NgFor_3_6.check_ngForOf(currVal_3_0_0,throwOnChange,false);
    this._NgFor_3_6.ngDoCheck(this,this._anchor_3,throwOnChange);
    const currVal_6_0_0:any = (this.context.node.children && this.context.node.expanded);
    this._NgIf_6_6.check_ngIf(currVal_6_0_0,throwOnChange,false);
    this._NgIf_6_6.ngDoCheck(this,this._anchor_6,throwOnChange);
    this._vc_3.detectChangesInNestedViews(throwOnChange);
    this._vc_6.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_3.destroyNestedViews();
    this._vc_6.destroyNestedViews();
  }
  createEmbeddedViewInternal(nodeIndex:number):import8.AppView<any> {
    if ((nodeIndex == 3)) { return new View_UITreeRow1(this.viewUtils,this,3,this._anchor_3,this._vc_3); }
    if ((nodeIndex == 6)) { return new View_UITreeRow5(this.viewUtils,this,6,this._anchor_6,this._vc_6); }
    return (null as any);
  }
}
