// primeng/components/slidemenu/slidemenu.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from '@angular/core/src/linker/ng_module_factory';
import * as import1 from 'primeng/components/slidemenu/slidemenu';
import * as import2 from '@angular/common/src/common_module';
import * as import3 from '@angular/common/src/localization';
import * as import4 from '@angular/core/src/di/injector';
import * as import5 from '@angular/core/src/i18n/tokens';
import * as import6 from '@angular/core/src/change_detection/change_detection_util';
import * as import7 from '@angular/core/src/linker/view';
import * as import8 from '@angular/core/src/linker/view_utils';
import * as import9 from '@angular/core/src/render/api';
import * as import10 from '@angular/core/src/metadata/view';
import * as import11 from 'primeng/components/dom/domhandler';
import * as import12 from '@angular/core/src/linker/view_type';
import * as import13 from '@angular/core/src/change_detection/constants';
import * as import14 from '@angular/core/src/linker/component_factory';
import * as import15 from '@angular/core/src/linker/element_ref';
import * as import16 from '@angular/core/src/linker/query_list';
import * as import17 from '../../../@angular/common/src/directives/ng_class.ngfactory';
import * as import18 from '../../../@angular/common/src/directives/ng_style.ngfactory';
import * as import19 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import20 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import21 from '@angular/router/src/router';
import * as import22 from '@angular/common/src/directives/ng_class';
import * as import23 from '@angular/common/src/directives/ng_style';
import * as import24 from '@angular/core/src/security';
import * as import25 from '@angular/core/src/linker/view_container';
import * as import26 from '../../../@angular/common/src/directives/ng_if.ngfactory';
import * as import27 from '@angular/core/src/linker/template_ref';
import * as import28 from '@angular/common/src/directives/ng_if';
import * as import29 from '../../../@angular/common/src/directives/ng_for.ngfactory';
import * as import30 from '@angular/common/src/directives/ng_for';
class SlideMenuModuleInjector extends import0.NgModuleInjector<import1.SlideMenuModule> {
  _CommonModule_0:import2.CommonModule;
  _SlideMenuModule_1:import1.SlideMenuModule;
  __NgLocalization_2:import3.NgLocaleLocalization;
  constructor(parent:import4.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_2():import3.NgLocaleLocalization {
    if ((this.__NgLocalization_2 == null)) { (this.__NgLocalization_2 = new import3.NgLocaleLocalization(this.parent.get(import5.LOCALE_ID))); }
    return this.__NgLocalization_2;
  }
  createInternal():import1.SlideMenuModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._SlideMenuModule_1 = new import1.SlideMenuModule();
    return this._SlideMenuModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import1.SlideMenuModule)) { return this._SlideMenuModule_1; }
    if ((token === import3.NgLocalization)) { return this._NgLocalization_2; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const SlideMenuModuleNgFactory:import0.NgModuleFactory<import1.SlideMenuModule> = new import0.NgModuleFactory(SlideMenuModuleInjector,import1.SlideMenuModule);
export class Wrapper_SlideMenu {
  /*private*/ _eventHandler:Function;
  context:import1.SlideMenu;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  /*private*/ _expr_5:any;
  /*private*/ _expr_6:any;
  /*private*/ _expr_7:any;
  /*private*/ _expr_8:any;
  constructor(p0:any,p1:any,p2:any) {
    this._changed = false;
    this.context = new import1.SlideMenu(p0,p1,p2);
    this._expr_0 = import6.UNINITIALIZED;
    this._expr_1 = import6.UNINITIALIZED;
    this._expr_2 = import6.UNINITIALIZED;
    this._expr_3 = import6.UNINITIALIZED;
    this._expr_4 = import6.UNINITIALIZED;
    this._expr_5 = import6.UNINITIALIZED;
    this._expr_6 = import6.UNINITIALIZED;
    this._expr_7 = import6.UNINITIALIZED;
    this._expr_8 = import6.UNINITIALIZED;
  }
  ngOnDetach(view:import7.AppView<any>,componentView:import7.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
    this.context.ngOnDestroy();
  }
  check_model(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.model = currValue;
      this._expr_0 = currValue;
    }
  }
  check_popup(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.popup = currValue;
      this._expr_1 = currValue;
    }
  }
  check_style(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_2,currValue))) {
      this._changed = true;
      this.context.style = currValue;
      this._expr_2 = currValue;
    }
  }
  check_styleClass(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_3,currValue))) {
      this._changed = true;
      this.context.styleClass = currValue;
      this._expr_3 = currValue;
    }
  }
  check_menuWidth(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_4,currValue))) {
      this._changed = true;
      this.context.menuWidth = currValue;
      this._expr_4 = currValue;
    }
  }
  check_viewportHeight(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_5,currValue))) {
      this._changed = true;
      this.context.viewportHeight = currValue;
      this._expr_5 = currValue;
    }
  }
  check_effectDuration(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_6,currValue))) {
      this._changed = true;
      this.context.effectDuration = currValue;
      this._expr_6 = currValue;
    }
  }
  check_easing(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_7,currValue))) {
      this._changed = true;
      this.context.easing = currValue;
      this._expr_7 = currValue;
    }
  }
  check_backLabel(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_8,currValue))) {
      this._changed = true;
      this.context.backLabel = currValue;
      this._expr_8 = currValue;
    }
  }
  ngDoCheck(view:import7.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    return changed;
  }
  checkHost(view:import7.AppView<any>,componentView:import7.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import7.AppView<any>,_eventHandler:any):void {
    this._eventHandler = _eventHandler;
  }
}
export class Wrapper_SlideMenuSub {
  /*private*/ _eventHandler:Function;
  context:import1.SlideMenuSub;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  /*private*/ _expr_5:any;
  constructor(p0:any,p1:any) {
    this._changed = false;
    this.context = new import1.SlideMenuSub(p0,p1);
    this._expr_0 = import6.UNINITIALIZED;
    this._expr_1 = import6.UNINITIALIZED;
    this._expr_2 = import6.UNINITIALIZED;
    this._expr_3 = import6.UNINITIALIZED;
    this._expr_4 = import6.UNINITIALIZED;
    this._expr_5 = import6.UNINITIALIZED;
  }
  ngOnDetach(view:import7.AppView<any>,componentView:import7.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
    this.context.ngOnDestroy();
  }
  check_item(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.item = currValue;
      this._expr_0 = currValue;
    }
  }
  check_root(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.root = currValue;
      this._expr_1 = currValue;
    }
  }
  check_backLabel(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_2,currValue))) {
      this._changed = true;
      this.context.backLabel = currValue;
      this._expr_2 = currValue;
    }
  }
  check_menuWidth(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_3,currValue))) {
      this._changed = true;
      this.context.menuWidth = currValue;
      this._expr_3 = currValue;
    }
  }
  check_effectDuration(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_4,currValue))) {
      this._changed = true;
      this.context.effectDuration = currValue;
      this._expr_4 = currValue;
    }
  }
  check_easing(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_5,currValue))) {
      this._changed = true;
      this.context.easing = currValue;
      this._expr_5 = currValue;
    }
  }
  ngDoCheck(view:import7.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    return changed;
  }
  checkHost(view:import7.AppView<any>,componentView:import7.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import7.AppView<any>,_eventHandler:any):void {
    this._eventHandler = _eventHandler;
  }
}
var renderType_SlideMenu_Host:import9.RenderComponentType = import8.createRenderComponentType('',0,import10.ViewEncapsulation.None,([] as any[]),{});
class View_SlideMenu_Host0 extends import7.AppView<any> {
  _el_0:any;
  compView_0:import7.AppView<import1.SlideMenu>;
  _DomHandler_0_3:import11.DomHandler;
  _SlideMenu_0_4:Wrapper_SlideMenu;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_SlideMenu_Host0,renderType_SlideMenu_Host,import12.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import13.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import14.ComponentRef<any> {
    this._el_0 = import8.selectOrCreateRenderHostElement(this.renderer,'p-slideMenu',import8.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this.compView_0 = new View_SlideMenu0(this.viewUtils,this,0,this._el_0);
    this._DomHandler_0_3 = new import11.DomHandler();
    this._SlideMenu_0_4 = new Wrapper_SlideMenu(new import15.ElementRef(this._el_0),this._DomHandler_0_3,this.renderer);
    this.compView_0.create(this._SlideMenu_0_4.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import14.ComponentRef_<any>(0,this,this._el_0,this._SlideMenu_0_4.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import11.DomHandler) && (0 === requestNodeIndex))) { return this._DomHandler_0_3; }
    if (((token === import1.SlideMenu) && (0 === requestNodeIndex))) { return this._SlideMenu_0_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._SlideMenu_0_4.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
    if (!throwOnChange) { if ((this.numberOfChecks === 0)) { this._SlideMenu_0_4.context.ngAfterViewInit(); } }
  }
  destroyInternal():void {
    this.compView_0.destroy();
    this._SlideMenu_0_4.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
export const SlideMenuNgFactory:import14.ComponentFactory<import1.SlideMenu> = new import14.ComponentFactory<import1.SlideMenu>('p-slideMenu',View_SlideMenu_Host0,import1.SlideMenu);
const styles_SlideMenu:any[] = ([] as any[]);
var renderType_SlideMenu:import9.RenderComponentType = import8.createRenderComponentType('',0,import10.ViewEncapsulation.None,styles_SlideMenu,{});
export class View_SlideMenu0 extends import7.AppView<import1.SlideMenu> {
  _viewQuery_container_0:import16.QueryList<any>;
  _viewQuery_backward_1:import16.QueryList<any>;
  _text_0:any;
  _el_1:any;
  _NgClass_1_3:import17.Wrapper_NgClass;
  _NgStyle_1_4:import18.Wrapper_NgStyle;
  _text_2:any;
  _el_3:any;
  _text_4:any;
  _el_5:any;
  _text_6:any;
  _el_7:any;
  compView_7:import7.AppView<import1.SlideMenuSub>;
  _SlideMenuSub_7_3:Wrapper_SlideMenuSub;
  _text_8:any;
  _text_9:any;
  _el_10:any;
  _text_11:any;
  _el_12:any;
  _el_13:any;
  _text_14:any;
  _text_15:any;
  _text_16:any;
  _text_17:any;
  _text_18:any;
  _map_25:any;
  /*private*/ _expr_26:any;
  /*private*/ _expr_27:any;
  /*private*/ _expr_28:any;
  /*private*/ _expr_29:any;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_SlideMenu0,renderType_SlideMenu,import12.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import13.ChangeDetectorStatus.CheckAlways);
    this._map_25 = import8.pureProxy2((p0:any,p1:any):{[key: string]:any} => {
      return {
        'ui-menu ui-slidemenu ui-widget ui-widget-content ui-corner-all': p0,
        'ui-menu-dynamic ui-shadow': p1
      }
      ;
    });
    this._expr_26 = import6.UNINITIALIZED;
    this._expr_27 = import6.UNINITIALIZED;
    this._expr_28 = import6.UNINITIALIZED;
    this._expr_29 = import6.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import14.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._viewQuery_container_0 = new import16.QueryList<any>();
    this._viewQuery_backward_1 = new import16.QueryList<any>();
    this._text_0 = this.renderer.createText(parentRenderNode,'\n        ',(null as any));
    this._el_1 = import8.createRenderElement(this.renderer,parentRenderNode,'div',import8.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_1_3 = new import17.Wrapper_NgClass(this.parentView.injectorGet(import19.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import20.KeyValueDiffers,this.parentIndex),new import15.ElementRef(this._el_1),this.renderer);
    this._NgStyle_1_4 = new import18.Wrapper_NgStyle(this.parentView.injectorGet(import20.KeyValueDiffers,this.parentIndex),new import15.ElementRef(this._el_1),this.renderer);
    this._text_2 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_3 = import8.createRenderElement(this.renderer,this._el_1,'div',new import8.InlineArray2(2,'class','ui-slidemenu-wrapper'),(null as any));
    this._text_4 = this.renderer.createText(this._el_3,'\n                ',(null as any));
    this._el_5 = import8.createRenderElement(this.renderer,this._el_3,'div',new import8.InlineArray2(2,'class','ui-slidemenu-content'),(null as any));
    this._text_6 = this.renderer.createText(this._el_5,'\n                    ',(null as any));
    this._el_7 = import8.createRenderElement(this.renderer,this._el_5,'p-slideMenuSub',new import8.InlineArray2(2,'root','root'),(null as any));
    this.compView_7 = new View_SlideMenuSub0(this.viewUtils,this,7,this._el_7);
    this._SlideMenuSub_7_3 = new Wrapper_SlideMenuSub(this.parentView.injectorGet(import1.SlideMenu,this.parentIndex),this.parentView.injectorGet(import21.Router,this.parentIndex));
    this.compView_7.create(this._SlideMenuSub_7_3.context);
    this._text_8 = this.renderer.createText(this._el_5,'\n                ',(null as any));
    this._text_9 = this.renderer.createText(this._el_3,'\n                ',(null as any));
    this._el_10 = import8.createRenderElement(this.renderer,this._el_3,'div',new import8.InlineArray2(2,'class','ui-slidemenu-backward ui-widget-header ui-corner-all'),(null as any));
    this._text_11 = this.renderer.createText(this._el_10,'\n                    ',(null as any));
    this._el_12 = import8.createRenderElement(this.renderer,this._el_10,'span',new import8.InlineArray2(2,'class','fa fa-fw fa-caret-left'),(null as any));
    this._el_13 = import8.createRenderElement(this.renderer,this._el_10,'span',import8.EMPTY_INLINE_ARRAY,(null as any));
    this._text_14 = this.renderer.createText(this._el_13,'',(null as any));
    this._text_15 = this.renderer.createText(this._el_10,'\n                ',(null as any));
    this._text_16 = this.renderer.createText(this._el_3,'\n            ',(null as any));
    this._text_17 = this.renderer.createText(this._el_1,'\n        ',(null as any));
    this._text_18 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    var disposable_0:Function = import8.subscribeToRenderElement(this,this._el_1,new import8.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_1));
    var disposable_1:Function = import8.subscribeToRenderElement(this,this._el_10,new import8.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_10));
    this._viewQuery_container_0.reset([new import15.ElementRef(this._el_1)]);
    this.context.containerViewChild = this._viewQuery_container_0.first;
    this._viewQuery_backward_1.reset([new import15.ElementRef(this._el_10)]);
    this.context.backwardViewChild = this._viewQuery_backward_1.first;
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2,
      this._el_3,
      this._text_4,
      this._el_5,
      this._text_6,
      this._el_7,
      this._text_8,
      this._text_9,
      this._el_10,
      this._text_11,
      this._el_12,
      this._el_13,
      this._text_14,
      this._text_15,
      this._text_16,
      this._text_17,
      this._text_18
    ]
    ),[
      disposable_0,
      disposable_1
    ]
    );
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import1.SlideMenuSub) && (7 === requestNodeIndex))) { return this._SlideMenuSub_7_3.context; }
    if (((token === import22.NgClass) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 17)))) { return this._NgClass_1_3.context; }
    if (((token === import23.NgStyle) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 17)))) { return this._NgStyle_1_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1_0_0:any = this.context.styleClass;
    this._NgClass_1_3.check_klass(currVal_1_0_0,throwOnChange,false);
    const currVal_1_0_1:any = this._map_25(true,this.context.popup);
    this._NgClass_1_3.check_ngClass(currVal_1_0_1,throwOnChange,false);
    this._NgClass_1_3.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_1_1_0:any = this.context.style;
    this._NgStyle_1_4.check_ngStyle(currVal_1_1_0,throwOnChange,false);
    this._NgStyle_1_4.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_7_0_0:any = this.context.model;
    this._SlideMenuSub_7_3.check_item(currVal_7_0_0,throwOnChange,false);
    const currVal_7_0_1:any = 'root';
    this._SlideMenuSub_7_3.check_root(currVal_7_0_1,throwOnChange,false);
    const currVal_7_0_2:any = this.context.menuWidth;
    this._SlideMenuSub_7_3.check_menuWidth(currVal_7_0_2,throwOnChange,false);
    const currVal_7_0_3:any = this.context.effectDuration;
    this._SlideMenuSub_7_3.check_effectDuration(currVal_7_0_3,throwOnChange,false);
    const currVal_7_0_4:any = this.context.easing;
    this._SlideMenuSub_7_3.check_easing(currVal_7_0_4,throwOnChange,false);
    this._SlideMenuSub_7_3.ngDoCheck(this,this._el_7,throwOnChange);
    const currVal_26:any = this.context.viewportHeight;
    if (import8.checkBinding(throwOnChange,this._expr_26,currVal_26)) {
      this.renderer.setElementStyle(this._el_3,'height',((this.viewUtils.sanitizer.sanitize(import24.SecurityContext.STYLE,currVal_26) == null)? (null as any): (this.viewUtils.sanitizer.sanitize(import24.SecurityContext.STYLE,currVal_26).toString() + 'px')));
      this._expr_26 = currVal_26;
    }
    const currVal_27:any = this.context.viewportContentHeight;
    if (import8.checkBinding(throwOnChange,this._expr_27,currVal_27)) {
      this.renderer.setElementStyle(this._el_5,'height',((this.viewUtils.sanitizer.sanitize(import24.SecurityContext.STYLE,currVal_27) == null)? (null as any): (this.viewUtils.sanitizer.sanitize(import24.SecurityContext.STYLE,currVal_27).toString() + 'px')));
      this._expr_27 = currVal_27;
    }
    const currVal_28:any = (this.context.left? 'block': 'none');
    if (import8.checkBinding(throwOnChange,this._expr_28,currVal_28)) {
      this.renderer.setElementStyle(this._el_10,'display',((this.viewUtils.sanitizer.sanitize(import24.SecurityContext.STYLE,currVal_28) == null)? (null as any): this.viewUtils.sanitizer.sanitize(import24.SecurityContext.STYLE,currVal_28).toString()));
      this._expr_28 = currVal_28;
    }
    const currVal_29:any = import8.inlineInterpolate(1,'',this.context.backLabel,'');
    if (import8.checkBinding(throwOnChange,this._expr_29,currVal_29)) {
      this.renderer.setText(this._text_14,currVal_29);
      this._expr_29 = currVal_29;
    }
    this.compView_7.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_7.destroy();
    this._SlideMenuSub_7_3.ngOnDestroy();
  }
  handleEvent_1(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.context.onClick($event)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
  handleEvent_10(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.context.goBack()) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
var renderType_SlideMenuSub_Host:import9.RenderComponentType = import8.createRenderComponentType('',0,import10.ViewEncapsulation.None,([] as any[]),{});
class View_SlideMenuSub_Host0 extends import7.AppView<any> {
  _el_0:any;
  compView_0:import7.AppView<import1.SlideMenuSub>;
  _SlideMenuSub_0_3:Wrapper_SlideMenuSub;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_SlideMenuSub_Host0,renderType_SlideMenuSub_Host,import12.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import13.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import14.ComponentRef<any> {
    this._el_0 = import8.selectOrCreateRenderHostElement(this.renderer,'p-slideMenuSub',import8.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this.compView_0 = new View_SlideMenuSub0(this.viewUtils,this,0,this._el_0);
    this._SlideMenuSub_0_3 = new Wrapper_SlideMenuSub(this.injectorGet(import1.SlideMenu,this.parentIndex),this.injectorGet(import21.Router,this.parentIndex));
    this.compView_0.create(this._SlideMenuSub_0_3.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import14.ComponentRef_<any>(0,this,this._el_0,this._SlideMenuSub_0_3.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import1.SlideMenuSub) && (0 === requestNodeIndex))) { return this._SlideMenuSub_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._SlideMenuSub_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
    this._SlideMenuSub_0_3.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
export const SlideMenuSubNgFactory:import14.ComponentFactory<import1.SlideMenuSub> = new import14.ComponentFactory<import1.SlideMenuSub>('p-slideMenuSub',View_SlideMenuSub_Host0,import1.SlideMenuSub);
const styles_SlideMenuSub:any[] = ([] as any[]);
class View_SlideMenuSub2 extends import7.AppView<any> {
  _el_0:any;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import25.ViewContainer) {
    super(View_SlideMenuSub2,renderType_SlideMenuSub,import12.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import13.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import14.ComponentRef<any> {
    this._el_0 = import8.createRenderElement(this.renderer,(null as any),'span',new import8.InlineArray2(2,'class','ui-submenu-icon fa fa-fw fa-caret-right'),(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return (null as any);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_SlideMenuSub3 extends import7.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import17.Wrapper_NgClass;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import25.ViewContainer) {
    super(View_SlideMenuSub3,renderType_SlideMenuSub,import12.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import13.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import14.ComponentRef<any> {
    this._el_0 = import8.createRenderElement(this.renderer,(null as any),'span',new import8.InlineArray2(2,'class','ui-menuitem-icon fa fa-fw'),(null as any));
    this._NgClass_0_3 = new import17.Wrapper_NgClass(this.parentView.parentView.parentView.injectorGet(import19.IterableDiffers,this.parentView.parentView.parentIndex),this.parentView.parentView.parentView.injectorGet(import20.KeyValueDiffers,this.parentView.parentView.parentIndex),new import15.ElementRef(this._el_0),this.renderer);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import22.NgClass) && (0 === requestNodeIndex))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = 'ui-menuitem-icon fa fa-fw';
    this._NgClass_0_3.check_klass(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = this.parentView.context.$implicit.icon;
    this._NgClass_0_3.check_ngClass(currVal_0_0_1,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_SlideMenuSub4 extends import7.AppView<any> {
  _el_0:any;
  compView_0:import7.AppView<import1.SlideMenuSub>;
  _SlideMenuSub_0_3:Wrapper_SlideMenuSub;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import25.ViewContainer) {
    super(View_SlideMenuSub4,renderType_SlideMenuSub,import12.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import13.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import14.ComponentRef<any> {
    this._el_0 = import8.createRenderElement(this.renderer,(null as any),'p-slideMenuSub',new import8.InlineArray2(2,'class','ui-submenu'),(null as any));
    this.compView_0 = new View_SlideMenuSub0(this.viewUtils,this,0,this._el_0);
    this._SlideMenuSub_0_3 = new Wrapper_SlideMenuSub(this.parentView.parentView.parentView.injectorGet(import1.SlideMenu,this.parentView.parentView.parentIndex),this.parentView.parentView.parentView.injectorGet(import21.Router,this.parentView.parentView.parentIndex));
    this.compView_0.create(this._SlideMenuSub_0_3.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import1.SlideMenuSub) && (0 === requestNodeIndex))) { return this._SlideMenuSub_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this.parentView.context.$implicit;
    this._SlideMenuSub_0_3.check_item(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = this.parentView.parentView.context.menuWidth;
    this._SlideMenuSub_0_3.check_menuWidth(currVal_0_0_1,throwOnChange,false);
    this._SlideMenuSub_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
    this._SlideMenuSub_0_3.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_SlideMenuSub1 extends import7.AppView<any> {
  _text_0:any;
  _el_1:any;
  _NgClass_1_3:import17.Wrapper_NgClass;
  _text_2:any;
  _el_3:any;
  _NgClass_3_3:import17.Wrapper_NgClass;
  _text_4:any;
  _anchor_5:any;
  /*private*/ _vc_5:import25.ViewContainer;
  _TemplateRef_5_5:any;
  _NgIf_5_6:import26.Wrapper_NgIf;
  _text_6:any;
  _anchor_7:any;
  /*private*/ _vc_7:import25.ViewContainer;
  _TemplateRef_7_5:any;
  _NgIf_7_6:import26.Wrapper_NgIf;
  _text_8:any;
  _el_9:any;
  _text_10:any;
  _text_11:any;
  _text_12:any;
  _anchor_13:any;
  /*private*/ _vc_13:import25.ViewContainer;
  _TemplateRef_13_5:any;
  _NgIf_13_6:import26.Wrapper_NgIf;
  _text_14:any;
  _text_15:any;
  _map_27:any;
  /*private*/ _expr_28:any;
  _map_29:any;
  /*private*/ _expr_30:any;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import25.ViewContainer) {
    super(View_SlideMenuSub1,renderType_SlideMenuSub,import12.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import13.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_27 = import8.pureProxy3((p0:any,p1:any,p2:any):{[key: string]:any} => {
      return {
        'ui-menuitem ui-widget ui-corner-all': p0,
        'ui-menu-parent': p1,
        'ui-menuitem-active': p2
      }
      ;
    });
    this._expr_28 = import6.UNINITIALIZED;
    this._map_29 = import8.pureProxy3((p0:any,p1:any,p2:any):{[key: string]:any} => {
      return {
        'ui-state-hover': p0,
        'ui-menuitem-link-parent': p1,
        'ui-state-disabled': p2
      }
      ;
    });
    this._expr_30 = import6.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import14.ComponentRef<any> {
    this._text_0 = this.renderer.createText((null as any),'\n                ',(null as any));
    this._el_1 = import8.createRenderElement(this.renderer,(null as any),'li',import8.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_1_3 = new import17.Wrapper_NgClass(this.parentView.parentView.injectorGet(import19.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import20.KeyValueDiffers,this.parentView.parentIndex),new import15.ElementRef(this._el_1),this.renderer);
    this._text_2 = this.renderer.createText(this._el_1,'\n                    ',(null as any));
    this._el_3 = import8.createRenderElement(this.renderer,this._el_1,'a',new import8.InlineArray2(2,'class','ui-menuitem-link ui-corner-all'),(null as any));
    this._NgClass_3_3 = new import17.Wrapper_NgClass(this.parentView.parentView.injectorGet(import19.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import20.KeyValueDiffers,this.parentView.parentIndex),new import15.ElementRef(this._el_3),this.renderer);
    this._text_4 = this.renderer.createText(this._el_3,'\n                        ',(null as any));
    this._anchor_5 = this.renderer.createTemplateAnchor(this._el_3,(null as any));
    this._vc_5 = new import25.ViewContainer(5,3,this,this._anchor_5);
    this._TemplateRef_5_5 = new import27.TemplateRef_(this,5,this._anchor_5);
    this._NgIf_5_6 = new import26.Wrapper_NgIf(this._vc_5.vcRef,this._TemplateRef_5_5);
    this._text_6 = this.renderer.createText(this._el_3,'\n                        ',(null as any));
    this._anchor_7 = this.renderer.createTemplateAnchor(this._el_3,(null as any));
    this._vc_7 = new import25.ViewContainer(7,3,this,this._anchor_7);
    this._TemplateRef_7_5 = new import27.TemplateRef_(this,7,this._anchor_7);
    this._NgIf_7_6 = new import26.Wrapper_NgIf(this._vc_7.vcRef,this._TemplateRef_7_5);
    this._text_8 = this.renderer.createText(this._el_3,'\n                        ',(null as any));
    this._el_9 = import8.createRenderElement(this.renderer,this._el_3,'span',new import8.InlineArray2(2,'class','ui-menuitem-text'),(null as any));
    this._text_10 = this.renderer.createText(this._el_9,'',(null as any));
    this._text_11 = this.renderer.createText(this._el_3,'\n                    ',(null as any));
    this._text_12 = this.renderer.createText(this._el_1,'\n                    ',(null as any));
    this._anchor_13 = this.renderer.createTemplateAnchor(this._el_1,(null as any));
    this._vc_13 = new import25.ViewContainer(13,1,this,this._anchor_13);
    this._TemplateRef_13_5 = new import27.TemplateRef_(this,13,this._anchor_13);
    this._NgIf_13_6 = new import26.Wrapper_NgIf(this._vc_13.vcRef,this._TemplateRef_13_5);
    this._text_14 = this.renderer.createText(this._el_1,'\n                ',(null as any));
    this._text_15 = this.renderer.createText((null as any),'\n            ',(null as any));
    var disposable_0:Function = import8.subscribeToRenderElement(this,this._el_3,new import8.InlineArray8(6,'click',(null as any),'mouseenter',(null as any),'mouseleave',(null as any)),this.eventHandler(this.handleEvent_3));
    this.init(this._text_15,((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2,
      this._el_3,
      this._text_4,
      this._anchor_5,
      this._text_6,
      this._anchor_7,
      this._text_8,
      this._el_9,
      this._text_10,
      this._text_11,
      this._text_12,
      this._anchor_13,
      this._text_14,
      this._text_15
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import27.TemplateRef) && (5 === requestNodeIndex))) { return this._TemplateRef_5_5; }
    if (((token === import28.NgIf) && (5 === requestNodeIndex))) { return this._NgIf_5_6.context; }
    if (((token === import27.TemplateRef) && (7 === requestNodeIndex))) { return this._TemplateRef_7_5; }
    if (((token === import28.NgIf) && (7 === requestNodeIndex))) { return this._NgIf_7_6.context; }
    if (((token === import22.NgClass) && ((3 <= requestNodeIndex) && (requestNodeIndex <= 11)))) { return this._NgClass_3_3.context; }
    if (((token === import27.TemplateRef) && (13 === requestNodeIndex))) { return this._TemplateRef_13_5; }
    if (((token === import28.NgIf) && (13 === requestNodeIndex))) { return this._NgIf_13_6.context; }
    if (((token === import22.NgClass) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 14)))) { return this._NgClass_1_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1_0_0:any = this._map_27(true,this.context.$implicit.items,(this._el_1 == this.parentView.context.activeItem));
    this._NgClass_1_3.check_ngClass(currVal_1_0_0,throwOnChange,false);
    this._NgClass_1_3.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_3_0_0:any = 'ui-menuitem-link ui-corner-all';
    this._NgClass_3_3.check_klass(currVal_3_0_0,throwOnChange,false);
    const currVal_3_0_1:any = this._map_29(((this._el_3 == this.parentView.context.hoveredLink) && !this.context.$implicit.disabled),this.context.$implicit.items,this.context.$implicit.disabled);
    this._NgClass_3_3.check_ngClass(currVal_3_0_1,throwOnChange,false);
    this._NgClass_3_3.ngDoCheck(this,this._el_3,throwOnChange);
    const currVal_5_0_0:any = this.context.$implicit.items;
    this._NgIf_5_6.check_ngIf(currVal_5_0_0,throwOnChange,false);
    this._NgIf_5_6.ngDoCheck(this,this._anchor_5,throwOnChange);
    const currVal_7_0_0:any = this.context.$implicit.icon;
    this._NgIf_7_6.check_ngIf(currVal_7_0_0,throwOnChange,false);
    this._NgIf_7_6.ngDoCheck(this,this._anchor_7,throwOnChange);
    const currVal_13_0_0:any = this.context.$implicit.items;
    this._NgIf_13_6.check_ngIf(currVal_13_0_0,throwOnChange,false);
    this._NgIf_13_6.ngDoCheck(this,this._anchor_13,throwOnChange);
    this._vc_5.detectChangesInNestedViews(throwOnChange);
    this._vc_7.detectChangesInNestedViews(throwOnChange);
    this._vc_13.detectChangesInNestedViews(throwOnChange);
    const currVal_28:any = (this.context.$implicit.url || '#');
    if (import8.checkBinding(throwOnChange,this._expr_28,currVal_28)) {
      this.renderer.setElementProperty(this._el_3,'href',this.viewUtils.sanitizer.sanitize(import24.SecurityContext.URL,currVal_28));
      this._expr_28 = currVal_28;
    }
    const currVal_30:any = import8.inlineInterpolate(1,'',this.context.$implicit.label,'');
    if (import8.checkBinding(throwOnChange,this._expr_30,currVal_30)) {
      this.renderer.setText(this._text_10,currVal_30);
      this._expr_30 = currVal_30;
    }
  }
  destroyInternal():void {
    this._vc_5.destroyNestedViews();
    this._vc_7.destroyNestedViews();
    this._vc_13.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._text_0,ctx);
    cb(this._el_1,ctx);
    cb(this._text_15,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import7.AppView<any> {
    if ((nodeIndex == 5)) { return new View_SlideMenuSub2(this.viewUtils,this,5,this._anchor_5,this._vc_5); }
    if ((nodeIndex == 7)) { return new View_SlideMenuSub3(this.viewUtils,this,7,this._anchor_7,this._vc_7); }
    if ((nodeIndex == 13)) { return new View_SlideMenuSub4(this.viewUtils,this,13,this._anchor_13,this._vc_13); }
    return (null as any);
  }
  handleEvent_3(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.context.itemClick($event,this.context.$implicit,this._el_1)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'mouseenter')) {
      const pd_sub_1:any = ((<any>(this.parentView.context.hoveredLink = this._el_3)) !== false);
      result = (pd_sub_1 && result);
    }
    if ((eventName == 'mouseleave')) {
      const pd_sub_2:any = ((<any>(this.parentView.context.hoveredLink = (null as any))) !== false);
      result = (pd_sub_2 && result);
    }
    return result;
  }
}
var renderType_SlideMenuSub:import9.RenderComponentType = import8.createRenderComponentType('',0,import10.ViewEncapsulation.None,styles_SlideMenuSub,{});
export class View_SlideMenuSub0 extends import7.AppView<import1.SlideMenuSub> {
  _text_0:any;
  _el_1:any;
  _NgClass_1_3:import17.Wrapper_NgClass;
  _text_2:any;
  _anchor_3:any;
  /*private*/ _vc_3:import25.ViewContainer;
  _TemplateRef_3_5:any;
  _NgFor_3_6:import29.Wrapper_NgFor;
  _text_4:any;
  _text_5:any;
  /*private*/ _expr_10:any;
  /*private*/ _expr_11:any;
  /*private*/ _expr_12:any;
  /*private*/ _expr_13:any;
  /*private*/ _expr_14:any;
  _map_15:any;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_SlideMenuSub0,renderType_SlideMenuSub,import12.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import13.ChangeDetectorStatus.CheckAlways);
    this._expr_10 = import6.UNINITIALIZED;
    this._expr_11 = import6.UNINITIALIZED;
    this._expr_12 = import6.UNINITIALIZED;
    this._expr_13 = import6.UNINITIALIZED;
    this._expr_14 = import6.UNINITIALIZED;
    this._map_15 = import8.pureProxy2((p0:any,p1:any):{[key: string]:any} => {
      return {
        'ui-helper-reset ui-menu-rootlist': p0,
        'ui-widget-content ui-corner-all ui-helper-clearfix ui-menu-child': p1
      }
      ;
    });
  }
  createInternal(rootSelector:string):import14.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n        ',(null as any));
    this._el_1 = import8.createRenderElement(this.renderer,parentRenderNode,'ul',new import8.InlineArray2(2,'class','ui-menu-list'),(null as any));
    this._NgClass_1_3 = new import17.Wrapper_NgClass(this.parentView.injectorGet(import19.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import20.KeyValueDiffers,this.parentIndex),new import15.ElementRef(this._el_1),this.renderer);
    this._text_2 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._anchor_3 = this.renderer.createTemplateAnchor(this._el_1,(null as any));
    this._vc_3 = new import25.ViewContainer(3,1,this,this._anchor_3);
    this._TemplateRef_3_5 = new import27.TemplateRef_(this,3,this._anchor_3);
    this._NgFor_3_6 = new import29.Wrapper_NgFor(this._vc_3.vcRef,this._TemplateRef_3_5,this.parentView.injectorGet(import19.IterableDiffers,this.parentIndex),this.ref);
    this._text_4 = this.renderer.createText(this._el_1,'\n        ',(null as any));
    this._text_5 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2,
      this._anchor_3,
      this._text_4,
      this._text_5
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import27.TemplateRef) && (3 === requestNodeIndex))) { return this._TemplateRef_3_5; }
    if (((token === import30.NgFor) && (3 === requestNodeIndex))) { return this._NgFor_3_6.context; }
    if (((token === import22.NgClass) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 4)))) { return this._NgClass_1_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1_0_0:any = 'ui-menu-list';
    this._NgClass_1_3.check_klass(currVal_1_0_0,throwOnChange,false);
    const currVal_1_0_1:any = this._map_15(this.context.root,!this.context.root);
    this._NgClass_1_3.check_ngClass(currVal_1_0_1,throwOnChange,false);
    this._NgClass_1_3.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_3_0_0:any = (this.context.root? this.context.item: this.context.item.items);
    this._NgFor_3_6.check_ngForOf(currVal_3_0_0,throwOnChange,false);
    this._NgFor_3_6.ngDoCheck(this,this._anchor_3,throwOnChange);
    this._vc_3.detectChangesInNestedViews(throwOnChange);
    const currVal_10:any = this.context.menuWidth;
    if (import8.checkBinding(throwOnChange,this._expr_10,currVal_10)) {
      this.renderer.setElementStyle(this._el_1,'width',((this.viewUtils.sanitizer.sanitize(import24.SecurityContext.STYLE,currVal_10) == null)? (null as any): (this.viewUtils.sanitizer.sanitize(import24.SecurityContext.STYLE,currVal_10).toString() + 'px')));
      this._expr_10 = currVal_10;
    }
    const currVal_11:any = (this.context.root? this.context.slideMenu.left: this.context.slideMenu.menuWidth);
    if (import8.checkBinding(throwOnChange,this._expr_11,currVal_11)) {
      this.renderer.setElementStyle(this._el_1,'left',((this.viewUtils.sanitizer.sanitize(import24.SecurityContext.STYLE,currVal_11) == null)? (null as any): (this.viewUtils.sanitizer.sanitize(import24.SecurityContext.STYLE,currVal_11).toString() + 'px')));
      this._expr_11 = currVal_11;
    }
    const currVal_12:any = (this.context.root? 'left': 'none');
    if (import8.checkBinding(throwOnChange,this._expr_12,currVal_12)) {
      this.renderer.setElementStyle(this._el_1,'transitionProperty',((this.viewUtils.sanitizer.sanitize(import24.SecurityContext.STYLE,currVal_12) == null)? (null as any): this.viewUtils.sanitizer.sanitize(import24.SecurityContext.STYLE,currVal_12).toString()));
      this._expr_12 = currVal_12;
    }
    const currVal_13:any = (this.context.effectDuration + 'ms');
    if (import8.checkBinding(throwOnChange,this._expr_13,currVal_13)) {
      this.renderer.setElementStyle(this._el_1,'transitionDuration',((this.viewUtils.sanitizer.sanitize(import24.SecurityContext.STYLE,currVal_13) == null)? (null as any): this.viewUtils.sanitizer.sanitize(import24.SecurityContext.STYLE,currVal_13).toString()));
      this._expr_13 = currVal_13;
    }
    const currVal_14:any = this.context.easing;
    if (import8.checkBinding(throwOnChange,this._expr_14,currVal_14)) {
      this.renderer.setElementStyle(this._el_1,'transitionTimingFunction',((this.viewUtils.sanitizer.sanitize(import24.SecurityContext.STYLE,currVal_14) == null)? (null as any): this.viewUtils.sanitizer.sanitize(import24.SecurityContext.STYLE,currVal_14).toString()));
      this._expr_14 = currVal_14;
    }
  }
  destroyInternal():void {
    this._vc_3.destroyNestedViews();
  }
  createEmbeddedViewInternal(nodeIndex:number):import7.AppView<any> {
    if ((nodeIndex == 3)) { return new View_SlideMenuSub1(this.viewUtils,this,3,this._anchor_3,this._vc_3); }
    return (null as any);
  }
}
