// primeng/components/lightbox/lightbox.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from '@angular/core/src/linker/ng_module_factory';
import * as import1 from 'primeng/components/lightbox/lightbox';
import * as import2 from '@angular/common/src/common_module';
import * as import3 from '@angular/common/src/localization';
import * as import4 from '@angular/core/src/di/injector';
import * as import5 from '@angular/core/src/i18n/tokens';
import * as import6 from '@angular/core/src/change_detection/change_detection_util';
import * as import7 from '@angular/core/src/linker/view';
import * as import8 from '@angular/core/src/linker/view_utils';
import * as import9 from '@angular/core/src/render/api';
import * as import10 from '@angular/core/src/metadata/view';
import * as import11 from 'primeng/components/dom/domhandler';
import * as import12 from '@angular/core/src/linker/view_type';
import * as import13 from '@angular/core/src/change_detection/constants';
import * as import14 from '@angular/core/src/linker/component_factory';
import * as import15 from '@angular/core/src/linker/element_ref';
import * as import16 from '@angular/core/src/linker/view_container';
import * as import17 from '@angular/core/src/security';
import * as import18 from '../../../@angular/common/src/directives/ng_style.ngfactory';
import * as import19 from '../../../@angular/common/src/directives/ng_for.ngfactory';
import * as import20 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import21 from '@angular/core/src/linker/template_ref';
import * as import22 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import23 from '@angular/common/src/directives/ng_for';
import * as import24 from '@angular/common/src/directives/ng_style';
import * as import25 from '../../../@angular/common/src/directives/ng_if.ngfactory';
import * as import26 from '../../../@angular/common/src/directives/ng_class.ngfactory';
import * as import27 from '@angular/common/src/directives/ng_if';
import * as import28 from '@angular/common/src/directives/ng_class';
class LightboxModuleInjector extends import0.NgModuleInjector<import1.LightboxModule> {
  _CommonModule_0:import2.CommonModule;
  _LightboxModule_1:import1.LightboxModule;
  __NgLocalization_2:import3.NgLocaleLocalization;
  constructor(parent:import4.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_2():import3.NgLocaleLocalization {
    if ((this.__NgLocalization_2 == null)) { (this.__NgLocalization_2 = new import3.NgLocaleLocalization(this.parent.get(import5.LOCALE_ID))); }
    return this.__NgLocalization_2;
  }
  createInternal():import1.LightboxModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._LightboxModule_1 = new import1.LightboxModule();
    return this._LightboxModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import1.LightboxModule)) { return this._LightboxModule_1; }
    if ((token === import3.NgLocalization)) { return this._NgLocalization_2; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const LightboxModuleNgFactory:import0.NgModuleFactory<import1.LightboxModule> = new import0.NgModuleFactory(LightboxModuleInjector,import1.LightboxModule);
export class Wrapper_Lightbox {
  /*private*/ _eventHandler:Function;
  context:import1.Lightbox;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  /*private*/ _expr_5:any;
  constructor(p0:any,p1:any,p2:any) {
    this._changed = false;
    this.context = new import1.Lightbox(p0,p1,p2);
    this._expr_0 = import6.UNINITIALIZED;
    this._expr_1 = import6.UNINITIALIZED;
    this._expr_2 = import6.UNINITIALIZED;
    this._expr_3 = import6.UNINITIALIZED;
    this._expr_4 = import6.UNINITIALIZED;
    this._expr_5 = import6.UNINITIALIZED;
  }
  ngOnDetach(view:import7.AppView<any>,componentView:import7.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
    this.context.ngOnDestroy();
  }
  check_images(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.images = currValue;
      this._expr_0 = currValue;
    }
  }
  check_type(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.type = currValue;
      this._expr_1 = currValue;
    }
  }
  check_style(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_2,currValue))) {
      this._changed = true;
      this.context.style = currValue;
      this._expr_2 = currValue;
    }
  }
  check_styleClass(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_3,currValue))) {
      this._changed = true;
      this.context.styleClass = currValue;
      this._expr_3 = currValue;
    }
  }
  check_easing(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_4,currValue))) {
      this._changed = true;
      this.context.easing = currValue;
      this._expr_4 = currValue;
    }
  }
  check_effectDuration(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_5,currValue))) {
      this._changed = true;
      this.context.effectDuration = currValue;
      this._expr_5 = currValue;
    }
  }
  ngDoCheck(view:import7.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    return changed;
  }
  checkHost(view:import7.AppView<any>,componentView:import7.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import7.AppView<any>,_eventHandler:any):void {
    this._eventHandler = _eventHandler;
  }
}
var renderType_Lightbox_Host:import9.RenderComponentType = import8.createRenderComponentType('',0,import10.ViewEncapsulation.None,([] as any[]),{});
class View_Lightbox_Host0 extends import7.AppView<any> {
  _el_0:any;
  compView_0:import7.AppView<import1.Lightbox>;
  _DomHandler_0_3:import11.DomHandler;
  _Lightbox_0_4:Wrapper_Lightbox;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_Lightbox_Host0,renderType_Lightbox_Host,import12.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import13.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import14.ComponentRef<any> {
    this._el_0 = import8.selectOrCreateRenderHostElement(this.renderer,'p-lightbox',import8.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this.compView_0 = new View_Lightbox0(this.viewUtils,this,0,this._el_0);
    this._DomHandler_0_3 = new import11.DomHandler();
    this._Lightbox_0_4 = new Wrapper_Lightbox(new import15.ElementRef(this._el_0),this._DomHandler_0_3,this.renderer);
    this.compView_0.create(this._Lightbox_0_4.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import14.ComponentRef_<any>(0,this,this._el_0,this._Lightbox_0_4.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import11.DomHandler) && (0 === requestNodeIndex))) { return this._DomHandler_0_3; }
    if (((token === import1.Lightbox) && (0 === requestNodeIndex))) { return this._Lightbox_0_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._Lightbox_0_4.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
    if (!throwOnChange) { if ((this.numberOfChecks === 0)) { this._Lightbox_0_4.context.ngAfterViewInit(); } }
  }
  destroyInternal():void {
    this.compView_0.destroy();
    this._Lightbox_0_4.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  visitProjectableNodesInternal(nodeIndex:number,ngContentIndex:number,cb:any,ctx:any):void {
    if (((nodeIndex == 0) && (ngContentIndex == 0))) {  }
    if (((nodeIndex == 0) && (ngContentIndex == 1))) {  }
  }
}
export const LightboxNgFactory:import14.ComponentFactory<import1.Lightbox> = new import14.ComponentFactory<import1.Lightbox>('p-lightbox',View_Lightbox_Host0,import1.Lightbox);
const styles_Lightbox:any[] = ([] as any[]);
class View_Lightbox2 extends import7.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  /*private*/ _expr_4:any;
  /*private*/ _expr_5:any;
  /*private*/ _expr_6:any;
  /*private*/ _expr_7:any;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_Lightbox2,renderType_Lightbox,import12.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import13.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_4 = import6.UNINITIALIZED;
    this._expr_5 = import6.UNINITIALIZED;
    this._expr_6 = import6.UNINITIALIZED;
    this._expr_7 = import6.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import14.ComponentRef<any> {
    this._el_0 = import8.createRenderElement(this.renderer,(null as any),'a',import8.EMPTY_INLINE_ARRAY,(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    this._el_2 = import8.createRenderElement(this.renderer,this._el_0,'img',import8.EMPTY_INLINE_ARRAY,(null as any));
    this._text_3 = this.renderer.createText(this._el_0,'\n            ',(null as any));
    var disposable_0:Function = import8.subscribeToRenderElement(this,this._el_0,new import8.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3
    ]
    ),[disposable_0]);
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_4:any = this.context.$implicit.source;
    if (import8.checkBinding(throwOnChange,this._expr_4,currVal_4)) {
      this.renderer.setElementProperty(this._el_0,'href',this.viewUtils.sanitizer.sanitize(import17.SecurityContext.URL,currVal_4));
      this._expr_4 = currVal_4;
    }
    const currVal_5:any = this.context.$implicit.thumbnail;
    if (import8.checkBinding(throwOnChange,this._expr_5,currVal_5)) {
      this.renderer.setElementProperty(this._el_2,'src',this.viewUtils.sanitizer.sanitize(import17.SecurityContext.URL,currVal_5));
      this._expr_5 = currVal_5;
    }
    const currVal_6:any = this.context.$implicit.title;
    if (import8.checkBinding(throwOnChange,this._expr_6,currVal_6)) {
      this.renderer.setElementProperty(this._el_2,'title',currVal_6);
      this._expr_6 = currVal_6;
    }
    const currVal_7:any = this.context.$implicit.alt;
    if (import8.checkBinding(throwOnChange,this._expr_7,currVal_7)) {
      this.renderer.setElementProperty(this._el_2,'alt',currVal_7);
      this._expr_7 = currVal_7;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.parentView.context.onImageClick($event,this.context.$implicit,this.context.index,(<View_Lightbox0>this.parentView.parentView)._el_12)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
class View_Lightbox1 extends import7.AppView<any> {
  _el_0:any;
  _NgStyle_0_3:import18.Wrapper_NgStyle;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import16.ViewContainer;
  _TemplateRef_2_5:any;
  _NgFor_2_6:import19.Wrapper_NgFor;
  _text_3:any;
  /*private*/ _expr_8:any;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_Lightbox1,renderType_Lightbox,import12.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import13.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_8 = import6.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import14.ComponentRef<any> {
    this._el_0 = import8.createRenderElement(this.renderer,(null as any),'div',import8.EMPTY_INLINE_ARRAY,(null as any));
    this._NgStyle_0_3 = new import18.Wrapper_NgStyle(this.parentView.injectorGet(import20.KeyValueDiffers,this.parentIndex),new import15.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n            ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import16.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import21.TemplateRef_(this,2,this._anchor_2);
    this._NgFor_2_6 = new import19.Wrapper_NgFor(this._vc_2.vcRef,this._TemplateRef_2_5,this.parentView.injectorGet(import22.IterableDiffers,this.parentIndex),this.parentView.ref);
    this._text_3 = this.renderer.createText(this._el_0,'\n        ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import21.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import23.NgFor) && (2 === requestNodeIndex))) { return this._NgFor_2_6.context; }
    if (((token === import24.NgStyle) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 3)))) { return this._NgStyle_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this.parentView.context.style;
    this._NgStyle_0_3.check_ngStyle(currVal_0_0_0,throwOnChange,false);
    this._NgStyle_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_2_0_0:any = this.parentView.context.images;
    this._NgFor_2_6.check_ngForOf(currVal_2_0_0,throwOnChange,false);
    this._NgFor_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
    const currVal_8:any = this.parentView.context.styleClass;
    if (import8.checkBinding(throwOnChange,this._expr_8,currVal_8)) {
      this.renderer.setElementProperty(this._el_0,'className',currVal_8);
      this._expr_8 = currVal_8;
    }
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import7.AppView<any> {
    if ((nodeIndex == 2)) { return new View_Lightbox2(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    return (null as any);
  }
}
class View_Lightbox3 extends import7.AppView<any> {
  _el_0:any;
  _NgStyle_0_3:import18.Wrapper_NgStyle;
  _text_1:any;
  _text_2:any;
  /*private*/ _expr_4:any;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_Lightbox3,renderType_Lightbox,import12.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import13.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_4 = import6.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import14.ComponentRef<any> {
    this._el_0 = import8.createRenderElement(this.renderer,(null as any),'span',import8.EMPTY_INLINE_ARRAY,(null as any));
    this._NgStyle_0_3 = new import18.Wrapper_NgStyle(this.parentView.injectorGet(import20.KeyValueDiffers,this.parentIndex),new import15.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n            ',(null as any));
    this.projectNodes(this._el_0,0);
    this._text_2 = this.renderer.createText(this._el_0,'\n        ',(null as any));
    var disposable_0:Function = import8.subscribeToRenderElement(this,this._el_0,new import8.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._text_2
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import24.NgStyle) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 2)))) { return this._NgStyle_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this.parentView.context.style;
    this._NgStyle_0_3.check_ngStyle(currVal_0_0_0,throwOnChange,false);
    this._NgStyle_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_4:any = this.parentView.context.styleClass;
    if (import8.checkBinding(throwOnChange,this._expr_4,currVal_4)) {
      this.renderer.setElementProperty(this._el_0,'className',currVal_4);
      this._expr_4 = currVal_4;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.context.onLinkClick($event,(<View_Lightbox0>this.parentView)._el_12)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
var renderType_Lightbox:import9.RenderComponentType = import8.createRenderComponentType('',2,import10.ViewEncapsulation.None,styles_Lightbox,{});
export class View_Lightbox0 extends import7.AppView<import1.Lightbox> {
  _text_0:any;
  _anchor_1:any;
  /*private*/ _vc_1:import16.ViewContainer;
  _TemplateRef_1_5:any;
  _NgIf_1_6:import25.Wrapper_NgIf;
  _text_2:any;
  _anchor_3:any;
  /*private*/ _vc_3:import16.ViewContainer;
  _TemplateRef_3_5:any;
  _NgIf_3_6:import25.Wrapper_NgIf;
  _text_4:any;
  _el_5:any;
  _text_6:any;
  _el_7:any;
  _text_8:any;
  _el_9:any;
  _NgClass_9_3:import26.Wrapper_NgClass;
  _el_10:any;
  _text_11:any;
  _el_12:any;
  _NgClass_12_3:import26.Wrapper_NgClass;
  _text_13:any;
  _el_14:any;
  _text_15:any;
  _text_16:any;
  _text_17:any;
  _el_18:any;
  _NgClass_18_3:import26.Wrapper_NgClass;
  _el_19:any;
  _text_20:any;
  _text_21:any;
  _el_22:any;
  _text_23:any;
  _el_24:any;
  _text_25:any;
  _el_26:any;
  _el_27:any;
  _text_28:any;
  _el_29:any;
  _text_30:any;
  _text_31:any;
  _text_32:any;
  /*private*/ _expr_42:any;
  /*private*/ _expr_43:any;
  /*private*/ _expr_44:any;
  /*private*/ _expr_45:any;
  /*private*/ _expr_46:any;
  /*private*/ _expr_47:any;
  _map_48:any;
  /*private*/ _expr_49:any;
  /*private*/ _expr_50:any;
  /*private*/ _expr_51:any;
  _map_52:any;
  /*private*/ _expr_53:any;
  /*private*/ _expr_54:any;
  _map_55:any;
  /*private*/ _expr_56:any;
  /*private*/ _expr_57:any;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_Lightbox0,renderType_Lightbox,import12.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import13.ChangeDetectorStatus.CheckAlways);
    this._expr_42 = import6.UNINITIALIZED;
    this._expr_43 = import6.UNINITIALIZED;
    this._expr_44 = import6.UNINITIALIZED;
    this._expr_45 = import6.UNINITIALIZED;
    this._expr_46 = import6.UNINITIALIZED;
    this._expr_47 = import6.UNINITIALIZED;
    this._map_48 = import8.pureProxy1((p0:any):{[key: string]:any} => {
      return {'ui-helper-hidden': p0};
    });
    this._expr_49 = import6.UNINITIALIZED;
    this._expr_50 = import6.UNINITIALIZED;
    this._expr_51 = import6.UNINITIALIZED;
    this._map_52 = import8.pureProxy1((p0:any):{[key: string]:any} => {
      return {'ui-lightbox-loading': p0};
    });
    this._expr_53 = import6.UNINITIALIZED;
    this._expr_54 = import6.UNINITIALIZED;
    this._map_55 = import8.pureProxy1((p0:any):{[key: string]:any} => {
      return {'ui-helper-hidden': p0};
    });
    this._expr_56 = import6.UNINITIALIZED;
    this._expr_57 = import6.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import14.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n        ',(null as any));
    this._anchor_1 = this.renderer.createTemplateAnchor(parentRenderNode,(null as any));
    this._vc_1 = new import16.ViewContainer(1,(null as any),this,this._anchor_1);
    this._TemplateRef_1_5 = new import21.TemplateRef_(this,1,this._anchor_1);
    this._NgIf_1_6 = new import25.Wrapper_NgIf(this._vc_1.vcRef,this._TemplateRef_1_5);
    this._text_2 = this.renderer.createText(parentRenderNode,'\n        ',(null as any));
    this._anchor_3 = this.renderer.createTemplateAnchor(parentRenderNode,(null as any));
    this._vc_3 = new import16.ViewContainer(3,(null as any),this,this._anchor_3);
    this._TemplateRef_3_5 = new import21.TemplateRef_(this,3,this._anchor_3);
    this._NgIf_3_6 = new import25.Wrapper_NgIf(this._vc_3.vcRef,this._TemplateRef_3_5);
    this._text_4 = this.renderer.createText(parentRenderNode,'\n        ',(null as any));
    this._el_5 = import8.createRenderElement(this.renderer,parentRenderNode,'div',new import8.InlineArray2(2,'class','ui-lightbox ui-widget ui-helper-hidden ui-corner-all ui-shadow'),(null as any));
    this._text_6 = this.renderer.createText(this._el_5,'\n           ',(null as any));
    this._el_7 = import8.createRenderElement(this.renderer,this._el_5,'div',new import8.InlineArray2(2,'class','ui-lightbox-content-wrapper'),(null as any));
    this._text_8 = this.renderer.createText(this._el_7,'\n              ',(null as any));
    this._el_9 = import8.createRenderElement(this.renderer,this._el_7,'a',new import8.InlineArray2(2,'class','ui-state-default ui-lightbox-nav-left ui-corner-right'),(null as any));
    this._NgClass_9_3 = new import26.Wrapper_NgClass(this.parentView.injectorGet(import22.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import20.KeyValueDiffers,this.parentIndex),new import15.ElementRef(this._el_9),this.renderer);
    this._el_10 = import8.createRenderElement(this.renderer,this._el_9,'span',new import8.InlineArray2(2,'class','fa fa-fw fa-caret-left'),(null as any));
    this._text_11 = this.renderer.createText(this._el_7,'\n              ',(null as any));
    this._el_12 = import8.createRenderElement(this.renderer,this._el_7,'div',new import8.InlineArray2(2,'class','ui-lightbox-content ui-corner-all'),(null as any));
    this._NgClass_12_3 = new import26.Wrapper_NgClass(this.parentView.injectorGet(import22.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import20.KeyValueDiffers,this.parentIndex),new import15.ElementRef(this._el_12),this.renderer);
    this._text_13 = this.renderer.createText(this._el_12,'\n                ',(null as any));
    this._el_14 = import8.createRenderElement(this.renderer,this._el_12,'img',new import8.InlineArray2(2,'style','display:none'),(null as any));
    this._text_15 = this.renderer.createText(this._el_12,'\n                ',(null as any));
    this.projectNodes(this._el_12,1);
    this._text_16 = this.renderer.createText(this._el_12,'\n              ',(null as any));
    this._text_17 = this.renderer.createText(this._el_7,'\n              ',(null as any));
    this._el_18 = import8.createRenderElement(this.renderer,this._el_7,'a',new import8.InlineArray2(2,'class','ui-state-default ui-lightbox-nav-right ui-corner-left ui-helper-hidden'),(null as any));
    this._NgClass_18_3 = new import26.Wrapper_NgClass(this.parentView.injectorGet(import22.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import20.KeyValueDiffers,this.parentIndex),new import15.ElementRef(this._el_18),this.renderer);
    this._el_19 = import8.createRenderElement(this.renderer,this._el_18,'span',new import8.InlineArray2(2,'class','fa fa-fw fa-caret-right'),(null as any));
    this._text_20 = this.renderer.createText(this._el_7,'\n           ',(null as any));
    this._text_21 = this.renderer.createText(this._el_5,'\n           ',(null as any));
    this._el_22 = import8.createRenderElement(this.renderer,this._el_5,'div',new import8.InlineArray2(2,'class','ui-lightbox-caption ui-widget-header'),(null as any));
    this._text_23 = this.renderer.createText(this._el_22,'\n              ',(null as any));
    this._el_24 = import8.createRenderElement(this.renderer,this._el_22,'span',new import8.InlineArray2(2,'class','ui-lightbox-caption-text'),(null as any));
    this._text_25 = this.renderer.createText(this._el_24,'',(null as any));
    this._el_26 = import8.createRenderElement(this.renderer,this._el_22,'a',new import8.InlineArray4(4,'class','ui-lightbox-close ui-corner-all','href','#'),(null as any));
    this._el_27 = import8.createRenderElement(this.renderer,this._el_26,'span',new import8.InlineArray2(2,'class','fa fa-fw fa-close'),(null as any));
    this._text_28 = this.renderer.createText(this._el_22,'\n              ',(null as any));
    this._el_29 = import8.createRenderElement(this.renderer,this._el_22,'div',new import8.InlineArray2(2,'style','clear:both'),(null as any));
    this._text_30 = this.renderer.createText(this._el_22,'\n           ',(null as any));
    this._text_31 = this.renderer.createText(this._el_5,'\n        ',(null as any));
    this._text_32 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    var disposable_0:Function = import8.subscribeToRenderElement(this,this._el_5,new import8.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_5));
    var disposable_1:Function = import8.subscribeToRenderElement(this,this._el_9,new import8.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_9));
    var disposable_2:Function = import8.subscribeToRenderElement(this,this._el_14,new import8.InlineArray2(2,'load',(null as any)),this.eventHandler(this.handleEvent_14));
    var disposable_3:Function = import8.subscribeToRenderElement(this,this._el_18,new import8.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_18));
    var disposable_4:Function = import8.subscribeToRenderElement(this,this._el_26,new import8.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_26));
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._anchor_1,
      this._text_2,
      this._anchor_3,
      this._text_4,
      this._el_5,
      this._text_6,
      this._el_7,
      this._text_8,
      this._el_9,
      this._el_10,
      this._text_11,
      this._el_12,
      this._text_13,
      this._el_14,
      this._text_15,
      this._text_16,
      this._text_17,
      this._el_18,
      this._el_19,
      this._text_20,
      this._text_21,
      this._el_22,
      this._text_23,
      this._el_24,
      this._text_25,
      this._el_26,
      this._el_27,
      this._text_28,
      this._el_29,
      this._text_30,
      this._text_31,
      this._text_32
    ]
    ),[
      disposable_0,
      disposable_1,
      disposable_2,
      disposable_3,
      disposable_4
    ]
    );
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import21.TemplateRef) && (1 === requestNodeIndex))) { return this._TemplateRef_1_5; }
    if (((token === import27.NgIf) && (1 === requestNodeIndex))) { return this._NgIf_1_6.context; }
    if (((token === import21.TemplateRef) && (3 === requestNodeIndex))) { return this._TemplateRef_3_5; }
    if (((token === import27.NgIf) && (3 === requestNodeIndex))) { return this._NgIf_3_6.context; }
    if (((token === import28.NgClass) && ((9 <= requestNodeIndex) && (requestNodeIndex <= 10)))) { return this._NgClass_9_3.context; }
    if (((token === import28.NgClass) && ((12 <= requestNodeIndex) && (requestNodeIndex <= 16)))) { return this._NgClass_12_3.context; }
    if (((token === import28.NgClass) && ((18 <= requestNodeIndex) && (requestNodeIndex <= 19)))) { return this._NgClass_18_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1_0_0:any = (this.context.type == 'image');
    this._NgIf_1_6.check_ngIf(currVal_1_0_0,throwOnChange,false);
    this._NgIf_1_6.ngDoCheck(this,this._anchor_1,throwOnChange);
    const currVal_3_0_0:any = (this.context.type == 'content');
    this._NgIf_3_6.check_ngIf(currVal_3_0_0,throwOnChange,false);
    this._NgIf_3_6.ngDoCheck(this,this._anchor_3,throwOnChange);
    const currVal_9_0_0:any = 'ui-state-default ui-lightbox-nav-left ui-corner-right';
    this._NgClass_9_3.check_klass(currVal_9_0_0,throwOnChange,false);
    const currVal_9_0_1:any = this._map_48(!this.context.leftVisible);
    this._NgClass_9_3.check_ngClass(currVal_9_0_1,throwOnChange,false);
    this._NgClass_9_3.ngDoCheck(this,this._el_9,throwOnChange);
    const currVal_12_0_0:any = 'ui-lightbox-content ui-corner-all';
    this._NgClass_12_3.check_klass(currVal_12_0_0,throwOnChange,false);
    const currVal_12_0_1:any = this._map_52(this.context.loading);
    this._NgClass_12_3.check_ngClass(currVal_12_0_1,throwOnChange,false);
    this._NgClass_12_3.ngDoCheck(this,this._el_12,throwOnChange);
    const currVal_18_0_0:any = 'ui-state-default ui-lightbox-nav-right ui-corner-left ui-helper-hidden';
    this._NgClass_18_3.check_klass(currVal_18_0_0,throwOnChange,false);
    const currVal_18_0_1:any = this._map_55(!this.context.rightVisible);
    this._NgClass_18_3.check_ngClass(currVal_18_0_1,throwOnChange,false);
    this._NgClass_18_3.ngDoCheck(this,this._el_18,throwOnChange);
    this._vc_1.detectChangesInNestedViews(throwOnChange);
    this._vc_3.detectChangesInNestedViews(throwOnChange);
    const currVal_42:any = (this.context.visible? 'block': 'none');
    if (import8.checkBinding(throwOnChange,this._expr_42,currVal_42)) {
      this.renderer.setElementStyle(this._el_5,'display',((this.viewUtils.sanitizer.sanitize(import17.SecurityContext.STYLE,currVal_42) == null)? (null as any): this.viewUtils.sanitizer.sanitize(import17.SecurityContext.STYLE,currVal_42).toString()));
      this._expr_42 = currVal_42;
    }
    const currVal_43:any = this.context.zindex;
    if (import8.checkBinding(throwOnChange,this._expr_43,currVal_43)) {
      this.renderer.setElementStyle(this._el_5,'zIndex',((this.viewUtils.sanitizer.sanitize(import17.SecurityContext.STYLE,currVal_43) == null)? (null as any): this.viewUtils.sanitizer.sanitize(import17.SecurityContext.STYLE,currVal_43).toString()));
      this._expr_43 = currVal_43;
    }
    const currVal_44:any = 'all';
    if (import8.checkBinding(throwOnChange,this._expr_44,currVal_44)) {
      this.renderer.setElementStyle(this._el_5,'transitionProperty',((this.viewUtils.sanitizer.sanitize(import17.SecurityContext.STYLE,currVal_44) == null)? (null as any): this.viewUtils.sanitizer.sanitize(import17.SecurityContext.STYLE,currVal_44).toString()));
      this._expr_44 = currVal_44;
    }
    const currVal_45:any = this.context.effectDuration;
    if (import8.checkBinding(throwOnChange,this._expr_45,currVal_45)) {
      this.renderer.setElementStyle(this._el_5,'transitionDuration',((this.viewUtils.sanitizer.sanitize(import17.SecurityContext.STYLE,currVal_45) == null)? (null as any): this.viewUtils.sanitizer.sanitize(import17.SecurityContext.STYLE,currVal_45).toString()));
      this._expr_45 = currVal_45;
    }
    const currVal_46:any = this.context.easing;
    if (import8.checkBinding(throwOnChange,this._expr_46,currVal_46)) {
      this.renderer.setElementStyle(this._el_5,'transitionTimingFunction',((this.viewUtils.sanitizer.sanitize(import17.SecurityContext.STYLE,currVal_46) == null)? (null as any): this.viewUtils.sanitizer.sanitize(import17.SecurityContext.STYLE,currVal_46).toString()));
      this._expr_46 = currVal_46;
    }
    const currVal_47:any = (this.context.zindex + 1);
    if (import8.checkBinding(throwOnChange,this._expr_47,currVal_47)) {
      this.renderer.setElementStyle(this._el_9,'zIndex',((this.viewUtils.sanitizer.sanitize(import17.SecurityContext.STYLE,currVal_47) == null)? (null as any): this.viewUtils.sanitizer.sanitize(import17.SecurityContext.STYLE,currVal_47).toString()));
      this._expr_47 = currVal_47;
    }
    const currVal_49:any = 'width,height';
    if (import8.checkBinding(throwOnChange,this._expr_49,currVal_49)) {
      this.renderer.setElementStyle(this._el_12,'transitionProperty',((this.viewUtils.sanitizer.sanitize(import17.SecurityContext.STYLE,currVal_49) == null)? (null as any): this.viewUtils.sanitizer.sanitize(import17.SecurityContext.STYLE,currVal_49).toString()));
      this._expr_49 = currVal_49;
    }
    const currVal_50:any = this.context.effectDuration;
    if (import8.checkBinding(throwOnChange,this._expr_50,currVal_50)) {
      this.renderer.setElementStyle(this._el_12,'transitionDuration',((this.viewUtils.sanitizer.sanitize(import17.SecurityContext.STYLE,currVal_50) == null)? (null as any): this.viewUtils.sanitizer.sanitize(import17.SecurityContext.STYLE,currVal_50).toString()));
      this._expr_50 = currVal_50;
    }
    const currVal_51:any = this.context.easing;
    if (import8.checkBinding(throwOnChange,this._expr_51,currVal_51)) {
      this.renderer.setElementStyle(this._el_12,'transitionTimingFunction',((this.viewUtils.sanitizer.sanitize(import17.SecurityContext.STYLE,currVal_51) == null)? (null as any): this.viewUtils.sanitizer.sanitize(import17.SecurityContext.STYLE,currVal_51).toString()));
      this._expr_51 = currVal_51;
    }
    const currVal_53:any = (this.context.currentImage? (this.context.currentImage.source || ''): '');
    if (import8.checkBinding(throwOnChange,this._expr_53,currVal_53)) {
      this.renderer.setElementProperty(this._el_14,'src',this.viewUtils.sanitizer.sanitize(import17.SecurityContext.URL,currVal_53));
      this._expr_53 = currVal_53;
    }
    const currVal_54:any = (this.context.zindex + 1);
    if (import8.checkBinding(throwOnChange,this._expr_54,currVal_54)) {
      this.renderer.setElementStyle(this._el_18,'zIndex',((this.viewUtils.sanitizer.sanitize(import17.SecurityContext.STYLE,currVal_54) == null)? (null as any): this.viewUtils.sanitizer.sanitize(import17.SecurityContext.STYLE,currVal_54).toString()));
      this._expr_54 = currVal_54;
    }
    const currVal_56:any = (this.context.captionText? 'block': 'none');
    if (import8.checkBinding(throwOnChange,this._expr_56,currVal_56)) {
      this.renderer.setElementStyle(this._el_22,'display',((this.viewUtils.sanitizer.sanitize(import17.SecurityContext.STYLE,currVal_56) == null)? (null as any): this.viewUtils.sanitizer.sanitize(import17.SecurityContext.STYLE,currVal_56).toString()));
      this._expr_56 = currVal_56;
    }
    const currVal_57:any = import8.inlineInterpolate(1,'',this.context.captionText,'');
    if (import8.checkBinding(throwOnChange,this._expr_57,currVal_57)) {
      this.renderer.setText(this._text_25,currVal_57);
      this._expr_57 = currVal_57;
    }
  }
  destroyInternal():void {
    this._vc_1.destroyNestedViews();
    this._vc_3.destroyNestedViews();
  }
  createEmbeddedViewInternal(nodeIndex:number):import7.AppView<any> {
    if ((nodeIndex == 1)) { return new View_Lightbox1(this.viewUtils,this,1,this._anchor_1,this._vc_1); }
    if ((nodeIndex == 3)) { return new View_Lightbox3(this.viewUtils,this,3,this._anchor_3,this._vc_3); }
    return (null as any);
  }
  handleEvent_5(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>(this.context.preventDocumentClickListener = true)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
  handleEvent_9(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.context.prev(this._el_14)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
  handleEvent_14(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'load')) {
      const pd_sub_0:any = ((<any>this.context.onImageLoad($event,this._el_12)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
  handleEvent_18(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.context.next(this._el_14)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
  handleEvent_26(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.context.hide($event)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
