// primeng/components/chips/chips.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from '@angular/core/src/linker/ng_module_factory';
import * as import1 from 'primeng/components/chips/chips';
import * as import2 from '@angular/common/src/common_module';
import * as import3 from 'primeng/components/inputtext/inputtext';
import * as import4 from 'primeng/components/common/shared';
import * as import5 from '@angular/common/src/localization';
import * as import6 from '@angular/core/src/di/injector';
import * as import7 from '@angular/core/src/i18n/tokens';
import * as import8 from '@angular/core/src/change_detection/change_detection_util';
import * as import9 from '@angular/core/src/linker/view';
import * as import10 from '@angular/core/src/linker/view_utils';
import * as import11 from '@angular/core/src/render/api';
import * as import12 from '@angular/core/src/metadata/view';
import * as import13 from 'primeng/components/dom/domhandler';
import * as import14 from '@angular/core/src/linker/query_list';
import * as import15 from '@angular/core/src/linker/view_type';
import * as import16 from '@angular/core/src/change_detection/constants';
import * as import17 from '@angular/core/src/linker/component_factory';
import * as import18 from '@angular/core/src/linker/element_ref';
import * as import19 from '@angular/forms/src/directives/control_value_accessor';
import * as import20 from '@angular/core/src/linker/view_container';
import * as import21 from '../common/shared.ngfactory';
import * as import22 from '@angular/core/src/linker/template_ref';
import * as import23 from '../../../@angular/common/src/directives/ng_if.ngfactory';
import * as import24 from '@angular/common/src/directives/ng_if';
import * as import25 from '../../../@angular/common/src/directives/ng_class.ngfactory';
import * as import26 from '../../../@angular/common/src/directives/ng_style.ngfactory';
import * as import27 from '../../../@angular/common/src/directives/ng_for.ngfactory';
import * as import28 from '../inputtext/inputtext.ngfactory';
import * as import29 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import30 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import31 from '@angular/common/src/directives/ng_for';
import * as import32 from '@angular/common/src/directives/ng_class';
import * as import33 from '@angular/common/src/directives/ng_style';
class ChipsModuleInjector extends import0.NgModuleInjector<import1.ChipsModule> {
  _CommonModule_0:import2.CommonModule;
  _InputTextModule_1:import3.InputTextModule;
  _SharedModule_2:import4.SharedModule;
  _ChipsModule_3:import1.ChipsModule;
  __NgLocalization_4:import5.NgLocaleLocalization;
  constructor(parent:import6.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_4():import5.NgLocaleLocalization {
    if ((this.__NgLocalization_4 == null)) { (this.__NgLocalization_4 = new import5.NgLocaleLocalization(this.parent.get(import7.LOCALE_ID))); }
    return this.__NgLocalization_4;
  }
  createInternal():import1.ChipsModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._InputTextModule_1 = new import3.InputTextModule();
    this._SharedModule_2 = new import4.SharedModule();
    this._ChipsModule_3 = new import1.ChipsModule();
    return this._ChipsModule_3;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.InputTextModule)) { return this._InputTextModule_1; }
    if ((token === import4.SharedModule)) { return this._SharedModule_2; }
    if ((token === import1.ChipsModule)) { return this._ChipsModule_3; }
    if ((token === import5.NgLocalization)) { return this._NgLocalization_4; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const ChipsModuleNgFactory:import0.NgModuleFactory<import1.ChipsModule> = new import0.NgModuleFactory(ChipsModuleInjector,import1.ChipsModule);
export class Wrapper_Chips {
  /*private*/ _eventHandler:Function;
  context:import1.Chips;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  /*private*/ _expr_5:any;
  subscription0:any;
  subscription1:any;
  constructor(p0:any,p1:any) {
    this._changed = false;
    this.context = new import1.Chips(p0,p1);
    this._expr_0 = import8.UNINITIALIZED;
    this._expr_1 = import8.UNINITIALIZED;
    this._expr_2 = import8.UNINITIALIZED;
    this._expr_3 = import8.UNINITIALIZED;
    this._expr_4 = import8.UNINITIALIZED;
    this._expr_5 = import8.UNINITIALIZED;
  }
  ngOnDetach(view:import9.AppView<any>,componentView:import9.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
    (this.subscription0 && this.subscription0.unsubscribe());
    (this.subscription1 && this.subscription1.unsubscribe());
  }
  check_style(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.style = currValue;
      this._expr_0 = currValue;
    }
  }
  check_styleClass(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.styleClass = currValue;
      this._expr_1 = currValue;
    }
  }
  check_disabled(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_2,currValue))) {
      this._changed = true;
      this.context.disabled = currValue;
      this._expr_2 = currValue;
    }
  }
  check_field(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_3,currValue))) {
      this._changed = true;
      this.context.field = currValue;
      this._expr_3 = currValue;
    }
  }
  check_placeholder(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_4,currValue))) {
      this._changed = true;
      this.context.placeholder = currValue;
      this._expr_4 = currValue;
    }
  }
  check_max(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_5,currValue))) {
      this._changed = true;
      this.context.max = currValue;
      this._expr_5 = currValue;
    }
  }
  ngDoCheck(view:import9.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    return changed;
  }
  checkHost(view:import9.AppView<any>,componentView:import9.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import9.AppView<any>,_eventHandler:any,emit0:boolean,emit1:boolean):void {
    this._eventHandler = _eventHandler;
    if (emit0) { (this.subscription0 = this.context.onAdd.subscribe(_eventHandler.bind(view,'onAdd'))); }
    if (emit1) { (this.subscription1 = this.context.onRemove.subscribe(_eventHandler.bind(view,'onRemove'))); }
  }
}
var renderType_Chips_Host:import11.RenderComponentType = import10.createRenderComponentType('',0,import12.ViewEncapsulation.None,([] as any[]),{});
class View_Chips_Host0 extends import9.AppView<any> {
  _el_0:any;
  compView_0:import9.AppView<import1.Chips>;
  _DomHandler_0_3:import13.DomHandler;
  _Chips_0_4:Wrapper_Chips;
  __NG_VALUE_ACCESSOR_0_5:any[];
  _query_TemplateRef_0_0:import14.QueryList<any>;
  constructor(viewUtils:import10.ViewUtils,parentView:import9.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_Chips_Host0,renderType_Chips_Host,import15.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import16.ChangeDetectorStatus.CheckAlways);
  }
  get _NG_VALUE_ACCESSOR_0_5():any[] {
    if ((this.__NG_VALUE_ACCESSOR_0_5 == null)) { (this.__NG_VALUE_ACCESSOR_0_5 = [this._Chips_0_4.context]); }
    return this.__NG_VALUE_ACCESSOR_0_5;
  }
  createInternal(rootSelector:string):import17.ComponentRef<any> {
    this._el_0 = import10.selectOrCreateRenderHostElement(this.renderer,'p-chips',import10.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this.compView_0 = new View_Chips0(this.viewUtils,this,0,this._el_0);
    this._DomHandler_0_3 = new import13.DomHandler();
    this._Chips_0_4 = new Wrapper_Chips(new import18.ElementRef(this._el_0),this._DomHandler_0_3);
    this._query_TemplateRef_0_0 = new import14.QueryList<any>();
    this.compView_0.create(this._Chips_0_4.context);
    this._query_TemplateRef_0_0.reset(([] as any[]));
    this._Chips_0_4.context.itemTemplate = this._query_TemplateRef_0_0.first;
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import17.ComponentRef_<any>(0,this,this._el_0,this._Chips_0_4.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import13.DomHandler) && (0 === requestNodeIndex))) { return this._DomHandler_0_3; }
    if (((token === import1.Chips) && (0 === requestNodeIndex))) { return this._Chips_0_4.context; }
    if (((token === import19.NG_VALUE_ACCESSOR) && (0 === requestNodeIndex))) { return this._NG_VALUE_ACCESSOR_0_5; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._Chips_0_4.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
    this._Chips_0_4.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
export const ChipsNgFactory:import17.ComponentFactory<import1.Chips> = new import17.ComponentFactory<import1.Chips>('p-chips',View_Chips_Host0,import1.Chips);
const styles_Chips:any[] = ([] as any[]);
class View_Chips2 extends import9.AppView<any> {
  _el_0:any;
  constructor(viewUtils:import10.ViewUtils,parentView:import9.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import20.ViewContainer) {
    super(View_Chips2,renderType_Chips,import15.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import16.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import17.ComponentRef<any> {
    this._el_0 = import10.createRenderElement(this.renderer,(null as any),'span',new import10.InlineArray2(2,'class','ui-chips-token-icon fa fa-fw fa-close'),(null as any));
    var disposable_0:Function = import10.subscribeToRenderElement(this,this._el_0,new import10.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),[disposable_0]);
    return (null as any);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.parentView.context.removeItem($event,this.parentView.context.index)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
class View_Chips3 extends import9.AppView<any> {
  _el_0:any;
  _text_1:any;
  /*private*/ _expr_2:any;
  constructor(viewUtils:import10.ViewUtils,parentView:import9.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import20.ViewContainer) {
    super(View_Chips3,renderType_Chips,import15.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import16.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_2 = import8.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import17.ComponentRef<any> {
    this._el_0 = import10.createRenderElement(this.renderer,(null as any),'span',new import10.InlineArray2(2,'class','ui-chips-token-label'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1
    ]
    ),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2:any = import10.inlineInterpolate(1,'',(this.parentView.parentView.context.field? this.parentView.parentView.context.resolveFieldData(this.parentView.context.$implicit,this.parentView.parentView.context.field): this.parentView.context.$implicit),'');
    if (import10.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setText(this._text_1,currVal_2);
      this._expr_2 = currVal_2;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_Chips5 extends import9.AppView<any> {
  _el_0:any;
  constructor(viewUtils:import10.ViewUtils,parentView:import9.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import20.ViewContainer) {
    super(View_Chips5,renderType_Chips,import15.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import16.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import17.ComponentRef<any> {
    this._el_0 = this.renderer.createTemplateAnchor((null as any),(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): ([] as any[])),(null as any));
    return (null as any);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_Chips4 extends import9.AppView<any> {
  _anchor_0:any;
  /*private*/ _vc_0:import20.ViewContainer;
  _TemplateRef_0_5:any;
  _TemplateWrapper_0_6:import21.Wrapper_TemplateWrapper;
  _el_1:any;
  constructor(viewUtils:import10.ViewUtils,parentView:import9.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import20.ViewContainer) {
    super(View_Chips4,renderType_Chips,import15.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import16.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import17.ComponentRef<any> {
    this._anchor_0 = this.renderer.createTemplateAnchor((null as any),(null as any));
    this._vc_0 = new import20.ViewContainer(0,(null as any),this,this._anchor_0);
    this._TemplateRef_0_5 = new import22.TemplateRef_(this,0,this._anchor_0);
    this._TemplateWrapper_0_6 = new import21.Wrapper_TemplateWrapper(this._vc_0.vcRef);
    this._el_1 = this.renderer.createTemplateAnchor((null as any),(null as any));
    this.init(this._el_1,((<any>this.renderer).directRenderer? (null as any): [this._anchor_0]),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import22.TemplateRef) && (0 === requestNodeIndex))) { return this._TemplateRef_0_5; }
    if (((token === import4.TemplateWrapper) && (0 === requestNodeIndex))) { return this._TemplateWrapper_0_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this.parentView.context.$implicit;
    this._TemplateWrapper_0_6.check_item(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = this.parentView.parentView.context.itemTemplate;
    this._TemplateWrapper_0_6.check_templateRef(currVal_0_0_1,throwOnChange,false);
    this._TemplateWrapper_0_6.ngDoCheck(this,this._anchor_0,throwOnChange);
    this._vc_0.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_0.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._vc_0.nativeElement,ctx);
    this._vc_0.visitNestedViewRootNodes(cb,ctx);
    cb(this._el_1,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import9.AppView<any> {
    if ((nodeIndex == 0)) { return new View_Chips5(this.viewUtils,this,0,this._anchor_0,this._vc_0); }
    return (null as any);
  }
}
class View_Chips1 extends import9.AppView<any> {
  _el_0:any;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import20.ViewContainer;
  _TemplateRef_2_5:any;
  _NgIf_2_6:import23.Wrapper_NgIf;
  _text_3:any;
  _anchor_4:any;
  /*private*/ _vc_4:import20.ViewContainer;
  _TemplateRef_4_5:any;
  _NgIf_4_6:import23.Wrapper_NgIf;
  _text_5:any;
  _anchor_6:any;
  /*private*/ _vc_6:import20.ViewContainer;
  _TemplateRef_6_5:any;
  _NgIf_6_6:import23.Wrapper_NgIf;
  _text_7:any;
  constructor(viewUtils:import10.ViewUtils,parentView:import9.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import20.ViewContainer) {
    super(View_Chips1,renderType_Chips,import15.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import16.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import17.ComponentRef<any> {
    this._el_0 = import10.createRenderElement(this.renderer,(null as any),'li',new import10.InlineArray2(2,'class','ui-chips-token ui-state-highlight ui-corner-all'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import20.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import22.TemplateRef_(this,2,this._anchor_2);
    this._NgIf_2_6 = new import23.Wrapper_NgIf(this._vc_2.vcRef,this._TemplateRef_2_5);
    this._text_3 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this._anchor_4 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_4 = new import20.ViewContainer(4,0,this,this._anchor_4);
    this._TemplateRef_4_5 = new import22.TemplateRef_(this,4,this._anchor_4);
    this._NgIf_4_6 = new import23.Wrapper_NgIf(this._vc_4.vcRef,this._TemplateRef_4_5);
    this._text_5 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this._anchor_6 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_6 = new import20.ViewContainer(6,0,this,this._anchor_6);
    this._TemplateRef_6_5 = new import22.TemplateRef_(this,6,this._anchor_6);
    this._NgIf_6_6 = new import23.Wrapper_NgIf(this._vc_6.vcRef,this._TemplateRef_6_5);
    this._text_7 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3,
      this._anchor_4,
      this._text_5,
      this._anchor_6,
      this._text_7
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import22.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import24.NgIf) && (2 === requestNodeIndex))) { return this._NgIf_2_6.context; }
    if (((token === import22.TemplateRef) && (4 === requestNodeIndex))) { return this._TemplateRef_4_5; }
    if (((token === import24.NgIf) && (4 === requestNodeIndex))) { return this._NgIf_4_6.context; }
    if (((token === import22.TemplateRef) && (6 === requestNodeIndex))) { return this._TemplateRef_6_5; }
    if (((token === import24.NgIf) && (6 === requestNodeIndex))) { return this._NgIf_6_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2_0_0:boolean = (!this.parentView.context.itemTemplate && !this.parentView.context.disabled);
    this._NgIf_2_6.check_ngIf(currVal_2_0_0,throwOnChange,false);
    this._NgIf_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    const currVal_4_0_0:boolean = !this.parentView.context.itemTemplate;
    this._NgIf_4_6.check_ngIf(currVal_4_0_0,throwOnChange,false);
    this._NgIf_4_6.ngDoCheck(this,this._anchor_4,throwOnChange);
    const currVal_6_0_0:any = this.parentView.context.itemTemplate;
    this._NgIf_6_6.check_ngIf(currVal_6_0_0,throwOnChange,false);
    this._NgIf_6_6.ngDoCheck(this,this._anchor_6,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
    this._vc_4.detectChangesInNestedViews(throwOnChange);
    this._vc_6.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
    this._vc_4.destroyNestedViews();
    this._vc_6.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import9.AppView<any> {
    if ((nodeIndex == 2)) { return new View_Chips2(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    if ((nodeIndex == 4)) { return new View_Chips3(this.viewUtils,this,4,this._anchor_4,this._vc_4); }
    if ((nodeIndex == 6)) { return new View_Chips4(this.viewUtils,this,6,this._anchor_6,this._vc_6); }
    return (null as any);
  }
}
var renderType_Chips:import11.RenderComponentType = import10.createRenderComponentType('',0,import12.ViewEncapsulation.None,styles_Chips,{});
export class View_Chips0 extends import9.AppView<import1.Chips> {
  _text_0:any;
  _el_1:any;
  _NgClass_1_3:import25.Wrapper_NgClass;
  _NgStyle_1_4:import26.Wrapper_NgStyle;
  _text_2:any;
  _el_3:any;
  _NgClass_3_3:import25.Wrapper_NgClass;
  _text_4:any;
  _anchor_5:any;
  /*private*/ _vc_5:import20.ViewContainer;
  _TemplateRef_5_5:any;
  _NgFor_5_6:import27.Wrapper_NgFor;
  _text_6:any;
  _el_7:any;
  _text_8:any;
  _el_9:any;
  _InputText_9_3:import28.Wrapper_InputText;
  _text_10:any;
  _text_11:any;
  _text_12:any;
  _text_13:any;
  _map_21:any;
  /*private*/ _expr_22:any;
  /*private*/ _expr_23:any;
  /*private*/ _expr_24:any;
  constructor(viewUtils:import10.ViewUtils,parentView:import9.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_Chips0,renderType_Chips,import15.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import16.ChangeDetectorStatus.CheckAlways);
    this._map_21 = import10.pureProxy3((p0:any,p1:any,p2:any):{[key: string]:any} => {
      return {
        'ui-inputtext ui-state-default ui-corner-all': p0,
        'ui-state-focus': p1,
        'ui-state-disabled': p2
      }
      ;
    });
    this._expr_22 = import8.UNINITIALIZED;
    this._expr_23 = import8.UNINITIALIZED;
    this._expr_24 = import8.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import17.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n        ',(null as any));
    this._el_1 = import10.createRenderElement(this.renderer,parentRenderNode,'div',import10.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_1_3 = new import25.Wrapper_NgClass(this.parentView.injectorGet(import29.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import30.KeyValueDiffers,this.parentIndex),new import18.ElementRef(this._el_1),this.renderer);
    this._NgStyle_1_4 = new import26.Wrapper_NgStyle(this.parentView.injectorGet(import30.KeyValueDiffers,this.parentIndex),new import18.ElementRef(this._el_1),this.renderer);
    this._text_2 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_3 = import10.createRenderElement(this.renderer,this._el_1,'ul',import10.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_3_3 = new import25.Wrapper_NgClass(this.parentView.injectorGet(import29.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import30.KeyValueDiffers,this.parentIndex),new import18.ElementRef(this._el_3),this.renderer);
    this._text_4 = this.renderer.createText(this._el_3,'\n                ',(null as any));
    this._anchor_5 = this.renderer.createTemplateAnchor(this._el_3,(null as any));
    this._vc_5 = new import20.ViewContainer(5,3,this,this._anchor_5);
    this._TemplateRef_5_5 = new import22.TemplateRef_(this,5,this._anchor_5);
    this._NgFor_5_6 = new import27.Wrapper_NgFor(this._vc_5.vcRef,this._TemplateRef_5_5,this.parentView.injectorGet(import29.IterableDiffers,this.parentIndex),this.ref);
    this._text_6 = this.renderer.createText(this._el_3,'\n                ',(null as any));
    this._el_7 = import10.createRenderElement(this.renderer,this._el_3,'li',new import10.InlineArray2(2,'class','ui-chips-input-token'),(null as any));
    this._text_8 = this.renderer.createText(this._el_7,'\n                    ',(null as any));
    this._el_9 = import10.createRenderElement(this.renderer,this._el_7,'input',new import10.InlineArray4(4,'pInputText','','type','text'),(null as any));
    this._InputText_9_3 = new import28.Wrapper_InputText(new import18.ElementRef(this._el_9));
    this._text_10 = this.renderer.createText(this._el_7,'\n                ',(null as any));
    this._text_11 = this.renderer.createText(this._el_3,'\n            ',(null as any));
    this._text_12 = this.renderer.createText(this._el_1,'\n        ',(null as any));
    this._text_13 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    var disposable_0:Function = import10.subscribeToRenderElement(this,this._el_3,new import10.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_3));
    var disposable_1:Function = import10.subscribeToRenderElement(this,this._el_9,new import10.InlineArray16(10,'keydown',(null as any),'focus',(null as any),'blur',(null as any),'mouseover',(null as any),'mouseout',(null as any)),this.eventHandler(this.handleEvent_9));
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2,
      this._el_3,
      this._text_4,
      this._anchor_5,
      this._text_6,
      this._el_7,
      this._text_8,
      this._el_9,
      this._text_10,
      this._text_11,
      this._text_12,
      this._text_13
    ]
    ),[
      disposable_0,
      disposable_1
    ]
    );
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import22.TemplateRef) && (5 === requestNodeIndex))) { return this._TemplateRef_5_5; }
    if (((token === import31.NgFor) && (5 === requestNodeIndex))) { return this._NgFor_5_6.context; }
    if (((token === import3.InputText) && (9 === requestNodeIndex))) { return this._InputText_9_3.context; }
    if (((token === import32.NgClass) && ((3 <= requestNodeIndex) && (requestNodeIndex <= 11)))) { return this._NgClass_3_3.context; }
    if (((token === import32.NgClass) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 12)))) { return this._NgClass_1_3.context; }
    if (((token === import33.NgStyle) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 12)))) { return this._NgStyle_1_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1_0_0:any = this.context.styleClass;
    this._NgClass_1_3.check_klass(currVal_1_0_0,throwOnChange,false);
    const currVal_1_0_1:any = 'ui-chips ui-widget';
    this._NgClass_1_3.check_ngClass(currVal_1_0_1,throwOnChange,false);
    this._NgClass_1_3.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_1_1_0:any = this.context.style;
    this._NgStyle_1_4.check_ngStyle(currVal_1_1_0,throwOnChange,false);
    this._NgStyle_1_4.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_3_0_0:any = this._map_21(true,this.context.focus,this.context.disabled);
    this._NgClass_3_3.check_ngClass(currVal_3_0_0,throwOnChange,false);
    this._NgClass_3_3.ngDoCheck(this,this._el_3,throwOnChange);
    const currVal_5_0_0:any = this.context.value;
    this._NgFor_5_6.check_ngForOf(currVal_5_0_0,throwOnChange,false);
    this._NgFor_5_6.ngDoCheck(this,this._anchor_5,throwOnChange);
    this._InputText_9_3.ngDoCheck(this,this._el_9,throwOnChange);
    this._vc_5.detectChangesInNestedViews(throwOnChange);
    const currVal_22:any = this.context.placeholder;
    if (import10.checkBinding(throwOnChange,this._expr_22,currVal_22)) {
      this.renderer.setElementAttribute(this._el_9,'placeholder',((currVal_22 == null)? (null as any): currVal_22.toString()));
      this._expr_22 = currVal_22;
    }
    const currVal_23:any = (this.context.maxedOut || this.context.disabled);
    if (import10.checkBinding(throwOnChange,this._expr_23,currVal_23)) {
      this.renderer.setElementProperty(this._el_9,'disabled',currVal_23);
      this._expr_23 = currVal_23;
    }
    const currVal_24:any = this.context.disabled;
    if (import10.checkBinding(throwOnChange,this._expr_24,currVal_24)) {
      this.renderer.setElementProperty(this._el_9,'disabled',currVal_24);
      this._expr_24 = currVal_24;
    }
    this._InputText_9_3.checkHost(this,this,this._el_9,throwOnChange);
  }
  destroyInternal():void {
    this._vc_5.destroyNestedViews();
  }
  createEmbeddedViewInternal(nodeIndex:number):import9.AppView<any> {
    if ((nodeIndex == 5)) { return new View_Chips1(this.viewUtils,this,5,this._anchor_5,this._vc_5); }
    return (null as any);
  }
  handleEvent_3(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this._el_9.focus()) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
  handleEvent_9(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    result = (this._InputText_9_3.handleEvent(eventName,$event) && result);
    if ((eventName == 'keydown')) {
      const pd_sub_0:any = ((<any>this.context.onKeydown($event,this._el_9)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'focus')) {
      const pd_sub_1:any = ((<any>this.context.onFocus()) !== false);
      result = (pd_sub_1 && result);
    }
    if ((eventName == 'blur')) {
      const pd_sub_2:any = ((<any>this.context.onBlur()) !== false);
      result = (pd_sub_2 && result);
    }
    return result;
  }
}
