// angular2-autosize/src/autosize.directive.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
import * as import0 from 'angular2-autosize/src/autosize.directive';
export var Wrapper_Autosize = (function () {
    function Wrapper_Autosize(p0) {
        this._changed = false;
        this.context = new import0.Autosize(p0);
    }
    Wrapper_Autosize.prototype.ngOnDetach = function (view, componentView, el) {
    };
    Wrapper_Autosize.prototype.ngOnDestroy = function () {
    };
    Wrapper_Autosize.prototype.ngDoCheck = function (view, el, throwOnChange) {
        var changed = this._changed;
        this._changed = false;
        return changed;
    };
    Wrapper_Autosize.prototype.checkHost = function (view, componentView, el, throwOnChange) {
    };
    Wrapper_Autosize.prototype.handleEvent = function (eventName, $event) {
        var result = true;
        if ((eventName == 'input')) {
            var pd_sub_0 = (this.context.onInput($event.target) !== false);
            result = (pd_sub_0 && result);
        }
        return result;
    };
    Wrapper_Autosize.prototype.subscribe = function (view, _eventHandler) {
        this._eventHandler = _eventHandler;
    };
    return Wrapper_Autosize;
}());
