// primeng/components/terminal/terminal.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from '@angular/core/src/linker/ng_module_factory';
import * as import1 from 'primeng/components/terminal/terminal';
import * as import2 from '@angular/common/src/common_module';
import * as import3 from '@angular/forms/src/directives';
import * as import4 from '@angular/forms/src/form_providers';
import * as import5 from '@angular/common/src/localization';
import * as import6 from '@angular/forms/src/directives/radio_control_value_accessor';
import * as import7 from '@angular/core/src/di/injector';
import * as import8 from '@angular/core/src/i18n/tokens';
import * as import9 from '@angular/core/src/change_detection/change_detection_util';
import * as import10 from '@angular/core/src/linker/view';
import * as import11 from '@angular/core/src/linker/view_utils';
import * as import12 from '@angular/core/src/render/api';
import * as import13 from '@angular/core/src/metadata/view';
import * as import14 from 'primeng/components/dom/domhandler';
import * as import15 from '@angular/core/src/linker/view_type';
import * as import16 from '@angular/core/src/change_detection/constants';
import * as import17 from '@angular/core/src/linker/component_factory';
import * as import18 from '@angular/core/src/linker/element_ref';
import * as import19 from '@angular/core/src/linker/view_container';
import * as import20 from '../../../@angular/common/src/directives/ng_class.ngfactory';
import * as import21 from '../../../@angular/common/src/directives/ng_style.ngfactory';
import * as import22 from '../../../@angular/common/src/directives/ng_if.ngfactory';
import * as import23 from '../../../@angular/common/src/directives/ng_for.ngfactory';
import * as import24 from '../../../@angular/forms/src/directives/default_value_accessor.ngfactory';
import * as import25 from '../../../@angular/forms/src/directives/ng_model.ngfactory';
import * as import26 from '../../../@angular/forms/src/directives/ng_control_status.ngfactory';
import * as import27 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import28 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import29 from '@angular/core/src/linker/template_ref';
import * as import30 from '@angular/common/src/directives/ng_if';
import * as import31 from '@angular/common/src/directives/ng_for';
import * as import32 from '@angular/forms/src/directives/default_value_accessor';
import * as import33 from '@angular/forms/src/directives/control_value_accessor';
import * as import34 from '@angular/forms/src/directives/ng_model';
import * as import35 from '@angular/forms/src/directives/ng_control';
import * as import36 from '@angular/forms/src/directives/ng_control_status';
import * as import37 from '@angular/common/src/directives/ng_class';
import * as import38 from '@angular/common/src/directives/ng_style';
class TerminalModuleInjector extends import0.NgModuleInjector<import1.TerminalModule> {
  _CommonModule_0:import2.CommonModule;
  _InternalFormsSharedModule_1:import3.InternalFormsSharedModule;
  _FormsModule_2:import4.FormsModule;
  _TerminalModule_3:import1.TerminalModule;
  __NgLocalization_4:import5.NgLocaleLocalization;
  __RadioControlRegistry_5:import6.RadioControlRegistry;
  constructor(parent:import7.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_4():import5.NgLocaleLocalization {
    if ((this.__NgLocalization_4 == null)) { (this.__NgLocalization_4 = new import5.NgLocaleLocalization(this.parent.get(import8.LOCALE_ID))); }
    return this.__NgLocalization_4;
  }
  get _RadioControlRegistry_5():import6.RadioControlRegistry {
    if ((this.__RadioControlRegistry_5 == null)) { (this.__RadioControlRegistry_5 = new import6.RadioControlRegistry()); }
    return this.__RadioControlRegistry_5;
  }
  createInternal():import1.TerminalModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._InternalFormsSharedModule_1 = new import3.InternalFormsSharedModule();
    this._FormsModule_2 = new import4.FormsModule();
    this._TerminalModule_3 = new import1.TerminalModule();
    return this._TerminalModule_3;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.InternalFormsSharedModule)) { return this._InternalFormsSharedModule_1; }
    if ((token === import4.FormsModule)) { return this._FormsModule_2; }
    if ((token === import1.TerminalModule)) { return this._TerminalModule_3; }
    if ((token === import5.NgLocalization)) { return this._NgLocalization_4; }
    if ((token === import6.RadioControlRegistry)) { return this._RadioControlRegistry_5; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const TerminalModuleNgFactory:import0.NgModuleFactory<import1.TerminalModule> = new import0.NgModuleFactory(TerminalModuleInjector,import1.TerminalModule);
export class Wrapper_Terminal {
  /*private*/ _eventHandler:Function;
  context:import1.Terminal;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  subscription0:any;
  subscription1:any;
  constructor(p0:any,p1:any) {
    this._changed = false;
    this.context = new import1.Terminal(p0,p1);
    this._expr_0 = import9.UNINITIALIZED;
    this._expr_1 = import9.UNINITIALIZED;
    this._expr_2 = import9.UNINITIALIZED;
    this._expr_3 = import9.UNINITIALIZED;
    this._expr_4 = import9.UNINITIALIZED;
  }
  ngOnDetach(view:import10.AppView<any>,componentView:import10.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
    (this.subscription0 && this.subscription0.unsubscribe());
    (this.subscription1 && this.subscription1.unsubscribe());
  }
  check_welcomeMessage(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import11.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.welcomeMessage = currValue;
      this._expr_0 = currValue;
    }
  }
  check_prompt(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import11.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.prompt = currValue;
      this._expr_1 = currValue;
    }
  }
  check_style(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import11.checkBinding(throwOnChange,this._expr_2,currValue))) {
      this._changed = true;
      this.context.style = currValue;
      this._expr_2 = currValue;
    }
  }
  check_styleClass(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import11.checkBinding(throwOnChange,this._expr_3,currValue))) {
      this._changed = true;
      this.context.styleClass = currValue;
      this._expr_3 = currValue;
    }
  }
  check_response(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import11.checkBinding(throwOnChange,this._expr_4,currValue))) {
      this._changed = true;
      this.context.response = currValue;
      this._expr_4 = currValue;
    }
  }
  ngDoCheck(view:import10.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    return changed;
  }
  checkHost(view:import10.AppView<any>,componentView:import10.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import10.AppView<any>,_eventHandler:any,emit0:boolean,emit1:boolean):void {
    this._eventHandler = _eventHandler;
    if (emit0) { (this.subscription0 = this.context.responseChange.subscribe(_eventHandler.bind(view,'responseChange'))); }
    if (emit1) { (this.subscription1 = this.context.handler.subscribe(_eventHandler.bind(view,'handler'))); }
  }
}
var renderType_Terminal_Host:import12.RenderComponentType = import11.createRenderComponentType('',0,import13.ViewEncapsulation.None,([] as any[]),{});
class View_Terminal_Host0 extends import10.AppView<any> {
  _el_0:any;
  compView_0:import10.AppView<import1.Terminal>;
  _DomHandler_0_3:import14.DomHandler;
  _Terminal_0_4:Wrapper_Terminal;
  constructor(viewUtils:import11.ViewUtils,parentView:import10.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_Terminal_Host0,renderType_Terminal_Host,import15.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import16.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import17.ComponentRef<any> {
    this._el_0 = import11.selectOrCreateRenderHostElement(this.renderer,'p-terminal',import11.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this.compView_0 = new View_Terminal0(this.viewUtils,this,0,this._el_0);
    this._DomHandler_0_3 = new import14.DomHandler();
    this._Terminal_0_4 = new Wrapper_Terminal(new import18.ElementRef(this._el_0),this._DomHandler_0_3);
    this.compView_0.create(this._Terminal_0_4.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import17.ComponentRef_<any>(0,this,this._el_0,this._Terminal_0_4.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import14.DomHandler) && (0 === requestNodeIndex))) { return this._DomHandler_0_3; }
    if (((token === import1.Terminal) && (0 === requestNodeIndex))) { return this._Terminal_0_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._Terminal_0_4.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
    if (!throwOnChange) {
      if ((this.numberOfChecks === 0)) { this._Terminal_0_4.context.ngAfterViewInit(); }
      this._Terminal_0_4.context.ngAfterViewChecked();
    }
  }
  destroyInternal():void {
    this.compView_0.destroy();
    this._Terminal_0_4.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
export const TerminalNgFactory:import17.ComponentFactory<import1.Terminal> = new import17.ComponentFactory<import1.Terminal>('p-terminal',View_Terminal_Host0,import1.Terminal);
const styles_Terminal:any[] = ([] as any[]);
class View_Terminal1 extends import10.AppView<any> {
  _el_0:any;
  _text_1:any;
  /*private*/ _expr_2:any;
  constructor(viewUtils:import11.ViewUtils,parentView:import10.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import19.ViewContainer) {
    super(View_Terminal1,renderType_Terminal,import15.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import16.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_2 = import9.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import17.ComponentRef<any> {
    this._el_0 = import11.createRenderElement(this.renderer,(null as any),'div',import11.EMPTY_INLINE_ARRAY,(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1
    ]
    ),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2:any = import11.inlineInterpolate(1,'',this.parentView.context.welcomeMessage,'');
    if (import11.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setText(this._text_1,currVal_2);
      this._expr_2 = currVal_2;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_Terminal2 extends import10.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _text_4:any;
  _el_5:any;
  _text_6:any;
  _text_7:any;
  _el_8:any;
  _text_9:any;
  _text_10:any;
  /*private*/ _expr_11:any;
  /*private*/ _expr_12:any;
  /*private*/ _expr_13:any;
  constructor(viewUtils:import11.ViewUtils,parentView:import10.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import19.ViewContainer) {
    super(View_Terminal2,renderType_Terminal,import15.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import16.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_11 = import9.UNINITIALIZED;
    this._expr_12 = import9.UNINITIALIZED;
    this._expr_13 = import9.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import17.ComponentRef<any> {
    this._el_0 = import11.createRenderElement(this.renderer,(null as any),'div',import11.EMPTY_INLINE_ARRAY,(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this._el_2 = import11.createRenderElement(this.renderer,this._el_0,'span',import11.EMPTY_INLINE_ARRAY,(null as any));
    this._text_3 = this.renderer.createText(this._el_2,'',(null as any));
    this._text_4 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this._el_5 = import11.createRenderElement(this.renderer,this._el_0,'span',new import11.InlineArray2(2,'class','ui-terminal-command'),(null as any));
    this._text_6 = this.renderer.createText(this._el_5,'',(null as any));
    this._text_7 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this._el_8 = import11.createRenderElement(this.renderer,this._el_0,'div',import11.EMPTY_INLINE_ARRAY,(null as any));
    this._text_9 = this.renderer.createText(this._el_8,'',(null as any));
    this._text_10 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._text_4,
      this._el_5,
      this._text_6,
      this._text_7,
      this._el_8,
      this._text_9,
      this._text_10
    ]
    ),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_11:any = import11.inlineInterpolate(1,'',this.parentView.context.prompt,'');
    if (import11.checkBinding(throwOnChange,this._expr_11,currVal_11)) {
      this.renderer.setText(this._text_3,currVal_11);
      this._expr_11 = currVal_11;
    }
    const currVal_12:any = import11.inlineInterpolate(1,'',this.context.$implicit.text,'');
    if (import11.checkBinding(throwOnChange,this._expr_12,currVal_12)) {
      this.renderer.setText(this._text_6,currVal_12);
      this._expr_12 = currVal_12;
    }
    const currVal_13:any = import11.inlineInterpolate(1,'',this.context.$implicit.response,'');
    if (import11.checkBinding(throwOnChange,this._expr_13,currVal_13)) {
      this.renderer.setText(this._text_9,currVal_13);
      this._expr_13 = currVal_13;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
var renderType_Terminal:import12.RenderComponentType = import11.createRenderComponentType('',0,import13.ViewEncapsulation.None,styles_Terminal,{});
export class View_Terminal0 extends import10.AppView<import1.Terminal> {
  _text_0:any;
  _el_1:any;
  _NgClass_1_3:import20.Wrapper_NgClass;
  _NgStyle_1_4:import21.Wrapper_NgStyle;
  _text_2:any;
  _anchor_3:any;
  /*private*/ _vc_3:import19.ViewContainer;
  _TemplateRef_3_5:any;
  _NgIf_3_6:import22.Wrapper_NgIf;
  _text_4:any;
  _el_5:any;
  _text_6:any;
  _anchor_7:any;
  /*private*/ _vc_7:import19.ViewContainer;
  _TemplateRef_7_5:any;
  _NgFor_7_6:import23.Wrapper_NgFor;
  _text_8:any;
  _text_9:any;
  _el_10:any;
  _text_11:any;
  _el_12:any;
  _text_13:any;
  _text_14:any;
  _el_15:any;
  _DefaultValueAccessor_15_3:import24.Wrapper_DefaultValueAccessor;
  _NG_VALUE_ACCESSOR_15_4:any[];
  _NgModel_15_5:import25.Wrapper_NgModel;
  _NgControl_15_6:any;
  _NgControlStatus_15_7:import26.Wrapper_NgControlStatus;
  _text_16:any;
  _text_17:any;
  _text_18:any;
  /*private*/ _expr_32:any;
  constructor(viewUtils:import11.ViewUtils,parentView:import10.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_Terminal0,renderType_Terminal,import15.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import16.ChangeDetectorStatus.CheckAlways);
    this._expr_32 = import9.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import17.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n        ',(null as any));
    this._el_1 = import11.createRenderElement(this.renderer,parentRenderNode,'div',import11.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_1_3 = new import20.Wrapper_NgClass(this.parentView.injectorGet(import27.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import28.KeyValueDiffers,this.parentIndex),new import18.ElementRef(this._el_1),this.renderer);
    this._NgStyle_1_4 = new import21.Wrapper_NgStyle(this.parentView.injectorGet(import28.KeyValueDiffers,this.parentIndex),new import18.ElementRef(this._el_1),this.renderer);
    this._text_2 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._anchor_3 = this.renderer.createTemplateAnchor(this._el_1,(null as any));
    this._vc_3 = new import19.ViewContainer(3,1,this,this._anchor_3);
    this._TemplateRef_3_5 = new import29.TemplateRef_(this,3,this._anchor_3);
    this._NgIf_3_6 = new import22.Wrapper_NgIf(this._vc_3.vcRef,this._TemplateRef_3_5);
    this._text_4 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_5 = import11.createRenderElement(this.renderer,this._el_1,'div',new import11.InlineArray2(2,'class','ui-terminal-content'),(null as any));
    this._text_6 = this.renderer.createText(this._el_5,'\n                ',(null as any));
    this._anchor_7 = this.renderer.createTemplateAnchor(this._el_5,(null as any));
    this._vc_7 = new import19.ViewContainer(7,5,this,this._anchor_7);
    this._TemplateRef_7_5 = new import29.TemplateRef_(this,7,this._anchor_7);
    this._NgFor_7_6 = new import23.Wrapper_NgFor(this._vc_7.vcRef,this._TemplateRef_7_5,this.parentView.injectorGet(import27.IterableDiffers,this.parentIndex),this.ref);
    this._text_8 = this.renderer.createText(this._el_5,'\n            ',(null as any));
    this._text_9 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_10 = import11.createRenderElement(this.renderer,this._el_1,'div',import11.EMPTY_INLINE_ARRAY,(null as any));
    this._text_11 = this.renderer.createText(this._el_10,'\n                ',(null as any));
    this._el_12 = import11.createRenderElement(this.renderer,this._el_10,'span',new import11.InlineArray2(2,'class','ui-terminal-content-prompt'),(null as any));
    this._text_13 = this.renderer.createText(this._el_12,'',(null as any));
    this._text_14 = this.renderer.createText(this._el_10,'\n                ',(null as any));
    this._el_15 = import11.createRenderElement(this.renderer,this._el_10,'input',new import11.InlineArray8(8,'autocomplete','off','autofocus','','class','ui-terminal-input','type','text'),(null as any));
    this._DefaultValueAccessor_15_3 = new import24.Wrapper_DefaultValueAccessor(this.renderer,new import18.ElementRef(this._el_15));
    this._NG_VALUE_ACCESSOR_15_4 = [this._DefaultValueAccessor_15_3.context];
    this._NgModel_15_5 = new import25.Wrapper_NgModel((null as any),(null as any),(null as any),this._NG_VALUE_ACCESSOR_15_4);
    this._NgControl_15_6 = this._NgModel_15_5.context;
    this._NgControlStatus_15_7 = new import26.Wrapper_NgControlStatus(this._NgControl_15_6);
    this._text_16 = this.renderer.createText(this._el_10,'\n            ',(null as any));
    this._text_17 = this.renderer.createText(this._el_1,'\n        ',(null as any));
    this._text_18 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    var disposable_0:Function = import11.subscribeToRenderElement(this,this._el_1,new import11.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_1));
    var disposable_1:Function = import11.subscribeToRenderElement(this,this._el_15,new import11.InlineArray8(8,'ngModelChange',(null as any),'keydown',(null as any),'input',(null as any),'blur',(null as any)),this.eventHandler(this.handleEvent_15));
    this._NgModel_15_5.subscribe(this,this.eventHandler(this.handleEvent_15),true);
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2,
      this._anchor_3,
      this._text_4,
      this._el_5,
      this._text_6,
      this._anchor_7,
      this._text_8,
      this._text_9,
      this._el_10,
      this._text_11,
      this._el_12,
      this._text_13,
      this._text_14,
      this._el_15,
      this._text_16,
      this._text_17,
      this._text_18
    ]
    ),[
      disposable_0,
      disposable_1
    ]
    );
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import29.TemplateRef) && (3 === requestNodeIndex))) { return this._TemplateRef_3_5; }
    if (((token === import30.NgIf) && (3 === requestNodeIndex))) { return this._NgIf_3_6.context; }
    if (((token === import29.TemplateRef) && (7 === requestNodeIndex))) { return this._TemplateRef_7_5; }
    if (((token === import31.NgFor) && (7 === requestNodeIndex))) { return this._NgFor_7_6.context; }
    if (((token === import32.DefaultValueAccessor) && (15 === requestNodeIndex))) { return this._DefaultValueAccessor_15_3.context; }
    if (((token === import33.NG_VALUE_ACCESSOR) && (15 === requestNodeIndex))) { return this._NG_VALUE_ACCESSOR_15_4; }
    if (((token === import34.NgModel) && (15 === requestNodeIndex))) { return this._NgModel_15_5.context; }
    if (((token === import35.NgControl) && (15 === requestNodeIndex))) { return this._NgControl_15_6; }
    if (((token === import36.NgControlStatus) && (15 === requestNodeIndex))) { return this._NgControlStatus_15_7.context; }
    if (((token === import37.NgClass) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 17)))) { return this._NgClass_1_3.context; }
    if (((token === import38.NgStyle) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 17)))) { return this._NgStyle_1_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1_0_0:any = this.context.styleClass;
    this._NgClass_1_3.check_klass(currVal_1_0_0,throwOnChange,false);
    const currVal_1_0_1:any = 'ui-terminal ui-widget ui-widget-content ui-corner-all';
    this._NgClass_1_3.check_ngClass(currVal_1_0_1,throwOnChange,false);
    this._NgClass_1_3.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_1_1_0:any = this.context.style;
    this._NgStyle_1_4.check_ngStyle(currVal_1_1_0,throwOnChange,false);
    this._NgStyle_1_4.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_3_0_0:any = this.context.welcomeMessage;
    this._NgIf_3_6.check_ngIf(currVal_3_0_0,throwOnChange,false);
    this._NgIf_3_6.ngDoCheck(this,this._anchor_3,throwOnChange);
    const currVal_7_0_0:any = this.context.commands;
    this._NgFor_7_6.check_ngForOf(currVal_7_0_0,throwOnChange,false);
    this._NgFor_7_6.ngDoCheck(this,this._anchor_7,throwOnChange);
    this._DefaultValueAccessor_15_3.ngDoCheck(this,this._el_15,throwOnChange);
    const currVal_15_1_0:any = this.context.command;
    this._NgModel_15_5.check_model(currVal_15_1_0,throwOnChange,false);
    this._NgModel_15_5.ngDoCheck(this,this._el_15,throwOnChange);
    this._NgControlStatus_15_7.ngDoCheck(this,this._el_15,throwOnChange);
    this._vc_3.detectChangesInNestedViews(throwOnChange);
    this._vc_7.detectChangesInNestedViews(throwOnChange);
    const currVal_32:any = import11.inlineInterpolate(1,'',this.context.prompt,'');
    if (import11.checkBinding(throwOnChange,this._expr_32,currVal_32)) {
      this.renderer.setText(this._text_13,currVal_32);
      this._expr_32 = currVal_32;
    }
    this._NgControlStatus_15_7.checkHost(this,this,this._el_15,throwOnChange);
  }
  destroyInternal():void {
    this._vc_3.destroyNestedViews();
    this._vc_7.destroyNestedViews();
    this._NgModel_15_5.ngOnDestroy();
  }
  createEmbeddedViewInternal(nodeIndex:number):import10.AppView<any> {
    if ((nodeIndex == 3)) { return new View_Terminal1(this.viewUtils,this,3,this._anchor_3,this._vc_3); }
    if ((nodeIndex == 7)) { return new View_Terminal2(this.viewUtils,this,7,this._anchor_7,this._vc_7); }
    return (null as any);
  }
  handleEvent_1(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.context.focus(this._el_15)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
  handleEvent_15(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    result = (this._DefaultValueAccessor_15_3.handleEvent(eventName,$event) && result);
    if ((eventName == 'ngModelChange')) {
      const pd_sub_0:any = ((<any>(this.context.command = $event)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'keydown')) {
      const pd_sub_1:any = ((<any>this.context.handleCommand($event)) !== false);
      result = (pd_sub_1 && result);
    }
    return result;
  }
}
