// primeng/components/fileupload/fileupload.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from '@angular/core/src/linker/ng_module_factory';
import * as import1 from 'primeng/components/fileupload/fileupload';
import * as import2 from '@angular/common/src/common_module';
import * as import3 from 'primeng/components/common/shared';
import * as import4 from 'primeng/components/button/button';
import * as import5 from 'primeng/components/progressbar/progressbar';
import * as import6 from 'primeng/components/messages/messages';
import * as import7 from '@angular/common/src/localization';
import * as import8 from '@angular/core/src/di/injector';
import * as import9 from '@angular/core/src/i18n/tokens';
import * as import10 from '@angular/core/src/change_detection/change_detection_util';
import * as import11 from '@angular/core/src/linker/view';
import * as import12 from '@angular/core/src/linker/view_utils';
import * as import13 from '@angular/core/src/render/api';
import * as import14 from '@angular/core/src/metadata/view';
import * as import15 from '@angular/core/src/linker/query_list';
import * as import16 from '@angular/core/src/linker/view_type';
import * as import17 from '@angular/core/src/change_detection/constants';
import * as import18 from '@angular/core/src/linker/component_factory';
import * as import19 from '@angular/platform-browser/src/security/dom_sanitization_service';
import * as import20 from 'primeng/components/dom/domhandler';
import * as import21 from '../button/button.ngfactory';
import * as import22 from '@angular/core/src/linker/view_container';
import * as import23 from '@angular/core/src/linker/element_ref';
import * as import24 from '../progressbar/progressbar.ngfactory';
import * as import25 from '@angular/core/src/security';
import * as import26 from '../../../@angular/common/src/directives/ng_if.ngfactory';
import * as import27 from '@angular/core/src/linker/template_ref';
import * as import28 from '@angular/common/src/directives/ng_if';
import * as import29 from '../../../@angular/common/src/directives/ng_for.ngfactory';
import * as import30 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import31 from '@angular/common/src/directives/ng_for';
import * as import32 from '../../../@angular/common/src/directives/ng_class.ngfactory';
import * as import33 from '../../../@angular/common/src/directives/ng_style.ngfactory';
import * as import34 from '../messages/messages.ngfactory';
import * as import35 from '../common/shared.ngfactory';
import * as import36 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import37 from '@angular/common/src/directives/ng_class';
import * as import38 from '@angular/common/src/directives/ng_style';
class FileUploadModuleInjector extends import0.NgModuleInjector<import1.FileUploadModule> {
  _CommonModule_0:import2.CommonModule;
  _SharedModule_1:import3.SharedModule;
  _ButtonModule_2:import4.ButtonModule;
  _ProgressBarModule_3:import5.ProgressBarModule;
  _MessagesModule_4:import6.MessagesModule;
  _FileUploadModule_5:import1.FileUploadModule;
  __NgLocalization_6:import7.NgLocaleLocalization;
  constructor(parent:import8.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_6():import7.NgLocaleLocalization {
    if ((this.__NgLocalization_6 == null)) { (this.__NgLocalization_6 = new import7.NgLocaleLocalization(this.parent.get(import9.LOCALE_ID))); }
    return this.__NgLocalization_6;
  }
  createInternal():import1.FileUploadModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._SharedModule_1 = new import3.SharedModule();
    this._ButtonModule_2 = new import4.ButtonModule();
    this._ProgressBarModule_3 = new import5.ProgressBarModule();
    this._MessagesModule_4 = new import6.MessagesModule();
    this._FileUploadModule_5 = new import1.FileUploadModule();
    return this._FileUploadModule_5;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.SharedModule)) { return this._SharedModule_1; }
    if ((token === import4.ButtonModule)) { return this._ButtonModule_2; }
    if ((token === import5.ProgressBarModule)) { return this._ProgressBarModule_3; }
    if ((token === import6.MessagesModule)) { return this._MessagesModule_4; }
    if ((token === import1.FileUploadModule)) { return this._FileUploadModule_5; }
    if ((token === import7.NgLocalization)) { return this._NgLocalization_6; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const FileUploadModuleNgFactory:import0.NgModuleFactory<import1.FileUploadModule> = new import0.NgModuleFactory(FileUploadModuleInjector,import1.FileUploadModule);
export class Wrapper_FileUpload {
  /*private*/ _eventHandler:Function;
  context:import1.FileUpload;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  /*private*/ _expr_5:any;
  /*private*/ _expr_6:any;
  /*private*/ _expr_7:any;
  /*private*/ _expr_8:any;
  /*private*/ _expr_9:any;
  /*private*/ _expr_10:any;
  /*private*/ _expr_11:any;
  /*private*/ _expr_12:any;
  /*private*/ _expr_13:any;
  /*private*/ _expr_14:any;
  subscription0:any;
  subscription1:any;
  subscription2:any;
  subscription3:any;
  subscription4:any;
  subscription5:any;
  constructor(p0:any) {
    this._changed = false;
    this.context = new import1.FileUpload(p0);
    this._expr_0 = import10.UNINITIALIZED;
    this._expr_1 = import10.UNINITIALIZED;
    this._expr_2 = import10.UNINITIALIZED;
    this._expr_3 = import10.UNINITIALIZED;
    this._expr_4 = import10.UNINITIALIZED;
    this._expr_5 = import10.UNINITIALIZED;
    this._expr_6 = import10.UNINITIALIZED;
    this._expr_7 = import10.UNINITIALIZED;
    this._expr_8 = import10.UNINITIALIZED;
    this._expr_9 = import10.UNINITIALIZED;
    this._expr_10 = import10.UNINITIALIZED;
    this._expr_11 = import10.UNINITIALIZED;
    this._expr_12 = import10.UNINITIALIZED;
    this._expr_13 = import10.UNINITIALIZED;
    this._expr_14 = import10.UNINITIALIZED;
  }
  ngOnDetach(view:import11.AppView<any>,componentView:import11.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
    (this.subscription0 && this.subscription0.unsubscribe());
    (this.subscription1 && this.subscription1.unsubscribe());
    (this.subscription2 && this.subscription2.unsubscribe());
    (this.subscription3 && this.subscription3.unsubscribe());
    (this.subscription4 && this.subscription4.unsubscribe());
    (this.subscription5 && this.subscription5.unsubscribe());
  }
  check_name(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import12.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.name = currValue;
      this._expr_0 = currValue;
    }
  }
  check_url(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import12.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.url = currValue;
      this._expr_1 = currValue;
    }
  }
  check_multiple(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import12.checkBinding(throwOnChange,this._expr_2,currValue))) {
      this._changed = true;
      this.context.multiple = currValue;
      this._expr_2 = currValue;
    }
  }
  check_accept(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import12.checkBinding(throwOnChange,this._expr_3,currValue))) {
      this._changed = true;
      this.context.accept = currValue;
      this._expr_3 = currValue;
    }
  }
  check_disabled(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import12.checkBinding(throwOnChange,this._expr_4,currValue))) {
      this._changed = true;
      this.context.disabled = currValue;
      this._expr_4 = currValue;
    }
  }
  check_auto(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import12.checkBinding(throwOnChange,this._expr_5,currValue))) {
      this._changed = true;
      this.context.auto = currValue;
      this._expr_5 = currValue;
    }
  }
  check_maxFileSize(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import12.checkBinding(throwOnChange,this._expr_6,currValue))) {
      this._changed = true;
      this.context.maxFileSize = currValue;
      this._expr_6 = currValue;
    }
  }
  check_invalidFileSizeMessageSummary(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import12.checkBinding(throwOnChange,this._expr_7,currValue))) {
      this._changed = true;
      this.context.invalidFileSizeMessageSummary = currValue;
      this._expr_7 = currValue;
    }
  }
  check_invalidFileSizeMessageDetail(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import12.checkBinding(throwOnChange,this._expr_8,currValue))) {
      this._changed = true;
      this.context.invalidFileSizeMessageDetail = currValue;
      this._expr_8 = currValue;
    }
  }
  check_style(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import12.checkBinding(throwOnChange,this._expr_9,currValue))) {
      this._changed = true;
      this.context.style = currValue;
      this._expr_9 = currValue;
    }
  }
  check_styleClass(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import12.checkBinding(throwOnChange,this._expr_10,currValue))) {
      this._changed = true;
      this.context.styleClass = currValue;
      this._expr_10 = currValue;
    }
  }
  check_previewWidth(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import12.checkBinding(throwOnChange,this._expr_11,currValue))) {
      this._changed = true;
      this.context.previewWidth = currValue;
      this._expr_11 = currValue;
    }
  }
  check_chooseLabel(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import12.checkBinding(throwOnChange,this._expr_12,currValue))) {
      this._changed = true;
      this.context.chooseLabel = currValue;
      this._expr_12 = currValue;
    }
  }
  check_uploadLabel(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import12.checkBinding(throwOnChange,this._expr_13,currValue))) {
      this._changed = true;
      this.context.uploadLabel = currValue;
      this._expr_13 = currValue;
    }
  }
  check_cancelLabel(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import12.checkBinding(throwOnChange,this._expr_14,currValue))) {
      this._changed = true;
      this.context.cancelLabel = currValue;
      this._expr_14 = currValue;
    }
  }
  ngDoCheck(view:import11.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    if (!throwOnChange) { if ((view.numberOfChecks === 0)) { this.context.ngOnInit(); } }
    return changed;
  }
  checkHost(view:import11.AppView<any>,componentView:import11.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import11.AppView<any>,_eventHandler:any,emit0:boolean,emit1:boolean,emit2:boolean,emit3:boolean,emit4:boolean,emit5:boolean):void {
    this._eventHandler = _eventHandler;
    if (emit0) { (this.subscription0 = this.context.onBeforeUpload.subscribe(_eventHandler.bind(view,'onBeforeUpload'))); }
    if (emit1) { (this.subscription1 = this.context.onBeforeSend.subscribe(_eventHandler.bind(view,'onBeforeSend'))); }
    if (emit2) { (this.subscription2 = this.context.onUpload.subscribe(_eventHandler.bind(view,'onUpload'))); }
    if (emit3) { (this.subscription3 = this.context.onError.subscribe(_eventHandler.bind(view,'onError'))); }
    if (emit4) { (this.subscription4 = this.context.onClear.subscribe(_eventHandler.bind(view,'onClear'))); }
    if (emit5) { (this.subscription5 = this.context.onSelect.subscribe(_eventHandler.bind(view,'onSelect'))); }
  }
}
var renderType_FileUpload_Host:import13.RenderComponentType = import12.createRenderComponentType('',0,import14.ViewEncapsulation.None,([] as any[]),{});
class View_FileUpload_Host0 extends import11.AppView<any> {
  _el_0:any;
  compView_0:import11.AppView<import1.FileUpload>;
  _FileUpload_0_3:Wrapper_FileUpload;
  _query_PrimeTemplate_0_0:import15.QueryList<any>;
  constructor(viewUtils:import12.ViewUtils,parentView:import11.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_FileUpload_Host0,renderType_FileUpload_Host,import16.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import17.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import18.ComponentRef<any> {
    this._el_0 = import12.selectOrCreateRenderHostElement(this.renderer,'p-fileUpload',import12.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this.compView_0 = new View_FileUpload0(this.viewUtils,this,0,this._el_0);
    this._FileUpload_0_3 = new Wrapper_FileUpload(this.injectorGet(import19.DomSanitizer,this.parentIndex));
    this._query_PrimeTemplate_0_0 = new import15.QueryList<any>();
    this.compView_0.create(this._FileUpload_0_3.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import18.ComponentRef_<any>(0,this,this._el_0,this._FileUpload_0_3.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import1.FileUpload) && (0 === requestNodeIndex))) { return this._FileUpload_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._FileUpload_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    if (!throwOnChange) {
      if (this._query_PrimeTemplate_0_0.dirty) {
        this._query_PrimeTemplate_0_0.reset(([] as any[]));
        this._FileUpload_0_3.context.templates = this._query_PrimeTemplate_0_0;
        this._query_PrimeTemplate_0_0.notifyOnChanges();
      }
      if ((this.numberOfChecks === 0)) { this._FileUpload_0_3.context.ngAfterContentInit(); }
    }
    this.compView_0.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
    this._FileUpload_0_3.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
export const FileUploadNgFactory:import18.ComponentFactory<import1.FileUpload> = new import18.ComponentFactory<import1.FileUpload>('p-fileUpload',View_FileUpload_Host0,import1.FileUpload);
const styles_FileUpload:any[] = ([] as any[]);
class View_FileUpload1 extends import11.AppView<any> {
  _el_0:any;
  _DomHandler_0_3:import20.DomHandler;
  _Button_0_4:import21.Wrapper_Button;
  /*private*/ _expr_3:any;
  constructor(viewUtils:import12.ViewUtils,parentView:import11.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import22.ViewContainer) {
    super(View_FileUpload1,renderType_FileUpload,import16.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import17.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_3 = import10.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import18.ComponentRef<any> {
    this._el_0 = import12.createRenderElement(this.renderer,(null as any),'button',new import12.InlineArray8(6,'icon','fa-upload','pButton','','type','button'),(null as any));
    this._DomHandler_0_3 = new import20.DomHandler();
    this._Button_0_4 = new import21.Wrapper_Button(new import23.ElementRef(this._el_0),this._DomHandler_0_3);
    var disposable_0:Function = import12.subscribeToRenderElement(this,this._el_0,new import12.InlineArray16(14,'click',(null as any),'mouseenter',(null as any),'mouseleave',(null as any),'mousedown',(null as any),'mouseup',(null as any),'focus',(null as any),'blur',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import20.DomHandler) && (0 === requestNodeIndex))) { return this._DomHandler_0_3; }
    if (((token === import4.Button) && (0 === requestNodeIndex))) { return this._Button_0_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this.parentView.context.uploadLabel;
    this._Button_0_4.check_label(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = 'fa-upload';
    this._Button_0_4.check_icon(currVal_0_0_1,throwOnChange,false);
    this._Button_0_4.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_3:boolean = !this.parentView.context.hasFiles();
    if (import12.checkBinding(throwOnChange,this._expr_3,currVal_3)) {
      this.renderer.setElementProperty(this._el_0,'disabled',currVal_3);
      this._expr_3 = currVal_3;
    }
    this._Button_0_4.checkHost(this,this,this._el_0,throwOnChange);
    if (!throwOnChange) { if ((this.numberOfChecks === 0)) { this._Button_0_4.context.ngAfterViewInit(); } }
  }
  destroyInternal():void {
    this._Button_0_4.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    result = (this._Button_0_4.handleEvent(eventName,$event) && result);
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.context.upload()) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
class View_FileUpload2 extends import11.AppView<any> {
  _el_0:any;
  _DomHandler_0_3:import20.DomHandler;
  _Button_0_4:import21.Wrapper_Button;
  /*private*/ _expr_3:any;
  constructor(viewUtils:import12.ViewUtils,parentView:import11.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import22.ViewContainer) {
    super(View_FileUpload2,renderType_FileUpload,import16.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import17.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_3 = import10.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import18.ComponentRef<any> {
    this._el_0 = import12.createRenderElement(this.renderer,(null as any),'button',new import12.InlineArray8(6,'icon','fa-close','pButton','','type','button'),(null as any));
    this._DomHandler_0_3 = new import20.DomHandler();
    this._Button_0_4 = new import21.Wrapper_Button(new import23.ElementRef(this._el_0),this._DomHandler_0_3);
    var disposable_0:Function = import12.subscribeToRenderElement(this,this._el_0,new import12.InlineArray16(14,'click',(null as any),'mouseenter',(null as any),'mouseleave',(null as any),'mousedown',(null as any),'mouseup',(null as any),'focus',(null as any),'blur',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import20.DomHandler) && (0 === requestNodeIndex))) { return this._DomHandler_0_3; }
    if (((token === import4.Button) && (0 === requestNodeIndex))) { return this._Button_0_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this.parentView.context.cancelLabel;
    this._Button_0_4.check_label(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = 'fa-close';
    this._Button_0_4.check_icon(currVal_0_0_1,throwOnChange,false);
    this._Button_0_4.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_3:boolean = !this.parentView.context.hasFiles();
    if (import12.checkBinding(throwOnChange,this._expr_3,currVal_3)) {
      this.renderer.setElementProperty(this._el_0,'disabled',currVal_3);
      this._expr_3 = currVal_3;
    }
    this._Button_0_4.checkHost(this,this,this._el_0,throwOnChange);
    if (!throwOnChange) { if ((this.numberOfChecks === 0)) { this._Button_0_4.context.ngAfterViewInit(); } }
  }
  destroyInternal():void {
    this._Button_0_4.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    result = (this._Button_0_4.handleEvent(eventName,$event) && result);
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.context.clear()) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
class View_FileUpload3 extends import11.AppView<any> {
  _el_0:any;
  compView_0:import11.AppView<import5.ProgressBar>;
  _ProgressBar_0_3:import24.Wrapper_ProgressBar;
  constructor(viewUtils:import12.ViewUtils,parentView:import11.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import22.ViewContainer) {
    super(View_FileUpload3,renderType_FileUpload,import16.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import17.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import18.ComponentRef<any> {
    this._el_0 = import12.createRenderElement(this.renderer,(null as any),'p-progressBar',import12.EMPTY_INLINE_ARRAY,(null as any));
    this.compView_0 = new import24.View_ProgressBar0(this.viewUtils,this,0,this._el_0);
    this._ProgressBar_0_3 = new import24.Wrapper_ProgressBar();
    this.compView_0.create(this._ProgressBar_0_3.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import5.ProgressBar) && (0 === requestNodeIndex))) { return this._ProgressBar_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this.parentView.context.progress;
    this._ProgressBar_0_3.check_value(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = false;
    this._ProgressBar_0_3.check_showValue(currVal_0_0_1,throwOnChange,false);
    this._ProgressBar_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_FileUpload7 extends import11.AppView<any> {
  _el_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  constructor(viewUtils:import12.ViewUtils,parentView:import11.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import22.ViewContainer) {
    super(View_FileUpload7,renderType_FileUpload,import16.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import17.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_1 = import10.UNINITIALIZED;
    this._expr_2 = import10.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import18.ComponentRef<any> {
    this._el_0 = import12.createRenderElement(this.renderer,(null as any),'img',import12.EMPTY_INLINE_ARRAY,(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1:any = this.parentView.context.$implicit.objectURL;
    if (import12.checkBinding(throwOnChange,this._expr_1,currVal_1)) {
      this.renderer.setElementProperty(this._el_0,'src',this.viewUtils.sanitizer.sanitize(import25.SecurityContext.URL,currVal_1));
      this._expr_1 = currVal_1;
    }
    const currVal_2:any = this.parentView.parentView.parentView.parentView.context.previewWidth;
    if (import12.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setElementProperty(this._el_0,'width',currVal_2);
      this._expr_2 = currVal_2;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_FileUpload6 extends import11.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _anchor_3:any;
  /*private*/ _vc_3:import22.ViewContainer;
  _TemplateRef_3_5:any;
  _NgIf_3_6:import26.Wrapper_NgIf;
  _text_4:any;
  _el_5:any;
  _text_6:any;
  _text_7:any;
  _el_8:any;
  _text_9:any;
  _text_10:any;
  _el_11:any;
  _el_12:any;
  _DomHandler_12_3:import20.DomHandler;
  _Button_12_4:import21.Wrapper_Button;
  _text_13:any;
  /*private*/ _expr_19:any;
  /*private*/ _expr_20:any;
  constructor(viewUtils:import12.ViewUtils,parentView:import11.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import22.ViewContainer) {
    super(View_FileUpload6,renderType_FileUpload,import16.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import17.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_19 = import10.UNINITIALIZED;
    this._expr_20 = import10.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import18.ComponentRef<any> {
    this._el_0 = import12.createRenderElement(this.renderer,(null as any),'div',new import12.InlineArray2(2,'class','ui-fileupload-row'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                            ',(null as any));
    this._el_2 = import12.createRenderElement(this.renderer,this._el_0,'div',import12.EMPTY_INLINE_ARRAY,(null as any));
    this._anchor_3 = this.renderer.createTemplateAnchor(this._el_2,(null as any));
    this._vc_3 = new import22.ViewContainer(3,2,this,this._anchor_3);
    this._TemplateRef_3_5 = new import27.TemplateRef_(this,3,this._anchor_3);
    this._NgIf_3_6 = new import26.Wrapper_NgIf(this._vc_3.vcRef,this._TemplateRef_3_5);
    this._text_4 = this.renderer.createText(this._el_0,'\n                            ',(null as any));
    this._el_5 = import12.createRenderElement(this.renderer,this._el_0,'div',import12.EMPTY_INLINE_ARRAY,(null as any));
    this._text_6 = this.renderer.createText(this._el_5,'',(null as any));
    this._text_7 = this.renderer.createText(this._el_0,'\n                            ',(null as any));
    this._el_8 = import12.createRenderElement(this.renderer,this._el_0,'div',import12.EMPTY_INLINE_ARRAY,(null as any));
    this._text_9 = this.renderer.createText(this._el_8,'',(null as any));
    this._text_10 = this.renderer.createText(this._el_0,'\n                            ',(null as any));
    this._el_11 = import12.createRenderElement(this.renderer,this._el_0,'div',import12.EMPTY_INLINE_ARRAY,(null as any));
    this._el_12 = import12.createRenderElement(this.renderer,this._el_11,'button',new import12.InlineArray8(6,'icon','fa-close','pButton','','type','button'),(null as any));
    this._DomHandler_12_3 = new import20.DomHandler();
    this._Button_12_4 = new import21.Wrapper_Button(new import23.ElementRef(this._el_12),this._DomHandler_12_3);
    this._text_13 = this.renderer.createText(this._el_0,'\n                        ',(null as any));
    var disposable_0:Function = import12.subscribeToRenderElement(this,this._el_12,new import12.InlineArray16(14,'click',(null as any),'mouseenter',(null as any),'mouseleave',(null as any),'mousedown',(null as any),'mouseup',(null as any),'focus',(null as any),'blur',(null as any)),this.eventHandler(this.handleEvent_12));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._anchor_3,
      this._text_4,
      this._el_5,
      this._text_6,
      this._text_7,
      this._el_8,
      this._text_9,
      this._text_10,
      this._el_11,
      this._el_12,
      this._text_13
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import27.TemplateRef) && (3 === requestNodeIndex))) { return this._TemplateRef_3_5; }
    if (((token === import28.NgIf) && (3 === requestNodeIndex))) { return this._NgIf_3_6.context; }
    if (((token === import20.DomHandler) && (12 === requestNodeIndex))) { return this._DomHandler_12_3; }
    if (((token === import4.Button) && (12 === requestNodeIndex))) { return this._Button_12_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_3_0_0:any = this.parentView.parentView.parentView.context.isImage(this.context.$implicit);
    this._NgIf_3_6.check_ngIf(currVal_3_0_0,throwOnChange,false);
    this._NgIf_3_6.ngDoCheck(this,this._anchor_3,throwOnChange);
    const currVal_12_0_0:any = 'fa-close';
    this._Button_12_4.check_icon(currVal_12_0_0,throwOnChange,false);
    this._Button_12_4.ngDoCheck(this,this._el_12,throwOnChange);
    this._vc_3.detectChangesInNestedViews(throwOnChange);
    const currVal_19:any = import12.inlineInterpolate(1,'',this.context.$implicit.name,'');
    if (import12.checkBinding(throwOnChange,this._expr_19,currVal_19)) {
      this.renderer.setText(this._text_6,currVal_19);
      this._expr_19 = currVal_19;
    }
    const currVal_20:any = import12.inlineInterpolate(1,'',this.parentView.parentView.parentView.context.formatSize(this.context.$implicit.size),'');
    if (import12.checkBinding(throwOnChange,this._expr_20,currVal_20)) {
      this.renderer.setText(this._text_9,currVal_20);
      this._expr_20 = currVal_20;
    }
    this._Button_12_4.checkHost(this,this,this._el_12,throwOnChange);
    if (!throwOnChange) { if ((this.numberOfChecks === 0)) { this._Button_12_4.context.ngAfterViewInit(); } }
  }
  destroyInternal():void {
    this._vc_3.destroyNestedViews();
    this._Button_12_4.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import11.AppView<any> {
    if ((nodeIndex == 3)) { return new View_FileUpload7(this.viewUtils,this,3,this._anchor_3,this._vc_3); }
    return (null as any);
  }
  handleEvent_12(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    result = (this._Button_12_4.handleEvent(eventName,$event) && result);
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.parentView.parentView.context.remove(this.parentView.parentView.parentView.context.i)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
class View_FileUpload5 extends import11.AppView<any> {
  _el_0:any;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import22.ViewContainer;
  _TemplateRef_2_5:any;
  _NgFor_2_6:import29.Wrapper_NgFor;
  _text_3:any;
  constructor(viewUtils:import12.ViewUtils,parentView:import11.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import22.ViewContainer) {
    super(View_FileUpload5,renderType_FileUpload,import16.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import17.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import18.ComponentRef<any> {
    this._el_0 = import12.createRenderElement(this.renderer,(null as any),'div',import12.EMPTY_INLINE_ARRAY,(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                        ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import22.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import27.TemplateRef_(this,2,this._anchor_2);
    this._NgFor_2_6 = new import29.Wrapper_NgFor(this._vc_2.vcRef,this._TemplateRef_2_5,this.parentView.parentView.parentView.injectorGet(import30.IterableDiffers,this.parentView.parentView.parentIndex),this.parentView.parentView.ref);
    this._text_3 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import27.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import31.NgFor) && (2 === requestNodeIndex))) { return this._NgFor_2_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2_0_0:any = this.parentView.parentView.context.files;
    this._NgFor_2_6.check_ngForOf(currVal_2_0_0,throwOnChange,false);
    this._NgFor_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import11.AppView<any> {
    if ((nodeIndex == 2)) { return new View_FileUpload6(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    return (null as any);
  }
}
class View_FileUpload9 extends import11.AppView<any> {
  _el_0:any;
  constructor(viewUtils:import12.ViewUtils,parentView:import11.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import22.ViewContainer) {
    super(View_FileUpload9,renderType_FileUpload,import16.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import17.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import18.ComponentRef<any> {
    this._el_0 = this.renderer.createTemplateAnchor((null as any),(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): ([] as any[])),(null as any));
    return (null as any);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_FileUpload8 extends import11.AppView<any> {
  _el_0:any;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import22.ViewContainer;
  _TemplateRef_2_5:any;
  _NgFor_2_6:import29.Wrapper_NgFor;
  _text_3:any;
  constructor(viewUtils:import12.ViewUtils,parentView:import11.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import22.ViewContainer) {
    super(View_FileUpload8,renderType_FileUpload,import16.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import17.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import18.ComponentRef<any> {
    this._el_0 = import12.createRenderElement(this.renderer,(null as any),'div',import12.EMPTY_INLINE_ARRAY,(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                        ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import22.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import27.TemplateRef_(this,2,this._anchor_2);
    this._NgFor_2_6 = new import29.Wrapper_NgFor(this._vc_2.vcRef,this._TemplateRef_2_5,this.parentView.parentView.parentView.injectorGet(import30.IterableDiffers,this.parentView.parentView.parentIndex),this.parentView.parentView.ref);
    this._text_3 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import27.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import31.NgFor) && (2 === requestNodeIndex))) { return this._NgFor_2_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2_0_0:any = this.parentView.parentView.context.files;
    this._NgFor_2_6.check_ngForOf(currVal_2_0_0,throwOnChange,false);
    const currVal_2_0_1:any = this.parentView.parentView.context.fileTemplate;
    this._NgFor_2_6.check_ngForTemplate(currVal_2_0_1,throwOnChange,false);
    this._NgFor_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import11.AppView<any> {
    if ((nodeIndex == 2)) { return new View_FileUpload9(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    return (null as any);
  }
}
class View_FileUpload4 extends import11.AppView<any> {
  _el_0:any;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import22.ViewContainer;
  _TemplateRef_2_5:any;
  _NgIf_2_6:import26.Wrapper_NgIf;
  _text_3:any;
  _anchor_4:any;
  /*private*/ _vc_4:import22.ViewContainer;
  _TemplateRef_4_5:any;
  _NgIf_4_6:import26.Wrapper_NgIf;
  _text_5:any;
  constructor(viewUtils:import12.ViewUtils,parentView:import11.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import22.ViewContainer) {
    super(View_FileUpload4,renderType_FileUpload,import16.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import17.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import18.ComponentRef<any> {
    this._el_0 = import12.createRenderElement(this.renderer,(null as any),'div',new import12.InlineArray2(2,'class','ui-fileupload-files'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import22.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import27.TemplateRef_(this,2,this._anchor_2);
    this._NgIf_2_6 = new import26.Wrapper_NgIf(this._vc_2.vcRef,this._TemplateRef_2_5);
    this._text_3 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this._anchor_4 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_4 = new import22.ViewContainer(4,0,this,this._anchor_4);
    this._TemplateRef_4_5 = new import27.TemplateRef_(this,4,this._anchor_4);
    this._NgIf_4_6 = new import26.Wrapper_NgIf(this._vc_4.vcRef,this._TemplateRef_4_5);
    this._text_5 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3,
      this._anchor_4,
      this._text_5
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import27.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import28.NgIf) && (2 === requestNodeIndex))) { return this._NgIf_2_6.context; }
    if (((token === import27.TemplateRef) && (4 === requestNodeIndex))) { return this._TemplateRef_4_5; }
    if (((token === import28.NgIf) && (4 === requestNodeIndex))) { return this._NgIf_4_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2_0_0:boolean = !this.parentView.context.fileTemplate;
    this._NgIf_2_6.check_ngIf(currVal_2_0_0,throwOnChange,false);
    this._NgIf_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    const currVal_4_0_0:any = this.parentView.context.fileTemplate;
    this._NgIf_4_6.check_ngIf(currVal_4_0_0,throwOnChange,false);
    this._NgIf_4_6.ngDoCheck(this,this._anchor_4,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
    this._vc_4.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
    this._vc_4.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import11.AppView<any> {
    if ((nodeIndex == 2)) { return new View_FileUpload5(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    if ((nodeIndex == 4)) { return new View_FileUpload8(this.viewUtils,this,4,this._anchor_4,this._vc_4); }
    return (null as any);
  }
}
var renderType_FileUpload:import13.RenderComponentType = import12.createRenderComponentType('',0,import14.ViewEncapsulation.None,styles_FileUpload,{});
export class View_FileUpload0 extends import11.AppView<import1.FileUpload> {
  _text_0:any;
  _el_1:any;
  _NgClass_1_3:import32.Wrapper_NgClass;
  _NgStyle_1_4:import33.Wrapper_NgStyle;
  _text_2:any;
  _el_3:any;
  _text_4:any;
  _el_5:any;
  _DomHandler_5_3:import20.DomHandler;
  _Button_5_4:import21.Wrapper_Button;
  _text_6:any;
  _el_7:any;
  _text_8:any;
  _text_9:any;
  _anchor_10:any;
  /*private*/ _vc_10:import22.ViewContainer;
  _TemplateRef_10_5:any;
  _NgIf_10_6:import26.Wrapper_NgIf;
  _text_11:any;
  _anchor_12:any;
  /*private*/ _vc_12:import22.ViewContainer;
  _TemplateRef_12_5:any;
  _NgIf_12_6:import26.Wrapper_NgIf;
  _text_13:any;
  _text_14:any;
  _el_15:any;
  _NgClass_15_3:import32.Wrapper_NgClass;
  _text_16:any;
  _anchor_17:any;
  /*private*/ _vc_17:import22.ViewContainer;
  _TemplateRef_17_5:any;
  _NgIf_17_6:import26.Wrapper_NgIf;
  _text_18:any;
  _el_19:any;
  compView_19:import11.AppView<import6.Messages>;
  _Messages_19_3:import34.Wrapper_Messages;
  _text_20:any;
  _anchor_21:any;
  /*private*/ _vc_21:import22.ViewContainer;
  _TemplateRef_21_5:any;
  _NgIf_21_6:import26.Wrapper_NgIf;
  _text_22:any;
  _el_23:any;
  /*private*/ _vc_23:import22.ViewContainer;
  compView_23:import11.AppView<import3.TemplateLoader>;
  _TemplateLoader_23_5:import35.Wrapper_TemplateLoader;
  _text_24:any;
  _text_25:any;
  _text_26:any;
  /*private*/ _expr_49:any;
  /*private*/ _expr_50:any;
  /*private*/ _expr_51:any;
  /*private*/ _expr_52:any;
  _map_53:any;
  constructor(viewUtils:import12.ViewUtils,parentView:import11.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_FileUpload0,renderType_FileUpload,import16.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import17.ChangeDetectorStatus.CheckAlways);
    this._expr_49 = import10.UNINITIALIZED;
    this._expr_50 = import10.UNINITIALIZED;
    this._expr_51 = import10.UNINITIALIZED;
    this._expr_52 = import10.UNINITIALIZED;
    this._map_53 = import12.pureProxy2((p0:any,p1:any):{[key: string]:any} => {
      return {
        'ui-fileupload-content ui-widget-content ui-corner-bottom': p0,
        'ui-fileupload-highlight': p1
      }
      ;
    });
  }
  createInternal(rootSelector:string):import18.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n        ',(null as any));
    this._el_1 = import12.createRenderElement(this.renderer,parentRenderNode,'div',import12.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_1_3 = new import32.Wrapper_NgClass(this.parentView.injectorGet(import30.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import36.KeyValueDiffers,this.parentIndex),new import23.ElementRef(this._el_1),this.renderer);
    this._NgStyle_1_4 = new import33.Wrapper_NgStyle(this.parentView.injectorGet(import36.KeyValueDiffers,this.parentIndex),new import23.ElementRef(this._el_1),this.renderer);
    this._text_2 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_3 = import12.createRenderElement(this.renderer,this._el_1,'div',new import12.InlineArray2(2,'class','ui-fileupload-buttonbar ui-widget-header ui-corner-top'),(null as any));
    this._text_4 = this.renderer.createText(this._el_3,'\n                ',(null as any));
    this._el_5 = import12.createRenderElement(this.renderer,this._el_3,'button',new import12.InlineArray8(8,'class','ui-fileupload-choose','icon','fa-plus','pButton','','type','button'),(null as any));
    this._DomHandler_5_3 = new import20.DomHandler();
    this._Button_5_4 = new import21.Wrapper_Button(new import23.ElementRef(this._el_5),this._DomHandler_5_3);
    this._text_6 = this.renderer.createText(this._el_5,' \n                    ',(null as any));
    this._el_7 = import12.createRenderElement(this.renderer,this._el_5,'input',new import12.InlineArray2(2,'type','file'),(null as any));
    this._text_8 = this.renderer.createText(this._el_5,'\n                ',(null as any));
    this._text_9 = this.renderer.createText(this._el_3,'\n\n                ',(null as any));
    this._anchor_10 = this.renderer.createTemplateAnchor(this._el_3,(null as any));
    this._vc_10 = new import22.ViewContainer(10,3,this,this._anchor_10);
    this._TemplateRef_10_5 = new import27.TemplateRef_(this,10,this._anchor_10);
    this._NgIf_10_6 = new import26.Wrapper_NgIf(this._vc_10.vcRef,this._TemplateRef_10_5);
    this._text_11 = this.renderer.createText(this._el_3,'\n                ',(null as any));
    this._anchor_12 = this.renderer.createTemplateAnchor(this._el_3,(null as any));
    this._vc_12 = new import22.ViewContainer(12,3,this,this._anchor_12);
    this._TemplateRef_12_5 = new import27.TemplateRef_(this,12,this._anchor_12);
    this._NgIf_12_6 = new import26.Wrapper_NgIf(this._vc_12.vcRef,this._TemplateRef_12_5);
    this._text_13 = this.renderer.createText(this._el_3,'\n            ',(null as any));
    this._text_14 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_15 = import12.createRenderElement(this.renderer,this._el_1,'div',import12.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_15_3 = new import32.Wrapper_NgClass(this.parentView.injectorGet(import30.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import36.KeyValueDiffers,this.parentIndex),new import23.ElementRef(this._el_15),this.renderer);
    this._text_16 = this.renderer.createText(this._el_15,'\n                ',(null as any));
    this._anchor_17 = this.renderer.createTemplateAnchor(this._el_15,(null as any));
    this._vc_17 = new import22.ViewContainer(17,15,this,this._anchor_17);
    this._TemplateRef_17_5 = new import27.TemplateRef_(this,17,this._anchor_17);
    this._NgIf_17_6 = new import26.Wrapper_NgIf(this._vc_17.vcRef,this._TemplateRef_17_5);
    this._text_18 = this.renderer.createText(this._el_15,'\n                \n                ',(null as any));
    this._el_19 = import12.createRenderElement(this.renderer,this._el_15,'p-messages',import12.EMPTY_INLINE_ARRAY,(null as any));
    this.compView_19 = new import34.View_Messages0(this.viewUtils,this,19,this._el_19);
    this._Messages_19_3 = new import34.Wrapper_Messages();
    this.compView_19.create(this._Messages_19_3.context);
    this._text_20 = this.renderer.createText(this._el_15,'\n                \n                ',(null as any));
    this._anchor_21 = this.renderer.createTemplateAnchor(this._el_15,(null as any));
    this._vc_21 = new import22.ViewContainer(21,15,this,this._anchor_21);
    this._TemplateRef_21_5 = new import27.TemplateRef_(this,21,this._anchor_21);
    this._NgIf_21_6 = new import26.Wrapper_NgIf(this._vc_21.vcRef,this._TemplateRef_21_5);
    this._text_22 = this.renderer.createText(this._el_15,'\n                \n                ',(null as any));
    this._el_23 = import12.createRenderElement(this.renderer,this._el_15,'p-templateLoader',import12.EMPTY_INLINE_ARRAY,(null as any));
    this._vc_23 = new import22.ViewContainer(23,15,this,this._el_23);
    this.compView_23 = new import35.View_TemplateLoader0(this.viewUtils,this,23,this._el_23);
    this._TemplateLoader_23_5 = new import35.Wrapper_TemplateLoader(this._vc_23.vcRef);
    this.compView_23.create(this._TemplateLoader_23_5.context);
    this._text_24 = this.renderer.createText(this._el_15,'\n            ',(null as any));
    this._text_25 = this.renderer.createText(this._el_1,'\n        ',(null as any));
    this._text_26 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    var disposable_0:Function = import12.subscribeToRenderElement(this,this._el_5,new import12.InlineArray16(14,'click',(null as any),'mouseenter',(null as any),'mouseleave',(null as any),'mousedown',(null as any),'mouseup',(null as any),'focus',(null as any),'blur',(null as any)),this.eventHandler(this.handleEvent_5));
    var disposable_1:Function = import12.subscribeToRenderElement(this,this._el_7,new import12.InlineArray2(2,'change',(null as any)),this.eventHandler(this.handleEvent_7));
    var disposable_2:Function = import12.subscribeToRenderElement(this,this._el_15,new import12.InlineArray8(8,'dragenter',(null as any),'dragover',(null as any),'dragleave',(null as any),'drop',(null as any)),this.eventHandler(this.handleEvent_15));
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2,
      this._el_3,
      this._text_4,
      this._el_5,
      this._text_6,
      this._el_7,
      this._text_8,
      this._text_9,
      this._anchor_10,
      this._text_11,
      this._anchor_12,
      this._text_13,
      this._text_14,
      this._el_15,
      this._text_16,
      this._anchor_17,
      this._text_18,
      this._el_19,
      this._text_20,
      this._anchor_21,
      this._text_22,
      this._el_23,
      this._text_24,
      this._text_25,
      this._text_26
    ]
    ),[
      disposable_0,
      disposable_1,
      disposable_2
    ]
    );
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import20.DomHandler) && ((5 <= requestNodeIndex) && (requestNodeIndex <= 8)))) { return this._DomHandler_5_3; }
    if (((token === import4.Button) && ((5 <= requestNodeIndex) && (requestNodeIndex <= 8)))) { return this._Button_5_4.context; }
    if (((token === import27.TemplateRef) && (10 === requestNodeIndex))) { return this._TemplateRef_10_5; }
    if (((token === import28.NgIf) && (10 === requestNodeIndex))) { return this._NgIf_10_6.context; }
    if (((token === import27.TemplateRef) && (12 === requestNodeIndex))) { return this._TemplateRef_12_5; }
    if (((token === import28.NgIf) && (12 === requestNodeIndex))) { return this._NgIf_12_6.context; }
    if (((token === import27.TemplateRef) && (17 === requestNodeIndex))) { return this._TemplateRef_17_5; }
    if (((token === import28.NgIf) && (17 === requestNodeIndex))) { return this._NgIf_17_6.context; }
    if (((token === import6.Messages) && (19 === requestNodeIndex))) { return this._Messages_19_3.context; }
    if (((token === import27.TemplateRef) && (21 === requestNodeIndex))) { return this._TemplateRef_21_5; }
    if (((token === import28.NgIf) && (21 === requestNodeIndex))) { return this._NgIf_21_6.context; }
    if (((token === import3.TemplateLoader) && (23 === requestNodeIndex))) { return this._TemplateLoader_23_5.context; }
    if (((token === import37.NgClass) && ((15 <= requestNodeIndex) && (requestNodeIndex <= 24)))) { return this._NgClass_15_3.context; }
    if (((token === import37.NgClass) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 25)))) { return this._NgClass_1_3.context; }
    if (((token === import38.NgStyle) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 25)))) { return this._NgStyle_1_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1_0_0:any = this.context.styleClass;
    this._NgClass_1_3.check_klass(currVal_1_0_0,throwOnChange,false);
    const currVal_1_0_1:any = 'ui-fileupload ui-widget';
    this._NgClass_1_3.check_ngClass(currVal_1_0_1,throwOnChange,false);
    this._NgClass_1_3.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_1_1_0:any = this.context.style;
    this._NgStyle_1_4.check_ngStyle(currVal_1_1_0,throwOnChange,false);
    this._NgStyle_1_4.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_5_0_0:any = this.context.chooseLabel;
    this._Button_5_4.check_label(currVal_5_0_0,throwOnChange,false);
    const currVal_5_0_1:any = 'fa-plus';
    this._Button_5_4.check_icon(currVal_5_0_1,throwOnChange,false);
    this._Button_5_4.ngDoCheck(this,this._el_5,throwOnChange);
    const currVal_10_0_0:boolean = !this.context.auto;
    this._NgIf_10_6.check_ngIf(currVal_10_0_0,throwOnChange,false);
    this._NgIf_10_6.ngDoCheck(this,this._anchor_10,throwOnChange);
    const currVal_12_0_0:boolean = !this.context.auto;
    this._NgIf_12_6.check_ngIf(currVal_12_0_0,throwOnChange,false);
    this._NgIf_12_6.ngDoCheck(this,this._anchor_12,throwOnChange);
    const currVal_15_0_0:any = this._map_53(true,this.context.dragHighlight);
    this._NgClass_15_3.check_ngClass(currVal_15_0_0,throwOnChange,false);
    this._NgClass_15_3.ngDoCheck(this,this._el_15,throwOnChange);
    const currVal_17_0_0:any = this.context.hasFiles();
    this._NgIf_17_6.check_ngIf(currVal_17_0_0,throwOnChange,false);
    this._NgIf_17_6.ngDoCheck(this,this._anchor_17,throwOnChange);
    const currVal_19_0_0:any = this.context.msgs;
    this._Messages_19_3.check_value(currVal_19_0_0,throwOnChange,false);
    this._Messages_19_3.ngDoCheck(this,this._el_19,throwOnChange);
    const currVal_21_0_0:any = this.context.hasFiles();
    this._NgIf_21_6.check_ngIf(currVal_21_0_0,throwOnChange,false);
    this._NgIf_21_6.ngDoCheck(this,this._anchor_21,throwOnChange);
    const currVal_23_0_0:any = this.context.contentTemplate;
    this._TemplateLoader_23_5.check_template(currVal_23_0_0,throwOnChange,false);
    this._TemplateLoader_23_5.ngDoCheck(this,this._el_23,throwOnChange);
    this._vc_10.detectChangesInNestedViews(throwOnChange);
    this._vc_12.detectChangesInNestedViews(throwOnChange);
    this._vc_17.detectChangesInNestedViews(throwOnChange);
    this._vc_21.detectChangesInNestedViews(throwOnChange);
    this._vc_23.detectChangesInNestedViews(throwOnChange);
    const currVal_49:any = this.context.disabled;
    if (import12.checkBinding(throwOnChange,this._expr_49,currVal_49)) {
      this.renderer.setElementProperty(this._el_5,'disabled',currVal_49);
      this._expr_49 = currVal_49;
    }
    this._Button_5_4.checkHost(this,this,this._el_5,throwOnChange);
    const currVal_50:any = this.context.multiple;
    if (import12.checkBinding(throwOnChange,this._expr_50,currVal_50)) {
      this.renderer.setElementProperty(this._el_7,'multiple',currVal_50);
      this._expr_50 = currVal_50;
    }
    const currVal_51:any = this.context.accept;
    if (import12.checkBinding(throwOnChange,this._expr_51,currVal_51)) {
      this.renderer.setElementProperty(this._el_7,'accept',currVal_51);
      this._expr_51 = currVal_51;
    }
    const currVal_52:any = this.context.disabled;
    if (import12.checkBinding(throwOnChange,this._expr_52,currVal_52)) {
      this.renderer.setElementProperty(this._el_7,'disabled',currVal_52);
      this._expr_52 = currVal_52;
    }
    this.compView_19.internalDetectChanges(throwOnChange);
    this.compView_23.internalDetectChanges(throwOnChange);
    if (!throwOnChange) { if ((this.numberOfChecks === 0)) { this._Button_5_4.context.ngAfterViewInit(); } }
  }
  destroyInternal():void {
    this._vc_10.destroyNestedViews();
    this._vc_12.destroyNestedViews();
    this._vc_17.destroyNestedViews();
    this._vc_21.destroyNestedViews();
    this._vc_23.destroyNestedViews();
    this.compView_19.destroy();
    this.compView_23.destroy();
    this._Button_5_4.ngOnDestroy();
  }
  createEmbeddedViewInternal(nodeIndex:number):import11.AppView<any> {
    if ((nodeIndex == 10)) { return new View_FileUpload1(this.viewUtils,this,10,this._anchor_10,this._vc_10); }
    if ((nodeIndex == 12)) { return new View_FileUpload2(this.viewUtils,this,12,this._anchor_12,this._vc_12); }
    if ((nodeIndex == 17)) { return new View_FileUpload3(this.viewUtils,this,17,this._anchor_17,this._vc_17); }
    if ((nodeIndex == 21)) { return new View_FileUpload4(this.viewUtils,this,21,this._anchor_21,this._vc_21); }
    return (null as any);
  }
  handleEvent_5(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    result = (this._Button_5_4.handleEvent(eventName,$event) && result);
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.context.onChooseClick($event,this._el_7)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
  handleEvent_7(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'change')) {
      const pd_sub_0:any = ((<any>this.context.onFileSelect($event)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
  handleEvent_15(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'dragenter')) {
      const pd_sub_0:any = ((<any>this.context.onDragEnter($event)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'dragover')) {
      const pd_sub_1:any = ((<any>this.context.onDragOver($event)) !== false);
      result = (pd_sub_1 && result);
    }
    if ((eventName == 'dragleave')) {
      const pd_sub_2:any = ((<any>this.context.onDragLeave($event)) !== false);
      result = (pd_sub_2 && result);
    }
    if ((eventName == 'drop')) {
      const pd_sub_3:any = ((<any>this.context.onDrop($event)) !== false);
      result = (pd_sub_3 && result);
    }
    return result;
  }
}
