// @angular/http/src/http_module.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from '@angular/core/src/linker/ng_module_factory';
import * as import1 from '@angular/http/src/http_module';
import * as import2 from '@angular/http/src/backends/browser_xhr';
import * as import3 from '@angular/http/src/base_response_options';
import * as import4 from '@angular/http/src/backends/xhr_backend';
import * as import5 from '@angular/http/src/base_request_options';
import * as import6 from '@angular/core/src/di/injector';
import * as import7 from '@angular/http/src/interfaces';
import * as import8 from '@angular/http/src/http';
import * as import9 from '@angular/http/src/backends/browser_jsonp';
import * as import10 from '@angular/http/src/backends/jsonp_backend';
class HttpModuleInjector extends import0.NgModuleInjector<import1.HttpModule> {
  _HttpModule_0:import1.HttpModule;
  __BrowserXhr_1:import2.BrowserXhr;
  __ResponseOptions_2:import3.BaseResponseOptions;
  __XSRFStrategy_3:any;
  __XHRBackend_4:import4.XHRBackend;
  __RequestOptions_5:import5.BaseRequestOptions;
  __Http_6:any;
  constructor(parent:import6.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _BrowserXhr_1():import2.BrowserXhr {
    if ((this.__BrowserXhr_1 == null)) { (this.__BrowserXhr_1 = new import2.BrowserXhr()); }
    return this.__BrowserXhr_1;
  }
  get _ResponseOptions_2():import3.BaseResponseOptions {
    if ((this.__ResponseOptions_2 == null)) { (this.__ResponseOptions_2 = new import3.BaseResponseOptions()); }
    return this.__ResponseOptions_2;
  }
  get _XSRFStrategy_3():any {
    if ((this.__XSRFStrategy_3 == null)) { (this.__XSRFStrategy_3 = import1._createDefaultCookieXSRFStrategy()); }
    return this.__XSRFStrategy_3;
  }
  get _XHRBackend_4():import4.XHRBackend {
    if ((this.__XHRBackend_4 == null)) { (this.__XHRBackend_4 = new import4.XHRBackend(this._BrowserXhr_1,this._ResponseOptions_2,this._XSRFStrategy_3)); }
    return this.__XHRBackend_4;
  }
  get _RequestOptions_5():import5.BaseRequestOptions {
    if ((this.__RequestOptions_5 == null)) { (this.__RequestOptions_5 = new import5.BaseRequestOptions()); }
    return this.__RequestOptions_5;
  }
  get _Http_6():any {
    if ((this.__Http_6 == null)) { (this.__Http_6 = import1.httpFactory(this._XHRBackend_4,this._RequestOptions_5)); }
    return this.__Http_6;
  }
  createInternal():import1.HttpModule {
    this._HttpModule_0 = new import1.HttpModule();
    return this._HttpModule_0;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import1.HttpModule)) { return this._HttpModule_0; }
    if ((token === import2.BrowserXhr)) { return this._BrowserXhr_1; }
    if ((token === import3.ResponseOptions)) { return this._ResponseOptions_2; }
    if ((token === import7.XSRFStrategy)) { return this._XSRFStrategy_3; }
    if ((token === import4.XHRBackend)) { return this._XHRBackend_4; }
    if ((token === import5.RequestOptions)) { return this._RequestOptions_5; }
    if ((token === import8.Http)) { return this._Http_6; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const HttpModuleNgFactory:import0.NgModuleFactory<import1.HttpModule> = new import0.NgModuleFactory(HttpModuleInjector,import1.HttpModule);
class JsonpModuleInjector extends import0.NgModuleInjector<import1.JsonpModule> {
  _JsonpModule_0:import1.JsonpModule;
  __BrowserJsonp_1:import9.BrowserJsonp;
  __ResponseOptions_2:import3.BaseResponseOptions;
  __JSONPBackend_3:import10.JSONPBackend_;
  __RequestOptions_4:import5.BaseRequestOptions;
  __Jsonp_5:any;
  constructor(parent:import6.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _BrowserJsonp_1():import9.BrowserJsonp {
    if ((this.__BrowserJsonp_1 == null)) { (this.__BrowserJsonp_1 = new import9.BrowserJsonp()); }
    return this.__BrowserJsonp_1;
  }
  get _ResponseOptions_2():import3.BaseResponseOptions {
    if ((this.__ResponseOptions_2 == null)) { (this.__ResponseOptions_2 = new import3.BaseResponseOptions()); }
    return this.__ResponseOptions_2;
  }
  get _JSONPBackend_3():import10.JSONPBackend_ {
    if ((this.__JSONPBackend_3 == null)) { (this.__JSONPBackend_3 = new import10.JSONPBackend_(this._BrowserJsonp_1,this._ResponseOptions_2)); }
    return this.__JSONPBackend_3;
  }
  get _RequestOptions_4():import5.BaseRequestOptions {
    if ((this.__RequestOptions_4 == null)) { (this.__RequestOptions_4 = new import5.BaseRequestOptions()); }
    return this.__RequestOptions_4;
  }
  get _Jsonp_5():any {
    if ((this.__Jsonp_5 == null)) { (this.__Jsonp_5 = import1.jsonpFactory(this._JSONPBackend_3,this._RequestOptions_4)); }
    return this.__Jsonp_5;
  }
  createInternal():import1.JsonpModule {
    this._JsonpModule_0 = new import1.JsonpModule();
    return this._JsonpModule_0;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import1.JsonpModule)) { return this._JsonpModule_0; }
    if ((token === import9.BrowserJsonp)) { return this._BrowserJsonp_1; }
    if ((token === import3.ResponseOptions)) { return this._ResponseOptions_2; }
    if ((token === import10.JSONPBackend)) { return this._JSONPBackend_3; }
    if ((token === import5.RequestOptions)) { return this._RequestOptions_4; }
    if ((token === import8.Jsonp)) { return this._Jsonp_5; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const JsonpModuleNgFactory:import0.NgModuleFactory<import1.JsonpModule> = new import0.NgModuleFactory(JsonpModuleInjector,import1.JsonpModule);
