// primeng/components/autocomplete/autocomplete.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from '@angular/core/src/linker/ng_module_factory';
import * as import1 from 'primeng/components/autocomplete/autocomplete';
import * as import2 from '@angular/common/src/common_module';
import * as import3 from 'primeng/components/inputtext/inputtext';
import * as import4 from 'primeng/components/button/button';
import * as import5 from 'primeng/components/common/shared';
import * as import6 from '@angular/common/src/localization';
import * as import7 from '@angular/core/src/di/injector';
import * as import8 from '@angular/core/src/i18n/tokens';
import * as import9 from '@angular/core/src/change_detection/change_detection_util';
import * as import10 from '@angular/core/src/linker/view';
import * as import11 from '@angular/core/src/linker/view_utils';
import * as import12 from '@angular/core/src/render/api';
import * as import13 from '@angular/core/src/metadata/view';
import * as import14 from 'primeng/components/dom/domhandler';
import * as import15 from '@angular/core/src/linker/query_list';
import * as import16 from '@angular/core/src/linker/view_type';
import * as import17 from '@angular/core/src/change_detection/constants';
import * as import18 from '@angular/core/src/linker/component_factory';
import * as import19 from '@angular/core/src/linker/element_ref';
import * as import20 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import21 from '@angular/forms/src/directives/control_value_accessor';
import * as import22 from '../../../@angular/common/src/directives/ng_class.ngfactory';
import * as import23 from '../../../@angular/common/src/directives/ng_style.ngfactory';
import * as import24 from '../inputtext/inputtext.ngfactory';
import * as import25 from '@angular/core/src/linker/view_container';
import * as import26 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import27 from '@angular/common/src/directives/ng_class';
import * as import28 from '@angular/common/src/directives/ng_style';
import * as import29 from '../../../@angular/common/src/directives/ng_for.ngfactory';
import * as import30 from '@angular/core/src/linker/template_ref';
import * as import31 from '@angular/common/src/directives/ng_for';
import * as import32 from '../button/button.ngfactory';
import * as import33 from '../common/shared.ngfactory';
import * as import34 from '../../../@angular/common/src/directives/ng_if.ngfactory';
import * as import35 from '@angular/common/src/directives/ng_if';
import * as import36 from '@angular/core/src/security';
class AutoCompleteModuleInjector extends import0.NgModuleInjector<import1.AutoCompleteModule> {
  _CommonModule_0:import2.CommonModule;
  _InputTextModule_1:import3.InputTextModule;
  _ButtonModule_2:import4.ButtonModule;
  _SharedModule_3:import5.SharedModule;
  _AutoCompleteModule_4:import1.AutoCompleteModule;
  __NgLocalization_5:import6.NgLocaleLocalization;
  constructor(parent:import7.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_5():import6.NgLocaleLocalization {
    if ((this.__NgLocalization_5 == null)) { (this.__NgLocalization_5 = new import6.NgLocaleLocalization(this.parent.get(import8.LOCALE_ID))); }
    return this.__NgLocalization_5;
  }
  createInternal():import1.AutoCompleteModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._InputTextModule_1 = new import3.InputTextModule();
    this._ButtonModule_2 = new import4.ButtonModule();
    this._SharedModule_3 = new import5.SharedModule();
    this._AutoCompleteModule_4 = new import1.AutoCompleteModule();
    return this._AutoCompleteModule_4;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.InputTextModule)) { return this._InputTextModule_1; }
    if ((token === import4.ButtonModule)) { return this._ButtonModule_2; }
    if ((token === import5.SharedModule)) { return this._SharedModule_3; }
    if ((token === import1.AutoCompleteModule)) { return this._AutoCompleteModule_4; }
    if ((token === import6.NgLocalization)) { return this._NgLocalization_5; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const AutoCompleteModuleNgFactory:import0.NgModuleFactory<import1.AutoCompleteModule> = new import0.NgModuleFactory(AutoCompleteModuleInjector,import1.AutoCompleteModule);
export class Wrapper_AutoComplete {
  /*private*/ _eventHandler:Function;
  context:import1.AutoComplete;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  /*private*/ _expr_5:any;
  /*private*/ _expr_6:any;
  /*private*/ _expr_7:any;
  /*private*/ _expr_8:any;
  /*private*/ _expr_9:any;
  /*private*/ _expr_10:any;
  /*private*/ _expr_11:any;
  /*private*/ _expr_12:any;
  /*private*/ _expr_13:any;
  /*private*/ _expr_14:any;
  /*private*/ _expr_15:any;
  /*private*/ _expr_16:any;
  /*private*/ _expr_17:any;
  /*private*/ _expr_18:any;
  subscription0:any;
  subscription1:any;
  subscription2:any;
  subscription3:any;
  constructor(p0:any,p1:any,p2:any,p3:any) {
    this._changed = false;
    this.context = new import1.AutoComplete(p0,p1,p2,p3);
    this._expr_0 = import9.UNINITIALIZED;
    this._expr_1 = import9.UNINITIALIZED;
    this._expr_2 = import9.UNINITIALIZED;
    this._expr_3 = import9.UNINITIALIZED;
    this._expr_4 = import9.UNINITIALIZED;
    this._expr_5 = import9.UNINITIALIZED;
    this._expr_6 = import9.UNINITIALIZED;
    this._expr_7 = import9.UNINITIALIZED;
    this._expr_8 = import9.UNINITIALIZED;
    this._expr_9 = import9.UNINITIALIZED;
    this._expr_10 = import9.UNINITIALIZED;
    this._expr_11 = import9.UNINITIALIZED;
    this._expr_12 = import9.UNINITIALIZED;
    this._expr_13 = import9.UNINITIALIZED;
    this._expr_14 = import9.UNINITIALIZED;
    this._expr_15 = import9.UNINITIALIZED;
    this._expr_16 = import9.UNINITIALIZED;
    this._expr_17 = import9.UNINITIALIZED;
    this._expr_18 = import9.UNINITIALIZED;
  }
  ngOnDetach(view:import10.AppView<any>,componentView:import10.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
    this.context.ngOnDestroy();
    (this.subscription0 && this.subscription0.unsubscribe());
    (this.subscription1 && this.subscription1.unsubscribe());
    (this.subscription2 && this.subscription2.unsubscribe());
    (this.subscription3 && this.subscription3.unsubscribe());
  }
  check_minLength(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import11.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.minLength = currValue;
      this._expr_0 = currValue;
    }
  }
  check_delay(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import11.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.delay = currValue;
      this._expr_1 = currValue;
    }
  }
  check_style(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import11.checkBinding(throwOnChange,this._expr_2,currValue))) {
      this._changed = true;
      this.context.style = currValue;
      this._expr_2 = currValue;
    }
  }
  check_styleClass(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import11.checkBinding(throwOnChange,this._expr_3,currValue))) {
      this._changed = true;
      this.context.styleClass = currValue;
      this._expr_3 = currValue;
    }
  }
  check_inputStyle(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import11.checkBinding(throwOnChange,this._expr_4,currValue))) {
      this._changed = true;
      this.context.inputStyle = currValue;
      this._expr_4 = currValue;
    }
  }
  check_inputStyleClass(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import11.checkBinding(throwOnChange,this._expr_5,currValue))) {
      this._changed = true;
      this.context.inputStyleClass = currValue;
      this._expr_5 = currValue;
    }
  }
  check_placeholder(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import11.checkBinding(throwOnChange,this._expr_6,currValue))) {
      this._changed = true;
      this.context.placeholder = currValue;
      this._expr_6 = currValue;
    }
  }
  check_readonly(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import11.checkBinding(throwOnChange,this._expr_7,currValue))) {
      this._changed = true;
      this.context.readonly = currValue;
      this._expr_7 = currValue;
    }
  }
  check_disabled(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import11.checkBinding(throwOnChange,this._expr_8,currValue))) {
      this._changed = true;
      this.context.disabled = currValue;
      this._expr_8 = currValue;
    }
  }
  check_maxlength(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import11.checkBinding(throwOnChange,this._expr_9,currValue))) {
      this._changed = true;
      this.context.maxlength = currValue;
      this._expr_9 = currValue;
    }
  }
  check_size(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import11.checkBinding(throwOnChange,this._expr_10,currValue))) {
      this._changed = true;
      this.context.size = currValue;
      this._expr_10 = currValue;
    }
  }
  check_suggestions(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import11.checkBinding(throwOnChange,this._expr_11,currValue))) {
      this._changed = true;
      this.context.suggestions = currValue;
      this._expr_11 = currValue;
    }
  }
  check_appendTo(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import11.checkBinding(throwOnChange,this._expr_12,currValue))) {
      this._changed = true;
      this.context.appendTo = currValue;
      this._expr_12 = currValue;
    }
  }
  check_field(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import11.checkBinding(throwOnChange,this._expr_13,currValue))) {
      this._changed = true;
      this.context.field = currValue;
      this._expr_13 = currValue;
    }
  }
  check_scrollHeight(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import11.checkBinding(throwOnChange,this._expr_14,currValue))) {
      this._changed = true;
      this.context.scrollHeight = currValue;
      this._expr_14 = currValue;
    }
  }
  check_dropdown(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import11.checkBinding(throwOnChange,this._expr_15,currValue))) {
      this._changed = true;
      this.context.dropdown = currValue;
      this._expr_15 = currValue;
    }
  }
  check_multiple(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import11.checkBinding(throwOnChange,this._expr_16,currValue))) {
      this._changed = true;
      this.context.multiple = currValue;
      this._expr_16 = currValue;
    }
  }
  ngDoCheck(view:import10.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    if (!throwOnChange) { this.context.ngDoCheck(); }
    return changed;
  }
  checkHost(view:import10.AppView<any>,componentView:import10.AppView<any>,el:any,throwOnChange:boolean):void {
    const currVal_17:any = this.context.filled;
    if (import11.checkBinding(throwOnChange,this._expr_17,currVal_17)) {
      view.renderer.setElementClass(el,'ui-inputwrapper-filled',currVal_17);
      this._expr_17 = currVal_17;
    }
    const currVal_18:any = this.context.focus;
    if (import11.checkBinding(throwOnChange,this._expr_18,currVal_18)) {
      view.renderer.setElementClass(el,'ui-inputwrapper-focus',currVal_18);
      this._expr_18 = currVal_18;
    }
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import10.AppView<any>,_eventHandler:any,emit0:boolean,emit1:boolean,emit2:boolean,emit3:boolean):void {
    this._eventHandler = _eventHandler;
    if (emit0) { (this.subscription0 = this.context.completeMethod.subscribe(_eventHandler.bind(view,'completeMethod'))); }
    if (emit1) { (this.subscription1 = this.context.onSelect.subscribe(_eventHandler.bind(view,'onSelect'))); }
    if (emit2) { (this.subscription2 = this.context.onUnselect.subscribe(_eventHandler.bind(view,'onUnselect'))); }
    if (emit3) { (this.subscription3 = this.context.onDropdownClick.subscribe(_eventHandler.bind(view,'onDropdownClick'))); }
  }
}
var renderType_AutoComplete_Host:import12.RenderComponentType = import11.createRenderComponentType('',0,import13.ViewEncapsulation.None,([] as any[]),{});
class View_AutoComplete_Host0 extends import10.AppView<any> {
  _el_0:any;
  compView_0:import10.AppView<import1.AutoComplete>;
  _DomHandler_0_3:import14.DomHandler;
  _AutoComplete_0_4:Wrapper_AutoComplete;
  __NG_VALUE_ACCESSOR_0_5:any[];
  _query_TemplateRef_0_0:import15.QueryList<any>;
  constructor(viewUtils:import11.ViewUtils,parentView:import10.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_AutoComplete_Host0,renderType_AutoComplete_Host,import16.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import17.ChangeDetectorStatus.CheckAlways);
  }
  get _NG_VALUE_ACCESSOR_0_5():any[] {
    if ((this.__NG_VALUE_ACCESSOR_0_5 == null)) { (this.__NG_VALUE_ACCESSOR_0_5 = [this._AutoComplete_0_4.context]); }
    return this.__NG_VALUE_ACCESSOR_0_5;
  }
  createInternal(rootSelector:string):import18.ComponentRef<any> {
    this._el_0 = import11.selectOrCreateRenderHostElement(this.renderer,'p-autoComplete',import11.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this.compView_0 = new View_AutoComplete0(this.viewUtils,this,0,this._el_0);
    this._DomHandler_0_3 = new import14.DomHandler();
    this._AutoComplete_0_4 = new Wrapper_AutoComplete(new import19.ElementRef(this._el_0),this._DomHandler_0_3,this.injectorGet(import20.IterableDiffers,this.parentIndex),this.renderer);
    this._query_TemplateRef_0_0 = new import15.QueryList<any>();
    this.compView_0.create(this._AutoComplete_0_4.context);
    this._query_TemplateRef_0_0.reset(([] as any[]));
    this._AutoComplete_0_4.context.itemTemplate = this._query_TemplateRef_0_0.first;
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import18.ComponentRef_<any>(0,this,this._el_0,this._AutoComplete_0_4.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import14.DomHandler) && (0 === requestNodeIndex))) { return this._DomHandler_0_3; }
    if (((token === import1.AutoComplete) && (0 === requestNodeIndex))) { return this._AutoComplete_0_4.context; }
    if (((token === import21.NG_VALUE_ACCESSOR) && (0 === requestNodeIndex))) { return this._NG_VALUE_ACCESSOR_0_5; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._AutoComplete_0_4.ngDoCheck(this,this._el_0,throwOnChange);
    this._AutoComplete_0_4.checkHost(this,this.compView_0,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
    if (!throwOnChange) {
      if ((this.numberOfChecks === 0)) { this._AutoComplete_0_4.context.ngAfterViewInit(); }
      this._AutoComplete_0_4.context.ngAfterViewChecked();
    }
  }
  destroyInternal():void {
    this.compView_0.destroy();
    this._AutoComplete_0_4.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
export const AutoCompleteNgFactory:import18.ComponentFactory<import1.AutoComplete> = new import18.ComponentFactory<import1.AutoComplete>('p-autoComplete',View_AutoComplete_Host0,import1.AutoComplete);
const styles_AutoComplete:any[] = ([] as any[]);
class View_AutoComplete1 extends import10.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import22.Wrapper_NgClass;
  _NgStyle_0_4:import23.Wrapper_NgStyle;
  _InputText_0_5:import24.Wrapper_InputText;
  /*private*/ _expr_4:any;
  /*private*/ _expr_5:any;
  /*private*/ _expr_6:any;
  /*private*/ _expr_7:any;
  /*private*/ _expr_8:any;
  /*private*/ _expr_9:any;
  _map_10:any;
  constructor(viewUtils:import11.ViewUtils,parentView:import10.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import25.ViewContainer) {
    super(View_AutoComplete1,renderType_AutoComplete,import16.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import17.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_4 = import9.UNINITIALIZED;
    this._expr_5 = import9.UNINITIALIZED;
    this._expr_6 = import9.UNINITIALIZED;
    this._expr_7 = import9.UNINITIALIZED;
    this._expr_8 = import9.UNINITIALIZED;
    this._expr_9 = import9.UNINITIALIZED;
    this._map_10 = import11.pureProxy2((p0:any,p1:any):{[key: string]:any} => {
      return {
        'ui-autocomplete-input': p0,
        'ui-autocomplete-dd-input': p1
      }
      ;
    });
  }
  createInternal(rootSelector:string):import18.ComponentRef<any> {
    this._el_0 = import11.createRenderElement(this.renderer,(null as any),'input',new import11.InlineArray8(6,'autocomplete','off','pInputText','','type','text'),(null as any));
    this._NgClass_0_3 = new import22.Wrapper_NgClass(this.parentView.parentView.injectorGet(import20.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import26.KeyValueDiffers,this.parentView.parentIndex),new import19.ElementRef(this._el_0),this.renderer);
    this._NgStyle_0_4 = new import23.Wrapper_NgStyle(this.parentView.parentView.injectorGet(import26.KeyValueDiffers,this.parentView.parentIndex),new import19.ElementRef(this._el_0),this.renderer);
    this._InputText_0_5 = new import24.Wrapper_InputText(new import19.ElementRef(this._el_0));
    var disposable_0:Function = import11.subscribeToRenderElement(this,this._el_0,new import11.InlineArray16(12,'input',(null as any),'keydown',(null as any),'focus',(null as any),'blur',(null as any),'mouseover',(null as any),'mouseout',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import27.NgClass) && (0 === requestNodeIndex))) { return this._NgClass_0_3.context; }
    if (((token === import28.NgStyle) && (0 === requestNodeIndex))) { return this._NgStyle_0_4.context; }
    if (((token === import3.InputText) && (0 === requestNodeIndex))) { return this._InputText_0_5.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this.parentView.context.inputStyleClass;
    this._NgClass_0_3.check_klass(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = this._map_10(true,this.parentView.context.dropdown);
    this._NgClass_0_3.check_ngClass(currVal_0_0_1,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_0_1_0:any = this.parentView.context.inputStyle;
    this._NgStyle_0_4.check_ngStyle(currVal_0_1_0,throwOnChange,false);
    this._NgStyle_0_4.ngDoCheck(this,this._el_0,throwOnChange);
    this._InputText_0_5.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_4:any = (this.parentView.context.value? (this.parentView.context.field? (this.parentView.context.resolveFieldData(this.parentView.context.value) || this.parentView.context.value): this.parentView.context.value): (null as any));
    if (import11.checkBinding(throwOnChange,this._expr_4,currVal_4)) {
      this.renderer.setElementProperty(this._el_0,'value',currVal_4);
      this._expr_4 = currVal_4;
    }
    const currVal_5:any = this.parentView.context.placeholder;
    if (import11.checkBinding(throwOnChange,this._expr_5,currVal_5)) {
      this.renderer.setElementAttribute(this._el_0,'placeholder',((currVal_5 == null)? (null as any): currVal_5.toString()));
      this._expr_5 = currVal_5;
    }
    const currVal_6:any = this.parentView.context.size;
    if (import11.checkBinding(throwOnChange,this._expr_6,currVal_6)) {
      this.renderer.setElementAttribute(this._el_0,'size',((currVal_6 == null)? (null as any): currVal_6.toString()));
      this._expr_6 = currVal_6;
    }
    const currVal_7:any = this.parentView.context.maxlength;
    if (import11.checkBinding(throwOnChange,this._expr_7,currVal_7)) {
      this.renderer.setElementAttribute(this._el_0,'maxlength',((currVal_7 == null)? (null as any): currVal_7.toString()));
      this._expr_7 = currVal_7;
    }
    const currVal_8:any = this.parentView.context.readonly;
    if (import11.checkBinding(throwOnChange,this._expr_8,currVal_8)) {
      this.renderer.setElementAttribute(this._el_0,'readonly',((currVal_8 == null)? (null as any): currVal_8.toString()));
      this._expr_8 = currVal_8;
    }
    const currVal_9:any = this.parentView.context.disabled;
    if (import11.checkBinding(throwOnChange,this._expr_9,currVal_9)) {
      this.renderer.setElementProperty(this._el_0,'disabled',currVal_9);
      this._expr_9 = currVal_9;
    }
    this._InputText_0_5.checkHost(this,this,this._el_0,throwOnChange);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    result = (this._InputText_0_5.handleEvent(eventName,$event) && result);
    if ((eventName == 'input')) {
      const pd_sub_0:any = ((<any>this.parentView.context.onInput($event)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'keydown')) {
      const pd_sub_1:any = ((<any>this.parentView.context.onKeydown($event)) !== false);
      result = (pd_sub_1 && result);
    }
    if ((eventName == 'focus')) {
      const pd_sub_2:any = ((<any>this.parentView.context.onFocus()) !== false);
      result = (pd_sub_2 && result);
    }
    if ((eventName == 'blur')) {
      const pd_sub_3:any = ((<any>this.parentView.context.onBlur()) !== false);
      result = (pd_sub_3 && result);
    }
    return result;
  }
}
class View_AutoComplete3 extends import10.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _el_4:any;
  _text_5:any;
  _text_6:any;
  /*private*/ _expr_7:any;
  constructor(viewUtils:import11.ViewUtils,parentView:import10.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import25.ViewContainer) {
    super(View_AutoComplete3,renderType_AutoComplete,import16.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import17.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_7 = import9.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import18.ComponentRef<any> {
    this._el_0 = import11.createRenderElement(this.renderer,(null as any),'li',new import11.InlineArray2(2,'class','ui-autocomplete-token ui-state-highlight ui-corner-all'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this._el_2 = import11.createRenderElement(this.renderer,this._el_0,'span',new import11.InlineArray2(2,'class','ui-autocomplete-token-icon fa fa-fw fa-close'),(null as any));
    this._text_3 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this._el_4 = import11.createRenderElement(this.renderer,this._el_0,'span',new import11.InlineArray2(2,'class','ui-autocomplete-token-label'),(null as any));
    this._text_5 = this.renderer.createText(this._el_4,'',(null as any));
    this._text_6 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    var disposable_0:Function = import11.subscribeToRenderElement(this,this._el_2,new import11.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_2));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._el_4,
      this._text_5,
      this._text_6
    ]
    ),[disposable_0]);
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_7:any = import11.inlineInterpolate(1,'',(this.parentView.parentView.context.field? this.context.$implicit[this.parentView.parentView.context.field]: this.context.$implicit),'');
    if (import11.checkBinding(throwOnChange,this._expr_7,currVal_7)) {
      this.renderer.setText(this._text_5,currVal_7);
      this._expr_7 = currVal_7;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  handleEvent_2(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.parentView.context.removeItem(this._el_0)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
class View_AutoComplete2 extends import10.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import22.Wrapper_NgClass;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import25.ViewContainer;
  _TemplateRef_2_5:any;
  _NgFor_2_6:import29.Wrapper_NgFor;
  _text_3:any;
  _el_4:any;
  _text_5:any;
  _el_6:any;
  _InputText_6_3:import24.Wrapper_InputText;
  _text_7:any;
  _text_8:any;
  _map_14:any;
  /*private*/ _expr_15:any;
  /*private*/ _expr_16:any;
  constructor(viewUtils:import11.ViewUtils,parentView:import10.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import25.ViewContainer) {
    super(View_AutoComplete2,renderType_AutoComplete,import16.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import17.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_14 = import11.pureProxy1((p0:any):{[key: string]:any} => {
      return {'ui-state-disabled': p0};
    });
    this._expr_15 = import9.UNINITIALIZED;
    this._expr_16 = import9.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import18.ComponentRef<any> {
    this._el_0 = import11.createRenderElement(this.renderer,(null as any),'ul',new import11.InlineArray2(2,'class','ui-autocomplete-multiple-container ui-widget ui-inputtext ui-state-default ui-corner-all'),(null as any));
    this._NgClass_0_3 = new import22.Wrapper_NgClass(this.parentView.parentView.injectorGet(import20.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import26.KeyValueDiffers,this.parentView.parentIndex),new import19.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import25.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import30.TemplateRef_(this,2,this._anchor_2);
    this._NgFor_2_6 = new import29.Wrapper_NgFor(this._vc_2.vcRef,this._TemplateRef_2_5,this.parentView.parentView.injectorGet(import20.IterableDiffers,this.parentView.parentIndex),this.parentView.ref);
    this._text_3 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    this._el_4 = import11.createRenderElement(this.renderer,this._el_0,'li',new import11.InlineArray2(2,'class','ui-autocomplete-input-token'),(null as any));
    this._text_5 = this.renderer.createText(this._el_4,'\n                    ',(null as any));
    this._el_6 = import11.createRenderElement(this.renderer,this._el_4,'input',new import11.InlineArray8(6,'autocomplete','off','pInputText','','type','text'),(null as any));
    this._InputText_6_3 = new import24.Wrapper_InputText(new import19.ElementRef(this._el_6));
    this._text_7 = this.renderer.createText(this._el_4,'\n                ',(null as any));
    this._text_8 = this.renderer.createText(this._el_0,'\n            ',(null as any));
    var disposable_0:Function = import11.subscribeToRenderElement(this,this._el_0,new import11.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_0));
    var disposable_1:Function = import11.subscribeToRenderElement(this,this._el_6,new import11.InlineArray16(12,'input',(null as any),'keydown',(null as any),'focus',(null as any),'blur',(null as any),'mouseover',(null as any),'mouseout',(null as any)),this.eventHandler(this.handleEvent_6));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3,
      this._el_4,
      this._text_5,
      this._el_6,
      this._text_7,
      this._text_8
    ]
    ),[
      disposable_0,
      disposable_1
    ]
    );
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import30.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import31.NgFor) && (2 === requestNodeIndex))) { return this._NgFor_2_6.context; }
    if (((token === import3.InputText) && (6 === requestNodeIndex))) { return this._InputText_6_3.context; }
    if (((token === import27.NgClass) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 8)))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = 'ui-autocomplete-multiple-container ui-widget ui-inputtext ui-state-default ui-corner-all';
    this._NgClass_0_3.check_klass(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = this._map_14(this.parentView.context.disabled);
    this._NgClass_0_3.check_ngClass(currVal_0_0_1,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_2_0_0:any = this.parentView.context.value;
    this._NgFor_2_6.check_ngForOf(currVal_2_0_0,throwOnChange,false);
    this._NgFor_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    this._InputText_6_3.ngDoCheck(this,this._el_6,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
    const currVal_15:any = this.parentView.context.disabled;
    if (import11.checkBinding(throwOnChange,this._expr_15,currVal_15)) {
      this.renderer.setElementProperty(this._el_6,'disabled',currVal_15);
      this._expr_15 = currVal_15;
    }
    const currVal_16:any = this.parentView.context.placeholder;
    if (import11.checkBinding(throwOnChange,this._expr_16,currVal_16)) {
      this.renderer.setElementAttribute(this._el_6,'placeholder',((currVal_16 == null)? (null as any): currVal_16.toString()));
      this._expr_16 = currVal_16;
    }
    this._InputText_6_3.checkHost(this,this,this._el_6,throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import10.AppView<any> {
    if ((nodeIndex == 2)) { return new View_AutoComplete3(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    return (null as any);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this._el_6.focus()) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
  handleEvent_6(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    result = (this._InputText_6_3.handleEvent(eventName,$event) && result);
    if ((eventName == 'input')) {
      const pd_sub_0:any = ((<any>this.parentView.context.onInput($event)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'keydown')) {
      const pd_sub_1:any = ((<any>this.parentView.context.onKeydown($event)) !== false);
      result = (pd_sub_1 && result);
    }
    if ((eventName == 'focus')) {
      const pd_sub_2:any = ((<any>this.parentView.context.onFocus()) !== false);
      result = (pd_sub_2 && result);
    }
    if ((eventName == 'blur')) {
      const pd_sub_3:any = ((<any>this.parentView.context.onBlur()) !== false);
      result = (pd_sub_3 && result);
    }
    return result;
  }
}
class View_AutoComplete4 extends import10.AppView<any> {
  _el_0:any;
  _DomHandler_0_3:import14.DomHandler;
  _Button_0_4:import32.Wrapper_Button;
  /*private*/ _expr_3:any;
  constructor(viewUtils:import11.ViewUtils,parentView:import10.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import25.ViewContainer) {
    super(View_AutoComplete4,renderType_AutoComplete,import16.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import17.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_3 = import9.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import18.ComponentRef<any> {
    this._el_0 = import11.createRenderElement(this.renderer,(null as any),'button',new import11.InlineArray8(8,'class','ui-autocomplete-dropdown','icon','fa-fw fa-caret-down','pButton','','type','button'),(null as any));
    this._DomHandler_0_3 = new import14.DomHandler();
    this._Button_0_4 = new import32.Wrapper_Button(new import19.ElementRef(this._el_0),this._DomHandler_0_3);
    var disposable_0:Function = import11.subscribeToRenderElement(this,this._el_0,new import11.InlineArray16(14,'click',(null as any),'focus',(null as any),'blur',(null as any),'mouseenter',(null as any),'mouseleave',(null as any),'mousedown',(null as any),'mouseup',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import14.DomHandler) && (0 === requestNodeIndex))) { return this._DomHandler_0_3; }
    if (((token === import4.Button) && (0 === requestNodeIndex))) { return this._Button_0_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = 'fa-fw fa-caret-down';
    this._Button_0_4.check_icon(currVal_0_0_0,throwOnChange,false);
    this._Button_0_4.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_3:any = this.parentView.context.disabled;
    if (import11.checkBinding(throwOnChange,this._expr_3,currVal_3)) {
      this.renderer.setElementProperty(this._el_0,'disabled',currVal_3);
      this._expr_3 = currVal_3;
    }
    this._Button_0_4.checkHost(this,this,this._el_0,throwOnChange);
    if (!throwOnChange) { if ((this.numberOfChecks === 0)) { this._Button_0_4.context.ngAfterViewInit(); } }
  }
  destroyInternal():void {
    this._Button_0_4.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    result = (this._Button_0_4.handleEvent(eventName,$event) && result);
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.context.handleDropdownClick($event)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'focus')) {
      const pd_sub_1:any = ((<any>this.parentView.context.onDropdownFocus($event)) !== false);
      result = (pd_sub_1 && result);
    }
    if ((eventName == 'blur')) {
      const pd_sub_2:any = ((<any>this.parentView.context.onDropdownBlur($event)) !== false);
      result = (pd_sub_2 && result);
    }
    return result;
  }
}
class View_AutoComplete6 extends import10.AppView<any> {
  _el_0:any;
  _text_1:any;
  /*private*/ _expr_2:any;
  constructor(viewUtils:import11.ViewUtils,parentView:import10.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import25.ViewContainer) {
    super(View_AutoComplete6,renderType_AutoComplete,import16.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import17.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_2 = import9.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import18.ComponentRef<any> {
    this._el_0 = import11.createRenderElement(this.renderer,(null as any),'span',import11.EMPTY_INLINE_ARRAY,(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1
    ]
    ),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2:any = import11.inlineInterpolate(1,'',(this.parentView.parentView.context.field? this.parentView.context.$implicit[this.parentView.parentView.context.field]: this.parentView.context.$implicit),'');
    if (import11.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setText(this._text_1,currVal_2);
      this._expr_2 = currVal_2;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_AutoComplete8 extends import10.AppView<any> {
  _el_0:any;
  constructor(viewUtils:import11.ViewUtils,parentView:import10.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import25.ViewContainer) {
    super(View_AutoComplete8,renderType_AutoComplete,import16.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import17.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import18.ComponentRef<any> {
    this._el_0 = this.renderer.createTemplateAnchor((null as any),(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): ([] as any[])),(null as any));
    return (null as any);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_AutoComplete7 extends import10.AppView<any> {
  _anchor_0:any;
  /*private*/ _vc_0:import25.ViewContainer;
  _TemplateRef_0_5:any;
  _TemplateWrapper_0_6:import33.Wrapper_TemplateWrapper;
  _el_1:any;
  constructor(viewUtils:import11.ViewUtils,parentView:import10.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import25.ViewContainer) {
    super(View_AutoComplete7,renderType_AutoComplete,import16.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import17.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import18.ComponentRef<any> {
    this._anchor_0 = this.renderer.createTemplateAnchor((null as any),(null as any));
    this._vc_0 = new import25.ViewContainer(0,(null as any),this,this._anchor_0);
    this._TemplateRef_0_5 = new import30.TemplateRef_(this,0,this._anchor_0);
    this._TemplateWrapper_0_6 = new import33.Wrapper_TemplateWrapper(this._vc_0.vcRef);
    this._el_1 = this.renderer.createTemplateAnchor((null as any),(null as any));
    this.init(this._el_1,((<any>this.renderer).directRenderer? (null as any): [this._anchor_0]),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import30.TemplateRef) && (0 === requestNodeIndex))) { return this._TemplateRef_0_5; }
    if (((token === import5.TemplateWrapper) && (0 === requestNodeIndex))) { return this._TemplateWrapper_0_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this.parentView.context.$implicit;
    this._TemplateWrapper_0_6.check_item(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = this.parentView.parentView.context.itemTemplate;
    this._TemplateWrapper_0_6.check_templateRef(currVal_0_0_1,throwOnChange,false);
    this._TemplateWrapper_0_6.ngDoCheck(this,this._anchor_0,throwOnChange);
    this._vc_0.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_0.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._vc_0.nativeElement,ctx);
    this._vc_0.visitNestedViewRootNodes(cb,ctx);
    cb(this._el_1,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import10.AppView<any> {
    if ((nodeIndex == 0)) { return new View_AutoComplete8(this.viewUtils,this,0,this._anchor_0,this._vc_0); }
    return (null as any);
  }
}
class View_AutoComplete5 extends import10.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import22.Wrapper_NgClass;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import25.ViewContainer;
  _TemplateRef_2_5:any;
  _NgIf_2_6:import34.Wrapper_NgIf;
  _text_3:any;
  _anchor_4:any;
  /*private*/ _vc_4:import25.ViewContainer;
  _TemplateRef_4_5:any;
  _NgIf_4_6:import34.Wrapper_NgIf;
  _text_5:any;
  _map_13:any;
  constructor(viewUtils:import11.ViewUtils,parentView:import10.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import25.ViewContainer) {
    super(View_AutoComplete5,renderType_AutoComplete,import16.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import17.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_13 = import11.pureProxy2((p0:any,p1:any):{[key: string]:any} => {
      return {
        'ui-autocomplete-list-item ui-corner-all': p0,
        'ui-state-highlight': p1
      }
      ;
    });
  }
  createInternal(rootSelector:string):import18.ComponentRef<any> {
    this._el_0 = import11.createRenderElement(this.renderer,(null as any),'li',import11.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_0_3 = new import22.Wrapper_NgClass(this.parentView.parentView.injectorGet(import20.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import26.KeyValueDiffers,this.parentView.parentIndex),new import19.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n                        ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import25.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import30.TemplateRef_(this,2,this._anchor_2);
    this._NgIf_2_6 = new import34.Wrapper_NgIf(this._vc_2.vcRef,this._TemplateRef_2_5);
    this._text_3 = this.renderer.createText(this._el_0,'\n                        ',(null as any));
    this._anchor_4 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_4 = new import25.ViewContainer(4,0,this,this._anchor_4);
    this._TemplateRef_4_5 = new import30.TemplateRef_(this,4,this._anchor_4);
    this._NgIf_4_6 = new import34.Wrapper_NgIf(this._vc_4.vcRef,this._TemplateRef_4_5);
    this._text_5 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    var disposable_0:Function = import11.subscribeToRenderElement(this,this._el_0,new import11.InlineArray8(6,'mouseenter',(null as any),'mouseleave',(null as any),'click',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3,
      this._anchor_4,
      this._text_5
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import30.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import35.NgIf) && (2 === requestNodeIndex))) { return this._NgIf_2_6.context; }
    if (((token === import30.TemplateRef) && (4 === requestNodeIndex))) { return this._TemplateRef_4_5; }
    if (((token === import35.NgIf) && (4 === requestNodeIndex))) { return this._NgIf_4_6.context; }
    if (((token === import27.NgClass) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 5)))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this._map_13(true,(this.parentView.context.highlightOption == this.context.$implicit));
    this._NgClass_0_3.check_ngClass(currVal_0_0_0,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_2_0_0:boolean = !this.parentView.context.itemTemplate;
    this._NgIf_2_6.check_ngIf(currVal_2_0_0,throwOnChange,false);
    this._NgIf_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    const currVal_4_0_0:any = this.parentView.context.itemTemplate;
    this._NgIf_4_6.check_ngIf(currVal_4_0_0,throwOnChange,false);
    this._NgIf_4_6.ngDoCheck(this,this._anchor_4,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
    this._vc_4.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
    this._vc_4.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import10.AppView<any> {
    if ((nodeIndex == 2)) { return new View_AutoComplete6(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    if ((nodeIndex == 4)) { return new View_AutoComplete7(this.viewUtils,this,4,this._anchor_4,this._vc_4); }
    return (null as any);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'mouseenter')) {
      const pd_sub_0:any = ((<any>(this.parentView.context.highlightOption = this.context.$implicit)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'mouseleave')) {
      const pd_sub_1:any = ((<any>(this.parentView.context.highlightOption = (null as any))) !== false);
      result = (pd_sub_1 && result);
    }
    if ((eventName == 'click')) {
      const pd_sub_2:any = ((<any>this.parentView.context.selectItem(this.context.$implicit)) !== false);
      result = (pd_sub_2 && result);
    }
    return result;
  }
}
var renderType_AutoComplete:import12.RenderComponentType = import11.createRenderComponentType('',0,import13.ViewEncapsulation.None,styles_AutoComplete,{});
export class View_AutoComplete0 extends import10.AppView<import1.AutoComplete> {
  _text_0:any;
  _el_1:any;
  _NgClass_1_3:import22.Wrapper_NgClass;
  _NgStyle_1_4:import23.Wrapper_NgStyle;
  _text_2:any;
  _anchor_3:any;
  /*private*/ _vc_3:import25.ViewContainer;
  _TemplateRef_3_5:any;
  _NgIf_3_6:import34.Wrapper_NgIf;
  _anchor_4:any;
  /*private*/ _vc_4:import25.ViewContainer;
  _TemplateRef_4_5:any;
  _NgIf_4_6:import34.Wrapper_NgIf;
  _anchor_5:any;
  /*private*/ _vc_5:import25.ViewContainer;
  _TemplateRef_5_5:any;
  _NgIf_5_6:import34.Wrapper_NgIf;
  _text_6:any;
  _el_7:any;
  _text_8:any;
  _el_9:any;
  _text_10:any;
  _anchor_11:any;
  /*private*/ _vc_11:import25.ViewContainer;
  _TemplateRef_11_5:any;
  _NgFor_11_6:import29.Wrapper_NgFor;
  _text_12:any;
  _text_13:any;
  _text_14:any;
  _text_15:any;
  _map_30:any;
  /*private*/ _expr_31:any;
  /*private*/ _expr_32:any;
  /*private*/ _expr_33:any;
  constructor(viewUtils:import11.ViewUtils,parentView:import10.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_AutoComplete0,renderType_AutoComplete,import16.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import17.ChangeDetectorStatus.CheckAlways);
    this._map_30 = import11.pureProxy3((p0:any,p1:any,p2:any):{[key: string]:any} => {
      return {
        'ui-autocomplete ui-widget': p0,
        'ui-autocomplete-dd': p1,
        'ui-autocomplete-multiple': p2
      }
      ;
    });
    this._expr_31 = import9.UNINITIALIZED;
    this._expr_32 = import9.UNINITIALIZED;
    this._expr_33 = import9.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import18.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n        ',(null as any));
    this._el_1 = import11.createRenderElement(this.renderer,parentRenderNode,'span',import11.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_1_3 = new import22.Wrapper_NgClass(this.parentView.injectorGet(import20.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import26.KeyValueDiffers,this.parentIndex),new import19.ElementRef(this._el_1),this.renderer);
    this._NgStyle_1_4 = new import23.Wrapper_NgStyle(this.parentView.injectorGet(import26.KeyValueDiffers,this.parentIndex),new import19.ElementRef(this._el_1),this.renderer);
    this._text_2 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._anchor_3 = this.renderer.createTemplateAnchor(this._el_1,(null as any));
    this._vc_3 = new import25.ViewContainer(3,1,this,this._anchor_3);
    this._TemplateRef_3_5 = new import30.TemplateRef_(this,3,this._anchor_3);
    this._NgIf_3_6 = new import34.Wrapper_NgIf(this._vc_3.vcRef,this._TemplateRef_3_5);
    this._anchor_4 = this.renderer.createTemplateAnchor(this._el_1,(null as any));
    this._vc_4 = new import25.ViewContainer(4,1,this,this._anchor_4);
    this._TemplateRef_4_5 = new import30.TemplateRef_(this,4,this._anchor_4);
    this._NgIf_4_6 = new import34.Wrapper_NgIf(this._vc_4.vcRef,this._TemplateRef_4_5);
    this._anchor_5 = this.renderer.createTemplateAnchor(this._el_1,(null as any));
    this._vc_5 = new import25.ViewContainer(5,1,this,this._anchor_5);
    this._TemplateRef_5_5 = new import30.TemplateRef_(this,5,this._anchor_5);
    this._NgIf_5_6 = new import34.Wrapper_NgIf(this._vc_5.vcRef,this._TemplateRef_5_5);
    this._text_6 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_7 = import11.createRenderElement(this.renderer,this._el_1,'div',new import11.InlineArray2(2,'class','ui-autocomplete-panel ui-widget-content ui-corner-all ui-shadow'),(null as any));
    this._text_8 = this.renderer.createText(this._el_7,'\n                ',(null as any));
    this._el_9 = import11.createRenderElement(this.renderer,this._el_7,'ul',new import11.InlineArray2(2,'class','ui-autocomplete-items ui-autocomplete-list ui-widget-content ui-widget ui-corner-all ui-helper-reset'),(null as any));
    this._text_10 = this.renderer.createText(this._el_9,'\n                    ',(null as any));
    this._anchor_11 = this.renderer.createTemplateAnchor(this._el_9,(null as any));
    this._vc_11 = new import25.ViewContainer(11,9,this,this._anchor_11);
    this._TemplateRef_11_5 = new import30.TemplateRef_(this,11,this._anchor_11);
    this._NgFor_11_6 = new import29.Wrapper_NgFor(this._vc_11.vcRef,this._TemplateRef_11_5,this.parentView.injectorGet(import20.IterableDiffers,this.parentIndex),this.ref);
    this._text_12 = this.renderer.createText(this._el_9,'\n                ',(null as any));
    this._text_13 = this.renderer.createText(this._el_7,'\n            ',(null as any));
    this._text_14 = this.renderer.createText(this._el_1,'\n        ',(null as any));
    this._text_15 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2,
      this._anchor_3,
      this._anchor_4,
      this._anchor_5,
      this._text_6,
      this._el_7,
      this._text_8,
      this._el_9,
      this._text_10,
      this._anchor_11,
      this._text_12,
      this._text_13,
      this._text_14,
      this._text_15
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import30.TemplateRef) && (3 === requestNodeIndex))) { return this._TemplateRef_3_5; }
    if (((token === import35.NgIf) && (3 === requestNodeIndex))) { return this._NgIf_3_6.context; }
    if (((token === import30.TemplateRef) && (4 === requestNodeIndex))) { return this._TemplateRef_4_5; }
    if (((token === import35.NgIf) && (4 === requestNodeIndex))) { return this._NgIf_4_6.context; }
    if (((token === import30.TemplateRef) && (5 === requestNodeIndex))) { return this._TemplateRef_5_5; }
    if (((token === import35.NgIf) && (5 === requestNodeIndex))) { return this._NgIf_5_6.context; }
    if (((token === import30.TemplateRef) && (11 === requestNodeIndex))) { return this._TemplateRef_11_5; }
    if (((token === import31.NgFor) && (11 === requestNodeIndex))) { return this._NgFor_11_6.context; }
    if (((token === import27.NgClass) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 14)))) { return this._NgClass_1_3.context; }
    if (((token === import28.NgStyle) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 14)))) { return this._NgStyle_1_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1_0_0:any = this.context.styleClass;
    this._NgClass_1_3.check_klass(currVal_1_0_0,throwOnChange,false);
    const currVal_1_0_1:any = this._map_30(true,this.context.dropdown,this.context.multiple);
    this._NgClass_1_3.check_ngClass(currVal_1_0_1,throwOnChange,false);
    this._NgClass_1_3.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_1_1_0:any = this.context.style;
    this._NgStyle_1_4.check_ngStyle(currVal_1_1_0,throwOnChange,false);
    this._NgStyle_1_4.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_3_0_0:boolean = !this.context.multiple;
    this._NgIf_3_6.check_ngIf(currVal_3_0_0,throwOnChange,false);
    this._NgIf_3_6.ngDoCheck(this,this._anchor_3,throwOnChange);
    const currVal_4_0_0:any = this.context.multiple;
    this._NgIf_4_6.check_ngIf(currVal_4_0_0,throwOnChange,false);
    this._NgIf_4_6.ngDoCheck(this,this._anchor_4,throwOnChange);
    const currVal_5_0_0:any = this.context.dropdown;
    this._NgIf_5_6.check_ngIf(currVal_5_0_0,throwOnChange,false);
    this._NgIf_5_6.ngDoCheck(this,this._anchor_5,throwOnChange);
    const currVal_11_0_0:any = this.context.suggestions;
    this._NgFor_11_6.check_ngForOf(currVal_11_0_0,throwOnChange,false);
    this._NgFor_11_6.ngDoCheck(this,this._anchor_11,throwOnChange);
    this._vc_3.detectChangesInNestedViews(throwOnChange);
    this._vc_4.detectChangesInNestedViews(throwOnChange);
    this._vc_5.detectChangesInNestedViews(throwOnChange);
    this._vc_11.detectChangesInNestedViews(throwOnChange);
    const currVal_31:any = (this.context.panelVisible? 'block': 'none');
    if (import11.checkBinding(throwOnChange,this._expr_31,currVal_31)) {
      this.renderer.setElementStyle(this._el_7,'display',((this.viewUtils.sanitizer.sanitize(import36.SecurityContext.STYLE,currVal_31) == null)? (null as any): this.viewUtils.sanitizer.sanitize(import36.SecurityContext.STYLE,currVal_31).toString()));
      this._expr_31 = currVal_31;
    }
    const currVal_32:any = (this.context.appendTo? 'auto': '100%');
    if (import11.checkBinding(throwOnChange,this._expr_32,currVal_32)) {
      this.renderer.setElementStyle(this._el_7,'width',((this.viewUtils.sanitizer.sanitize(import36.SecurityContext.STYLE,currVal_32) == null)? (null as any): this.viewUtils.sanitizer.sanitize(import36.SecurityContext.STYLE,currVal_32).toString()));
      this._expr_32 = currVal_32;
    }
    const currVal_33:any = this.context.scrollHeight;
    if (import11.checkBinding(throwOnChange,this._expr_33,currVal_33)) {
      this.renderer.setElementStyle(this._el_7,'max-height',((this.viewUtils.sanitizer.sanitize(import36.SecurityContext.STYLE,currVal_33) == null)? (null as any): this.viewUtils.sanitizer.sanitize(import36.SecurityContext.STYLE,currVal_33).toString()));
      this._expr_33 = currVal_33;
    }
  }
  destroyInternal():void {
    this._vc_3.destroyNestedViews();
    this._vc_4.destroyNestedViews();
    this._vc_5.destroyNestedViews();
    this._vc_11.destroyNestedViews();
  }
  createEmbeddedViewInternal(nodeIndex:number):import10.AppView<any> {
    if ((nodeIndex == 3)) { return new View_AutoComplete1(this.viewUtils,this,3,this._anchor_3,this._vc_3); }
    if ((nodeIndex == 4)) { return new View_AutoComplete2(this.viewUtils,this,4,this._anchor_4,this._vc_4); }
    if ((nodeIndex == 5)) { return new View_AutoComplete4(this.viewUtils,this,5,this._anchor_5,this._vc_5); }
    if ((nodeIndex == 11)) { return new View_AutoComplete5(this.viewUtils,this,11,this._anchor_11,this._vc_11); }
    return (null as any);
  }
}
