// ng2-slim-loading-bar/src/slim-loading-bar.component.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
import * as import0 from 'ng2-slim-loading-bar/src/slim-loading-bar.component';
import * as import1 from '@angular/core/src/change_detection/change_detection_util';
import * as import2 from '@angular/core/src/linker/view';
import * as import3 from '@angular/core/src/linker/view_utils';
import * as import5 from '@angular/core/src/metadata/view';
import * as import6 from '@angular/core/src/linker/view_type';
import * as import7 from '@angular/core/src/change_detection/constants';
import * as import8 from '@angular/core/src/linker/component_factory';
import * as import9 from 'ng2-slim-loading-bar/src/slim-loading-bar.service';
import * as import10 from '@angular/core/src/security';
export var Wrapper_SlimLoadingBarComponent = (function () {
    function Wrapper_SlimLoadingBarComponent(p0) {
        this._changed = false;
        this.context = new import0.SlimLoadingBarComponent(p0);
        this._expr_0 = import1.UNINITIALIZED;
        this._expr_1 = import1.UNINITIALIZED;
        this._expr_2 = import1.UNINITIALIZED;
        this._expr_3 = import1.UNINITIALIZED;
    }
    Wrapper_SlimLoadingBarComponent.prototype.ngOnDetach = function (view, componentView, el) {
    };
    Wrapper_SlimLoadingBarComponent.prototype.ngOnDestroy = function () {
    };
    Wrapper_SlimLoadingBarComponent.prototype.check_progress = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_0, currValue))) {
            this._changed = true;
            this.context.progress = currValue;
            this._expr_0 = currValue;
        }
    };
    Wrapper_SlimLoadingBarComponent.prototype.check_color = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_1, currValue))) {
            this._changed = true;
            this.context.color = currValue;
            this._expr_1 = currValue;
        }
    };
    Wrapper_SlimLoadingBarComponent.prototype.check_height = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_2, currValue))) {
            this._changed = true;
            this.context.height = currValue;
            this._expr_2 = currValue;
        }
    };
    Wrapper_SlimLoadingBarComponent.prototype.check_show = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_3, currValue))) {
            this._changed = true;
            this.context.show = currValue;
            this._expr_3 = currValue;
        }
    };
    Wrapper_SlimLoadingBarComponent.prototype.ngDoCheck = function (view, el, throwOnChange) {
        var changed = this._changed;
        this._changed = false;
        if (!throwOnChange) {
            if ((view.numberOfChecks === 0)) {
                this.context.ngOnInit();
            }
        }
        return changed;
    };
    Wrapper_SlimLoadingBarComponent.prototype.checkHost = function (view, componentView, el, throwOnChange) {
    };
    Wrapper_SlimLoadingBarComponent.prototype.handleEvent = function (eventName, $event) {
        var result = true;
        return result;
    };
    Wrapper_SlimLoadingBarComponent.prototype.subscribe = function (view, _eventHandler) {
        this._eventHandler = _eventHandler;
    };
    return Wrapper_SlimLoadingBarComponent;
}());
var renderType_SlimLoadingBarComponent_Host = import3.createRenderComponentType('', 0, import5.ViewEncapsulation.None, [], {});
var View_SlimLoadingBarComponent_Host0 = (function (_super) {
    __extends(View_SlimLoadingBarComponent_Host0, _super);
    function View_SlimLoadingBarComponent_Host0(viewUtils, parentView, parentIndex, parentElement) {
        _super.call(this, View_SlimLoadingBarComponent_Host0, renderType_SlimLoadingBarComponent_Host, import6.ViewType.HOST, viewUtils, parentView, parentIndex, parentElement, import7.ChangeDetectorStatus.CheckAlways);
    }
    View_SlimLoadingBarComponent_Host0.prototype.createInternal = function (rootSelector) {
        this._el_0 = import3.selectOrCreateRenderHostElement(this.renderer, 'ng2-slim-loading-bar', import3.EMPTY_INLINE_ARRAY, rootSelector, null);
        this.compView_0 = new View_SlimLoadingBarComponent0(this.viewUtils, this, 0, this._el_0);
        this._SlimLoadingBarComponent_0_3 = new Wrapper_SlimLoadingBarComponent(this.injectorGet(import9.SlimLoadingBarService, this.parentIndex));
        this.compView_0.create(this._SlimLoadingBarComponent_0_3.context);
        this.init(this._el_0, (this.renderer.directRenderer ? null : [this._el_0]), null);
        return new import8.ComponentRef_(0, this, this._el_0, this._SlimLoadingBarComponent_0_3.context);
    };
    View_SlimLoadingBarComponent_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import0.SlimLoadingBarComponent) && (0 === requestNodeIndex))) {
            return this._SlimLoadingBarComponent_0_3.context;
        }
        return notFoundResult;
    };
    View_SlimLoadingBarComponent_Host0.prototype.detectChangesInternal = function (throwOnChange) {
        this._SlimLoadingBarComponent_0_3.ngDoCheck(this, this._el_0, throwOnChange);
        this.compView_0.internalDetectChanges(throwOnChange);
    };
    View_SlimLoadingBarComponent_Host0.prototype.destroyInternal = function () {
        this.compView_0.destroy();
    };
    View_SlimLoadingBarComponent_Host0.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    return View_SlimLoadingBarComponent_Host0;
}(import2.AppView));
export var SlimLoadingBarComponentNgFactory = new import8.ComponentFactory('ng2-slim-loading-bar', View_SlimLoadingBarComponent_Host0, import0.SlimLoadingBarComponent);
var styles_SlimLoadingBarComponent = [];
var renderType_SlimLoadingBarComponent = import3.createRenderComponentType('', 0, import5.ViewEncapsulation.None, styles_SlimLoadingBarComponent, {});
export var View_SlimLoadingBarComponent0 = (function (_super) {
    __extends(View_SlimLoadingBarComponent0, _super);
    function View_SlimLoadingBarComponent0(viewUtils, parentView, parentIndex, parentElement) {
        _super.call(this, View_SlimLoadingBarComponent0, renderType_SlimLoadingBarComponent, import6.ViewType.COMPONENT, viewUtils, parentView, parentIndex, parentElement, import7.ChangeDetectorStatus.CheckAlways);
        this._expr_5 = import1.UNINITIALIZED;
        this._expr_6 = import1.UNINITIALIZED;
        this._expr_7 = import1.UNINITIALIZED;
        this._expr_8 = import1.UNINITIALIZED;
        this._expr_9 = import1.UNINITIALIZED;
    }
    View_SlimLoadingBarComponent0.prototype.createInternal = function (rootSelector) {
        var parentRenderNode = this.renderer.createViewRoot(this.parentElement);
        this._text_0 = this.renderer.createText(parentRenderNode, '\n', null);
        this._el_1 = import3.createRenderElement(this.renderer, parentRenderNode, 'div', new import3.InlineArray2(2, 'class', 'slim-loading-bar'), null);
        this._text_2 = this.renderer.createText(this._el_1, '\n    ', null);
        this._el_3 = import3.createRenderElement(this.renderer, this._el_1, 'div', new import3.InlineArray2(2, 'class', 'slim-loading-bar-progress'), null);
        this._text_4 = this.renderer.createText(this._el_1, '\n', null);
        this.init(null, (this.renderer.directRenderer ? null : [
            this._text_0,
            this._el_1,
            this._text_2,
            this._el_3,
            this._text_4
        ]), null);
        return null;
    };
    View_SlimLoadingBarComponent0.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_5 = (this.context.progress + '%');
        if (import3.checkBinding(throwOnChange, this._expr_5, currVal_5)) {
            this.renderer.setElementStyle(this._el_3, 'width', ((this.viewUtils.sanitizer.sanitize(import10.SecurityContext.STYLE, currVal_5) == null) ? null : this.viewUtils.sanitizer.sanitize(import10.SecurityContext.STYLE, currVal_5).toString()));
            this._expr_5 = currVal_5;
        }
        var currVal_6 = this.context.color;
        if (import3.checkBinding(throwOnChange, this._expr_6, currVal_6)) {
            this.renderer.setElementStyle(this._el_3, 'backgroundColor', ((this.viewUtils.sanitizer.sanitize(import10.SecurityContext.STYLE, currVal_6) == null) ? null : this.viewUtils.sanitizer.sanitize(import10.SecurityContext.STYLE, currVal_6).toString()));
            this._expr_6 = currVal_6;
        }
        var currVal_7 = this.context.color;
        if (import3.checkBinding(throwOnChange, this._expr_7, currVal_7)) {
            this.renderer.setElementStyle(this._el_3, 'color', ((this.viewUtils.sanitizer.sanitize(import10.SecurityContext.STYLE, currVal_7) == null) ? null : this.viewUtils.sanitizer.sanitize(import10.SecurityContext.STYLE, currVal_7).toString()));
            this._expr_7 = currVal_7;
        }
        var currVal_8 = this.context.height;
        if (import3.checkBinding(throwOnChange, this._expr_8, currVal_8)) {
            this.renderer.setElementStyle(this._el_3, 'height', ((this.viewUtils.sanitizer.sanitize(import10.SecurityContext.STYLE, currVal_8) == null) ? null : this.viewUtils.sanitizer.sanitize(import10.SecurityContext.STYLE, currVal_8).toString()));
            this._expr_8 = currVal_8;
        }
        var currVal_9 = (this.context.show ? '1' : '0');
        if (import3.checkBinding(throwOnChange, this._expr_9, currVal_9)) {
            this.renderer.setElementStyle(this._el_3, 'opacity', ((this.viewUtils.sanitizer.sanitize(import10.SecurityContext.STYLE, currVal_9) == null) ? null : this.viewUtils.sanitizer.sanitize(import10.SecurityContext.STYLE, currVal_9).toString()));
            this._expr_9 = currVal_9;
        }
    };
    return View_SlimLoadingBarComponent0;
}(import2.AppView));
