// primeng/components/dialog/dialog.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from '@angular/core/src/linker/ng_module_factory';
import * as import1 from 'primeng/components/dialog/dialog';
import * as import2 from '@angular/common/src/common_module';
import * as import3 from 'primeng/components/common/shared';
import * as import4 from '@angular/common/src/localization';
import * as import5 from '@angular/core/src/di/injector';
import * as import6 from '@angular/core/src/i18n/tokens';
import * as import7 from '@angular/core/src/change_detection/change_detection_util';
import * as import8 from '@angular/core/src/linker/view';
import * as import9 from '@angular/core/src/linker/view_utils';
import * as import10 from '@angular/core/src/render/api';
import * as import11 from '@angular/core/src/metadata/view';
import * as import12 from 'primeng/components/dom/domhandler';
import * as import13 from '@angular/core/src/linker/query_list';
import * as import14 from '@angular/core/src/linker/view_type';
import * as import15 from '@angular/core/src/change_detection/constants';
import * as import16 from '@angular/core/src/linker/component_factory';
import * as import17 from '@angular/core/src/linker/element_ref';
import * as import18 from '@angular/core/src/animation/animation_transition';
import * as import19 from '@angular/core/src/animation/animation_sequence_player';
import * as import20 from '@angular/core/src/animation/animation_styles';
import * as import21 from '@angular/core/src/animation/animation_style_util';
import * as import22 from '@angular/core/src/animation/animation_keyframe';
import * as import23 from '@angular/core/src/animation/animation_player';
import * as import24 from '@angular/core/src/linker/view_container';
import * as import25 from '../../../@angular/common/src/directives/ng_class.ngfactory';
import * as import26 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import27 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import28 from '@angular/common/src/directives/ng_class';
import * as import29 from '../../../@angular/common/src/directives/ng_style.ngfactory';
import * as import30 from '../../../@angular/common/src/directives/ng_if.ngfactory';
import * as import31 from '@angular/core/src/linker/template_ref';
import * as import32 from '@angular/common/src/directives/ng_if';
import * as import33 from '@angular/common/src/directives/ng_style';
import * as import34 from '@angular/core/src/security';
class DialogModuleInjector extends import0.NgModuleInjector<import1.DialogModule> {
  _CommonModule_0:import2.CommonModule;
  _SharedModule_1:import3.SharedModule;
  _DialogModule_2:import1.DialogModule;
  __NgLocalization_3:import4.NgLocaleLocalization;
  constructor(parent:import5.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_3():import4.NgLocaleLocalization {
    if ((this.__NgLocalization_3 == null)) { (this.__NgLocalization_3 = new import4.NgLocaleLocalization(this.parent.get(import6.LOCALE_ID))); }
    return this.__NgLocalization_3;
  }
  createInternal():import1.DialogModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._SharedModule_1 = new import3.SharedModule();
    this._DialogModule_2 = new import1.DialogModule();
    return this._DialogModule_2;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.SharedModule)) { return this._SharedModule_1; }
    if ((token === import1.DialogModule)) { return this._DialogModule_2; }
    if ((token === import4.NgLocalization)) { return this._NgLocalization_3; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const DialogModuleNgFactory:import0.NgModuleFactory<import1.DialogModule> = new import0.NgModuleFactory(DialogModuleInjector,import1.DialogModule);
export class Wrapper_Dialog {
  /*private*/ _eventHandler:Function;
  context:import1.Dialog;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  /*private*/ _expr_5:any;
  /*private*/ _expr_6:any;
  /*private*/ _expr_7:any;
  /*private*/ _expr_8:any;
  /*private*/ _expr_9:any;
  /*private*/ _expr_10:any;
  /*private*/ _expr_11:any;
  /*private*/ _expr_12:any;
  /*private*/ _expr_13:any;
  /*private*/ _expr_14:any;
  /*private*/ _expr_15:any;
  /*private*/ _expr_16:any;
  subscription0:any;
  subscription1:any;
  subscription2:any;
  subscription3:any;
  subscription4:any;
  constructor(p0:any,p1:any,p2:any) {
    this._changed = false;
    this.context = new import1.Dialog(p0,p1,p2);
    this._expr_0 = import7.UNINITIALIZED;
    this._expr_1 = import7.UNINITIALIZED;
    this._expr_2 = import7.UNINITIALIZED;
    this._expr_3 = import7.UNINITIALIZED;
    this._expr_4 = import7.UNINITIALIZED;
    this._expr_5 = import7.UNINITIALIZED;
    this._expr_6 = import7.UNINITIALIZED;
    this._expr_7 = import7.UNINITIALIZED;
    this._expr_8 = import7.UNINITIALIZED;
    this._expr_9 = import7.UNINITIALIZED;
    this._expr_10 = import7.UNINITIALIZED;
    this._expr_11 = import7.UNINITIALIZED;
    this._expr_12 = import7.UNINITIALIZED;
    this._expr_13 = import7.UNINITIALIZED;
    this._expr_14 = import7.UNINITIALIZED;
    this._expr_15 = import7.UNINITIALIZED;
    this._expr_16 = import7.UNINITIALIZED;
  }
  ngOnDetach(view:import8.AppView<any>,componentView:import8.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
    this.context.ngOnDestroy();
    (this.subscription0 && this.subscription0.unsubscribe());
    (this.subscription1 && this.subscription1.unsubscribe());
    (this.subscription2 && this.subscription2.unsubscribe());
    (this.subscription3 && this.subscription3.unsubscribe());
    (this.subscription4 && this.subscription4.unsubscribe());
  }
  check_header(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.header = currValue;
      this._expr_0 = currValue;
    }
  }
  check_draggable(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.draggable = currValue;
      this._expr_1 = currValue;
    }
  }
  check_resizable(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_2,currValue))) {
      this._changed = true;
      this.context.resizable = currValue;
      this._expr_2 = currValue;
    }
  }
  check_minWidth(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_3,currValue))) {
      this._changed = true;
      this.context.minWidth = currValue;
      this._expr_3 = currValue;
    }
  }
  check_minHeight(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_4,currValue))) {
      this._changed = true;
      this.context.minHeight = currValue;
      this._expr_4 = currValue;
    }
  }
  check_width(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_5,currValue))) {
      this._changed = true;
      this.context.width = currValue;
      this._expr_5 = currValue;
    }
  }
  check_height(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_6,currValue))) {
      this._changed = true;
      this.context.height = currValue;
      this._expr_6 = currValue;
    }
  }
  check_contentHeight(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_7,currValue))) {
      this._changed = true;
      this.context.contentHeight = currValue;
      this._expr_7 = currValue;
    }
  }
  check_modal(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_8,currValue))) {
      this._changed = true;
      this.context.modal = currValue;
      this._expr_8 = currValue;
    }
  }
  check_closeOnEscape(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_9,currValue))) {
      this._changed = true;
      this.context.closeOnEscape = currValue;
      this._expr_9 = currValue;
    }
  }
  check_rtl(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_10,currValue))) {
      this._changed = true;
      this.context.rtl = currValue;
      this._expr_10 = currValue;
    }
  }
  check_closable(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_11,currValue))) {
      this._changed = true;
      this.context.closable = currValue;
      this._expr_11 = currValue;
    }
  }
  check_responsive(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_12,currValue))) {
      this._changed = true;
      this.context.responsive = currValue;
      this._expr_12 = currValue;
    }
  }
  check_appendTo(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_13,currValue))) {
      this._changed = true;
      this.context.appendTo = currValue;
      this._expr_13 = currValue;
    }
  }
  check_style(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_14,currValue))) {
      this._changed = true;
      this.context.style = currValue;
      this._expr_14 = currValue;
    }
  }
  check_styleClass(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_15,currValue))) {
      this._changed = true;
      this.context.styleClass = currValue;
      this._expr_15 = currValue;
    }
  }
  check_visible(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_16,currValue))) {
      this._changed = true;
      this.context.visible = currValue;
      this._expr_16 = currValue;
    }
  }
  ngDoCheck(view:import8.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    return changed;
  }
  checkHost(view:import8.AppView<any>,componentView:import8.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import8.AppView<any>,_eventHandler:any,emit0:boolean,emit1:boolean,emit2:boolean,emit3:boolean,emit4:boolean):void {
    this._eventHandler = _eventHandler;
    if (emit0) { (this.subscription0 = this.context.onBeforeShow.subscribe(_eventHandler.bind(view,'onBeforeShow'))); }
    if (emit1) { (this.subscription1 = this.context.onAfterShow.subscribe(_eventHandler.bind(view,'onAfterShow'))); }
    if (emit2) { (this.subscription2 = this.context.onBeforeHide.subscribe(_eventHandler.bind(view,'onBeforeHide'))); }
    if (emit3) { (this.subscription3 = this.context.onAfterHide.subscribe(_eventHandler.bind(view,'onAfterHide'))); }
    if (emit4) { (this.subscription4 = this.context.visibleChange.subscribe(_eventHandler.bind(view,'visibleChange'))); }
  }
}
var renderType_Dialog_Host:import10.RenderComponentType = import9.createRenderComponentType('',0,import11.ViewEncapsulation.None,([] as any[]),{});
class View_Dialog_Host0 extends import8.AppView<any> {
  _el_0:any;
  compView_0:import8.AppView<import1.Dialog>;
  _DomHandler_0_3:import12.DomHandler;
  _Dialog_0_4:Wrapper_Dialog;
  _query_Header_0_0:import13.QueryList<any>;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_Dialog_Host0,renderType_Dialog_Host,import14.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    this._el_0 = import9.selectOrCreateRenderHostElement(this.renderer,'p-dialog',import9.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this.compView_0 = new View_Dialog0(this.viewUtils,this,0,this._el_0);
    this._DomHandler_0_3 = new import12.DomHandler();
    this._Dialog_0_4 = new Wrapper_Dialog(new import17.ElementRef(this._el_0),this._DomHandler_0_3,this.renderer);
    this._query_Header_0_0 = new import13.QueryList<any>();
    this.compView_0.create(this._Dialog_0_4.context);
    this._query_Header_0_0.reset(([] as any[]));
    this._Dialog_0_4.context.headerFacet = this._query_Header_0_0.first;
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import16.ComponentRef_<any>(0,this,this._el_0,this._Dialog_0_4.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import12.DomHandler) && (0 === requestNodeIndex))) { return this._DomHandler_0_3; }
    if (((token === import1.Dialog) && (0 === requestNodeIndex))) { return this._Dialog_0_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._Dialog_0_4.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
    if (!throwOnChange) {
      if ((this.numberOfChecks === 0)) { this._Dialog_0_4.context.ngAfterViewInit(); }
      this._Dialog_0_4.context.ngAfterViewChecked();
    }
  }
  destroyInternal():void {
    this.compView_0.destroy();
    this._Dialog_0_4.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  visitProjectableNodesInternal(nodeIndex:number,ngContentIndex:number,cb:any,ctx:any):void {
    if (((nodeIndex == 0) && (ngContentIndex == 0))) {  }
    if (((nodeIndex == 0) && (ngContentIndex == 1))) {  }
    if (((nodeIndex == 0) && (ngContentIndex == 2))) {  }
  }
}
export const DialogNgFactory:import16.ComponentFactory<import1.Dialog> = new import16.ComponentFactory<import1.Dialog>('p-dialog',View_Dialog_Host0,import1.Dialog);
const styles_Dialog:any[] = ([] as any[]);
var Dialog_dialogState_states:any = {
  'hidden': {'opacity': '0'},
  'visible': {'opacity': '1'},
  '*': {}
}
;
function Dialog_dialogState_factory(view:import8.AppView<any>,element:any,currentState:any,nextState:any):import18.AnimationTransition {
  var previousPlayers:any = view.animationContext.getAnimationPlayers(element,((nextState == 'void')? (null as any): 'dialogState'));
  var collectedStyles:any = {};
  var player:any = (null as any);
  var totalTime:any = 0;
  var defaultStateStyles:any = Dialog_dialogState_states['*'];
  var startStateStyles:any = Dialog_dialogState_states[currentState];
  if ((startStateStyles == (null as any))) { (startStateStyles = defaultStateStyles); }
  var endStateStyles:any = Dialog_dialogState_states[nextState];
  if ((endStateStyles == (null as any))) { (endStateStyles = defaultStateStyles); }
  if (((player == (null as any)) && ((currentState == 'visible') && (nextState == 'hidden')))) {
      player = new import19.AnimationSequencePlayer([view.renderer.animate(element,new import20.AnimationStyles(import21.collectAndResolveStyles(collectedStyles,[startStateStyles])),import21.balanceAnimationKeyframes(collectedStyles,endStateStyles,[
        new import22.AnimationKeyframe(0,new import20.AnimationStyles(import21.collectAndResolveStyles(collectedStyles,[{}]))),
        new import22.AnimationKeyframe(1,new import20.AnimationStyles(import21.collectAndResolveStyles(collectedStyles,[{}])))
      ]
    ),400,0,'ease-in',previousPlayers)]);
    totalTime = 400;
  }
  if (((player == (null as any)) && ((currentState == 'hidden') && (nextState == 'visible')))) {
      player = new import19.AnimationSequencePlayer([view.renderer.animate(element,new import20.AnimationStyles(import21.collectAndResolveStyles(collectedStyles,[startStateStyles])),import21.balanceAnimationKeyframes(collectedStyles,endStateStyles,[
        new import22.AnimationKeyframe(0,new import20.AnimationStyles(import21.collectAndResolveStyles(collectedStyles,[{}]))),
        new import22.AnimationKeyframe(1,new import20.AnimationStyles(import21.collectAndResolveStyles(collectedStyles,[{}])))
      ]
    ),400,0,'ease-out',previousPlayers)]);
    totalTime = 400;
  }
  if ((player == (null as any))) { (player = new import23.NoOpAnimationPlayer()); }
  player.onDone(():void => {
    player.destroy();
    import21.renderStyles(element,view.renderer,import21.prepareFinalAnimationStyles(startStateStyles,endStateStyles));
  });
  new import19.AnimationSequencePlayer(previousPlayers).destroy();
  import21.renderStyles(element,view.renderer,import21.clearStyles(startStateStyles));
  view.animationContext.queueAnimation(element,'dialogState',player);
  return new import18.AnimationTransition(player,currentState,nextState,totalTime);
}
class View_Dialog1 extends import8.AppView<any> {
  _el_0:any;
  _text_1:any;
  /*private*/ _expr_2:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import24.ViewContainer) {
    super(View_Dialog1,renderType_Dialog,import14.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_2 = import7.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'span',new import9.InlineArray2(2,'class','ui-dialog-title'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1
    ]
    ),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2:any = import9.inlineInterpolate(1,'',this.parentView.context.header,'');
    if (import9.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setText(this._text_1,currVal_2);
      this._expr_2 = currVal_2;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_Dialog2 extends import8.AppView<any> {
  _el_0:any;
  _text_1:any;
  _text_2:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import24.ViewContainer) {
    super(View_Dialog2,renderType_Dialog,import14.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'span',new import9.InlineArray2(2,'class','ui-dialog-title'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this.projectNodes(this._el_0,0);
    this._text_2 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._text_2
    ]
    ),(null as any));
    return (null as any);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_Dialog3 extends import8.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import25.Wrapper_NgClass;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _map_5:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import24.ViewContainer) {
    super(View_Dialog3,renderType_Dialog,import14.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_5 = import9.pureProxy2((p0:any,p1:any):{[key: string]:any} => {
      return {
        'ui-dialog-titlebar-icon ui-dialog-titlebar-close ui-corner-all': p0,
        'ui-state-hover': p1
      }
      ;
    });
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'a',new import9.InlineArray4(4,'href','#','role','button'),(null as any));
    this._NgClass_0_3 = new import25.Wrapper_NgClass(this.parentView.parentView.injectorGet(import26.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import27.KeyValueDiffers,this.parentView.parentIndex),new import17.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this._el_2 = import9.createRenderElement(this.renderer,this._el_0,'span',new import9.InlineArray2(2,'class','fa fa-fw fa-close'),(null as any));
    this._text_3 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    var disposable_0:Function = import9.subscribeToRenderElement(this,this._el_0,new import9.InlineArray8(6,'click',(null as any),'mouseenter',(null as any),'mouseleave',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import28.NgClass) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 3)))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this._map_5(true,this.parentView.context.hoverCloseIcon);
    this._NgClass_0_3.check_ngClass(currVal_0_0_0,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.context.hide($event)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'mouseenter')) {
      const pd_sub_1:any = ((<any>(this.parentView.context.hoverCloseIcon = true)) !== false);
      result = (pd_sub_1 && result);
    }
    if ((eventName == 'mouseleave')) {
      const pd_sub_2:any = ((<any>(this.parentView.context.hoverCloseIcon = false)) !== false);
      result = (pd_sub_2 && result);
    }
    return result;
  }
}
class View_Dialog4 extends import8.AppView<any> {
  _el_0:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import24.ViewContainer) {
    super(View_Dialog4,renderType_Dialog,import14.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'div',new import9.InlineArray4(4,'class','ui-resizable-handle ui-resizable-se ui-icon ui-icon-gripsmall-diagonal-se','style','z-index: 90;'),(null as any));
    var disposable_0:Function = import9.subscribeToRenderElement(this,this._el_0,new import9.InlineArray2(2,'mousedown',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),[disposable_0]);
    return (null as any);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'mousedown')) {
      const pd_sub_0:any = ((<any>this.parentView.context.initResize($event)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
var renderType_Dialog:import10.RenderComponentType = import9.createRenderComponentType('',3,import11.ViewEncapsulation.None,styles_Dialog,{dialogState: Dialog_dialogState_factory});
export class View_Dialog0 extends import8.AppView<import1.Dialog> {
  _viewQuery_container_0:import13.QueryList<any>;
  _viewQuery_content_1:import13.QueryList<any>;
  _text_0:any;
  _el_1:any;
  _NgClass_1_3:import25.Wrapper_NgClass;
  _NgStyle_1_4:import29.Wrapper_NgStyle;
  _text_2:any;
  _el_3:any;
  _text_4:any;
  _anchor_5:any;
  /*private*/ _vc_5:import24.ViewContainer;
  _TemplateRef_5_5:any;
  _NgIf_5_6:import30.Wrapper_NgIf;
  _text_6:any;
  _anchor_7:any;
  /*private*/ _vc_7:import24.ViewContainer;
  _TemplateRef_7_5:any;
  _NgIf_7_6:import30.Wrapper_NgIf;
  _text_8:any;
  _anchor_9:any;
  /*private*/ _vc_9:import24.ViewContainer;
  _TemplateRef_9_5:any;
  _NgIf_9_6:import30.Wrapper_NgIf;
  _text_10:any;
  _text_11:any;
  _el_12:any;
  _text_13:any;
  _text_14:any;
  _text_15:any;
  _text_16:any;
  _anchor_17:any;
  /*private*/ _vc_17:import24.ViewContainer;
  _TemplateRef_17_5:any;
  _NgIf_17_6:import30.Wrapper_NgIf;
  _text_18:any;
  _text_19:any;
  /*private*/ _expr_36:any;
  /*private*/ _expr_37:any;
  /*private*/ _expr_38:any;
  /*private*/ _expr_39:any;
  _map_40:any;
  /*private*/ _expr_41:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_Dialog0,renderType_Dialog,import14.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways);
    this._expr_36 = import7.UNINITIALIZED;
    this._expr_37 = import7.UNINITIALIZED;
    this._expr_38 = import7.UNINITIALIZED;
    this._expr_39 = import7.UNINITIALIZED;
    this._map_40 = import9.pureProxy3((p0:any,p1:any,p2:any):{[key: string]:any} => {
      return {
        'ui-dialog ui-widget ui-widget-content ui-corner-all ui-shadow': p0,
        'ui-dialog-rtl': p1,
        'ui-dialog-draggable': p2
      }
      ;
    });
    this._expr_41 = import7.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._viewQuery_container_0 = new import13.QueryList<any>();
    this._viewQuery_content_1 = new import13.QueryList<any>();
    this._text_0 = this.renderer.createText(parentRenderNode,'\n        ',(null as any));
    this._el_1 = import9.createRenderElement(this.renderer,parentRenderNode,'div',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_1_3 = new import25.Wrapper_NgClass(this.parentView.injectorGet(import26.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import27.KeyValueDiffers,this.parentIndex),new import17.ElementRef(this._el_1),this.renderer);
    this._NgStyle_1_4 = new import29.Wrapper_NgStyle(this.parentView.injectorGet(import27.KeyValueDiffers,this.parentIndex),new import17.ElementRef(this._el_1),this.renderer);
    this._text_2 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_3 = import9.createRenderElement(this.renderer,this._el_1,'div',new import9.InlineArray2(2,'class','ui-dialog-titlebar ui-widget-header ui-helper-clearfix ui-corner-top'),(null as any));
    this._text_4 = this.renderer.createText(this._el_3,'\n                ',(null as any));
    this._anchor_5 = this.renderer.createTemplateAnchor(this._el_3,(null as any));
    this._vc_5 = new import24.ViewContainer(5,3,this,this._anchor_5);
    this._TemplateRef_5_5 = new import31.TemplateRef_(this,5,this._anchor_5);
    this._NgIf_5_6 = new import30.Wrapper_NgIf(this._vc_5.vcRef,this._TemplateRef_5_5);
    this._text_6 = this.renderer.createText(this._el_3,'\n                ',(null as any));
    this._anchor_7 = this.renderer.createTemplateAnchor(this._el_3,(null as any));
    this._vc_7 = new import24.ViewContainer(7,3,this,this._anchor_7);
    this._TemplateRef_7_5 = new import31.TemplateRef_(this,7,this._anchor_7);
    this._NgIf_7_6 = new import30.Wrapper_NgIf(this._vc_7.vcRef,this._TemplateRef_7_5);
    this._text_8 = this.renderer.createText(this._el_3,'\n                ',(null as any));
    this._anchor_9 = this.renderer.createTemplateAnchor(this._el_3,(null as any));
    this._vc_9 = new import24.ViewContainer(9,3,this,this._anchor_9);
    this._TemplateRef_9_5 = new import31.TemplateRef_(this,9,this._anchor_9);
    this._NgIf_9_6 = new import30.Wrapper_NgIf(this._vc_9.vcRef,this._TemplateRef_9_5);
    this._text_10 = this.renderer.createText(this._el_3,'\n            ',(null as any));
    this._text_11 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_12 = import9.createRenderElement(this.renderer,this._el_1,'div',new import9.InlineArray2(2,'class','ui-dialog-content ui-widget-content'),(null as any));
    this._text_13 = this.renderer.createText(this._el_12,'\n                ',(null as any));
    this.projectNodes(this._el_12,1);
    this._text_14 = this.renderer.createText(this._el_12,'\n            ',(null as any));
    this._text_15 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this.projectNodes(this._el_1,2);
    this._text_16 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._anchor_17 = this.renderer.createTemplateAnchor(this._el_1,(null as any));
    this._vc_17 = new import24.ViewContainer(17,1,this,this._anchor_17);
    this._TemplateRef_17_5 = new import31.TemplateRef_(this,17,this._anchor_17);
    this._NgIf_17_6 = new import30.Wrapper_NgIf(this._vc_17.vcRef,this._TemplateRef_17_5);
    this._text_18 = this.renderer.createText(this._el_1,'\n        ',(null as any));
    this._text_19 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    var disposable_0:Function = import9.subscribeToRenderElement(this,this._el_1,new import9.InlineArray2(2,'mousedown',(null as any)),this.eventHandler(this.handleEvent_1));
    var disposable_1:Function = import9.subscribeToRenderElement(this,this._el_3,new import9.InlineArray4(4,'mousedown',(null as any),'mouseup',(null as any)),this.eventHandler(this.handleEvent_3));
    this._viewQuery_container_0.reset([new import17.ElementRef(this._el_1)]);
    this.context.containerViewChild = this._viewQuery_container_0.first;
    this._viewQuery_content_1.reset([new import17.ElementRef(this._el_12)]);
    this.context.contentViewChild = this._viewQuery_content_1.first;
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2,
      this._el_3,
      this._text_4,
      this._anchor_5,
      this._text_6,
      this._anchor_7,
      this._text_8,
      this._anchor_9,
      this._text_10,
      this._text_11,
      this._el_12,
      this._text_13,
      this._text_14,
      this._text_15,
      this._text_16,
      this._anchor_17,
      this._text_18,
      this._text_19
    ]
    ),[
      disposable_0,
      disposable_1
    ]
    );
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import31.TemplateRef) && (5 === requestNodeIndex))) { return this._TemplateRef_5_5; }
    if (((token === import32.NgIf) && (5 === requestNodeIndex))) { return this._NgIf_5_6.context; }
    if (((token === import31.TemplateRef) && (7 === requestNodeIndex))) { return this._TemplateRef_7_5; }
    if (((token === import32.NgIf) && (7 === requestNodeIndex))) { return this._NgIf_7_6.context; }
    if (((token === import31.TemplateRef) && (9 === requestNodeIndex))) { return this._TemplateRef_9_5; }
    if (((token === import32.NgIf) && (9 === requestNodeIndex))) { return this._NgIf_9_6.context; }
    if (((token === import31.TemplateRef) && (17 === requestNodeIndex))) { return this._TemplateRef_17_5; }
    if (((token === import32.NgIf) && (17 === requestNodeIndex))) { return this._NgIf_17_6.context; }
    if (((token === import28.NgClass) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 18)))) { return this._NgClass_1_3.context; }
    if (((token === import33.NgStyle) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 18)))) { return this._NgStyle_1_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_39:any = (this.context.visible? 'visible': 'hidden');
    if (import9.checkBinding(throwOnChange,this._expr_39,currVal_39)) {
      var animationTransition_dialogState:any = this.componentType.animations['dialogState'](this,this._el_1,((this._expr_39 == import7.UNINITIALIZED)? 'void': this._expr_39),((currVal_39 == import7.UNINITIALIZED)? 'void': currVal_39));
      this._expr_39 = currVal_39;
    }
    const currVal_1_0_0:any = this.context.styleClass;
    this._NgClass_1_3.check_klass(currVal_1_0_0,throwOnChange,false);
    const currVal_1_0_1:any = this._map_40(true,this.context.rtl,this.context.draggable);
    this._NgClass_1_3.check_ngClass(currVal_1_0_1,throwOnChange,false);
    this._NgClass_1_3.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_1_1_0:any = this.context.style;
    this._NgStyle_1_4.check_ngStyle(currVal_1_1_0,throwOnChange,false);
    this._NgStyle_1_4.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_5_0_0:any = this.context.header;
    this._NgIf_5_6.check_ngIf(currVal_5_0_0,throwOnChange,false);
    this._NgIf_5_6.ngDoCheck(this,this._anchor_5,throwOnChange);
    const currVal_7_0_0:any = this.context.headerFacet;
    this._NgIf_7_6.check_ngIf(currVal_7_0_0,throwOnChange,false);
    this._NgIf_7_6.ngDoCheck(this,this._anchor_7,throwOnChange);
    const currVal_9_0_0:any = this.context.closable;
    this._NgIf_9_6.check_ngIf(currVal_9_0_0,throwOnChange,false);
    this._NgIf_9_6.ngDoCheck(this,this._anchor_9,throwOnChange);
    const currVal_17_0_0:any = this.context.resizable;
    this._NgIf_17_6.check_ngIf(currVal_17_0_0,throwOnChange,false);
    this._NgIf_17_6.ngDoCheck(this,this._anchor_17,throwOnChange);
    this._vc_5.detectChangesInNestedViews(throwOnChange);
    this._vc_7.detectChangesInNestedViews(throwOnChange);
    this._vc_9.detectChangesInNestedViews(throwOnChange);
    this._vc_17.detectChangesInNestedViews(throwOnChange);
    const currVal_36:any = (this.context.visible? 'block': 'none');
    if (import9.checkBinding(throwOnChange,this._expr_36,currVal_36)) {
      this.renderer.setElementStyle(this._el_1,'display',((this.viewUtils.sanitizer.sanitize(import34.SecurityContext.STYLE,currVal_36) == null)? (null as any): this.viewUtils.sanitizer.sanitize(import34.SecurityContext.STYLE,currVal_36).toString()));
      this._expr_36 = currVal_36;
    }
    const currVal_37:any = this.context.width;
    if (import9.checkBinding(throwOnChange,this._expr_37,currVal_37)) {
      this.renderer.setElementStyle(this._el_1,'width',((this.viewUtils.sanitizer.sanitize(import34.SecurityContext.STYLE,currVal_37) == null)? (null as any): (this.viewUtils.sanitizer.sanitize(import34.SecurityContext.STYLE,currVal_37).toString() + 'px')));
      this._expr_37 = currVal_37;
    }
    const currVal_38:any = this.context.height;
    if (import9.checkBinding(throwOnChange,this._expr_38,currVal_38)) {
      this.renderer.setElementStyle(this._el_1,'height',((this.viewUtils.sanitizer.sanitize(import34.SecurityContext.STYLE,currVal_38) == null)? (null as any): (this.viewUtils.sanitizer.sanitize(import34.SecurityContext.STYLE,currVal_38).toString() + 'px')));
      this._expr_38 = currVal_38;
    }
    const currVal_41:any = this.context.contentHeight;
    if (import9.checkBinding(throwOnChange,this._expr_41,currVal_41)) {
      this.renderer.setElementStyle(this._el_12,'height',((this.viewUtils.sanitizer.sanitize(import34.SecurityContext.STYLE,currVal_41) == null)? (null as any): (this.viewUtils.sanitizer.sanitize(import34.SecurityContext.STYLE,currVal_41).toString() + 'px')));
      this._expr_41 = currVal_41;
    }
  }
  destroyInternal():void {
    this._vc_5.destroyNestedViews();
    this._vc_7.destroyNestedViews();
    this._vc_9.destroyNestedViews();
    this._vc_17.destroyNestedViews();
  }
  detachInternal():void {
    var animationTransition_dialogState:any = this.componentType.animations['dialogState'](this,this._el_1,this._expr_39,'void');
  }
  createEmbeddedViewInternal(nodeIndex:number):import8.AppView<any> {
    if ((nodeIndex == 5)) { return new View_Dialog1(this.viewUtils,this,5,this._anchor_5,this._vc_5); }
    if ((nodeIndex == 7)) { return new View_Dialog2(this.viewUtils,this,7,this._anchor_7,this._vc_7); }
    if ((nodeIndex == 9)) { return new View_Dialog3(this.viewUtils,this,9,this._anchor_9,this._vc_9); }
    if ((nodeIndex == 17)) { return new View_Dialog4(this.viewUtils,this,17,this._anchor_17,this._vc_17); }
    return (null as any);
  }
  handleEvent_1(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'mousedown')) {
      const pd_sub_0:any = ((<any>this.context.moveOnTop()) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
  handleEvent_3(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'mousedown')) {
      const pd_sub_0:any = ((<any>this.context.initDrag($event)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'mouseup')) {
      const pd_sub_1:any = ((<any>this.context.endDrag($event)) !== false);
      result = (pd_sub_1 && result);
    }
    return result;
  }
}
