// primeng/components/button/button.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
import * as import0 from '@angular/core/src/linker/ng_module_factory';
import * as import1 from 'primeng/components/button/button';
import * as import2 from '@angular/common/src/common_module';
import * as import3 from '@angular/common/src/localization';
import * as import5 from '@angular/core/src/i18n/tokens';
import * as import6 from '@angular/core/src/change_detection/change_detection_util';
import * as import8 from '@angular/core/src/linker/view_utils';
var ButtonModuleInjector = (function (_super) {
    __extends(ButtonModuleInjector, _super);
    function ButtonModuleInjector(parent) {
        _super.call(this, parent, [], []);
    }
    Object.defineProperty(ButtonModuleInjector.prototype, "_NgLocalization_2", {
        get: function () {
            if ((this.__NgLocalization_2 == null)) {
                (this.__NgLocalization_2 = new import3.NgLocaleLocalization(this.parent.get(import5.LOCALE_ID)));
            }
            return this.__NgLocalization_2;
        },
        enumerable: true,
        configurable: true
    });
    ButtonModuleInjector.prototype.createInternal = function () {
        this._CommonModule_0 = new import2.CommonModule();
        this._ButtonModule_1 = new import1.ButtonModule();
        return this._ButtonModule_1;
    };
    ButtonModuleInjector.prototype.getInternal = function (token, notFoundResult) {
        if ((token === import2.CommonModule)) {
            return this._CommonModule_0;
        }
        if ((token === import1.ButtonModule)) {
            return this._ButtonModule_1;
        }
        if ((token === import3.NgLocalization)) {
            return this._NgLocalization_2;
        }
        return notFoundResult;
    };
    ButtonModuleInjector.prototype.destroyInternal = function () {
    };
    return ButtonModuleInjector;
}(import0.NgModuleInjector));
export var ButtonModuleNgFactory = new import0.NgModuleFactory(ButtonModuleInjector, import1.ButtonModule);
export var Wrapper_Button = (function () {
    function Wrapper_Button(p0, p1) {
        this._changed = false;
        this.context = new import1.Button(p0, p1);
        this._expr_0 = import6.UNINITIALIZED;
        this._expr_1 = import6.UNINITIALIZED;
        this._expr_2 = import6.UNINITIALIZED;
        this._expr_3 = import6.UNINITIALIZED;
        this._expr_4 = import6.UNINITIALIZED;
        this._expr_5 = import6.UNINITIALIZED;
        this._expr_6 = import6.UNINITIALIZED;
        this._expr_7 = import6.UNINITIALIZED;
    }
    Wrapper_Button.prototype.ngOnDetach = function (view, componentView, el) {
    };
    Wrapper_Button.prototype.ngOnDestroy = function () {
        this.context.ngOnDestroy();
    };
    Wrapper_Button.prototype.check_iconPos = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import8.checkBinding(throwOnChange, this._expr_0, currValue))) {
            this._changed = true;
            this.context.iconPos = currValue;
            this._expr_0 = currValue;
        }
    };
    Wrapper_Button.prototype.check_cornerStyleClass = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import8.checkBinding(throwOnChange, this._expr_1, currValue))) {
            this._changed = true;
            this.context.cornerStyleClass = currValue;
            this._expr_1 = currValue;
        }
    };
    Wrapper_Button.prototype.check_label = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import8.checkBinding(throwOnChange, this._expr_2, currValue))) {
            this._changed = true;
            this.context.label = currValue;
            this._expr_2 = currValue;
        }
    };
    Wrapper_Button.prototype.check_icon = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import8.checkBinding(throwOnChange, this._expr_3, currValue))) {
            this._changed = true;
            this.context.icon = currValue;
            this._expr_3 = currValue;
        }
    };
    Wrapper_Button.prototype.ngDoCheck = function (view, el, throwOnChange) {
        var changed = this._changed;
        this._changed = false;
        return changed;
    };
    Wrapper_Button.prototype.checkHost = function (view, componentView, el, throwOnChange) {
        var currVal_4 = (this.context.hover && !this.context.isDisabled());
        if (import8.checkBinding(throwOnChange, this._expr_4, currVal_4)) {
            view.renderer.setElementClass(el, 'ui-state-hover', currVal_4);
            this._expr_4 = currVal_4;
        }
        var currVal_5 = this.context.focus;
        if (import8.checkBinding(throwOnChange, this._expr_5, currVal_5)) {
            view.renderer.setElementClass(el, 'ui-state-focus', currVal_5);
            this._expr_5 = currVal_5;
        }
        var currVal_6 = this.context.active;
        if (import8.checkBinding(throwOnChange, this._expr_6, currVal_6)) {
            view.renderer.setElementClass(el, 'ui-state-active', currVal_6);
            this._expr_6 = currVal_6;
        }
        var currVal_7 = this.context.isDisabled();
        if (import8.checkBinding(throwOnChange, this._expr_7, currVal_7)) {
            view.renderer.setElementClass(el, 'ui-state-disabled', currVal_7);
            this._expr_7 = currVal_7;
        }
    };
    Wrapper_Button.prototype.handleEvent = function (eventName, $event) {
        var result = true;
        if ((eventName == 'mouseenter')) {
            var pd_sub_0 = (this.context.onMouseenter($event) !== false);
            result = (pd_sub_0 && result);
        }
        if ((eventName == 'mouseleave')) {
            var pd_sub_1 = (this.context.onMouseleave($event) !== false);
            result = (pd_sub_1 && result);
        }
        if ((eventName == 'mousedown')) {
            var pd_sub_2 = (this.context.onMouseDown($event) !== false);
            result = (pd_sub_2 && result);
        }
        if ((eventName == 'mouseup')) {
            var pd_sub_3 = (this.context.onMouseUp($event) !== false);
            result = (pd_sub_3 && result);
        }
        if ((eventName == 'focus')) {
            var pd_sub_4 = (this.context.onFocus($event) !== false);
            result = (pd_sub_4 && result);
        }
        if ((eventName == 'blur')) {
            var pd_sub_5 = (this.context.onBlur($event) !== false);
            result = (pd_sub_5 && result);
        }
        return result;
    };
    Wrapper_Button.prototype.subscribe = function (view, _eventHandler) {
        this._eventHandler = _eventHandler;
    };
    return Wrapper_Button;
}());
