// primeng/components/panel/panel.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from '@angular/core/src/linker/ng_module_factory';
import * as import1 from 'primeng/components/panel/panel';
import * as import2 from '@angular/common/src/common_module';
import * as import3 from '@angular/common/src/localization';
import * as import4 from '@angular/core/src/di/injector';
import * as import5 from '@angular/core/src/i18n/tokens';
import * as import6 from '@angular/core/src/change_detection/change_detection_util';
import * as import7 from '@angular/core/src/linker/view';
import * as import8 from '@angular/core/src/linker/view_utils';
import * as import9 from '@angular/core/src/render/api';
import * as import10 from '@angular/core/src/metadata/view';
import * as import11 from '@angular/core/src/linker/view_type';
import * as import12 from '@angular/core/src/change_detection/constants';
import * as import13 from '@angular/core/src/linker/component_factory';
import * as import14 from '@angular/core/src/linker/element_ref';
import * as import15 from '@angular/core/src/animation/animation_transition';
import * as import16 from '@angular/core/src/animation/animation_sequence_player';
import * as import17 from '@angular/core/src/animation/animation_styles';
import * as import18 from '@angular/core/src/animation/animation_style_util';
import * as import19 from '@angular/core/src/animation/animation_keyframe';
import * as import20 from '@angular/core/src/animation/animation_player';
import * as import21 from '@angular/core/src/linker/view_container';
import * as import22 from '../../../@angular/common/src/directives/ng_class.ngfactory';
import * as import23 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import24 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import25 from '@angular/common/src/directives/ng_class';
import * as import26 from '../../../@angular/common/src/directives/ng_style.ngfactory';
import * as import27 from '../../../@angular/common/src/directives/ng_if.ngfactory';
import * as import28 from '@angular/core/src/linker/template_ref';
import * as import29 from '@angular/common/src/directives/ng_if';
import * as import30 from '@angular/common/src/directives/ng_style';
class PanelModuleInjector extends import0.NgModuleInjector<import1.PanelModule> {
  _CommonModule_0:import2.CommonModule;
  _PanelModule_1:import1.PanelModule;
  __NgLocalization_2:import3.NgLocaleLocalization;
  constructor(parent:import4.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_2():import3.NgLocaleLocalization {
    if ((this.__NgLocalization_2 == null)) { (this.__NgLocalization_2 = new import3.NgLocaleLocalization(this.parent.get(import5.LOCALE_ID))); }
    return this.__NgLocalization_2;
  }
  createInternal():import1.PanelModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._PanelModule_1 = new import1.PanelModule();
    return this._PanelModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import1.PanelModule)) { return this._PanelModule_1; }
    if ((token === import3.NgLocalization)) { return this._NgLocalization_2; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const PanelModuleNgFactory:import0.NgModuleFactory<import1.PanelModule> = new import0.NgModuleFactory(PanelModuleInjector,import1.PanelModule);
export class Wrapper_Panel {
  /*private*/ _eventHandler:Function;
  context:import1.Panel;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  subscription0:any;
  subscription1:any;
  subscription2:any;
  constructor(p0:any) {
    this._changed = false;
    this.context = new import1.Panel(p0);
    this._expr_0 = import6.UNINITIALIZED;
    this._expr_1 = import6.UNINITIALIZED;
    this._expr_2 = import6.UNINITIALIZED;
    this._expr_3 = import6.UNINITIALIZED;
    this._expr_4 = import6.UNINITIALIZED;
  }
  ngOnDetach(view:import7.AppView<any>,componentView:import7.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
    (this.subscription0 && this.subscription0.unsubscribe());
    (this.subscription1 && this.subscription1.unsubscribe());
    (this.subscription2 && this.subscription2.unsubscribe());
  }
  check_toggleable(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.toggleable = currValue;
      this._expr_0 = currValue;
    }
  }
  check_header(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.header = currValue;
      this._expr_1 = currValue;
    }
  }
  check_collapsed(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_2,currValue))) {
      this._changed = true;
      this.context.collapsed = currValue;
      this._expr_2 = currValue;
    }
  }
  check_style(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_3,currValue))) {
      this._changed = true;
      this.context.style = currValue;
      this._expr_3 = currValue;
    }
  }
  check_styleClass(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_4,currValue))) {
      this._changed = true;
      this.context.styleClass = currValue;
      this._expr_4 = currValue;
    }
  }
  ngDoCheck(view:import7.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    return changed;
  }
  checkHost(view:import7.AppView<any>,componentView:import7.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import7.AppView<any>,_eventHandler:any,emit0:boolean,emit1:boolean,emit2:boolean):void {
    this._eventHandler = _eventHandler;
    if (emit0) { (this.subscription0 = this.context.collapsedChange.subscribe(_eventHandler.bind(view,'collapsedChange'))); }
    if (emit1) { (this.subscription1 = this.context.onBeforeToggle.subscribe(_eventHandler.bind(view,'onBeforeToggle'))); }
    if (emit2) { (this.subscription2 = this.context.onAfterToggle.subscribe(_eventHandler.bind(view,'onAfterToggle'))); }
  }
}
var renderType_Panel_Host:import9.RenderComponentType = import8.createRenderComponentType('',0,import10.ViewEncapsulation.None,([] as any[]),{});
class View_Panel_Host0 extends import7.AppView<any> {
  _el_0:any;
  compView_0:import7.AppView<import1.Panel>;
  _Panel_0_3:Wrapper_Panel;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_Panel_Host0,renderType_Panel_Host,import11.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import12.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import13.ComponentRef<any> {
    this._el_0 = import8.selectOrCreateRenderHostElement(this.renderer,'p-panel',import8.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this.compView_0 = new View_Panel0(this.viewUtils,this,0,this._el_0);
    this._Panel_0_3 = new Wrapper_Panel(new import14.ElementRef(this._el_0));
    this.compView_0.create(this._Panel_0_3.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import13.ComponentRef_<any>(0,this,this._el_0,this._Panel_0_3.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import1.Panel) && (0 === requestNodeIndex))) { return this._Panel_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._Panel_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
    this._Panel_0_3.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  visitProjectableNodesInternal(nodeIndex:number,ngContentIndex:number,cb:any,ctx:any):void {
    if (((nodeIndex == 0) && (ngContentIndex == 0))) {  }
    if (((nodeIndex == 0) && (ngContentIndex == 1))) {  }
  }
}
export const PanelNgFactory:import13.ComponentFactory<import1.Panel> = new import13.ComponentFactory<import1.Panel>('p-panel',View_Panel_Host0,import1.Panel);
const styles_Panel:any[] = ([] as any[]);
var Panel_panelContent_states:any = {
  'hidden': {'height': '0px'},
  'visible': {'height': '*'},
  '*': {}
}
;
function Panel_panelContent_factory(view:import7.AppView<any>,element:any,currentState:any,nextState:any):import15.AnimationTransition {
  var previousPlayers:any = view.animationContext.getAnimationPlayers(element,((nextState == 'void')? (null as any): 'panelContent'));
  var collectedStyles:any = {};
  var player:any = (null as any);
  var totalTime:any = 0;
  var defaultStateStyles:any = Panel_panelContent_states['*'];
  var startStateStyles:any = Panel_panelContent_states[currentState];
  if ((startStateStyles == (null as any))) { (startStateStyles = defaultStateStyles); }
  var endStateStyles:any = Panel_panelContent_states[nextState];
  if ((endStateStyles == (null as any))) { (endStateStyles = defaultStateStyles); }
  if (((player == (null as any)) && ((currentState == 'visible') && (nextState == 'hidden')))) {
      player = new import16.AnimationSequencePlayer([view.renderer.animate(element,new import17.AnimationStyles(import18.collectAndResolveStyles(collectedStyles,[startStateStyles])),import18.balanceAnimationKeyframes(collectedStyles,endStateStyles,[
        new import19.AnimationKeyframe(0,new import17.AnimationStyles(import18.collectAndResolveStyles(collectedStyles,[{}]))),
        new import19.AnimationKeyframe(1,new import17.AnimationStyles(import18.collectAndResolveStyles(collectedStyles,[{}])))
      ]
    ),400,0,'cubic-bezier(0.86, 0, 0.07, 1)',previousPlayers)]);
    totalTime = 400;
  }
  if (((player == (null as any)) && ((currentState == 'hidden') && (nextState == 'visible')))) {
      player = new import16.AnimationSequencePlayer([view.renderer.animate(element,new import17.AnimationStyles(import18.collectAndResolveStyles(collectedStyles,[startStateStyles])),import18.balanceAnimationKeyframes(collectedStyles,endStateStyles,[
        new import19.AnimationKeyframe(0,new import17.AnimationStyles(import18.collectAndResolveStyles(collectedStyles,[{}]))),
        new import19.AnimationKeyframe(1,new import17.AnimationStyles(import18.collectAndResolveStyles(collectedStyles,[{}])))
      ]
    ),400,0,'cubic-bezier(0.86, 0, 0.07, 1)',previousPlayers)]);
    totalTime = 400;
  }
  if ((player == (null as any))) { (player = new import20.NoOpAnimationPlayer()); }
  player.onDone(():void => {
    player.destroy();
    import18.renderStyles(element,view.renderer,import18.prepareFinalAnimationStyles(startStateStyles,endStateStyles));
  });
  new import16.AnimationSequencePlayer(previousPlayers).destroy();
  import18.renderStyles(element,view.renderer,import18.clearStyles(startStateStyles));
  view.animationContext.queueAnimation(element,'panelContent',player);
  return new import15.AnimationTransition(player,currentState,nextState,totalTime);
}
class View_Panel1 extends import7.AppView<any> {
  _el_0:any;
  _text_1:any;
  /*private*/ _expr_2:any;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import21.ViewContainer) {
    super(View_Panel1,renderType_Panel,import11.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import12.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_2 = import6.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import13.ComponentRef<any> {
    this._el_0 = import8.createRenderElement(this.renderer,(null as any),'span',new import8.InlineArray2(2,'class','ui-panel-title'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1
    ]
    ),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2:any = import8.inlineInterpolate(1,'',this.parentView.context.header,'');
    if (import8.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setText(this._text_1,currVal_2);
      this._expr_2 = currVal_2;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_Panel2 extends import7.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import22.Wrapper_NgClass;
  _text_1:any;
  _el_2:any;
  _NgClass_2_3:import22.Wrapper_NgClass;
  _text_3:any;
  _map_6:any;
  _map_7:any;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import21.ViewContainer) {
    super(View_Panel2,renderType_Panel,import11.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import12.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_6 = import8.pureProxy1((p0:any):{[key: string]:any} => {
      return {'ui-state-hover': p0};
    });
    this._map_7 = import8.pureProxy2((p0:any,p1:any):{[key: string]:any} => {
      return {
        'fa-minus': p0,
        'fa-plus': p1
      }
      ;
    });
  }
  createInternal(rootSelector:string):import13.ComponentRef<any> {
    this._el_0 = import8.createRenderElement(this.renderer,(null as any),'a',new import8.InlineArray4(4,'class','ui-panel-titlebar-icon ui-panel-titlebar-toggler ui-corner-all ui-state-default','href','#'),(null as any));
    this._NgClass_0_3 = new import22.Wrapper_NgClass(this.parentView.parentView.injectorGet(import23.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import24.KeyValueDiffers,this.parentView.parentIndex),new import14.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this._el_2 = import8.createRenderElement(this.renderer,this._el_0,'span',new import8.InlineArray2(2,'class','fa fa-fw'),(null as any));
    this._NgClass_2_3 = new import22.Wrapper_NgClass(this.parentView.parentView.injectorGet(import23.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import24.KeyValueDiffers,this.parentView.parentIndex),new import14.ElementRef(this._el_2),this.renderer);
    this._text_3 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    var disposable_0:Function = import8.subscribeToRenderElement(this,this._el_0,new import8.InlineArray8(6,'mouseenter',(null as any),'mouseleave',(null as any),'click',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import25.NgClass) && (2 === requestNodeIndex))) { return this._NgClass_2_3.context; }
    if (((token === import25.NgClass) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 3)))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = 'ui-panel-titlebar-icon ui-panel-titlebar-toggler ui-corner-all ui-state-default';
    this._NgClass_0_3.check_klass(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = this._map_6(this.parentView.context.hoverToggler);
    this._NgClass_0_3.check_ngClass(currVal_0_0_1,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_2_0_0:any = 'fa fa-fw';
    this._NgClass_2_3.check_klass(currVal_2_0_0,throwOnChange,false);
    const currVal_2_0_1:any = this._map_7(!this.parentView.context.collapsed,this.parentView.context.collapsed);
    this._NgClass_2_3.check_ngClass(currVal_2_0_1,throwOnChange,false);
    this._NgClass_2_3.ngDoCheck(this,this._el_2,throwOnChange);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'mouseenter')) {
      const pd_sub_0:any = ((<any>(this.parentView.context.hoverToggler = true)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'mouseleave')) {
      const pd_sub_1:any = ((<any>(this.parentView.context.hoverToggler = false)) !== false);
      result = (pd_sub_1 && result);
    }
    if ((eventName == 'click')) {
      const pd_sub_2:any = ((<any>this.parentView.context.toggle($event)) !== false);
      result = (pd_sub_2 && result);
    }
    return result;
  }
}
var renderType_Panel:import9.RenderComponentType = import8.createRenderComponentType('',2,import10.ViewEncapsulation.None,styles_Panel,{panelContent: Panel_panelContent_factory});
export class View_Panel0 extends import7.AppView<import1.Panel> {
  _text_0:any;
  _el_1:any;
  _NgClass_1_3:import22.Wrapper_NgClass;
  _NgStyle_1_4:import26.Wrapper_NgStyle;
  _text_2:any;
  _el_3:any;
  _text_4:any;
  _anchor_5:any;
  /*private*/ _vc_5:import21.ViewContainer;
  _TemplateRef_5_5:any;
  _NgIf_5_6:import27.Wrapper_NgIf;
  _text_6:any;
  _text_7:any;
  _anchor_8:any;
  /*private*/ _vc_8:import21.ViewContainer;
  _TemplateRef_8_5:any;
  _NgIf_8_6:import27.Wrapper_NgIf;
  _text_9:any;
  _text_10:any;
  _el_11:any;
  _NgClass_11_3:import22.Wrapper_NgClass;
  _text_12:any;
  _el_13:any;
  _text_14:any;
  _text_15:any;
  _text_16:any;
  _text_17:any;
  _text_18:any;
  /*private*/ _expr_28:any;
  _map_29:any;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_Panel0,renderType_Panel,import11.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import12.ChangeDetectorStatus.CheckAlways);
    this._expr_28 = import6.UNINITIALIZED;
    this._map_29 = import8.pureProxy1((p0:any):{[key: string]:any} => {
      return {'ui-panel-content-wrapper-overflown': p0};
    });
  }
  createInternal(rootSelector:string):import13.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n        ',(null as any));
    this._el_1 = import8.createRenderElement(this.renderer,parentRenderNode,'div',import8.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_1_3 = new import22.Wrapper_NgClass(this.parentView.injectorGet(import23.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import24.KeyValueDiffers,this.parentIndex),new import14.ElementRef(this._el_1),this.renderer);
    this._NgStyle_1_4 = new import26.Wrapper_NgStyle(this.parentView.injectorGet(import24.KeyValueDiffers,this.parentIndex),new import14.ElementRef(this._el_1),this.renderer);
    this._text_2 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_3 = import8.createRenderElement(this.renderer,this._el_1,'div',new import8.InlineArray2(2,'class','ui-panel-titlebar ui-widget-header ui-helper-clearfix ui-corner-all'),(null as any));
    this._text_4 = this.renderer.createText(this._el_3,'\n                ',(null as any));
    this._anchor_5 = this.renderer.createTemplateAnchor(this._el_3,(null as any));
    this._vc_5 = new import21.ViewContainer(5,3,this,this._anchor_5);
    this._TemplateRef_5_5 = new import28.TemplateRef_(this,5,this._anchor_5);
    this._NgIf_5_6 = new import27.Wrapper_NgIf(this._vc_5.vcRef,this._TemplateRef_5_5);
    this._text_6 = this.renderer.createText(this._el_3,'\n                ',(null as any));
    this.projectNodes(this._el_3,0);
    this._text_7 = this.renderer.createText(this._el_3,'\n                ',(null as any));
    this._anchor_8 = this.renderer.createTemplateAnchor(this._el_3,(null as any));
    this._vc_8 = new import21.ViewContainer(8,3,this,this._anchor_8);
    this._TemplateRef_8_5 = new import28.TemplateRef_(this,8,this._anchor_8);
    this._NgIf_8_6 = new import27.Wrapper_NgIf(this._vc_8.vcRef,this._TemplateRef_8_5);
    this._text_9 = this.renderer.createText(this._el_3,'\n            ',(null as any));
    this._text_10 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_11 = import8.createRenderElement(this.renderer,this._el_1,'div',new import8.InlineArray2(2,'class','ui-panel-content-wrapper'),(null as any));
    this._NgClass_11_3 = new import22.Wrapper_NgClass(this.parentView.injectorGet(import23.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import24.KeyValueDiffers,this.parentIndex),new import14.ElementRef(this._el_11),this.renderer);
    this._text_12 = this.renderer.createText(this._el_11,'\n                ',(null as any));
    this._el_13 = import8.createRenderElement(this.renderer,this._el_11,'div',new import8.InlineArray2(2,'class','ui-panel-content ui-widget-content'),(null as any));
    this._text_14 = this.renderer.createText(this._el_13,'\n                    ',(null as any));
    this.projectNodes(this._el_13,1);
    this._text_15 = this.renderer.createText(this._el_13,'\n                ',(null as any));
    this._text_16 = this.renderer.createText(this._el_11,'\n            ',(null as any));
    this._text_17 = this.renderer.createText(this._el_1,'\n        ',(null as any));
    this._text_18 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2,
      this._el_3,
      this._text_4,
      this._anchor_5,
      this._text_6,
      this._text_7,
      this._anchor_8,
      this._text_9,
      this._text_10,
      this._el_11,
      this._text_12,
      this._el_13,
      this._text_14,
      this._text_15,
      this._text_16,
      this._text_17,
      this._text_18
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import28.TemplateRef) && (5 === requestNodeIndex))) { return this._TemplateRef_5_5; }
    if (((token === import29.NgIf) && (5 === requestNodeIndex))) { return this._NgIf_5_6.context; }
    if (((token === import28.TemplateRef) && (8 === requestNodeIndex))) { return this._TemplateRef_8_5; }
    if (((token === import29.NgIf) && (8 === requestNodeIndex))) { return this._NgIf_8_6.context; }
    if (((token === import25.NgClass) && ((11 <= requestNodeIndex) && (requestNodeIndex <= 16)))) { return this._NgClass_11_3.context; }
    if (((token === import25.NgClass) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 17)))) { return this._NgClass_1_3.context; }
    if (((token === import30.NgStyle) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 17)))) { return this._NgStyle_1_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_28:any = (this.context.collapsed? 'hidden': 'visible');
    if (import8.checkBinding(throwOnChange,this._expr_28,currVal_28)) {
      var animationTransition_panelContent:any = this.componentType.animations['panelContent'](this,this._el_11,((this._expr_28 == import6.UNINITIALIZED)? 'void': this._expr_28),((currVal_28 == import6.UNINITIALIZED)? 'void': currVal_28));
      this._expr_28 = currVal_28;
    }
    const currVal_1_0_0:any = this.context.styleClass;
    this._NgClass_1_3.check_klass(currVal_1_0_0,throwOnChange,false);
    const currVal_1_0_1:any = 'ui-panel ui-widget ui-widget-content ui-corner-all';
    this._NgClass_1_3.check_ngClass(currVal_1_0_1,throwOnChange,false);
    this._NgClass_1_3.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_1_1_0:any = this.context.style;
    this._NgStyle_1_4.check_ngStyle(currVal_1_1_0,throwOnChange,false);
    this._NgStyle_1_4.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_5_0_0:any = this.context.header;
    this._NgIf_5_6.check_ngIf(currVal_5_0_0,throwOnChange,false);
    this._NgIf_5_6.ngDoCheck(this,this._anchor_5,throwOnChange);
    const currVal_8_0_0:any = this.context.toggleable;
    this._NgIf_8_6.check_ngIf(currVal_8_0_0,throwOnChange,false);
    this._NgIf_8_6.ngDoCheck(this,this._anchor_8,throwOnChange);
    const currVal_11_0_0:any = 'ui-panel-content-wrapper';
    this._NgClass_11_3.check_klass(currVal_11_0_0,throwOnChange,false);
    const currVal_11_0_1:any = this._map_29((this.context.collapsed || this.context.animating));
    this._NgClass_11_3.check_ngClass(currVal_11_0_1,throwOnChange,false);
    this._NgClass_11_3.ngDoCheck(this,this._el_11,throwOnChange);
    this._vc_5.detectChangesInNestedViews(throwOnChange);
    this._vc_8.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_5.destroyNestedViews();
    this._vc_8.destroyNestedViews();
  }
  detachInternal():void {
    var animationTransition_panelContent:any = this.componentType.animations['panelContent'](this,this._el_11,this._expr_28,'void');
  }
  createEmbeddedViewInternal(nodeIndex:number):import7.AppView<any> {
    if ((nodeIndex == 5)) { return new View_Panel1(this.viewUtils,this,5,this._anchor_5,this._vc_5); }
    if ((nodeIndex == 8)) { return new View_Panel2(this.viewUtils,this,8,this._anchor_8,this._vc_8); }
    return (null as any);
  }
}
