// @angular/forms/src/directives/ng_form.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from '@angular/forms/src/directives/ng_form';
import * as import1 from '@angular/core/src/linker/view';
export class Wrapper_NgForm {
  /*private*/ _eventHandler:Function;
  context:import0.NgForm;
  /*private*/ _changed:boolean;
  subscription0:any;
  constructor(p0:any,p1:any) {
    this._changed = false;
    this.context = new import0.NgForm(p0,p1);
  }
  ngOnDetach(view:import1.AppView<any>,componentView:import1.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
    (this.subscription0 && this.subscription0.unsubscribe());
  }
  ngDoCheck(view:import1.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    return changed;
  }
  checkHost(view:import1.AppView<any>,componentView:import1.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    if ((eventName == 'submit')) {
      const pd_sub_0:any = ((<any>this.context.onSubmit($event)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'reset')) {
      const pd_sub_1:any = ((<any>this.context.onReset()) !== false);
      result = (pd_sub_1 && result);
    }
    return result;
  }
  subscribe(view:import1.AppView<any>,_eventHandler:any,emit0:boolean):void {
    this._eventHandler = _eventHandler;
    if (emit0) { (this.subscription0 = this.context.ngSubmit.subscribe(_eventHandler.bind(view,'ngSubmit'))); }
  }
}
