// primeng/components/accordion/accordion.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from '@angular/core/src/linker/ng_module_factory';
import * as import1 from 'primeng/components/accordion/accordion';
import * as import2 from '@angular/common/src/common_module';
import * as import3 from '@angular/common/src/localization';
import * as import4 from '@angular/core/src/di/injector';
import * as import5 from '@angular/core/src/i18n/tokens';
import * as import6 from '@angular/core/src/change_detection/change_detection_util';
import * as import7 from '@angular/core/src/linker/view';
import * as import8 from '@angular/core/src/linker/view_utils';
import * as import9 from '@angular/core/src/render/api';
import * as import10 from '@angular/core/src/metadata/view';
import * as import11 from '@angular/core/src/linker/view_type';
import * as import12 from '@angular/core/src/change_detection/constants';
import * as import13 from '@angular/core/src/linker/component_factory';
import * as import14 from '@angular/core/src/linker/element_ref';
import * as import15 from '../../../@angular/common/src/directives/ng_class.ngfactory';
import * as import16 from '../../../@angular/common/src/directives/ng_style.ngfactory';
import * as import17 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import18 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import19 from '@angular/common/src/directives/ng_class';
import * as import20 from '@angular/common/src/directives/ng_style';
import * as import21 from '@angular/core/src/linker/query_list';
import * as import22 from '@angular/core/src/animation/animation_transition';
import * as import23 from '@angular/core/src/animation/animation_sequence_player';
import * as import24 from '@angular/core/src/animation/animation_styles';
import * as import25 from '@angular/core/src/animation/animation_style_util';
import * as import26 from '@angular/core/src/animation/animation_keyframe';
import * as import27 from '@angular/core/src/animation/animation_player';
import * as import28 from '@angular/core/src/linker/view_container';
import * as import29 from '../../../@angular/common/src/directives/ng_if.ngfactory';
import * as import30 from '@angular/core/src/linker/template_ref';
import * as import31 from '@angular/common/src/directives/ng_if';
class AccordionModuleInjector extends import0.NgModuleInjector<import1.AccordionModule> {
  _CommonModule_0:import2.CommonModule;
  _AccordionModule_1:import1.AccordionModule;
  __NgLocalization_2:import3.NgLocaleLocalization;
  constructor(parent:import4.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_2():import3.NgLocaleLocalization {
    if ((this.__NgLocalization_2 == null)) { (this.__NgLocalization_2 = new import3.NgLocaleLocalization(this.parent.get(import5.LOCALE_ID))); }
    return this.__NgLocalization_2;
  }
  createInternal():import1.AccordionModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._AccordionModule_1 = new import1.AccordionModule();
    return this._AccordionModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import1.AccordionModule)) { return this._AccordionModule_1; }
    if ((token === import3.NgLocalization)) { return this._NgLocalization_2; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const AccordionModuleNgFactory:import0.NgModuleFactory<import1.AccordionModule> = new import0.NgModuleFactory(AccordionModuleInjector,import1.AccordionModule);
export class Wrapper_Accordion {
  /*private*/ _eventHandler:Function;
  context:import1.Accordion;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  subscription0:any;
  subscription1:any;
  constructor(p0:any) {
    this._changed = false;
    this.context = new import1.Accordion(p0);
    this._expr_0 = import6.UNINITIALIZED;
    this._expr_1 = import6.UNINITIALIZED;
    this._expr_2 = import6.UNINITIALIZED;
  }
  ngOnDetach(view:import7.AppView<any>,componentView:import7.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
    (this.subscription0 && this.subscription0.unsubscribe());
    (this.subscription1 && this.subscription1.unsubscribe());
  }
  check_multiple(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.multiple = currValue;
      this._expr_0 = currValue;
    }
  }
  check_style(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.style = currValue;
      this._expr_1 = currValue;
    }
  }
  check_styleClass(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_2,currValue))) {
      this._changed = true;
      this.context.styleClass = currValue;
      this._expr_2 = currValue;
    }
  }
  ngDoCheck(view:import7.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    return changed;
  }
  checkHost(view:import7.AppView<any>,componentView:import7.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import7.AppView<any>,_eventHandler:any,emit0:boolean,emit1:boolean):void {
    this._eventHandler = _eventHandler;
    if (emit0) { (this.subscription0 = this.context.onClose.subscribe(_eventHandler.bind(view,'onClose'))); }
    if (emit1) { (this.subscription1 = this.context.onOpen.subscribe(_eventHandler.bind(view,'onOpen'))); }
  }
}
export class Wrapper_AccordionTab {
  /*private*/ _eventHandler:Function;
  context:import1.AccordionTab;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  subscription0:any;
  constructor(p0:any) {
    this._changed = false;
    this.context = new import1.AccordionTab(p0);
    this._expr_0 = import6.UNINITIALIZED;
    this._expr_1 = import6.UNINITIALIZED;
    this._expr_2 = import6.UNINITIALIZED;
  }
  ngOnDetach(view:import7.AppView<any>,componentView:import7.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
    (this.subscription0 && this.subscription0.unsubscribe());
  }
  check_header(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.header = currValue;
      this._expr_0 = currValue;
    }
  }
  check_selected(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.selected = currValue;
      this._expr_1 = currValue;
    }
  }
  check_disabled(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_2,currValue))) {
      this._changed = true;
      this.context.disabled = currValue;
      this._expr_2 = currValue;
    }
  }
  ngDoCheck(view:import7.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    return changed;
  }
  checkHost(view:import7.AppView<any>,componentView:import7.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import7.AppView<any>,_eventHandler:any,emit0:boolean):void {
    this._eventHandler = _eventHandler;
    if (emit0) { (this.subscription0 = this.context.selectedChange.subscribe(_eventHandler.bind(view,'selectedChange'))); }
  }
}
var renderType_Accordion_Host:import9.RenderComponentType = import8.createRenderComponentType('',0,import10.ViewEncapsulation.None,([] as any[]),{});
class View_Accordion_Host0 extends import7.AppView<any> {
  _el_0:any;
  compView_0:import7.AppView<import1.Accordion>;
  _Accordion_0_3:Wrapper_Accordion;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_Accordion_Host0,renderType_Accordion_Host,import11.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import12.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import13.ComponentRef<any> {
    this._el_0 = import8.selectOrCreateRenderHostElement(this.renderer,'p-accordion',import8.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this.compView_0 = new View_Accordion0(this.viewUtils,this,0,this._el_0);
    this._Accordion_0_3 = new Wrapper_Accordion(new import14.ElementRef(this._el_0));
    this.compView_0.create(this._Accordion_0_3.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import13.ComponentRef_<any>(0,this,this._el_0,this._Accordion_0_3.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import1.Accordion) && (0 === requestNodeIndex))) { return this._Accordion_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._Accordion_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
    this._Accordion_0_3.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  visitProjectableNodesInternal(nodeIndex:number,ngContentIndex:number,cb:any,ctx:any):void {
    if (((nodeIndex == 0) && (ngContentIndex == 0))) {  }
  }
}
export const AccordionNgFactory:import13.ComponentFactory<import1.Accordion> = new import13.ComponentFactory<import1.Accordion>('p-accordion',View_Accordion_Host0,import1.Accordion);
const styles_Accordion:any[] = ([] as any[]);
var renderType_Accordion:import9.RenderComponentType = import8.createRenderComponentType('',1,import10.ViewEncapsulation.None,styles_Accordion,{});
export class View_Accordion0 extends import7.AppView<import1.Accordion> {
  _text_0:any;
  _el_1:any;
  _NgClass_1_3:import15.Wrapper_NgClass;
  _NgStyle_1_4:import16.Wrapper_NgStyle;
  _text_2:any;
  _text_3:any;
  _text_4:any;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_Accordion0,renderType_Accordion,import11.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import12.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import13.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n        ',(null as any));
    this._el_1 = import8.createRenderElement(this.renderer,parentRenderNode,'div',import8.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_1_3 = new import15.Wrapper_NgClass(this.parentView.injectorGet(import17.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import18.KeyValueDiffers,this.parentIndex),new import14.ElementRef(this._el_1),this.renderer);
    this._NgStyle_1_4 = new import16.Wrapper_NgStyle(this.parentView.injectorGet(import18.KeyValueDiffers,this.parentIndex),new import14.ElementRef(this._el_1),this.renderer);
    this._text_2 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this.projectNodes(this._el_1,0);
    this._text_3 = this.renderer.createText(this._el_1,'\n        ',(null as any));
    this._text_4 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2,
      this._text_3,
      this._text_4
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import19.NgClass) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 3)))) { return this._NgClass_1_3.context; }
    if (((token === import20.NgStyle) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 3)))) { return this._NgStyle_1_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1_0_0:any = this.context.styleClass;
    this._NgClass_1_3.check_klass(currVal_1_0_0,throwOnChange,false);
    const currVal_1_0_1:any = 'ui-accordion ui-widget ui-helper-reset';
    this._NgClass_1_3.check_ngClass(currVal_1_0_1,throwOnChange,false);
    this._NgClass_1_3.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_1_1_0:any = this.context.style;
    this._NgStyle_1_4.check_ngStyle(currVal_1_1_0,throwOnChange,false);
    this._NgStyle_1_4.ngDoCheck(this,this._el_1,throwOnChange);
  }
}
var renderType_AccordionTab_Host:import9.RenderComponentType = import8.createRenderComponentType('',0,import10.ViewEncapsulation.None,([] as any[]),{});
class View_AccordionTab_Host0 extends import7.AppView<any> {
  _el_0:any;
  compView_0:import7.AppView<import1.AccordionTab>;
  _AccordionTab_0_3:Wrapper_AccordionTab;
  _query_Header_0_0:import21.QueryList<any>;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_AccordionTab_Host0,renderType_AccordionTab_Host,import11.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import12.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import13.ComponentRef<any> {
    this._el_0 = import8.selectOrCreateRenderHostElement(this.renderer,'p-accordionTab',import8.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this.compView_0 = new View_AccordionTab0(this.viewUtils,this,0,this._el_0);
    this._AccordionTab_0_3 = new Wrapper_AccordionTab(this.injectorGet(import1.Accordion,this.parentIndex));
    this._query_Header_0_0 = new import21.QueryList<any>();
    this.compView_0.create(this._AccordionTab_0_3.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import13.ComponentRef_<any>(0,this,this._el_0,this._AccordionTab_0_3.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import1.AccordionTab) && (0 === requestNodeIndex))) { return this._AccordionTab_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._AccordionTab_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    if (!throwOnChange) { if (this._query_Header_0_0.dirty) {
      this._query_Header_0_0.reset(([] as any[]));
      this._AccordionTab_0_3.context.headerFacet = this._query_Header_0_0;
      this._query_Header_0_0.notifyOnChanges();
    } }
    this.compView_0.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
    this._AccordionTab_0_3.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  visitProjectableNodesInternal(nodeIndex:number,ngContentIndex:number,cb:any,ctx:any):void {
    if (((nodeIndex == 0) && (ngContentIndex == 0))) {  }
    if (((nodeIndex == 0) && (ngContentIndex == 1))) {  }
  }
}
export const AccordionTabNgFactory:import13.ComponentFactory<import1.AccordionTab> = new import13.ComponentFactory<import1.AccordionTab>('p-accordionTab',View_AccordionTab_Host0,import1.AccordionTab);
const styles_AccordionTab:any[] = ([] as any[]);
var AccordionTab_tabContent_states:any = {
  'hidden': {'height': '0px'},
  'visible': {'height': '*'},
  '*': {}
}
;
function AccordionTab_tabContent_factory(view:import7.AppView<any>,element:any,currentState:any,nextState:any):import22.AnimationTransition {
  var previousPlayers:any = view.animationContext.getAnimationPlayers(element,((nextState == 'void')? (null as any): 'tabContent'));
  var collectedStyles:any = {};
  var player:any = (null as any);
  var totalTime:any = 0;
  var defaultStateStyles:any = AccordionTab_tabContent_states['*'];
  var startStateStyles:any = AccordionTab_tabContent_states[currentState];
  if ((startStateStyles == (null as any))) { (startStateStyles = defaultStateStyles); }
  var endStateStyles:any = AccordionTab_tabContent_states[nextState];
  if ((endStateStyles == (null as any))) { (endStateStyles = defaultStateStyles); }
  if (((player == (null as any)) && ((currentState == 'visible') && (nextState == 'hidden')))) {
      player = new import23.AnimationSequencePlayer([view.renderer.animate(element,new import24.AnimationStyles(import25.collectAndResolveStyles(collectedStyles,[startStateStyles])),import25.balanceAnimationKeyframes(collectedStyles,endStateStyles,[
        new import26.AnimationKeyframe(0,new import24.AnimationStyles(import25.collectAndResolveStyles(collectedStyles,[{}]))),
        new import26.AnimationKeyframe(1,new import24.AnimationStyles(import25.collectAndResolveStyles(collectedStyles,[{}])))
      ]
    ),400,0,'cubic-bezier(0.86, 0, 0.07, 1)',previousPlayers)]);
    totalTime = 400;
  }
  if (((player == (null as any)) && ((currentState == 'hidden') && (nextState == 'visible')))) {
      player = new import23.AnimationSequencePlayer([view.renderer.animate(element,new import24.AnimationStyles(import25.collectAndResolveStyles(collectedStyles,[startStateStyles])),import25.balanceAnimationKeyframes(collectedStyles,endStateStyles,[
        new import26.AnimationKeyframe(0,new import24.AnimationStyles(import25.collectAndResolveStyles(collectedStyles,[{}]))),
        new import26.AnimationKeyframe(1,new import24.AnimationStyles(import25.collectAndResolveStyles(collectedStyles,[{}])))
      ]
    ),400,0,'cubic-bezier(0.86, 0, 0.07, 1)',previousPlayers)]);
    totalTime = 400;
  }
  if ((player == (null as any))) { (player = new import27.NoOpAnimationPlayer()); }
  player.onDone(():void => {
    player.destroy();
    import25.renderStyles(element,view.renderer,import25.prepareFinalAnimationStyles(startStateStyles,endStateStyles));
  });
  new import23.AnimationSequencePlayer(previousPlayers).destroy();
  import25.renderStyles(element,view.renderer,import25.clearStyles(startStateStyles));
  view.animationContext.queueAnimation(element,'tabContent',player);
  return new import22.AnimationTransition(player,currentState,nextState,totalTime);
}
class View_AccordionTab1 extends import7.AppView<any> {
  _el_0:any;
  _text_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import28.ViewContainer) {
    super(View_AccordionTab1,renderType_AccordionTab,import11.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import12.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_2 = import6.UNINITIALIZED;
    this._expr_3 = import6.UNINITIALIZED;
    this._expr_4 = import6.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import13.ComponentRef<any> {
    this._el_0 = import8.createRenderElement(this.renderer,(null as any),'a',new import8.InlineArray4(4,'href','#','role','tab'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1
    ]
    ),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2:any = this.parentView.context.selected;
    if (import8.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setElementAttribute(this._el_0,'aria-expanded',((currVal_2 == null)? (null as any): currVal_2.toString()));
      this._expr_2 = currVal_2;
    }
    const currVal_3:any = this.parentView.context.selected;
    if (import8.checkBinding(throwOnChange,this._expr_3,currVal_3)) {
      this.renderer.setElementAttribute(this._el_0,'aria-selected',((currVal_3 == null)? (null as any): currVal_3.toString()));
      this._expr_3 = currVal_3;
    }
    const currVal_4:any = import8.inlineInterpolate(1,'',this.parentView.context.header,'');
    if (import8.checkBinding(throwOnChange,this._expr_4,currVal_4)) {
      this.renderer.setText(this._text_1,currVal_4);
      this._expr_4 = currVal_4;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_AccordionTab2 extends import7.AppView<any> {
  _el_0:any;
  _text_1:any;
  _text_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import28.ViewContainer) {
    super(View_AccordionTab2,renderType_AccordionTab,import11.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import12.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_3 = import6.UNINITIALIZED;
    this._expr_4 = import6.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import13.ComponentRef<any> {
    this._el_0 = import8.createRenderElement(this.renderer,(null as any),'a',new import8.InlineArray4(4,'href','#','role','tab'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    this.projectNodes(this._el_0,0);
    this._text_2 = this.renderer.createText(this._el_0,'\n            ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._text_2
    ]
    ),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_3:any = this.parentView.context.selected;
    if (import8.checkBinding(throwOnChange,this._expr_3,currVal_3)) {
      this.renderer.setElementAttribute(this._el_0,'aria-expanded',((currVal_3 == null)? (null as any): currVal_3.toString()));
      this._expr_3 = currVal_3;
    }
    const currVal_4:any = this.parentView.context.selected;
    if (import8.checkBinding(throwOnChange,this._expr_4,currVal_4)) {
      this.renderer.setElementAttribute(this._el_0,'aria-selected',((currVal_4 == null)? (null as any): currVal_4.toString()));
      this._expr_4 = currVal_4;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
var renderType_AccordionTab:import9.RenderComponentType = import8.createRenderComponentType('',2,import10.ViewEncapsulation.None,styles_AccordionTab,{tabContent: AccordionTab_tabContent_factory});
export class View_AccordionTab0 extends import7.AppView<import1.AccordionTab> {
  _text_0:any;
  _el_1:any;
  _NgClass_1_3:import15.Wrapper_NgClass;
  _text_2:any;
  _el_3:any;
  _NgClass_3_3:import15.Wrapper_NgClass;
  _text_4:any;
  _anchor_5:any;
  /*private*/ _vc_5:import28.ViewContainer;
  _TemplateRef_5_5:any;
  _NgIf_5_6:import29.Wrapper_NgIf;
  _text_6:any;
  _anchor_7:any;
  /*private*/ _vc_7:import28.ViewContainer;
  _TemplateRef_7_5:any;
  _NgIf_7_6:import29.Wrapper_NgIf;
  _text_8:any;
  _text_9:any;
  _el_10:any;
  _NgClass_10_3:import15.Wrapper_NgClass;
  _text_11:any;
  _el_12:any;
  _text_13:any;
  _text_14:any;
  _text_15:any;
  _text_16:any;
  _map_26:any;
  _map_27:any;
  /*private*/ _expr_28:any;
  /*private*/ _expr_29:any;
  _map_30:any;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_AccordionTab0,renderType_AccordionTab,import11.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import12.ChangeDetectorStatus.CheckAlways);
    this._map_26 = import8.pureProxy3((p0:any,p1:any,p2:any):{[key: string]:any} => {
      return {
        'ui-state-active': p0,
        'ui-state-hover': p1,
        'ui-state-disabled': p2
      }
      ;
    });
    this._map_27 = import8.pureProxy2((p0:any,p1:any):{[key: string]:any} => {
      return {
        'fa-caret-down': p0,
        'fa-caret-right': p1
      }
      ;
    });
    this._expr_28 = import6.UNINITIALIZED;
    this._expr_29 = import6.UNINITIALIZED;
    this._map_30 = import8.pureProxy1((p0:any):{[key: string]:any} => {
      return {'ui-accordion-content-wrapper-overflown': p0};
    });
  }
  createInternal(rootSelector:string):import13.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n        ',(null as any));
    this._el_1 = import8.createRenderElement(this.renderer,parentRenderNode,'div',new import8.InlineArray2(2,'class','ui-accordion-header ui-state-default ui-corner-all'),(null as any));
    this._NgClass_1_3 = new import15.Wrapper_NgClass(this.parentView.injectorGet(import17.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import18.KeyValueDiffers,this.parentIndex),new import14.ElementRef(this._el_1),this.renderer);
    this._text_2 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_3 = import8.createRenderElement(this.renderer,this._el_1,'span',new import8.InlineArray2(2,'class','fa fa-fw'),(null as any));
    this._NgClass_3_3 = new import15.Wrapper_NgClass(this.parentView.injectorGet(import17.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import18.KeyValueDiffers,this.parentIndex),new import14.ElementRef(this._el_3),this.renderer);
    this._text_4 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._anchor_5 = this.renderer.createTemplateAnchor(this._el_1,(null as any));
    this._vc_5 = new import28.ViewContainer(5,1,this,this._anchor_5);
    this._TemplateRef_5_5 = new import30.TemplateRef_(this,5,this._anchor_5);
    this._NgIf_5_6 = new import29.Wrapper_NgIf(this._vc_5.vcRef,this._TemplateRef_5_5);
    this._text_6 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._anchor_7 = this.renderer.createTemplateAnchor(this._el_1,(null as any));
    this._vc_7 = new import28.ViewContainer(7,1,this,this._anchor_7);
    this._TemplateRef_7_5 = new import30.TemplateRef_(this,7,this._anchor_7);
    this._NgIf_7_6 = new import29.Wrapper_NgIf(this._vc_7.vcRef,this._TemplateRef_7_5);
    this._text_8 = this.renderer.createText(this._el_1,'\n        ',(null as any));
    this._text_9 = this.renderer.createText(parentRenderNode,'\n        ',(null as any));
    this._el_10 = import8.createRenderElement(this.renderer,parentRenderNode,'div',new import8.InlineArray4(4,'class','ui-accordion-content-wrapper','role','tabpanel'),(null as any));
    this._NgClass_10_3 = new import15.Wrapper_NgClass(this.parentView.injectorGet(import17.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import18.KeyValueDiffers,this.parentIndex),new import14.ElementRef(this._el_10),this.renderer);
    this._text_11 = this.renderer.createText(this._el_10,'\n            ',(null as any));
    this._el_12 = import8.createRenderElement(this.renderer,this._el_10,'div',new import8.InlineArray2(2,'class','ui-accordion-content ui-widget-content'),(null as any));
    this._text_13 = this.renderer.createText(this._el_12,'\n                ',(null as any));
    this.projectNodes(this._el_12,1);
    this._text_14 = this.renderer.createText(this._el_12,'\n            ',(null as any));
    this._text_15 = this.renderer.createText(this._el_10,'\n        ',(null as any));
    this._text_16 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    var disposable_0:Function = import8.subscribeToRenderElement(this,this._el_1,new import8.InlineArray8(6,'mouseenter',(null as any),'mouseleave',(null as any),'click',(null as any)),this.eventHandler(this.handleEvent_1));
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2,
      this._el_3,
      this._text_4,
      this._anchor_5,
      this._text_6,
      this._anchor_7,
      this._text_8,
      this._text_9,
      this._el_10,
      this._text_11,
      this._el_12,
      this._text_13,
      this._text_14,
      this._text_15,
      this._text_16
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import19.NgClass) && (3 === requestNodeIndex))) { return this._NgClass_3_3.context; }
    if (((token === import30.TemplateRef) && (5 === requestNodeIndex))) { return this._TemplateRef_5_5; }
    if (((token === import31.NgIf) && (5 === requestNodeIndex))) { return this._NgIf_5_6.context; }
    if (((token === import30.TemplateRef) && (7 === requestNodeIndex))) { return this._TemplateRef_7_5; }
    if (((token === import31.NgIf) && (7 === requestNodeIndex))) { return this._NgIf_7_6.context; }
    if (((token === import19.NgClass) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 8)))) { return this._NgClass_1_3.context; }
    if (((token === import19.NgClass) && ((10 <= requestNodeIndex) && (requestNodeIndex <= 15)))) { return this._NgClass_10_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_28:any = (this.context.selected? 'visible': 'hidden');
    if (import8.checkBinding(throwOnChange,this._expr_28,currVal_28)) {
      var animationTransition_tabContent:any = this.componentType.animations['tabContent'](this,this._el_10,((this._expr_28 == import6.UNINITIALIZED)? 'void': this._expr_28),((currVal_28 == import6.UNINITIALIZED)? 'void': currVal_28));
      this._expr_28 = currVal_28;
    }
    const currVal_1_0_0:any = 'ui-accordion-header ui-state-default ui-corner-all';
    this._NgClass_1_3.check_klass(currVal_1_0_0,throwOnChange,false);
    const currVal_1_0_1:any = this._map_26(this.context.selected,(this.context.hover && !this.context.disabled),this.context.disabled);
    this._NgClass_1_3.check_ngClass(currVal_1_0_1,throwOnChange,false);
    this._NgClass_1_3.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_3_0_0:any = 'fa fa-fw';
    this._NgClass_3_3.check_klass(currVal_3_0_0,throwOnChange,false);
    const currVal_3_0_1:any = this._map_27(this.context.selected,!this.context.selected);
    this._NgClass_3_3.check_ngClass(currVal_3_0_1,throwOnChange,false);
    this._NgClass_3_3.ngDoCheck(this,this._el_3,throwOnChange);
    const currVal_5_0_0:boolean = !this.context.hasHeaderFacet;
    this._NgIf_5_6.check_ngIf(currVal_5_0_0,throwOnChange,false);
    this._NgIf_5_6.ngDoCheck(this,this._anchor_5,throwOnChange);
    const currVal_7_0_0:any = this.context.hasHeaderFacet;
    this._NgIf_7_6.check_ngIf(currVal_7_0_0,throwOnChange,false);
    this._NgIf_7_6.ngDoCheck(this,this._anchor_7,throwOnChange);
    const currVal_10_0_0:any = 'ui-accordion-content-wrapper';
    this._NgClass_10_3.check_klass(currVal_10_0_0,throwOnChange,false);
    const currVal_10_0_1:any = this._map_30((!this.context.selected || this.context.animating));
    this._NgClass_10_3.check_ngClass(currVal_10_0_1,throwOnChange,false);
    this._NgClass_10_3.ngDoCheck(this,this._el_10,throwOnChange);
    this._vc_5.detectChangesInNestedViews(throwOnChange);
    this._vc_7.detectChangesInNestedViews(throwOnChange);
    const currVal_29:boolean = !this.context.selected;
    if (import8.checkBinding(throwOnChange,this._expr_29,currVal_29)) {
      this.renderer.setElementAttribute(this._el_10,'aria-hidden',((currVal_29 == null)? (null as any): currVal_29.toString()));
      this._expr_29 = currVal_29;
    }
  }
  destroyInternal():void {
    this._vc_5.destroyNestedViews();
    this._vc_7.destroyNestedViews();
  }
  detachInternal():void {
    var animationTransition_tabContent:any = this.componentType.animations['tabContent'](this,this._el_10,this._expr_28,'void');
  }
  createEmbeddedViewInternal(nodeIndex:number):import7.AppView<any> {
    if ((nodeIndex == 5)) { return new View_AccordionTab1(this.viewUtils,this,5,this._anchor_5,this._vc_5); }
    if ((nodeIndex == 7)) { return new View_AccordionTab2(this.viewUtils,this,7,this._anchor_7,this._vc_7); }
    return (null as any);
  }
  handleEvent_1(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'mouseenter')) {
      const pd_sub_0:any = ((<any>(this.context.hover = true)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'mouseleave')) {
      const pd_sub_1:any = ((<any>(this.context.hover = false)) !== false);
      result = (pd_sub_1 && result);
    }
    if ((eventName == 'click')) {
      const pd_sub_2:any = ((<any>this.context.toggle($event)) !== false);
      result = (pd_sub_2 && result);
    }
    return result;
  }
}
