// primeng/components/checkbox/checkbox.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from '@angular/core/src/linker/ng_module_factory';
import * as import1 from 'primeng/components/checkbox/checkbox';
import * as import2 from '@angular/common/src/common_module';
import * as import3 from '@angular/common/src/localization';
import * as import4 from '@angular/core/src/di/injector';
import * as import5 from '@angular/core/src/i18n/tokens';
import * as import6 from '@angular/core/src/change_detection/change_detection_util';
import * as import7 from '@angular/core/src/linker/view';
import * as import8 from '@angular/core/src/linker/view_utils';
import * as import9 from '@angular/core/src/render/api';
import * as import10 from '@angular/core/src/metadata/view';
import * as import11 from '@angular/core/src/linker/view_type';
import * as import12 from '@angular/core/src/change_detection/constants';
import * as import13 from '@angular/core/src/linker/component_factory';
import * as import14 from '@angular/forms/src/directives/control_value_accessor';
import * as import15 from '@angular/core/src/linker/view_container';
import * as import16 from '../../../@angular/common/src/directives/ng_class.ngfactory';
import * as import17 from '../../../@angular/common/src/directives/ng_if.ngfactory';
import * as import18 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import19 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import20 from '@angular/core/src/linker/element_ref';
import * as import21 from '@angular/core/src/linker/template_ref';
import * as import22 from '@angular/common/src/directives/ng_class';
import * as import23 from '@angular/common/src/directives/ng_if';
class CheckboxModuleInjector extends import0.NgModuleInjector<import1.CheckboxModule> {
  _CommonModule_0:import2.CommonModule;
  _CheckboxModule_1:import1.CheckboxModule;
  __NgLocalization_2:import3.NgLocaleLocalization;
  constructor(parent:import4.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_2():import3.NgLocaleLocalization {
    if ((this.__NgLocalization_2 == null)) { (this.__NgLocalization_2 = new import3.NgLocaleLocalization(this.parent.get(import5.LOCALE_ID))); }
    return this.__NgLocalization_2;
  }
  createInternal():import1.CheckboxModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._CheckboxModule_1 = new import1.CheckboxModule();
    return this._CheckboxModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import1.CheckboxModule)) { return this._CheckboxModule_1; }
    if ((token === import3.NgLocalization)) { return this._NgLocalization_2; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const CheckboxModuleNgFactory:import0.NgModuleFactory<import1.CheckboxModule> = new import0.NgModuleFactory(CheckboxModuleInjector,import1.CheckboxModule);
export class Wrapper_Checkbox {
  /*private*/ _eventHandler:Function;
  context:import1.Checkbox;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  subscription0:any;
  constructor() {
    this._changed = false;
    this.context = new import1.Checkbox();
    this._expr_0 = import6.UNINITIALIZED;
    this._expr_1 = import6.UNINITIALIZED;
    this._expr_2 = import6.UNINITIALIZED;
    this._expr_3 = import6.UNINITIALIZED;
    this._expr_4 = import6.UNINITIALIZED;
  }
  ngOnDetach(view:import7.AppView<any>,componentView:import7.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
    (this.subscription0 && this.subscription0.unsubscribe());
  }
  check_value(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.value = currValue;
      this._expr_0 = currValue;
    }
  }
  check_name(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.name = currValue;
      this._expr_1 = currValue;
    }
  }
  check_disabled(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_2,currValue))) {
      this._changed = true;
      this.context.disabled = currValue;
      this._expr_2 = currValue;
    }
  }
  check_binary(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_3,currValue))) {
      this._changed = true;
      this.context.binary = currValue;
      this._expr_3 = currValue;
    }
  }
  check_label(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_4,currValue))) {
      this._changed = true;
      this.context.label = currValue;
      this._expr_4 = currValue;
    }
  }
  ngDoCheck(view:import7.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    return changed;
  }
  checkHost(view:import7.AppView<any>,componentView:import7.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import7.AppView<any>,_eventHandler:any,emit0:boolean):void {
    this._eventHandler = _eventHandler;
    if (emit0) { (this.subscription0 = this.context.onChange.subscribe(_eventHandler.bind(view,'onChange'))); }
  }
}
var renderType_Checkbox_Host:import9.RenderComponentType = import8.createRenderComponentType('',0,import10.ViewEncapsulation.None,([] as any[]),{});
class View_Checkbox_Host0 extends import7.AppView<any> {
  _el_0:any;
  compView_0:import7.AppView<import1.Checkbox>;
  _Checkbox_0_3:Wrapper_Checkbox;
  __NG_VALUE_ACCESSOR_0_4:any[];
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_Checkbox_Host0,renderType_Checkbox_Host,import11.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import12.ChangeDetectorStatus.CheckAlways);
  }
  get _NG_VALUE_ACCESSOR_0_4():any[] {
    if ((this.__NG_VALUE_ACCESSOR_0_4 == null)) { (this.__NG_VALUE_ACCESSOR_0_4 = [this._Checkbox_0_3.context]); }
    return this.__NG_VALUE_ACCESSOR_0_4;
  }
  createInternal(rootSelector:string):import13.ComponentRef<any> {
    this._el_0 = import8.selectOrCreateRenderHostElement(this.renderer,'p-checkbox',import8.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this.compView_0 = new View_Checkbox0(this.viewUtils,this,0,this._el_0);
    this._Checkbox_0_3 = new Wrapper_Checkbox();
    this.compView_0.create(this._Checkbox_0_3.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import13.ComponentRef_<any>(0,this,this._el_0,this._Checkbox_0_3.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import1.Checkbox) && (0 === requestNodeIndex))) { return this._Checkbox_0_3.context; }
    if (((token === import14.NG_VALUE_ACCESSOR) && (0 === requestNodeIndex))) { return this._NG_VALUE_ACCESSOR_0_4; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._Checkbox_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
    this._Checkbox_0_3.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
export const CheckboxNgFactory:import13.ComponentFactory<import1.Checkbox> = new import13.ComponentFactory<import1.Checkbox>('p-checkbox',View_Checkbox_Host0,import1.Checkbox);
const styles_Checkbox:any[] = ([] as any[]);
class View_Checkbox1 extends import7.AppView<any> {
  _el_0:any;
  _text_1:any;
  /*private*/ _expr_2:any;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import15.ViewContainer) {
    super(View_Checkbox1,renderType_Checkbox,import11.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import12.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_2 = import6.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import13.ComponentRef<any> {
    this._el_0 = import8.createRenderElement(this.renderer,(null as any),'label',new import8.InlineArray2(2,'class','ui-chkbox-label'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'',(null as any));
    var disposable_0:Function = import8.subscribeToRenderElement(this,this._el_0,new import8.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1
    ]
    ),[disposable_0]);
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2:any = import8.inlineInterpolate(1,'',this.parentView.context.label,'');
    if (import8.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setText(this._text_1,currVal_2);
      this._expr_2 = currVal_2;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.context.onClick($event,(<View_Checkbox0>this.parentView)._el_5,true)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
var renderType_Checkbox:import9.RenderComponentType = import8.createRenderComponentType('',0,import10.ViewEncapsulation.None,styles_Checkbox,{});
export class View_Checkbox0 extends import7.AppView<import1.Checkbox> {
  _text_0:any;
  _el_1:any;
  _text_2:any;
  _el_3:any;
  _text_4:any;
  _el_5:any;
  _NgClass_5_3:import16.Wrapper_NgClass;
  _text_6:any;
  _text_7:any;
  _el_8:any;
  _NgClass_8_3:import16.Wrapper_NgClass;
  _text_9:any;
  _el_10:any;
  _NgClass_10_3:import16.Wrapper_NgClass;
  _text_11:any;
  _text_12:any;
  _text_13:any;
  _anchor_14:any;
  /*private*/ _vc_14:import15.ViewContainer;
  _TemplateRef_14_5:any;
  _NgIf_14_6:import17.Wrapper_NgIf;
  _text_15:any;
  /*private*/ _expr_22:any;
  /*private*/ _expr_23:any;
  /*private*/ _expr_24:any;
  /*private*/ _expr_25:any;
  _map_26:any;
  _map_27:any;
  _map_28:any;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_Checkbox0,renderType_Checkbox,import11.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import12.ChangeDetectorStatus.CheckAlways);
    this._expr_22 = import6.UNINITIALIZED;
    this._expr_23 = import6.UNINITIALIZED;
    this._expr_24 = import6.UNINITIALIZED;
    this._expr_25 = import6.UNINITIALIZED;
    this._map_26 = import8.pureProxy1((p0:any):{[key: string]:any} => {
      return {'ui-state-focus': p0};
    });
    this._map_27 = import8.pureProxy4((p0:any,p1:any,p2:any,p3:any):{[key: string]:any} => {
      return {
        'ui-state-hover': p0,
        'ui-state-active': p1,
        'ui-state-disabled': p2,
        'ui-state-focus': p3
      }
      ;
    });
    this._map_28 = import8.pureProxy1((p0:any):{[key: string]:any} => {
      return {'fa fa-check': p0};
    });
  }
  createInternal(rootSelector:string):import13.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n        ',(null as any));
    this._el_1 = import8.createRenderElement(this.renderer,parentRenderNode,'div',new import8.InlineArray2(2,'class','ui-chkbox ui-widget'),(null as any));
    this._text_2 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_3 = import8.createRenderElement(this.renderer,this._el_1,'div',new import8.InlineArray2(2,'class','ui-helper-hidden-accessible'),(null as any));
    this._text_4 = this.renderer.createText(this._el_3,'\n                ',(null as any));
    this._el_5 = import8.createRenderElement(this.renderer,this._el_3,'input',new import8.InlineArray2(2,'type','checkbox'),(null as any));
    this._NgClass_5_3 = new import16.Wrapper_NgClass(this.parentView.injectorGet(import18.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import19.KeyValueDiffers,this.parentIndex),new import20.ElementRef(this._el_5),this.renderer);
    this._text_6 = this.renderer.createText(this._el_3,'\n            ',(null as any));
    this._text_7 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_8 = import8.createRenderElement(this.renderer,this._el_1,'div',new import8.InlineArray2(2,'class','ui-chkbox-box ui-widget ui-corner-all ui-state-default'),(null as any));
    this._NgClass_8_3 = new import16.Wrapper_NgClass(this.parentView.injectorGet(import18.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import19.KeyValueDiffers,this.parentIndex),new import20.ElementRef(this._el_8),this.renderer);
    this._text_9 = this.renderer.createText(this._el_8,'\n                ',(null as any));
    this._el_10 = import8.createRenderElement(this.renderer,this._el_8,'span',new import8.InlineArray2(2,'class','ui-chkbox-icon ui-c'),(null as any));
    this._NgClass_10_3 = new import16.Wrapper_NgClass(this.parentView.injectorGet(import18.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import19.KeyValueDiffers,this.parentIndex),new import20.ElementRef(this._el_10),this.renderer);
    this._text_11 = this.renderer.createText(this._el_8,'\n            ',(null as any));
    this._text_12 = this.renderer.createText(this._el_1,'\n        ',(null as any));
    this._text_13 = this.renderer.createText(parentRenderNode,'\n        ',(null as any));
    this._anchor_14 = this.renderer.createTemplateAnchor(parentRenderNode,(null as any));
    this._vc_14 = new import15.ViewContainer(14,(null as any),this,this._anchor_14);
    this._TemplateRef_14_5 = new import21.TemplateRef_(this,14,this._anchor_14);
    this._NgIf_14_6 = new import17.Wrapper_NgIf(this._vc_14.vcRef,this._TemplateRef_14_5);
    this._text_15 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    var disposable_0:Function = import8.subscribeToRenderElement(this,this._el_5,new import8.InlineArray8(6,'focus',(null as any),'blur',(null as any),'change',(null as any)),this.eventHandler(this.handleEvent_5));
    var disposable_1:Function = import8.subscribeToRenderElement(this,this._el_8,new import8.InlineArray8(6,'click',(null as any),'mouseover',(null as any),'mouseout',(null as any)),this.eventHandler(this.handleEvent_8));
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2,
      this._el_3,
      this._text_4,
      this._el_5,
      this._text_6,
      this._text_7,
      this._el_8,
      this._text_9,
      this._el_10,
      this._text_11,
      this._text_12,
      this._text_13,
      this._anchor_14,
      this._text_15
    ]
    ),[
      disposable_0,
      disposable_1
    ]
    );
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import22.NgClass) && (5 === requestNodeIndex))) { return this._NgClass_5_3.context; }
    if (((token === import22.NgClass) && (10 === requestNodeIndex))) { return this._NgClass_10_3.context; }
    if (((token === import22.NgClass) && ((8 <= requestNodeIndex) && (requestNodeIndex <= 11)))) { return this._NgClass_8_3.context; }
    if (((token === import21.TemplateRef) && (14 === requestNodeIndex))) { return this._TemplateRef_14_5; }
    if (((token === import23.NgIf) && (14 === requestNodeIndex))) { return this._NgIf_14_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_5_0_0:any = this._map_26(this.context.focused);
    this._NgClass_5_3.check_ngClass(currVal_5_0_0,throwOnChange,false);
    this._NgClass_5_3.ngDoCheck(this,this._el_5,throwOnChange);
    const currVal_8_0_0:any = 'ui-chkbox-box ui-widget ui-corner-all ui-state-default';
    this._NgClass_8_3.check_klass(currVal_8_0_0,throwOnChange,false);
    const currVal_8_0_1:any = this._map_27((this.context.hover && !this.context.disabled),this.context.checked,this.context.disabled,this.context.focused);
    this._NgClass_8_3.check_ngClass(currVal_8_0_1,throwOnChange,false);
    this._NgClass_8_3.ngDoCheck(this,this._el_8,throwOnChange);
    const currVal_10_0_0:any = 'ui-chkbox-icon ui-c';
    this._NgClass_10_3.check_klass(currVal_10_0_0,throwOnChange,false);
    const currVal_10_0_1:any = this._map_28(this.context.checked);
    this._NgClass_10_3.check_ngClass(currVal_10_0_1,throwOnChange,false);
    this._NgClass_10_3.ngDoCheck(this,this._el_10,throwOnChange);
    const currVal_14_0_0:any = this.context.label;
    this._NgIf_14_6.check_ngIf(currVal_14_0_0,throwOnChange,false);
    this._NgIf_14_6.ngDoCheck(this,this._anchor_14,throwOnChange);
    this._vc_14.detectChangesInNestedViews(throwOnChange);
    const currVal_22:any = this.context.name;
    if (import8.checkBinding(throwOnChange,this._expr_22,currVal_22)) {
      this.renderer.setElementProperty(this._el_5,'name',currVal_22);
      this._expr_22 = currVal_22;
    }
    const currVal_23:any = this.context.value;
    if (import8.checkBinding(throwOnChange,this._expr_23,currVal_23)) {
      this.renderer.setElementProperty(this._el_5,'value',currVal_23);
      this._expr_23 = currVal_23;
    }
    const currVal_24:any = this.context.checked;
    if (import8.checkBinding(throwOnChange,this._expr_24,currVal_24)) {
      this.renderer.setElementProperty(this._el_5,'checked',currVal_24);
      this._expr_24 = currVal_24;
    }
    const currVal_25:any = this.context.disabled;
    if (import8.checkBinding(throwOnChange,this._expr_25,currVal_25)) {
      this.renderer.setElementProperty(this._el_5,'disabled',currVal_25);
      this._expr_25 = currVal_25;
    }
  }
  destroyInternal():void {
    this._vc_14.destroyNestedViews();
  }
  createEmbeddedViewInternal(nodeIndex:number):import7.AppView<any> {
    if ((nodeIndex == 14)) { return new View_Checkbox1(this.viewUtils,this,14,this._anchor_14,this._vc_14); }
    return (null as any);
  }
  handleEvent_5(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'focus')) {
      const pd_sub_0:any = ((<any>this.context.onFocus($event)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'blur')) {
      const pd_sub_1:any = ((<any>this.context.onBlur($event)) !== false);
      result = (pd_sub_1 && result);
    }
    if ((eventName == 'change')) {
      const pd_sub_2:any = ((<any>this.context.handleChange($event)) !== false);
      result = (pd_sub_2 && result);
    }
    return result;
  }
  handleEvent_8(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.context.onClick($event,this._el_5,true)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'mouseover')) {
      const pd_sub_1:any = ((<any>(this.context.hover = true)) !== false);
      result = (pd_sub_1 && result);
    }
    if ((eventName == 'mouseout')) {
      const pd_sub_2:any = ((<any>(this.context.hover = false)) !== false);
      result = (pd_sub_2 && result);
    }
    return result;
  }
}
