// primeng/components/picklist/picklist.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from '@angular/core/src/linker/ng_module_factory';
import * as import1 from 'primeng/components/picklist/picklist';
import * as import2 from '@angular/common/src/common_module';
import * as import3 from 'primeng/components/button/button';
import * as import4 from 'primeng/components/common/shared';
import * as import5 from '@angular/common/src/localization';
import * as import6 from '@angular/core/src/di/injector';
import * as import7 from '@angular/core/src/i18n/tokens';
import * as import8 from '@angular/core/src/change_detection/change_detection_util';
import * as import9 from '@angular/core/src/linker/view';
import * as import10 from '@angular/core/src/linker/view_utils';
import * as import11 from '@angular/core/src/render/api';
import * as import12 from '@angular/core/src/metadata/view';
import * as import13 from 'primeng/components/dom/domhandler';
import * as import14 from '@angular/core/src/linker/query_list';
import * as import15 from '@angular/core/src/linker/view_type';
import * as import16 from '@angular/core/src/change_detection/constants';
import * as import17 from '@angular/core/src/linker/component_factory';
import * as import18 from '@angular/core/src/linker/element_ref';
import * as import19 from '../button/button.ngfactory';
import * as import20 from '@angular/core/src/linker/view_container';
import * as import21 from '../../../@angular/common/src/directives/ng_class.ngfactory';
import * as import22 from '../common/shared.ngfactory';
import * as import23 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import24 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import25 from '@angular/core/src/linker/template_ref';
import * as import26 from '@angular/common/src/directives/ng_class';
import * as import27 from '../../../@angular/common/src/directives/ng_style.ngfactory';
import * as import28 from '../../../@angular/common/src/directives/ng_if.ngfactory';
import * as import29 from '../../../@angular/common/src/directives/ng_for.ngfactory';
import * as import30 from '@angular/common/src/directives/ng_if';
import * as import31 from '@angular/common/src/directives/ng_for';
import * as import32 from '@angular/common/src/directives/ng_style';
class PickListModuleInjector extends import0.NgModuleInjector<import1.PickListModule> {
  _CommonModule_0:import2.CommonModule;
  _ButtonModule_1:import3.ButtonModule;
  _SharedModule_2:import4.SharedModule;
  _PickListModule_3:import1.PickListModule;
  __NgLocalization_4:import5.NgLocaleLocalization;
  constructor(parent:import6.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_4():import5.NgLocaleLocalization {
    if ((this.__NgLocalization_4 == null)) { (this.__NgLocalization_4 = new import5.NgLocaleLocalization(this.parent.get(import7.LOCALE_ID))); }
    return this.__NgLocalization_4;
  }
  createInternal():import1.PickListModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._ButtonModule_1 = new import3.ButtonModule();
    this._SharedModule_2 = new import4.SharedModule();
    this._PickListModule_3 = new import1.PickListModule();
    return this._PickListModule_3;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.ButtonModule)) { return this._ButtonModule_1; }
    if ((token === import4.SharedModule)) { return this._SharedModule_2; }
    if ((token === import1.PickListModule)) { return this._PickListModule_3; }
    if ((token === import5.NgLocalization)) { return this._NgLocalization_4; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const PickListModuleNgFactory:import0.NgModuleFactory<import1.PickListModule> = new import0.NgModuleFactory(PickListModuleInjector,import1.PickListModule);
export class Wrapper_PickList {
  /*private*/ _eventHandler:Function;
  context:import1.PickList;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  /*private*/ _expr_5:any;
  /*private*/ _expr_6:any;
  /*private*/ _expr_7:any;
  /*private*/ _expr_8:any;
  /*private*/ _expr_9:any;
  /*private*/ _expr_10:any;
  subscription0:any;
  subscription1:any;
  constructor(p0:any,p1:any) {
    this._changed = false;
    this.context = new import1.PickList(p0,p1);
    this._expr_0 = import8.UNINITIALIZED;
    this._expr_1 = import8.UNINITIALIZED;
    this._expr_2 = import8.UNINITIALIZED;
    this._expr_3 = import8.UNINITIALIZED;
    this._expr_4 = import8.UNINITIALIZED;
    this._expr_5 = import8.UNINITIALIZED;
    this._expr_6 = import8.UNINITIALIZED;
    this._expr_7 = import8.UNINITIALIZED;
    this._expr_8 = import8.UNINITIALIZED;
    this._expr_9 = import8.UNINITIALIZED;
    this._expr_10 = import8.UNINITIALIZED;
  }
  ngOnDetach(view:import9.AppView<any>,componentView:import9.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
    this.context.ngOnDestroy();
    (this.subscription0 && this.subscription0.unsubscribe());
    (this.subscription1 && this.subscription1.unsubscribe());
  }
  check_source(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.source = currValue;
      this._expr_0 = currValue;
    }
  }
  check_target(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.target = currValue;
      this._expr_1 = currValue;
    }
  }
  check_sourceHeader(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_2,currValue))) {
      this._changed = true;
      this.context.sourceHeader = currValue;
      this._expr_2 = currValue;
    }
  }
  check_targetHeader(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_3,currValue))) {
      this._changed = true;
      this.context.targetHeader = currValue;
      this._expr_3 = currValue;
    }
  }
  check_responsive(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_4,currValue))) {
      this._changed = true;
      this.context.responsive = currValue;
      this._expr_4 = currValue;
    }
  }
  check_style(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_5,currValue))) {
      this._changed = true;
      this.context.style = currValue;
      this._expr_5 = currValue;
    }
  }
  check_styleClass(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_6,currValue))) {
      this._changed = true;
      this.context.styleClass = currValue;
      this._expr_6 = currValue;
    }
  }
  check_sourceStyle(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_7,currValue))) {
      this._changed = true;
      this.context.sourceStyle = currValue;
      this._expr_7 = currValue;
    }
  }
  check_targetStyle(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_8,currValue))) {
      this._changed = true;
      this.context.targetStyle = currValue;
      this._expr_8 = currValue;
    }
  }
  check_showSourceControls(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_9,currValue))) {
      this._changed = true;
      this.context.showSourceControls = currValue;
      this._expr_9 = currValue;
    }
  }
  check_showTargetControls(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_10,currValue))) {
      this._changed = true;
      this.context.showTargetControls = currValue;
      this._expr_10 = currValue;
    }
  }
  ngDoCheck(view:import9.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    return changed;
  }
  checkHost(view:import9.AppView<any>,componentView:import9.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import9.AppView<any>,_eventHandler:any,emit0:boolean,emit1:boolean):void {
    this._eventHandler = _eventHandler;
    if (emit0) { (this.subscription0 = this.context.onMoveToSource.subscribe(_eventHandler.bind(view,'onMoveToSource'))); }
    if (emit1) { (this.subscription1 = this.context.onMoveToTarget.subscribe(_eventHandler.bind(view,'onMoveToTarget'))); }
  }
}
var renderType_PickList_Host:import11.RenderComponentType = import10.createRenderComponentType('',0,import12.ViewEncapsulation.None,([] as any[]),{});
class View_PickList_Host0 extends import9.AppView<any> {
  _el_0:any;
  compView_0:import9.AppView<import1.PickList>;
  _DomHandler_0_3:import13.DomHandler;
  _PickList_0_4:Wrapper_PickList;
  _query_TemplateRef_0_0:import14.QueryList<any>;
  constructor(viewUtils:import10.ViewUtils,parentView:import9.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_PickList_Host0,renderType_PickList_Host,import15.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import16.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import17.ComponentRef<any> {
    this._el_0 = import10.selectOrCreateRenderHostElement(this.renderer,'p-pickList',import10.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this.compView_0 = new View_PickList0(this.viewUtils,this,0,this._el_0);
    this._DomHandler_0_3 = new import13.DomHandler();
    this._PickList_0_4 = new Wrapper_PickList(new import18.ElementRef(this._el_0),this._DomHandler_0_3);
    this._query_TemplateRef_0_0 = new import14.QueryList<any>();
    this.compView_0.create(this._PickList_0_4.context);
    this._query_TemplateRef_0_0.reset(([] as any[]));
    this._PickList_0_4.context.itemTemplate = this._query_TemplateRef_0_0.first;
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import17.ComponentRef_<any>(0,this,this._el_0,this._PickList_0_4.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import13.DomHandler) && (0 === requestNodeIndex))) { return this._DomHandler_0_3; }
    if (((token === import1.PickList) && (0 === requestNodeIndex))) { return this._PickList_0_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._PickList_0_4.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
    if (!throwOnChange) { this._PickList_0_4.context.ngAfterViewChecked(); }
  }
  destroyInternal():void {
    this.compView_0.destroy();
    this._PickList_0_4.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
export const PickListNgFactory:import17.ComponentFactory<import1.PickList> = new import17.ComponentFactory<import1.PickList>('p-pickList',View_PickList_Host0,import1.PickList);
const styles_PickList:any[] = ([] as any[]);
class View_PickList1 extends import9.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _el_4:any;
  _DomHandler_4_3:import13.DomHandler;
  _Button_4_4:import19.Wrapper_Button;
  _text_5:any;
  _el_6:any;
  _DomHandler_6_3:import13.DomHandler;
  _Button_6_4:import19.Wrapper_Button;
  _text_7:any;
  _el_8:any;
  _DomHandler_8_3:import13.DomHandler;
  _Button_8_4:import19.Wrapper_Button;
  _text_9:any;
  _el_10:any;
  _DomHandler_10_3:import13.DomHandler;
  _Button_10_4:import19.Wrapper_Button;
  _text_11:any;
  _text_12:any;
  constructor(viewUtils:import10.ViewUtils,parentView:import9.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import20.ViewContainer) {
    super(View_PickList1,renderType_PickList,import15.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import16.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import17.ComponentRef<any> {
    this._el_0 = import10.createRenderElement(this.renderer,(null as any),'div',new import10.InlineArray2(2,'class','ui-picklist-source-controls ui-picklist-buttons'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    this._el_2 = import10.createRenderElement(this.renderer,this._el_0,'div',new import10.InlineArray2(2,'class','ui-picklist-buttons-cell'),(null as any));
    this._text_3 = this.renderer.createText(this._el_2,'\n                    ',(null as any));
    this._el_4 = import10.createRenderElement(this.renderer,this._el_2,'button',new import10.InlineArray8(6,'icon','fa-angle-up','pButton','','type','button'),(null as any));
    this._DomHandler_4_3 = new import13.DomHandler();
    this._Button_4_4 = new import19.Wrapper_Button(new import18.ElementRef(this._el_4),this._DomHandler_4_3);
    this._text_5 = this.renderer.createText(this._el_2,'\n                    ',(null as any));
    this._el_6 = import10.createRenderElement(this.renderer,this._el_2,'button',new import10.InlineArray8(6,'icon','fa-angle-double-up','pButton','','type','button'),(null as any));
    this._DomHandler_6_3 = new import13.DomHandler();
    this._Button_6_4 = new import19.Wrapper_Button(new import18.ElementRef(this._el_6),this._DomHandler_6_3);
    this._text_7 = this.renderer.createText(this._el_2,'\n                    ',(null as any));
    this._el_8 = import10.createRenderElement(this.renderer,this._el_2,'button',new import10.InlineArray8(6,'icon','fa-angle-down','pButton','','type','button'),(null as any));
    this._DomHandler_8_3 = new import13.DomHandler();
    this._Button_8_4 = new import19.Wrapper_Button(new import18.ElementRef(this._el_8),this._DomHandler_8_3);
    this._text_9 = this.renderer.createText(this._el_2,'\n                    ',(null as any));
    this._el_10 = import10.createRenderElement(this.renderer,this._el_2,'button',new import10.InlineArray8(6,'icon','fa-angle-double-down','pButton','','type','button'),(null as any));
    this._DomHandler_10_3 = new import13.DomHandler();
    this._Button_10_4 = new import19.Wrapper_Button(new import18.ElementRef(this._el_10),this._DomHandler_10_3);
    this._text_11 = this.renderer.createText(this._el_2,'\n                ',(null as any));
    this._text_12 = this.renderer.createText(this._el_0,'\n            ',(null as any));
    var disposable_0:Function = import10.subscribeToRenderElement(this,this._el_4,new import10.InlineArray16(14,'click',(null as any),'mouseenter',(null as any),'mouseleave',(null as any),'mousedown',(null as any),'mouseup',(null as any),'focus',(null as any),'blur',(null as any)),this.eventHandler(this.handleEvent_4));
    var disposable_1:Function = import10.subscribeToRenderElement(this,this._el_6,new import10.InlineArray16(14,'click',(null as any),'mouseenter',(null as any),'mouseleave',(null as any),'mousedown',(null as any),'mouseup',(null as any),'focus',(null as any),'blur',(null as any)),this.eventHandler(this.handleEvent_6));
    var disposable_2:Function = import10.subscribeToRenderElement(this,this._el_8,new import10.InlineArray16(14,'click',(null as any),'mouseenter',(null as any),'mouseleave',(null as any),'mousedown',(null as any),'mouseup',(null as any),'focus',(null as any),'blur',(null as any)),this.eventHandler(this.handleEvent_8));
    var disposable_3:Function = import10.subscribeToRenderElement(this,this._el_10,new import10.InlineArray16(14,'click',(null as any),'mouseenter',(null as any),'mouseleave',(null as any),'mousedown',(null as any),'mouseup',(null as any),'focus',(null as any),'blur',(null as any)),this.eventHandler(this.handleEvent_10));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._el_4,
      this._text_5,
      this._el_6,
      this._text_7,
      this._el_8,
      this._text_9,
      this._el_10,
      this._text_11,
      this._text_12
    ]
    ),[
      disposable_0,
      disposable_1,
      disposable_2,
      disposable_3
    ]
    );
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import13.DomHandler) && (4 === requestNodeIndex))) { return this._DomHandler_4_3; }
    if (((token === import3.Button) && (4 === requestNodeIndex))) { return this._Button_4_4.context; }
    if (((token === import13.DomHandler) && (6 === requestNodeIndex))) { return this._DomHandler_6_3; }
    if (((token === import3.Button) && (6 === requestNodeIndex))) { return this._Button_6_4.context; }
    if (((token === import13.DomHandler) && (8 === requestNodeIndex))) { return this._DomHandler_8_3; }
    if (((token === import3.Button) && (8 === requestNodeIndex))) { return this._Button_8_4.context; }
    if (((token === import13.DomHandler) && (10 === requestNodeIndex))) { return this._DomHandler_10_3; }
    if (((token === import3.Button) && (10 === requestNodeIndex))) { return this._Button_10_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_4_0_0:any = 'fa-angle-up';
    this._Button_4_4.check_icon(currVal_4_0_0,throwOnChange,false);
    this._Button_4_4.ngDoCheck(this,this._el_4,throwOnChange);
    const currVal_6_0_0:any = 'fa-angle-double-up';
    this._Button_6_4.check_icon(currVal_6_0_0,throwOnChange,false);
    this._Button_6_4.ngDoCheck(this,this._el_6,throwOnChange);
    const currVal_8_0_0:any = 'fa-angle-down';
    this._Button_8_4.check_icon(currVal_8_0_0,throwOnChange,false);
    this._Button_8_4.ngDoCheck(this,this._el_8,throwOnChange);
    const currVal_10_0_0:any = 'fa-angle-double-down';
    this._Button_10_4.check_icon(currVal_10_0_0,throwOnChange,false);
    this._Button_10_4.ngDoCheck(this,this._el_10,throwOnChange);
    this._Button_4_4.checkHost(this,this,this._el_4,throwOnChange);
    this._Button_6_4.checkHost(this,this,this._el_6,throwOnChange);
    this._Button_8_4.checkHost(this,this,this._el_8,throwOnChange);
    this._Button_10_4.checkHost(this,this,this._el_10,throwOnChange);
    if (!throwOnChange) {
      if ((this.numberOfChecks === 0)) { this._Button_4_4.context.ngAfterViewInit(); }
      if ((this.numberOfChecks === 0)) { this._Button_6_4.context.ngAfterViewInit(); }
      if ((this.numberOfChecks === 0)) { this._Button_8_4.context.ngAfterViewInit(); }
      if ((this.numberOfChecks === 0)) { this._Button_10_4.context.ngAfterViewInit(); }
    }
  }
  destroyInternal():void {
    this._Button_4_4.ngOnDestroy();
    this._Button_6_4.ngOnDestroy();
    this._Button_8_4.ngOnDestroy();
    this._Button_10_4.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  handleEvent_4(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    result = (this._Button_4_4.handleEvent(eventName,$event) && result);
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.context.moveUp((<View_PickList0>this.parentView)._el_9,this.parentView.context.source,this.parentView.context.selectedItemsSource)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
  handleEvent_6(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    result = (this._Button_6_4.handleEvent(eventName,$event) && result);
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.context.moveTop((<View_PickList0>this.parentView)._el_9,this.parentView.context.source,this.parentView.context.selectedItemsSource)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
  handleEvent_8(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    result = (this._Button_8_4.handleEvent(eventName,$event) && result);
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.context.moveDown((<View_PickList0>this.parentView)._el_9,this.parentView.context.source,this.parentView.context.selectedItemsSource)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
  handleEvent_10(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    result = (this._Button_10_4.handleEvent(eventName,$event) && result);
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.context.moveBottom((<View_PickList0>this.parentView)._el_9,this.parentView.context.source,this.parentView.context.selectedItemsSource)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
class View_PickList2 extends import9.AppView<any> {
  _el_0:any;
  _text_1:any;
  /*private*/ _expr_2:any;
  constructor(viewUtils:import10.ViewUtils,parentView:import9.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import20.ViewContainer) {
    super(View_PickList2,renderType_PickList,import15.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import16.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_2 = import8.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import17.ComponentRef<any> {
    this._el_0 = import10.createRenderElement(this.renderer,(null as any),'div',new import10.InlineArray2(2,'class','ui-picklist-caption ui-widget-header ui-corner-tl ui-corner-tr'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1
    ]
    ),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2:any = import10.inlineInterpolate(1,'',this.parentView.context.sourceHeader,'');
    if (import10.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setText(this._text_1,currVal_2);
      this._expr_2 = currVal_2;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_PickList4 extends import9.AppView<any> {
  _el_0:any;
  constructor(viewUtils:import10.ViewUtils,parentView:import9.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import20.ViewContainer) {
    super(View_PickList4,renderType_PickList,import15.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import16.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import17.ComponentRef<any> {
    this._el_0 = this.renderer.createTemplateAnchor((null as any),(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): ([] as any[])),(null as any));
    return (null as any);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_PickList3 extends import9.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import21.Wrapper_NgClass;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import20.ViewContainer;
  _TemplateRef_2_5:any;
  _TemplateWrapper_2_6:import22.Wrapper_TemplateWrapper;
  _text_3:any;
  _map_8:any;
  constructor(viewUtils:import10.ViewUtils,parentView:import9.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import20.ViewContainer) {
    super(View_PickList3,renderType_PickList,import15.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import16.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_8 = import10.pureProxy3((p0:any,p1:any,p2:any):{[key: string]:any} => {
      return {
        'ui-picklist-item': p0,
        'ui-state-hover': p1,
        'ui-state-highlight': p2
      }
      ;
    });
  }
  createInternal(rootSelector:string):import17.ComponentRef<any> {
    this._el_0 = import10.createRenderElement(this.renderer,(null as any),'li',import10.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_0_3 = new import21.Wrapper_NgClass(this.parentView.parentView.injectorGet(import23.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import24.KeyValueDiffers,this.parentView.parentIndex),new import18.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n                        ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import20.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import25.TemplateRef_(this,2,this._anchor_2);
    this._TemplateWrapper_2_6 = new import22.Wrapper_TemplateWrapper(this._vc_2.vcRef);
    this._text_3 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    var disposable_0:Function = import10.subscribeToRenderElement(this,this._el_0,new import10.InlineArray8(6,'mouseenter',(null as any),'mouseleave',(null as any),'click',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import25.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import4.TemplateWrapper) && (2 === requestNodeIndex))) { return this._TemplateWrapper_2_6.context; }
    if (((token === import26.NgClass) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 3)))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this._map_8(true,(this.parentView.context.hoveredItem == this.context.$implicit),this.parentView.context.isSelected(this.context.$implicit,this.parentView.context.selectedItemsSource));
    this._NgClass_0_3.check_ngClass(currVal_0_0_0,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_2_0_0:any = this.context.$implicit;
    this._TemplateWrapper_2_6.check_item(currVal_2_0_0,throwOnChange,false);
    const currVal_2_0_1:any = this.parentView.context.itemTemplate;
    this._TemplateWrapper_2_6.check_templateRef(currVal_2_0_1,throwOnChange,false);
    this._TemplateWrapper_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import9.AppView<any> {
    if ((nodeIndex == 2)) { return new View_PickList4(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    return (null as any);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'mouseenter')) {
      const pd_sub_0:any = ((<any>(this.parentView.context.hoveredItem = this.context.$implicit)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'mouseleave')) {
      const pd_sub_1:any = ((<any>(this.parentView.context.hoveredItem = (null as any))) !== false);
      result = (pd_sub_1 && result);
    }
    if ((eventName == 'click')) {
      const pd_sub_2:any = ((<any>this.parentView.context.onItemClick($event,this.context.$implicit,this.parentView.context.selectedItemsSource)) !== false);
      result = (pd_sub_2 && result);
    }
    return result;
  }
}
class View_PickList5 extends import9.AppView<any> {
  _el_0:any;
  _text_1:any;
  /*private*/ _expr_2:any;
  constructor(viewUtils:import10.ViewUtils,parentView:import9.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import20.ViewContainer) {
    super(View_PickList5,renderType_PickList,import15.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import16.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_2 = import8.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import17.ComponentRef<any> {
    this._el_0 = import10.createRenderElement(this.renderer,(null as any),'div',new import10.InlineArray2(2,'class','ui-picklist-caption ui-widget-header ui-corner-tl ui-corner-tr'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1
    ]
    ),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2:any = import10.inlineInterpolate(1,'',this.parentView.context.targetHeader,'');
    if (import10.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setText(this._text_1,currVal_2);
      this._expr_2 = currVal_2;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_PickList7 extends import9.AppView<any> {
  _el_0:any;
  constructor(viewUtils:import10.ViewUtils,parentView:import9.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import20.ViewContainer) {
    super(View_PickList7,renderType_PickList,import15.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import16.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import17.ComponentRef<any> {
    this._el_0 = this.renderer.createTemplateAnchor((null as any),(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): ([] as any[])),(null as any));
    return (null as any);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_PickList6 extends import9.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import21.Wrapper_NgClass;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import20.ViewContainer;
  _TemplateRef_2_5:any;
  _TemplateWrapper_2_6:import22.Wrapper_TemplateWrapper;
  _text_3:any;
  _map_8:any;
  constructor(viewUtils:import10.ViewUtils,parentView:import9.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import20.ViewContainer) {
    super(View_PickList6,renderType_PickList,import15.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import16.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_8 = import10.pureProxy3((p0:any,p1:any,p2:any):{[key: string]:any} => {
      return {
        'ui-picklist-item': p0,
        'ui-state-hover': p1,
        'ui-state-highlight': p2
      }
      ;
    });
  }
  createInternal(rootSelector:string):import17.ComponentRef<any> {
    this._el_0 = import10.createRenderElement(this.renderer,(null as any),'li',import10.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_0_3 = new import21.Wrapper_NgClass(this.parentView.parentView.injectorGet(import23.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import24.KeyValueDiffers,this.parentView.parentIndex),new import18.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n                        ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import20.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import25.TemplateRef_(this,2,this._anchor_2);
    this._TemplateWrapper_2_6 = new import22.Wrapper_TemplateWrapper(this._vc_2.vcRef);
    this._text_3 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    var disposable_0:Function = import10.subscribeToRenderElement(this,this._el_0,new import10.InlineArray8(6,'mouseenter',(null as any),'mouseleave',(null as any),'click',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import25.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import4.TemplateWrapper) && (2 === requestNodeIndex))) { return this._TemplateWrapper_2_6.context; }
    if (((token === import26.NgClass) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 3)))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this._map_8(true,(this.parentView.context.hoveredItem == this.context.$implicit),this.parentView.context.isSelected(this.context.$implicit,this.parentView.context.selectedItemsTarget));
    this._NgClass_0_3.check_ngClass(currVal_0_0_0,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_2_0_0:any = this.context.$implicit;
    this._TemplateWrapper_2_6.check_item(currVal_2_0_0,throwOnChange,false);
    const currVal_2_0_1:any = this.parentView.context.itemTemplate;
    this._TemplateWrapper_2_6.check_templateRef(currVal_2_0_1,throwOnChange,false);
    this._TemplateWrapper_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import9.AppView<any> {
    if ((nodeIndex == 2)) { return new View_PickList7(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    return (null as any);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'mouseenter')) {
      const pd_sub_0:any = ((<any>(this.parentView.context.hoveredItem = this.context.$implicit)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'mouseleave')) {
      const pd_sub_1:any = ((<any>(this.parentView.context.hoveredItem = (null as any))) !== false);
      result = (pd_sub_1 && result);
    }
    if ((eventName == 'click')) {
      const pd_sub_2:any = ((<any>this.parentView.context.onItemClick($event,this.context.$implicit,this.parentView.context.selectedItemsTarget)) !== false);
      result = (pd_sub_2 && result);
    }
    return result;
  }
}
class View_PickList8 extends import9.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _el_4:any;
  _DomHandler_4_3:import13.DomHandler;
  _Button_4_4:import19.Wrapper_Button;
  _text_5:any;
  _el_6:any;
  _DomHandler_6_3:import13.DomHandler;
  _Button_6_4:import19.Wrapper_Button;
  _text_7:any;
  _el_8:any;
  _DomHandler_8_3:import13.DomHandler;
  _Button_8_4:import19.Wrapper_Button;
  _text_9:any;
  _el_10:any;
  _DomHandler_10_3:import13.DomHandler;
  _Button_10_4:import19.Wrapper_Button;
  _text_11:any;
  _text_12:any;
  constructor(viewUtils:import10.ViewUtils,parentView:import9.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import20.ViewContainer) {
    super(View_PickList8,renderType_PickList,import15.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import16.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import17.ComponentRef<any> {
    this._el_0 = import10.createRenderElement(this.renderer,(null as any),'div',new import10.InlineArray2(2,'class','ui-picklist-target-controls ui-picklist-buttons'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    this._el_2 = import10.createRenderElement(this.renderer,this._el_0,'div',new import10.InlineArray2(2,'class','ui-picklist-buttons-cell'),(null as any));
    this._text_3 = this.renderer.createText(this._el_2,'\n                    ',(null as any));
    this._el_4 = import10.createRenderElement(this.renderer,this._el_2,'button',new import10.InlineArray8(6,'icon','fa-angle-up','pButton','','type','button'),(null as any));
    this._DomHandler_4_3 = new import13.DomHandler();
    this._Button_4_4 = new import19.Wrapper_Button(new import18.ElementRef(this._el_4),this._DomHandler_4_3);
    this._text_5 = this.renderer.createText(this._el_2,'\n                    ',(null as any));
    this._el_6 = import10.createRenderElement(this.renderer,this._el_2,'button',new import10.InlineArray8(6,'icon','fa-angle-double-up','pButton','','type','button'),(null as any));
    this._DomHandler_6_3 = new import13.DomHandler();
    this._Button_6_4 = new import19.Wrapper_Button(new import18.ElementRef(this._el_6),this._DomHandler_6_3);
    this._text_7 = this.renderer.createText(this._el_2,'\n                    ',(null as any));
    this._el_8 = import10.createRenderElement(this.renderer,this._el_2,'button',new import10.InlineArray8(6,'icon','fa-angle-down','pButton','','type','button'),(null as any));
    this._DomHandler_8_3 = new import13.DomHandler();
    this._Button_8_4 = new import19.Wrapper_Button(new import18.ElementRef(this._el_8),this._DomHandler_8_3);
    this._text_9 = this.renderer.createText(this._el_2,'\n                    ',(null as any));
    this._el_10 = import10.createRenderElement(this.renderer,this._el_2,'button',new import10.InlineArray8(6,'icon','fa-angle-double-down','pButton','','type','button'),(null as any));
    this._DomHandler_10_3 = new import13.DomHandler();
    this._Button_10_4 = new import19.Wrapper_Button(new import18.ElementRef(this._el_10),this._DomHandler_10_3);
    this._text_11 = this.renderer.createText(this._el_2,'\n                ',(null as any));
    this._text_12 = this.renderer.createText(this._el_0,'\n            ',(null as any));
    var disposable_0:Function = import10.subscribeToRenderElement(this,this._el_4,new import10.InlineArray16(14,'click',(null as any),'mouseenter',(null as any),'mouseleave',(null as any),'mousedown',(null as any),'mouseup',(null as any),'focus',(null as any),'blur',(null as any)),this.eventHandler(this.handleEvent_4));
    var disposable_1:Function = import10.subscribeToRenderElement(this,this._el_6,new import10.InlineArray16(14,'click',(null as any),'mouseenter',(null as any),'mouseleave',(null as any),'mousedown',(null as any),'mouseup',(null as any),'focus',(null as any),'blur',(null as any)),this.eventHandler(this.handleEvent_6));
    var disposable_2:Function = import10.subscribeToRenderElement(this,this._el_8,new import10.InlineArray16(14,'click',(null as any),'mouseenter',(null as any),'mouseleave',(null as any),'mousedown',(null as any),'mouseup',(null as any),'focus',(null as any),'blur',(null as any)),this.eventHandler(this.handleEvent_8));
    var disposable_3:Function = import10.subscribeToRenderElement(this,this._el_10,new import10.InlineArray16(14,'click',(null as any),'mouseenter',(null as any),'mouseleave',(null as any),'mousedown',(null as any),'mouseup',(null as any),'focus',(null as any),'blur',(null as any)),this.eventHandler(this.handleEvent_10));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._el_4,
      this._text_5,
      this._el_6,
      this._text_7,
      this._el_8,
      this._text_9,
      this._el_10,
      this._text_11,
      this._text_12
    ]
    ),[
      disposable_0,
      disposable_1,
      disposable_2,
      disposable_3
    ]
    );
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import13.DomHandler) && (4 === requestNodeIndex))) { return this._DomHandler_4_3; }
    if (((token === import3.Button) && (4 === requestNodeIndex))) { return this._Button_4_4.context; }
    if (((token === import13.DomHandler) && (6 === requestNodeIndex))) { return this._DomHandler_6_3; }
    if (((token === import3.Button) && (6 === requestNodeIndex))) { return this._Button_6_4.context; }
    if (((token === import13.DomHandler) && (8 === requestNodeIndex))) { return this._DomHandler_8_3; }
    if (((token === import3.Button) && (8 === requestNodeIndex))) { return this._Button_8_4.context; }
    if (((token === import13.DomHandler) && (10 === requestNodeIndex))) { return this._DomHandler_10_3; }
    if (((token === import3.Button) && (10 === requestNodeIndex))) { return this._Button_10_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_4_0_0:any = 'fa-angle-up';
    this._Button_4_4.check_icon(currVal_4_0_0,throwOnChange,false);
    this._Button_4_4.ngDoCheck(this,this._el_4,throwOnChange);
    const currVal_6_0_0:any = 'fa-angle-double-up';
    this._Button_6_4.check_icon(currVal_6_0_0,throwOnChange,false);
    this._Button_6_4.ngDoCheck(this,this._el_6,throwOnChange);
    const currVal_8_0_0:any = 'fa-angle-down';
    this._Button_8_4.check_icon(currVal_8_0_0,throwOnChange,false);
    this._Button_8_4.ngDoCheck(this,this._el_8,throwOnChange);
    const currVal_10_0_0:any = 'fa-angle-double-down';
    this._Button_10_4.check_icon(currVal_10_0_0,throwOnChange,false);
    this._Button_10_4.ngDoCheck(this,this._el_10,throwOnChange);
    this._Button_4_4.checkHost(this,this,this._el_4,throwOnChange);
    this._Button_6_4.checkHost(this,this,this._el_6,throwOnChange);
    this._Button_8_4.checkHost(this,this,this._el_8,throwOnChange);
    this._Button_10_4.checkHost(this,this,this._el_10,throwOnChange);
    if (!throwOnChange) {
      if ((this.numberOfChecks === 0)) { this._Button_4_4.context.ngAfterViewInit(); }
      if ((this.numberOfChecks === 0)) { this._Button_6_4.context.ngAfterViewInit(); }
      if ((this.numberOfChecks === 0)) { this._Button_8_4.context.ngAfterViewInit(); }
      if ((this.numberOfChecks === 0)) { this._Button_10_4.context.ngAfterViewInit(); }
    }
  }
  destroyInternal():void {
    this._Button_4_4.ngOnDestroy();
    this._Button_6_4.ngOnDestroy();
    this._Button_8_4.ngOnDestroy();
    this._Button_10_4.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  handleEvent_4(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    result = (this._Button_4_4.handleEvent(eventName,$event) && result);
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.context.moveUp((<View_PickList0>this.parentView)._el_33,this.parentView.context.target,this.parentView.context.selectedItemsTarget)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
  handleEvent_6(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    result = (this._Button_6_4.handleEvent(eventName,$event) && result);
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.context.moveTop((<View_PickList0>this.parentView)._el_33,this.parentView.context.target,this.parentView.context.selectedItemsTarget)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
  handleEvent_8(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    result = (this._Button_8_4.handleEvent(eventName,$event) && result);
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.context.moveDown((<View_PickList0>this.parentView)._el_33,this.parentView.context.target,this.parentView.context.selectedItemsTarget)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
  handleEvent_10(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    result = (this._Button_10_4.handleEvent(eventName,$event) && result);
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.context.moveBottom((<View_PickList0>this.parentView)._el_33,this.parentView.context.target,this.parentView.context.selectedItemsTarget)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
var renderType_PickList:import11.RenderComponentType = import10.createRenderComponentType('',0,import12.ViewEncapsulation.None,styles_PickList,{});
export class View_PickList0 extends import9.AppView<import1.PickList> {
  _text_0:any;
  _el_1:any;
  _NgClass_1_3:import21.Wrapper_NgClass;
  _NgStyle_1_4:import27.Wrapper_NgStyle;
  _text_2:any;
  _anchor_3:any;
  /*private*/ _vc_3:import20.ViewContainer;
  _TemplateRef_3_5:any;
  _NgIf_3_6:import28.Wrapper_NgIf;
  _text_4:any;
  _el_5:any;
  _NgClass_5_3:import21.Wrapper_NgClass;
  _text_6:any;
  _anchor_7:any;
  /*private*/ _vc_7:import20.ViewContainer;
  _TemplateRef_7_5:any;
  _NgIf_7_6:import28.Wrapper_NgIf;
  _text_8:any;
  _el_9:any;
  _NgStyle_9_3:import27.Wrapper_NgStyle;
  _text_10:any;
  _anchor_11:any;
  /*private*/ _vc_11:import20.ViewContainer;
  _TemplateRef_11_5:any;
  _NgFor_11_6:import29.Wrapper_NgFor;
  _text_12:any;
  _text_13:any;
  _text_14:any;
  _el_15:any;
  _text_16:any;
  _el_17:any;
  _text_18:any;
  _el_19:any;
  _DomHandler_19_3:import13.DomHandler;
  _Button_19_4:import19.Wrapper_Button;
  _text_20:any;
  _el_21:any;
  _DomHandler_21_3:import13.DomHandler;
  _Button_21_4:import19.Wrapper_Button;
  _text_22:any;
  _el_23:any;
  _DomHandler_23_3:import13.DomHandler;
  _Button_23_4:import19.Wrapper_Button;
  _text_24:any;
  _el_25:any;
  _DomHandler_25_3:import13.DomHandler;
  _Button_25_4:import19.Wrapper_Button;
  _text_26:any;
  _text_27:any;
  _text_28:any;
  _el_29:any;
  _NgClass_29_3:import21.Wrapper_NgClass;
  _text_30:any;
  _anchor_31:any;
  /*private*/ _vc_31:import20.ViewContainer;
  _TemplateRef_31_5:any;
  _NgIf_31_6:import28.Wrapper_NgIf;
  _text_32:any;
  _el_33:any;
  _NgStyle_33_3:import27.Wrapper_NgStyle;
  _text_34:any;
  _anchor_35:any;
  /*private*/ _vc_35:import20.ViewContainer;
  _TemplateRef_35_5:any;
  _NgFor_35_6:import29.Wrapper_NgFor;
  _text_36:any;
  _text_37:any;
  _text_38:any;
  _anchor_39:any;
  /*private*/ _vc_39:import20.ViewContainer;
  _TemplateRef_39_5:any;
  _NgIf_39_6:import28.Wrapper_NgIf;
  _text_40:any;
  _text_41:any;
  _map_74:any;
  _map_75:any;
  _map_76:any;
  constructor(viewUtils:import10.ViewUtils,parentView:import9.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_PickList0,renderType_PickList,import15.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import16.ChangeDetectorStatus.CheckAlways);
    this._map_74 = import10.pureProxy2((p0:any,p1:any):{[key: string]:any} => {
      return {
        'ui-picklist ui-widget ui-helper-clearfix': p0,
        'ui-picklist-responsive': p1
      }
      ;
    });
    this._map_75 = import10.pureProxy1((p0:any):{[key: string]:any} => {
      return {'ui-picklist-listwrapper-nocontrols': p0};
    });
    this._map_76 = import10.pureProxy1((p0:any):{[key: string]:any} => {
      return {'ui-picklist-listwrapper-nocontrols': p0};
    });
  }
  createInternal(rootSelector:string):import17.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n        ',(null as any));
    this._el_1 = import10.createRenderElement(this.renderer,parentRenderNode,'div',import10.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_1_3 = new import21.Wrapper_NgClass(this.parentView.injectorGet(import23.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import24.KeyValueDiffers,this.parentIndex),new import18.ElementRef(this._el_1),this.renderer);
    this._NgStyle_1_4 = new import27.Wrapper_NgStyle(this.parentView.injectorGet(import24.KeyValueDiffers,this.parentIndex),new import18.ElementRef(this._el_1),this.renderer);
    this._text_2 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._anchor_3 = this.renderer.createTemplateAnchor(this._el_1,(null as any));
    this._vc_3 = new import20.ViewContainer(3,1,this,this._anchor_3);
    this._TemplateRef_3_5 = new import25.TemplateRef_(this,3,this._anchor_3);
    this._NgIf_3_6 = new import28.Wrapper_NgIf(this._vc_3.vcRef,this._TemplateRef_3_5);
    this._text_4 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_5 = import10.createRenderElement(this.renderer,this._el_1,'div',new import10.InlineArray2(2,'class','ui-picklist-listwrapper ui-picklist-source-wrapper'),(null as any));
    this._NgClass_5_3 = new import21.Wrapper_NgClass(this.parentView.injectorGet(import23.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import24.KeyValueDiffers,this.parentIndex),new import18.ElementRef(this._el_5),this.renderer);
    this._text_6 = this.renderer.createText(this._el_5,'\n                ',(null as any));
    this._anchor_7 = this.renderer.createTemplateAnchor(this._el_5,(null as any));
    this._vc_7 = new import20.ViewContainer(7,5,this,this._anchor_7);
    this._TemplateRef_7_5 = new import25.TemplateRef_(this,7,this._anchor_7);
    this._NgIf_7_6 = new import28.Wrapper_NgIf(this._vc_7.vcRef,this._TemplateRef_7_5);
    this._text_8 = this.renderer.createText(this._el_5,'\n                ',(null as any));
    this._el_9 = import10.createRenderElement(this.renderer,this._el_5,'ul',new import10.InlineArray2(2,'class','ui-widget-content ui-picklist-list ui-picklist-source ui-corner-bottom'),(null as any));
    this._NgStyle_9_3 = new import27.Wrapper_NgStyle(this.parentView.injectorGet(import24.KeyValueDiffers,this.parentIndex),new import18.ElementRef(this._el_9),this.renderer);
    this._text_10 = this.renderer.createText(this._el_9,'\n                    ',(null as any));
    this._anchor_11 = this.renderer.createTemplateAnchor(this._el_9,(null as any));
    this._vc_11 = new import20.ViewContainer(11,9,this,this._anchor_11);
    this._TemplateRef_11_5 = new import25.TemplateRef_(this,11,this._anchor_11);
    this._NgFor_11_6 = new import29.Wrapper_NgFor(this._vc_11.vcRef,this._TemplateRef_11_5,this.parentView.injectorGet(import23.IterableDiffers,this.parentIndex),this.ref);
    this._text_12 = this.renderer.createText(this._el_9,'\n                ',(null as any));
    this._text_13 = this.renderer.createText(this._el_5,'\n            ',(null as any));
    this._text_14 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_15 = import10.createRenderElement(this.renderer,this._el_1,'div',new import10.InlineArray2(2,'class','ui-picklist-buttons'),(null as any));
    this._text_16 = this.renderer.createText(this._el_15,'\n                ',(null as any));
    this._el_17 = import10.createRenderElement(this.renderer,this._el_15,'div',new import10.InlineArray2(2,'class','ui-picklist-buttons-cell'),(null as any));
    this._text_18 = this.renderer.createText(this._el_17,'\n                    ',(null as any));
    this._el_19 = import10.createRenderElement(this.renderer,this._el_17,'button',new import10.InlineArray8(6,'icon','fa-angle-right','pButton','','type','button'),(null as any));
    this._DomHandler_19_3 = new import13.DomHandler();
    this._Button_19_4 = new import19.Wrapper_Button(new import18.ElementRef(this._el_19),this._DomHandler_19_3);
    this._text_20 = this.renderer.createText(this._el_17,'\n                    ',(null as any));
    this._el_21 = import10.createRenderElement(this.renderer,this._el_17,'button',new import10.InlineArray8(6,'icon','fa-angle-double-right','pButton','','type','button'),(null as any));
    this._DomHandler_21_3 = new import13.DomHandler();
    this._Button_21_4 = new import19.Wrapper_Button(new import18.ElementRef(this._el_21),this._DomHandler_21_3);
    this._text_22 = this.renderer.createText(this._el_17,'\n                    ',(null as any));
    this._el_23 = import10.createRenderElement(this.renderer,this._el_17,'button',new import10.InlineArray8(6,'icon','fa-angle-left','pButton','','type','button'),(null as any));
    this._DomHandler_23_3 = new import13.DomHandler();
    this._Button_23_4 = new import19.Wrapper_Button(new import18.ElementRef(this._el_23),this._DomHandler_23_3);
    this._text_24 = this.renderer.createText(this._el_17,'\n                    ',(null as any));
    this._el_25 = import10.createRenderElement(this.renderer,this._el_17,'button',new import10.InlineArray8(6,'icon','fa-angle-double-left','pButton','','type','button'),(null as any));
    this._DomHandler_25_3 = new import13.DomHandler();
    this._Button_25_4 = new import19.Wrapper_Button(new import18.ElementRef(this._el_25),this._DomHandler_25_3);
    this._text_26 = this.renderer.createText(this._el_17,'\n                ',(null as any));
    this._text_27 = this.renderer.createText(this._el_15,'\n            ',(null as any));
    this._text_28 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_29 = import10.createRenderElement(this.renderer,this._el_1,'div',new import10.InlineArray2(2,'class','ui-picklist-listwrapper ui-picklist-target-wrapper'),(null as any));
    this._NgClass_29_3 = new import21.Wrapper_NgClass(this.parentView.injectorGet(import23.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import24.KeyValueDiffers,this.parentIndex),new import18.ElementRef(this._el_29),this.renderer);
    this._text_30 = this.renderer.createText(this._el_29,'\n                ',(null as any));
    this._anchor_31 = this.renderer.createTemplateAnchor(this._el_29,(null as any));
    this._vc_31 = new import20.ViewContainer(31,29,this,this._anchor_31);
    this._TemplateRef_31_5 = new import25.TemplateRef_(this,31,this._anchor_31);
    this._NgIf_31_6 = new import28.Wrapper_NgIf(this._vc_31.vcRef,this._TemplateRef_31_5);
    this._text_32 = this.renderer.createText(this._el_29,'\n                ',(null as any));
    this._el_33 = import10.createRenderElement(this.renderer,this._el_29,'ul',new import10.InlineArray2(2,'class','ui-widget-content ui-picklist-list ui-picklist-target ui-corner-bottom'),(null as any));
    this._NgStyle_33_3 = new import27.Wrapper_NgStyle(this.parentView.injectorGet(import24.KeyValueDiffers,this.parentIndex),new import18.ElementRef(this._el_33),this.renderer);
    this._text_34 = this.renderer.createText(this._el_33,'\n                    ',(null as any));
    this._anchor_35 = this.renderer.createTemplateAnchor(this._el_33,(null as any));
    this._vc_35 = new import20.ViewContainer(35,33,this,this._anchor_35);
    this._TemplateRef_35_5 = new import25.TemplateRef_(this,35,this._anchor_35);
    this._NgFor_35_6 = new import29.Wrapper_NgFor(this._vc_35.vcRef,this._TemplateRef_35_5,this.parentView.injectorGet(import23.IterableDiffers,this.parentIndex),this.ref);
    this._text_36 = this.renderer.createText(this._el_33,'\n                ',(null as any));
    this._text_37 = this.renderer.createText(this._el_29,'\n            ',(null as any));
    this._text_38 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._anchor_39 = this.renderer.createTemplateAnchor(this._el_1,(null as any));
    this._vc_39 = new import20.ViewContainer(39,1,this,this._anchor_39);
    this._TemplateRef_39_5 = new import25.TemplateRef_(this,39,this._anchor_39);
    this._NgIf_39_6 = new import28.Wrapper_NgIf(this._vc_39.vcRef,this._TemplateRef_39_5);
    this._text_40 = this.renderer.createText(this._el_1,'\n        ',(null as any));
    this._text_41 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    var disposable_0:Function = import10.subscribeToRenderElement(this,this._el_19,new import10.InlineArray16(14,'click',(null as any),'mouseenter',(null as any),'mouseleave',(null as any),'mousedown',(null as any),'mouseup',(null as any),'focus',(null as any),'blur',(null as any)),this.eventHandler(this.handleEvent_19));
    var disposable_1:Function = import10.subscribeToRenderElement(this,this._el_21,new import10.InlineArray16(14,'click',(null as any),'mouseenter',(null as any),'mouseleave',(null as any),'mousedown',(null as any),'mouseup',(null as any),'focus',(null as any),'blur',(null as any)),this.eventHandler(this.handleEvent_21));
    var disposable_2:Function = import10.subscribeToRenderElement(this,this._el_23,new import10.InlineArray16(14,'click',(null as any),'mouseenter',(null as any),'mouseleave',(null as any),'mousedown',(null as any),'mouseup',(null as any),'focus',(null as any),'blur',(null as any)),this.eventHandler(this.handleEvent_23));
    var disposable_3:Function = import10.subscribeToRenderElement(this,this._el_25,new import10.InlineArray16(14,'click',(null as any),'mouseenter',(null as any),'mouseleave',(null as any),'mousedown',(null as any),'mouseup',(null as any),'focus',(null as any),'blur',(null as any)),this.eventHandler(this.handleEvent_25));
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2,
      this._anchor_3,
      this._text_4,
      this._el_5,
      this._text_6,
      this._anchor_7,
      this._text_8,
      this._el_9,
      this._text_10,
      this._anchor_11,
      this._text_12,
      this._text_13,
      this._text_14,
      this._el_15,
      this._text_16,
      this._el_17,
      this._text_18,
      this._el_19,
      this._text_20,
      this._el_21,
      this._text_22,
      this._el_23,
      this._text_24,
      this._el_25,
      this._text_26,
      this._text_27,
      this._text_28,
      this._el_29,
      this._text_30,
      this._anchor_31,
      this._text_32,
      this._el_33,
      this._text_34,
      this._anchor_35,
      this._text_36,
      this._text_37,
      this._text_38,
      this._anchor_39,
      this._text_40,
      this._text_41
    ]
    ),[
      disposable_0,
      disposable_1,
      disposable_2,
      disposable_3
    ]
    );
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import25.TemplateRef) && (3 === requestNodeIndex))) { return this._TemplateRef_3_5; }
    if (((token === import30.NgIf) && (3 === requestNodeIndex))) { return this._NgIf_3_6.context; }
    if (((token === import25.TemplateRef) && (7 === requestNodeIndex))) { return this._TemplateRef_7_5; }
    if (((token === import30.NgIf) && (7 === requestNodeIndex))) { return this._NgIf_7_6.context; }
    if (((token === import25.TemplateRef) && (11 === requestNodeIndex))) { return this._TemplateRef_11_5; }
    if (((token === import31.NgFor) && (11 === requestNodeIndex))) { return this._NgFor_11_6.context; }
    if (((token === import32.NgStyle) && ((9 <= requestNodeIndex) && (requestNodeIndex <= 12)))) { return this._NgStyle_9_3.context; }
    if (((token === import26.NgClass) && ((5 <= requestNodeIndex) && (requestNodeIndex <= 13)))) { return this._NgClass_5_3.context; }
    if (((token === import13.DomHandler) && (19 === requestNodeIndex))) { return this._DomHandler_19_3; }
    if (((token === import3.Button) && (19 === requestNodeIndex))) { return this._Button_19_4.context; }
    if (((token === import13.DomHandler) && (21 === requestNodeIndex))) { return this._DomHandler_21_3; }
    if (((token === import3.Button) && (21 === requestNodeIndex))) { return this._Button_21_4.context; }
    if (((token === import13.DomHandler) && (23 === requestNodeIndex))) { return this._DomHandler_23_3; }
    if (((token === import3.Button) && (23 === requestNodeIndex))) { return this._Button_23_4.context; }
    if (((token === import13.DomHandler) && (25 === requestNodeIndex))) { return this._DomHandler_25_3; }
    if (((token === import3.Button) && (25 === requestNodeIndex))) { return this._Button_25_4.context; }
    if (((token === import25.TemplateRef) && (31 === requestNodeIndex))) { return this._TemplateRef_31_5; }
    if (((token === import30.NgIf) && (31 === requestNodeIndex))) { return this._NgIf_31_6.context; }
    if (((token === import25.TemplateRef) && (35 === requestNodeIndex))) { return this._TemplateRef_35_5; }
    if (((token === import31.NgFor) && (35 === requestNodeIndex))) { return this._NgFor_35_6.context; }
    if (((token === import32.NgStyle) && ((33 <= requestNodeIndex) && (requestNodeIndex <= 36)))) { return this._NgStyle_33_3.context; }
    if (((token === import26.NgClass) && ((29 <= requestNodeIndex) && (requestNodeIndex <= 37)))) { return this._NgClass_29_3.context; }
    if (((token === import25.TemplateRef) && (39 === requestNodeIndex))) { return this._TemplateRef_39_5; }
    if (((token === import30.NgIf) && (39 === requestNodeIndex))) { return this._NgIf_39_6.context; }
    if (((token === import26.NgClass) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 40)))) { return this._NgClass_1_3.context; }
    if (((token === import32.NgStyle) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 40)))) { return this._NgStyle_1_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1_0_0:any = this.context.styleClass;
    this._NgClass_1_3.check_klass(currVal_1_0_0,throwOnChange,false);
    const currVal_1_0_1:any = this._map_74(true,this.context.responsive);
    this._NgClass_1_3.check_ngClass(currVal_1_0_1,throwOnChange,false);
    this._NgClass_1_3.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_1_1_0:any = this.context.style;
    this._NgStyle_1_4.check_ngStyle(currVal_1_1_0,throwOnChange,false);
    this._NgStyle_1_4.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_3_0_0:any = this.context.showSourceControls;
    this._NgIf_3_6.check_ngIf(currVal_3_0_0,throwOnChange,false);
    this._NgIf_3_6.ngDoCheck(this,this._anchor_3,throwOnChange);
    const currVal_5_0_0:any = 'ui-picklist-listwrapper ui-picklist-source-wrapper';
    this._NgClass_5_3.check_klass(currVal_5_0_0,throwOnChange,false);
    const currVal_5_0_1:any = this._map_75(!this.context.showSourceControls);
    this._NgClass_5_3.check_ngClass(currVal_5_0_1,throwOnChange,false);
    this._NgClass_5_3.ngDoCheck(this,this._el_5,throwOnChange);
    const currVal_7_0_0:any = this.context.sourceHeader;
    this._NgIf_7_6.check_ngIf(currVal_7_0_0,throwOnChange,false);
    this._NgIf_7_6.ngDoCheck(this,this._anchor_7,throwOnChange);
    const currVal_9_0_0:any = this.context.sourceStyle;
    this._NgStyle_9_3.check_ngStyle(currVal_9_0_0,throwOnChange,false);
    this._NgStyle_9_3.ngDoCheck(this,this._el_9,throwOnChange);
    const currVal_11_0_0:any = this.context.source;
    this._NgFor_11_6.check_ngForOf(currVal_11_0_0,throwOnChange,false);
    this._NgFor_11_6.ngDoCheck(this,this._anchor_11,throwOnChange);
    const currVal_19_0_0:any = 'fa-angle-right';
    this._Button_19_4.check_icon(currVal_19_0_0,throwOnChange,false);
    this._Button_19_4.ngDoCheck(this,this._el_19,throwOnChange);
    const currVal_21_0_0:any = 'fa-angle-double-right';
    this._Button_21_4.check_icon(currVal_21_0_0,throwOnChange,false);
    this._Button_21_4.ngDoCheck(this,this._el_21,throwOnChange);
    const currVal_23_0_0:any = 'fa-angle-left';
    this._Button_23_4.check_icon(currVal_23_0_0,throwOnChange,false);
    this._Button_23_4.ngDoCheck(this,this._el_23,throwOnChange);
    const currVal_25_0_0:any = 'fa-angle-double-left';
    this._Button_25_4.check_icon(currVal_25_0_0,throwOnChange,false);
    this._Button_25_4.ngDoCheck(this,this._el_25,throwOnChange);
    const currVal_29_0_0:any = 'ui-picklist-listwrapper ui-picklist-target-wrapper';
    this._NgClass_29_3.check_klass(currVal_29_0_0,throwOnChange,false);
    const currVal_29_0_1:any = this._map_76(!this.context.showSourceControls);
    this._NgClass_29_3.check_ngClass(currVal_29_0_1,throwOnChange,false);
    this._NgClass_29_3.ngDoCheck(this,this._el_29,throwOnChange);
    const currVal_31_0_0:any = this.context.targetHeader;
    this._NgIf_31_6.check_ngIf(currVal_31_0_0,throwOnChange,false);
    this._NgIf_31_6.ngDoCheck(this,this._anchor_31,throwOnChange);
    const currVal_33_0_0:any = this.context.targetStyle;
    this._NgStyle_33_3.check_ngStyle(currVal_33_0_0,throwOnChange,false);
    this._NgStyle_33_3.ngDoCheck(this,this._el_33,throwOnChange);
    const currVal_35_0_0:any = this.context.target;
    this._NgFor_35_6.check_ngForOf(currVal_35_0_0,throwOnChange,false);
    this._NgFor_35_6.ngDoCheck(this,this._anchor_35,throwOnChange);
    const currVal_39_0_0:any = this.context.showTargetControls;
    this._NgIf_39_6.check_ngIf(currVal_39_0_0,throwOnChange,false);
    this._NgIf_39_6.ngDoCheck(this,this._anchor_39,throwOnChange);
    this._vc_3.detectChangesInNestedViews(throwOnChange);
    this._vc_7.detectChangesInNestedViews(throwOnChange);
    this._vc_11.detectChangesInNestedViews(throwOnChange);
    this._vc_31.detectChangesInNestedViews(throwOnChange);
    this._vc_35.detectChangesInNestedViews(throwOnChange);
    this._vc_39.detectChangesInNestedViews(throwOnChange);
    this._Button_19_4.checkHost(this,this,this._el_19,throwOnChange);
    this._Button_21_4.checkHost(this,this,this._el_21,throwOnChange);
    this._Button_23_4.checkHost(this,this,this._el_23,throwOnChange);
    this._Button_25_4.checkHost(this,this,this._el_25,throwOnChange);
    if (!throwOnChange) {
      if ((this.numberOfChecks === 0)) { this._Button_19_4.context.ngAfterViewInit(); }
      if ((this.numberOfChecks === 0)) { this._Button_21_4.context.ngAfterViewInit(); }
      if ((this.numberOfChecks === 0)) { this._Button_23_4.context.ngAfterViewInit(); }
      if ((this.numberOfChecks === 0)) { this._Button_25_4.context.ngAfterViewInit(); }
    }
  }
  destroyInternal():void {
    this._vc_3.destroyNestedViews();
    this._vc_7.destroyNestedViews();
    this._vc_11.destroyNestedViews();
    this._vc_31.destroyNestedViews();
    this._vc_35.destroyNestedViews();
    this._vc_39.destroyNestedViews();
    this._Button_19_4.ngOnDestroy();
    this._Button_21_4.ngOnDestroy();
    this._Button_23_4.ngOnDestroy();
    this._Button_25_4.ngOnDestroy();
  }
  createEmbeddedViewInternal(nodeIndex:number):import9.AppView<any> {
    if ((nodeIndex == 3)) { return new View_PickList1(this.viewUtils,this,3,this._anchor_3,this._vc_3); }
    if ((nodeIndex == 7)) { return new View_PickList2(this.viewUtils,this,7,this._anchor_7,this._vc_7); }
    if ((nodeIndex == 11)) { return new View_PickList3(this.viewUtils,this,11,this._anchor_11,this._vc_11); }
    if ((nodeIndex == 31)) { return new View_PickList5(this.viewUtils,this,31,this._anchor_31,this._vc_31); }
    if ((nodeIndex == 35)) { return new View_PickList6(this.viewUtils,this,35,this._anchor_35,this._vc_35); }
    if ((nodeIndex == 39)) { return new View_PickList8(this.viewUtils,this,39,this._anchor_39,this._vc_39); }
    return (null as any);
  }
  handleEvent_19(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    result = (this._Button_19_4.handleEvent(eventName,$event) && result);
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.context.moveRight(this._el_33)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
  handleEvent_21(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    result = (this._Button_21_4.handleEvent(eventName,$event) && result);
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.context.moveAllRight()) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
  handleEvent_23(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    result = (this._Button_23_4.handleEvent(eventName,$event) && result);
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.context.moveLeft(this._el_9)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
  handleEvent_25(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    result = (this._Button_25_4.handleEvent(eventName,$event) && result);
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.context.moveAllLeft()) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
