// primeng/components/dropdown/dropdown.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from '@angular/core/src/linker/ng_module_factory';
import * as import1 from 'primeng/components/dropdown/dropdown';
import * as import2 from '@angular/common/src/common_module';
import * as import3 from 'primeng/components/common/shared';
import * as import4 from '@angular/common/src/localization';
import * as import5 from '@angular/core/src/di/injector';
import * as import6 from '@angular/core/src/i18n/tokens';
import * as import7 from '@angular/core/src/change_detection/change_detection_util';
import * as import8 from '@angular/core/src/linker/view';
import * as import9 from '@angular/core/src/linker/view_utils';
import * as import10 from '@angular/core/src/render/api';
import * as import11 from '@angular/core/src/metadata/view';
import * as import12 from 'primeng/components/dom/domhandler';
import * as import13 from '@angular/core/src/linker/query_list';
import * as import14 from '@angular/core/src/linker/view_type';
import * as import15 from '@angular/core/src/change_detection/constants';
import * as import16 from '@angular/core/src/linker/component_factory';
import * as import17 from '@angular/core/src/linker/element_ref';
import * as import18 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import19 from '@angular/forms/src/directives/control_value_accessor';
import * as import20 from '@angular/core/src/linker/view_container';
import * as import21 from '../../../@angular/common/src/directives/ng_class.ngfactory';
import * as import22 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import23 from '@angular/common/src/directives/ng_class';
import * as import24 from '../common/shared.ngfactory';
import * as import25 from '@angular/core/src/linker/template_ref';
import * as import26 from '../../../@angular/common/src/directives/ng_if.ngfactory';
import * as import27 from '@angular/common/src/directives/ng_if';
import * as import28 from '../../../@angular/common/src/directives/ng_style.ngfactory';
import * as import29 from '../../../@angular/common/src/directives/ng_for.ngfactory';
import * as import30 from '@angular/common/src/directives/ng_for';
import * as import31 from '@angular/common/src/directives/ng_style';
import * as import32 from '@angular/core/src/security';
class DropdownModuleInjector extends import0.NgModuleInjector<import1.DropdownModule> {
  _CommonModule_0:import2.CommonModule;
  _SharedModule_1:import3.SharedModule;
  _DropdownModule_2:import1.DropdownModule;
  __NgLocalization_3:import4.NgLocaleLocalization;
  constructor(parent:import5.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_3():import4.NgLocaleLocalization {
    if ((this.__NgLocalization_3 == null)) { (this.__NgLocalization_3 = new import4.NgLocaleLocalization(this.parent.get(import6.LOCALE_ID))); }
    return this.__NgLocalization_3;
  }
  createInternal():import1.DropdownModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._SharedModule_1 = new import3.SharedModule();
    this._DropdownModule_2 = new import1.DropdownModule();
    return this._DropdownModule_2;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.SharedModule)) { return this._SharedModule_1; }
    if ((token === import1.DropdownModule)) { return this._DropdownModule_2; }
    if ((token === import4.NgLocalization)) { return this._NgLocalization_3; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const DropdownModuleNgFactory:import0.NgModuleFactory<import1.DropdownModule> = new import0.NgModuleFactory(DropdownModuleInjector,import1.DropdownModule);
export class Wrapper_Dropdown {
  /*private*/ _eventHandler:Function;
  context:import1.Dropdown;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  /*private*/ _expr_5:any;
  /*private*/ _expr_6:any;
  /*private*/ _expr_7:any;
  /*private*/ _expr_8:any;
  /*private*/ _expr_9:any;
  subscription0:any;
  subscription1:any;
  subscription2:any;
  constructor(p0:any,p1:any,p2:any,p3:any) {
    this._changed = false;
    this.context = new import1.Dropdown(p0,p1,p2,p3);
    this._expr_0 = import7.UNINITIALIZED;
    this._expr_1 = import7.UNINITIALIZED;
    this._expr_2 = import7.UNINITIALIZED;
    this._expr_3 = import7.UNINITIALIZED;
    this._expr_4 = import7.UNINITIALIZED;
    this._expr_5 = import7.UNINITIALIZED;
    this._expr_6 = import7.UNINITIALIZED;
    this._expr_7 = import7.UNINITIALIZED;
    this._expr_8 = import7.UNINITIALIZED;
    this._expr_9 = import7.UNINITIALIZED;
  }
  ngOnDetach(view:import8.AppView<any>,componentView:import8.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
    this.context.ngOnDestroy();
    (this.subscription0 && this.subscription0.unsubscribe());
    (this.subscription1 && this.subscription1.unsubscribe());
    (this.subscription2 && this.subscription2.unsubscribe());
  }
  check_options(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.options = currValue;
      this._expr_0 = currValue;
    }
  }
  check_scrollHeight(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.scrollHeight = currValue;
      this._expr_1 = currValue;
    }
  }
  check_filter(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_2,currValue))) {
      this._changed = true;
      this.context.filter = currValue;
      this._expr_2 = currValue;
    }
  }
  check_style(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_3,currValue))) {
      this._changed = true;
      this.context.style = currValue;
      this._expr_3 = currValue;
    }
  }
  check_styleClass(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_4,currValue))) {
      this._changed = true;
      this.context.styleClass = currValue;
      this._expr_4 = currValue;
    }
  }
  check_disabled(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_5,currValue))) {
      this._changed = true;
      this.context.disabled = currValue;
      this._expr_5 = currValue;
    }
  }
  check_autoWidth(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_6,currValue))) {
      this._changed = true;
      this.context.autoWidth = currValue;
      this._expr_6 = currValue;
    }
  }
  check_required(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_7,currValue))) {
      this._changed = true;
      this.context.required = currValue;
      this._expr_7 = currValue;
    }
  }
  check_editable(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_8,currValue))) {
      this._changed = true;
      this.context.editable = currValue;
      this._expr_8 = currValue;
    }
  }
  check_appendTo(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_9,currValue))) {
      this._changed = true;
      this.context.appendTo = currValue;
      this._expr_9 = currValue;
    }
  }
  ngDoCheck(view:import8.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    if (!throwOnChange) {
      if ((view.numberOfChecks === 0)) { this.context.ngOnInit(); }
      this.context.ngDoCheck();
    }
    return changed;
  }
  checkHost(view:import8.AppView<any>,componentView:import8.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import8.AppView<any>,_eventHandler:any,emit0:boolean,emit1:boolean,emit2:boolean):void {
    this._eventHandler = _eventHandler;
    if (emit0) { (this.subscription0 = this.context.onChange.subscribe(_eventHandler.bind(view,'onChange'))); }
    if (emit1) { (this.subscription1 = this.context.onFocus.subscribe(_eventHandler.bind(view,'onFocus'))); }
    if (emit2) { (this.subscription2 = this.context.onBlur.subscribe(_eventHandler.bind(view,'onBlur'))); }
  }
}
var renderType_Dropdown_Host:import10.RenderComponentType = import9.createRenderComponentType('',0,import11.ViewEncapsulation.None,([] as any[]),{});
class View_Dropdown_Host0 extends import8.AppView<any> {
  _el_0:any;
  compView_0:import8.AppView<import1.Dropdown>;
  _DomHandler_0_3:import12.DomHandler;
  _Dropdown_0_4:Wrapper_Dropdown;
  __NG_VALUE_ACCESSOR_0_5:any[];
  _query_TemplateRef_0_0:import13.QueryList<any>;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_Dropdown_Host0,renderType_Dropdown_Host,import14.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways);
  }
  get _NG_VALUE_ACCESSOR_0_5():any[] {
    if ((this.__NG_VALUE_ACCESSOR_0_5 == null)) { (this.__NG_VALUE_ACCESSOR_0_5 = [this._Dropdown_0_4.context]); }
    return this.__NG_VALUE_ACCESSOR_0_5;
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    this._el_0 = import9.selectOrCreateRenderHostElement(this.renderer,'p-dropdown',import9.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this.compView_0 = new View_Dropdown0(this.viewUtils,this,0,this._el_0);
    this._DomHandler_0_3 = new import12.DomHandler();
    this._Dropdown_0_4 = new Wrapper_Dropdown(new import17.ElementRef(this._el_0),this._DomHandler_0_3,this.renderer,this.injectorGet(import18.IterableDiffers,this.parentIndex));
    this._query_TemplateRef_0_0 = new import13.QueryList<any>();
    this.compView_0.create(this._Dropdown_0_4.context);
    this._query_TemplateRef_0_0.reset(([] as any[]));
    this._Dropdown_0_4.context.itemTemplate = this._query_TemplateRef_0_0.first;
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import16.ComponentRef_<any>(0,this,this._el_0,this._Dropdown_0_4.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import12.DomHandler) && (0 === requestNodeIndex))) { return this._DomHandler_0_3; }
    if (((token === import1.Dropdown) && (0 === requestNodeIndex))) { return this._Dropdown_0_4.context; }
    if (((token === import19.NG_VALUE_ACCESSOR) && (0 === requestNodeIndex))) { return this._NG_VALUE_ACCESSOR_0_5; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._Dropdown_0_4.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
    if (!throwOnChange) {
      if ((this.numberOfChecks === 0)) { this._Dropdown_0_4.context.ngAfterViewInit(); }
      this._Dropdown_0_4.context.ngAfterViewChecked();
    }
  }
  destroyInternal():void {
    this.compView_0.destroy();
    this._Dropdown_0_4.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
export const DropdownNgFactory:import16.ComponentFactory<import1.Dropdown> = new import16.ComponentFactory<import1.Dropdown>('p-dropdown',View_Dropdown_Host0,import1.Dropdown);
const styles_Dropdown:any[] = ([] as any[]);
class View_Dropdown1 extends import8.AppView<any> {
  _el_0:any;
  _text_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import20.ViewContainer) {
    super(View_Dropdown1,renderType_Dropdown,import14.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_2 = import7.UNINITIALIZED;
    this._expr_3 = import7.UNINITIALIZED;
    this._expr_4 = import7.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'option',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1
    ]
    ),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2:any = this.context.$implicit.value;
    if (import9.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setElementProperty(this._el_0,'value',currVal_2);
      this._expr_2 = currVal_2;
    }
    const currVal_3:any = (this.parentView.context.selectedOption == this.context.$implicit);
    if (import9.checkBinding(throwOnChange,this._expr_3,currVal_3)) {
      this.renderer.setElementProperty(this._el_0,'selected',currVal_3);
      this._expr_3 = currVal_3;
    }
    const currVal_4:any = import9.inlineInterpolate(1,'',this.context.$implicit.label,'');
    if (import9.checkBinding(throwOnChange,this._expr_4,currVal_4)) {
      this.renderer.setText(this._text_1,currVal_4);
      this._expr_4 = currVal_4;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_Dropdown2 extends import8.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import21.Wrapper_NgClass;
  _text_1:any;
  _map_3:any;
  /*private*/ _expr_4:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import20.ViewContainer) {
    super(View_Dropdown2,renderType_Dropdown,import14.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_3 = import9.pureProxy2((p0:any,p1:any):{[key: string]:any} => {
      return {
        'ui-dropdown-label ui-inputtext ui-corner-all': p0,
        'ui-dropdown-label-empty': p1
      }
      ;
    });
    this._expr_4 = import7.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'label',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_0_3 = new import21.Wrapper_NgClass(this.parentView.parentView.injectorGet(import18.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import22.KeyValueDiffers,this.parentView.parentIndex),new import17.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import23.NgClass) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 1)))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this._map_3(true,!this.parentView.context.label);
    this._NgClass_0_3.check_ngClass(currVal_0_0_0,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_4:any = import9.inlineInterpolate(1,'',(this.parentView.context.label || 'empty'),'');
    if (import9.checkBinding(throwOnChange,this._expr_4,currVal_4)) {
      this.renderer.setText(this._text_1,currVal_4);
      this._expr_4 = currVal_4;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_Dropdown3 extends import8.AppView<any> {
  _el_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import20.ViewContainer) {
    super(View_Dropdown3,renderType_Dropdown,import14.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_1 = import7.UNINITIALIZED;
    this._expr_2 = import7.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'input',new import9.InlineArray4(4,'class','ui-dropdown-label ui-inputtext ui-corner-all','type','text'),(null as any));
    var disposable_0:Function = import9.subscribeToRenderElement(this,this._el_0,new import9.InlineArray8(8,'click',(null as any),'input',(null as any),'focus',(null as any),'blur',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),[disposable_0]);
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1:any = this.parentView.context.label;
    if (import9.checkBinding(throwOnChange,this._expr_1,currVal_1)) {
      this.renderer.setElementProperty(this._el_0,'value',currVal_1);
      this._expr_1 = currVal_1;
    }
    const currVal_2:any = this.parentView.context.disabled;
    if (import9.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setElementProperty(this._el_0,'disabled',currVal_2);
      this._expr_2 = currVal_2;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.context.onEditableInputClick($event)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'input')) {
      const pd_sub_1:any = ((<any>this.parentView.context.onEditableInputChange($event)) !== false);
      result = (pd_sub_1 && result);
    }
    if ((eventName == 'focus')) {
      const pd_sub_2:any = ((<any>this.parentView.context.onEditableInputFocus($event)) !== false);
      result = (pd_sub_2 && result);
    }
    if ((eventName == 'blur')) {
      const pd_sub_3:any = ((<any>this.parentView.context.onInputBlur($event)) !== false);
      result = (pd_sub_3 && result);
    }
    return result;
  }
}
class View_Dropdown4 extends import8.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _el_4:any;
  _text_5:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import20.ViewContainer) {
    super(View_Dropdown4,renderType_Dropdown,import14.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'div',new import9.InlineArray2(2,'class','ui-dropdown-filter-container'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this._el_2 = import9.createRenderElement(this.renderer,this._el_0,'input',new import9.InlineArray8(6,'autocomplete','off','class','ui-dropdown-filter ui-inputtext ui-widget ui-state-default ui-corner-all','type','text'),(null as any));
    this._text_3 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this._el_4 = import9.createRenderElement(this.renderer,this._el_0,'span',new import9.InlineArray2(2,'class','fa fa-search'),(null as any));
    this._text_5 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    var disposable_0:Function = import9.subscribeToRenderElement(this,this._el_0,new import9.InlineArray4(4,'input',(null as any),'click',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._el_4,
      this._text_5
    ]
    ),[disposable_0]);
    return (null as any);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'input')) {
      const pd_sub_0:any = ((<any>this.parentView.context.onFilter($event)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'click')) {
      const pd_sub_1:any = ((<any>$event.stopPropagation()) !== false);
      result = (pd_sub_1 && result);
    }
    return result;
  }
}
class View_Dropdown6 extends import8.AppView<any> {
  _el_0:any;
  _text_1:any;
  /*private*/ _expr_2:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import20.ViewContainer) {
    super(View_Dropdown6,renderType_Dropdown,import14.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_2 = import7.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'span',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1
    ]
    ),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2:any = import9.inlineInterpolate(1,'',(this.parentView.context.$implicit.label || 'empty'),'');
    if (import9.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setText(this._text_1,currVal_2);
      this._expr_2 = currVal_2;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_Dropdown8 extends import8.AppView<any> {
  _el_0:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import20.ViewContainer) {
    super(View_Dropdown8,renderType_Dropdown,import14.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    this._el_0 = this.renderer.createTemplateAnchor((null as any),(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): ([] as any[])),(null as any));
    return (null as any);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_Dropdown7 extends import8.AppView<any> {
  _anchor_0:any;
  /*private*/ _vc_0:import20.ViewContainer;
  _TemplateRef_0_5:any;
  _TemplateWrapper_0_6:import24.Wrapper_TemplateWrapper;
  _el_1:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import20.ViewContainer) {
    super(View_Dropdown7,renderType_Dropdown,import14.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    this._anchor_0 = this.renderer.createTemplateAnchor((null as any),(null as any));
    this._vc_0 = new import20.ViewContainer(0,(null as any),this,this._anchor_0);
    this._TemplateRef_0_5 = new import25.TemplateRef_(this,0,this._anchor_0);
    this._TemplateWrapper_0_6 = new import24.Wrapper_TemplateWrapper(this._vc_0.vcRef);
    this._el_1 = this.renderer.createTemplateAnchor((null as any),(null as any));
    this.init(this._el_1,((<any>this.renderer).directRenderer? (null as any): [this._anchor_0]),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import25.TemplateRef) && (0 === requestNodeIndex))) { return this._TemplateRef_0_5; }
    if (((token === import3.TemplateWrapper) && (0 === requestNodeIndex))) { return this._TemplateWrapper_0_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this.parentView.context.$implicit;
    this._TemplateWrapper_0_6.check_item(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = this.parentView.parentView.context.itemTemplate;
    this._TemplateWrapper_0_6.check_templateRef(currVal_0_0_1,throwOnChange,false);
    this._TemplateWrapper_0_6.ngDoCheck(this,this._anchor_0,throwOnChange);
    this._vc_0.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_0.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._vc_0.nativeElement,ctx);
    this._vc_0.visitNestedViewRootNodes(cb,ctx);
    cb(this._el_1,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import8.AppView<any> {
    if ((nodeIndex == 0)) { return new View_Dropdown8(this.viewUtils,this,0,this._anchor_0,this._vc_0); }
    return (null as any);
  }
}
class View_Dropdown5 extends import8.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import21.Wrapper_NgClass;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import20.ViewContainer;
  _TemplateRef_2_5:any;
  _NgIf_2_6:import26.Wrapper_NgIf;
  _text_3:any;
  _anchor_4:any;
  /*private*/ _vc_4:import20.ViewContainer;
  _TemplateRef_4_5:any;
  _NgIf_4_6:import26.Wrapper_NgIf;
  _text_5:any;
  _map_13:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import20.ViewContainer) {
    super(View_Dropdown5,renderType_Dropdown,import14.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_13 = import9.pureProxy4((p0:any,p1:any,p2:any,p3:any):{[key: string]:any} => {
      return {
        'ui-dropdown-item ui-corner-all': p0,
        'ui-state-hover': p1,
        'ui-state-highlight': p2,
        'ui-dropdown-item-empty': p3
      }
      ;
    });
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'li',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_0_3 = new import21.Wrapper_NgClass(this.parentView.parentView.injectorGet(import18.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import22.KeyValueDiffers,this.parentView.parentIndex),new import17.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n                            ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import20.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import25.TemplateRef_(this,2,this._anchor_2);
    this._NgIf_2_6 = new import26.Wrapper_NgIf(this._vc_2.vcRef,this._TemplateRef_2_5);
    this._text_3 = this.renderer.createText(this._el_0,'\n                            ',(null as any));
    this._anchor_4 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_4 = new import20.ViewContainer(4,0,this,this._anchor_4);
    this._TemplateRef_4_5 = new import25.TemplateRef_(this,4,this._anchor_4);
    this._NgIf_4_6 = new import26.Wrapper_NgIf(this._vc_4.vcRef,this._TemplateRef_4_5);
    this._text_5 = this.renderer.createText(this._el_0,'\n                        ',(null as any));
    var disposable_0:Function = import9.subscribeToRenderElement(this,this._el_0,new import9.InlineArray8(6,'click',(null as any),'mouseenter',(null as any),'mouseleave',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3,
      this._anchor_4,
      this._text_5
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import25.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import27.NgIf) && (2 === requestNodeIndex))) { return this._NgIf_2_6.context; }
    if (((token === import25.TemplateRef) && (4 === requestNodeIndex))) { return this._TemplateRef_4_5; }
    if (((token === import27.NgIf) && (4 === requestNodeIndex))) { return this._NgIf_4_6.context; }
    if (((token === import23.NgClass) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 5)))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this._map_13(true,(this.parentView.context.hoveredItem == this._el_0),(this.parentView.context.selectedOption == this.context.$implicit),(!this.context.$implicit.label || (this.context.$implicit.label.length === 0)));
    this._NgClass_0_3.check_ngClass(currVal_0_0_0,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_2_0_0:boolean = !this.parentView.context.itemTemplate;
    this._NgIf_2_6.check_ngIf(currVal_2_0_0,throwOnChange,false);
    this._NgIf_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    const currVal_4_0_0:any = this.parentView.context.itemTemplate;
    this._NgIf_4_6.check_ngIf(currVal_4_0_0,throwOnChange,false);
    this._NgIf_4_6.ngDoCheck(this,this._anchor_4,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
    this._vc_4.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
    this._vc_4.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import8.AppView<any> {
    if ((nodeIndex == 2)) { return new View_Dropdown6(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    if ((nodeIndex == 4)) { return new View_Dropdown7(this.viewUtils,this,4,this._anchor_4,this._vc_4); }
    return (null as any);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.context.onItemClick($event,this.context.$implicit)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'mouseenter')) {
      const pd_sub_1:any = ((<any>(this.parentView.context.hoveredItem = this._el_0)) !== false);
      result = (pd_sub_1 && result);
    }
    if ((eventName == 'mouseleave')) {
      const pd_sub_2:any = ((<any>(this.parentView.context.hoveredItem = (null as any))) !== false);
      result = (pd_sub_2 && result);
    }
    return result;
  }
}
var renderType_Dropdown:import10.RenderComponentType = import9.createRenderComponentType('',0,import11.ViewEncapsulation.None,styles_Dropdown,{});
export class View_Dropdown0 extends import8.AppView<import1.Dropdown> {
  _viewQuery_container_0:import13.QueryList<any>;
  _viewQuery_panel_1:import13.QueryList<any>;
  _viewQuery_itemswrapper_2:import13.QueryList<any>;
  _text_0:any;
  _el_1:any;
  _NgClass_1_3:import21.Wrapper_NgClass;
  _NgStyle_1_4:import28.Wrapper_NgStyle;
  _text_2:any;
  _el_3:any;
  _text_4:any;
  _el_5:any;
  _text_6:any;
  _anchor_7:any;
  /*private*/ _vc_7:import20.ViewContainer;
  _TemplateRef_7_5:any;
  _NgFor_7_6:import29.Wrapper_NgFor;
  _text_8:any;
  _text_9:any;
  _text_10:any;
  _el_11:any;
  _text_12:any;
  _el_13:any;
  _text_14:any;
  _text_15:any;
  _anchor_16:any;
  /*private*/ _vc_16:import20.ViewContainer;
  _TemplateRef_16_5:any;
  _NgIf_16_6:import26.Wrapper_NgIf;
  _text_17:any;
  _anchor_18:any;
  /*private*/ _vc_18:import20.ViewContainer;
  _TemplateRef_18_5:any;
  _NgIf_18_6:import26.Wrapper_NgIf;
  _text_19:any;
  _el_20:any;
  _NgClass_20_3:import21.Wrapper_NgClass;
  _text_21:any;
  _el_22:any;
  _text_23:any;
  _text_24:any;
  _el_25:any;
  _text_26:any;
  _anchor_27:any;
  /*private*/ _vc_27:import20.ViewContainer;
  _TemplateRef_27_5:any;
  _NgIf_27_6:import26.Wrapper_NgIf;
  _text_28:any;
  _el_29:any;
  _text_30:any;
  _el_31:any;
  _text_32:any;
  _anchor_33:any;
  /*private*/ _vc_33:import20.ViewContainer;
  _TemplateRef_33_5:any;
  _NgFor_33_6:import29.Wrapper_NgFor;
  _text_34:any;
  _text_35:any;
  _text_36:any;
  _text_37:any;
  _text_38:any;
  _map_60:any;
  /*private*/ _expr_61:any;
  /*private*/ _expr_62:any;
  _map_63:any;
  /*private*/ _expr_64:any;
  /*private*/ _expr_65:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_Dropdown0,renderType_Dropdown,import14.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways);
    this._map_60 = import9.pureProxy5((p0:any,p1:any,p2:any,p3:any,p4:any):{[key: string]:any} => {
      return {
        'ui-dropdown ui-widget ui-state-default ui-corner-all ui-helper-clearfix': p0,
        'ui-state-hover': p1,
        'ui-state-focus': p2,
        'ui-state-disabled': p3,
        'ui-dropdown-open': p4
      }
      ;
    });
    this._expr_61 = import7.UNINITIALIZED;
    this._expr_62 = import7.UNINITIALIZED;
    this._map_63 = import9.pureProxy2((p0:any,p1:any):{[key: string]:any} => {
      return {
        'ui-state-hover': p0,
        'ui-state-focus': p1
      }
      ;
    });
    this._expr_64 = import7.UNINITIALIZED;
    this._expr_65 = import7.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._viewQuery_container_0 = new import13.QueryList<any>();
    this._viewQuery_panel_1 = new import13.QueryList<any>();
    this._viewQuery_itemswrapper_2 = new import13.QueryList<any>();
    this._text_0 = this.renderer.createText(parentRenderNode,'\n         ',(null as any));
    this._el_1 = import9.createRenderElement(this.renderer,parentRenderNode,'div',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_1_3 = new import21.Wrapper_NgClass(this.parentView.injectorGet(import18.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import22.KeyValueDiffers,this.parentIndex),new import17.ElementRef(this._el_1),this.renderer);
    this._NgStyle_1_4 = new import28.Wrapper_NgStyle(this.parentView.injectorGet(import22.KeyValueDiffers,this.parentIndex),new import17.ElementRef(this._el_1),this.renderer);
    this._text_2 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_3 = import9.createRenderElement(this.renderer,this._el_1,'div',new import9.InlineArray2(2,'class','ui-helper-hidden-accessible'),(null as any));
    this._text_4 = this.renderer.createText(this._el_3,'\n                ',(null as any));
    this._el_5 = import9.createRenderElement(this.renderer,this._el_3,'select',new import9.InlineArray2(2,'tabindex','-1'),(null as any));
    this._text_6 = this.renderer.createText(this._el_5,'\n                    ',(null as any));
    this._anchor_7 = this.renderer.createTemplateAnchor(this._el_5,(null as any));
    this._vc_7 = new import20.ViewContainer(7,5,this,this._anchor_7);
    this._TemplateRef_7_5 = new import25.TemplateRef_(this,7,this._anchor_7);
    this._NgFor_7_6 = new import29.Wrapper_NgFor(this._vc_7.vcRef,this._TemplateRef_7_5,this.parentView.injectorGet(import18.IterableDiffers,this.parentIndex),this.ref);
    this._text_8 = this.renderer.createText(this._el_5,'\n                ',(null as any));
    this._text_9 = this.renderer.createText(this._el_3,'\n            ',(null as any));
    this._text_10 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_11 = import9.createRenderElement(this.renderer,this._el_1,'div',new import9.InlineArray2(2,'class','ui-helper-hidden-accessible'),(null as any));
    this._text_12 = this.renderer.createText(this._el_11,'\n                ',(null as any));
    this._el_13 = import9.createRenderElement(this.renderer,this._el_11,'input',new import9.InlineArray4(4,'readonly','','type','text'),(null as any));
    this._text_14 = this.renderer.createText(this._el_11,'\n            ',(null as any));
    this._text_15 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._anchor_16 = this.renderer.createTemplateAnchor(this._el_1,(null as any));
    this._vc_16 = new import20.ViewContainer(16,1,this,this._anchor_16);
    this._TemplateRef_16_5 = new import25.TemplateRef_(this,16,this._anchor_16);
    this._NgIf_16_6 = new import26.Wrapper_NgIf(this._vc_16.vcRef,this._TemplateRef_16_5);
    this._text_17 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._anchor_18 = this.renderer.createTemplateAnchor(this._el_1,(null as any));
    this._vc_18 = new import20.ViewContainer(18,1,this,this._anchor_18);
    this._TemplateRef_18_5 = new import25.TemplateRef_(this,18,this._anchor_18);
    this._NgIf_18_6 = new import26.Wrapper_NgIf(this._vc_18.vcRef,this._TemplateRef_18_5);
    this._text_19 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_20 = import9.createRenderElement(this.renderer,this._el_1,'div',new import9.InlineArray2(2,'class','ui-dropdown-trigger ui-state-default ui-corner-right'),(null as any));
    this._NgClass_20_3 = new import21.Wrapper_NgClass(this.parentView.injectorGet(import18.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import22.KeyValueDiffers,this.parentIndex),new import17.ElementRef(this._el_20),this.renderer);
    this._text_21 = this.renderer.createText(this._el_20,'\n                ',(null as any));
    this._el_22 = import9.createRenderElement(this.renderer,this._el_20,'span',new import9.InlineArray2(2,'class','fa fa-fw fa-caret-down ui-c'),(null as any));
    this._text_23 = this.renderer.createText(this._el_20,'\n            ',(null as any));
    this._text_24 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_25 = import9.createRenderElement(this.renderer,this._el_1,'div',new import9.InlineArray2(2,'class','ui-dropdown-panel ui-widget-content ui-corner-all ui-helper-hidden ui-shadow'),(null as any));
    this._text_26 = this.renderer.createText(this._el_25,'\n                ',(null as any));
    this._anchor_27 = this.renderer.createTemplateAnchor(this._el_25,(null as any));
    this._vc_27 = new import20.ViewContainer(27,25,this,this._anchor_27);
    this._TemplateRef_27_5 = new import25.TemplateRef_(this,27,this._anchor_27);
    this._NgIf_27_6 = new import26.Wrapper_NgIf(this._vc_27.vcRef,this._TemplateRef_27_5);
    this._text_28 = this.renderer.createText(this._el_25,'\n                ',(null as any));
    this._el_29 = import9.createRenderElement(this.renderer,this._el_25,'div',new import9.InlineArray2(2,'class','ui-dropdown-items-wrapper'),(null as any));
    this._text_30 = this.renderer.createText(this._el_29,'\n                    ',(null as any));
    this._el_31 = import9.createRenderElement(this.renderer,this._el_29,'ul',new import9.InlineArray2(2,'class','ui-dropdown-items ui-dropdown-list ui-widget-content ui-widget ui-corner-all ui-helper-reset'),(null as any));
    this._text_32 = this.renderer.createText(this._el_31,'\n                        ',(null as any));
    this._anchor_33 = this.renderer.createTemplateAnchor(this._el_31,(null as any));
    this._vc_33 = new import20.ViewContainer(33,31,this,this._anchor_33);
    this._TemplateRef_33_5 = new import25.TemplateRef_(this,33,this._anchor_33);
    this._NgFor_33_6 = new import29.Wrapper_NgFor(this._vc_33.vcRef,this._TemplateRef_33_5,this.parentView.injectorGet(import18.IterableDiffers,this.parentIndex),this.ref);
    this._text_34 = this.renderer.createText(this._el_31,'\n                    ',(null as any));
    this._text_35 = this.renderer.createText(this._el_29,'\n                ',(null as any));
    this._text_36 = this.renderer.createText(this._el_25,'\n            ',(null as any));
    this._text_37 = this.renderer.createText(this._el_1,'\n        ',(null as any));
    this._text_38 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    var disposable_0:Function = import9.subscribeToRenderElement(this,this._el_1,new import9.InlineArray8(6,'mouseenter',(null as any),'mouseleave',(null as any),'click',(null as any)),this.eventHandler(this.handleEvent_1));
    var disposable_1:Function = import9.subscribeToRenderElement(this,this._el_13,new import9.InlineArray8(6,'focus',(null as any),'blur',(null as any),'keydown',(null as any)),this.eventHandler(this.handleEvent_13));
    this._viewQuery_container_0.reset([new import17.ElementRef(this._el_1)]);
    this.context.containerViewChild = this._viewQuery_container_0.first;
    this._viewQuery_panel_1.reset([new import17.ElementRef(this._el_25)]);
    this.context.panelViewChild = this._viewQuery_panel_1.first;
    this._viewQuery_itemswrapper_2.reset([new import17.ElementRef(this._el_29)]);
    this.context.itemsWrapperViewChild = this._viewQuery_itemswrapper_2.first;
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2,
      this._el_3,
      this._text_4,
      this._el_5,
      this._text_6,
      this._anchor_7,
      this._text_8,
      this._text_9,
      this._text_10,
      this._el_11,
      this._text_12,
      this._el_13,
      this._text_14,
      this._text_15,
      this._anchor_16,
      this._text_17,
      this._anchor_18,
      this._text_19,
      this._el_20,
      this._text_21,
      this._el_22,
      this._text_23,
      this._text_24,
      this._el_25,
      this._text_26,
      this._anchor_27,
      this._text_28,
      this._el_29,
      this._text_30,
      this._el_31,
      this._text_32,
      this._anchor_33,
      this._text_34,
      this._text_35,
      this._text_36,
      this._text_37,
      this._text_38
    ]
    ),[
      disposable_0,
      disposable_1
    ]
    );
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import25.TemplateRef) && (7 === requestNodeIndex))) { return this._TemplateRef_7_5; }
    if (((token === import30.NgFor) && (7 === requestNodeIndex))) { return this._NgFor_7_6.context; }
    if (((token === import25.TemplateRef) && (16 === requestNodeIndex))) { return this._TemplateRef_16_5; }
    if (((token === import27.NgIf) && (16 === requestNodeIndex))) { return this._NgIf_16_6.context; }
    if (((token === import25.TemplateRef) && (18 === requestNodeIndex))) { return this._TemplateRef_18_5; }
    if (((token === import27.NgIf) && (18 === requestNodeIndex))) { return this._NgIf_18_6.context; }
    if (((token === import23.NgClass) && ((20 <= requestNodeIndex) && (requestNodeIndex <= 23)))) { return this._NgClass_20_3.context; }
    if (((token === import25.TemplateRef) && (27 === requestNodeIndex))) { return this._TemplateRef_27_5; }
    if (((token === import27.NgIf) && (27 === requestNodeIndex))) { return this._NgIf_27_6.context; }
    if (((token === import25.TemplateRef) && (33 === requestNodeIndex))) { return this._TemplateRef_33_5; }
    if (((token === import30.NgFor) && (33 === requestNodeIndex))) { return this._NgFor_33_6.context; }
    if (((token === import23.NgClass) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 37)))) { return this._NgClass_1_3.context; }
    if (((token === import31.NgStyle) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 37)))) { return this._NgStyle_1_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1_0_0:any = this.context.styleClass;
    this._NgClass_1_3.check_klass(currVal_1_0_0,throwOnChange,false);
    const currVal_1_0_1:any = this._map_60(true,(this.context.hover && !this.context.disabled),this.context.focus,this.context.disabled,this.context.panelVisible);
    this._NgClass_1_3.check_ngClass(currVal_1_0_1,throwOnChange,false);
    this._NgClass_1_3.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_1_1_0:any = this.context.style;
    this._NgStyle_1_4.check_ngStyle(currVal_1_1_0,throwOnChange,false);
    this._NgStyle_1_4.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_7_0_0:any = this.context.options;
    this._NgFor_7_6.check_ngForOf(currVal_7_0_0,throwOnChange,false);
    this._NgFor_7_6.ngDoCheck(this,this._anchor_7,throwOnChange);
    const currVal_16_0_0:boolean = !this.context.editable;
    this._NgIf_16_6.check_ngIf(currVal_16_0_0,throwOnChange,false);
    this._NgIf_16_6.ngDoCheck(this,this._anchor_16,throwOnChange);
    const currVal_18_0_0:any = this.context.editable;
    this._NgIf_18_6.check_ngIf(currVal_18_0_0,throwOnChange,false);
    this._NgIf_18_6.ngDoCheck(this,this._anchor_18,throwOnChange);
    const currVal_20_0_0:any = 'ui-dropdown-trigger ui-state-default ui-corner-right';
    this._NgClass_20_3.check_klass(currVal_20_0_0,throwOnChange,false);
    const currVal_20_0_1:any = this._map_63((this.context.hover && !this.context.disabled),this.context.focus);
    this._NgClass_20_3.check_ngClass(currVal_20_0_1,throwOnChange,false);
    this._NgClass_20_3.ngDoCheck(this,this._el_20,throwOnChange);
    const currVal_27_0_0:any = this.context.filter;
    this._NgIf_27_6.check_ngIf(currVal_27_0_0,throwOnChange,false);
    this._NgIf_27_6.ngDoCheck(this,this._anchor_27,throwOnChange);
    const currVal_33_0_0:any = this.context.optionsToDisplay;
    this._NgFor_33_6.check_ngForOf(currVal_33_0_0,throwOnChange,false);
    this._NgFor_33_6.ngDoCheck(this,this._anchor_33,throwOnChange);
    this._vc_7.detectChangesInNestedViews(throwOnChange);
    this._vc_16.detectChangesInNestedViews(throwOnChange);
    this._vc_18.detectChangesInNestedViews(throwOnChange);
    this._vc_27.detectChangesInNestedViews(throwOnChange);
    this._vc_33.detectChangesInNestedViews(throwOnChange);
    const currVal_61:any = this.context.required;
    if (import9.checkBinding(throwOnChange,this._expr_61,currVal_61)) {
      this.renderer.setElementProperty(this._el_5,'required',currVal_61);
      this._expr_61 = currVal_61;
    }
    const currVal_62:any = this.context.disabled;
    if (import9.checkBinding(throwOnChange,this._expr_62,currVal_62)) {
      this.renderer.setElementProperty(this._el_13,'disabled',currVal_62);
      this._expr_62 = currVal_62;
    }
    const currVal_64:any = (this.context.panelVisible? 'block': 'none');
    if (import9.checkBinding(throwOnChange,this._expr_64,currVal_64)) {
      this.renderer.setElementStyle(this._el_25,'display',((this.viewUtils.sanitizer.sanitize(import32.SecurityContext.STYLE,currVal_64) == null)? (null as any): this.viewUtils.sanitizer.sanitize(import32.SecurityContext.STYLE,currVal_64).toString()));
      this._expr_64 = currVal_64;
    }
    const currVal_65:any = (this.context.scrollHeight || 'auto');
    if (import9.checkBinding(throwOnChange,this._expr_65,currVal_65)) {
      this.renderer.setElementStyle(this._el_29,'max-height',((this.viewUtils.sanitizer.sanitize(import32.SecurityContext.STYLE,currVal_65) == null)? (null as any): this.viewUtils.sanitizer.sanitize(import32.SecurityContext.STYLE,currVal_65).toString()));
      this._expr_65 = currVal_65;
    }
  }
  destroyInternal():void {
    this._vc_7.destroyNestedViews();
    this._vc_16.destroyNestedViews();
    this._vc_18.destroyNestedViews();
    this._vc_27.destroyNestedViews();
    this._vc_33.destroyNestedViews();
  }
  createEmbeddedViewInternal(nodeIndex:number):import8.AppView<any> {
    if ((nodeIndex == 7)) { return new View_Dropdown1(this.viewUtils,this,7,this._anchor_7,this._vc_7); }
    if ((nodeIndex == 16)) { return new View_Dropdown2(this.viewUtils,this,16,this._anchor_16,this._vc_16); }
    if ((nodeIndex == 18)) { return new View_Dropdown3(this.viewUtils,this,18,this._anchor_18,this._vc_18); }
    if ((nodeIndex == 27)) { return new View_Dropdown4(this.viewUtils,this,27,this._anchor_27,this._vc_27); }
    if ((nodeIndex == 33)) { return new View_Dropdown5(this.viewUtils,this,33,this._anchor_33,this._vc_33); }
    return (null as any);
  }
  handleEvent_1(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'mouseenter')) {
      const pd_sub_0:any = ((<any>this.context.onMouseenter($event)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'mouseleave')) {
      const pd_sub_1:any = ((<any>this.context.onMouseleave($event)) !== false);
      result = (pd_sub_1 && result);
    }
    if ((eventName == 'click')) {
      const pd_sub_2:any = ((<any>this.context.onMouseclick($event,this._el_13)) !== false);
      result = (pd_sub_2 && result);
    }
    return result;
  }
  handleEvent_13(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'focus')) {
      const pd_sub_0:any = ((<any>this.context.onInputFocus($event)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'blur')) {
      const pd_sub_1:any = ((<any>this.context.onInputBlur($event)) !== false);
      result = (pd_sub_1 && result);
    }
    if ((eventName == 'keydown')) {
      const pd_sub_2:any = ((<any>this.context.onKeydown($event)) !== false);
      result = (pd_sub_2 && result);
    }
    return result;
  }
}
