// primeng/components/multiselect/multiselect.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from '@angular/core/src/linker/ng_module_factory';
import * as import1 from 'primeng/components/multiselect/multiselect';
import * as import2 from '@angular/common/src/common_module';
import * as import3 from '@angular/common/src/localization';
import * as import4 from '@angular/core/src/di/injector';
import * as import5 from '@angular/core/src/i18n/tokens';
import * as import6 from '@angular/core/src/change_detection/change_detection_util';
import * as import7 from '@angular/core/src/linker/view';
import * as import8 from '@angular/core/src/linker/view_utils';
import * as import9 from '@angular/core/src/render/api';
import * as import10 from '@angular/core/src/metadata/view';
import * as import11 from 'primeng/components/dom/domhandler';
import * as import12 from '@angular/core/src/linker/view_type';
import * as import13 from '@angular/core/src/change_detection/constants';
import * as import14 from '@angular/core/src/linker/component_factory';
import * as import15 from '@angular/core/src/linker/element_ref';
import * as import16 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import17 from '@angular/forms/src/directives/control_value_accessor';
import * as import18 from '../../../@angular/common/src/directives/ng_class.ngfactory';
import * as import19 from '@angular/core/src/linker/view_container';
import * as import20 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import21 from '@angular/common/src/directives/ng_class';
import * as import22 from '@angular/core/src/security';
import * as import23 from '../../../@angular/common/src/directives/ng_style.ngfactory';
import * as import24 from '../../../@angular/common/src/directives/ng_for.ngfactory';
import * as import25 from '@angular/core/src/linker/template_ref';
import * as import26 from '@angular/common/src/directives/ng_for';
import * as import27 from '@angular/common/src/directives/ng_style';
class MultiSelectModuleInjector extends import0.NgModuleInjector<import1.MultiSelectModule> {
  _CommonModule_0:import2.CommonModule;
  _MultiSelectModule_1:import1.MultiSelectModule;
  __NgLocalization_2:import3.NgLocaleLocalization;
  constructor(parent:import4.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_2():import3.NgLocaleLocalization {
    if ((this.__NgLocalization_2 == null)) { (this.__NgLocalization_2 = new import3.NgLocaleLocalization(this.parent.get(import5.LOCALE_ID))); }
    return this.__NgLocalization_2;
  }
  createInternal():import1.MultiSelectModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._MultiSelectModule_1 = new import1.MultiSelectModule();
    return this._MultiSelectModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import1.MultiSelectModule)) { return this._MultiSelectModule_1; }
    if ((token === import3.NgLocalization)) { return this._NgLocalization_2; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const MultiSelectModuleNgFactory:import0.NgModuleFactory<import1.MultiSelectModule> = new import0.NgModuleFactory(MultiSelectModuleInjector,import1.MultiSelectModule);
export class Wrapper_MultiSelect {
  /*private*/ _eventHandler:Function;
  context:import1.MultiSelect;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  /*private*/ _expr_5:any;
  /*private*/ _expr_6:any;
  subscription0:any;
  constructor(p0:any,p1:any,p2:any,p3:any) {
    this._changed = false;
    this.context = new import1.MultiSelect(p0,p1,p2,p3);
    this._expr_0 = import6.UNINITIALIZED;
    this._expr_1 = import6.UNINITIALIZED;
    this._expr_2 = import6.UNINITIALIZED;
    this._expr_3 = import6.UNINITIALIZED;
    this._expr_4 = import6.UNINITIALIZED;
    this._expr_5 = import6.UNINITIALIZED;
    this._expr_6 = import6.UNINITIALIZED;
  }
  ngOnDetach(view:import7.AppView<any>,componentView:import7.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
    this.context.ngOnDestroy();
    (this.subscription0 && this.subscription0.unsubscribe());
  }
  check_options(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.options = currValue;
      this._expr_0 = currValue;
    }
  }
  check_scrollHeight(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.scrollHeight = currValue;
      this._expr_1 = currValue;
    }
  }
  check_defaultLabel(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_2,currValue))) {
      this._changed = true;
      this.context.defaultLabel = currValue;
      this._expr_2 = currValue;
    }
  }
  check_style(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_3,currValue))) {
      this._changed = true;
      this.context.style = currValue;
      this._expr_3 = currValue;
    }
  }
  check_styleClass(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_4,currValue))) {
      this._changed = true;
      this.context.styleClass = currValue;
      this._expr_4 = currValue;
    }
  }
  check_disabled(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_5,currValue))) {
      this._changed = true;
      this.context.disabled = currValue;
      this._expr_5 = currValue;
    }
  }
  check_overlayVisible(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_6,currValue))) {
      this._changed = true;
      this.context.overlayVisible = currValue;
      this._expr_6 = currValue;
    }
  }
  ngDoCheck(view:import7.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    if (!throwOnChange) {
      if ((view.numberOfChecks === 0)) { this.context.ngOnInit(); }
      this.context.ngDoCheck();
    }
    return changed;
  }
  checkHost(view:import7.AppView<any>,componentView:import7.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import7.AppView<any>,_eventHandler:any,emit0:boolean):void {
    this._eventHandler = _eventHandler;
    if (emit0) { (this.subscription0 = this.context.onChange.subscribe(_eventHandler.bind(view,'onChange'))); }
  }
}
var renderType_MultiSelect_Host:import9.RenderComponentType = import8.createRenderComponentType('',0,import10.ViewEncapsulation.None,([] as any[]),{});
class View_MultiSelect_Host0 extends import7.AppView<any> {
  _el_0:any;
  compView_0:import7.AppView<import1.MultiSelect>;
  _DomHandler_0_3:import11.DomHandler;
  _MultiSelect_0_4:Wrapper_MultiSelect;
  __NG_VALUE_ACCESSOR_0_5:any[];
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_MultiSelect_Host0,renderType_MultiSelect_Host,import12.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import13.ChangeDetectorStatus.CheckAlways);
  }
  get _NG_VALUE_ACCESSOR_0_5():any[] {
    if ((this.__NG_VALUE_ACCESSOR_0_5 == null)) { (this.__NG_VALUE_ACCESSOR_0_5 = [this._MultiSelect_0_4.context]); }
    return this.__NG_VALUE_ACCESSOR_0_5;
  }
  createInternal(rootSelector:string):import14.ComponentRef<any> {
    this._el_0 = import8.selectOrCreateRenderHostElement(this.renderer,'p-multiSelect',import8.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this.compView_0 = new View_MultiSelect0(this.viewUtils,this,0,this._el_0);
    this._DomHandler_0_3 = new import11.DomHandler();
    this._MultiSelect_0_4 = new Wrapper_MultiSelect(new import15.ElementRef(this._el_0),this._DomHandler_0_3,this.renderer,this.injectorGet(import16.IterableDiffers,this.parentIndex));
    this.compView_0.create(this._MultiSelect_0_4.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import14.ComponentRef_<any>(0,this,this._el_0,this._MultiSelect_0_4.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import11.DomHandler) && (0 === requestNodeIndex))) { return this._DomHandler_0_3; }
    if (((token === import1.MultiSelect) && (0 === requestNodeIndex))) { return this._MultiSelect_0_4.context; }
    if (((token === import17.NG_VALUE_ACCESSOR) && (0 === requestNodeIndex))) { return this._NG_VALUE_ACCESSOR_0_5; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._MultiSelect_0_4.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
    if (!throwOnChange) {
      if ((this.numberOfChecks === 0)) { this._MultiSelect_0_4.context.ngAfterViewInit(); }
      this._MultiSelect_0_4.context.ngAfterViewChecked();
    }
  }
  destroyInternal():void {
    this.compView_0.destroy();
    this._MultiSelect_0_4.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
export const MultiSelectNgFactory:import14.ComponentFactory<import1.MultiSelect> = new import14.ComponentFactory<import1.MultiSelect>('p-multiSelect',View_MultiSelect_Host0,import1.MultiSelect);
const styles_MultiSelect:any[] = ([] as any[]);
class View_MultiSelect1 extends import7.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import18.Wrapper_NgClass;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _el_4:any;
  _text_5:any;
  _el_6:any;
  _text_7:any;
  _text_8:any;
  _el_9:any;
  _NgClass_9_3:import18.Wrapper_NgClass;
  _text_10:any;
  _el_11:any;
  _NgClass_11_3:import18.Wrapper_NgClass;
  _text_12:any;
  _text_13:any;
  _text_14:any;
  _el_15:any;
  _text_16:any;
  _text_17:any;
  /*private*/ _expr_21:any;
  _map_22:any;
  /*private*/ _expr_23:any;
  _map_24:any;
  _map_25:any;
  /*private*/ _expr_26:any;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import19.ViewContainer) {
    super(View_MultiSelect1,renderType_MultiSelect,import12.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import13.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_21 = import6.UNINITIALIZED;
    this._map_22 = import8.pureProxy2((p0:any,p1:any):{[key: string]:any} => {
      return {
        'ui-state-highlight': p0,
        'ui-state-hover': p1
      }
      ;
    });
    this._expr_23 = import6.UNINITIALIZED;
    this._map_24 = import8.pureProxy1((p0:any):{[key: string]:any} => {
      return {'ui-state-active': p0};
    });
    this._map_25 = import8.pureProxy1((p0:any):{[key: string]:any} => {
      return {'fa fa-check': p0};
    });
    this._expr_26 = import6.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import14.ComponentRef<any> {
    this._el_0 = import8.createRenderElement(this.renderer,(null as any),'li',new import8.InlineArray2(2,'class','ui-multiselect-item ui-corner-all'),(null as any));
    this._NgClass_0_3 = new import18.Wrapper_NgClass(this.parentView.parentView.injectorGet(import16.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import20.KeyValueDiffers,this.parentView.parentIndex),new import15.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n                            ',(null as any));
    this._el_2 = import8.createRenderElement(this.renderer,this._el_0,'div',new import8.InlineArray2(2,'class','ui-chkbox ui-widget'),(null as any));
    this._text_3 = this.renderer.createText(this._el_2,'\n                                ',(null as any));
    this._el_4 = import8.createRenderElement(this.renderer,this._el_2,'div',new import8.InlineArray2(2,'class','ui-helper-hidden-accessible'),(null as any));
    this._text_5 = this.renderer.createText(this._el_4,'\n                                    ',(null as any));
    this._el_6 = import8.createRenderElement(this.renderer,this._el_4,'input',new import8.InlineArray4(4,'readonly','readonly','type','checkbox'),(null as any));
    this._text_7 = this.renderer.createText(this._el_4,'\n                                ',(null as any));
    this._text_8 = this.renderer.createText(this._el_2,'\n                                ',(null as any));
    this._el_9 = import8.createRenderElement(this.renderer,this._el_2,'div',new import8.InlineArray2(2,'class','ui-chkbox-box ui-widget ui-corner-all ui-state-default'),(null as any));
    this._NgClass_9_3 = new import18.Wrapper_NgClass(this.parentView.parentView.injectorGet(import16.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import20.KeyValueDiffers,this.parentView.parentIndex),new import15.ElementRef(this._el_9),this.renderer);
    this._text_10 = this.renderer.createText(this._el_9,'\n                                    ',(null as any));
    this._el_11 = import8.createRenderElement(this.renderer,this._el_9,'span',new import8.InlineArray2(2,'class','ui-chkbox-icon ui-c'),(null as any));
    this._NgClass_11_3 = new import18.Wrapper_NgClass(this.parentView.parentView.injectorGet(import16.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import20.KeyValueDiffers,this.parentView.parentIndex),new import15.ElementRef(this._el_11),this.renderer);
    this._text_12 = this.renderer.createText(this._el_9,'\n                                ',(null as any));
    this._text_13 = this.renderer.createText(this._el_2,'\n                            ',(null as any));
    this._text_14 = this.renderer.createText(this._el_0,'\n                            ',(null as any));
    this._el_15 = import8.createRenderElement(this.renderer,this._el_0,'label',import8.EMPTY_INLINE_ARRAY,(null as any));
    this._text_16 = this.renderer.createText(this._el_15,'',(null as any));
    this._text_17 = this.renderer.createText(this._el_0,'\n                        ',(null as any));
    var disposable_0:Function = import8.subscribeToRenderElement(this,this._el_0,new import8.InlineArray8(6,'click',(null as any),'mouseenter',(null as any),'mouseleave',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._el_4,
      this._text_5,
      this._el_6,
      this._text_7,
      this._text_8,
      this._el_9,
      this._text_10,
      this._el_11,
      this._text_12,
      this._text_13,
      this._text_14,
      this._el_15,
      this._text_16,
      this._text_17
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import21.NgClass) && (11 === requestNodeIndex))) { return this._NgClass_11_3.context; }
    if (((token === import21.NgClass) && ((9 <= requestNodeIndex) && (requestNodeIndex <= 12)))) { return this._NgClass_9_3.context; }
    if (((token === import21.NgClass) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 17)))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = 'ui-multiselect-item ui-corner-all';
    this._NgClass_0_3.check_klass(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = this._map_22(this.parentView.context.isSelected(this.context.$implicit.value),(this.parentView.context.hoveredItem == this._el_0));
    this._NgClass_0_3.check_ngClass(currVal_0_0_1,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_9_0_0:any = 'ui-chkbox-box ui-widget ui-corner-all ui-state-default';
    this._NgClass_9_3.check_klass(currVal_9_0_0,throwOnChange,false);
    const currVal_9_0_1:any = this._map_24(this.parentView.context.isSelected(this.context.$implicit.value));
    this._NgClass_9_3.check_ngClass(currVal_9_0_1,throwOnChange,false);
    this._NgClass_9_3.ngDoCheck(this,this._el_9,throwOnChange);
    const currVal_11_0_0:any = 'ui-chkbox-icon ui-c';
    this._NgClass_11_3.check_klass(currVal_11_0_0,throwOnChange,false);
    const currVal_11_0_1:any = this._map_25(this.parentView.context.isSelected(this.context.$implicit.value));
    this._NgClass_11_3.check_ngClass(currVal_11_0_1,throwOnChange,false);
    this._NgClass_11_3.ngDoCheck(this,this._el_11,throwOnChange);
    const currVal_21:any = (this.parentView.context.isItemVisible(this.context.$implicit)? 'block': 'none');
    if (import8.checkBinding(throwOnChange,this._expr_21,currVal_21)) {
      this.renderer.setElementStyle(this._el_0,'display',((this.viewUtils.sanitizer.sanitize(import22.SecurityContext.STYLE,currVal_21) == null)? (null as any): this.viewUtils.sanitizer.sanitize(import22.SecurityContext.STYLE,currVal_21).toString()));
      this._expr_21 = currVal_21;
    }
    const currVal_23:any = this.parentView.context.isSelected(this.context.$implicit.value);
    if (import8.checkBinding(throwOnChange,this._expr_23,currVal_23)) {
      this.renderer.setElementProperty(this._el_6,'checked',currVal_23);
      this._expr_23 = currVal_23;
    }
    const currVal_26:any = import8.inlineInterpolate(1,'',this.context.$implicit.label,'');
    if (import8.checkBinding(throwOnChange,this._expr_26,currVal_26)) {
      this.renderer.setText(this._text_16,currVal_26);
      this._expr_26 = currVal_26;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.context.onItemClick($event,this.context.$implicit.value)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'mouseenter')) {
      const pd_sub_1:any = ((<any>(this.parentView.context.hoveredItem = this._el_0)) !== false);
      result = (pd_sub_1 && result);
    }
    if ((eventName == 'mouseleave')) {
      const pd_sub_2:any = ((<any>(this.parentView.context.hoveredItem = (null as any))) !== false);
      result = (pd_sub_2 && result);
    }
    return result;
  }
}
var renderType_MultiSelect:import9.RenderComponentType = import8.createRenderComponentType('',0,import10.ViewEncapsulation.None,styles_MultiSelect,{});
export class View_MultiSelect0 extends import7.AppView<import1.MultiSelect> {
  _text_0:any;
  _el_1:any;
  _NgClass_1_3:import18.Wrapper_NgClass;
  _NgStyle_1_4:import23.Wrapper_NgStyle;
  _text_2:any;
  _el_3:any;
  _text_4:any;
  _el_5:any;
  _text_6:any;
  _text_7:any;
  _el_8:any;
  _text_9:any;
  _el_10:any;
  _NgClass_10_3:import18.Wrapper_NgClass;
  _text_11:any;
  _text_12:any;
  _text_13:any;
  _el_14:any;
  _NgClass_14_3:import18.Wrapper_NgClass;
  _text_15:any;
  _el_16:any;
  _text_17:any;
  _text_18:any;
  _el_19:any;
  _text_20:any;
  _el_21:any;
  _text_22:any;
  _el_23:any;
  _text_24:any;
  _el_25:any;
  _text_26:any;
  _el_27:any;
  _text_28:any;
  _text_29:any;
  _el_30:any;
  _NgClass_30_3:import18.Wrapper_NgClass;
  _text_31:any;
  _el_32:any;
  _NgClass_32_3:import18.Wrapper_NgClass;
  _text_33:any;
  _text_34:any;
  _text_35:any;
  _el_36:any;
  _text_37:any;
  _el_38:any;
  _text_39:any;
  _el_40:any;
  _text_41:any;
  _text_42:any;
  _el_43:any;
  _text_44:any;
  _el_45:any;
  _text_46:any;
  _text_47:any;
  _text_48:any;
  _el_49:any;
  _text_50:any;
  _el_51:any;
  _text_52:any;
  _anchor_53:any;
  /*private*/ _vc_53:import19.ViewContainer;
  _TemplateRef_53_5:any;
  _NgFor_53_6:import24.Wrapper_NgFor;
  _text_54:any;
  _text_55:any;
  _text_56:any;
  _text_57:any;
  _text_58:any;
  _map_68:any;
  /*private*/ _expr_69:any;
  /*private*/ _expr_70:any;
  _map_71:any;
  /*private*/ _expr_72:any;
  _map_73:any;
  /*private*/ _expr_74:any;
  /*private*/ _expr_75:any;
  _map_76:any;
  _map_77:any;
  /*private*/ _expr_78:any;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_MultiSelect0,renderType_MultiSelect,import12.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import13.ChangeDetectorStatus.CheckAlways);
    this._map_68 = import8.pureProxy3((p0:any,p1:any,p2:any):{[key: string]:any} => {
      return {
        'ui-multiselect ui-widget ui-state-default ui-corner-all': p0,
        'ui-state-focus': p1,
        'ui-state-disabled': p2
      }
      ;
    });
    this._expr_69 = import6.UNINITIALIZED;
    this._expr_70 = import6.UNINITIALIZED;
    this._map_71 = import8.pureProxy3((p0:any,p1:any,p2:any):{[key: string]:any} => {
      return {
        'ui-multiselect-label ui-corner-all': p0,
        'ui-state-hover': p1,
        'ui-state-focus': p2
      }
      ;
    });
    this._expr_72 = import6.UNINITIALIZED;
    this._map_73 = import8.pureProxy3((p0:any,p1:any,p2:any):{[key: string]:any} => {
      return {
        'ui-multiselect-trigger ui-state-default ui-corner-right': p0,
        'ui-state-hover': p1,
        'ui-state-focus': p2
      }
      ;
    });
    this._expr_74 = import6.UNINITIALIZED;
    this._expr_75 = import6.UNINITIALIZED;
    this._map_76 = import8.pureProxy1((p0:any):{[key: string]:any} => {
      return {'ui-state-hover': p0};
    });
    this._map_77 = import8.pureProxy1((p0:any):{[key: string]:any} => {
      return {'fa fa-check': p0};
    });
    this._expr_78 = import6.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import14.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n        ',(null as any));
    this._el_1 = import8.createRenderElement(this.renderer,parentRenderNode,'div',import8.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_1_3 = new import18.Wrapper_NgClass(this.parentView.injectorGet(import16.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import20.KeyValueDiffers,this.parentIndex),new import15.ElementRef(this._el_1),this.renderer);
    this._NgStyle_1_4 = new import23.Wrapper_NgStyle(this.parentView.injectorGet(import20.KeyValueDiffers,this.parentIndex),new import15.ElementRef(this._el_1),this.renderer);
    this._text_2 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_3 = import8.createRenderElement(this.renderer,this._el_1,'div',new import8.InlineArray2(2,'class','ui-helper-hidden-accessible'),(null as any));
    this._text_4 = this.renderer.createText(this._el_3,'\n                ',(null as any));
    this._el_5 = import8.createRenderElement(this.renderer,this._el_3,'input',new import8.InlineArray4(4,'readonly','readonly','type','text'),(null as any));
    this._text_6 = this.renderer.createText(this._el_3,'\n            ',(null as any));
    this._text_7 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_8 = import8.createRenderElement(this.renderer,this._el_1,'div',new import8.InlineArray2(2,'class','ui-multiselect-label-container'),(null as any));
    this._text_9 = this.renderer.createText(this._el_8,'\n                ',(null as any));
    this._el_10 = import8.createRenderElement(this.renderer,this._el_8,'label',import8.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_10_3 = new import18.Wrapper_NgClass(this.parentView.injectorGet(import16.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import20.KeyValueDiffers,this.parentIndex),new import15.ElementRef(this._el_10),this.renderer);
    this._text_11 = this.renderer.createText(this._el_10,'',(null as any));
    this._text_12 = this.renderer.createText(this._el_8,'\n            ',(null as any));
    this._text_13 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_14 = import8.createRenderElement(this.renderer,this._el_1,'div',import8.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_14_3 = new import18.Wrapper_NgClass(this.parentView.injectorGet(import16.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import20.KeyValueDiffers,this.parentIndex),new import15.ElementRef(this._el_14),this.renderer);
    this._text_15 = this.renderer.createText(this._el_14,'\n                ',(null as any));
    this._el_16 = import8.createRenderElement(this.renderer,this._el_14,'span',new import8.InlineArray2(2,'class','fa fa-fw fa-caret-down ui-c'),(null as any));
    this._text_17 = this.renderer.createText(this._el_14,'\n            ',(null as any));
    this._text_18 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_19 = import8.createRenderElement(this.renderer,this._el_1,'div',new import8.InlineArray2(2,'class','ui-multiselect-panel ui-widget ui-widget-content ui-corner-all ui-shadow'),(null as any));
    this._text_20 = this.renderer.createText(this._el_19,'\n                ',(null as any));
    this._el_21 = import8.createRenderElement(this.renderer,this._el_19,'div',new import8.InlineArray2(2,'class','ui-widget-header ui-corner-all ui-multiselect-header ui-helper-clearfix'),(null as any));
    this._text_22 = this.renderer.createText(this._el_21,'\n                    ',(null as any));
    this._el_23 = import8.createRenderElement(this.renderer,this._el_21,'div',new import8.InlineArray2(2,'class','ui-chkbox ui-widget'),(null as any));
    this._text_24 = this.renderer.createText(this._el_23,'\n                        ',(null as any));
    this._el_25 = import8.createRenderElement(this.renderer,this._el_23,'div',new import8.InlineArray2(2,'class','ui-helper-hidden-accessible'),(null as any));
    this._text_26 = this.renderer.createText(this._el_25,'\n                            ',(null as any));
    this._el_27 = import8.createRenderElement(this.renderer,this._el_25,'input',new import8.InlineArray4(4,'readonly','readonly','type','checkbox'),(null as any));
    this._text_28 = this.renderer.createText(this._el_25,'\n                        ',(null as any));
    this._text_29 = this.renderer.createText(this._el_23,'\n                        ',(null as any));
    this._el_30 = import8.createRenderElement(this.renderer,this._el_23,'div',new import8.InlineArray2(2,'class','ui-chkbox-box ui-widget ui-corner-all ui-state-default'),(null as any));
    this._NgClass_30_3 = new import18.Wrapper_NgClass(this.parentView.injectorGet(import16.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import20.KeyValueDiffers,this.parentIndex),new import15.ElementRef(this._el_30),this.renderer);
    this._text_31 = this.renderer.createText(this._el_30,'\n                            ',(null as any));
    this._el_32 = import8.createRenderElement(this.renderer,this._el_30,'span',new import8.InlineArray2(2,'class','ui-chkbox-icon ui-c'),(null as any));
    this._NgClass_32_3 = new import18.Wrapper_NgClass(this.parentView.injectorGet(import16.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import20.KeyValueDiffers,this.parentIndex),new import15.ElementRef(this._el_32),this.renderer);
    this._text_33 = this.renderer.createText(this._el_30,'\n                        ',(null as any));
    this._text_34 = this.renderer.createText(this._el_23,'\n                    ',(null as any));
    this._text_35 = this.renderer.createText(this._el_21,'\n                    ',(null as any));
    this._el_36 = import8.createRenderElement(this.renderer,this._el_21,'div',new import8.InlineArray2(2,'class','ui-multiselect-filter-container'),(null as any));
    this._text_37 = this.renderer.createText(this._el_36,'\n                        ',(null as any));
    this._el_38 = import8.createRenderElement(this.renderer,this._el_36,'input',new import8.InlineArray8(6,'class','ui-inputtext ui-widget ui-state-default ui-corner-all','role','textbox','type','text'),(null as any));
    this._text_39 = this.renderer.createText(this._el_36,'\n                        ',(null as any));
    this._el_40 = import8.createRenderElement(this.renderer,this._el_36,'span',new import8.InlineArray2(2,'class','fa fa-fw fa-search'),(null as any));
    this._text_41 = this.renderer.createText(this._el_36,'\n                    ',(null as any));
    this._text_42 = this.renderer.createText(this._el_21,'\n                    ',(null as any));
    this._el_43 = import8.createRenderElement(this.renderer,this._el_21,'a',new import8.InlineArray4(4,'class','ui-multiselect-close ui-corner-all','href','#'),(null as any));
    this._text_44 = this.renderer.createText(this._el_43,'\n                        ',(null as any));
    this._el_45 = import8.createRenderElement(this.renderer,this._el_43,'span',new import8.InlineArray2(2,'class','fa fa-close'),(null as any));
    this._text_46 = this.renderer.createText(this._el_43,'\n                    ',(null as any));
    this._text_47 = this.renderer.createText(this._el_21,'\n                ',(null as any));
    this._text_48 = this.renderer.createText(this._el_19,'\n                ',(null as any));
    this._el_49 = import8.createRenderElement(this.renderer,this._el_19,'div',new import8.InlineArray2(2,'class','ui-multiselect-items-wrapper'),(null as any));
    this._text_50 = this.renderer.createText(this._el_49,'\n                    ',(null as any));
    this._el_51 = import8.createRenderElement(this.renderer,this._el_49,'ul',new import8.InlineArray2(2,'class','ui-multiselect-items ui-multiselect-list ui-widget-content ui-widget ui-corner-all ui-helper-reset'),(null as any));
    this._text_52 = this.renderer.createText(this._el_51,'\n                        ',(null as any));
    this._anchor_53 = this.renderer.createTemplateAnchor(this._el_51,(null as any));
    this._vc_53 = new import19.ViewContainer(53,51,this,this._anchor_53);
    this._TemplateRef_53_5 = new import25.TemplateRef_(this,53,this._anchor_53);
    this._NgFor_53_6 = new import24.Wrapper_NgFor(this._vc_53.vcRef,this._TemplateRef_53_5,this.parentView.injectorGet(import16.IterableDiffers,this.parentIndex),this.ref);
    this._text_54 = this.renderer.createText(this._el_51,'\n                    ',(null as any));
    this._text_55 = this.renderer.createText(this._el_49,'\n                ',(null as any));
    this._text_56 = this.renderer.createText(this._el_19,'\n            ',(null as any));
    this._text_57 = this.renderer.createText(this._el_1,'\n        ',(null as any));
    this._text_58 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    var disposable_0:Function = import8.subscribeToRenderElement(this,this._el_1,new import8.InlineArray8(6,'mouseenter',(null as any),'mouseleave',(null as any),'click',(null as any)),this.eventHandler(this.handleEvent_1));
    var disposable_1:Function = import8.subscribeToRenderElement(this,this._el_5,new import8.InlineArray4(4,'focus',(null as any),'blur',(null as any)),this.eventHandler(this.handleEvent_5));
    var disposable_2:Function = import8.subscribeToRenderElement(this,this._el_19,new import8.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_19));
    var disposable_3:Function = import8.subscribeToRenderElement(this,this._el_30,new import8.InlineArray8(6,'mouseenter',(null as any),'mouseleave',(null as any),'click',(null as any)),this.eventHandler(this.handleEvent_30));
    var disposable_4:Function = import8.subscribeToRenderElement(this,this._el_38,new import8.InlineArray2(2,'input',(null as any)),this.eventHandler(this.handleEvent_38));
    var disposable_5:Function = import8.subscribeToRenderElement(this,this._el_43,new import8.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_43));
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2,
      this._el_3,
      this._text_4,
      this._el_5,
      this._text_6,
      this._text_7,
      this._el_8,
      this._text_9,
      this._el_10,
      this._text_11,
      this._text_12,
      this._text_13,
      this._el_14,
      this._text_15,
      this._el_16,
      this._text_17,
      this._text_18,
      this._el_19,
      this._text_20,
      this._el_21,
      this._text_22,
      this._el_23,
      this._text_24,
      this._el_25,
      this._text_26,
      this._el_27,
      this._text_28,
      this._text_29,
      this._el_30,
      this._text_31,
      this._el_32,
      this._text_33,
      this._text_34,
      this._text_35,
      this._el_36,
      this._text_37,
      this._el_38,
      this._text_39,
      this._el_40,
      this._text_41,
      this._text_42,
      this._el_43,
      this._text_44,
      this._el_45,
      this._text_46,
      this._text_47,
      this._text_48,
      this._el_49,
      this._text_50,
      this._el_51,
      this._text_52,
      this._anchor_53,
      this._text_54,
      this._text_55,
      this._text_56,
      this._text_57,
      this._text_58
    ]
    ),[
      disposable_0,
      disposable_1,
      disposable_2,
      disposable_3,
      disposable_4,
      disposable_5
    ]
    );
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import21.NgClass) && ((10 <= requestNodeIndex) && (requestNodeIndex <= 11)))) { return this._NgClass_10_3.context; }
    if (((token === import21.NgClass) && ((14 <= requestNodeIndex) && (requestNodeIndex <= 17)))) { return this._NgClass_14_3.context; }
    if (((token === import21.NgClass) && (32 === requestNodeIndex))) { return this._NgClass_32_3.context; }
    if (((token === import21.NgClass) && ((30 <= requestNodeIndex) && (requestNodeIndex <= 33)))) { return this._NgClass_30_3.context; }
    if (((token === import25.TemplateRef) && (53 === requestNodeIndex))) { return this._TemplateRef_53_5; }
    if (((token === import26.NgFor) && (53 === requestNodeIndex))) { return this._NgFor_53_6.context; }
    if (((token === import21.NgClass) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 57)))) { return this._NgClass_1_3.context; }
    if (((token === import27.NgStyle) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 57)))) { return this._NgStyle_1_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1_0_0:any = this.context.styleClass;
    this._NgClass_1_3.check_klass(currVal_1_0_0,throwOnChange,false);
    const currVal_1_0_1:any = this._map_68(true,this.context.focus,this.context.disabled);
    this._NgClass_1_3.check_ngClass(currVal_1_0_1,throwOnChange,false);
    this._NgClass_1_3.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_1_1_0:any = this.context.style;
    this._NgStyle_1_4.check_ngStyle(currVal_1_1_0,throwOnChange,false);
    this._NgStyle_1_4.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_10_0_0:any = this._map_71(true,this.context.hover,this.context.focus);
    this._NgClass_10_3.check_ngClass(currVal_10_0_0,throwOnChange,false);
    this._NgClass_10_3.ngDoCheck(this,this._el_10,throwOnChange);
    const currVal_14_0_0:any = this._map_73(true,this.context.hover,this.context.focus);
    this._NgClass_14_3.check_ngClass(currVal_14_0_0,throwOnChange,false);
    this._NgClass_14_3.ngDoCheck(this,this._el_14,throwOnChange);
    const currVal_30_0_0:any = 'ui-chkbox-box ui-widget ui-corner-all ui-state-default';
    this._NgClass_30_3.check_klass(currVal_30_0_0,throwOnChange,false);
    const currVal_30_0_1:any = this._map_76(this.context.hoverToggleAll);
    this._NgClass_30_3.check_ngClass(currVal_30_0_1,throwOnChange,false);
    this._NgClass_30_3.ngDoCheck(this,this._el_30,throwOnChange);
    const currVal_32_0_0:any = 'ui-chkbox-icon ui-c';
    this._NgClass_32_3.check_klass(currVal_32_0_0,throwOnChange,false);
    const currVal_32_0_1:any = this._map_77(this.context.isAllChecked());
    this._NgClass_32_3.check_ngClass(currVal_32_0_1,throwOnChange,false);
    this._NgClass_32_3.ngDoCheck(this,this._el_32,throwOnChange);
    const currVal_53_0_0:any = this.context.options;
    this._NgFor_53_6.check_ngForOf(currVal_53_0_0,throwOnChange,false);
    this._NgFor_53_6.ngDoCheck(this,this._anchor_53,throwOnChange);
    this._vc_53.detectChangesInNestedViews(throwOnChange);
    const currVal_69:any = this.context.disabled;
    if (import8.checkBinding(throwOnChange,this._expr_69,currVal_69)) {
      this.renderer.setElementProperty(this._el_5,'disabled',currVal_69);
      this._expr_69 = currVal_69;
    }
    const currVal_70:any = this.context.valuesAsString;
    if (import8.checkBinding(throwOnChange,this._expr_70,currVal_70)) {
      this.renderer.setElementProperty(this._el_8,'title',currVal_70);
      this._expr_70 = currVal_70;
    }
    const currVal_72:any = import8.inlineInterpolate(1,'',this.context.valuesAsString,'');
    if (import8.checkBinding(throwOnChange,this._expr_72,currVal_72)) {
      this.renderer.setText(this._text_11,currVal_72);
      this._expr_72 = currVal_72;
    }
    const currVal_74:any = (this.context.overlayVisible? 'block': 'none');
    if (import8.checkBinding(throwOnChange,this._expr_74,currVal_74)) {
      this.renderer.setElementStyle(this._el_19,'display',((this.viewUtils.sanitizer.sanitize(import22.SecurityContext.STYLE,currVal_74) == null)? (null as any): this.viewUtils.sanitizer.sanitize(import22.SecurityContext.STYLE,currVal_74).toString()));
      this._expr_74 = currVal_74;
    }
    const currVal_75:any = this.context.isAllChecked();
    if (import8.checkBinding(throwOnChange,this._expr_75,currVal_75)) {
      this.renderer.setElementProperty(this._el_27,'checked',currVal_75);
      this._expr_75 = currVal_75;
    }
    const currVal_78:any = (this.context.scrollHeight || 'auto');
    if (import8.checkBinding(throwOnChange,this._expr_78,currVal_78)) {
      this.renderer.setElementStyle(this._el_51,'max-height',((this.viewUtils.sanitizer.sanitize(import22.SecurityContext.STYLE,currVal_78) == null)? (null as any): this.viewUtils.sanitizer.sanitize(import22.SecurityContext.STYLE,currVal_78).toString()));
      this._expr_78 = currVal_78;
    }
  }
  destroyInternal():void {
    this._vc_53.destroyNestedViews();
  }
  createEmbeddedViewInternal(nodeIndex:number):import7.AppView<any> {
    if ((nodeIndex == 53)) { return new View_MultiSelect1(this.viewUtils,this,53,this._anchor_53,this._vc_53); }
    return (null as any);
  }
  handleEvent_1(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'mouseenter')) {
      const pd_sub_0:any = ((<any>this.context.onMouseenter($event)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'mouseleave')) {
      const pd_sub_1:any = ((<any>this.context.onMouseleave($event)) !== false);
      result = (pd_sub_1 && result);
    }
    if ((eventName == 'click')) {
      const pd_sub_2:any = ((<any>this.context.onMouseclick($event,this._el_5)) !== false);
      result = (pd_sub_2 && result);
    }
    return result;
  }
  handleEvent_5(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'focus')) {
      const pd_sub_0:any = ((<any>this.context.onFocus($event)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'blur')) {
      const pd_sub_1:any = ((<any>this.context.onBlur($event)) !== false);
      result = (pd_sub_1 && result);
    }
    return result;
  }
  handleEvent_19(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>(this.context.panelClick = true)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
  handleEvent_30(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'mouseenter')) {
      const pd_sub_0:any = ((<any>(this.context.hoverToggleAll = true)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'mouseleave')) {
      const pd_sub_1:any = ((<any>(this.context.hoverToggleAll = false)) !== false);
      result = (pd_sub_1 && result);
    }
    if ((eventName == 'click')) {
      const pd_sub_2:any = ((<any>this.context.toggleAll($event,this._el_27)) !== false);
      result = (pd_sub_2 && result);
    }
    return result;
  }
  handleEvent_38(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'input')) {
      const pd_sub_0:any = ((<any>this.context.onFilter($event)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
  handleEvent_43(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.context.close($event)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
