// primeng/components/panelmenu/panelmenu.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from '@angular/core/src/linker/ng_module_factory';
import * as import1 from 'primeng/components/panelmenu/panelmenu';
import * as import2 from '@angular/common/src/common_module';
import * as import3 from '@angular/common/src/localization';
import * as import4 from '@angular/core/src/di/injector';
import * as import5 from '@angular/core/src/i18n/tokens';
import * as import6 from '@angular/core/src/change_detection/change_detection_util';
import * as import7 from '@angular/core/src/linker/view';
import * as import8 from '@angular/core/src/linker/view_utils';
import * as import9 from '@angular/core/src/render/api';
import * as import10 from '@angular/core/src/metadata/view';
import * as import11 from '@angular/core/src/linker/view_type';
import * as import12 from '@angular/core/src/change_detection/constants';
import * as import13 from '@angular/core/src/linker/component_factory';
import * as import14 from '@angular/router/src/router';
import * as import15 from '@angular/core/src/animation/animation_transition';
import * as import16 from '@angular/core/src/animation/animation_sequence_player';
import * as import17 from '@angular/core/src/animation/animation_styles';
import * as import18 from '@angular/core/src/animation/animation_style_util';
import * as import19 from '@angular/core/src/animation/animation_keyframe';
import * as import20 from '@angular/core/src/animation/animation_player';
import * as import21 from '../../../@angular/common/src/directives/ng_class.ngfactory';
import * as import22 from '@angular/core/src/linker/view_container';
import * as import23 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import24 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import25 from '@angular/core/src/linker/element_ref';
import * as import26 from '@angular/common/src/directives/ng_class';
import * as import27 from '../../../@angular/common/src/directives/ng_if.ngfactory';
import * as import28 from '@angular/core/src/linker/template_ref';
import * as import29 from '@angular/common/src/directives/ng_if';
import * as import30 from '@angular/core/src/security';
import * as import31 from '../../../@angular/common/src/directives/ng_style.ngfactory';
import * as import32 from '../../../@angular/common/src/directives/ng_for.ngfactory';
import * as import33 from '@angular/common/src/directives/ng_for';
import * as import34 from '@angular/common/src/directives/ng_style';
class PanelMenuModuleInjector extends import0.NgModuleInjector<import1.PanelMenuModule> {
  _CommonModule_0:import2.CommonModule;
  _PanelMenuModule_1:import1.PanelMenuModule;
  __NgLocalization_2:import3.NgLocaleLocalization;
  constructor(parent:import4.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_2():import3.NgLocaleLocalization {
    if ((this.__NgLocalization_2 == null)) { (this.__NgLocalization_2 = new import3.NgLocaleLocalization(this.parent.get(import5.LOCALE_ID))); }
    return this.__NgLocalization_2;
  }
  createInternal():import1.PanelMenuModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._PanelMenuModule_1 = new import1.PanelMenuModule();
    return this._PanelMenuModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import1.PanelMenuModule)) { return this._PanelMenuModule_1; }
    if ((token === import3.NgLocalization)) { return this._NgLocalization_2; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const PanelMenuModuleNgFactory:import0.NgModuleFactory<import1.PanelMenuModule> = new import0.NgModuleFactory(PanelMenuModuleInjector,import1.PanelMenuModule);
export class Wrapper_PanelMenu {
  /*private*/ _eventHandler:Function;
  context:import1.PanelMenu;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  constructor(p0:any) {
    this._changed = false;
    this.context = new import1.PanelMenu(p0);
    this._expr_0 = import6.UNINITIALIZED;
    this._expr_1 = import6.UNINITIALIZED;
    this._expr_2 = import6.UNINITIALIZED;
  }
  ngOnDetach(view:import7.AppView<any>,componentView:import7.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
    this.context.ngOnDestroy();
  }
  check_model(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.model = currValue;
      this._expr_0 = currValue;
    }
  }
  check_style(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.style = currValue;
      this._expr_1 = currValue;
    }
  }
  check_styleClass(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_2,currValue))) {
      this._changed = true;
      this.context.styleClass = currValue;
      this._expr_2 = currValue;
    }
  }
  ngDoCheck(view:import7.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    return changed;
  }
  checkHost(view:import7.AppView<any>,componentView:import7.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import7.AppView<any>,_eventHandler:any):void {
    this._eventHandler = _eventHandler;
  }
}
export class Wrapper_PanelMenuSub {
  /*private*/ _eventHandler:Function;
  context:import1.PanelMenuSub;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  constructor(p0:any) {
    this._changed = false;
    this.context = new import1.PanelMenuSub(p0);
    this._expr_0 = import6.UNINITIALIZED;
    this._expr_1 = import6.UNINITIALIZED;
  }
  ngOnDetach(view:import7.AppView<any>,componentView:import7.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
  }
  check_item(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.item = currValue;
      this._expr_0 = currValue;
    }
  }
  check_expanded(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.expanded = currValue;
      this._expr_1 = currValue;
    }
  }
  ngDoCheck(view:import7.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    return changed;
  }
  checkHost(view:import7.AppView<any>,componentView:import7.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import7.AppView<any>,_eventHandler:any):void {
    this._eventHandler = _eventHandler;
  }
}
var renderType_PanelMenu_Host:import9.RenderComponentType = import8.createRenderComponentType('',0,import10.ViewEncapsulation.None,([] as any[]),{});
class View_PanelMenu_Host0 extends import7.AppView<any> {
  _el_0:any;
  compView_0:import7.AppView<import1.PanelMenu>;
  _PanelMenu_0_3:Wrapper_PanelMenu;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_PanelMenu_Host0,renderType_PanelMenu_Host,import11.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import12.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import13.ComponentRef<any> {
    this._el_0 = import8.selectOrCreateRenderHostElement(this.renderer,'p-panelMenu',import8.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this.compView_0 = new View_PanelMenu0(this.viewUtils,this,0,this._el_0);
    this._PanelMenu_0_3 = new Wrapper_PanelMenu(this.injectorGet(import14.Router,this.parentIndex));
    this.compView_0.create(this._PanelMenu_0_3.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import13.ComponentRef_<any>(0,this,this._el_0,this._PanelMenu_0_3.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import1.PanelMenu) && (0 === requestNodeIndex))) { return this._PanelMenu_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._PanelMenu_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
    this._PanelMenu_0_3.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
export const PanelMenuNgFactory:import13.ComponentFactory<import1.PanelMenu> = new import13.ComponentFactory<import1.PanelMenu>('p-panelMenu',View_PanelMenu_Host0,import1.PanelMenu);
const styles_PanelMenu:any[] = ([] as any[]);
var PanelMenu_rootItem_states:any = {
  'hidden': {'height': '0px'},
  'visible': {'height': '*'},
  '*': {}
}
;
function PanelMenu_rootItem_factory(view:import7.AppView<any>,element:any,currentState:any,nextState:any):import15.AnimationTransition {
  var previousPlayers:any = view.animationContext.getAnimationPlayers(element,((nextState == 'void')? (null as any): 'rootItem'));
  var collectedStyles:any = {};
  var player:any = (null as any);
  var totalTime:any = 0;
  var defaultStateStyles:any = PanelMenu_rootItem_states['*'];
  var startStateStyles:any = PanelMenu_rootItem_states[currentState];
  if ((startStateStyles == (null as any))) { (startStateStyles = defaultStateStyles); }
  var endStateStyles:any = PanelMenu_rootItem_states[nextState];
  if ((endStateStyles == (null as any))) { (endStateStyles = defaultStateStyles); }
  if (((player == (null as any)) && ((currentState == 'visible') && (nextState == 'hidden')))) {
      player = new import16.AnimationSequencePlayer([view.renderer.animate(element,new import17.AnimationStyles(import18.collectAndResolveStyles(collectedStyles,[startStateStyles])),import18.balanceAnimationKeyframes(collectedStyles,endStateStyles,[
        new import19.AnimationKeyframe(0,new import17.AnimationStyles(import18.collectAndResolveStyles(collectedStyles,[{}]))),
        new import19.AnimationKeyframe(1,new import17.AnimationStyles(import18.collectAndResolveStyles(collectedStyles,[{}])))
      ]
    ),400,0,'cubic-bezier(0.86, 0, 0.07, 1)',previousPlayers)]);
    totalTime = 400;
  }
  if (((player == (null as any)) && ((currentState == 'hidden') && (nextState == 'visible')))) {
      player = new import16.AnimationSequencePlayer([view.renderer.animate(element,new import17.AnimationStyles(import18.collectAndResolveStyles(collectedStyles,[startStateStyles])),import18.balanceAnimationKeyframes(collectedStyles,endStateStyles,[
        new import19.AnimationKeyframe(0,new import17.AnimationStyles(import18.collectAndResolveStyles(collectedStyles,[{}]))),
        new import19.AnimationKeyframe(1,new import17.AnimationStyles(import18.collectAndResolveStyles(collectedStyles,[{}])))
      ]
    ),400,0,'cubic-bezier(0.86, 0, 0.07, 1)',previousPlayers)]);
    totalTime = 400;
  }
  if ((player == (null as any))) { (player = new import20.NoOpAnimationPlayer()); }
  player.onDone(():void => {
    player.destroy();
    import18.renderStyles(element,view.renderer,import18.prepareFinalAnimationStyles(startStateStyles,endStateStyles));
  });
  new import16.AnimationSequencePlayer(previousPlayers).destroy();
  import18.renderStyles(element,view.renderer,import18.clearStyles(startStateStyles));
  view.animationContext.queueAnimation(element,'rootItem',player);
  return new import15.AnimationTransition(player,currentState,nextState,totalTime);
}
class View_PanelMenu2 extends import7.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import21.Wrapper_NgClass;
  _map_2:any;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import22.ViewContainer) {
    super(View_PanelMenu2,renderType_PanelMenu,import11.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import12.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_2 = import8.pureProxy2((p0:any,p1:any):{[key: string]:any} => {
      return {
        'fa-caret-right': p0,
        'fa-caret-down': p1
      }
      ;
    });
  }
  createInternal(rootSelector:string):import13.ComponentRef<any> {
    this._el_0 = import8.createRenderElement(this.renderer,(null as any),'span',new import8.InlineArray2(2,'class','ui-panelmenu-icon fa'),(null as any));
    this._NgClass_0_3 = new import21.Wrapper_NgClass(this.parentView.parentView.parentView.injectorGet(import23.IterableDiffers,this.parentView.parentView.parentIndex),this.parentView.parentView.parentView.injectorGet(import24.KeyValueDiffers,this.parentView.parentView.parentIndex),new import25.ElementRef(this._el_0),this.renderer);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import26.NgClass) && (0 === requestNodeIndex))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = 'ui-panelmenu-icon fa';
    this._NgClass_0_3.check_klass(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = this._map_2(!this.parentView.context.$implicit.expanded,this.parentView.context.$implicit.expanded);
    this._NgClass_0_3.check_ngClass(currVal_0_0_1,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_PanelMenu3 extends import7.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import21.Wrapper_NgClass;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import22.ViewContainer) {
    super(View_PanelMenu3,renderType_PanelMenu,import11.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import12.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import13.ComponentRef<any> {
    this._el_0 = import8.createRenderElement(this.renderer,(null as any),'span',new import8.InlineArray2(2,'class','ui-menuitem-icon fa'),(null as any));
    this._NgClass_0_3 = new import21.Wrapper_NgClass(this.parentView.parentView.parentView.injectorGet(import23.IterableDiffers,this.parentView.parentView.parentIndex),this.parentView.parentView.parentView.injectorGet(import24.KeyValueDiffers,this.parentView.parentView.parentIndex),new import25.ElementRef(this._el_0),this.renderer);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import26.NgClass) && (0 === requestNodeIndex))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = 'ui-menuitem-icon fa';
    this._NgClass_0_3.check_klass(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = this.parentView.context.$implicit.icon;
    this._NgClass_0_3.check_ngClass(currVal_0_0_1,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_PanelMenu4 extends import7.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import21.Wrapper_NgClass;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _el_4:any;
  compView_4:import7.AppView<import1.PanelMenuSub>;
  _PanelMenuSub_4_3:Wrapper_PanelMenuSub;
  _text_5:any;
  _text_6:any;
  /*private*/ _expr_10:any;
  _map_11:any;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import22.ViewContainer) {
    super(View_PanelMenu4,renderType_PanelMenu,import11.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import12.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_10 = import6.UNINITIALIZED;
    this._map_11 = import8.pureProxy1((p0:any):{[key: string]:any} => {
      return {'ui-panelmenu-content-wrapper-overflown': p0};
    });
  }
  createInternal(rootSelector:string):import13.ComponentRef<any> {
    this._el_0 = import8.createRenderElement(this.renderer,(null as any),'div',new import8.InlineArray2(2,'class','ui-panelmenu-content-wrapper'),(null as any));
    this._NgClass_0_3 = new import21.Wrapper_NgClass(this.parentView.parentView.parentView.injectorGet(import23.IterableDiffers,this.parentView.parentView.parentIndex),this.parentView.parentView.parentView.injectorGet(import24.KeyValueDiffers,this.parentView.parentView.parentIndex),new import25.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this._el_2 = import8.createRenderElement(this.renderer,this._el_0,'div',new import8.InlineArray2(2,'class','ui-panelmenu-content ui-widget-content'),(null as any));
    this._text_3 = this.renderer.createText(this._el_2,'\n                        ',(null as any));
    this._el_4 = import8.createRenderElement(this.renderer,this._el_2,'p-panelMenuSub',import8.EMPTY_INLINE_ARRAY,(null as any));
    this.compView_4 = new View_PanelMenuSub0(this.viewUtils,this,4,this._el_4);
    this._PanelMenuSub_4_3 = new Wrapper_PanelMenuSub(this.parentView.parentView.parentView.injectorGet(import14.Router,this.parentView.parentView.parentIndex));
    this.compView_4.create(this._PanelMenuSub_4_3.context);
    this._text_5 = this.renderer.createText(this._el_2,'\n                    ',(null as any));
    this._text_6 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._el_4,
      this._text_5,
      this._text_6
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import1.PanelMenuSub) && (4 === requestNodeIndex))) { return this._PanelMenuSub_4_3.context; }
    if (((token === import26.NgClass) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 6)))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_10:any = (this.parentView.context.$implicit.expanded? 'visible': 'hidden');
    if (import8.checkBinding(throwOnChange,this._expr_10,currVal_10)) {
      var animationTransition_rootItem:any = this.componentType.animations['rootItem'](this,this._el_0,((this._expr_10 == import6.UNINITIALIZED)? 'void': this._expr_10),((currVal_10 == import6.UNINITIALIZED)? 'void': currVal_10));
      this._expr_10 = currVal_10;
    }
    const currVal_0_0_0:any = 'ui-panelmenu-content-wrapper';
    this._NgClass_0_3.check_klass(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = this._map_11((!this.parentView.context.$implicit.expanded || this.parentView.parentView.context.animating));
    this._NgClass_0_3.check_ngClass(currVal_0_0_1,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_4_0_0:any = this.parentView.context.$implicit;
    this._PanelMenuSub_4_3.check_item(currVal_4_0_0,throwOnChange,false);
    const currVal_4_0_1:any = true;
    this._PanelMenuSub_4_3.check_expanded(currVal_4_0_1,throwOnChange,false);
    this._PanelMenuSub_4_3.ngDoCheck(this,this._el_4,throwOnChange);
    this.compView_4.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_4.destroy();
  }
  detachInternal():void {
    var animationTransition_rootItem:any = this.componentType.animations['rootItem'](this,this._el_0,this._expr_10,'void');
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_PanelMenu1 extends import7.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _NgClass_2_3:import21.Wrapper_NgClass;
  _text_3:any;
  _el_4:any;
  _NgClass_4_3:import21.Wrapper_NgClass;
  _text_5:any;
  _anchor_6:any;
  /*private*/ _vc_6:import22.ViewContainer;
  _TemplateRef_6_5:any;
  _NgIf_6_6:import27.Wrapper_NgIf;
  _anchor_7:any;
  /*private*/ _vc_7:import22.ViewContainer;
  _TemplateRef_7_5:any;
  _NgIf_7_6:import27.Wrapper_NgIf;
  _el_8:any;
  _text_9:any;
  _text_10:any;
  _text_11:any;
  _text_12:any;
  _anchor_13:any;
  /*private*/ _vc_13:import22.ViewContainer;
  _TemplateRef_13_5:any;
  _NgIf_13_6:import27.Wrapper_NgIf;
  _text_14:any;
  _map_26:any;
  /*private*/ _expr_27:any;
  _map_28:any;
  /*private*/ _expr_29:any;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import22.ViewContainer) {
    super(View_PanelMenu1,renderType_PanelMenu,import11.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import12.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_26 = import8.pureProxy6((p0:any,p1:any,p2:any,p3:any,p4:any,p5:any):{[key: string]:any} => {
      return {
        'ui-widget ui-panelmenu-header ui-state-default': p0,
        'ui-corner-top': p1,
        'ui-corner-bottom': p2,
        'ui-state-active': p3,
        'ui-state-hover': p4,
        'ui-state-disabled': p5
      }
      ;
    });
    this._expr_27 = import6.UNINITIALIZED;
    this._map_28 = import8.pureProxy1((p0:any):{[key: string]:any} => {
      return {'ui-panelmenu-headerlink-hasicon': p0};
    });
    this._expr_29 = import6.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import13.ComponentRef<any> {
    this._el_0 = import8.createRenderElement(this.renderer,(null as any),'div',new import8.InlineArray2(2,'class','ui-panelmenu-panel'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    this._el_2 = import8.createRenderElement(this.renderer,this._el_0,'div',new import8.InlineArray2(2,'tabindex','0'),(null as any));
    this._NgClass_2_3 = new import21.Wrapper_NgClass(this.parentView.parentView.injectorGet(import23.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import24.KeyValueDiffers,this.parentView.parentIndex),new import25.ElementRef(this._el_2),this.renderer);
    this._text_3 = this.renderer.createText(this._el_2,'\n                    ',(null as any));
    this._el_4 = import8.createRenderElement(this.renderer,this._el_2,'a',import8.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_4_3 = new import21.Wrapper_NgClass(this.parentView.parentView.injectorGet(import23.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import24.KeyValueDiffers,this.parentView.parentIndex),new import25.ElementRef(this._el_4),this.renderer);
    this._text_5 = this.renderer.createText(this._el_4,'\n                        ',(null as any));
    this._anchor_6 = this.renderer.createTemplateAnchor(this._el_4,(null as any));
    this._vc_6 = new import22.ViewContainer(6,4,this,this._anchor_6);
    this._TemplateRef_6_5 = new import28.TemplateRef_(this,6,this._anchor_6);
    this._NgIf_6_6 = new import27.Wrapper_NgIf(this._vc_6.vcRef,this._TemplateRef_6_5);
    this._anchor_7 = this.renderer.createTemplateAnchor(this._el_4,(null as any));
    this._vc_7 = new import22.ViewContainer(7,4,this,this._anchor_7);
    this._TemplateRef_7_5 = new import28.TemplateRef_(this,7,this._anchor_7);
    this._NgIf_7_6 = new import27.Wrapper_NgIf(this._vc_7.vcRef,this._TemplateRef_7_5);
    this._el_8 = import8.createRenderElement(this.renderer,this._el_4,'span',new import8.InlineArray2(2,'class','ui-menuitem-text'),(null as any));
    this._text_9 = this.renderer.createText(this._el_8,'',(null as any));
    this._text_10 = this.renderer.createText(this._el_4,'\n                    ',(null as any));
    this._text_11 = this.renderer.createText(this._el_2,'\n                ',(null as any));
    this._text_12 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    this._anchor_13 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_13 = new import22.ViewContainer(13,0,this,this._anchor_13);
    this._TemplateRef_13_5 = new import28.TemplateRef_(this,13,this._anchor_13);
    this._NgIf_13_6 = new import27.Wrapper_NgIf(this._vc_13.vcRef,this._TemplateRef_13_5);
    this._text_14 = this.renderer.createText(this._el_0,'\n            ',(null as any));
    var disposable_0:Function = import8.subscribeToRenderElement(this,this._el_4,new import8.InlineArray8(6,'mouseenter',(null as any),'mouseleave',(null as any),'click',(null as any)),this.eventHandler(this.handleEvent_4));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._el_4,
      this._text_5,
      this._anchor_6,
      this._anchor_7,
      this._el_8,
      this._text_9,
      this._text_10,
      this._text_11,
      this._text_12,
      this._anchor_13,
      this._text_14
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import28.TemplateRef) && (6 === requestNodeIndex))) { return this._TemplateRef_6_5; }
    if (((token === import29.NgIf) && (6 === requestNodeIndex))) { return this._NgIf_6_6.context; }
    if (((token === import28.TemplateRef) && (7 === requestNodeIndex))) { return this._TemplateRef_7_5; }
    if (((token === import29.NgIf) && (7 === requestNodeIndex))) { return this._NgIf_7_6.context; }
    if (((token === import26.NgClass) && ((4 <= requestNodeIndex) && (requestNodeIndex <= 10)))) { return this._NgClass_4_3.context; }
    if (((token === import26.NgClass) && ((2 <= requestNodeIndex) && (requestNodeIndex <= 11)))) { return this._NgClass_2_3.context; }
    if (((token === import28.TemplateRef) && (13 === requestNodeIndex))) { return this._TemplateRef_13_5; }
    if (((token === import29.NgIf) && (13 === requestNodeIndex))) { return this._NgIf_13_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2_0_0:any = this._map_26(true,this.context.first,(this.context.last && !this.context.$implicit.expanded),this.context.$implicit.expanded,((this.context.$implicit == this.parentView.context.hoveredItem) && !this.context.$implicit.disabled),this.context.$implicit.disabled);
    this._NgClass_2_3.check_ngClass(currVal_2_0_0,throwOnChange,false);
    this._NgClass_2_3.ngDoCheck(this,this._el_2,throwOnChange);
    const currVal_4_0_0:any = this._map_28(this.context.$implicit.icon);
    this._NgClass_4_3.check_ngClass(currVal_4_0_0,throwOnChange,false);
    this._NgClass_4_3.ngDoCheck(this,this._el_4,throwOnChange);
    const currVal_6_0_0:any = this.context.$implicit.items;
    this._NgIf_6_6.check_ngIf(currVal_6_0_0,throwOnChange,false);
    this._NgIf_6_6.ngDoCheck(this,this._anchor_6,throwOnChange);
    const currVal_7_0_0:any = this.context.$implicit.icon;
    this._NgIf_7_6.check_ngIf(currVal_7_0_0,throwOnChange,false);
    this._NgIf_7_6.ngDoCheck(this,this._anchor_7,throwOnChange);
    const currVal_13_0_0:any = this.context.$implicit.items;
    this._NgIf_13_6.check_ngIf(currVal_13_0_0,throwOnChange,false);
    this._NgIf_13_6.ngDoCheck(this,this._anchor_13,throwOnChange);
    this._vc_6.detectChangesInNestedViews(throwOnChange);
    this._vc_7.detectChangesInNestedViews(throwOnChange);
    this._vc_13.detectChangesInNestedViews(throwOnChange);
    const currVal_27:any = (this.context.$implicit.url || '#');
    if (import8.checkBinding(throwOnChange,this._expr_27,currVal_27)) {
      this.renderer.setElementProperty(this._el_4,'href',this.viewUtils.sanitizer.sanitize(import30.SecurityContext.URL,currVal_27));
      this._expr_27 = currVal_27;
    }
    const currVal_29:any = import8.inlineInterpolate(1,'',this.context.$implicit.label,'');
    if (import8.checkBinding(throwOnChange,this._expr_29,currVal_29)) {
      this.renderer.setText(this._text_9,currVal_29);
      this._expr_29 = currVal_29;
    }
  }
  destroyInternal():void {
    this._vc_6.destroyNestedViews();
    this._vc_7.destroyNestedViews();
    this._vc_13.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import7.AppView<any> {
    if ((nodeIndex == 6)) { return new View_PanelMenu2(this.viewUtils,this,6,this._anchor_6,this._vc_6); }
    if ((nodeIndex == 7)) { return new View_PanelMenu3(this.viewUtils,this,7,this._anchor_7,this._vc_7); }
    if ((nodeIndex == 13)) { return new View_PanelMenu4(this.viewUtils,this,13,this._anchor_13,this._vc_13); }
    return (null as any);
  }
  handleEvent_4(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'mouseenter')) {
      const pd_sub_0:any = ((<any>(this.parentView.context.hoveredItem = this.context.$implicit)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'mouseleave')) {
      const pd_sub_1:any = ((<any>(this.parentView.context.hoveredItem = (null as any))) !== false);
      result = (pd_sub_1 && result);
    }
    if ((eventName == 'click')) {
      const pd_sub_2:any = ((<any>this.parentView.context.handleClick($event,this.context.$implicit)) !== false);
      result = (pd_sub_2 && result);
    }
    return result;
  }
}
var renderType_PanelMenu:import9.RenderComponentType = import8.createRenderComponentType('',0,import10.ViewEncapsulation.None,styles_PanelMenu,{rootItem: PanelMenu_rootItem_factory});
export class View_PanelMenu0 extends import7.AppView<import1.PanelMenu> {
  _text_0:any;
  _el_1:any;
  _NgClass_1_3:import21.Wrapper_NgClass;
  _NgStyle_1_4:import31.Wrapper_NgStyle;
  _text_2:any;
  _anchor_3:any;
  /*private*/ _vc_3:import22.ViewContainer;
  _TemplateRef_3_5:any;
  _NgFor_3_6:import32.Wrapper_NgFor;
  _text_4:any;
  _text_5:any;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_PanelMenu0,renderType_PanelMenu,import11.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import12.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import13.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n        ',(null as any));
    this._el_1 = import8.createRenderElement(this.renderer,parentRenderNode,'div',import8.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_1_3 = new import21.Wrapper_NgClass(this.parentView.injectorGet(import23.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import24.KeyValueDiffers,this.parentIndex),new import25.ElementRef(this._el_1),this.renderer);
    this._NgStyle_1_4 = new import31.Wrapper_NgStyle(this.parentView.injectorGet(import24.KeyValueDiffers,this.parentIndex),new import25.ElementRef(this._el_1),this.renderer);
    this._text_2 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._anchor_3 = this.renderer.createTemplateAnchor(this._el_1,(null as any));
    this._vc_3 = new import22.ViewContainer(3,1,this,this._anchor_3);
    this._TemplateRef_3_5 = new import28.TemplateRef_(this,3,this._anchor_3);
    this._NgFor_3_6 = new import32.Wrapper_NgFor(this._vc_3.vcRef,this._TemplateRef_3_5,this.parentView.injectorGet(import23.IterableDiffers,this.parentIndex),this.ref);
    this._text_4 = this.renderer.createText(this._el_1,'\n        ',(null as any));
    this._text_5 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2,
      this._anchor_3,
      this._text_4,
      this._text_5
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import28.TemplateRef) && (3 === requestNodeIndex))) { return this._TemplateRef_3_5; }
    if (((token === import33.NgFor) && (3 === requestNodeIndex))) { return this._NgFor_3_6.context; }
    if (((token === import26.NgClass) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 4)))) { return this._NgClass_1_3.context; }
    if (((token === import34.NgStyle) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 4)))) { return this._NgStyle_1_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1_0_0:any = this.context.styleClass;
    this._NgClass_1_3.check_klass(currVal_1_0_0,throwOnChange,false);
    const currVal_1_0_1:any = 'ui-panelmenu ui-widget';
    this._NgClass_1_3.check_ngClass(currVal_1_0_1,throwOnChange,false);
    this._NgClass_1_3.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_1_1_0:any = this.context.style;
    this._NgStyle_1_4.check_ngStyle(currVal_1_1_0,throwOnChange,false);
    this._NgStyle_1_4.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_3_0_0:any = this.context.model;
    this._NgFor_3_6.check_ngForOf(currVal_3_0_0,throwOnChange,false);
    this._NgFor_3_6.ngDoCheck(this,this._anchor_3,throwOnChange);
    this._vc_3.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_3.destroyNestedViews();
  }
  createEmbeddedViewInternal(nodeIndex:number):import7.AppView<any> {
    if ((nodeIndex == 3)) { return new View_PanelMenu1(this.viewUtils,this,3,this._anchor_3,this._vc_3); }
    return (null as any);
  }
}
var renderType_PanelMenuSub_Host:import9.RenderComponentType = import8.createRenderComponentType('',0,import10.ViewEncapsulation.None,([] as any[]),{});
class View_PanelMenuSub_Host0 extends import7.AppView<any> {
  _el_0:any;
  compView_0:import7.AppView<import1.PanelMenuSub>;
  _PanelMenuSub_0_3:Wrapper_PanelMenuSub;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_PanelMenuSub_Host0,renderType_PanelMenuSub_Host,import11.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import12.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import13.ComponentRef<any> {
    this._el_0 = import8.selectOrCreateRenderHostElement(this.renderer,'p-panelMenuSub',import8.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this.compView_0 = new View_PanelMenuSub0(this.viewUtils,this,0,this._el_0);
    this._PanelMenuSub_0_3 = new Wrapper_PanelMenuSub(this.injectorGet(import14.Router,this.parentIndex));
    this.compView_0.create(this._PanelMenuSub_0_3.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import13.ComponentRef_<any>(0,this,this._el_0,this._PanelMenuSub_0_3.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import1.PanelMenuSub) && (0 === requestNodeIndex))) { return this._PanelMenuSub_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._PanelMenuSub_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
export const PanelMenuSubNgFactory:import13.ComponentFactory<import1.PanelMenuSub> = new import13.ComponentFactory<import1.PanelMenuSub>('p-panelMenuSub',View_PanelMenuSub_Host0,import1.PanelMenuSub);
const styles_PanelMenuSub:any[] = ([] as any[]);
class View_PanelMenuSub2 extends import7.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import21.Wrapper_NgClass;
  _map_2:any;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import22.ViewContainer) {
    super(View_PanelMenuSub2,renderType_PanelMenuSub,import11.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import12.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_2 = import8.pureProxy2((p0:any,p1:any):{[key: string]:any} => {
      return {
        'fa-caret-right': p0,
        'fa-caret-down': p1
      }
      ;
    });
  }
  createInternal(rootSelector:string):import13.ComponentRef<any> {
    this._el_0 = import8.createRenderElement(this.renderer,(null as any),'span',new import8.InlineArray2(2,'class','ui-panelmenu-icon fa fa-fw'),(null as any));
    this._NgClass_0_3 = new import21.Wrapper_NgClass(this.parentView.parentView.parentView.injectorGet(import23.IterableDiffers,this.parentView.parentView.parentIndex),this.parentView.parentView.parentView.injectorGet(import24.KeyValueDiffers,this.parentView.parentView.parentIndex),new import25.ElementRef(this._el_0),this.renderer);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import26.NgClass) && (0 === requestNodeIndex))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = 'ui-panelmenu-icon fa fa-fw';
    this._NgClass_0_3.check_klass(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = this._map_2(!this.parentView.context.$implicit.expanded,this.parentView.context.$implicit.expanded);
    this._NgClass_0_3.check_ngClass(currVal_0_0_1,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_PanelMenuSub3 extends import7.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import21.Wrapper_NgClass;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import22.ViewContainer) {
    super(View_PanelMenuSub3,renderType_PanelMenuSub,import11.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import12.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import13.ComponentRef<any> {
    this._el_0 = import8.createRenderElement(this.renderer,(null as any),'span',new import8.InlineArray2(2,'class','ui-menuitem-icon fa fa-fw'),(null as any));
    this._NgClass_0_3 = new import21.Wrapper_NgClass(this.parentView.parentView.parentView.injectorGet(import23.IterableDiffers,this.parentView.parentView.parentIndex),this.parentView.parentView.parentView.injectorGet(import24.KeyValueDiffers,this.parentView.parentView.parentIndex),new import25.ElementRef(this._el_0),this.renderer);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import26.NgClass) && (0 === requestNodeIndex))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = 'ui-menuitem-icon fa fa-fw';
    this._NgClass_0_3.check_klass(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = this.parentView.context.$implicit.icon;
    this._NgClass_0_3.check_ngClass(currVal_0_0_1,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_PanelMenuSub4 extends import7.AppView<any> {
  _el_0:any;
  compView_0:import7.AppView<import1.PanelMenuSub>;
  _PanelMenuSub_0_3:Wrapper_PanelMenuSub;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import22.ViewContainer) {
    super(View_PanelMenuSub4,renderType_PanelMenuSub,import11.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import12.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import13.ComponentRef<any> {
    this._el_0 = import8.createRenderElement(this.renderer,(null as any),'p-panelMenuSub',import8.EMPTY_INLINE_ARRAY,(null as any));
    this.compView_0 = new View_PanelMenuSub0(this.viewUtils,this,0,this._el_0);
    this._PanelMenuSub_0_3 = new Wrapper_PanelMenuSub(this.parentView.parentView.parentView.injectorGet(import14.Router,this.parentView.parentView.parentIndex));
    this.compView_0.create(this._PanelMenuSub_0_3.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import1.PanelMenuSub) && (0 === requestNodeIndex))) { return this._PanelMenuSub_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this.parentView.context.$implicit;
    this._PanelMenuSub_0_3.check_item(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = this.parentView.context.$implicit.expanded;
    this._PanelMenuSub_0_3.check_expanded(currVal_0_0_1,throwOnChange,false);
    this._PanelMenuSub_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_PanelMenuSub1 extends import7.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import21.Wrapper_NgClass;
  _text_1:any;
  _el_2:any;
  _NgClass_2_3:import21.Wrapper_NgClass;
  _text_3:any;
  _anchor_4:any;
  /*private*/ _vc_4:import22.ViewContainer;
  _TemplateRef_4_5:any;
  _NgIf_4_6:import27.Wrapper_NgIf;
  _anchor_5:any;
  /*private*/ _vc_5:import22.ViewContainer;
  _TemplateRef_5_5:any;
  _NgIf_5_6:import27.Wrapper_NgIf;
  _el_6:any;
  _text_7:any;
  _text_8:any;
  _text_9:any;
  _anchor_10:any;
  /*private*/ _vc_10:import22.ViewContainer;
  _TemplateRef_10_5:any;
  _NgIf_10_6:import27.Wrapper_NgIf;
  _text_11:any;
  _map_23:any;
  /*private*/ _expr_24:any;
  _map_25:any;
  /*private*/ _expr_26:any;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import22.ViewContainer) {
    super(View_PanelMenuSub1,renderType_PanelMenuSub,import11.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import12.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_23 = import8.pureProxy1((p0:any):{[key: string]:any} => {
      return {'ui-menu-parent': p0};
    });
    this._expr_24 = import6.UNINITIALIZED;
    this._map_25 = import8.pureProxy3((p0:any,p1:any,p2:any):{[key: string]:any} => {
      return {
        'ui-menuitem-link-hasicon': p0,
        'ui-state-hover': p1,
        'ui-state-disabled': p2
      }
      ;
    });
    this._expr_26 = import6.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import13.ComponentRef<any> {
    this._el_0 = import8.createRenderElement(this.renderer,(null as any),'li',new import8.InlineArray2(2,'class','ui-menuitem ui-corner-all'),(null as any));
    this._NgClass_0_3 = new import21.Wrapper_NgClass(this.parentView.parentView.injectorGet(import23.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import24.KeyValueDiffers,this.parentView.parentIndex),new import25.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    this._el_2 = import8.createRenderElement(this.renderer,this._el_0,'a',new import8.InlineArray2(2,'class','ui-menuitem-link ui-corner-all'),(null as any));
    this._NgClass_2_3 = new import21.Wrapper_NgClass(this.parentView.parentView.injectorGet(import23.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import24.KeyValueDiffers,this.parentView.parentIndex),new import25.ElementRef(this._el_2),this.renderer);
    this._text_3 = this.renderer.createText(this._el_2,'\n                    ',(null as any));
    this._anchor_4 = this.renderer.createTemplateAnchor(this._el_2,(null as any));
    this._vc_4 = new import22.ViewContainer(4,2,this,this._anchor_4);
    this._TemplateRef_4_5 = new import28.TemplateRef_(this,4,this._anchor_4);
    this._NgIf_4_6 = new import27.Wrapper_NgIf(this._vc_4.vcRef,this._TemplateRef_4_5);
    this._anchor_5 = this.renderer.createTemplateAnchor(this._el_2,(null as any));
    this._vc_5 = new import22.ViewContainer(5,2,this,this._anchor_5);
    this._TemplateRef_5_5 = new import28.TemplateRef_(this,5,this._anchor_5);
    this._NgIf_5_6 = new import27.Wrapper_NgIf(this._vc_5.vcRef,this._TemplateRef_5_5);
    this._el_6 = import8.createRenderElement(this.renderer,this._el_2,'span',new import8.InlineArray2(2,'class','ui-menuitem-text'),(null as any));
    this._text_7 = this.renderer.createText(this._el_6,'',(null as any));
    this._text_8 = this.renderer.createText(this._el_2,'\n                ',(null as any));
    this._text_9 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    this._anchor_10 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_10 = new import22.ViewContainer(10,0,this,this._anchor_10);
    this._TemplateRef_10_5 = new import28.TemplateRef_(this,10,this._anchor_10);
    this._NgIf_10_6 = new import27.Wrapper_NgIf(this._vc_10.vcRef,this._TemplateRef_10_5);
    this._text_11 = this.renderer.createText(this._el_0,'\n            ',(null as any));
    var disposable_0:Function = import8.subscribeToRenderElement(this,this._el_2,new import8.InlineArray8(6,'click',(null as any),'mouseenter',(null as any),'mouseleave',(null as any)),this.eventHandler(this.handleEvent_2));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._anchor_4,
      this._anchor_5,
      this._el_6,
      this._text_7,
      this._text_8,
      this._text_9,
      this._anchor_10,
      this._text_11
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import28.TemplateRef) && (4 === requestNodeIndex))) { return this._TemplateRef_4_5; }
    if (((token === import29.NgIf) && (4 === requestNodeIndex))) { return this._NgIf_4_6.context; }
    if (((token === import28.TemplateRef) && (5 === requestNodeIndex))) { return this._TemplateRef_5_5; }
    if (((token === import29.NgIf) && (5 === requestNodeIndex))) { return this._NgIf_5_6.context; }
    if (((token === import26.NgClass) && ((2 <= requestNodeIndex) && (requestNodeIndex <= 8)))) { return this._NgClass_2_3.context; }
    if (((token === import28.TemplateRef) && (10 === requestNodeIndex))) { return this._TemplateRef_10_5; }
    if (((token === import29.NgIf) && (10 === requestNodeIndex))) { return this._NgIf_10_6.context; }
    if (((token === import26.NgClass) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 11)))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = 'ui-menuitem ui-corner-all';
    this._NgClass_0_3.check_klass(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = this._map_23(this.context.$implicit.items);
    this._NgClass_0_3.check_ngClass(currVal_0_0_1,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_2_0_0:any = 'ui-menuitem-link ui-corner-all';
    this._NgClass_2_3.check_klass(currVal_2_0_0,throwOnChange,false);
    const currVal_2_0_1:any = this._map_25((this.context.$implicit.icon && this.context.$implicit.items),((this.parentView.context.hoveredLink == this._el_2) && !this.context.$implicit.disabled),this.context.$implicit.disabled);
    this._NgClass_2_3.check_ngClass(currVal_2_0_1,throwOnChange,false);
    this._NgClass_2_3.ngDoCheck(this,this._el_2,throwOnChange);
    const currVal_4_0_0:any = this.context.$implicit.items;
    this._NgIf_4_6.check_ngIf(currVal_4_0_0,throwOnChange,false);
    this._NgIf_4_6.ngDoCheck(this,this._anchor_4,throwOnChange);
    const currVal_5_0_0:any = this.context.$implicit.icon;
    this._NgIf_5_6.check_ngIf(currVal_5_0_0,throwOnChange,false);
    this._NgIf_5_6.ngDoCheck(this,this._anchor_5,throwOnChange);
    const currVal_10_0_0:any = this.context.$implicit.items;
    this._NgIf_10_6.check_ngIf(currVal_10_0_0,throwOnChange,false);
    this._NgIf_10_6.ngDoCheck(this,this._anchor_10,throwOnChange);
    this._vc_4.detectChangesInNestedViews(throwOnChange);
    this._vc_5.detectChangesInNestedViews(throwOnChange);
    this._vc_10.detectChangesInNestedViews(throwOnChange);
    const currVal_24:any = (this.context.$implicit.url || '#');
    if (import8.checkBinding(throwOnChange,this._expr_24,currVal_24)) {
      this.renderer.setElementProperty(this._el_2,'href',this.viewUtils.sanitizer.sanitize(import30.SecurityContext.URL,currVal_24));
      this._expr_24 = currVal_24;
    }
    const currVal_26:any = import8.inlineInterpolate(1,'',this.context.$implicit.label,'');
    if (import8.checkBinding(throwOnChange,this._expr_26,currVal_26)) {
      this.renderer.setText(this._text_7,currVal_26);
      this._expr_26 = currVal_26;
    }
  }
  destroyInternal():void {
    this._vc_4.destroyNestedViews();
    this._vc_5.destroyNestedViews();
    this._vc_10.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import7.AppView<any> {
    if ((nodeIndex == 4)) { return new View_PanelMenuSub2(this.viewUtils,this,4,this._anchor_4,this._vc_4); }
    if ((nodeIndex == 5)) { return new View_PanelMenuSub3(this.viewUtils,this,5,this._anchor_5,this._vc_5); }
    if ((nodeIndex == 10)) { return new View_PanelMenuSub4(this.viewUtils,this,10,this._anchor_10,this._vc_10); }
    return (null as any);
  }
  handleEvent_2(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.context.handleClick($event,this.context.$implicit)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'mouseenter')) {
      const pd_sub_1:any = ((<any>(this.parentView.context.hoveredLink = this._el_2)) !== false);
      result = (pd_sub_1 && result);
    }
    if ((eventName == 'mouseleave')) {
      const pd_sub_2:any = ((<any>(this.parentView.context.hoveredLink = (null as any))) !== false);
      result = (pd_sub_2 && result);
    }
    return result;
  }
}
var renderType_PanelMenuSub:import9.RenderComponentType = import8.createRenderComponentType('',0,import10.ViewEncapsulation.None,styles_PanelMenuSub,{});
export class View_PanelMenuSub0 extends import7.AppView<import1.PanelMenuSub> {
  _text_0:any;
  _el_1:any;
  _text_2:any;
  _anchor_3:any;
  /*private*/ _vc_3:import22.ViewContainer;
  _TemplateRef_3_5:any;
  _NgFor_3_6:import32.Wrapper_NgFor;
  _text_4:any;
  _text_5:any;
  /*private*/ _expr_9:any;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_PanelMenuSub0,renderType_PanelMenuSub,import11.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import12.ChangeDetectorStatus.CheckAlways);
    this._expr_9 = import6.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import13.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n        ',(null as any));
    this._el_1 = import8.createRenderElement(this.renderer,parentRenderNode,'ul',new import8.InlineArray2(2,'class','ui-menu-list ui-helper-reset'),(null as any));
    this._text_2 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._anchor_3 = this.renderer.createTemplateAnchor(this._el_1,(null as any));
    this._vc_3 = new import22.ViewContainer(3,1,this,this._anchor_3);
    this._TemplateRef_3_5 = new import28.TemplateRef_(this,3,this._anchor_3);
    this._NgFor_3_6 = new import32.Wrapper_NgFor(this._vc_3.vcRef,this._TemplateRef_3_5,this.parentView.injectorGet(import23.IterableDiffers,this.parentIndex),this.ref);
    this._text_4 = this.renderer.createText(this._el_1,'\n        ',(null as any));
    this._text_5 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2,
      this._anchor_3,
      this._text_4,
      this._text_5
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import28.TemplateRef) && (3 === requestNodeIndex))) { return this._TemplateRef_3_5; }
    if (((token === import33.NgFor) && (3 === requestNodeIndex))) { return this._NgFor_3_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_3_0_0:any = this.context.item.items;
    this._NgFor_3_6.check_ngForOf(currVal_3_0_0,throwOnChange,false);
    this._NgFor_3_6.ngDoCheck(this,this._anchor_3,throwOnChange);
    this._vc_3.detectChangesInNestedViews(throwOnChange);
    const currVal_9:any = (this.context.expanded? 'block': 'none');
    if (import8.checkBinding(throwOnChange,this._expr_9,currVal_9)) {
      this.renderer.setElementStyle(this._el_1,'display',((this.viewUtils.sanitizer.sanitize(import30.SecurityContext.STYLE,currVal_9) == null)? (null as any): this.viewUtils.sanitizer.sanitize(import30.SecurityContext.STYLE,currVal_9).toString()));
      this._expr_9 = currVal_9;
    }
  }
  destroyInternal():void {
    this._vc_3.destroyNestedViews();
  }
  createEmbeddedViewInternal(nodeIndex:number):import7.AppView<any> {
    if ((nodeIndex == 3)) { return new View_PanelMenuSub1(this.viewUtils,this,3,this._anchor_3,this._vc_3); }
    return (null as any);
  }
}
