// ng2-file-upload/file-upload/file-drop.directive.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from 'ng2-file-upload/file-upload/file-drop.directive';
import * as import1 from '@angular/core/src/change_detection/change_detection_util';
import * as import2 from '@angular/core/src/linker/view';
import * as import3 from '@angular/core/src/linker/view_utils';
export class Wrapper_FileDropDirective {
  /*private*/ _eventHandler:Function;
  context:import0.FileDropDirective;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  subscription0:any;
  subscription1:any;
  constructor(p0:any) {
    this._changed = false;
    this.context = new import0.FileDropDirective(p0);
    this._expr_0 = import1.UNINITIALIZED;
  }
  ngOnDetach(view:import2.AppView<any>,componentView:import2.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
    (this.subscription0 && this.subscription0.unsubscribe());
    (this.subscription1 && this.subscription1.unsubscribe());
  }
  check_uploader(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.uploader = currValue;
      this._expr_0 = currValue;
    }
  }
  ngDoCheck(view:import2.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    return changed;
  }
  checkHost(view:import2.AppView<any>,componentView:import2.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    if ((eventName == 'drop')) {
      const pd_sub_0:any = ((<any>this.context.onDrop($event)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'dragover')) {
      const pd_sub_1:any = ((<any>this.context.onDragOver($event)) !== false);
      result = (pd_sub_1 && result);
    }
    if ((eventName == 'dragleave')) {
      const pd_sub_2:any = ((<any>this.context.onDragLeave($event)) !== false);
      result = (pd_sub_2 && result);
    }
    return result;
  }
  subscribe(view:import2.AppView<any>,_eventHandler:any,emit0:boolean,emit1:boolean):void {
    this._eventHandler = _eventHandler;
    if (emit0) { (this.subscription0 = this.context.fileOver.subscribe(_eventHandler.bind(view,'fileOver'))); }
    if (emit1) { (this.subscription1 = this.context.onFileDrop.subscribe(_eventHandler.bind(view,'onFileDrop'))); }
  }
}
