// primeng/components/galleria/galleria.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from '@angular/core/src/linker/ng_module_factory';
import * as import1 from 'primeng/components/galleria/galleria';
import * as import2 from '@angular/common/src/common_module';
import * as import3 from '@angular/common/src/localization';
import * as import4 from '@angular/core/src/di/injector';
import * as import5 from '@angular/core/src/i18n/tokens';
import * as import6 from '@angular/core/src/change_detection/change_detection_util';
import * as import7 from '@angular/core/src/linker/view';
import * as import8 from '@angular/core/src/linker/view_utils';
import * as import9 from '@angular/core/src/render/api';
import * as import10 from '@angular/core/src/metadata/view';
import * as import11 from 'primeng/components/dom/domhandler';
import * as import12 from '@angular/core/src/linker/view_type';
import * as import13 from '@angular/core/src/change_detection/constants';
import * as import14 from '@angular/core/src/linker/component_factory';
import * as import15 from '@angular/core/src/linker/element_ref';
import * as import16 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import17 from '../../../@angular/common/src/directives/ng_class.ngfactory';
import * as import18 from '@angular/core/src/linker/view_container';
import * as import19 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import20 from '@angular/common/src/directives/ng_class';
import * as import21 from '@angular/core/src/security';
import * as import22 from '../../../@angular/common/src/directives/ng_for.ngfactory';
import * as import23 from '@angular/core/src/linker/template_ref';
import * as import24 from '@angular/common/src/directives/ng_for';
import * as import25 from '../../../@angular/common/src/directives/ng_style.ngfactory';
import * as import26 from '../../../@angular/common/src/directives/ng_if.ngfactory';
import * as import27 from '@angular/common/src/directives/ng_if';
import * as import28 from '@angular/common/src/directives/ng_style';
class GalleriaModuleInjector extends import0.NgModuleInjector<import1.GalleriaModule> {
  _CommonModule_0:import2.CommonModule;
  _GalleriaModule_1:import1.GalleriaModule;
  __NgLocalization_2:import3.NgLocaleLocalization;
  constructor(parent:import4.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_2():import3.NgLocaleLocalization {
    if ((this.__NgLocalization_2 == null)) { (this.__NgLocalization_2 = new import3.NgLocaleLocalization(this.parent.get(import5.LOCALE_ID))); }
    return this.__NgLocalization_2;
  }
  createInternal():import1.GalleriaModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._GalleriaModule_1 = new import1.GalleriaModule();
    return this._GalleriaModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import1.GalleriaModule)) { return this._GalleriaModule_1; }
    if ((token === import3.NgLocalization)) { return this._NgLocalization_2; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const GalleriaModuleNgFactory:import0.NgModuleFactory<import1.GalleriaModule> = new import0.NgModuleFactory(GalleriaModuleInjector,import1.GalleriaModule);
export class Wrapper_Galleria {
  /*private*/ _eventHandler:Function;
  context:import1.Galleria;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  /*private*/ _expr_5:any;
  /*private*/ _expr_6:any;
  /*private*/ _expr_7:any;
  /*private*/ _expr_8:any;
  /*private*/ _expr_9:any;
  /*private*/ _expr_10:any;
  /*private*/ _expr_11:any;
  subscription0:any;
  constructor(p0:any,p1:any,p2:any) {
    this._changed = false;
    this.context = new import1.Galleria(p0,p1,p2);
    this._expr_0 = import6.UNINITIALIZED;
    this._expr_1 = import6.UNINITIALIZED;
    this._expr_2 = import6.UNINITIALIZED;
    this._expr_3 = import6.UNINITIALIZED;
    this._expr_4 = import6.UNINITIALIZED;
    this._expr_5 = import6.UNINITIALIZED;
    this._expr_6 = import6.UNINITIALIZED;
    this._expr_7 = import6.UNINITIALIZED;
    this._expr_8 = import6.UNINITIALIZED;
    this._expr_9 = import6.UNINITIALIZED;
    this._expr_10 = import6.UNINITIALIZED;
    this._expr_11 = import6.UNINITIALIZED;
  }
  ngOnDetach(view:import7.AppView<any>,componentView:import7.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
    this.context.ngOnDestroy();
    (this.subscription0 && this.subscription0.unsubscribe());
  }
  check_images(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.images = currValue;
      this._expr_0 = currValue;
    }
  }
  check_style(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.style = currValue;
      this._expr_1 = currValue;
    }
  }
  check_styleClass(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_2,currValue))) {
      this._changed = true;
      this.context.styleClass = currValue;
      this._expr_2 = currValue;
    }
  }
  check_panelWidth(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_3,currValue))) {
      this._changed = true;
      this.context.panelWidth = currValue;
      this._expr_3 = currValue;
    }
  }
  check_panelHeight(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_4,currValue))) {
      this._changed = true;
      this.context.panelHeight = currValue;
      this._expr_4 = currValue;
    }
  }
  check_frameWidth(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_5,currValue))) {
      this._changed = true;
      this.context.frameWidth = currValue;
      this._expr_5 = currValue;
    }
  }
  check_frameHeight(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_6,currValue))) {
      this._changed = true;
      this.context.frameHeight = currValue;
      this._expr_6 = currValue;
    }
  }
  check_activeIndex(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_7,currValue))) {
      this._changed = true;
      this.context.activeIndex = currValue;
      this._expr_7 = currValue;
    }
  }
  check_showFilmstrip(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_8,currValue))) {
      this._changed = true;
      this.context.showFilmstrip = currValue;
      this._expr_8 = currValue;
    }
  }
  check_autoPlay(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_9,currValue))) {
      this._changed = true;
      this.context.autoPlay = currValue;
      this._expr_9 = currValue;
    }
  }
  check_transitionInterval(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_10,currValue))) {
      this._changed = true;
      this.context.transitionInterval = currValue;
      this._expr_10 = currValue;
    }
  }
  check_showCaption(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_11,currValue))) {
      this._changed = true;
      this.context.showCaption = currValue;
      this._expr_11 = currValue;
    }
  }
  ngDoCheck(view:import7.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    if (!throwOnChange) { this.context.ngDoCheck(); }
    return changed;
  }
  checkHost(view:import7.AppView<any>,componentView:import7.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import7.AppView<any>,_eventHandler:any,emit0:boolean):void {
    this._eventHandler = _eventHandler;
    if (emit0) { (this.subscription0 = this.context.onImageClicked.subscribe(_eventHandler.bind(view,'onImageClicked'))); }
  }
}
var renderType_Galleria_Host:import9.RenderComponentType = import8.createRenderComponentType('',0,import10.ViewEncapsulation.None,([] as any[]),{});
class View_Galleria_Host0 extends import7.AppView<any> {
  _el_0:any;
  compView_0:import7.AppView<import1.Galleria>;
  _DomHandler_0_3:import11.DomHandler;
  _Galleria_0_4:Wrapper_Galleria;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_Galleria_Host0,renderType_Galleria_Host,import12.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import13.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import14.ComponentRef<any> {
    this._el_0 = import8.selectOrCreateRenderHostElement(this.renderer,'p-galleria',import8.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this.compView_0 = new View_Galleria0(this.viewUtils,this,0,this._el_0);
    this._DomHandler_0_3 = new import11.DomHandler();
    this._Galleria_0_4 = new Wrapper_Galleria(new import15.ElementRef(this._el_0),this._DomHandler_0_3,this.injectorGet(import16.IterableDiffers,this.parentIndex));
    this.compView_0.create(this._Galleria_0_4.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import14.ComponentRef_<any>(0,this,this._el_0,this._Galleria_0_4.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import11.DomHandler) && (0 === requestNodeIndex))) { return this._DomHandler_0_3; }
    if (((token === import1.Galleria) && (0 === requestNodeIndex))) { return this._Galleria_0_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._Galleria_0_4.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
    if (!throwOnChange) {
      if ((this.numberOfChecks === 0)) { this._Galleria_0_4.context.ngAfterViewInit(); }
      this._Galleria_0_4.context.ngAfterViewChecked();
    }
  }
  destroyInternal():void {
    this.compView_0.destroy();
    this._Galleria_0_4.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
export const GalleriaNgFactory:import14.ComponentFactory<import1.Galleria> = new import14.ComponentFactory<import1.Galleria>('p-galleria',View_Galleria_Host0,import1.Galleria);
const styles_Galleria:any[] = ([] as any[]);
class View_Galleria1 extends import7.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import17.Wrapper_NgClass;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  /*private*/ _expr_5:any;
  /*private*/ _expr_6:any;
  _map_7:any;
  /*private*/ _expr_8:any;
  /*private*/ _expr_9:any;
  /*private*/ _expr_10:any;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import18.ViewContainer) {
    super(View_Galleria1,renderType_Galleria,import12.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import13.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_5 = import6.UNINITIALIZED;
    this._expr_6 = import6.UNINITIALIZED;
    this._map_7 = import8.pureProxy1((p0:any):{[key: string]:any} => {
      return {'ui-helper-hidden': p0};
    });
    this._expr_8 = import6.UNINITIALIZED;
    this._expr_9 = import6.UNINITIALIZED;
    this._expr_10 = import6.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import14.ComponentRef<any> {
    this._el_0 = import8.createRenderElement(this.renderer,(null as any),'li',new import8.InlineArray2(2,'class','ui-galleria-panel'),(null as any));
    this._NgClass_0_3 = new import17.Wrapper_NgClass(this.parentView.parentView.injectorGet(import16.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import19.KeyValueDiffers,this.parentView.parentIndex),new import15.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this._el_2 = import8.createRenderElement(this.renderer,this._el_0,'img',new import8.InlineArray2(2,'class','ui-panel-images'),(null as any));
    this._text_3 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    var disposable_0:Function = import8.subscribeToRenderElement(this,this._el_0,new import8.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import20.NgClass) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 3)))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = 'ui-galleria-panel';
    this._NgClass_0_3.check_klass(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = this._map_7((this.context.index != this.parentView.context.activeIndex));
    this._NgClass_0_3.check_ngClass(currVal_0_0_1,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_5:any = this.parentView.context.panelWidth;
    if (import8.checkBinding(throwOnChange,this._expr_5,currVal_5)) {
      this.renderer.setElementStyle(this._el_0,'width',((this.viewUtils.sanitizer.sanitize(import21.SecurityContext.STYLE,currVal_5) == null)? (null as any): (this.viewUtils.sanitizer.sanitize(import21.SecurityContext.STYLE,currVal_5).toString() + 'px')));
      this._expr_5 = currVal_5;
    }
    const currVal_6:any = this.parentView.context.panelHeight;
    if (import8.checkBinding(throwOnChange,this._expr_6,currVal_6)) {
      this.renderer.setElementStyle(this._el_0,'height',((this.viewUtils.sanitizer.sanitize(import21.SecurityContext.STYLE,currVal_6) == null)? (null as any): (this.viewUtils.sanitizer.sanitize(import21.SecurityContext.STYLE,currVal_6).toString() + 'px')));
      this._expr_6 = currVal_6;
    }
    const currVal_8:any = this.context.$implicit.source;
    if (import8.checkBinding(throwOnChange,this._expr_8,currVal_8)) {
      this.renderer.setElementProperty(this._el_2,'src',this.viewUtils.sanitizer.sanitize(import21.SecurityContext.URL,currVal_8));
      this._expr_8 = currVal_8;
    }
    const currVal_9:any = this.context.$implicit.alt;
    if (import8.checkBinding(throwOnChange,this._expr_9,currVal_9)) {
      this.renderer.setElementProperty(this._el_2,'alt',currVal_9);
      this._expr_9 = currVal_9;
    }
    const currVal_10:any = this.context.$implicit.title;
    if (import8.checkBinding(throwOnChange,this._expr_10,currVal_10)) {
      this.renderer.setElementProperty(this._el_2,'title',currVal_10);
      this._expr_10 = currVal_10;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.context.clickImage($event,this.context.$implicit,this.context.index)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
class View_Galleria3 extends import7.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import17.Wrapper_NgClass;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _el_4:any;
  _text_5:any;
  _text_6:any;
  /*private*/ _expr_8:any;
  /*private*/ _expr_9:any;
  /*private*/ _expr_10:any;
  _map_11:any;
  /*private*/ _expr_12:any;
  /*private*/ _expr_13:any;
  /*private*/ _expr_14:any;
  /*private*/ _expr_15:any;
  /*private*/ _expr_16:any;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import18.ViewContainer) {
    super(View_Galleria3,renderType_Galleria,import12.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import13.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_8 = import6.UNINITIALIZED;
    this._expr_9 = import6.UNINITIALIZED;
    this._expr_10 = import6.UNINITIALIZED;
    this._map_11 = import8.pureProxy1((p0:any):{[key: string]:any} => {
      return {'ui-galleria-frame-active': p0};
    });
    this._expr_12 = import6.UNINITIALIZED;
    this._expr_13 = import6.UNINITIALIZED;
    this._expr_14 = import6.UNINITIALIZED;
    this._expr_15 = import6.UNINITIALIZED;
    this._expr_16 = import6.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import14.ComponentRef<any> {
    this._el_0 = import8.createRenderElement(this.renderer,(null as any),'li',new import8.InlineArray2(2,'class','ui-galleria-frame'),(null as any));
    this._NgClass_0_3 = new import17.Wrapper_NgClass(this.parentView.parentView.parentView.injectorGet(import16.IterableDiffers,this.parentView.parentView.parentIndex),this.parentView.parentView.parentView.injectorGet(import19.KeyValueDiffers,this.parentView.parentView.parentIndex),new import15.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n                        ',(null as any));
    this._el_2 = import8.createRenderElement(this.renderer,this._el_0,'div',new import8.InlineArray2(2,'class','ui-galleria-frame-content'),(null as any));
    this._text_3 = this.renderer.createText(this._el_2,'\n                            ',(null as any));
    this._el_4 = import8.createRenderElement(this.renderer,this._el_2,'img',new import8.InlineArray2(2,'class','ui-galleria-frame-image'),(null as any));
    this._text_5 = this.renderer.createText(this._el_2,'\n                        ',(null as any));
    this._text_6 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    var disposable_0:Function = import8.subscribeToRenderElement(this,this._el_0,new import8.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._el_4,
      this._text_5,
      this._text_6
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import20.NgClass) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 6)))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = 'ui-galleria-frame';
    this._NgClass_0_3.check_klass(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = this._map_11((this.context.index == this.parentView.parentView.context.activeIndex));
    this._NgClass_0_3.check_ngClass(currVal_0_0_1,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_8:any = this.parentView.parentView.context.frameWidth;
    if (import8.checkBinding(throwOnChange,this._expr_8,currVal_8)) {
      this.renderer.setElementStyle(this._el_0,'width',((this.viewUtils.sanitizer.sanitize(import21.SecurityContext.STYLE,currVal_8) == null)? (null as any): (this.viewUtils.sanitizer.sanitize(import21.SecurityContext.STYLE,currVal_8).toString() + 'px')));
      this._expr_8 = currVal_8;
    }
    const currVal_9:any = this.parentView.parentView.context.frameHeight;
    if (import8.checkBinding(throwOnChange,this._expr_9,currVal_9)) {
      this.renderer.setElementStyle(this._el_0,'height',((this.viewUtils.sanitizer.sanitize(import21.SecurityContext.STYLE,currVal_9) == null)? (null as any): (this.viewUtils.sanitizer.sanitize(import21.SecurityContext.STYLE,currVal_9).toString() + 'px')));
      this._expr_9 = currVal_9;
    }
    const currVal_10:any = 'opacity 0.75s ease';
    if (import8.checkBinding(throwOnChange,this._expr_10,currVal_10)) {
      this.renderer.setElementStyle(this._el_0,'transition',((this.viewUtils.sanitizer.sanitize(import21.SecurityContext.STYLE,currVal_10) == null)? (null as any): this.viewUtils.sanitizer.sanitize(import21.SecurityContext.STYLE,currVal_10).toString()));
      this._expr_10 = currVal_10;
    }
    const currVal_12:any = this.context.$implicit.source;
    if (import8.checkBinding(throwOnChange,this._expr_12,currVal_12)) {
      this.renderer.setElementProperty(this._el_4,'src',this.viewUtils.sanitizer.sanitize(import21.SecurityContext.URL,currVal_12));
      this._expr_12 = currVal_12;
    }
    const currVal_13:any = this.context.$implicit.alt;
    if (import8.checkBinding(throwOnChange,this._expr_13,currVal_13)) {
      this.renderer.setElementProperty(this._el_4,'alt',currVal_13);
      this._expr_13 = currVal_13;
    }
    const currVal_14:any = this.context.$implicit.title;
    if (import8.checkBinding(throwOnChange,this._expr_14,currVal_14)) {
      this.renderer.setElementProperty(this._el_4,'title',currVal_14);
      this._expr_14 = currVal_14;
    }
    const currVal_15:any = this.parentView.parentView.context.frameWidth;
    if (import8.checkBinding(throwOnChange,this._expr_15,currVal_15)) {
      this.renderer.setElementStyle(this._el_4,'width',((this.viewUtils.sanitizer.sanitize(import21.SecurityContext.STYLE,currVal_15) == null)? (null as any): (this.viewUtils.sanitizer.sanitize(import21.SecurityContext.STYLE,currVal_15).toString() + 'px')));
      this._expr_15 = currVal_15;
    }
    const currVal_16:any = this.parentView.parentView.context.frameHeight;
    if (import8.checkBinding(throwOnChange,this._expr_16,currVal_16)) {
      this.renderer.setElementStyle(this._el_4,'height',((this.viewUtils.sanitizer.sanitize(import21.SecurityContext.STYLE,currVal_16) == null)? (null as any): (this.viewUtils.sanitizer.sanitize(import21.SecurityContext.STYLE,currVal_16).toString() + 'px')));
      this._expr_16 = currVal_16;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.parentView.context.frameClick(this._el_0)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
class View_Galleria2 extends import7.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import17.Wrapper_NgClass;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _anchor_4:any;
  /*private*/ _vc_4:import18.ViewContainer;
  _TemplateRef_4_5:any;
  _NgFor_4_6:import22.Wrapper_NgFor;
  _text_5:any;
  _text_6:any;
  _map_11:any;
  /*private*/ _expr_12:any;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import18.ViewContainer) {
    super(View_Galleria2,renderType_Galleria,import12.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import13.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_11 = import8.pureProxy1((p0:any):{[key: string]:any} => {
      return {'ui-galleria-filmstrip-wrapper': p0};
    });
    this._expr_12 = import6.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import14.ComponentRef<any> {
    this._el_0 = import8.createRenderElement(this.renderer,(null as any),'div',import8.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_0_3 = new import17.Wrapper_NgClass(this.parentView.parentView.injectorGet(import16.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import19.KeyValueDiffers,this.parentView.parentIndex),new import15.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    this._el_2 = import8.createRenderElement(this.renderer,this._el_0,'ul',new import8.InlineArray4(4,'class','ui-galleria-filmstrip','style','transition:left 1s'),(null as any));
    this._text_3 = this.renderer.createText(this._el_2,'\n                    ',(null as any));
    this._anchor_4 = this.renderer.createTemplateAnchor(this._el_2,(null as any));
    this._vc_4 = new import18.ViewContainer(4,2,this,this._anchor_4);
    this._TemplateRef_4_5 = new import23.TemplateRef_(this,4,this._anchor_4);
    this._NgFor_4_6 = new import22.Wrapper_NgFor(this._vc_4.vcRef,this._TemplateRef_4_5,this.parentView.parentView.injectorGet(import16.IterableDiffers,this.parentView.parentIndex),this.parentView.ref);
    this._text_5 = this.renderer.createText(this._el_2,'\n                ',(null as any));
    this._text_6 = this.renderer.createText(this._el_0,'\n            ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._anchor_4,
      this._text_5,
      this._text_6
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import23.TemplateRef) && (4 === requestNodeIndex))) { return this._TemplateRef_4_5; }
    if (((token === import24.NgFor) && (4 === requestNodeIndex))) { return this._NgFor_4_6.context; }
    if (((token === import20.NgClass) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 6)))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this._map_11(true);
    this._NgClass_0_3.check_ngClass(currVal_0_0_0,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_4_0_0:any = this.parentView.context.images;
    this._NgFor_4_6.check_ngForOf(currVal_4_0_0,throwOnChange,false);
    this._NgFor_4_6.ngDoCheck(this,this._anchor_4,throwOnChange);
    this._vc_4.detectChangesInNestedViews(throwOnChange);
    const currVal_12:any = this.parentView.context.stripLeft;
    if (import8.checkBinding(throwOnChange,this._expr_12,currVal_12)) {
      this.renderer.setElementStyle(this._el_2,'left',((this.viewUtils.sanitizer.sanitize(import21.SecurityContext.STYLE,currVal_12) == null)? (null as any): (this.viewUtils.sanitizer.sanitize(import21.SecurityContext.STYLE,currVal_12).toString() + 'px')));
      this._expr_12 = currVal_12;
    }
  }
  destroyInternal():void {
    this._vc_4.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import7.AppView<any> {
    if ((nodeIndex == 4)) { return new View_Galleria3(this.viewUtils,this,4,this._anchor_4,this._vc_4); }
    return (null as any);
  }
}
class View_Galleria4 extends import7.AppView<any> {
  _el_0:any;
  /*private*/ _expr_1:any;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import18.ViewContainer) {
    super(View_Galleria4,renderType_Galleria,import12.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import13.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_1 = import6.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import14.ComponentRef<any> {
    this._el_0 = import8.createRenderElement(this.renderer,(null as any),'div',new import8.InlineArray2(2,'class','ui-galleria-nav-prev fa fa-fw fa-chevron-circle-left'),(null as any));
    var disposable_0:Function = import8.subscribeToRenderElement(this,this._el_0,new import8.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),[disposable_0]);
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1:any = (this.parentView.context.frameHeight / 2);
    if (import8.checkBinding(throwOnChange,this._expr_1,currVal_1)) {
      this.renderer.setElementStyle(this._el_0,'bottom',((this.viewUtils.sanitizer.sanitize(import21.SecurityContext.STYLE,currVal_1) == null)? (null as any): (this.viewUtils.sanitizer.sanitize(import21.SecurityContext.STYLE,currVal_1).toString() + 'px')));
      this._expr_1 = currVal_1;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.context.clickNavLeft()) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
class View_Galleria5 extends import7.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _el_4:any;
  _text_5:any;
  _text_6:any;
  /*private*/ _expr_7:any;
  /*private*/ _expr_8:any;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import18.ViewContainer) {
    super(View_Galleria5,renderType_Galleria,import12.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import13.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_7 = import6.UNINITIALIZED;
    this._expr_8 = import6.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import14.ComponentRef<any> {
    this._el_0 = import8.createRenderElement(this.renderer,(null as any),'div',new import8.InlineArray4(4,'class','ui-galleria-caption','style','display:block'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    this._el_2 = import8.createRenderElement(this.renderer,this._el_0,'h4',import8.EMPTY_INLINE_ARRAY,(null as any));
    this._text_3 = this.renderer.createText(this._el_2,'',(null as any));
    this._el_4 = import8.createRenderElement(this.renderer,this._el_0,'p',import8.EMPTY_INLINE_ARRAY,(null as any));
    this._text_5 = this.renderer.createText(this._el_4,'',(null as any));
    this._text_6 = this.renderer.createText(this._el_0,'\n            ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._el_4,
      this._text_5,
      this._text_6
    ]
    ),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_7:any = import8.inlineInterpolate(1,'',((this.parentView.context.images[this.parentView.context.activeIndex] == null)? (null as any): this.parentView.context.images[this.parentView.context.activeIndex].title),'');
    if (import8.checkBinding(throwOnChange,this._expr_7,currVal_7)) {
      this.renderer.setText(this._text_3,currVal_7);
      this._expr_7 = currVal_7;
    }
    const currVal_8:any = import8.inlineInterpolate(1,'',((this.parentView.context.images[this.parentView.context.activeIndex] == null)? (null as any): this.parentView.context.images[this.parentView.context.activeIndex].alt),'');
    if (import8.checkBinding(throwOnChange,this._expr_8,currVal_8)) {
      this.renderer.setText(this._text_5,currVal_8);
      this._expr_8 = currVal_8;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
var renderType_Galleria:import9.RenderComponentType = import8.createRenderComponentType('',0,import10.ViewEncapsulation.None,styles_Galleria,{});
export class View_Galleria0 extends import7.AppView<import1.Galleria> {
  _text_0:any;
  _el_1:any;
  _NgClass_1_3:import17.Wrapper_NgClass;
  _NgStyle_1_4:import25.Wrapper_NgStyle;
  _text_2:any;
  _el_3:any;
  _text_4:any;
  _anchor_5:any;
  /*private*/ _vc_5:import18.ViewContainer;
  _TemplateRef_5_5:any;
  _NgFor_5_6:import22.Wrapper_NgFor;
  _text_6:any;
  _text_7:any;
  _anchor_8:any;
  /*private*/ _vc_8:import18.ViewContainer;
  _TemplateRef_8_5:any;
  _NgIf_8_6:import26.Wrapper_NgIf;
  _text_9:any;
  _anchor_10:any;
  /*private*/ _vc_10:import18.ViewContainer;
  _TemplateRef_10_5:any;
  _NgIf_10_6:import26.Wrapper_NgIf;
  _text_11:any;
  _el_12:any;
  _text_13:any;
  _anchor_14:any;
  /*private*/ _vc_14:import18.ViewContainer;
  _TemplateRef_14_5:any;
  _NgIf_14_6:import26.Wrapper_NgIf;
  _text_15:any;
  _text_16:any;
  /*private*/ _expr_31:any;
  _map_32:any;
  /*private*/ _expr_33:any;
  /*private*/ _expr_34:any;
  /*private*/ _expr_35:any;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_Galleria0,renderType_Galleria,import12.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import13.ChangeDetectorStatus.CheckAlways);
    this._expr_31 = import6.UNINITIALIZED;
    this._map_32 = import8.pureProxy1((p0:any):{[key: string]:any} => {
      return {'ui-galleria ui-widget ui-widget-content ui-corner-all': p0};
    });
    this._expr_33 = import6.UNINITIALIZED;
    this._expr_34 = import6.UNINITIALIZED;
    this._expr_35 = import6.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import14.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n        ',(null as any));
    this._el_1 = import8.createRenderElement(this.renderer,parentRenderNode,'div',import8.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_1_3 = new import17.Wrapper_NgClass(this.parentView.injectorGet(import16.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import19.KeyValueDiffers,this.parentIndex),new import15.ElementRef(this._el_1),this.renderer);
    this._NgStyle_1_4 = new import25.Wrapper_NgStyle(this.parentView.injectorGet(import19.KeyValueDiffers,this.parentIndex),new import15.ElementRef(this._el_1),this.renderer);
    this._text_2 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_3 = import8.createRenderElement(this.renderer,this._el_1,'ul',new import8.InlineArray2(2,'class','ui-galleria-panel-wrapper'),(null as any));
    this._text_4 = this.renderer.createText(this._el_3,'\n                ',(null as any));
    this._anchor_5 = this.renderer.createTemplateAnchor(this._el_3,(null as any));
    this._vc_5 = new import18.ViewContainer(5,3,this,this._anchor_5);
    this._TemplateRef_5_5 = new import23.TemplateRef_(this,5,this._anchor_5);
    this._NgFor_5_6 = new import22.Wrapper_NgFor(this._vc_5.vcRef,this._TemplateRef_5_5,this.parentView.injectorGet(import16.IterableDiffers,this.parentIndex),this.ref);
    this._text_6 = this.renderer.createText(this._el_3,'\n            ',(null as any));
    this._text_7 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._anchor_8 = this.renderer.createTemplateAnchor(this._el_1,(null as any));
    this._vc_8 = new import18.ViewContainer(8,1,this,this._anchor_8);
    this._TemplateRef_8_5 = new import23.TemplateRef_(this,8,this._anchor_8);
    this._NgIf_8_6 = new import26.Wrapper_NgIf(this._vc_8.vcRef,this._TemplateRef_8_5);
    this._text_9 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._anchor_10 = this.renderer.createTemplateAnchor(this._el_1,(null as any));
    this._vc_10 = new import18.ViewContainer(10,1,this,this._anchor_10);
    this._TemplateRef_10_5 = new import23.TemplateRef_(this,10,this._anchor_10);
    this._NgIf_10_6 = new import26.Wrapper_NgIf(this._vc_10.vcRef,this._TemplateRef_10_5);
    this._text_11 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_12 = import8.createRenderElement(this.renderer,this._el_1,'div',new import8.InlineArray2(2,'class','ui-galleria-nav-next fa fa-fw fa-chevron-circle-right'),(null as any));
    this._text_13 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._anchor_14 = this.renderer.createTemplateAnchor(this._el_1,(null as any));
    this._vc_14 = new import18.ViewContainer(14,1,this,this._anchor_14);
    this._TemplateRef_14_5 = new import23.TemplateRef_(this,14,this._anchor_14);
    this._NgIf_14_6 = new import26.Wrapper_NgIf(this._vc_14.vcRef,this._TemplateRef_14_5);
    this._text_15 = this.renderer.createText(this._el_1,'\n        ',(null as any));
    this._text_16 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    var disposable_0:Function = import8.subscribeToRenderElement(this,this._el_12,new import8.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_12));
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2,
      this._el_3,
      this._text_4,
      this._anchor_5,
      this._text_6,
      this._text_7,
      this._anchor_8,
      this._text_9,
      this._anchor_10,
      this._text_11,
      this._el_12,
      this._text_13,
      this._anchor_14,
      this._text_15,
      this._text_16
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import23.TemplateRef) && (5 === requestNodeIndex))) { return this._TemplateRef_5_5; }
    if (((token === import24.NgFor) && (5 === requestNodeIndex))) { return this._NgFor_5_6.context; }
    if (((token === import23.TemplateRef) && (8 === requestNodeIndex))) { return this._TemplateRef_8_5; }
    if (((token === import27.NgIf) && (8 === requestNodeIndex))) { return this._NgIf_8_6.context; }
    if (((token === import23.TemplateRef) && (10 === requestNodeIndex))) { return this._TemplateRef_10_5; }
    if (((token === import27.NgIf) && (10 === requestNodeIndex))) { return this._NgIf_10_6.context; }
    if (((token === import23.TemplateRef) && (14 === requestNodeIndex))) { return this._TemplateRef_14_5; }
    if (((token === import27.NgIf) && (14 === requestNodeIndex))) { return this._NgIf_14_6.context; }
    if (((token === import20.NgClass) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 15)))) { return this._NgClass_1_3.context; }
    if (((token === import28.NgStyle) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 15)))) { return this._NgStyle_1_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1_0_0:any = this.context.styleClass;
    this._NgClass_1_3.check_klass(currVal_1_0_0,throwOnChange,false);
    const currVal_1_0_1:any = this._map_32(true);
    this._NgClass_1_3.check_ngClass(currVal_1_0_1,throwOnChange,false);
    this._NgClass_1_3.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_1_1_0:any = this.context.style;
    this._NgStyle_1_4.check_ngStyle(currVal_1_1_0,throwOnChange,false);
    this._NgStyle_1_4.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_5_0_0:any = this.context.images;
    this._NgFor_5_6.check_ngForOf(currVal_5_0_0,throwOnChange,false);
    this._NgFor_5_6.ngDoCheck(this,this._anchor_5,throwOnChange);
    const currVal_8_0_0:any = this.context.showFilmstrip;
    this._NgIf_8_6.check_ngIf(currVal_8_0_0,throwOnChange,false);
    this._NgIf_8_6.ngDoCheck(this,this._anchor_8,throwOnChange);
    const currVal_10_0_0:any = (this.context.activeIndex !== 0);
    this._NgIf_10_6.check_ngIf(currVal_10_0_0,throwOnChange,false);
    this._NgIf_10_6.ngDoCheck(this,this._anchor_10,throwOnChange);
    const currVal_14_0_0:any = (this.context.showCaption && this.context.images);
    this._NgIf_14_6.check_ngIf(currVal_14_0_0,throwOnChange,false);
    this._NgIf_14_6.ngDoCheck(this,this._anchor_14,throwOnChange);
    this._vc_5.detectChangesInNestedViews(throwOnChange);
    this._vc_8.detectChangesInNestedViews(throwOnChange);
    this._vc_10.detectChangesInNestedViews(throwOnChange);
    this._vc_14.detectChangesInNestedViews(throwOnChange);
    const currVal_31:any = this.context.panelWidth;
    if (import8.checkBinding(throwOnChange,this._expr_31,currVal_31)) {
      this.renderer.setElementStyle(this._el_1,'width',((this.viewUtils.sanitizer.sanitize(import21.SecurityContext.STYLE,currVal_31) == null)? (null as any): (this.viewUtils.sanitizer.sanitize(import21.SecurityContext.STYLE,currVal_31).toString() + 'px')));
      this._expr_31 = currVal_31;
    }
    const currVal_33:any = this.context.panelWidth;
    if (import8.checkBinding(throwOnChange,this._expr_33,currVal_33)) {
      this.renderer.setElementStyle(this._el_3,'width',((this.viewUtils.sanitizer.sanitize(import21.SecurityContext.STYLE,currVal_33) == null)? (null as any): (this.viewUtils.sanitizer.sanitize(import21.SecurityContext.STYLE,currVal_33).toString() + 'px')));
      this._expr_33 = currVal_33;
    }
    const currVal_34:any = this.context.panelHeight;
    if (import8.checkBinding(throwOnChange,this._expr_34,currVal_34)) {
      this.renderer.setElementStyle(this._el_3,'height',((this.viewUtils.sanitizer.sanitize(import21.SecurityContext.STYLE,currVal_34) == null)? (null as any): (this.viewUtils.sanitizer.sanitize(import21.SecurityContext.STYLE,currVal_34).toString() + 'px')));
      this._expr_34 = currVal_34;
    }
    const currVal_35:any = (this.context.frameHeight / 2);
    if (import8.checkBinding(throwOnChange,this._expr_35,currVal_35)) {
      this.renderer.setElementStyle(this._el_12,'bottom',((this.viewUtils.sanitizer.sanitize(import21.SecurityContext.STYLE,currVal_35) == null)? (null as any): (this.viewUtils.sanitizer.sanitize(import21.SecurityContext.STYLE,currVal_35).toString() + 'px')));
      this._expr_35 = currVal_35;
    }
  }
  destroyInternal():void {
    this._vc_5.destroyNestedViews();
    this._vc_8.destroyNestedViews();
    this._vc_10.destroyNestedViews();
    this._vc_14.destroyNestedViews();
  }
  createEmbeddedViewInternal(nodeIndex:number):import7.AppView<any> {
    if ((nodeIndex == 5)) { return new View_Galleria1(this.viewUtils,this,5,this._anchor_5,this._vc_5); }
    if ((nodeIndex == 8)) { return new View_Galleria2(this.viewUtils,this,8,this._anchor_8,this._vc_8); }
    if ((nodeIndex == 10)) { return new View_Galleria4(this.viewUtils,this,10,this._anchor_10,this._vc_10); }
    if ((nodeIndex == 14)) { return new View_Galleria5(this.viewUtils,this,14,this._anchor_14,this._vc_14); }
    return (null as any);
  }
  handleEvent_12(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.context.clickNavRight()) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
