// primeng/components/splitbutton/splitbutton.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from '@angular/core/src/linker/ng_module_factory';
import * as import1 from 'primeng/components/splitbutton/splitbutton';
import * as import2 from '@angular/common/src/common_module';
import * as import3 from 'primeng/components/button/button';
import * as import4 from '@angular/common/src/localization';
import * as import5 from '@angular/core/src/di/injector';
import * as import6 from '@angular/core/src/i18n/tokens';
import * as import7 from '@angular/core/src/change_detection/change_detection_util';
import * as import8 from '@angular/core/src/linker/view';
import * as import9 from '@angular/core/src/linker/view_utils';
import * as import10 from '@angular/core/src/render/api';
import * as import11 from '@angular/core/src/metadata/view';
import * as import12 from 'primeng/components/dom/domhandler';
import * as import13 from '@angular/core/src/linker/view_type';
import * as import14 from '@angular/core/src/change_detection/constants';
import * as import15 from '@angular/core/src/linker/component_factory';
import * as import16 from '@angular/core/src/linker/element_ref';
import * as import17 from '@angular/router/src/router';
import * as import18 from '../../../@angular/common/src/directives/ng_class.ngfactory';
import * as import19 from '@angular/core/src/linker/view_container';
import * as import20 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import21 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import22 from '@angular/common/src/directives/ng_class';
import * as import23 from '../../../@angular/common/src/directives/ng_if.ngfactory';
import * as import24 from '@angular/core/src/linker/template_ref';
import * as import25 from '@angular/common/src/directives/ng_if';
import * as import26 from '@angular/core/src/security';
import * as import27 from '../../../@angular/common/src/directives/ng_style.ngfactory';
import * as import28 from '../button/button.ngfactory';
import * as import29 from '../../../@angular/common/src/directives/ng_for.ngfactory';
import * as import30 from '@angular/common/src/directives/ng_for';
import * as import31 from '@angular/common/src/directives/ng_style';
class SplitButtonModuleInjector extends import0.NgModuleInjector<import1.SplitButtonModule> {
  _CommonModule_0:import2.CommonModule;
  _ButtonModule_1:import3.ButtonModule;
  _SplitButtonModule_2:import1.SplitButtonModule;
  __NgLocalization_3:import4.NgLocaleLocalization;
  constructor(parent:import5.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_3():import4.NgLocaleLocalization {
    if ((this.__NgLocalization_3 == null)) { (this.__NgLocalization_3 = new import4.NgLocaleLocalization(this.parent.get(import6.LOCALE_ID))); }
    return this.__NgLocalization_3;
  }
  createInternal():import1.SplitButtonModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._ButtonModule_1 = new import3.ButtonModule();
    this._SplitButtonModule_2 = new import1.SplitButtonModule();
    return this._SplitButtonModule_2;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.ButtonModule)) { return this._ButtonModule_1; }
    if ((token === import1.SplitButtonModule)) { return this._SplitButtonModule_2; }
    if ((token === import4.NgLocalization)) { return this._NgLocalization_3; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const SplitButtonModuleNgFactory:import0.NgModuleFactory<import1.SplitButtonModule> = new import0.NgModuleFactory(SplitButtonModuleInjector,import1.SplitButtonModule);
export class Wrapper_SplitButton {
  /*private*/ _eventHandler:Function;
  context:import1.SplitButton;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  /*private*/ _expr_5:any;
  /*private*/ _expr_6:any;
  /*private*/ _expr_7:any;
  /*private*/ _expr_8:any;
  subscription0:any;
  constructor(p0:any,p1:any,p2:any,p3:any) {
    this._changed = false;
    this.context = new import1.SplitButton(p0,p1,p2,p3);
    this._expr_0 = import7.UNINITIALIZED;
    this._expr_1 = import7.UNINITIALIZED;
    this._expr_2 = import7.UNINITIALIZED;
    this._expr_3 = import7.UNINITIALIZED;
    this._expr_4 = import7.UNINITIALIZED;
    this._expr_5 = import7.UNINITIALIZED;
    this._expr_6 = import7.UNINITIALIZED;
    this._expr_7 = import7.UNINITIALIZED;
    this._expr_8 = import7.UNINITIALIZED;
  }
  ngOnDetach(view:import8.AppView<any>,componentView:import8.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
    this.context.ngOnDestroy();
    (this.subscription0 && this.subscription0.unsubscribe());
  }
  check_model(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.model = currValue;
      this._expr_0 = currValue;
    }
  }
  check_icon(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.icon = currValue;
      this._expr_1 = currValue;
    }
  }
  check_iconPos(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_2,currValue))) {
      this._changed = true;
      this.context.iconPos = currValue;
      this._expr_2 = currValue;
    }
  }
  check_label(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_3,currValue))) {
      this._changed = true;
      this.context.label = currValue;
      this._expr_3 = currValue;
    }
  }
  check_style(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_4,currValue))) {
      this._changed = true;
      this.context.style = currValue;
      this._expr_4 = currValue;
    }
  }
  check_styleClass(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_5,currValue))) {
      this._changed = true;
      this.context.styleClass = currValue;
      this._expr_5 = currValue;
    }
  }
  check_menuStyle(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_6,currValue))) {
      this._changed = true;
      this.context.menuStyle = currValue;
      this._expr_6 = currValue;
    }
  }
  check_menuStyleClass(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_7,currValue))) {
      this._changed = true;
      this.context.menuStyleClass = currValue;
      this._expr_7 = currValue;
    }
  }
  check_disabled(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_8,currValue))) {
      this._changed = true;
      this.context.disabled = currValue;
      this._expr_8 = currValue;
    }
  }
  ngDoCheck(view:import8.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    if (!throwOnChange) { if ((view.numberOfChecks === 0)) { this.context.ngOnInit(); } }
    return changed;
  }
  checkHost(view:import8.AppView<any>,componentView:import8.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import8.AppView<any>,_eventHandler:any,emit0:boolean):void {
    this._eventHandler = _eventHandler;
    if (emit0) { (this.subscription0 = this.context.onClick.subscribe(_eventHandler.bind(view,'onClick'))); }
  }
}
var renderType_SplitButton_Host:import10.RenderComponentType = import9.createRenderComponentType('',0,import11.ViewEncapsulation.None,([] as any[]),{});
class View_SplitButton_Host0 extends import8.AppView<any> {
  _el_0:any;
  compView_0:import8.AppView<import1.SplitButton>;
  _DomHandler_0_3:import12.DomHandler;
  _SplitButton_0_4:Wrapper_SplitButton;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_SplitButton_Host0,renderType_SplitButton_Host,import13.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.selectOrCreateRenderHostElement(this.renderer,'p-splitButton',import9.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this.compView_0 = new View_SplitButton0(this.viewUtils,this,0,this._el_0);
    this._DomHandler_0_3 = new import12.DomHandler();
    this._SplitButton_0_4 = new Wrapper_SplitButton(new import16.ElementRef(this._el_0),this._DomHandler_0_3,this.renderer,this.injectorGet(import17.Router,this.parentIndex));
    this.compView_0.create(this._SplitButton_0_4.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import15.ComponentRef_<any>(0,this,this._el_0,this._SplitButton_0_4.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import12.DomHandler) && (0 === requestNodeIndex))) { return this._DomHandler_0_3; }
    if (((token === import1.SplitButton) && (0 === requestNodeIndex))) { return this._SplitButton_0_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._SplitButton_0_4.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
    this._SplitButton_0_4.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
export const SplitButtonNgFactory:import15.ComponentFactory<import1.SplitButton> = new import15.ComponentFactory<import1.SplitButton>('p-splitButton',View_SplitButton_Host0,import1.SplitButton);
const styles_SplitButton:any[] = ([] as any[]);
class View_SplitButton2 extends import8.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import18.Wrapper_NgClass;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import19.ViewContainer) {
    super(View_SplitButton2,renderType_SplitButton,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'span',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_0_3 = new import18.Wrapper_NgClass(this.parentView.parentView.parentView.injectorGet(import20.IterableDiffers,this.parentView.parentView.parentIndex),this.parentView.parentView.parentView.injectorGet(import21.KeyValueDiffers,this.parentView.parentView.parentIndex),new import16.ElementRef(this._el_0),this.renderer);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import22.NgClass) && (0 === requestNodeIndex))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this.parentView.context.$implicit.icon;
    this._NgClass_0_3.check_klass(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = 'ui-menuitem-icon fa fa-fw';
    this._NgClass_0_3.check_ngClass(currVal_0_0_1,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_SplitButton1 extends import8.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _NgClass_2_3:import18.Wrapper_NgClass;
  _text_3:any;
  _anchor_4:any;
  /*private*/ _vc_4:import19.ViewContainer;
  _TemplateRef_4_5:any;
  _NgIf_4_6:import23.Wrapper_NgIf;
  _text_5:any;
  _el_6:any;
  _text_7:any;
  _text_8:any;
  _text_9:any;
  /*private*/ _expr_14:any;
  _map_15:any;
  /*private*/ _expr_16:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import19.ViewContainer) {
    super(View_SplitButton1,renderType_SplitButton,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_14 = import7.UNINITIALIZED;
    this._map_15 = import9.pureProxy3((p0:any,p1:any,p2:any):{[key: string]:any} => {
      return {
        'ui-menuitem-link ui-corner-all': p0,
        'ui-state-hover': p1,
        'ui-state-disabled': p2
      }
      ;
    });
    this._expr_16 = import7.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'li',new import9.InlineArray4(4,'class','ui-menuitem ui-widget ui-corner-all','role','menuitem'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                        ',(null as any));
    this._el_2 = import9.createRenderElement(this.renderer,this._el_0,'a',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_2_3 = new import18.Wrapper_NgClass(this.parentView.parentView.injectorGet(import20.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import21.KeyValueDiffers,this.parentView.parentIndex),new import16.ElementRef(this._el_2),this.renderer);
    this._text_3 = this.renderer.createText(this._el_2,'\n                            ',(null as any));
    this._anchor_4 = this.renderer.createTemplateAnchor(this._el_2,(null as any));
    this._vc_4 = new import19.ViewContainer(4,2,this,this._anchor_4);
    this._TemplateRef_4_5 = new import24.TemplateRef_(this,4,this._anchor_4);
    this._NgIf_4_6 = new import23.Wrapper_NgIf(this._vc_4.vcRef,this._TemplateRef_4_5);
    this._text_5 = this.renderer.createText(this._el_2,'\n                            ',(null as any));
    this._el_6 = import9.createRenderElement(this.renderer,this._el_2,'span',new import9.InlineArray2(2,'class','ui-menuitem-text'),(null as any));
    this._text_7 = this.renderer.createText(this._el_6,'',(null as any));
    this._text_8 = this.renderer.createText(this._el_2,'\n                        ',(null as any));
    this._text_9 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    var disposable_0:Function = import9.subscribeToRenderElement(this,this._el_0,new import9.InlineArray4(4,'mouseenter',(null as any),'mouseleave',(null as any)),this.eventHandler(this.handleEvent_0));
    var disposable_1:Function = import9.subscribeToRenderElement(this,this._el_2,new import9.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_2));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._anchor_4,
      this._text_5,
      this._el_6,
      this._text_7,
      this._text_8,
      this._text_9
    ]
    ),[
      disposable_0,
      disposable_1
    ]
    );
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import24.TemplateRef) && (4 === requestNodeIndex))) { return this._TemplateRef_4_5; }
    if (((token === import25.NgIf) && (4 === requestNodeIndex))) { return this._NgIf_4_6.context; }
    if (((token === import22.NgClass) && ((2 <= requestNodeIndex) && (requestNodeIndex <= 8)))) { return this._NgClass_2_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2_0_0:any = this._map_15(true,((this.parentView.context.hoveredItem == this.context.$implicit) && !this.context.$implicit.disabled),this.context.$implicit.disabled);
    this._NgClass_2_3.check_ngClass(currVal_2_0_0,throwOnChange,false);
    this._NgClass_2_3.ngDoCheck(this,this._el_2,throwOnChange);
    const currVal_4_0_0:any = this.context.$implicit.icon;
    this._NgIf_4_6.check_ngIf(currVal_4_0_0,throwOnChange,false);
    this._NgIf_4_6.ngDoCheck(this,this._anchor_4,throwOnChange);
    this._vc_4.detectChangesInNestedViews(throwOnChange);
    const currVal_14:any = (this.context.$implicit.url || '#');
    if (import9.checkBinding(throwOnChange,this._expr_14,currVal_14)) {
      this.renderer.setElementProperty(this._el_2,'href',this.viewUtils.sanitizer.sanitize(import26.SecurityContext.URL,currVal_14));
      this._expr_14 = currVal_14;
    }
    const currVal_16:any = import9.inlineInterpolate(1,'',this.context.$implicit.label,'');
    if (import9.checkBinding(throwOnChange,this._expr_16,currVal_16)) {
      this.renderer.setText(this._text_7,currVal_16);
      this._expr_16 = currVal_16;
    }
  }
  destroyInternal():void {
    this._vc_4.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import8.AppView<any> {
    if ((nodeIndex == 4)) { return new View_SplitButton2(this.viewUtils,this,4,this._anchor_4,this._vc_4); }
    return (null as any);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'mouseenter')) {
      const pd_sub_0:any = ((<any>(this.parentView.context.hoveredItem = this.context.$implicit)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'mouseleave')) {
      const pd_sub_1:any = ((<any>(this.parentView.context.hoveredItem = (null as any))) !== false);
      result = (pd_sub_1 && result);
    }
    return result;
  }
  handleEvent_2(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.context.itemClick($event,this.context.$implicit)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
var renderType_SplitButton:import10.RenderComponentType = import9.createRenderComponentType('',0,import11.ViewEncapsulation.None,styles_SplitButton,{});
export class View_SplitButton0 extends import8.AppView<import1.SplitButton> {
  _text_0:any;
  _el_1:any;
  _NgClass_1_3:import18.Wrapper_NgClass;
  _NgStyle_1_4:import27.Wrapper_NgStyle;
  _text_2:any;
  _el_3:any;
  _DomHandler_3_3:import12.DomHandler;
  _Button_3_4:import28.Wrapper_Button;
  _text_4:any;
  _el_5:any;
  _DomHandler_5_3:import12.DomHandler;
  _Button_5_4:import28.Wrapper_Button;
  _text_6:any;
  _el_7:any;
  _NgClass_7_3:import18.Wrapper_NgClass;
  _NgStyle_7_4:import27.Wrapper_NgStyle;
  _text_8:any;
  _el_9:any;
  _text_10:any;
  _anchor_11:any;
  /*private*/ _vc_11:import19.ViewContainer;
  _TemplateRef_11_5:any;
  _NgFor_11_6:import29.Wrapper_NgFor;
  _text_12:any;
  _text_13:any;
  _text_14:any;
  _text_15:any;
  _map_27:any;
  /*private*/ _expr_28:any;
  /*private*/ _expr_29:any;
  /*private*/ _expr_30:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_SplitButton0,renderType_SplitButton,import13.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways);
    this._map_27 = import9.pureProxy2((p0:any,p1:any):{[key: string]:any} => {
      return {
        'ui-splitbutton ui-buttonset ui-widget': p0,
        'ui-state-disabled': p1
      }
      ;
    });
    this._expr_28 = import7.UNINITIALIZED;
    this._expr_29 = import7.UNINITIALIZED;
    this._expr_30 = import7.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n        ',(null as any));
    this._el_1 = import9.createRenderElement(this.renderer,parentRenderNode,'div',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_1_3 = new import18.Wrapper_NgClass(this.parentView.injectorGet(import20.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import21.KeyValueDiffers,this.parentIndex),new import16.ElementRef(this._el_1),this.renderer);
    this._NgStyle_1_4 = new import27.Wrapper_NgStyle(this.parentView.injectorGet(import21.KeyValueDiffers,this.parentIndex),new import16.ElementRef(this._el_1),this.renderer);
    this._text_2 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_3 = import9.createRenderElement(this.renderer,this._el_1,'button',new import9.InlineArray8(6,'cornerStyleClass','ui-corner-left','pButton','','type','button'),(null as any));
    this._DomHandler_3_3 = new import12.DomHandler();
    this._Button_3_4 = new import28.Wrapper_Button(new import16.ElementRef(this._el_3),this._DomHandler_3_3);
    this._text_4 = this.renderer.createText(this._el_3,'\n            ',(null as any));
    this._el_5 = import9.createRenderElement(this.renderer,this._el_1,'button',new import9.InlineArray16(10,'class','ui-splitbutton-menubutton','cornerStyleClass','ui-corner-right','icon','fa-caret-down','pButton','','type','button'),(null as any));
    this._DomHandler_5_3 = new import12.DomHandler();
    this._Button_5_4 = new import28.Wrapper_Button(new import16.ElementRef(this._el_5),this._DomHandler_5_3);
    this._text_6 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_7 = import9.createRenderElement(this.renderer,this._el_1,'div',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_7_3 = new import18.Wrapper_NgClass(this.parentView.injectorGet(import20.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import21.KeyValueDiffers,this.parentIndex),new import16.ElementRef(this._el_7),this.renderer);
    this._NgStyle_7_4 = new import27.Wrapper_NgStyle(this.parentView.injectorGet(import21.KeyValueDiffers,this.parentIndex),new import16.ElementRef(this._el_7),this.renderer);
    this._text_8 = this.renderer.createText(this._el_7,'\n                ',(null as any));
    this._el_9 = import9.createRenderElement(this.renderer,this._el_7,'ul',new import9.InlineArray2(2,'class','ui-menu-list ui-helper-reset'),(null as any));
    this._text_10 = this.renderer.createText(this._el_9,'\n                    ',(null as any));
    this._anchor_11 = this.renderer.createTemplateAnchor(this._el_9,(null as any));
    this._vc_11 = new import19.ViewContainer(11,9,this,this._anchor_11);
    this._TemplateRef_11_5 = new import24.TemplateRef_(this,11,this._anchor_11);
    this._NgFor_11_6 = new import29.Wrapper_NgFor(this._vc_11.vcRef,this._TemplateRef_11_5,this.parentView.injectorGet(import20.IterableDiffers,this.parentIndex),this.ref);
    this._text_12 = this.renderer.createText(this._el_9,'\n                ',(null as any));
    this._text_13 = this.renderer.createText(this._el_7,'\n            ',(null as any));
    this._text_14 = this.renderer.createText(this._el_1,'\n        ',(null as any));
    this._text_15 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    var disposable_0:Function = import9.subscribeToRenderElement(this,this._el_3,new import9.InlineArray16(14,'click',(null as any),'mouseenter',(null as any),'mouseleave',(null as any),'mousedown',(null as any),'mouseup',(null as any),'focus',(null as any),'blur',(null as any)),this.eventHandler(this.handleEvent_3));
    var disposable_1:Function = import9.subscribeToRenderElement(this,this._el_5,new import9.InlineArray16(14,'click',(null as any),'mouseenter',(null as any),'mouseleave',(null as any),'mousedown',(null as any),'mouseup',(null as any),'focus',(null as any),'blur',(null as any)),this.eventHandler(this.handleEvent_5));
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2,
      this._el_3,
      this._text_4,
      this._el_5,
      this._text_6,
      this._el_7,
      this._text_8,
      this._el_9,
      this._text_10,
      this._anchor_11,
      this._text_12,
      this._text_13,
      this._text_14,
      this._text_15
    ]
    ),[
      disposable_0,
      disposable_1
    ]
    );
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import12.DomHandler) && ((3 <= requestNodeIndex) && (requestNodeIndex <= 4)))) { return this._DomHandler_3_3; }
    if (((token === import3.Button) && ((3 <= requestNodeIndex) && (requestNodeIndex <= 4)))) { return this._Button_3_4.context; }
    if (((token === import12.DomHandler) && (5 === requestNodeIndex))) { return this._DomHandler_5_3; }
    if (((token === import3.Button) && (5 === requestNodeIndex))) { return this._Button_5_4.context; }
    if (((token === import24.TemplateRef) && (11 === requestNodeIndex))) { return this._TemplateRef_11_5; }
    if (((token === import30.NgFor) && (11 === requestNodeIndex))) { return this._NgFor_11_6.context; }
    if (((token === import22.NgClass) && ((7 <= requestNodeIndex) && (requestNodeIndex <= 13)))) { return this._NgClass_7_3.context; }
    if (((token === import31.NgStyle) && ((7 <= requestNodeIndex) && (requestNodeIndex <= 13)))) { return this._NgStyle_7_4.context; }
    if (((token === import22.NgClass) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 14)))) { return this._NgClass_1_3.context; }
    if (((token === import31.NgStyle) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 14)))) { return this._NgStyle_1_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1_0_0:any = this.context.styleClass;
    this._NgClass_1_3.check_klass(currVal_1_0_0,throwOnChange,false);
    const currVal_1_0_1:any = this._map_27(true,this.context.disabled);
    this._NgClass_1_3.check_ngClass(currVal_1_0_1,throwOnChange,false);
    this._NgClass_1_3.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_1_1_0:any = this.context.style;
    this._NgStyle_1_4.check_ngStyle(currVal_1_1_0,throwOnChange,false);
    this._NgStyle_1_4.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_3_0_0:any = 'ui-corner-left';
    this._Button_3_4.check_cornerStyleClass(currVal_3_0_0,throwOnChange,false);
    const currVal_3_0_1:any = this.context.label;
    this._Button_3_4.check_label(currVal_3_0_1,throwOnChange,false);
    const currVal_3_0_2:any = this.context.icon;
    this._Button_3_4.check_icon(currVal_3_0_2,throwOnChange,false);
    this._Button_3_4.ngDoCheck(this,this._el_3,throwOnChange);
    const currVal_5_0_0:any = 'ui-corner-right';
    this._Button_5_4.check_cornerStyleClass(currVal_5_0_0,throwOnChange,false);
    const currVal_5_0_1:any = 'fa-caret-down';
    this._Button_5_4.check_icon(currVal_5_0_1,throwOnChange,false);
    this._Button_5_4.ngDoCheck(this,this._el_5,throwOnChange);
    const currVal_7_0_0:any = this.context.menuStyleClass;
    this._NgClass_7_3.check_klass(currVal_7_0_0,throwOnChange,false);
    const currVal_7_0_1:any = 'ui-menu ui-menu-dynamic ui-widget ui-widget-content ui-corner-all ui-helper-clearfix ui-shadow';
    this._NgClass_7_3.check_ngClass(currVal_7_0_1,throwOnChange,false);
    this._NgClass_7_3.ngDoCheck(this,this._el_7,throwOnChange);
    const currVal_7_1_0:any = this.context.menuStyle;
    this._NgStyle_7_4.check_ngStyle(currVal_7_1_0,throwOnChange,false);
    this._NgStyle_7_4.ngDoCheck(this,this._el_7,throwOnChange);
    const currVal_11_0_0:any = this.context.model;
    this._NgFor_11_6.check_ngForOf(currVal_11_0_0,throwOnChange,false);
    this._NgFor_11_6.ngDoCheck(this,this._anchor_11,throwOnChange);
    this._vc_11.detectChangesInNestedViews(throwOnChange);
    const currVal_28:any = this.context.disabled;
    if (import9.checkBinding(throwOnChange,this._expr_28,currVal_28)) {
      this.renderer.setElementProperty(this._el_3,'disabled',currVal_28);
      this._expr_28 = currVal_28;
    }
    this._Button_3_4.checkHost(this,this,this._el_3,throwOnChange);
    const currVal_29:any = this.context.disabled;
    if (import9.checkBinding(throwOnChange,this._expr_29,currVal_29)) {
      this.renderer.setElementProperty(this._el_5,'disabled',currVal_29);
      this._expr_29 = currVal_29;
    }
    this._Button_5_4.checkHost(this,this,this._el_5,throwOnChange);
    const currVal_30:any = (this.context.menuVisible? 'block': 'none');
    if (import9.checkBinding(throwOnChange,this._expr_30,currVal_30)) {
      this.renderer.setElementStyle(this._el_7,'display',((this.viewUtils.sanitizer.sanitize(import26.SecurityContext.STYLE,currVal_30) == null)? (null as any): this.viewUtils.sanitizer.sanitize(import26.SecurityContext.STYLE,currVal_30).toString()));
      this._expr_30 = currVal_30;
    }
    if (!throwOnChange) {
      if ((this.numberOfChecks === 0)) { this._Button_3_4.context.ngAfterViewInit(); }
      if ((this.numberOfChecks === 0)) { this._Button_5_4.context.ngAfterViewInit(); }
    }
  }
  destroyInternal():void {
    this._vc_11.destroyNestedViews();
    this._Button_3_4.ngOnDestroy();
    this._Button_5_4.ngOnDestroy();
  }
  createEmbeddedViewInternal(nodeIndex:number):import8.AppView<any> {
    if ((nodeIndex == 11)) { return new View_SplitButton1(this.viewUtils,this,11,this._anchor_11,this._vc_11); }
    return (null as any);
  }
  handleEvent_3(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    result = (this._Button_3_4.handleEvent(eventName,$event) && result);
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.context.onDefaultButtonClick($event)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
  handleEvent_5(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    result = (this._Button_5_4.handleEvent(eventName,$event) && result);
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.context.onDropdownClick($event,this._el_7,this._el_1)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
