// primeng/components/editor/editor.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from '@angular/core/src/linker/ng_module_factory';
import * as import1 from 'primeng/components/editor/editor';
import * as import2 from '@angular/common/src/common_module';
import * as import3 from '@angular/common/src/localization';
import * as import4 from '@angular/core/src/di/injector';
import * as import5 from '@angular/core/src/i18n/tokens';
import * as import6 from '@angular/core/src/change_detection/change_detection_util';
import * as import7 from '@angular/core/src/linker/view';
import * as import8 from '@angular/core/src/linker/view_utils';
import * as import9 from '@angular/core/src/render/api';
import * as import10 from '@angular/core/src/metadata/view';
import * as import11 from 'primeng/components/dom/domhandler';
import * as import12 from '@angular/core/src/linker/query_list';
import * as import13 from '@angular/core/src/linker/view_type';
import * as import14 from '@angular/core/src/change_detection/constants';
import * as import15 from '@angular/core/src/linker/component_factory';
import * as import16 from '@angular/core/src/linker/element_ref';
import * as import17 from '@angular/forms/src/directives/control_value_accessor';
import * as import18 from '@angular/core/src/linker/view_container';
import * as import19 from '../../../@angular/common/src/directives/ng_class.ngfactory';
import * as import20 from '../../../@angular/common/src/directives/ng_if.ngfactory';
import * as import21 from '../../../@angular/common/src/directives/ng_style.ngfactory';
import * as import22 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import23 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import24 from '@angular/core/src/linker/template_ref';
import * as import25 from '@angular/common/src/directives/ng_if';
import * as import26 from '@angular/common/src/directives/ng_style';
import * as import27 from '@angular/common/src/directives/ng_class';
class EditorModuleInjector extends import0.NgModuleInjector<import1.EditorModule> {
  _CommonModule_0:import2.CommonModule;
  _EditorModule_1:import1.EditorModule;
  __NgLocalization_2:import3.NgLocaleLocalization;
  constructor(parent:import4.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_2():import3.NgLocaleLocalization {
    if ((this.__NgLocalization_2 == null)) { (this.__NgLocalization_2 = new import3.NgLocaleLocalization(this.parent.get(import5.LOCALE_ID))); }
    return this.__NgLocalization_2;
  }
  createInternal():import1.EditorModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._EditorModule_1 = new import1.EditorModule();
    return this._EditorModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import1.EditorModule)) { return this._EditorModule_1; }
    if ((token === import3.NgLocalization)) { return this._NgLocalization_2; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const EditorModuleNgFactory:import0.NgModuleFactory<import1.EditorModule> = new import0.NgModuleFactory(EditorModuleInjector,import1.EditorModule);
export class Wrapper_Editor {
  /*private*/ _eventHandler:Function;
  context:import1.Editor;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  subscription0:any;
  subscription1:any;
  constructor(p0:any,p1:any) {
    this._changed = false;
    this.context = new import1.Editor(p0,p1);
    this._expr_0 = import6.UNINITIALIZED;
    this._expr_1 = import6.UNINITIALIZED;
    this._expr_2 = import6.UNINITIALIZED;
    this._expr_3 = import6.UNINITIALIZED;
    this._expr_4 = import6.UNINITIALIZED;
  }
  ngOnDetach(view:import7.AppView<any>,componentView:import7.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
    (this.subscription0 && this.subscription0.unsubscribe());
    (this.subscription1 && this.subscription1.unsubscribe());
  }
  check_style(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.style = currValue;
      this._expr_0 = currValue;
    }
  }
  check_styleClass(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.styleClass = currValue;
      this._expr_1 = currValue;
    }
  }
  check_placeholder(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_2,currValue))) {
      this._changed = true;
      this.context.placeholder = currValue;
      this._expr_2 = currValue;
    }
  }
  check_readOnly(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_3,currValue))) {
      this._changed = true;
      this.context.readOnly = currValue;
      this._expr_3 = currValue;
    }
  }
  check_formats(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_4,currValue))) {
      this._changed = true;
      this.context.formats = currValue;
      this._expr_4 = currValue;
    }
  }
  ngDoCheck(view:import7.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    return changed;
  }
  checkHost(view:import7.AppView<any>,componentView:import7.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import7.AppView<any>,_eventHandler:any,emit0:boolean,emit1:boolean):void {
    this._eventHandler = _eventHandler;
    if (emit0) { (this.subscription0 = this.context.onTextChange.subscribe(_eventHandler.bind(view,'onTextChange'))); }
    if (emit1) { (this.subscription1 = this.context.onSelectionChange.subscribe(_eventHandler.bind(view,'onSelectionChange'))); }
  }
}
var renderType_Editor_Host:import9.RenderComponentType = import8.createRenderComponentType('',0,import10.ViewEncapsulation.None,([] as any[]),{});
class View_Editor_Host0 extends import7.AppView<any> {
  _el_0:any;
  compView_0:import7.AppView<import1.Editor>;
  _DomHandler_0_3:import11.DomHandler;
  _Editor_0_4:Wrapper_Editor;
  __NG_VALUE_ACCESSOR_0_5:any[];
  _query_Header_0_0:import12.QueryList<any>;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_Editor_Host0,renderType_Editor_Host,import13.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways);
  }
  get _NG_VALUE_ACCESSOR_0_5():any[] {
    if ((this.__NG_VALUE_ACCESSOR_0_5 == null)) { (this.__NG_VALUE_ACCESSOR_0_5 = [this._Editor_0_4.context]); }
    return this.__NG_VALUE_ACCESSOR_0_5;
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import8.selectOrCreateRenderHostElement(this.renderer,'p-editor',import8.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this.compView_0 = new View_Editor0(this.viewUtils,this,0,this._el_0);
    this._DomHandler_0_3 = new import11.DomHandler();
    this._Editor_0_4 = new Wrapper_Editor(new import16.ElementRef(this._el_0),this._DomHandler_0_3);
    this._query_Header_0_0 = new import12.QueryList<any>();
    this.compView_0.create(this._Editor_0_4.context);
    this._query_Header_0_0.reset(([] as any[]));
    this._Editor_0_4.context.toolbar = this._query_Header_0_0.first;
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import15.ComponentRef_<any>(0,this,this._el_0,this._Editor_0_4.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import11.DomHandler) && (0 === requestNodeIndex))) { return this._DomHandler_0_3; }
    if (((token === import1.Editor) && (0 === requestNodeIndex))) { return this._Editor_0_4.context; }
    if (((token === import17.NG_VALUE_ACCESSOR) && (0 === requestNodeIndex))) { return this._NG_VALUE_ACCESSOR_0_5; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._Editor_0_4.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
    if (!throwOnChange) { if ((this.numberOfChecks === 0)) { this._Editor_0_4.context.ngAfterViewInit(); } }
  }
  destroyInternal():void {
    this.compView_0.destroy();
    this._Editor_0_4.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  visitProjectableNodesInternal(nodeIndex:number,ngContentIndex:number,cb:any,ctx:any):void {
    if (((nodeIndex == 0) && (ngContentIndex == 0))) {  }
  }
}
export const EditorNgFactory:import15.ComponentFactory<import1.Editor> = new import15.ComponentFactory<import1.Editor>('p-editor',View_Editor_Host0,import1.Editor);
const styles_Editor:any[] = ([] as any[]);
class View_Editor1 extends import7.AppView<any> {
  _el_0:any;
  _text_1:any;
  _text_2:any;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import18.ViewContainer) {
    super(View_Editor1,renderType_Editor,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import8.createRenderElement(this.renderer,(null as any),'div',new import8.InlineArray2(2,'class','ui-editor-toolbar ui-widget-header ui-corner-top'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    this.projectNodes(this._el_0,0);
    this._text_2 = this.renderer.createText(this._el_0,'\n            ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._text_2
    ]
    ),(null as any));
    return (null as any);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_Editor2 extends import7.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _el_4:any;
  _text_5:any;
  _el_6:any;
  _text_7:any;
  _text_8:any;
  _el_9:any;
  _text_10:any;
  _text_11:any;
  _el_12:any;
  _text_13:any;
  _text_14:any;
  _text_15:any;
  _el_16:any;
  _text_17:any;
  _el_18:any;
  _text_19:any;
  _text_20:any;
  _el_21:any;
  _text_22:any;
  _text_23:any;
  _el_24:any;
  _text_25:any;
  _text_26:any;
  _text_27:any;
  _text_28:any;
  _el_29:any;
  _text_30:any;
  _el_31:any;
  _text_32:any;
  _el_33:any;
  _text_34:any;
  _el_35:any;
  _text_36:any;
  _text_37:any;
  _el_38:any;
  _text_39:any;
  _el_40:any;
  _text_41:any;
  _el_42:any;
  _text_43:any;
  _text_44:any;
  _el_45:any;
  _text_46:any;
  _el_47:any;
  _text_48:any;
  _el_49:any;
  _text_50:any;
  _el_51:any;
  _text_52:any;
  _el_53:any;
  _text_54:any;
  _el_55:any;
  _text_56:any;
  _el_57:any;
  _text_58:any;
  _el_59:any;
  _text_60:any;
  _text_61:any;
  _text_62:any;
  _el_63:any;
  _text_64:any;
  _el_65:any;
  _text_66:any;
  _el_67:any;
  _text_68:any;
  _el_69:any;
  _text_70:any;
  _text_71:any;
  _el_72:any;
  _text_73:any;
  _el_74:any;
  _text_75:any;
  _text_76:any;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import18.ViewContainer) {
    super(View_Editor2,renderType_Editor,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import8.createRenderElement(this.renderer,(null as any),'div',new import8.InlineArray2(2,'class','ui-editor-toolbar ui-widget-header ui-corner-top'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    this._el_2 = import8.createRenderElement(this.renderer,this._el_0,'span',new import8.InlineArray2(2,'class','ql-formats'),(null as any));
    this._text_3 = this.renderer.createText(this._el_2,'\n                    ',(null as any));
    this._el_4 = import8.createRenderElement(this.renderer,this._el_2,'select',new import8.InlineArray2(2,'class','ql-header'),(null as any));
    this._text_5 = this.renderer.createText(this._el_4,'\n                      ',(null as any));
    this._el_6 = import8.createRenderElement(this.renderer,this._el_4,'option',new import8.InlineArray2(2,'value','1'),(null as any));
    this._text_7 = this.renderer.createText(this._el_6,'Heading',(null as any));
    this._text_8 = this.renderer.createText(this._el_4,'\n                      ',(null as any));
    this._el_9 = import8.createRenderElement(this.renderer,this._el_4,'option',new import8.InlineArray2(2,'value','2'),(null as any));
    this._text_10 = this.renderer.createText(this._el_9,'Subheading',(null as any));
    this._text_11 = this.renderer.createText(this._el_4,'\n                      ',(null as any));
    this._el_12 = import8.createRenderElement(this.renderer,this._el_4,'option',new import8.InlineArray2(2,'selected',''),(null as any));
    this._text_13 = this.renderer.createText(this._el_12,'Normal',(null as any));
    this._text_14 = this.renderer.createText(this._el_4,'\n                    ',(null as any));
    this._text_15 = this.renderer.createText(this._el_2,'\n                    ',(null as any));
    this._el_16 = import8.createRenderElement(this.renderer,this._el_2,'select',new import8.InlineArray2(2,'class','ql-font'),(null as any));
    this._text_17 = this.renderer.createText(this._el_16,'\n                      ',(null as any));
    this._el_18 = import8.createRenderElement(this.renderer,this._el_16,'option',new import8.InlineArray2(2,'selected',''),(null as any));
    this._text_19 = this.renderer.createText(this._el_18,'Sans Serif',(null as any));
    this._text_20 = this.renderer.createText(this._el_16,'\n                      ',(null as any));
    this._el_21 = import8.createRenderElement(this.renderer,this._el_16,'option',new import8.InlineArray2(2,'value','serif'),(null as any));
    this._text_22 = this.renderer.createText(this._el_21,'Serif',(null as any));
    this._text_23 = this.renderer.createText(this._el_16,'\n                      ',(null as any));
    this._el_24 = import8.createRenderElement(this.renderer,this._el_16,'option',new import8.InlineArray2(2,'value','monospace'),(null as any));
    this._text_25 = this.renderer.createText(this._el_24,'Monospace',(null as any));
    this._text_26 = this.renderer.createText(this._el_16,'\n                    ',(null as any));
    this._text_27 = this.renderer.createText(this._el_2,'\n                ',(null as any));
    this._text_28 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    this._el_29 = import8.createRenderElement(this.renderer,this._el_0,'span',new import8.InlineArray2(2,'class','ql-formats'),(null as any));
    this._text_30 = this.renderer.createText(this._el_29,'\n                    ',(null as any));
    this._el_31 = import8.createRenderElement(this.renderer,this._el_29,'button',new import8.InlineArray2(2,'class','ql-bold'),(null as any));
    this._text_32 = this.renderer.createText(this._el_29,'\n                    ',(null as any));
    this._el_33 = import8.createRenderElement(this.renderer,this._el_29,'button',new import8.InlineArray2(2,'class','ql-italic'),(null as any));
    this._text_34 = this.renderer.createText(this._el_29,'\n                    ',(null as any));
    this._el_35 = import8.createRenderElement(this.renderer,this._el_29,'button',new import8.InlineArray2(2,'class','ql-underline'),(null as any));
    this._text_36 = this.renderer.createText(this._el_29,'\n                ',(null as any));
    this._text_37 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    this._el_38 = import8.createRenderElement(this.renderer,this._el_0,'span',new import8.InlineArray2(2,'class','ql-formats'),(null as any));
    this._text_39 = this.renderer.createText(this._el_38,'\n                    ',(null as any));
    this._el_40 = import8.createRenderElement(this.renderer,this._el_38,'select',new import8.InlineArray2(2,'class','ql-color'),(null as any));
    this._text_41 = this.renderer.createText(this._el_38,'\n                    ',(null as any));
    this._el_42 = import8.createRenderElement(this.renderer,this._el_38,'select',new import8.InlineArray2(2,'class','ql-background'),(null as any));
    this._text_43 = this.renderer.createText(this._el_38,'\n                ',(null as any));
    this._text_44 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    this._el_45 = import8.createRenderElement(this.renderer,this._el_0,'span',new import8.InlineArray2(2,'class','ql-formats'),(null as any));
    this._text_46 = this.renderer.createText(this._el_45,'\n                    ',(null as any));
    this._el_47 = import8.createRenderElement(this.renderer,this._el_45,'button',new import8.InlineArray4(4,'class','ql-list','value','ordered'),(null as any));
    this._text_48 = this.renderer.createText(this._el_45,'\n                    ',(null as any));
    this._el_49 = import8.createRenderElement(this.renderer,this._el_45,'button',new import8.InlineArray4(4,'class','ql-list','value','bullet'),(null as any));
    this._text_50 = this.renderer.createText(this._el_45,'\n                    ',(null as any));
    this._el_51 = import8.createRenderElement(this.renderer,this._el_45,'select',new import8.InlineArray2(2,'class','ql-align'),(null as any));
    this._text_52 = this.renderer.createText(this._el_51,'\n                        ',(null as any));
    this._el_53 = import8.createRenderElement(this.renderer,this._el_51,'option',new import8.InlineArray2(2,'selected',''),(null as any));
    this._text_54 = this.renderer.createText(this._el_51,'\n                        ',(null as any));
    this._el_55 = import8.createRenderElement(this.renderer,this._el_51,'option',new import8.InlineArray2(2,'value','center'),(null as any));
    this._text_56 = this.renderer.createText(this._el_51,'\n                        ',(null as any));
    this._el_57 = import8.createRenderElement(this.renderer,this._el_51,'option',new import8.InlineArray2(2,'value','right'),(null as any));
    this._text_58 = this.renderer.createText(this._el_51,'\n                        ',(null as any));
    this._el_59 = import8.createRenderElement(this.renderer,this._el_51,'option',new import8.InlineArray2(2,'value','justify'),(null as any));
    this._text_60 = this.renderer.createText(this._el_51,'\n                    ',(null as any));
    this._text_61 = this.renderer.createText(this._el_45,'\n                ',(null as any));
    this._text_62 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    this._el_63 = import8.createRenderElement(this.renderer,this._el_0,'span',new import8.InlineArray2(2,'class','ql-formats'),(null as any));
    this._text_64 = this.renderer.createText(this._el_63,'\n                    ',(null as any));
    this._el_65 = import8.createRenderElement(this.renderer,this._el_63,'button',new import8.InlineArray2(2,'class','ql-link'),(null as any));
    this._text_66 = this.renderer.createText(this._el_63,'\n                    ',(null as any));
    this._el_67 = import8.createRenderElement(this.renderer,this._el_63,'button',new import8.InlineArray2(2,'class','ql-image'),(null as any));
    this._text_68 = this.renderer.createText(this._el_63,'\n                    ',(null as any));
    this._el_69 = import8.createRenderElement(this.renderer,this._el_63,'button',new import8.InlineArray2(2,'class','ql-code-block'),(null as any));
    this._text_70 = this.renderer.createText(this._el_63,'\n                ',(null as any));
    this._text_71 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    this._el_72 = import8.createRenderElement(this.renderer,this._el_0,'span',new import8.InlineArray2(2,'class','ql-formats'),(null as any));
    this._text_73 = this.renderer.createText(this._el_72,'\n                    ',(null as any));
    this._el_74 = import8.createRenderElement(this.renderer,this._el_72,'button',new import8.InlineArray2(2,'class','ql-clean'),(null as any));
    this._text_75 = this.renderer.createText(this._el_72,'\n                ',(null as any));
    this._text_76 = this.renderer.createText(this._el_0,'\n            ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._el_4,
      this._text_5,
      this._el_6,
      this._text_7,
      this._text_8,
      this._el_9,
      this._text_10,
      this._text_11,
      this._el_12,
      this._text_13,
      this._text_14,
      this._text_15,
      this._el_16,
      this._text_17,
      this._el_18,
      this._text_19,
      this._text_20,
      this._el_21,
      this._text_22,
      this._text_23,
      this._el_24,
      this._text_25,
      this._text_26,
      this._text_27,
      this._text_28,
      this._el_29,
      this._text_30,
      this._el_31,
      this._text_32,
      this._el_33,
      this._text_34,
      this._el_35,
      this._text_36,
      this._text_37,
      this._el_38,
      this._text_39,
      this._el_40,
      this._text_41,
      this._el_42,
      this._text_43,
      this._text_44,
      this._el_45,
      this._text_46,
      this._el_47,
      this._text_48,
      this._el_49,
      this._text_50,
      this._el_51,
      this._text_52,
      this._el_53,
      this._text_54,
      this._el_55,
      this._text_56,
      this._el_57,
      this._text_58,
      this._el_59,
      this._text_60,
      this._text_61,
      this._text_62,
      this._el_63,
      this._text_64,
      this._el_65,
      this._text_66,
      this._el_67,
      this._text_68,
      this._el_69,
      this._text_70,
      this._text_71,
      this._el_72,
      this._text_73,
      this._el_74,
      this._text_75,
      this._text_76
    ]
    ),(null as any));
    return (null as any);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
var renderType_Editor:import9.RenderComponentType = import8.createRenderComponentType('',1,import10.ViewEncapsulation.None,styles_Editor,{});
export class View_Editor0 extends import7.AppView<import1.Editor> {
  _text_0:any;
  _el_1:any;
  _NgClass_1_3:import19.Wrapper_NgClass;
  _text_2:any;
  _anchor_3:any;
  /*private*/ _vc_3:import18.ViewContainer;
  _TemplateRef_3_5:any;
  _NgIf_3_6:import20.Wrapper_NgIf;
  _text_4:any;
  _anchor_5:any;
  /*private*/ _vc_5:import18.ViewContainer;
  _TemplateRef_5_5:any;
  _NgIf_5_6:import20.Wrapper_NgIf;
  _text_6:any;
  _el_7:any;
  _NgStyle_7_3:import21.Wrapper_NgStyle;
  _text_8:any;
  _text_9:any;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_Editor0,renderType_Editor,import13.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n        ',(null as any));
    this._el_1 = import8.createRenderElement(this.renderer,parentRenderNode,'div',import8.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_1_3 = new import19.Wrapper_NgClass(this.parentView.injectorGet(import22.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import23.KeyValueDiffers,this.parentIndex),new import16.ElementRef(this._el_1),this.renderer);
    this._text_2 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._anchor_3 = this.renderer.createTemplateAnchor(this._el_1,(null as any));
    this._vc_3 = new import18.ViewContainer(3,1,this,this._anchor_3);
    this._TemplateRef_3_5 = new import24.TemplateRef_(this,3,this._anchor_3);
    this._NgIf_3_6 = new import20.Wrapper_NgIf(this._vc_3.vcRef,this._TemplateRef_3_5);
    this._text_4 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._anchor_5 = this.renderer.createTemplateAnchor(this._el_1,(null as any));
    this._vc_5 = new import18.ViewContainer(5,1,this,this._anchor_5);
    this._TemplateRef_5_5 = new import24.TemplateRef_(this,5,this._anchor_5);
    this._NgIf_5_6 = new import20.Wrapper_NgIf(this._vc_5.vcRef,this._TemplateRef_5_5);
    this._text_6 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_7 = import8.createRenderElement(this.renderer,this._el_1,'div',new import8.InlineArray2(2,'class','ui-editor-content'),(null as any));
    this._NgStyle_7_3 = new import21.Wrapper_NgStyle(this.parentView.injectorGet(import23.KeyValueDiffers,this.parentIndex),new import16.ElementRef(this._el_7),this.renderer);
    this._text_8 = this.renderer.createText(this._el_1,'\n        ',(null as any));
    this._text_9 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2,
      this._anchor_3,
      this._text_4,
      this._anchor_5,
      this._text_6,
      this._el_7,
      this._text_8,
      this._text_9
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import24.TemplateRef) && (3 === requestNodeIndex))) { return this._TemplateRef_3_5; }
    if (((token === import25.NgIf) && (3 === requestNodeIndex))) { return this._NgIf_3_6.context; }
    if (((token === import24.TemplateRef) && (5 === requestNodeIndex))) { return this._TemplateRef_5_5; }
    if (((token === import25.NgIf) && (5 === requestNodeIndex))) { return this._NgIf_5_6.context; }
    if (((token === import26.NgStyle) && (7 === requestNodeIndex))) { return this._NgStyle_7_3.context; }
    if (((token === import27.NgClass) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 8)))) { return this._NgClass_1_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1_0_0:any = this.context.styleClass;
    this._NgClass_1_3.check_klass(currVal_1_0_0,throwOnChange,false);
    const currVal_1_0_1:any = 'ui-widget ui-editor-container ui-corner-all';
    this._NgClass_1_3.check_ngClass(currVal_1_0_1,throwOnChange,false);
    this._NgClass_1_3.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_3_0_0:any = this.context.toolbar;
    this._NgIf_3_6.check_ngIf(currVal_3_0_0,throwOnChange,false);
    this._NgIf_3_6.ngDoCheck(this,this._anchor_3,throwOnChange);
    const currVal_5_0_0:boolean = !this.context.toolbar;
    this._NgIf_5_6.check_ngIf(currVal_5_0_0,throwOnChange,false);
    this._NgIf_5_6.ngDoCheck(this,this._anchor_5,throwOnChange);
    const currVal_7_0_0:any = this.context.style;
    this._NgStyle_7_3.check_ngStyle(currVal_7_0_0,throwOnChange,false);
    this._NgStyle_7_3.ngDoCheck(this,this._el_7,throwOnChange);
    this._vc_3.detectChangesInNestedViews(throwOnChange);
    this._vc_5.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_3.destroyNestedViews();
    this._vc_5.destroyNestedViews();
  }
  createEmbeddedViewInternal(nodeIndex:number):import7.AppView<any> {
    if ((nodeIndex == 3)) { return new View_Editor1(this.viewUtils,this,3,this._anchor_3,this._vc_3); }
    if ((nodeIndex == 5)) { return new View_Editor2(this.viewUtils,this,5,this._anchor_5,this._vc_5); }
    return (null as any);
  }
}
