// primeng/components/password/password.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from '@angular/core/src/linker/ng_module_factory';
import * as import1 from 'primeng/components/password/password';
import * as import2 from '@angular/common/src/common_module';
import * as import3 from '@angular/common/src/localization';
import * as import4 from '@angular/core/src/di/injector';
import * as import5 from '@angular/core/src/i18n/tokens';
import * as import6 from '@angular/core/src/change_detection/change_detection_util';
import * as import7 from '@angular/core/src/linker/view';
import * as import8 from '@angular/core/src/linker/view_utils';
class PasswordModuleInjector extends import0.NgModuleInjector<import1.PasswordModule> {
  _CommonModule_0:import2.CommonModule;
  _PasswordModule_1:import1.PasswordModule;
  __NgLocalization_2:import3.NgLocaleLocalization;
  constructor(parent:import4.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_2():import3.NgLocaleLocalization {
    if ((this.__NgLocalization_2 == null)) { (this.__NgLocalization_2 = new import3.NgLocaleLocalization(this.parent.get(import5.LOCALE_ID))); }
    return this.__NgLocalization_2;
  }
  createInternal():import1.PasswordModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._PasswordModule_1 = new import1.PasswordModule();
    return this._PasswordModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import1.PasswordModule)) { return this._PasswordModule_1; }
    if ((token === import3.NgLocalization)) { return this._NgLocalization_2; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const PasswordModuleNgFactory:import0.NgModuleFactory<import1.PasswordModule> = new import0.NgModuleFactory(PasswordModuleInjector,import1.PasswordModule);
export class Wrapper_Password {
  /*private*/ _eventHandler:Function;
  context:import1.Password;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  /*private*/ _expr_5:any;
  /*private*/ _expr_6:any;
  /*private*/ _expr_7:any;
  /*private*/ _expr_8:any;
  /*private*/ _expr_9:any;
  /*private*/ _expr_10:any;
  /*private*/ _expr_11:any;
  constructor(p0:any,p1:any) {
    this._changed = false;
    this.context = new import1.Password(p0,p1);
    this._expr_0 = import6.UNINITIALIZED;
    this._expr_1 = import6.UNINITIALIZED;
    this._expr_2 = import6.UNINITIALIZED;
    this._expr_3 = import6.UNINITIALIZED;
    this._expr_4 = import6.UNINITIALIZED;
    this._expr_5 = import6.UNINITIALIZED;
    this._expr_6 = import6.UNINITIALIZED;
    this._expr_7 = import6.UNINITIALIZED;
    this._expr_8 = import6.UNINITIALIZED;
    this._expr_9 = import6.UNINITIALIZED;
    this._expr_10 = import6.UNINITIALIZED;
    this._expr_11 = import6.UNINITIALIZED;
  }
  ngOnDetach(view:import7.AppView<any>,componentView:import7.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
    this.context.ngOnDestroy();
  }
  check_promptLabel(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.promptLabel = currValue;
      this._expr_0 = currValue;
    }
  }
  check_weakLabel(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.weakLabel = currValue;
      this._expr_1 = currValue;
    }
  }
  check_mediumLabel(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_2,currValue))) {
      this._changed = true;
      this.context.mediumLabel = currValue;
      this._expr_2 = currValue;
    }
  }
  check_strongLabel(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_3,currValue))) {
      this._changed = true;
      this.context.strongLabel = currValue;
      this._expr_3 = currValue;
    }
  }
  ngDoCheck(view:import7.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    return changed;
  }
  checkHost(view:import7.AppView<any>,componentView:import7.AppView<any>,el:any,throwOnChange:boolean):void {
    const currVal_4:any = true;
    if (import8.checkBinding(throwOnChange,this._expr_4,currVal_4)) {
      view.renderer.setElementClass(el,'ui-inputtext',currVal_4);
      this._expr_4 = currVal_4;
    }
    const currVal_5:any = true;
    if (import8.checkBinding(throwOnChange,this._expr_5,currVal_5)) {
      view.renderer.setElementClass(el,'ui-corner-all',currVal_5);
      this._expr_5 = currVal_5;
    }
    const currVal_6:any = true;
    if (import8.checkBinding(throwOnChange,this._expr_6,currVal_6)) {
      view.renderer.setElementClass(el,'ui-state-default',currVal_6);
      this._expr_6 = currVal_6;
    }
    const currVal_7:any = true;
    if (import8.checkBinding(throwOnChange,this._expr_7,currVal_7)) {
      view.renderer.setElementClass(el,'ui-widget',currVal_7);
      this._expr_7 = currVal_7;
    }
    const currVal_8:any = this.context.hover;
    if (import8.checkBinding(throwOnChange,this._expr_8,currVal_8)) {
      view.renderer.setElementClass(el,'ui-state-hover',currVal_8);
      this._expr_8 = currVal_8;
    }
    const currVal_9:any = this.context.focus;
    if (import8.checkBinding(throwOnChange,this._expr_9,currVal_9)) {
      view.renderer.setElementClass(el,'ui-state-focus',currVal_9);
      this._expr_9 = currVal_9;
    }
    const currVal_10:any = this.context.disabled;
    if (import8.checkBinding(throwOnChange,this._expr_10,currVal_10)) {
      view.renderer.setElementClass(el,'ui-state-disabled',currVal_10);
      this._expr_10 = currVal_10;
    }
    const currVal_11:any = this.context.filled;
    if (import8.checkBinding(throwOnChange,this._expr_11,currVal_11)) {
      view.renderer.setElementClass(el,'ui-state-filled',currVal_11);
      this._expr_11 = currVal_11;
    }
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    if ((eventName == 'mouseover')) {
      const pd_sub_0:any = ((<any>this.context.onMouseover($event)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'mouseout')) {
      const pd_sub_1:any = ((<any>this.context.onMouseout($event)) !== false);
      result = (pd_sub_1 && result);
    }
    if ((eventName == 'focus')) {
      const pd_sub_2:any = ((<any>this.context.onFocus($event)) !== false);
      result = (pd_sub_2 && result);
    }
    if ((eventName == 'blur')) {
      const pd_sub_3:any = ((<any>this.context.onBlur($event)) !== false);
      result = (pd_sub_3 && result);
    }
    if ((eventName == 'keyup')) {
      const pd_sub_4:any = ((<any>this.context.onKeyup($event)) !== false);
      result = (pd_sub_4 && result);
    }
    return result;
  }
  subscribe(view:import7.AppView<any>,_eventHandler:any):void {
    this._eventHandler = _eventHandler;
  }
}
