// primeng/components/carousel/carousel.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from '@angular/core/src/linker/ng_module_factory';
import * as import1 from 'primeng/components/carousel/carousel';
import * as import2 from '@angular/common/src/common_module';
import * as import3 from 'primeng/components/common/shared';
import * as import4 from '@angular/common/src/localization';
import * as import5 from '@angular/core/src/di/injector';
import * as import6 from '@angular/core/src/i18n/tokens';
import * as import7 from '@angular/core/src/change_detection/change_detection_util';
import * as import8 from '@angular/core/src/linker/view';
import * as import9 from '@angular/core/src/linker/view_utils';
import * as import10 from '@angular/core/src/render/api';
import * as import11 from '@angular/core/src/metadata/view';
import * as import12 from 'primeng/components/dom/domhandler';
import * as import13 from '@angular/core/src/linker/query_list';
import * as import14 from '@angular/core/src/linker/view_type';
import * as import15 from '@angular/core/src/change_detection/constants';
import * as import16 from '@angular/core/src/linker/component_factory';
import * as import17 from '@angular/core/src/linker/element_ref';
import * as import18 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import19 from '../../../@angular/common/src/directives/ng_class.ngfactory';
import * as import20 from '@angular/core/src/linker/view_container';
import * as import21 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import22 from '@angular/common/src/directives/ng_class';
import * as import23 from '../../../@angular/common/src/directives/ng_for.ngfactory';
import * as import24 from '@angular/core/src/linker/template_ref';
import * as import25 from '@angular/common/src/directives/ng_for';
import * as import26 from '@angular/core/src/security';
import * as import27 from '../common/shared.ngfactory';
import * as import28 from '../../../@angular/common/src/directives/ng_style.ngfactory';
import * as import29 from '../../../@angular/common/src/directives/ng_if.ngfactory';
import * as import30 from '@angular/common/src/directives/ng_if';
import * as import31 from '@angular/common/src/directives/ng_style';
class CarouselModuleInjector extends import0.NgModuleInjector<import1.CarouselModule> {
  _CommonModule_0:import2.CommonModule;
  _SharedModule_1:import3.SharedModule;
  _CarouselModule_2:import1.CarouselModule;
  __NgLocalization_3:import4.NgLocaleLocalization;
  constructor(parent:import5.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_3():import4.NgLocaleLocalization {
    if ((this.__NgLocalization_3 == null)) { (this.__NgLocalization_3 = new import4.NgLocaleLocalization(this.parent.get(import6.LOCALE_ID))); }
    return this.__NgLocalization_3;
  }
  createInternal():import1.CarouselModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._SharedModule_1 = new import3.SharedModule();
    this._CarouselModule_2 = new import1.CarouselModule();
    return this._CarouselModule_2;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.SharedModule)) { return this._SharedModule_1; }
    if ((token === import1.CarouselModule)) { return this._CarouselModule_2; }
    if ((token === import4.NgLocalization)) { return this._NgLocalization_3; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const CarouselModuleNgFactory:import0.NgModuleFactory<import1.CarouselModule> = new import0.NgModuleFactory(CarouselModuleInjector,import1.CarouselModule);
export class Wrapper_Carousel {
  /*private*/ _eventHandler:Function;
  context:import1.Carousel;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  /*private*/ _expr_5:any;
  /*private*/ _expr_6:any;
  /*private*/ _expr_7:any;
  /*private*/ _expr_8:any;
  /*private*/ _expr_9:any;
  /*private*/ _expr_10:any;
  /*private*/ _expr_11:any;
  /*private*/ _expr_12:any;
  constructor(p0:any,p1:any,p2:any,p3:any) {
    this._changed = false;
    this.context = new import1.Carousel(p0,p1,p2,p3);
    this._expr_0 = import7.UNINITIALIZED;
    this._expr_1 = import7.UNINITIALIZED;
    this._expr_2 = import7.UNINITIALIZED;
    this._expr_3 = import7.UNINITIALIZED;
    this._expr_4 = import7.UNINITIALIZED;
    this._expr_5 = import7.UNINITIALIZED;
    this._expr_6 = import7.UNINITIALIZED;
    this._expr_7 = import7.UNINITIALIZED;
    this._expr_8 = import7.UNINITIALIZED;
    this._expr_9 = import7.UNINITIALIZED;
    this._expr_10 = import7.UNINITIALIZED;
    this._expr_11 = import7.UNINITIALIZED;
    this._expr_12 = import7.UNINITIALIZED;
  }
  ngOnDetach(view:import8.AppView<any>,componentView:import8.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
    this.context.ngOnDestroy();
  }
  check_value(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.value = currValue;
      this._expr_0 = currValue;
    }
  }
  check_numVisible(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.numVisible = currValue;
      this._expr_1 = currValue;
    }
  }
  check_firstVisible(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_2,currValue))) {
      this._changed = true;
      this.context.firstVisible = currValue;
      this._expr_2 = currValue;
    }
  }
  check_headerText(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_3,currValue))) {
      this._changed = true;
      this.context.headerText = currValue;
      this._expr_3 = currValue;
    }
  }
  check_circular(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_4,currValue))) {
      this._changed = true;
      this.context.circular = currValue;
      this._expr_4 = currValue;
    }
  }
  check_breakpoint(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_5,currValue))) {
      this._changed = true;
      this.context.breakpoint = currValue;
      this._expr_5 = currValue;
    }
  }
  check_responsive(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_6,currValue))) {
      this._changed = true;
      this.context.responsive = currValue;
      this._expr_6 = currValue;
    }
  }
  check_autoplayInterval(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_7,currValue))) {
      this._changed = true;
      this.context.autoplayInterval = currValue;
      this._expr_7 = currValue;
    }
  }
  check_effectDuration(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_8,currValue))) {
      this._changed = true;
      this.context.effectDuration = currValue;
      this._expr_8 = currValue;
    }
  }
  check_easing(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_9,currValue))) {
      this._changed = true;
      this.context.easing = currValue;
      this._expr_9 = currValue;
    }
  }
  check_pageLinks(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_10,currValue))) {
      this._changed = true;
      this.context.pageLinks = currValue;
      this._expr_10 = currValue;
    }
  }
  check_style(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_11,currValue))) {
      this._changed = true;
      this.context.style = currValue;
      this._expr_11 = currValue;
    }
  }
  check_styleClass(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_12,currValue))) {
      this._changed = true;
      this.context.styleClass = currValue;
      this._expr_12 = currValue;
    }
  }
  ngDoCheck(view:import8.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    if (!throwOnChange) {
      if ((view.numberOfChecks === 0)) { this.context.ngOnInit(); }
      this.context.ngDoCheck();
    }
    return changed;
  }
  checkHost(view:import8.AppView<any>,componentView:import8.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import8.AppView<any>,_eventHandler:any):void {
    this._eventHandler = _eventHandler;
  }
}
var renderType_Carousel_Host:import10.RenderComponentType = import9.createRenderComponentType('',0,import11.ViewEncapsulation.None,([] as any[]),{});
class View_Carousel_Host0 extends import8.AppView<any> {
  _el_0:any;
  compView_0:import8.AppView<import1.Carousel>;
  _DomHandler_0_3:import12.DomHandler;
  _Carousel_0_4:Wrapper_Carousel;
  _query_TemplateRef_0_0:import13.QueryList<any>;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_Carousel_Host0,renderType_Carousel_Host,import14.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    this._el_0 = import9.selectOrCreateRenderHostElement(this.renderer,'p-carousel',import9.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this.compView_0 = new View_Carousel0(this.viewUtils,this,0,this._el_0);
    this._DomHandler_0_3 = new import12.DomHandler();
    this._Carousel_0_4 = new Wrapper_Carousel(new import17.ElementRef(this._el_0),this._DomHandler_0_3,this.injectorGet(import18.IterableDiffers,this.parentIndex),this.renderer);
    this._query_TemplateRef_0_0 = new import13.QueryList<any>();
    this.compView_0.create(this._Carousel_0_4.context);
    this._query_TemplateRef_0_0.reset(([] as any[]));
    this._Carousel_0_4.context.itemTemplate = this._query_TemplateRef_0_0.first;
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import16.ComponentRef_<any>(0,this,this._el_0,this._Carousel_0_4.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import12.DomHandler) && (0 === requestNodeIndex))) { return this._DomHandler_0_3; }
    if (((token === import1.Carousel) && (0 === requestNodeIndex))) { return this._Carousel_0_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._Carousel_0_4.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
    if (!throwOnChange) {
      if ((this.numberOfChecks === 0)) { this._Carousel_0_4.context.ngAfterViewInit(); }
      this._Carousel_0_4.context.ngAfterViewChecked();
    }
  }
  destroyInternal():void {
    this.compView_0.destroy();
    this._Carousel_0_4.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
export const CarouselNgFactory:import16.ComponentFactory<import1.Carousel> = new import16.ComponentFactory<import1.Carousel>('p-carousel',View_Carousel_Host0,import1.Carousel);
const styles_Carousel:any[] = ([] as any[]);
class View_Carousel2 extends import8.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import19.Wrapper_NgClass;
  _map_2:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import20.ViewContainer) {
    super(View_Carousel2,renderType_Carousel,import14.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_2 = import9.pureProxy1((p0:any):{[key: string]:any} => {
      return {'fa-dot-circle-o': p0};
    });
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'a',new import9.InlineArray4(4,'class','ui-carousel-page-link fa fa-circle-o','href','#'),(null as any));
    this._NgClass_0_3 = new import19.Wrapper_NgClass(this.parentView.parentView.parentView.injectorGet(import18.IterableDiffers,this.parentView.parentView.parentIndex),this.parentView.parentView.parentView.injectorGet(import21.KeyValueDiffers,this.parentView.parentView.parentIndex),new import17.ElementRef(this._el_0),this.renderer);
    var disposable_0:Function = import9.subscribeToRenderElement(this,this._el_0,new import9.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import22.NgClass) && (0 === requestNodeIndex))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = 'ui-carousel-page-link fa fa-circle-o';
    this._NgClass_0_3.check_klass(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = this._map_2((this.parentView.parentView.context.page === this.context.index));
    this._NgClass_0_3.check_ngClass(currVal_0_0_1,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.parentView.context.setPageWithLink($event,this.context.index)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
class View_Carousel1 extends import8.AppView<any> {
  _el_0:any;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import20.ViewContainer;
  _TemplateRef_2_5:any;
  _NgFor_2_6:import23.Wrapper_NgFor;
  _text_3:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import20.ViewContainer) {
    super(View_Carousel1,renderType_Carousel,import14.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'div',new import9.InlineArray2(2,'class','ui-carousel-page-links'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import20.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import24.TemplateRef_(this,2,this._anchor_2);
    this._NgFor_2_6 = new import23.Wrapper_NgFor(this._vc_2.vcRef,this._TemplateRef_2_5,this.parentView.parentView.injectorGet(import18.IterableDiffers,this.parentView.parentIndex),this.parentView.ref);
    this._text_3 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import24.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import25.NgFor) && (2 === requestNodeIndex))) { return this._NgFor_2_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2_0_0:any = this.parentView.context.anchorPageLinks;
    this._NgFor_2_6.check_ngForOf(currVal_2_0_0,throwOnChange,false);
    this._NgFor_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import8.AppView<any> {
    if ((nodeIndex == 2)) { return new View_Carousel2(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    return (null as any);
  }
}
class View_Carousel4 extends import8.AppView<any> {
  _el_0:any;
  _text_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import20.ViewContainer) {
    super(View_Carousel4,renderType_Carousel,import14.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_2 = import7.UNINITIALIZED;
    this._expr_3 = import7.UNINITIALIZED;
    this._expr_4 = import7.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'option',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1
    ]
    ),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2:any = this.context.$implicit;
    if (import9.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setElementProperty(this._el_0,'value',currVal_2);
      this._expr_2 = currVal_2;
    }
    const currVal_3:any = (this.parentView.parentView.context.value == this.context.$implicit);
    if (import9.checkBinding(throwOnChange,this._expr_3,currVal_3)) {
      this.renderer.setElementProperty(this._el_0,'selected',currVal_3);
      this._expr_3 = currVal_3;
    }
    const currVal_4:any = import9.inlineInterpolate(1,'',(this.context.$implicit + 1),'');
    if (import9.checkBinding(throwOnChange,this._expr_4,currVal_4)) {
      this.renderer.setText(this._text_1,currVal_4);
      this._expr_4 = currVal_4;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_Carousel3 extends import8.AppView<any> {
  _el_0:any;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import20.ViewContainer;
  _TemplateRef_2_5:any;
  _NgFor_2_6:import23.Wrapper_NgFor;
  _text_3:any;
  /*private*/ _expr_7:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import20.ViewContainer) {
    super(View_Carousel3,renderType_Carousel,import14.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_7 = import7.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'select',new import9.InlineArray2(2,'class','ui-carousel-dropdown ui-widget ui-state-default ui-corner-left'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import20.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import24.TemplateRef_(this,2,this._anchor_2);
    this._NgFor_2_6 = new import23.Wrapper_NgFor(this._vc_2.vcRef,this._TemplateRef_2_5,this.parentView.parentView.injectorGet(import18.IterableDiffers,this.parentView.parentIndex),this.parentView.ref);
    this._text_3 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    var disposable_0:Function = import9.subscribeToRenderElement(this,this._el_0,new import9.InlineArray2(2,'change',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import24.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import25.NgFor) && (2 === requestNodeIndex))) { return this._NgFor_2_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2_0_0:any = this.parentView.context.selectDropdownOptions;
    this._NgFor_2_6.check_ngForOf(currVal_2_0_0,throwOnChange,false);
    this._NgFor_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
    const currVal_7:any = this.parentView.context.page;
    if (import9.checkBinding(throwOnChange,this._expr_7,currVal_7)) {
      this.renderer.setElementProperty(this._el_0,'value',currVal_7);
      this._expr_7 = currVal_7;
    }
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import8.AppView<any> {
    if ((nodeIndex == 2)) { return new View_Carousel4(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    return (null as any);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'change')) {
      const pd_sub_0:any = ((<any>this.parentView.context.onDropdownChange($event.target.value)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
class View_Carousel6 extends import8.AppView<any> {
  _el_0:any;
  _text_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import20.ViewContainer) {
    super(View_Carousel6,renderType_Carousel,import14.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_2 = import7.UNINITIALIZED;
    this._expr_3 = import7.UNINITIALIZED;
    this._expr_4 = import7.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'option',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1
    ]
    ),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2:any = this.context.$implicit;
    if (import9.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setElementProperty(this._el_0,'value',currVal_2);
      this._expr_2 = currVal_2;
    }
    const currVal_3:any = (this.parentView.parentView.context.value == this.context.$implicit);
    if (import9.checkBinding(throwOnChange,this._expr_3,currVal_3)) {
      this.renderer.setElementProperty(this._el_0,'selected',currVal_3);
      this._expr_3 = currVal_3;
    }
    const currVal_4:any = import9.inlineInterpolate(1,'',(this.context.$implicit + 1),'');
    if (import9.checkBinding(throwOnChange,this._expr_4,currVal_4)) {
      this.renderer.setText(this._text_1,currVal_4);
      this._expr_4 = currVal_4;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_Carousel5 extends import8.AppView<any> {
  _el_0:any;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import20.ViewContainer;
  _TemplateRef_2_5:any;
  _NgFor_2_6:import23.Wrapper_NgFor;
  _text_3:any;
  /*private*/ _expr_7:any;
  /*private*/ _expr_8:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import20.ViewContainer) {
    super(View_Carousel5,renderType_Carousel,import14.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_7 = import7.UNINITIALIZED;
    this._expr_8 = import7.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'select',new import9.InlineArray2(2,'class','ui-carousel-mobiledropdown ui-widget ui-state-default ui-corner-left'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import20.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import24.TemplateRef_(this,2,this._anchor_2);
    this._NgFor_2_6 = new import23.Wrapper_NgFor(this._vc_2.vcRef,this._TemplateRef_2_5,this.parentView.parentView.injectorGet(import18.IterableDiffers,this.parentView.parentIndex),this.parentView.ref);
    this._text_3 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    var disposable_0:Function = import9.subscribeToRenderElement(this,this._el_0,new import9.InlineArray2(2,'change',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import24.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import25.NgFor) && (2 === requestNodeIndex))) { return this._NgFor_2_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2_0_0:any = this.parentView.context.mobileDropdownOptions;
    this._NgFor_2_6.check_ngForOf(currVal_2_0_0,throwOnChange,false);
    this._NgFor_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
    const currVal_7:any = this.parentView.context.page;
    if (import9.checkBinding(throwOnChange,this._expr_7,currVal_7)) {
      this.renderer.setElementProperty(this._el_0,'value',currVal_7);
      this._expr_7 = currVal_7;
    }
    const currVal_8:any = (this.parentView.context.shrinked? 'block': 'none');
    if (import9.checkBinding(throwOnChange,this._expr_8,currVal_8)) {
      this.renderer.setElementStyle(this._el_0,'display',((this.viewUtils.sanitizer.sanitize(import26.SecurityContext.STYLE,currVal_8) == null)? (null as any): this.viewUtils.sanitizer.sanitize(import26.SecurityContext.STYLE,currVal_8).toString()));
      this._expr_8 = currVal_8;
    }
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import8.AppView<any> {
    if ((nodeIndex == 2)) { return new View_Carousel6(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    return (null as any);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'change')) {
      const pd_sub_0:any = ((<any>this.parentView.context.onDropdownChange($event.target.value)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
class View_Carousel8 extends import8.AppView<any> {
  _el_0:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import20.ViewContainer) {
    super(View_Carousel8,renderType_Carousel,import14.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    this._el_0 = this.renderer.createTemplateAnchor((null as any),(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): ([] as any[])),(null as any));
    return (null as any);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_Carousel7 extends import8.AppView<any> {
  _el_0:any;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import20.ViewContainer;
  _TemplateRef_2_5:any;
  _TemplateWrapper_2_6:import27.Wrapper_TemplateWrapper;
  _text_3:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import20.ViewContainer) {
    super(View_Carousel7,renderType_Carousel,import14.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'li',new import9.InlineArray2(2,'class','ui-carousel-item ui-widget-content ui-corner-all'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                        ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import20.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import24.TemplateRef_(this,2,this._anchor_2);
    this._TemplateWrapper_2_6 = new import27.Wrapper_TemplateWrapper(this._vc_2.vcRef);
    this._text_3 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import24.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import3.TemplateWrapper) && (2 === requestNodeIndex))) { return this._TemplateWrapper_2_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2_0_0:any = this.context.$implicit;
    this._TemplateWrapper_2_6.check_item(currVal_2_0_0,throwOnChange,false);
    const currVal_2_0_1:any = this.parentView.context.itemTemplate;
    this._TemplateWrapper_2_6.check_templateRef(currVal_2_0_1,throwOnChange,false);
    this._TemplateWrapper_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import8.AppView<any> {
    if ((nodeIndex == 2)) { return new View_Carousel8(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    return (null as any);
  }
}
var renderType_Carousel:import10.RenderComponentType = import9.createRenderComponentType('',0,import11.ViewEncapsulation.None,styles_Carousel,{});
export class View_Carousel0 extends import8.AppView<import1.Carousel> {
  _text_0:any;
  _el_1:any;
  _NgClass_1_3:import19.Wrapper_NgClass;
  _NgStyle_1_4:import28.Wrapper_NgStyle;
  _text_2:any;
  _el_3:any;
  _text_4:any;
  _el_5:any;
  _text_6:any;
  _text_7:any;
  _el_8:any;
  _NgClass_8_3:import19.Wrapper_NgClass;
  _text_9:any;
  _el_10:any;
  _NgClass_10_3:import19.Wrapper_NgClass;
  _text_11:any;
  _anchor_12:any;
  /*private*/ _vc_12:import20.ViewContainer;
  _TemplateRef_12_5:any;
  _NgIf_12_6:import29.Wrapper_NgIf;
  _text_13:any;
  _anchor_14:any;
  /*private*/ _vc_14:import20.ViewContainer;
  _TemplateRef_14_5:any;
  _NgIf_14_6:import29.Wrapper_NgIf;
  _text_15:any;
  _anchor_16:any;
  /*private*/ _vc_16:import20.ViewContainer;
  _TemplateRef_16_5:any;
  _NgIf_16_6:import29.Wrapper_NgIf;
  _text_17:any;
  _text_18:any;
  _el_19:any;
  _text_20:any;
  _el_21:any;
  _text_22:any;
  _anchor_23:any;
  /*private*/ _vc_23:import20.ViewContainer;
  _TemplateRef_23_5:any;
  _NgFor_23_6:import23.Wrapper_NgFor;
  _text_24:any;
  _text_25:any;
  _text_26:any;
  _text_27:any;
  _map_44:any;
  /*private*/ _expr_45:any;
  _map_46:any;
  _map_47:any;
  /*private*/ _expr_48:any;
  /*private*/ _expr_49:any;
  /*private*/ _expr_50:any;
  /*private*/ _expr_51:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_Carousel0,renderType_Carousel,import14.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways);
    this._map_44 = import9.pureProxy1((p0:any):{[key: string]:any} => {
      return {'ui-carousel ui-widget ui-widget-content ui-corner-all': p0};
    });
    this._expr_45 = import7.UNINITIALIZED;
    this._map_46 = import9.pureProxy1((p0:any):{[key: string]:any} => {
      return {'ui-state-disabled': p0};
    });
    this._map_47 = import9.pureProxy1((p0:any):{[key: string]:any} => {
      return {'ui-state-disabled': p0};
    });
    this._expr_48 = import7.UNINITIALIZED;
    this._expr_49 = import7.UNINITIALIZED;
    this._expr_50 = import7.UNINITIALIZED;
    this._expr_51 = import7.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n        ',(null as any));
    this._el_1 = import9.createRenderElement(this.renderer,parentRenderNode,'div',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_1_3 = new import19.Wrapper_NgClass(this.parentView.injectorGet(import18.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import21.KeyValueDiffers,this.parentIndex),new import17.ElementRef(this._el_1),this.renderer);
    this._NgStyle_1_4 = new import28.Wrapper_NgStyle(this.parentView.injectorGet(import21.KeyValueDiffers,this.parentIndex),new import17.ElementRef(this._el_1),this.renderer);
    this._text_2 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_3 = import9.createRenderElement(this.renderer,this._el_1,'div',new import9.InlineArray2(2,'class','ui-carousel-header ui-widget-header ui-corner-all'),(null as any));
    this._text_4 = this.renderer.createText(this._el_3,'\n                ',(null as any));
    this._el_5 = import9.createRenderElement(this.renderer,this._el_3,'span',new import9.InlineArray2(2,'class','ui-carousel-header-title'),(null as any));
    this._text_6 = this.renderer.createText(this._el_5,'',(null as any));
    this._text_7 = this.renderer.createText(this._el_3,'\n                ',(null as any));
    this._el_8 = import9.createRenderElement(this.renderer,this._el_3,'span',new import9.InlineArray2(2,'class','ui-carousel-button ui-carousel-next-button fa fa-arrow-circle-right'),(null as any));
    this._NgClass_8_3 = new import19.Wrapper_NgClass(this.parentView.injectorGet(import18.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import21.KeyValueDiffers,this.parentIndex),new import17.ElementRef(this._el_8),this.renderer);
    this._text_9 = this.renderer.createText(this._el_3,'\n                ',(null as any));
    this._el_10 = import9.createRenderElement(this.renderer,this._el_3,'span',new import9.InlineArray2(2,'class','ui-carousel-button ui-carousel-prev-button fa fa-arrow-circle-left'),(null as any));
    this._NgClass_10_3 = new import19.Wrapper_NgClass(this.parentView.injectorGet(import18.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import21.KeyValueDiffers,this.parentIndex),new import17.ElementRef(this._el_10),this.renderer);
    this._text_11 = this.renderer.createText(this._el_3,'\n                ',(null as any));
    this._anchor_12 = this.renderer.createTemplateAnchor(this._el_3,(null as any));
    this._vc_12 = new import20.ViewContainer(12,3,this,this._anchor_12);
    this._TemplateRef_12_5 = new import24.TemplateRef_(this,12,this._anchor_12);
    this._NgIf_12_6 = new import29.Wrapper_NgIf(this._vc_12.vcRef,this._TemplateRef_12_5);
    this._text_13 = this.renderer.createText(this._el_3,'\n                ',(null as any));
    this._anchor_14 = this.renderer.createTemplateAnchor(this._el_3,(null as any));
    this._vc_14 = new import20.ViewContainer(14,3,this,this._anchor_14);
    this._TemplateRef_14_5 = new import24.TemplateRef_(this,14,this._anchor_14);
    this._NgIf_14_6 = new import29.Wrapper_NgIf(this._vc_14.vcRef,this._TemplateRef_14_5);
    this._text_15 = this.renderer.createText(this._el_3,'\n                ',(null as any));
    this._anchor_16 = this.renderer.createTemplateAnchor(this._el_3,(null as any));
    this._vc_16 = new import20.ViewContainer(16,3,this,this._anchor_16);
    this._TemplateRef_16_5 = new import24.TemplateRef_(this,16,this._anchor_16);
    this._NgIf_16_6 = new import29.Wrapper_NgIf(this._vc_16.vcRef,this._TemplateRef_16_5);
    this._text_17 = this.renderer.createText(this._el_3,'\n            ',(null as any));
    this._text_18 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_19 = import9.createRenderElement(this.renderer,this._el_1,'div',new import9.InlineArray2(2,'class','ui-carousel-viewport'),(null as any));
    this._text_20 = this.renderer.createText(this._el_19,'\n                ',(null as any));
    this._el_21 = import9.createRenderElement(this.renderer,this._el_19,'ul',new import9.InlineArray2(2,'class','ui-carousel-items'),(null as any));
    this._text_22 = this.renderer.createText(this._el_21,'\n                    ',(null as any));
    this._anchor_23 = this.renderer.createTemplateAnchor(this._el_21,(null as any));
    this._vc_23 = new import20.ViewContainer(23,21,this,this._anchor_23);
    this._TemplateRef_23_5 = new import24.TemplateRef_(this,23,this._anchor_23);
    this._NgFor_23_6 = new import23.Wrapper_NgFor(this._vc_23.vcRef,this._TemplateRef_23_5,this.parentView.injectorGet(import18.IterableDiffers,this.parentIndex),this.ref);
    this._text_24 = this.renderer.createText(this._el_21,'\n                ',(null as any));
    this._text_25 = this.renderer.createText(this._el_19,'\n            ',(null as any));
    this._text_26 = this.renderer.createText(this._el_1,'\n        ',(null as any));
    this._text_27 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    var disposable_0:Function = import9.subscribeToRenderElement(this,this._el_8,new import9.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_8));
    var disposable_1:Function = import9.subscribeToRenderElement(this,this._el_10,new import9.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_10));
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2,
      this._el_3,
      this._text_4,
      this._el_5,
      this._text_6,
      this._text_7,
      this._el_8,
      this._text_9,
      this._el_10,
      this._text_11,
      this._anchor_12,
      this._text_13,
      this._anchor_14,
      this._text_15,
      this._anchor_16,
      this._text_17,
      this._text_18,
      this._el_19,
      this._text_20,
      this._el_21,
      this._text_22,
      this._anchor_23,
      this._text_24,
      this._text_25,
      this._text_26,
      this._text_27
    ]
    ),[
      disposable_0,
      disposable_1
    ]
    );
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import22.NgClass) && (8 === requestNodeIndex))) { return this._NgClass_8_3.context; }
    if (((token === import22.NgClass) && (10 === requestNodeIndex))) { return this._NgClass_10_3.context; }
    if (((token === import24.TemplateRef) && (12 === requestNodeIndex))) { return this._TemplateRef_12_5; }
    if (((token === import30.NgIf) && (12 === requestNodeIndex))) { return this._NgIf_12_6.context; }
    if (((token === import24.TemplateRef) && (14 === requestNodeIndex))) { return this._TemplateRef_14_5; }
    if (((token === import30.NgIf) && (14 === requestNodeIndex))) { return this._NgIf_14_6.context; }
    if (((token === import24.TemplateRef) && (16 === requestNodeIndex))) { return this._TemplateRef_16_5; }
    if (((token === import30.NgIf) && (16 === requestNodeIndex))) { return this._NgIf_16_6.context; }
    if (((token === import24.TemplateRef) && (23 === requestNodeIndex))) { return this._TemplateRef_23_5; }
    if (((token === import25.NgFor) && (23 === requestNodeIndex))) { return this._NgFor_23_6.context; }
    if (((token === import22.NgClass) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 26)))) { return this._NgClass_1_3.context; }
    if (((token === import31.NgStyle) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 26)))) { return this._NgStyle_1_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1_0_0:any = this.context.styleClass;
    this._NgClass_1_3.check_klass(currVal_1_0_0,throwOnChange,false);
    const currVal_1_0_1:any = this._map_44(true);
    this._NgClass_1_3.check_ngClass(currVal_1_0_1,throwOnChange,false);
    this._NgClass_1_3.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_1_1_0:any = this.context.style;
    this._NgStyle_1_4.check_ngStyle(currVal_1_1_0,throwOnChange,false);
    this._NgStyle_1_4.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_8_0_0:any = 'ui-carousel-button ui-carousel-next-button fa fa-arrow-circle-right';
    this._NgClass_8_3.check_klass(currVal_8_0_0,throwOnChange,false);
    const currVal_8_0_1:any = this._map_46(((this.context.page === (this.context.totalPages - 1)) && !this.context.circular));
    this._NgClass_8_3.check_ngClass(currVal_8_0_1,throwOnChange,false);
    this._NgClass_8_3.ngDoCheck(this,this._el_8,throwOnChange);
    const currVal_10_0_0:any = 'ui-carousel-button ui-carousel-prev-button fa fa-arrow-circle-left';
    this._NgClass_10_3.check_klass(currVal_10_0_0,throwOnChange,false);
    const currVal_10_0_1:any = this._map_47(((this.context.page === 0) && !this.context.circular));
    this._NgClass_10_3.check_ngClass(currVal_10_0_1,throwOnChange,false);
    this._NgClass_10_3.ngDoCheck(this,this._el_10,throwOnChange);
    const currVal_12_0_0:any = this.context.displayPageLinks;
    this._NgIf_12_6.check_ngIf(currVal_12_0_0,throwOnChange,false);
    this._NgIf_12_6.ngDoCheck(this,this._anchor_12,throwOnChange);
    const currVal_14_0_0:any = this.context.displayPageDropdown;
    this._NgIf_14_6.check_ngIf(currVal_14_0_0,throwOnChange,false);
    this._NgIf_14_6.ngDoCheck(this,this._anchor_14,throwOnChange);
    const currVal_16_0_0:any = this.context.responsive;
    this._NgIf_16_6.check_ngIf(currVal_16_0_0,throwOnChange,false);
    this._NgIf_16_6.ngDoCheck(this,this._anchor_16,throwOnChange);
    const currVal_23_0_0:any = this.context.value;
    this._NgFor_23_6.check_ngForOf(currVal_23_0_0,throwOnChange,false);
    this._NgFor_23_6.ngDoCheck(this,this._anchor_23,throwOnChange);
    this._vc_12.detectChangesInNestedViews(throwOnChange);
    this._vc_14.detectChangesInNestedViews(throwOnChange);
    this._vc_16.detectChangesInNestedViews(throwOnChange);
    this._vc_23.detectChangesInNestedViews(throwOnChange);
    const currVal_45:any = import9.inlineInterpolate(1,'',this.context.headerText,'');
    if (import9.checkBinding(throwOnChange,this._expr_45,currVal_45)) {
      this.renderer.setText(this._text_6,currVal_45);
      this._expr_45 = currVal_45;
    }
    const currVal_48:any = this.context.left;
    if (import9.checkBinding(throwOnChange,this._expr_48,currVal_48)) {
      this.renderer.setElementStyle(this._el_21,'left',((this.viewUtils.sanitizer.sanitize(import26.SecurityContext.STYLE,currVal_48) == null)? (null as any): (this.viewUtils.sanitizer.sanitize(import26.SecurityContext.STYLE,currVal_48).toString() + 'px')));
      this._expr_48 = currVal_48;
    }
    const currVal_49:any = 'left';
    if (import9.checkBinding(throwOnChange,this._expr_49,currVal_49)) {
      this.renderer.setElementStyle(this._el_21,'transitionProperty',((this.viewUtils.sanitizer.sanitize(import26.SecurityContext.STYLE,currVal_49) == null)? (null as any): this.viewUtils.sanitizer.sanitize(import26.SecurityContext.STYLE,currVal_49).toString()));
      this._expr_49 = currVal_49;
    }
    const currVal_50:any = this.context.effectDuration;
    if (import9.checkBinding(throwOnChange,this._expr_50,currVal_50)) {
      this.renderer.setElementStyle(this._el_21,'transitionDuration',((this.viewUtils.sanitizer.sanitize(import26.SecurityContext.STYLE,currVal_50) == null)? (null as any): this.viewUtils.sanitizer.sanitize(import26.SecurityContext.STYLE,currVal_50).toString()));
      this._expr_50 = currVal_50;
    }
    const currVal_51:any = this.context.easing;
    if (import9.checkBinding(throwOnChange,this._expr_51,currVal_51)) {
      this.renderer.setElementStyle(this._el_21,'transitionTimingFunction',((this.viewUtils.sanitizer.sanitize(import26.SecurityContext.STYLE,currVal_51) == null)? (null as any): this.viewUtils.sanitizer.sanitize(import26.SecurityContext.STYLE,currVal_51).toString()));
      this._expr_51 = currVal_51;
    }
  }
  destroyInternal():void {
    this._vc_12.destroyNestedViews();
    this._vc_14.destroyNestedViews();
    this._vc_16.destroyNestedViews();
    this._vc_23.destroyNestedViews();
  }
  createEmbeddedViewInternal(nodeIndex:number):import8.AppView<any> {
    if ((nodeIndex == 12)) { return new View_Carousel1(this.viewUtils,this,12,this._anchor_12,this._vc_12); }
    if ((nodeIndex == 14)) { return new View_Carousel3(this.viewUtils,this,14,this._anchor_14,this._vc_14); }
    if ((nodeIndex == 16)) { return new View_Carousel5(this.viewUtils,this,16,this._anchor_16,this._vc_16); }
    if ((nodeIndex == 23)) { return new View_Carousel7(this.viewUtils,this,23,this._anchor_23,this._vc_23); }
    return (null as any);
  }
  handleEvent_8(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.context.onNextNav()) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
  handleEvent_10(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.context.onPrevNav()) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
