// primeng/components/datagrid/datagrid.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from '@angular/core/src/linker/ng_module_factory';
import * as import1 from 'primeng/components/datagrid/datagrid';
import * as import2 from '@angular/common/src/common_module';
import * as import3 from 'primeng/components/paginator/paginator';
import * as import4 from '@angular/common/src/localization';
import * as import5 from '@angular/core/src/di/injector';
import * as import6 from '@angular/core/src/i18n/tokens';
import * as import7 from '@angular/core/src/change_detection/change_detection_util';
import * as import8 from '@angular/core/src/linker/view';
import * as import9 from '@angular/core/src/linker/view_utils';
import * as import10 from '@angular/core/src/render/api';
import * as import11 from '@angular/core/src/metadata/view';
import * as import12 from '@angular/core/src/linker/query_list';
import * as import13 from '@angular/core/src/linker/view_type';
import * as import14 from '@angular/core/src/change_detection/constants';
import * as import15 from '@angular/core/src/linker/component_factory';
import * as import16 from '@angular/core/src/linker/element_ref';
import * as import17 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import18 from '@angular/core/src/linker/view_container';
import * as import19 from '../paginator/paginator.ngfactory';
import * as import20 from '../../../@angular/common/src/directives/ng_class.ngfactory';
import * as import21 from '../../../@angular/common/src/directives/ng_style.ngfactory';
import * as import22 from '../../../@angular/common/src/directives/ng_if.ngfactory';
import * as import23 from '../../../@angular/common/src/directives/ng_for.ngfactory';
import * as import24 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import25 from '@angular/core/src/linker/template_ref';
import * as import26 from '@angular/common/src/directives/ng_if';
import * as import27 from '@angular/common/src/directives/ng_for';
import * as import28 from '@angular/common/src/directives/ng_class';
import * as import29 from '@angular/common/src/directives/ng_style';
class DataGridModuleInjector extends import0.NgModuleInjector<import1.DataGridModule> {
  _CommonModule_0:import2.CommonModule;
  _PaginatorModule_1:import3.PaginatorModule;
  _DataGridModule_2:import1.DataGridModule;
  __NgLocalization_3:import4.NgLocaleLocalization;
  constructor(parent:import5.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_3():import4.NgLocaleLocalization {
    if ((this.__NgLocalization_3 == null)) { (this.__NgLocalization_3 = new import4.NgLocaleLocalization(this.parent.get(import6.LOCALE_ID))); }
    return this.__NgLocalization_3;
  }
  createInternal():import1.DataGridModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._PaginatorModule_1 = new import3.PaginatorModule();
    this._DataGridModule_2 = new import1.DataGridModule();
    return this._DataGridModule_2;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.PaginatorModule)) { return this._PaginatorModule_1; }
    if ((token === import1.DataGridModule)) { return this._DataGridModule_2; }
    if ((token === import4.NgLocalization)) { return this._NgLocalization_3; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const DataGridModuleNgFactory:import0.NgModuleFactory<import1.DataGridModule> = new import0.NgModuleFactory(DataGridModuleInjector,import1.DataGridModule);
export class Wrapper_DataGrid {
  /*private*/ _eventHandler:Function;
  context:import1.DataGrid;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  /*private*/ _expr_5:any;
  /*private*/ _expr_6:any;
  /*private*/ _expr_7:any;
  /*private*/ _expr_8:any;
  /*private*/ _expr_9:any;
  subscription0:any;
  constructor(p0:any,p1:any) {
    this._changed = false;
    this.context = new import1.DataGrid(p0,p1);
    this._expr_0 = import7.UNINITIALIZED;
    this._expr_1 = import7.UNINITIALIZED;
    this._expr_2 = import7.UNINITIALIZED;
    this._expr_3 = import7.UNINITIALIZED;
    this._expr_4 = import7.UNINITIALIZED;
    this._expr_5 = import7.UNINITIALIZED;
    this._expr_6 = import7.UNINITIALIZED;
    this._expr_7 = import7.UNINITIALIZED;
    this._expr_8 = import7.UNINITIALIZED;
    this._expr_9 = import7.UNINITIALIZED;
  }
  ngOnDetach(view:import8.AppView<any>,componentView:import8.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
    (this.subscription0 && this.subscription0.unsubscribe());
  }
  check_value(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.value = currValue;
      this._expr_0 = currValue;
    }
  }
  check_paginator(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.paginator = currValue;
      this._expr_1 = currValue;
    }
  }
  check_rows(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_2,currValue))) {
      this._changed = true;
      this.context.rows = currValue;
      this._expr_2 = currValue;
    }
  }
  check_totalRecords(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_3,currValue))) {
      this._changed = true;
      this.context.totalRecords = currValue;
      this._expr_3 = currValue;
    }
  }
  check_pageLinks(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_4,currValue))) {
      this._changed = true;
      this.context.pageLinks = currValue;
      this._expr_4 = currValue;
    }
  }
  check_rowsPerPageOptions(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_5,currValue))) {
      this._changed = true;
      this.context.rowsPerPageOptions = currValue;
      this._expr_5 = currValue;
    }
  }
  check_lazy(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_6,currValue))) {
      this._changed = true;
      this.context.lazy = currValue;
      this._expr_6 = currValue;
    }
  }
  check_style(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_7,currValue))) {
      this._changed = true;
      this.context.style = currValue;
      this._expr_7 = currValue;
    }
  }
  check_styleClass(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_8,currValue))) {
      this._changed = true;
      this.context.styleClass = currValue;
      this._expr_8 = currValue;
    }
  }
  check_paginatorPosition(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import9.checkBinding(throwOnChange,this._expr_9,currValue))) {
      this._changed = true;
      this.context.paginatorPosition = currValue;
      this._expr_9 = currValue;
    }
  }
  ngDoCheck(view:import8.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    if (!throwOnChange) { this.context.ngDoCheck(); }
    return changed;
  }
  checkHost(view:import8.AppView<any>,componentView:import8.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import8.AppView<any>,_eventHandler:any,emit0:boolean):void {
    this._eventHandler = _eventHandler;
    if (emit0) { (this.subscription0 = this.context.onLazyLoad.subscribe(_eventHandler.bind(view,'onLazyLoad'))); }
  }
}
var renderType_DataGrid_Host:import10.RenderComponentType = import9.createRenderComponentType('',0,import11.ViewEncapsulation.None,([] as any[]),{});
class View_DataGrid_Host0 extends import8.AppView<any> {
  _el_0:any;
  compView_0:import8.AppView<import1.DataGrid>;
  _DataGrid_0_3:Wrapper_DataGrid;
  _query_Header_0_0:import12.QueryList<any>;
  _query_Footer_0_1:import12.QueryList<any>;
  _query_TemplateRef_0_2:import12.QueryList<any>;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_DataGrid_Host0,renderType_DataGrid_Host,import13.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.selectOrCreateRenderHostElement(this.renderer,'p-dataGrid',import9.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this.compView_0 = new View_DataGrid0(this.viewUtils,this,0,this._el_0);
    this._DataGrid_0_3 = new Wrapper_DataGrid(new import16.ElementRef(this._el_0),this.injectorGet(import17.IterableDiffers,this.parentIndex));
    this._query_Header_0_0 = new import12.QueryList<any>();
    this._query_Footer_0_1 = new import12.QueryList<any>();
    this._query_TemplateRef_0_2 = new import12.QueryList<any>();
    this.compView_0.create(this._DataGrid_0_3.context);
    this._query_Header_0_0.reset(([] as any[]));
    this._DataGrid_0_3.context.header = this._query_Header_0_0.first;
    this._query_Footer_0_1.reset(([] as any[]));
    this._DataGrid_0_3.context.footer = this._query_Footer_0_1.first;
    this._query_TemplateRef_0_2.reset(([] as any[]));
    this._DataGrid_0_3.context.itemTemplate = this._query_TemplateRef_0_2.first;
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import15.ComponentRef_<any>(0,this,this._el_0,this._DataGrid_0_3.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import1.DataGrid) && (0 === requestNodeIndex))) { return this._DataGrid_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._DataGrid_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
    if (!throwOnChange) { if ((this.numberOfChecks === 0)) { this._DataGrid_0_3.context.ngAfterViewInit(); } }
  }
  destroyInternal():void {
    this.compView_0.destroy();
    this._DataGrid_0_3.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  visitProjectableNodesInternal(nodeIndex:number,ngContentIndex:number,cb:any,ctx:any):void {
    if (((nodeIndex == 0) && (ngContentIndex == 0))) {  }
    if (((nodeIndex == 0) && (ngContentIndex == 1))) {  }
  }
}
export const DataGridNgFactory:import15.ComponentFactory<import1.DataGrid> = new import15.ComponentFactory<import1.DataGrid>('p-dataGrid',View_DataGrid_Host0,import1.DataGrid);
const styles_DataGrid:any[] = ([] as any[]);
class View_DataGrid1 extends import8.AppView<any> {
  _el_0:any;
  _text_1:any;
  _text_2:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import18.ViewContainer) {
    super(View_DataGrid1,renderType_DataGrid,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'div',new import9.InlineArray2(2,'class','ui-datagrid-header ui-widget-header ui-corner-top'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    this.projectNodes(this._el_0,0);
    this._text_2 = this.renderer.createText(this._el_0,'\n            ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._text_2
    ]
    ),(null as any));
    return (null as any);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_DataGrid2 extends import8.AppView<any> {
  _el_0:any;
  compView_0:import8.AppView<import3.Paginator>;
  _Paginator_0_3:import19.Wrapper_Paginator;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import18.ViewContainer) {
    super(View_DataGrid2,renderType_DataGrid,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'p-paginator',new import9.InlineArray2(2,'styleClass','ui-paginator-bottom'),(null as any));
    this.compView_0 = new import19.View_Paginator0(this.viewUtils,this,0,this._el_0);
    this._Paginator_0_3 = new import19.Wrapper_Paginator();
    this.compView_0.create(this._Paginator_0_3.context);
    var disposable_0:Function = import9.subscribeToRenderElement(this,this._el_0,new import9.InlineArray2(2,'onPageChange',(null as any)),this.eventHandler(this.handleEvent_0));
    this._Paginator_0_3.subscribe(this,this.eventHandler(this.handleEvent_0),true);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import3.Paginator) && (0 === requestNodeIndex))) { return this._Paginator_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this.parentView.context.rows;
    this._Paginator_0_3.check_rows(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = this.parentView.context.pageLinks;
    this._Paginator_0_3.check_pageLinkSize(currVal_0_0_1,throwOnChange,false);
    const currVal_0_0_2:any = 'ui-paginator-bottom';
    this._Paginator_0_3.check_styleClass(currVal_0_0_2,throwOnChange,false);
    const currVal_0_0_3:any = this.parentView.context.rowsPerPageOptions;
    this._Paginator_0_3.check_rowsPerPageOptions(currVal_0_0_3,throwOnChange,false);
    const currVal_0_0_4:any = this.parentView.context.totalRecords;
    this._Paginator_0_3.check_totalRecords(currVal_0_0_4,throwOnChange,false);
    const currVal_0_0_5:any = this.parentView.context.first;
    this._Paginator_0_3.check_first(currVal_0_0_5,throwOnChange,false);
    this._Paginator_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
    this._Paginator_0_3.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'onPageChange')) {
      const pd_sub_0:any = ((<any>this.parentView.context.paginate($event)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
class View_DataGrid3 extends import8.AppView<any> {
  _el_0:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import18.ViewContainer) {
    super(View_DataGrid3,renderType_DataGrid,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = this.renderer.createTemplateAnchor((null as any),(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): ([] as any[])),(null as any));
    return (null as any);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_DataGrid4 extends import8.AppView<any> {
  _el_0:any;
  compView_0:import8.AppView<import3.Paginator>;
  _Paginator_0_3:import19.Wrapper_Paginator;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import18.ViewContainer) {
    super(View_DataGrid4,renderType_DataGrid,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'p-paginator',new import9.InlineArray2(2,'styleClass','ui-paginator-bottom'),(null as any));
    this.compView_0 = new import19.View_Paginator0(this.viewUtils,this,0,this._el_0);
    this._Paginator_0_3 = new import19.Wrapper_Paginator();
    this.compView_0.create(this._Paginator_0_3.context);
    var disposable_0:Function = import9.subscribeToRenderElement(this,this._el_0,new import9.InlineArray2(2,'onPageChange',(null as any)),this.eventHandler(this.handleEvent_0));
    this._Paginator_0_3.subscribe(this,this.eventHandler(this.handleEvent_0),true);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import3.Paginator) && (0 === requestNodeIndex))) { return this._Paginator_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this.parentView.context.rows;
    this._Paginator_0_3.check_rows(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = this.parentView.context.pageLinks;
    this._Paginator_0_3.check_pageLinkSize(currVal_0_0_1,throwOnChange,false);
    const currVal_0_0_2:any = 'ui-paginator-bottom';
    this._Paginator_0_3.check_styleClass(currVal_0_0_2,throwOnChange,false);
    const currVal_0_0_3:any = this.parentView.context.rowsPerPageOptions;
    this._Paginator_0_3.check_rowsPerPageOptions(currVal_0_0_3,throwOnChange,false);
    const currVal_0_0_4:any = this.parentView.context.totalRecords;
    this._Paginator_0_3.check_totalRecords(currVal_0_0_4,throwOnChange,false);
    const currVal_0_0_5:any = this.parentView.context.first;
    this._Paginator_0_3.check_first(currVal_0_0_5,throwOnChange,false);
    this._Paginator_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
    this._Paginator_0_3.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'onPageChange')) {
      const pd_sub_0:any = ((<any>this.parentView.context.paginate($event)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
class View_DataGrid5 extends import8.AppView<any> {
  _el_0:any;
  _text_1:any;
  _text_2:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import18.ViewContainer) {
    super(View_DataGrid5,renderType_DataGrid,import13.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    this._el_0 = import9.createRenderElement(this.renderer,(null as any),'div',new import9.InlineArray2(2,'class','ui-datagrid-footer ui-widget-header ui-corner-top'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    this.projectNodes(this._el_0,1);
    this._text_2 = this.renderer.createText(this._el_0,'\n            ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._text_2
    ]
    ),(null as any));
    return (null as any);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
var renderType_DataGrid:import10.RenderComponentType = import9.createRenderComponentType('',2,import11.ViewEncapsulation.None,styles_DataGrid,{});
export class View_DataGrid0 extends import8.AppView<import1.DataGrid> {
  _text_0:any;
  _el_1:any;
  _NgClass_1_3:import20.Wrapper_NgClass;
  _NgStyle_1_4:import21.Wrapper_NgStyle;
  _text_2:any;
  _anchor_3:any;
  /*private*/ _vc_3:import18.ViewContainer;
  _TemplateRef_3_5:any;
  _NgIf_3_6:import22.Wrapper_NgIf;
  _text_4:any;
  _anchor_5:any;
  /*private*/ _vc_5:import18.ViewContainer;
  _TemplateRef_5_5:any;
  _NgIf_5_6:import22.Wrapper_NgIf;
  _text_6:any;
  _el_7:any;
  _text_8:any;
  _el_9:any;
  _text_10:any;
  _anchor_11:any;
  /*private*/ _vc_11:import18.ViewContainer;
  _TemplateRef_11_5:any;
  _NgFor_11_6:import23.Wrapper_NgFor;
  _text_12:any;
  _text_13:any;
  _text_14:any;
  _anchor_15:any;
  /*private*/ _vc_15:import18.ViewContainer;
  _TemplateRef_15_5:any;
  _NgIf_15_6:import22.Wrapper_NgIf;
  _text_16:any;
  _anchor_17:any;
  /*private*/ _vc_17:import18.ViewContainer;
  _TemplateRef_17_5:any;
  _NgIf_17_6:import22.Wrapper_NgIf;
  _text_18:any;
  _text_19:any;
  constructor(viewUtils:import9.ViewUtils,parentView:import8.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_DataGrid0,renderType_DataGrid,import13.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import14.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import15.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n        ',(null as any));
    this._el_1 = import9.createRenderElement(this.renderer,parentRenderNode,'div',import9.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_1_3 = new import20.Wrapper_NgClass(this.parentView.injectorGet(import17.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import24.KeyValueDiffers,this.parentIndex),new import16.ElementRef(this._el_1),this.renderer);
    this._NgStyle_1_4 = new import21.Wrapper_NgStyle(this.parentView.injectorGet(import24.KeyValueDiffers,this.parentIndex),new import16.ElementRef(this._el_1),this.renderer);
    this._text_2 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._anchor_3 = this.renderer.createTemplateAnchor(this._el_1,(null as any));
    this._vc_3 = new import18.ViewContainer(3,1,this,this._anchor_3);
    this._TemplateRef_3_5 = new import25.TemplateRef_(this,3,this._anchor_3);
    this._NgIf_3_6 = new import22.Wrapper_NgIf(this._vc_3.vcRef,this._TemplateRef_3_5);
    this._text_4 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._anchor_5 = this.renderer.createTemplateAnchor(this._el_1,(null as any));
    this._vc_5 = new import18.ViewContainer(5,1,this,this._anchor_5);
    this._TemplateRef_5_5 = new import25.TemplateRef_(this,5,this._anchor_5);
    this._NgIf_5_6 = new import22.Wrapper_NgIf(this._vc_5.vcRef,this._TemplateRef_5_5);
    this._text_6 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_7 = import9.createRenderElement(this.renderer,this._el_1,'div',new import9.InlineArray2(2,'class','ui-datagrid-content ui-widget-content'),(null as any));
    this._text_8 = this.renderer.createText(this._el_7,'\n                ',(null as any));
    this._el_9 = import9.createRenderElement(this.renderer,this._el_7,'div',new import9.InlineArray2(2,'class','ui-g'),(null as any));
    this._text_10 = this.renderer.createText(this._el_9,'\n                    ',(null as any));
    this._anchor_11 = this.renderer.createTemplateAnchor(this._el_9,(null as any));
    this._vc_11 = new import18.ViewContainer(11,9,this,this._anchor_11);
    this._TemplateRef_11_5 = new import25.TemplateRef_(this,11,this._anchor_11);
    this._NgFor_11_6 = new import23.Wrapper_NgFor(this._vc_11.vcRef,this._TemplateRef_11_5,this.parentView.injectorGet(import17.IterableDiffers,this.parentIndex),this.ref);
    this._text_12 = this.renderer.createText(this._el_9,'\n                ',(null as any));
    this._text_13 = this.renderer.createText(this._el_7,'\n            ',(null as any));
    this._text_14 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._anchor_15 = this.renderer.createTemplateAnchor(this._el_1,(null as any));
    this._vc_15 = new import18.ViewContainer(15,1,this,this._anchor_15);
    this._TemplateRef_15_5 = new import25.TemplateRef_(this,15,this._anchor_15);
    this._NgIf_15_6 = new import22.Wrapper_NgIf(this._vc_15.vcRef,this._TemplateRef_15_5);
    this._text_16 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._anchor_17 = this.renderer.createTemplateAnchor(this._el_1,(null as any));
    this._vc_17 = new import18.ViewContainer(17,1,this,this._anchor_17);
    this._TemplateRef_17_5 = new import25.TemplateRef_(this,17,this._anchor_17);
    this._NgIf_17_6 = new import22.Wrapper_NgIf(this._vc_17.vcRef,this._TemplateRef_17_5);
    this._text_18 = this.renderer.createText(this._el_1,'\n        ',(null as any));
    this._text_19 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2,
      this._anchor_3,
      this._text_4,
      this._anchor_5,
      this._text_6,
      this._el_7,
      this._text_8,
      this._el_9,
      this._text_10,
      this._anchor_11,
      this._text_12,
      this._text_13,
      this._text_14,
      this._anchor_15,
      this._text_16,
      this._anchor_17,
      this._text_18,
      this._text_19
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import25.TemplateRef) && (3 === requestNodeIndex))) { return this._TemplateRef_3_5; }
    if (((token === import26.NgIf) && (3 === requestNodeIndex))) { return this._NgIf_3_6.context; }
    if (((token === import25.TemplateRef) && (5 === requestNodeIndex))) { return this._TemplateRef_5_5; }
    if (((token === import26.NgIf) && (5 === requestNodeIndex))) { return this._NgIf_5_6.context; }
    if (((token === import25.TemplateRef) && (11 === requestNodeIndex))) { return this._TemplateRef_11_5; }
    if (((token === import27.NgFor) && (11 === requestNodeIndex))) { return this._NgFor_11_6.context; }
    if (((token === import25.TemplateRef) && (15 === requestNodeIndex))) { return this._TemplateRef_15_5; }
    if (((token === import26.NgIf) && (15 === requestNodeIndex))) { return this._NgIf_15_6.context; }
    if (((token === import25.TemplateRef) && (17 === requestNodeIndex))) { return this._TemplateRef_17_5; }
    if (((token === import26.NgIf) && (17 === requestNodeIndex))) { return this._NgIf_17_6.context; }
    if (((token === import28.NgClass) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 18)))) { return this._NgClass_1_3.context; }
    if (((token === import29.NgStyle) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 18)))) { return this._NgStyle_1_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1_0_0:any = this.context.styleClass;
    this._NgClass_1_3.check_klass(currVal_1_0_0,throwOnChange,false);
    const currVal_1_0_1:any = 'ui-datagrid ui-widget';
    this._NgClass_1_3.check_ngClass(currVal_1_0_1,throwOnChange,false);
    this._NgClass_1_3.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_1_1_0:any = this.context.style;
    this._NgStyle_1_4.check_ngStyle(currVal_1_1_0,throwOnChange,false);
    this._NgStyle_1_4.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_3_0_0:any = this.context.header;
    this._NgIf_3_6.check_ngIf(currVal_3_0_0,throwOnChange,false);
    this._NgIf_3_6.ngDoCheck(this,this._anchor_3,throwOnChange);
    const currVal_5_0_0:any = ((this.context.paginator && (this.context.paginatorPosition != 'bottom')) || (this.context.paginatorPosition == 'both'));
    this._NgIf_5_6.check_ngIf(currVal_5_0_0,throwOnChange,false);
    this._NgIf_5_6.ngDoCheck(this,this._anchor_5,throwOnChange);
    const currVal_11_0_0:any = this.context.dataToRender;
    this._NgFor_11_6.check_ngForOf(currVal_11_0_0,throwOnChange,false);
    const currVal_11_0_1:any = this.context.itemTemplate;
    this._NgFor_11_6.check_ngForTemplate(currVal_11_0_1,throwOnChange,false);
    this._NgFor_11_6.ngDoCheck(this,this._anchor_11,throwOnChange);
    const currVal_15_0_0:any = ((this.context.paginator && (this.context.paginatorPosition != 'top')) || (this.context.paginatorPosition == 'both'));
    this._NgIf_15_6.check_ngIf(currVal_15_0_0,throwOnChange,false);
    this._NgIf_15_6.ngDoCheck(this,this._anchor_15,throwOnChange);
    const currVal_17_0_0:any = this.context.footer;
    this._NgIf_17_6.check_ngIf(currVal_17_0_0,throwOnChange,false);
    this._NgIf_17_6.ngDoCheck(this,this._anchor_17,throwOnChange);
    this._vc_3.detectChangesInNestedViews(throwOnChange);
    this._vc_5.detectChangesInNestedViews(throwOnChange);
    this._vc_11.detectChangesInNestedViews(throwOnChange);
    this._vc_15.detectChangesInNestedViews(throwOnChange);
    this._vc_17.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_3.destroyNestedViews();
    this._vc_5.destroyNestedViews();
    this._vc_11.destroyNestedViews();
    this._vc_15.destroyNestedViews();
    this._vc_17.destroyNestedViews();
  }
  createEmbeddedViewInternal(nodeIndex:number):import8.AppView<any> {
    if ((nodeIndex == 3)) { return new View_DataGrid1(this.viewUtils,this,3,this._anchor_3,this._vc_3); }
    if ((nodeIndex == 5)) { return new View_DataGrid2(this.viewUtils,this,5,this._anchor_5,this._vc_5); }
    if ((nodeIndex == 11)) { return new View_DataGrid3(this.viewUtils,this,11,this._anchor_11,this._vc_11); }
    if ((nodeIndex == 15)) { return new View_DataGrid4(this.viewUtils,this,15,this._anchor_15,this._vc_15); }
    if ((nodeIndex == 17)) { return new View_DataGrid5(this.viewUtils,this,17,this._anchor_17,this._vc_17); }
    return (null as any);
  }
}
