// angular2-material-datepicker/src/datepicker.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from 'angular2-material-datepicker/src/datepicker.component';
import * as import1 from '@angular/core/src/change_detection/change_detection_util';
import * as import2 from '@angular/core/src/linker/view';
import * as import3 from '@angular/core/src/linker/view_utils';
import * as import4 from '@angular/core/src/render/api';
import * as import5 from '@angular/core/src/metadata/view';
import * as import6 from '@angular/core/src/linker/view_type';
import * as import7 from '@angular/core/src/change_detection/constants';
import * as import8 from '@angular/core/src/linker/component_factory';
import * as import9 from '@angular/core/src/linker/element_ref';
import * as import10 from '@angular/core/src/animation/animation_transition';
import * as import11 from '@angular/core/src/animation/animation_sequence_player';
import * as import12 from '@angular/core/src/animation/animation_styles';
import * as import13 from '@angular/core/src/animation/animation_style_util';
import * as import14 from '@angular/core/src/animation/animation_keyframe';
import * as import15 from '@angular/core/src/animation/animation_player';
import * as import16 from '@angular/core/src/linker/view_container';
import * as import17 from '../../@angular/common/src/directives/ng_style.ngfactory';
import * as import18 from '../../@angular/common/src/directives/ng_if.ngfactory';
import * as import19 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import20 from '@angular/core/src/linker/template_ref';
import * as import21 from '@angular/common/src/directives/ng_if';
import * as import22 from '@angular/common/src/directives/ng_style';
import * as import23 from '../../@angular/common/src/directives/ng_for.ngfactory';
import * as import24 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import25 from '@angular/common/src/directives/ng_for';
import * as import26 from '../../@angular/forms/src/directives/default_value_accessor.ngfactory';
import * as import27 from '../../@angular/forms/src/directives/ng_model.ngfactory';
import * as import28 from '../../@angular/forms/src/directives/ng_control_status.ngfactory';
import * as import29 from '@angular/forms/src/directives/default_value_accessor';
import * as import30 from '@angular/forms/src/directives/control_value_accessor';
import * as import31 from '@angular/forms/src/directives/ng_model';
import * as import32 from '@angular/forms/src/directives/ng_control';
import * as import33 from '@angular/forms/src/directives/ng_control_status';
export class Wrapper_DatepickerComponent {
  /*private*/ _eventHandler:Function;
  context:import0.DatepickerComponent;
  /*private*/ _changed:boolean;
  /*private*/ _changes:{[key: string]:any};
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  /*private*/ _expr_5:any;
  /*private*/ _expr_6:any;
  /*private*/ _expr_7:any;
  /*private*/ _expr_8:any;
  /*private*/ _expr_9:any;
  /*private*/ _expr_10:any;
  /*private*/ _expr_11:any;
  /*private*/ _expr_12:any;
  /*private*/ _expr_13:any;
  /*private*/ _expr_14:any;
  subscription0:any;
  subscription1:any;
  constructor(p0:any,p1:any) {
    this._changed = false;
    this._changes = {};
    this.context = new import0.DatepickerComponent(p0,p1);
    this._expr_0 = import1.UNINITIALIZED;
    this._expr_1 = import1.UNINITIALIZED;
    this._expr_2 = import1.UNINITIALIZED;
    this._expr_3 = import1.UNINITIALIZED;
    this._expr_4 = import1.UNINITIALIZED;
    this._expr_5 = import1.UNINITIALIZED;
    this._expr_6 = import1.UNINITIALIZED;
    this._expr_7 = import1.UNINITIALIZED;
    this._expr_8 = import1.UNINITIALIZED;
    this._expr_9 = import1.UNINITIALIZED;
    this._expr_10 = import1.UNINITIALIZED;
    this._expr_11 = import1.UNINITIALIZED;
    this._expr_12 = import1.UNINITIALIZED;
    this._expr_13 = import1.UNINITIALIZED;
    this._expr_14 = import1.UNINITIALIZED;
  }
  ngOnDetach(view:import2.AppView<any>,componentView:import2.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
    this.context.ngOnDestroy();
    (this.subscription0 && this.subscription0.unsubscribe());
    (this.subscription1 && this.subscription1.unsubscribe());
  }
  check_date(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.date = currValue;
      this._changes['date'] = new import1.SimpleChange(this._expr_0,currValue);
      this._expr_0 = currValue;
    }
  }
  check_disabled(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.disabled = currValue;
      this._changes['disabled'] = new import1.SimpleChange(this._expr_1,currValue);
      this._expr_1 = currValue;
    }
  }
  check_accentColor(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_2,currValue))) {
      this._changed = true;
      this.context.accentColor = currValue;
      this._changes['accentColor'] = new import1.SimpleChange(this._expr_2,currValue);
      this._expr_2 = currValue;
    }
  }
  check_altInputStyle(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_3,currValue))) {
      this._changed = true;
      this.context.altInputStyle = currValue;
      this._changes['altInputStyle'] = new import1.SimpleChange(this._expr_3,currValue);
      this._expr_3 = currValue;
    }
  }
  check_dateFormat(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_4,currValue))) {
      this._changed = true;
      this.context.dateFormat = currValue;
      this._changes['dateFormat'] = new import1.SimpleChange(this._expr_4,currValue);
      this._expr_4 = currValue;
    }
  }
  check_fontFamily(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_5,currValue))) {
      this._changed = true;
      this.context.fontFamily = currValue;
      this._changes['fontFamily'] = new import1.SimpleChange(this._expr_5,currValue);
      this._expr_5 = currValue;
    }
  }
  check_rangeStart(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_6,currValue))) {
      this._changed = true;
      this.context.rangeStart = currValue;
      this._changes['rangeStart'] = new import1.SimpleChange(this._expr_6,currValue);
      this._expr_6 = currValue;
    }
  }
  check_rangeEnd(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_7,currValue))) {
      this._changed = true;
      this.context.rangeEnd = currValue;
      this._changes['rangeEnd'] = new import1.SimpleChange(this._expr_7,currValue);
      this._expr_7 = currValue;
    }
  }
  check_placeholder(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_8,currValue))) {
      this._changed = true;
      this.context.placeholder = currValue;
      this._changes['placeholder'] = new import1.SimpleChange(this._expr_8,currValue);
      this._expr_8 = currValue;
    }
  }
  check_inputText(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_9,currValue))) {
      this._changed = true;
      this.context.inputText = currValue;
      this._changes['inputText'] = new import1.SimpleChange(this._expr_9,currValue);
      this._expr_9 = currValue;
    }
  }
  check_showCalendar(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_10,currValue))) {
      this._changed = true;
      this.context.showCalendar = currValue;
      this._changes['showCalendar'] = new import1.SimpleChange(this._expr_10,currValue);
      this._expr_10 = currValue;
    }
  }
  check_calendarDays(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_11,currValue))) {
      this._changed = true;
      this.context.calendarDays = currValue;
      this._changes['calendarDays'] = new import1.SimpleChange(this._expr_11,currValue);
      this._expr_11 = currValue;
    }
  }
  check_currentMonth(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_12,currValue))) {
      this._changed = true;
      this.context.currentMonth = currValue;
      this._changes['currentMonth'] = new import1.SimpleChange(this._expr_12,currValue);
      this._expr_12 = currValue;
    }
  }
  check_dayNames(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_13,currValue))) {
      this._changed = true;
      this.context.dayNames = currValue;
      this._changes['dayNames'] = new import1.SimpleChange(this._expr_13,currValue);
      this._expr_13 = currValue;
    }
  }
  check_hoveredDay(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_14,currValue))) {
      this._changed = true;
      this.context.hoveredDay = currValue;
      this._changes['hoveredDay'] = new import1.SimpleChange(this._expr_14,currValue);
      this._expr_14 = currValue;
    }
  }
  ngDoCheck(view:import2.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    if (!throwOnChange) {
      if (changed) {
        this.context.ngOnChanges(this._changes);
        this._changes = {};
      }
      if ((view.numberOfChecks === 0)) { this.context.ngOnInit(); }
    }
    return changed;
  }
  checkHost(view:import2.AppView<any>,componentView:import2.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import2.AppView<any>,_eventHandler:any,emit0:boolean,emit1:boolean):void {
    this._eventHandler = _eventHandler;
    if (emit0) { (this.subscription0 = this.context.dateChange.subscribe(_eventHandler.bind(view,'dateChange'))); }
    if (emit1) { (this.subscription1 = this.context.onSelect.subscribe(_eventHandler.bind(view,'onSelect'))); }
  }
}
var renderType_DatepickerComponent_Host:import4.RenderComponentType = import3.createRenderComponentType('',0,import5.ViewEncapsulation.None,([] as any[]),{});
class View_DatepickerComponent_Host0 extends import2.AppView<any> {
  _el_0:any;
  compView_0:import2.AppView<import0.DatepickerComponent>;
  _DatepickerComponent_0_3:Wrapper_DatepickerComponent;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_DatepickerComponent_Host0,renderType_DatepickerComponent_Host,import6.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import7.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import8.ComponentRef<any> {
    this._el_0 = import3.selectOrCreateRenderHostElement(this.renderer,'material-datepicker',import3.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this.compView_0 = new View_DatepickerComponent0(this.viewUtils,this,0,this._el_0);
    this._DatepickerComponent_0_3 = new Wrapper_DatepickerComponent(this.renderer,new import9.ElementRef(this._el_0));
    this.compView_0.create(this._DatepickerComponent_0_3.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import8.ComponentRef_<any>(0,this,this._el_0,this._DatepickerComponent_0_3.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import0.DatepickerComponent) && (0 === requestNodeIndex))) { return this._DatepickerComponent_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._DatepickerComponent_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
    this._DatepickerComponent_0_3.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
export const DatepickerComponentNgFactory:import8.ComponentFactory<import0.DatepickerComponent> = new import8.ComponentFactory<import0.DatepickerComponent>('material-datepicker',View_DatepickerComponent_Host0,import0.DatepickerComponent);
const styles_DatepickerComponent:any[] = ['.datepicker[_ngcontent-%COMP%] {\n        position: relative;\n        display: inline-block;\n        color: #2b2b2b;\n        font-family: \'Helvetica Neue\', \'Helvetica\', \'Arial\', \'Calibri\', \'Roboto\';\n      }\n\n      .datepicker__calendar[_ngcontent-%COMP%] {\n        position: absolute;\n        overflow: hidden;\n        z-index: 1000;\n        top: 1.9em;\n        left: 0;\n        height: 24.25em;\n        width: 20.5em;\n        font-size: 14px;\n        background-color: #ffffff;\n        box-shadow: 0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23);\n        cursor: default;\n        -webkit-touch-callout: none;\n          -webkit-user-select: none;\n             -moz-user-select: none;\n              -ms-user-select: none;\n                  user-select: none;\n      }\n\n      .datepicker__calendar__cancel[_ngcontent-%COMP%] {\n        position: absolute;\n        bottom: 1em;\n        left: 1.8em;\n        color: #d8d8d8;\n        cursor: pointer;\n        -webkit-transition: 0.37s;\n        transition: 0.37s;\n      }\n\n      .datepicker__calendar__cancel[_ngcontent-%COMP%]:hover {\n        color: #b1b1b1;\n      }\n\n      .datepicker__calendar__content[_ngcontent-%COMP%] {\n        display: -webkit-box;\n        display: -ms-flexbox;\n        display: flex;\n        -ms-flex-flow: wrap;\n            flex-flow: wrap;\n        -webkit-box-pack: center;\n           -ms-flex-pack: center;\n         justify-content: center;\n        margin-top: 0.2em;\n      }\n\n      .datepicker__calendar__label[_ngcontent-%COMP%] {\n        display: inline-block;\n        width: 2.2em;\n        height: 2.2em;\n        margin: 0.2em;\n        line-height: 2.2em;\n        text-align: center;\n        color: #d8d8d8;\n      }\n\n      .datepicker__calendar__month[_ngcontent-%COMP%] {\n        display: -webkit-box;\n        display: -ms-flexbox;\n        display: flex;\n        -ms-flex-flow: wrap;\n            flex-flow: wrap;\n        -webkit-box-pack: center;\n           -ms-flex-pack: center;\n         justify-content: center;\n      }\n\n      .datepicker__calendar__month__day[_ngcontent-%COMP%] {\n        display: inline-block;\n        width: 2.2em;\n        height: 2.2em;\n        margin: 0.2em;\n        border-radius: 2.2em;\n        line-height: 2.2em;\n        text-align: center;\n        -webkit-transition: 0.37s;\n        transition: 0.37s;\n      }\n\n      .datepicker__calendar__nav[_ngcontent-%COMP%] {\n        display: -webkit-box;\n        display: -ms-flexbox;\n        display: flex;\n        -webkit-box-pack: center;\n           -ms-flex-pack: center;\n         justify-content: center;\n        -webkit-box-align: center;\n           -ms-flex-align: center;\n              align-items: center;\n        height: 3em;\n        background-color: #fff;\n        border-bottom: 1px solid #e8e8e8;\n      }\n\n      .datepicker__calendar__nav__arrow[_ngcontent-%COMP%] {\n        width: 0.8em;\n        height: 0.8em;\n        cursor: pointer;\n        -webkit-transition: 0.37s;\n        transition: 0.37s;\n      }\n\n      .datepicker__calendar__nav__arrow[_ngcontent-%COMP%]:hover {\n        -webkit-transform: scale(1.05);\n                transform: scale(1.05);\n      }\n\n      .datepicker__calendar__nav__chevron[_ngcontent-%COMP%] {\n        fill: #bbbbbb;\n        -webkit-transition: 0.37s;\n        transition: 0.37s;\n      }\n\n      .datepicker__calendar__nav__chevron[_ngcontent-%COMP%]:hover {\n        fill: #2b2b2b;\n      }\n\n      .datepicker__calendar__nav__header[_ngcontent-%COMP%] {\n        width: 11em;\n        margin: 0 1em;\n        text-align: center;\n      }\n\n      .datepicker__input[_ngcontent-%COMP%] {\n        outline: none;\n        border-radius: 0.1rem;\n        padding: .2em .6em;\n        font-size: 14px;\n      }'];
var DatepickerComponent_calendarAnimation_states:any = {
  '*': {},
  'left': {},
  'right': {}
}
;
function DatepickerComponent_calendarAnimation_factory(view:import2.AppView<any>,element:any,currentState:any,nextState:any):import10.AnimationTransition {
  var previousPlayers:any = view.animationContext.getAnimationPlayers(element,((nextState == 'void')? (null as any): 'calendarAnimation'));
  var collectedStyles:any = {};
  var player:any = (null as any);
  var totalTime:any = 0;
  var defaultStateStyles:any = DatepickerComponent_calendarAnimation_states['*'];
  var startStateStyles:any = DatepickerComponent_calendarAnimation_states[currentState];
  if ((startStateStyles == (null as any))) { (startStateStyles = defaultStateStyles); }
  var endStateStyles:any = DatepickerComponent_calendarAnimation_states[nextState];
  if ((endStateStyles == (null as any))) { (endStateStyles = defaultStateStyles); }
  if (((player == (null as any)) && (true && (nextState == 'left')))) {
      player = new import11.AnimationSequencePlayer([view.renderer.animate(element,new import12.AnimationStyles(import13.collectAndResolveStyles(collectedStyles,[startStateStyles])),[
        new import14.AnimationKeyframe(0,new import12.AnimationStyles(import13.collectAndResolveStyles(collectedStyles,[{'transform': 'true'}]))),
        new import14.AnimationKeyframe(0.5,new import12.AnimationStyles(import13.collectAndResolveStyles(collectedStyles,[{'transform': 'translateX(105%)'}]))),
        new import14.AnimationKeyframe(0.51,new import12.AnimationStyles(import13.collectAndResolveStyles(collectedStyles,[{'transform': 'translateX(-130%)'}]))),
        new import14.AnimationKeyframe(1,new import12.AnimationStyles(import13.collectAndResolveStyles(collectedStyles,[{'transform': 'translateX(0)'}])))
      ]
    ,180,0,(null as any),previousPlayers)]);
    totalTime = 180;
  }
  if (((player == (null as any)) && (true && (nextState == 'right')))) {
      player = new import11.AnimationSequencePlayer([view.renderer.animate(element,new import12.AnimationStyles(import13.collectAndResolveStyles(collectedStyles,[startStateStyles])),[
        new import14.AnimationKeyframe(0,new import12.AnimationStyles(import13.collectAndResolveStyles(collectedStyles,[{'transform': 'true'}]))),
        new import14.AnimationKeyframe(0.5,new import12.AnimationStyles(import13.collectAndResolveStyles(collectedStyles,[{'transform': 'translateX(-105%)'}]))),
        new import14.AnimationKeyframe(0.51,new import12.AnimationStyles(import13.collectAndResolveStyles(collectedStyles,[{'transform': 'translateX(130%)'}]))),
        new import14.AnimationKeyframe(1,new import12.AnimationStyles(import13.collectAndResolveStyles(collectedStyles,[{'transform': 'translateX(0)'}])))
      ]
    ,180,0,(null as any),previousPlayers)]);
    totalTime = 180;
  }
  if ((player == (null as any))) { (player = new import15.NoOpAnimationPlayer()); }
  player.onDone(():void => {
    player.destroy();
    import13.renderStyles(element,view.renderer,import13.prepareFinalAnimationStyles(startStateStyles,endStateStyles));
  });
  new import11.AnimationSequencePlayer(previousPlayers).destroy();
  import13.renderStyles(element,view.renderer,import13.clearStyles(startStateStyles));
  view.animationContext.queueAnimation(element,'calendarAnimation',player);
  return new import10.AnimationTransition(player,currentState,nextState,totalTime);
}
class View_DatepickerComponent2 extends import2.AppView<any> {
  _el_0:any;
  _text_1:any;
  /*private*/ _expr_2:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_DatepickerComponent2,renderType_DatepickerComponent,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import7.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_2 = import1.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import8.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'div',new import3.InlineArray2(2,'class','datepicker__calendar__label'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1
    ]
    ),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2:any = import3.inlineInterpolate(1,'\n            ',this.context.$implicit,'\n          ');
    if (import3.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setText(this._text_1,currVal_2);
      this._expr_2 = currVal_2;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_DatepickerComponent4 extends import2.AppView<any> {
  _el_0:any;
  _text_1:any;
  /*private*/ _expr_2:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_DatepickerComponent4,renderType_DatepickerComponent,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import7.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_2 = import1.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import8.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'span',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1
    ]
    ),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2:any = import3.inlineInterpolate(1,'\n                ',this.parentView.context.$implicit.getDate(),'\n              ');
    if (import3.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setText(this._text_1,currVal_2);
      this._expr_2 = currVal_2;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_DatepickerComponent3 extends import2.AppView<any> {
  _el_0:any;
  _NgStyle_0_3:import17.Wrapper_NgStyle;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import16.ViewContainer;
  _TemplateRef_2_5:any;
  _NgIf_2_6:import18.Wrapper_NgIf;
  _text_3:any;
  _map_8:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_DatepickerComponent3,renderType_DatepickerComponent,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import7.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_8 = import3.pureProxy4((p0:any,p1:any,p2:any,p3:any):{[key: string]:any} => {
      return {
        cursor: p0,
        'background-color': p1,
        color: p2,
        'pointer-events': p3
      }
      ;
    });
  }
  createInternal(rootSelector:string):import8.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'div',new import3.InlineArray2(2,'class','datepicker__calendar__month__day'),(null as any));
    this._NgStyle_0_3 = new import17.Wrapper_NgStyle(this.parentView.parentView.parentView.injectorGet(import19.KeyValueDiffers,this.parentView.parentView.parentIndex),new import9.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n              ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import16.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import20.TemplateRef_(this,2,this._anchor_2);
    this._NgIf_2_6 = new import18.Wrapper_NgIf(this._vc_2.vcRef,this._TemplateRef_2_5);
    this._text_3 = this.renderer.createText(this._el_0,'\n            ',(null as any));
    var disposable_0:Function = import3.subscribeToRenderElement(this,this._el_0,new import3.InlineArray8(6,'click',(null as any),'mouseenter',(null as any),'mouseleave',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import20.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import21.NgIf) && (2 === requestNodeIndex))) { return this._NgIf_2_6.context; }
    if (((token === import22.NgStyle) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 3)))) { return this._NgStyle_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this._map_8(((this.context.$implicit == 0)? 'initial': 'pointer'),this.parentView.parentView.context.getDayBackgroundColor(this.context.$implicit),(this.parentView.parentView.context.isHoveredDay(this.context.$implicit)? this.parentView.parentView.context.accentColor: this.parentView.parentView.context.getDayFontColor(this.context.$implicit)),((this.context.$implicit == 0)? 'none': ''));
    this._NgStyle_0_3.check_ngStyle(currVal_0_0_0,throwOnChange,false);
    this._NgStyle_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_2_0_0:any = (this.context.$implicit != 0);
    this._NgIf_2_6.check_ngIf(currVal_2_0_0,throwOnChange,false);
    this._NgIf_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import2.AppView<any> {
    if ((nodeIndex == 2)) { return new View_DatepickerComponent4(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    return (null as any);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.parentView.context.onSelectDay(this.context.$implicit)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'mouseenter')) {
      const pd_sub_1:any = ((<any>(this.parentView.parentView.context.hoveredDay = this.context.$implicit)) !== false);
      result = (pd_sub_1 && result);
    }
    if ((eventName == 'mouseleave')) {
      const pd_sub_2:any = ((<any>(this.parentView.parentView.context.hoveredDay = (null as any))) !== false);
      result = (pd_sub_2 && result);
    }
    return result;
  }
}
class View_DatepickerComponent1 extends import2.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _el_4:any;
  _text_5:any;
  _el_6:any;
  _text_7:any;
  _el_8:any;
  _text_9:any;
  _el_10:any;
  _text_11:any;
  _text_12:any;
  _el_13:any;
  _text_14:any;
  _el_15:any;
  _text_16:any;
  _text_17:any;
  _text_18:any;
  _text_19:any;
  _el_20:any;
  _text_21:any;
  _text_22:any;
  _el_23:any;
  _text_24:any;
  _el_25:any;
  _text_26:any;
  _el_27:any;
  _text_28:any;
  _el_29:any;
  _text_30:any;
  _text_31:any;
  _el_32:any;
  _text_33:any;
  _el_34:any;
  _text_35:any;
  _text_36:any;
  _text_37:any;
  _text_38:any;
  _text_39:any;
  _el_40:any;
  _text_41:any;
  _anchor_42:any;
  /*private*/ _vc_42:import16.ViewContainer;
  _TemplateRef_42_5:any;
  _NgFor_42_6:import23.Wrapper_NgFor;
  _text_43:any;
  _el_44:any;
  _text_45:any;
  _anchor_46:any;
  /*private*/ _vc_46:import16.ViewContainer;
  _TemplateRef_46_5:any;
  _NgFor_46_6:import23.Wrapper_NgFor;
  _text_47:any;
  _text_48:any;
  _el_49:any;
  _text_50:any;
  _text_51:any;
  _text_52:any;
  /*private*/ _expr_59:any;
  /*private*/ _expr_60:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_DatepickerComponent1,renderType_DatepickerComponent,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import7.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_59 = import1.UNINITIALIZED;
    this._expr_60 = import1.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import8.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'div',new import3.InlineArray2(2,'class','datepicker__calendar'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n        ',(null as any));
    this._el_2 = import3.createRenderElement(this.renderer,this._el_0,'div',new import3.InlineArray2(2,'class','datepicker__calendar__nav'),(null as any));
    this._text_3 = this.renderer.createText(this._el_2,'\n          ',(null as any));
    this._el_4 = import3.createRenderElement(this.renderer,this._el_2,'div',new import3.InlineArray2(2,'class','datepicker__calendar__nav__arrow'),(null as any));
    this._text_5 = this.renderer.createText(this._el_4,'\n          ',(null as any));
    this._el_6 = import3.createRenderElement(this.renderer,this._el_4,':svg:svg',new import3.InlineArray8(8,'class','datepicker__calendar__nav__chevron','viewBox','0 0 50 50','x','0px','y','0px'),(null as any));
    this._text_7 = this.renderer.createText(this._el_6,'\n            ',(null as any));
    this._el_8 = import3.createRenderElement(this.renderer,this._el_6,':svg:g',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._text_9 = this.renderer.createText(this._el_8,'\n                ',(null as any));
    this._el_10 = import3.createRenderElement(this.renderer,this._el_8,':svg:path',new import3.InlineArray2(2,'d','M39.7,7.1c0.5,0.5,0.5,1.2,0,1.7L29,19.6c-0.5,0.5-1.2,1.2-1.7,1.7L16.5,32.1c-0.5,0.5-1.2,0.5-1.7,0l-2.3-2.3\n                    c-0.5-0.5-1.2-1.2-1.7-1.7l-2.3-2.3c-0.5-0.5-0.5-1.2,0-1.7l10.8-10.8c0.5-0.5,1.2-1.2,1.7-1.7L31.7,0.8c0.5-0.5,1.2-0.5,1.7,0\n                    l2.3,2.3c0.5,0.5,1.2,1.2,1.7,1.7L39.7,7.1z'),(null as any));
    this._text_11 = this.renderer.createText(this._el_8,'\n            ',(null as any));
    this._text_12 = this.renderer.createText(this._el_6,'\n            ',(null as any));
    this._el_13 = import3.createRenderElement(this.renderer,this._el_6,':svg:g',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._text_14 = this.renderer.createText(this._el_13,'\n                ',(null as any));
    this._el_15 = import3.createRenderElement(this.renderer,this._el_13,':svg:path',new import3.InlineArray2(2,'d','M33.4,49c-0.5,0.5-1.2,0.5-1.7,0L20.9,38.2c-0.5-0.5-1.2-1.2-1.7-1.7L8.4,25.7c-0.5-0.5-0.5-1.2,0-1.7l2.3-2.3\n                    c0.5-0.5,1.2-1.2,1.7-1.7l2.3-2.3c0.5-0.5,1.2-0.5,1.7,0l10.8,10.8c0.5,0.5,1.2,1.2,1.7,1.7l10.8,10.8c0.5,0.5,0.5,1.2,0,1.7\n                    L37.4,45c-0.5,0.5-1.2,1.2-1.7,1.7L33.4,49z'),(null as any));
    this._text_16 = this.renderer.createText(this._el_13,'\n            ',(null as any));
    this._text_17 = this.renderer.createText(this._el_6,'\n          ',(null as any));
    this._text_18 = this.renderer.createText(this._el_4,'\n          ',(null as any));
    this._text_19 = this.renderer.createText(this._el_2,'\n          ',(null as any));
    this._el_20 = import3.createRenderElement(this.renderer,this._el_2,'div',new import3.InlineArray2(2,'class','datepicker__calendar__nav__header'),(null as any));
    this._text_21 = this.renderer.createText(this._el_20,'',(null as any));
    this._text_22 = this.renderer.createText(this._el_2,'\n          ',(null as any));
    this._el_23 = import3.createRenderElement(this.renderer,this._el_2,'div',new import3.InlineArray2(2,'class','datepicker__calendar__nav__arrow'),(null as any));
    this._text_24 = this.renderer.createText(this._el_23,'\n            ',(null as any));
    this._el_25 = import3.createRenderElement(this.renderer,this._el_23,':svg:svg',new import3.InlineArray8(8,'class','datepicker__calendar__nav__chevron','viewBox','0 0 50 50','x','0px','y','0px'),(null as any));
    this._text_26 = this.renderer.createText(this._el_25,'\n              ',(null as any));
    this._el_27 = import3.createRenderElement(this.renderer,this._el_25,':svg:g',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._text_28 = this.renderer.createText(this._el_27,'\n                ',(null as any));
    this._el_29 = import3.createRenderElement(this.renderer,this._el_27,':svg:path',new import3.InlineArray2(2,'d','M8.4,7.1c-0.5,0.5-0.5,1.2,0,1.7l10.8,10.8c0.5,0.5,1.2,1.2,1.7,1.7l10.8,10.8c0.5,0.5,1.2,0.5,1.7,0l2.3-2.3\n                    c0.5-0.5,1.2-1.2,1.7-1.7l2.3-2.3c0.5-0.5,0.5-1.2,0-1.7L29,13.2c-0.5-0.5-1.2-1.2-1.7-1.7L16.5,0.8c-0.5-0.5-1.2-0.5-1.7,0\n                    l-2.3,2.3c-0.5,0.5-1.2,1.2-1.7,1.7L8.4,7.1z'),(null as any));
    this._text_30 = this.renderer.createText(this._el_27,'\n              ',(null as any));
    this._text_31 = this.renderer.createText(this._el_25,'\n              ',(null as any));
    this._el_32 = import3.createRenderElement(this.renderer,this._el_25,':svg:g',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._text_33 = this.renderer.createText(this._el_32,'\n                ',(null as any));
    this._el_34 = import3.createRenderElement(this.renderer,this._el_32,':svg:path',new import3.InlineArray2(2,'d','M14.8,49c0.5,0.5,1.2,0.5,1.7,0l10.8-10.8c0.5-0.5,1.2-1.2,1.7-1.7l10.8-10.8c0.5-0.5,0.5-1.2,0-1.7l-2.3-2.3\n                    c-0.5-0.5-1.2-1.2-1.7-1.7l-2.3-2.3c-0.5-0.5-1.2-0.5-1.7,0L20.9,28.5c-0.5,0.5-1.2,1.2-1.7,1.7L8.4,40.9c-0.5,0.5-0.5,1.2,0,1.7\n                    l2.3,2.3c0.5,0.5,1.2,1.2,1.7,1.7L14.8,49z'),(null as any));
    this._text_35 = this.renderer.createText(this._el_32,'\n              ',(null as any));
    this._text_36 = this.renderer.createText(this._el_25,'\n            ',(null as any));
    this._text_37 = this.renderer.createText(this._el_23,'\n          ',(null as any));
    this._text_38 = this.renderer.createText(this._el_2,'\n        ',(null as any));
    this._text_39 = this.renderer.createText(this._el_0,'\n        ',(null as any));
    this._el_40 = import3.createRenderElement(this.renderer,this._el_0,'div',new import3.InlineArray2(2,'class','datepicker__calendar__content'),(null as any));
    this._text_41 = this.renderer.createText(this._el_40,'\n          ',(null as any));
    this._anchor_42 = this.renderer.createTemplateAnchor(this._el_40,(null as any));
    this._vc_42 = new import16.ViewContainer(42,40,this,this._anchor_42);
    this._TemplateRef_42_5 = new import20.TemplateRef_(this,42,this._anchor_42);
    this._NgFor_42_6 = new import23.Wrapper_NgFor(this._vc_42.vcRef,this._TemplateRef_42_5,this.parentView.parentView.injectorGet(import24.IterableDiffers,this.parentView.parentIndex),this.parentView.ref);
    this._text_43 = this.renderer.createText(this._el_40,'\n          ',(null as any));
    this._el_44 = import3.createRenderElement(this.renderer,this._el_40,'div',new import3.InlineArray2(2,'class','datepicker__calendar__month'),(null as any));
    this._text_45 = this.renderer.createText(this._el_44,'\n            ',(null as any));
    this._anchor_46 = this.renderer.createTemplateAnchor(this._el_44,(null as any));
    this._vc_46 = new import16.ViewContainer(46,44,this,this._anchor_46);
    this._TemplateRef_46_5 = new import20.TemplateRef_(this,46,this._anchor_46);
    this._NgFor_46_6 = new import23.Wrapper_NgFor(this._vc_46.vcRef,this._TemplateRef_46_5,this.parentView.parentView.injectorGet(import24.IterableDiffers,this.parentView.parentIndex),this.parentView.ref);
    this._text_47 = this.renderer.createText(this._el_44,'\n          ',(null as any));
    this._text_48 = this.renderer.createText(this._el_40,'\n          ',(null as any));
    this._el_49 = import3.createRenderElement(this.renderer,this._el_40,'div',new import3.InlineArray2(2,'class','datepicker__calendar__cancel'),(null as any));
    this._text_50 = this.renderer.createText(this._el_49,'\n            Cancel\n          ',(null as any));
    this._text_51 = this.renderer.createText(this._el_40,'\n        ',(null as any));
    this._text_52 = this.renderer.createText(this._el_0,'\n      ',(null as any));
    var disposable_0:Function = import3.subscribeToRenderElement(this,this._el_4,new import3.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_4));
    var disposable_1:Function = import3.subscribeToRenderElement(this,this._el_23,new import3.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_23));
    var disposable_2:Function = import3.subscribeToRenderElement(this,this._el_49,new import3.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_49));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._el_4,
      this._text_5,
      this._el_6,
      this._text_7,
      this._el_8,
      this._text_9,
      this._el_10,
      this._text_11,
      this._text_12,
      this._el_13,
      this._text_14,
      this._el_15,
      this._text_16,
      this._text_17,
      this._text_18,
      this._text_19,
      this._el_20,
      this._text_21,
      this._text_22,
      this._el_23,
      this._text_24,
      this._el_25,
      this._text_26,
      this._el_27,
      this._text_28,
      this._el_29,
      this._text_30,
      this._text_31,
      this._el_32,
      this._text_33,
      this._el_34,
      this._text_35,
      this._text_36,
      this._text_37,
      this._text_38,
      this._text_39,
      this._el_40,
      this._text_41,
      this._anchor_42,
      this._text_43,
      this._el_44,
      this._text_45,
      this._anchor_46,
      this._text_47,
      this._text_48,
      this._el_49,
      this._text_50,
      this._text_51,
      this._text_52
    ]
    ),[
      disposable_0,
      disposable_1,
      disposable_2
    ]
    );
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import20.TemplateRef) && (42 === requestNodeIndex))) { return this._TemplateRef_42_5; }
    if (((token === import25.NgFor) && (42 === requestNodeIndex))) { return this._NgFor_42_6.context; }
    if (((token === import20.TemplateRef) && (46 === requestNodeIndex))) { return this._TemplateRef_46_5; }
    if (((token === import25.NgFor) && (46 === requestNodeIndex))) { return this._NgFor_46_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_60:any = this.parentView.context.animate;
    if (import3.checkBinding(throwOnChange,this._expr_60,currVal_60)) {
      var animationTransition_calendarAnimation:any = this.componentType.animations['calendarAnimation'](this,this._el_44,((this._expr_60 == import1.UNINITIALIZED)? 'void': this._expr_60),((currVal_60 == import1.UNINITIALIZED)? 'void': currVal_60));
      this._expr_60 = currVal_60;
    }
    const currVal_42_0_0:any = this.parentView.context.dayNames;
    this._NgFor_42_6.check_ngForOf(currVal_42_0_0,throwOnChange,false);
    this._NgFor_42_6.ngDoCheck(this,this._anchor_42,throwOnChange);
    const currVal_46_0_0:any = this.parentView.context.calendarDays;
    this._NgFor_46_6.check_ngForOf(currVal_46_0_0,throwOnChange,false);
    this._NgFor_46_6.ngDoCheck(this,this._anchor_46,throwOnChange);
    this._vc_42.detectChangesInNestedViews(throwOnChange);
    this._vc_46.detectChangesInNestedViews(throwOnChange);
    const currVal_59:any = import3.inlineInterpolate(2,'\n            ',this.parentView.context.currentMonth,' ',this.parentView.context.currentYear,'\n          ');
    if (import3.checkBinding(throwOnChange,this._expr_59,currVal_59)) {
      this.renderer.setText(this._text_21,currVal_59);
      this._expr_59 = currVal_59;
    }
  }
  destroyInternal():void {
    this._vc_42.destroyNestedViews();
    this._vc_46.destroyNestedViews();
  }
  detachInternal():void {
    var animationTransition_calendarAnimation:any = this.componentType.animations['calendarAnimation'](this,this._el_44,this._expr_60,'void');
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import2.AppView<any> {
    if ((nodeIndex == 42)) { return new View_DatepickerComponent2(this.viewUtils,this,42,this._anchor_42,this._vc_42); }
    if ((nodeIndex == 46)) { return new View_DatepickerComponent3(this.viewUtils,this,46,this._anchor_46,this._vc_46); }
    return (null as any);
  }
  handleEvent_4(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.context.onArrowLeftClick()) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
  handleEvent_23(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.context.onArrowRightClick()) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
  handleEvent_49(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.context.onCancel()) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
var renderType_DatepickerComponent:import4.RenderComponentType = import3.createRenderComponentType('',0,import5.ViewEncapsulation.Emulated,styles_DatepickerComponent,{calendarAnimation: DatepickerComponent_calendarAnimation_factory});
export class View_DatepickerComponent0 extends import2.AppView<import0.DatepickerComponent> {
  _text_0:any;
  _el_1:any;
  _NgStyle_1_3:import17.Wrapper_NgStyle;
  _text_2:any;
  _el_3:any;
  _NgStyle_3_3:import17.Wrapper_NgStyle;
  _DefaultValueAccessor_3_4:import26.Wrapper_DefaultValueAccessor;
  _NG_VALUE_ACCESSOR_3_5:any[];
  _NgModel_3_6:import27.Wrapper_NgModel;
  _NgControl_3_7:any;
  _NgControlStatus_3_8:import28.Wrapper_NgControlStatus;
  _text_4:any;
  _anchor_5:any;
  /*private*/ _vc_5:import16.ViewContainer;
  _TemplateRef_5_5:any;
  _NgIf_5_6:import18.Wrapper_NgIf;
  _text_6:any;
  _text_7:any;
  _map_18:any;
  /*private*/ _expr_19:any;
  _map_20:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_DatepickerComponent0,renderType_DatepickerComponent,import6.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import7.ChangeDetectorStatus.CheckAlways);
    this._map_18 = import3.pureProxy1((p0:any):{[key: string]:any} => {
      return {'font-family': p0};
    });
    this._expr_19 = import1.UNINITIALIZED;
    this._map_20 = import3.pureProxy3((p0:any,p1:any,p2:any):{[key: string]:any} => {
      return {
        color: p0,
        'background-color': p1,
        border: p2
      }
      ;
    });
  }
  createInternal(rootSelector:string):import8.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    this._el_1 = import3.createRenderElement(this.renderer,parentRenderNode,'div',new import3.InlineArray2(2,'class','datepicker'),(null as any));
    this._NgStyle_1_3 = new import17.Wrapper_NgStyle(this.parentView.injectorGet(import19.KeyValueDiffers,this.parentIndex),new import9.ElementRef(this._el_1),this.renderer);
    this._text_2 = this.renderer.createText(this._el_1,'\n      ',(null as any));
    this._el_3 = import3.createRenderElement(this.renderer,this._el_1,'input',new import3.InlineArray4(4,'class','datepicker__input','readonly','true'),(null as any));
    this._NgStyle_3_3 = new import17.Wrapper_NgStyle(this.parentView.injectorGet(import19.KeyValueDiffers,this.parentIndex),new import9.ElementRef(this._el_3),this.renderer);
    this._DefaultValueAccessor_3_4 = new import26.Wrapper_DefaultValueAccessor(this.renderer,new import9.ElementRef(this._el_3));
    this._NG_VALUE_ACCESSOR_3_5 = [this._DefaultValueAccessor_3_4.context];
    this._NgModel_3_6 = new import27.Wrapper_NgModel((null as any),(null as any),(null as any),this._NG_VALUE_ACCESSOR_3_5);
    this._NgControl_3_7 = this._NgModel_3_6.context;
    this._NgControlStatus_3_8 = new import28.Wrapper_NgControlStatus(this._NgControl_3_7);
    this._text_4 = this.renderer.createText(this._el_1,'\n      ',(null as any));
    this._anchor_5 = this.renderer.createTemplateAnchor(this._el_1,(null as any));
    this._vc_5 = new import16.ViewContainer(5,1,this,this._anchor_5);
    this._TemplateRef_5_5 = new import20.TemplateRef_(this,5,this._anchor_5);
    this._NgIf_5_6 = new import18.Wrapper_NgIf(this._vc_5.vcRef,this._TemplateRef_5_5);
    this._text_6 = this.renderer.createText(this._el_1,'\n    ',(null as any));
    this._text_7 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    var disposable_0:Function = import3.subscribeToRenderElement(this,this._el_3,new import3.InlineArray8(8,'click',(null as any),'ngModelChange',(null as any),'input',(null as any),'blur',(null as any)),this.eventHandler(this.handleEvent_3));
    this._NgModel_3_6.subscribe(this,this.eventHandler(this.handleEvent_3),true);
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2,
      this._el_3,
      this._text_4,
      this._anchor_5,
      this._text_6,
      this._text_7
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import22.NgStyle) && (3 === requestNodeIndex))) { return this._NgStyle_3_3.context; }
    if (((token === import29.DefaultValueAccessor) && (3 === requestNodeIndex))) { return this._DefaultValueAccessor_3_4.context; }
    if (((token === import30.NG_VALUE_ACCESSOR) && (3 === requestNodeIndex))) { return this._NG_VALUE_ACCESSOR_3_5; }
    if (((token === import31.NgModel) && (3 === requestNodeIndex))) { return this._NgModel_3_6.context; }
    if (((token === import32.NgControl) && (3 === requestNodeIndex))) { return this._NgControl_3_7; }
    if (((token === import33.NgControlStatus) && (3 === requestNodeIndex))) { return this._NgControlStatus_3_8.context; }
    if (((token === import20.TemplateRef) && (5 === requestNodeIndex))) { return this._TemplateRef_5_5; }
    if (((token === import21.NgIf) && (5 === requestNodeIndex))) { return this._NgIf_5_6.context; }
    if (((token === import22.NgStyle) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 6)))) { return this._NgStyle_1_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1_0_0:any = this._map_18(this.context.fontFamily);
    this._NgStyle_1_3.check_ngStyle(currVal_1_0_0,throwOnChange,false);
    this._NgStyle_1_3.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_3_0_0:any = this._map_20((this.context.altInputStyle? this.context.colors['white']: this.context.colors['black']),(this.context.altInputStyle? this.context.accentColor: this.context.colors['white']),(this.context.altInputStyle? '': '1px solid #dadada'));
    this._NgStyle_3_3.check_ngStyle(currVal_3_0_0,throwOnChange,false);
    this._NgStyle_3_3.ngDoCheck(this,this._el_3,throwOnChange);
    this._DefaultValueAccessor_3_4.ngDoCheck(this,this._el_3,throwOnChange);
    const currVal_3_2_0:any = this.context.disabled;
    this._NgModel_3_6.check_isDisabled(currVal_3_2_0,throwOnChange,false);
    const currVal_3_2_1:any = this.context.inputText;
    this._NgModel_3_6.check_model(currVal_3_2_1,throwOnChange,false);
    this._NgModel_3_6.ngDoCheck(this,this._el_3,throwOnChange);
    this._NgControlStatus_3_8.ngDoCheck(this,this._el_3,throwOnChange);
    const currVal_5_0_0:any = this.context.showCalendar;
    this._NgIf_5_6.check_ngIf(currVal_5_0_0,throwOnChange,false);
    this._NgIf_5_6.ngDoCheck(this,this._anchor_5,throwOnChange);
    this._vc_5.detectChangesInNestedViews(throwOnChange);
    const currVal_19:any = this.context.placeholder;
    if (import3.checkBinding(throwOnChange,this._expr_19,currVal_19)) {
      this.renderer.setElementProperty(this._el_3,'placeholder',currVal_19);
      this._expr_19 = currVal_19;
    }
    this._NgControlStatus_3_8.checkHost(this,this,this._el_3,throwOnChange);
  }
  destroyInternal():void {
    this._vc_5.destroyNestedViews();
    this._NgModel_3_6.ngOnDestroy();
  }
  createEmbeddedViewInternal(nodeIndex:number):import2.AppView<any> {
    if ((nodeIndex == 5)) { return new View_DatepickerComponent1(this.viewUtils,this,5,this._anchor_5,this._vc_5); }
    return (null as any);
  }
  handleEvent_3(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    result = (this._DefaultValueAccessor_3_4.handleEvent(eventName,$event) && result);
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.context.onInputClick()) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'ngModelChange')) {
      const pd_sub_1:any = ((<any>(this.context.inputText = $event)) !== false);
      result = (pd_sub_1 && result);
    }
    return result;
  }
}
