// angular2-material-datepicker/src/datepicker.component.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
import * as import0 from 'angular2-material-datepicker/src/datepicker.component';
import * as import1 from '@angular/core/src/change_detection/change_detection_util';
import * as import2 from '@angular/core/src/linker/view';
import * as import3 from '@angular/core/src/linker/view_utils';
import * as import5 from '@angular/core/src/metadata/view';
import * as import6 from '@angular/core/src/linker/view_type';
import * as import7 from '@angular/core/src/change_detection/constants';
import * as import8 from '@angular/core/src/linker/component_factory';
import * as import9 from '@angular/core/src/linker/element_ref';
import * as import10 from '@angular/core/src/animation/animation_transition';
import * as import11 from '@angular/core/src/animation/animation_sequence_player';
import * as import12 from '@angular/core/src/animation/animation_styles';
import * as import13 from '@angular/core/src/animation/animation_style_util';
import * as import14 from '@angular/core/src/animation/animation_keyframe';
import * as import15 from '@angular/core/src/animation/animation_player';
import * as import16 from '@angular/core/src/linker/view_container';
import * as import17 from '../../@angular/common/src/directives/ng_style.ngfactory';
import * as import18 from '../../@angular/common/src/directives/ng_if.ngfactory';
import * as import19 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import20 from '@angular/core/src/linker/template_ref';
import * as import21 from '@angular/common/src/directives/ng_if';
import * as import22 from '@angular/common/src/directives/ng_style';
import * as import23 from '../../@angular/common/src/directives/ng_for.ngfactory';
import * as import24 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import25 from '@angular/common/src/directives/ng_for';
import * as import26 from '../../@angular/forms/src/directives/default_value_accessor.ngfactory';
import * as import27 from '../../@angular/forms/src/directives/ng_model.ngfactory';
import * as import28 from '../../@angular/forms/src/directives/ng_control_status.ngfactory';
import * as import29 from '@angular/forms/src/directives/default_value_accessor';
import * as import30 from '@angular/forms/src/directives/control_value_accessor';
import * as import31 from '@angular/forms/src/directives/ng_model';
import * as import32 from '@angular/forms/src/directives/ng_control';
import * as import33 from '@angular/forms/src/directives/ng_control_status';
export var Wrapper_DatepickerComponent = (function () {
    function Wrapper_DatepickerComponent(p0, p1) {
        this._changed = false;
        this._changes = {};
        this.context = new import0.DatepickerComponent(p0, p1);
        this._expr_0 = import1.UNINITIALIZED;
        this._expr_1 = import1.UNINITIALIZED;
        this._expr_2 = import1.UNINITIALIZED;
        this._expr_3 = import1.UNINITIALIZED;
        this._expr_4 = import1.UNINITIALIZED;
        this._expr_5 = import1.UNINITIALIZED;
        this._expr_6 = import1.UNINITIALIZED;
        this._expr_7 = import1.UNINITIALIZED;
        this._expr_8 = import1.UNINITIALIZED;
        this._expr_9 = import1.UNINITIALIZED;
        this._expr_10 = import1.UNINITIALIZED;
        this._expr_11 = import1.UNINITIALIZED;
        this._expr_12 = import1.UNINITIALIZED;
        this._expr_13 = import1.UNINITIALIZED;
        this._expr_14 = import1.UNINITIALIZED;
    }
    Wrapper_DatepickerComponent.prototype.ngOnDetach = function (view, componentView, el) {
    };
    Wrapper_DatepickerComponent.prototype.ngOnDestroy = function () {
        this.context.ngOnDestroy();
        (this.subscription0 && this.subscription0.unsubscribe());
        (this.subscription1 && this.subscription1.unsubscribe());
    };
    Wrapper_DatepickerComponent.prototype.check_date = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_0, currValue))) {
            this._changed = true;
            this.context.date = currValue;
            this._changes['date'] = new import1.SimpleChange(this._expr_0, currValue);
            this._expr_0 = currValue;
        }
    };
    Wrapper_DatepickerComponent.prototype.check_disabled = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_1, currValue))) {
            this._changed = true;
            this.context.disabled = currValue;
            this._changes['disabled'] = new import1.SimpleChange(this._expr_1, currValue);
            this._expr_1 = currValue;
        }
    };
    Wrapper_DatepickerComponent.prototype.check_accentColor = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_2, currValue))) {
            this._changed = true;
            this.context.accentColor = currValue;
            this._changes['accentColor'] = new import1.SimpleChange(this._expr_2, currValue);
            this._expr_2 = currValue;
        }
    };
    Wrapper_DatepickerComponent.prototype.check_altInputStyle = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_3, currValue))) {
            this._changed = true;
            this.context.altInputStyle = currValue;
            this._changes['altInputStyle'] = new import1.SimpleChange(this._expr_3, currValue);
            this._expr_3 = currValue;
        }
    };
    Wrapper_DatepickerComponent.prototype.check_dateFormat = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_4, currValue))) {
            this._changed = true;
            this.context.dateFormat = currValue;
            this._changes['dateFormat'] = new import1.SimpleChange(this._expr_4, currValue);
            this._expr_4 = currValue;
        }
    };
    Wrapper_DatepickerComponent.prototype.check_fontFamily = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_5, currValue))) {
            this._changed = true;
            this.context.fontFamily = currValue;
            this._changes['fontFamily'] = new import1.SimpleChange(this._expr_5, currValue);
            this._expr_5 = currValue;
        }
    };
    Wrapper_DatepickerComponent.prototype.check_rangeStart = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_6, currValue))) {
            this._changed = true;
            this.context.rangeStart = currValue;
            this._changes['rangeStart'] = new import1.SimpleChange(this._expr_6, currValue);
            this._expr_6 = currValue;
        }
    };
    Wrapper_DatepickerComponent.prototype.check_rangeEnd = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_7, currValue))) {
            this._changed = true;
            this.context.rangeEnd = currValue;
            this._changes['rangeEnd'] = new import1.SimpleChange(this._expr_7, currValue);
            this._expr_7 = currValue;
        }
    };
    Wrapper_DatepickerComponent.prototype.check_placeholder = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_8, currValue))) {
            this._changed = true;
            this.context.placeholder = currValue;
            this._changes['placeholder'] = new import1.SimpleChange(this._expr_8, currValue);
            this._expr_8 = currValue;
        }
    };
    Wrapper_DatepickerComponent.prototype.check_inputText = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_9, currValue))) {
            this._changed = true;
            this.context.inputText = currValue;
            this._changes['inputText'] = new import1.SimpleChange(this._expr_9, currValue);
            this._expr_9 = currValue;
        }
    };
    Wrapper_DatepickerComponent.prototype.check_showCalendar = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_10, currValue))) {
            this._changed = true;
            this.context.showCalendar = currValue;
            this._changes['showCalendar'] = new import1.SimpleChange(this._expr_10, currValue);
            this._expr_10 = currValue;
        }
    };
    Wrapper_DatepickerComponent.prototype.check_calendarDays = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_11, currValue))) {
            this._changed = true;
            this.context.calendarDays = currValue;
            this._changes['calendarDays'] = new import1.SimpleChange(this._expr_11, currValue);
            this._expr_11 = currValue;
        }
    };
    Wrapper_DatepickerComponent.prototype.check_currentMonth = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_12, currValue))) {
            this._changed = true;
            this.context.currentMonth = currValue;
            this._changes['currentMonth'] = new import1.SimpleChange(this._expr_12, currValue);
            this._expr_12 = currValue;
        }
    };
    Wrapper_DatepickerComponent.prototype.check_dayNames = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_13, currValue))) {
            this._changed = true;
            this.context.dayNames = currValue;
            this._changes['dayNames'] = new import1.SimpleChange(this._expr_13, currValue);
            this._expr_13 = currValue;
        }
    };
    Wrapper_DatepickerComponent.prototype.check_hoveredDay = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_14, currValue))) {
            this._changed = true;
            this.context.hoveredDay = currValue;
            this._changes['hoveredDay'] = new import1.SimpleChange(this._expr_14, currValue);
            this._expr_14 = currValue;
        }
    };
    Wrapper_DatepickerComponent.prototype.ngDoCheck = function (view, el, throwOnChange) {
        var changed = this._changed;
        this._changed = false;
        if (!throwOnChange) {
            if (changed) {
                this.context.ngOnChanges(this._changes);
                this._changes = {};
            }
            if ((view.numberOfChecks === 0)) {
                this.context.ngOnInit();
            }
        }
        return changed;
    };
    Wrapper_DatepickerComponent.prototype.checkHost = function (view, componentView, el, throwOnChange) {
    };
    Wrapper_DatepickerComponent.prototype.handleEvent = function (eventName, $event) {
        var result = true;
        return result;
    };
    Wrapper_DatepickerComponent.prototype.subscribe = function (view, _eventHandler, emit0, emit1) {
        this._eventHandler = _eventHandler;
        if (emit0) {
            (this.subscription0 = this.context.dateChange.subscribe(_eventHandler.bind(view, 'dateChange')));
        }
        if (emit1) {
            (this.subscription1 = this.context.onSelect.subscribe(_eventHandler.bind(view, 'onSelect')));
        }
    };
    return Wrapper_DatepickerComponent;
}());
var renderType_DatepickerComponent_Host = import3.createRenderComponentType('', 0, import5.ViewEncapsulation.None, [], {});
var View_DatepickerComponent_Host0 = (function (_super) {
    __extends(View_DatepickerComponent_Host0, _super);
    function View_DatepickerComponent_Host0(viewUtils, parentView, parentIndex, parentElement) {
        _super.call(this, View_DatepickerComponent_Host0, renderType_DatepickerComponent_Host, import6.ViewType.HOST, viewUtils, parentView, parentIndex, parentElement, import7.ChangeDetectorStatus.CheckAlways);
    }
    View_DatepickerComponent_Host0.prototype.createInternal = function (rootSelector) {
        this._el_0 = import3.selectOrCreateRenderHostElement(this.renderer, 'material-datepicker', import3.EMPTY_INLINE_ARRAY, rootSelector, null);
        this.compView_0 = new View_DatepickerComponent0(this.viewUtils, this, 0, this._el_0);
        this._DatepickerComponent_0_3 = new Wrapper_DatepickerComponent(this.renderer, new import9.ElementRef(this._el_0));
        this.compView_0.create(this._DatepickerComponent_0_3.context);
        this.init(this._el_0, (this.renderer.directRenderer ? null : [this._el_0]), null);
        return new import8.ComponentRef_(0, this, this._el_0, this._DatepickerComponent_0_3.context);
    };
    View_DatepickerComponent_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import0.DatepickerComponent) && (0 === requestNodeIndex))) {
            return this._DatepickerComponent_0_3.context;
        }
        return notFoundResult;
    };
    View_DatepickerComponent_Host0.prototype.detectChangesInternal = function (throwOnChange) {
        this._DatepickerComponent_0_3.ngDoCheck(this, this._el_0, throwOnChange);
        this.compView_0.internalDetectChanges(throwOnChange);
    };
    View_DatepickerComponent_Host0.prototype.destroyInternal = function () {
        this.compView_0.destroy();
        this._DatepickerComponent_0_3.ngOnDestroy();
    };
    View_DatepickerComponent_Host0.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    return View_DatepickerComponent_Host0;
}(import2.AppView));
export var DatepickerComponentNgFactory = new import8.ComponentFactory('material-datepicker', View_DatepickerComponent_Host0, import0.DatepickerComponent);
var styles_DatepickerComponent = ['.datepicker[_ngcontent-%COMP%] {\n        position: relative;\n        display: inline-block;\n        color: #2b2b2b;\n        font-family: \'Helvetica Neue\', \'Helvetica\', \'Arial\', \'Calibri\', \'Roboto\';\n      }\n\n      .datepicker__calendar[_ngcontent-%COMP%] {\n        position: absolute;\n        overflow: hidden;\n        z-index: 1000;\n        top: 1.9em;\n        left: 0;\n        height: 24.25em;\n        width: 20.5em;\n        font-size: 14px;\n        background-color: #ffffff;\n        box-shadow: 0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23);\n        cursor: default;\n        -webkit-touch-callout: none;\n          -webkit-user-select: none;\n             -moz-user-select: none;\n              -ms-user-select: none;\n                  user-select: none;\n      }\n\n      .datepicker__calendar__cancel[_ngcontent-%COMP%] {\n        position: absolute;\n        bottom: 1em;\n        left: 1.8em;\n        color: #d8d8d8;\n        cursor: pointer;\n        -webkit-transition: 0.37s;\n        transition: 0.37s;\n      }\n\n      .datepicker__calendar__cancel[_ngcontent-%COMP%]:hover {\n        color: #b1b1b1;\n      }\n\n      .datepicker__calendar__content[_ngcontent-%COMP%] {\n        display: -webkit-box;\n        display: -ms-flexbox;\n        display: flex;\n        -ms-flex-flow: wrap;\n            flex-flow: wrap;\n        -webkit-box-pack: center;\n           -ms-flex-pack: center;\n         justify-content: center;\n        margin-top: 0.2em;\n      }\n\n      .datepicker__calendar__label[_ngcontent-%COMP%] {\n        display: inline-block;\n        width: 2.2em;\n        height: 2.2em;\n        margin: 0.2em;\n        line-height: 2.2em;\n        text-align: center;\n        color: #d8d8d8;\n      }\n\n      .datepicker__calendar__month[_ngcontent-%COMP%] {\n        display: -webkit-box;\n        display: -ms-flexbox;\n        display: flex;\n        -ms-flex-flow: wrap;\n            flex-flow: wrap;\n        -webkit-box-pack: center;\n           -ms-flex-pack: center;\n         justify-content: center;\n      }\n\n      .datepicker__calendar__month__day[_ngcontent-%COMP%] {\n        display: inline-block;\n        width: 2.2em;\n        height: 2.2em;\n        margin: 0.2em;\n        border-radius: 2.2em;\n        line-height: 2.2em;\n        text-align: center;\n        -webkit-transition: 0.37s;\n        transition: 0.37s;\n      }\n\n      .datepicker__calendar__nav[_ngcontent-%COMP%] {\n        display: -webkit-box;\n        display: -ms-flexbox;\n        display: flex;\n        -webkit-box-pack: center;\n           -ms-flex-pack: center;\n         justify-content: center;\n        -webkit-box-align: center;\n           -ms-flex-align: center;\n              align-items: center;\n        height: 3em;\n        background-color: #fff;\n        border-bottom: 1px solid #e8e8e8;\n      }\n\n      .datepicker__calendar__nav__arrow[_ngcontent-%COMP%] {\n        width: 0.8em;\n        height: 0.8em;\n        cursor: pointer;\n        -webkit-transition: 0.37s;\n        transition: 0.37s;\n      }\n\n      .datepicker__calendar__nav__arrow[_ngcontent-%COMP%]:hover {\n        -webkit-transform: scale(1.05);\n                transform: scale(1.05);\n      }\n\n      .datepicker__calendar__nav__chevron[_ngcontent-%COMP%] {\n        fill: #bbbbbb;\n        -webkit-transition: 0.37s;\n        transition: 0.37s;\n      }\n\n      .datepicker__calendar__nav__chevron[_ngcontent-%COMP%]:hover {\n        fill: #2b2b2b;\n      }\n\n      .datepicker__calendar__nav__header[_ngcontent-%COMP%] {\n        width: 11em;\n        margin: 0 1em;\n        text-align: center;\n      }\n\n      .datepicker__input[_ngcontent-%COMP%] {\n        outline: none;\n        border-radius: 0.1rem;\n        padding: .2em .6em;\n        font-size: 14px;\n      }'];
var DatepickerComponent_calendarAnimation_states = {
    '*': {},
    'left': {},
    'right': {}
};
function DatepickerComponent_calendarAnimation_factory(view, element, currentState, nextState) {
    var previousPlayers = view.animationContext.getAnimationPlayers(element, ((nextState == 'void') ? null : 'calendarAnimation'));
    var collectedStyles = {};
    var player = null;
    var totalTime = 0;
    var defaultStateStyles = DatepickerComponent_calendarAnimation_states['*'];
    var startStateStyles = DatepickerComponent_calendarAnimation_states[currentState];
    if ((startStateStyles == null)) {
        (startStateStyles = defaultStateStyles);
    }
    var endStateStyles = DatepickerComponent_calendarAnimation_states[nextState];
    if ((endStateStyles == null)) {
        (endStateStyles = defaultStateStyles);
    }
    if (((player == null) && (true && (nextState == 'left')))) {
        player = new import11.AnimationSequencePlayer([view.renderer.animate(element, new import12.AnimationStyles(import13.collectAndResolveStyles(collectedStyles, [startStateStyles])), [
                new import14.AnimationKeyframe(0, new import12.AnimationStyles(import13.collectAndResolveStyles(collectedStyles, [{ 'transform': 'true' }]))),
                new import14.AnimationKeyframe(0.5, new import12.AnimationStyles(import13.collectAndResolveStyles(collectedStyles, [{ 'transform': 'translateX(105%)' }]))),
                new import14.AnimationKeyframe(0.51, new import12.AnimationStyles(import13.collectAndResolveStyles(collectedStyles, [{ 'transform': 'translateX(-130%)' }]))),
                new import14.AnimationKeyframe(1, new import12.AnimationStyles(import13.collectAndResolveStyles(collectedStyles, [{ 'transform': 'translateX(0)' }])))
            ], 180, 0, null, previousPlayers)]);
        totalTime = 180;
    }
    if (((player == null) && (true && (nextState == 'right')))) {
        player = new import11.AnimationSequencePlayer([view.renderer.animate(element, new import12.AnimationStyles(import13.collectAndResolveStyles(collectedStyles, [startStateStyles])), [
                new import14.AnimationKeyframe(0, new import12.AnimationStyles(import13.collectAndResolveStyles(collectedStyles, [{ 'transform': 'true' }]))),
                new import14.AnimationKeyframe(0.5, new import12.AnimationStyles(import13.collectAndResolveStyles(collectedStyles, [{ 'transform': 'translateX(-105%)' }]))),
                new import14.AnimationKeyframe(0.51, new import12.AnimationStyles(import13.collectAndResolveStyles(collectedStyles, [{ 'transform': 'translateX(130%)' }]))),
                new import14.AnimationKeyframe(1, new import12.AnimationStyles(import13.collectAndResolveStyles(collectedStyles, [{ 'transform': 'translateX(0)' }])))
            ], 180, 0, null, previousPlayers)]);
        totalTime = 180;
    }
    if ((player == null)) {
        (player = new import15.NoOpAnimationPlayer());
    }
    player.onDone(function () {
        player.destroy();
        import13.renderStyles(element, view.renderer, import13.prepareFinalAnimationStyles(startStateStyles, endStateStyles));
    });
    new import11.AnimationSequencePlayer(previousPlayers).destroy();
    import13.renderStyles(element, view.renderer, import13.clearStyles(startStateStyles));
    view.animationContext.queueAnimation(element, 'calendarAnimation', player);
    return new import10.AnimationTransition(player, currentState, nextState, totalTime);
}
var View_DatepickerComponent2 = (function (_super) {
    __extends(View_DatepickerComponent2, _super);
    function View_DatepickerComponent2(viewUtils, parentView, parentIndex, parentElement, declaredViewContainer) {
        _super.call(this, View_DatepickerComponent2, renderType_DatepickerComponent, import6.ViewType.EMBEDDED, viewUtils, parentView, parentIndex, parentElement, import7.ChangeDetectorStatus.CheckAlways, declaredViewContainer);
        this._expr_2 = import1.UNINITIALIZED;
    }
    View_DatepickerComponent2.prototype.createInternal = function (rootSelector) {
        this._el_0 = import3.createRenderElement(this.renderer, null, 'div', new import3.InlineArray2(2, 'class', 'datepicker__calendar__label'), null);
        this._text_1 = this.renderer.createText(this._el_0, '', null);
        this.init(this._el_0, (this.renderer.directRenderer ? null : [
            this._el_0,
            this._text_1
        ]), null);
        return null;
    };
    View_DatepickerComponent2.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_2 = import3.inlineInterpolate(1, '\n            ', this.context.$implicit, '\n          ');
        if (import3.checkBinding(throwOnChange, this._expr_2, currVal_2)) {
            this.renderer.setText(this._text_1, currVal_2);
            this._expr_2 = currVal_2;
        }
    };
    View_DatepickerComponent2.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    return View_DatepickerComponent2;
}(import2.AppView));
var View_DatepickerComponent4 = (function (_super) {
    __extends(View_DatepickerComponent4, _super);
    function View_DatepickerComponent4(viewUtils, parentView, parentIndex, parentElement, declaredViewContainer) {
        _super.call(this, View_DatepickerComponent4, renderType_DatepickerComponent, import6.ViewType.EMBEDDED, viewUtils, parentView, parentIndex, parentElement, import7.ChangeDetectorStatus.CheckAlways, declaredViewContainer);
        this._expr_2 = import1.UNINITIALIZED;
    }
    View_DatepickerComponent4.prototype.createInternal = function (rootSelector) {
        this._el_0 = import3.createRenderElement(this.renderer, null, 'span', import3.EMPTY_INLINE_ARRAY, null);
        this._text_1 = this.renderer.createText(this._el_0, '', null);
        this.init(this._el_0, (this.renderer.directRenderer ? null : [
            this._el_0,
            this._text_1
        ]), null);
        return null;
    };
    View_DatepickerComponent4.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_2 = import3.inlineInterpolate(1, '\n                ', this.parentView.context.$implicit.getDate(), '\n              ');
        if (import3.checkBinding(throwOnChange, this._expr_2, currVal_2)) {
            this.renderer.setText(this._text_1, currVal_2);
            this._expr_2 = currVal_2;
        }
    };
    View_DatepickerComponent4.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    return View_DatepickerComponent4;
}(import2.AppView));
var View_DatepickerComponent3 = (function (_super) {
    __extends(View_DatepickerComponent3, _super);
    function View_DatepickerComponent3(viewUtils, parentView, parentIndex, parentElement, declaredViewContainer) {
        _super.call(this, View_DatepickerComponent3, renderType_DatepickerComponent, import6.ViewType.EMBEDDED, viewUtils, parentView, parentIndex, parentElement, import7.ChangeDetectorStatus.CheckAlways, declaredViewContainer);
        this._map_8 = import3.pureProxy4(function (p0, p1, p2, p3) {
            return {
                cursor: p0,
                'background-color': p1,
                color: p2,
                'pointer-events': p3
            };
        });
    }
    View_DatepickerComponent3.prototype.createInternal = function (rootSelector) {
        this._el_0 = import3.createRenderElement(this.renderer, null, 'div', new import3.InlineArray2(2, 'class', 'datepicker__calendar__month__day'), null);
        this._NgStyle_0_3 = new import17.Wrapper_NgStyle(this.parentView.parentView.parentView.injectorGet(import19.KeyValueDiffers, this.parentView.parentView.parentIndex), new import9.ElementRef(this._el_0), this.renderer);
        this._text_1 = this.renderer.createText(this._el_0, '\n              ', null);
        this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0, null);
        this._vc_2 = new import16.ViewContainer(2, 0, this, this._anchor_2);
        this._TemplateRef_2_5 = new import20.TemplateRef_(this, 2, this._anchor_2);
        this._NgIf_2_6 = new import18.Wrapper_NgIf(this._vc_2.vcRef, this._TemplateRef_2_5);
        this._text_3 = this.renderer.createText(this._el_0, '\n            ', null);
        var disposable_0 = import3.subscribeToRenderElement(this, this._el_0, new import3.InlineArray8(6, 'click', null, 'mouseenter', null, 'mouseleave', null), this.eventHandler(this.handleEvent_0));
        this.init(this._el_0, (this.renderer.directRenderer ? null : [
            this._el_0,
            this._text_1,
            this._anchor_2,
            this._text_3
        ]), [disposable_0]);
        return null;
    };
    View_DatepickerComponent3.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import20.TemplateRef) && (2 === requestNodeIndex))) {
            return this._TemplateRef_2_5;
        }
        if (((token === import21.NgIf) && (2 === requestNodeIndex))) {
            return this._NgIf_2_6.context;
        }
        if (((token === import22.NgStyle) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 3)))) {
            return this._NgStyle_0_3.context;
        }
        return notFoundResult;
    };
    View_DatepickerComponent3.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_0_0_0 = this._map_8(((this.context.$implicit == 0) ? 'initial' : 'pointer'), this.parentView.parentView.context.getDayBackgroundColor(this.context.$implicit), (this.parentView.parentView.context.isHoveredDay(this.context.$implicit) ? this.parentView.parentView.context.accentColor : this.parentView.parentView.context.getDayFontColor(this.context.$implicit)), ((this.context.$implicit == 0) ? 'none' : ''));
        this._NgStyle_0_3.check_ngStyle(currVal_0_0_0, throwOnChange, false);
        this._NgStyle_0_3.ngDoCheck(this, this._el_0, throwOnChange);
        var currVal_2_0_0 = (this.context.$implicit != 0);
        this._NgIf_2_6.check_ngIf(currVal_2_0_0, throwOnChange, false);
        this._NgIf_2_6.ngDoCheck(this, this._anchor_2, throwOnChange);
        this._vc_2.detectChangesInNestedViews(throwOnChange);
    };
    View_DatepickerComponent3.prototype.destroyInternal = function () {
        this._vc_2.destroyNestedViews();
    };
    View_DatepickerComponent3.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    View_DatepickerComponent3.prototype.createEmbeddedViewInternal = function (nodeIndex) {
        if ((nodeIndex == 2)) {
            return new View_DatepickerComponent4(this.viewUtils, this, 2, this._anchor_2, this._vc_2);
        }
        return null;
    };
    View_DatepickerComponent3.prototype.handleEvent_0 = function (eventName, $event) {
        this.markPathToRootAsCheckOnce();
        var result = true;
        if ((eventName == 'click')) {
            var pd_sub_0 = (this.parentView.parentView.context.onSelectDay(this.context.$implicit) !== false);
            result = (pd_sub_0 && result);
        }
        if ((eventName == 'mouseenter')) {
            var pd_sub_1 = ((this.parentView.parentView.context.hoveredDay = this.context.$implicit) !== false);
            result = (pd_sub_1 && result);
        }
        if ((eventName == 'mouseleave')) {
            var pd_sub_2 = ((this.parentView.parentView.context.hoveredDay = null) !== false);
            result = (pd_sub_2 && result);
        }
        return result;
    };
    return View_DatepickerComponent3;
}(import2.AppView));
var View_DatepickerComponent1 = (function (_super) {
    __extends(View_DatepickerComponent1, _super);
    function View_DatepickerComponent1(viewUtils, parentView, parentIndex, parentElement, declaredViewContainer) {
        _super.call(this, View_DatepickerComponent1, renderType_DatepickerComponent, import6.ViewType.EMBEDDED, viewUtils, parentView, parentIndex, parentElement, import7.ChangeDetectorStatus.CheckAlways, declaredViewContainer);
        this._expr_59 = import1.UNINITIALIZED;
        this._expr_60 = import1.UNINITIALIZED;
    }
    View_DatepickerComponent1.prototype.createInternal = function (rootSelector) {
        this._el_0 = import3.createRenderElement(this.renderer, null, 'div', new import3.InlineArray2(2, 'class', 'datepicker__calendar'), null);
        this._text_1 = this.renderer.createText(this._el_0, '\n        ', null);
        this._el_2 = import3.createRenderElement(this.renderer, this._el_0, 'div', new import3.InlineArray2(2, 'class', 'datepicker__calendar__nav'), null);
        this._text_3 = this.renderer.createText(this._el_2, '\n          ', null);
        this._el_4 = import3.createRenderElement(this.renderer, this._el_2, 'div', new import3.InlineArray2(2, 'class', 'datepicker__calendar__nav__arrow'), null);
        this._text_5 = this.renderer.createText(this._el_4, '\n          ', null);
        this._el_6 = import3.createRenderElement(this.renderer, this._el_4, ':svg:svg', new import3.InlineArray8(8, 'class', 'datepicker__calendar__nav__chevron', 'viewBox', '0 0 50 50', 'x', '0px', 'y', '0px'), null);
        this._text_7 = this.renderer.createText(this._el_6, '\n            ', null);
        this._el_8 = import3.createRenderElement(this.renderer, this._el_6, ':svg:g', import3.EMPTY_INLINE_ARRAY, null);
        this._text_9 = this.renderer.createText(this._el_8, '\n                ', null);
        this._el_10 = import3.createRenderElement(this.renderer, this._el_8, ':svg:path', new import3.InlineArray2(2, 'd', 'M39.7,7.1c0.5,0.5,0.5,1.2,0,1.7L29,19.6c-0.5,0.5-1.2,1.2-1.7,1.7L16.5,32.1c-0.5,0.5-1.2,0.5-1.7,0l-2.3-2.3\n                    c-0.5-0.5-1.2-1.2-1.7-1.7l-2.3-2.3c-0.5-0.5-0.5-1.2,0-1.7l10.8-10.8c0.5-0.5,1.2-1.2,1.7-1.7L31.7,0.8c0.5-0.5,1.2-0.5,1.7,0\n                    l2.3,2.3c0.5,0.5,1.2,1.2,1.7,1.7L39.7,7.1z'), null);
        this._text_11 = this.renderer.createText(this._el_8, '\n            ', null);
        this._text_12 = this.renderer.createText(this._el_6, '\n            ', null);
        this._el_13 = import3.createRenderElement(this.renderer, this._el_6, ':svg:g', import3.EMPTY_INLINE_ARRAY, null);
        this._text_14 = this.renderer.createText(this._el_13, '\n                ', null);
        this._el_15 = import3.createRenderElement(this.renderer, this._el_13, ':svg:path', new import3.InlineArray2(2, 'd', 'M33.4,49c-0.5,0.5-1.2,0.5-1.7,0L20.9,38.2c-0.5-0.5-1.2-1.2-1.7-1.7L8.4,25.7c-0.5-0.5-0.5-1.2,0-1.7l2.3-2.3\n                    c0.5-0.5,1.2-1.2,1.7-1.7l2.3-2.3c0.5-0.5,1.2-0.5,1.7,0l10.8,10.8c0.5,0.5,1.2,1.2,1.7,1.7l10.8,10.8c0.5,0.5,0.5,1.2,0,1.7\n                    L37.4,45c-0.5,0.5-1.2,1.2-1.7,1.7L33.4,49z'), null);
        this._text_16 = this.renderer.createText(this._el_13, '\n            ', null);
        this._text_17 = this.renderer.createText(this._el_6, '\n          ', null);
        this._text_18 = this.renderer.createText(this._el_4, '\n          ', null);
        this._text_19 = this.renderer.createText(this._el_2, '\n          ', null);
        this._el_20 = import3.createRenderElement(this.renderer, this._el_2, 'div', new import3.InlineArray2(2, 'class', 'datepicker__calendar__nav__header'), null);
        this._text_21 = this.renderer.createText(this._el_20, '', null);
        this._text_22 = this.renderer.createText(this._el_2, '\n          ', null);
        this._el_23 = import3.createRenderElement(this.renderer, this._el_2, 'div', new import3.InlineArray2(2, 'class', 'datepicker__calendar__nav__arrow'), null);
        this._text_24 = this.renderer.createText(this._el_23, '\n            ', null);
        this._el_25 = import3.createRenderElement(this.renderer, this._el_23, ':svg:svg', new import3.InlineArray8(8, 'class', 'datepicker__calendar__nav__chevron', 'viewBox', '0 0 50 50', 'x', '0px', 'y', '0px'), null);
        this._text_26 = this.renderer.createText(this._el_25, '\n              ', null);
        this._el_27 = import3.createRenderElement(this.renderer, this._el_25, ':svg:g', import3.EMPTY_INLINE_ARRAY, null);
        this._text_28 = this.renderer.createText(this._el_27, '\n                ', null);
        this._el_29 = import3.createRenderElement(this.renderer, this._el_27, ':svg:path', new import3.InlineArray2(2, 'd', 'M8.4,7.1c-0.5,0.5-0.5,1.2,0,1.7l10.8,10.8c0.5,0.5,1.2,1.2,1.7,1.7l10.8,10.8c0.5,0.5,1.2,0.5,1.7,0l2.3-2.3\n                    c0.5-0.5,1.2-1.2,1.7-1.7l2.3-2.3c0.5-0.5,0.5-1.2,0-1.7L29,13.2c-0.5-0.5-1.2-1.2-1.7-1.7L16.5,0.8c-0.5-0.5-1.2-0.5-1.7,0\n                    l-2.3,2.3c-0.5,0.5-1.2,1.2-1.7,1.7L8.4,7.1z'), null);
        this._text_30 = this.renderer.createText(this._el_27, '\n              ', null);
        this._text_31 = this.renderer.createText(this._el_25, '\n              ', null);
        this._el_32 = import3.createRenderElement(this.renderer, this._el_25, ':svg:g', import3.EMPTY_INLINE_ARRAY, null);
        this._text_33 = this.renderer.createText(this._el_32, '\n                ', null);
        this._el_34 = import3.createRenderElement(this.renderer, this._el_32, ':svg:path', new import3.InlineArray2(2, 'd', 'M14.8,49c0.5,0.5,1.2,0.5,1.7,0l10.8-10.8c0.5-0.5,1.2-1.2,1.7-1.7l10.8-10.8c0.5-0.5,0.5-1.2,0-1.7l-2.3-2.3\n                    c-0.5-0.5-1.2-1.2-1.7-1.7l-2.3-2.3c-0.5-0.5-1.2-0.5-1.7,0L20.9,28.5c-0.5,0.5-1.2,1.2-1.7,1.7L8.4,40.9c-0.5,0.5-0.5,1.2,0,1.7\n                    l2.3,2.3c0.5,0.5,1.2,1.2,1.7,1.7L14.8,49z'), null);
        this._text_35 = this.renderer.createText(this._el_32, '\n              ', null);
        this._text_36 = this.renderer.createText(this._el_25, '\n            ', null);
        this._text_37 = this.renderer.createText(this._el_23, '\n          ', null);
        this._text_38 = this.renderer.createText(this._el_2, '\n        ', null);
        this._text_39 = this.renderer.createText(this._el_0, '\n        ', null);
        this._el_40 = import3.createRenderElement(this.renderer, this._el_0, 'div', new import3.InlineArray2(2, 'class', 'datepicker__calendar__content'), null);
        this._text_41 = this.renderer.createText(this._el_40, '\n          ', null);
        this._anchor_42 = this.renderer.createTemplateAnchor(this._el_40, null);
        this._vc_42 = new import16.ViewContainer(42, 40, this, this._anchor_42);
        this._TemplateRef_42_5 = new import20.TemplateRef_(this, 42, this._anchor_42);
        this._NgFor_42_6 = new import23.Wrapper_NgFor(this._vc_42.vcRef, this._TemplateRef_42_5, this.parentView.parentView.injectorGet(import24.IterableDiffers, this.parentView.parentIndex), this.parentView.ref);
        this._text_43 = this.renderer.createText(this._el_40, '\n          ', null);
        this._el_44 = import3.createRenderElement(this.renderer, this._el_40, 'div', new import3.InlineArray2(2, 'class', 'datepicker__calendar__month'), null);
        this._text_45 = this.renderer.createText(this._el_44, '\n            ', null);
        this._anchor_46 = this.renderer.createTemplateAnchor(this._el_44, null);
        this._vc_46 = new import16.ViewContainer(46, 44, this, this._anchor_46);
        this._TemplateRef_46_5 = new import20.TemplateRef_(this, 46, this._anchor_46);
        this._NgFor_46_6 = new import23.Wrapper_NgFor(this._vc_46.vcRef, this._TemplateRef_46_5, this.parentView.parentView.injectorGet(import24.IterableDiffers, this.parentView.parentIndex), this.parentView.ref);
        this._text_47 = this.renderer.createText(this._el_44, '\n          ', null);
        this._text_48 = this.renderer.createText(this._el_40, '\n          ', null);
        this._el_49 = import3.createRenderElement(this.renderer, this._el_40, 'div', new import3.InlineArray2(2, 'class', 'datepicker__calendar__cancel'), null);
        this._text_50 = this.renderer.createText(this._el_49, '\n            Cancel\n          ', null);
        this._text_51 = this.renderer.createText(this._el_40, '\n        ', null);
        this._text_52 = this.renderer.createText(this._el_0, '\n      ', null);
        var disposable_0 = import3.subscribeToRenderElement(this, this._el_4, new import3.InlineArray2(2, 'click', null), this.eventHandler(this.handleEvent_4));
        var disposable_1 = import3.subscribeToRenderElement(this, this._el_23, new import3.InlineArray2(2, 'click', null), this.eventHandler(this.handleEvent_23));
        var disposable_2 = import3.subscribeToRenderElement(this, this._el_49, new import3.InlineArray2(2, 'click', null), this.eventHandler(this.handleEvent_49));
        this.init(this._el_0, (this.renderer.directRenderer ? null : [
            this._el_0,
            this._text_1,
            this._el_2,
            this._text_3,
            this._el_4,
            this._text_5,
            this._el_6,
            this._text_7,
            this._el_8,
            this._text_9,
            this._el_10,
            this._text_11,
            this._text_12,
            this._el_13,
            this._text_14,
            this._el_15,
            this._text_16,
            this._text_17,
            this._text_18,
            this._text_19,
            this._el_20,
            this._text_21,
            this._text_22,
            this._el_23,
            this._text_24,
            this._el_25,
            this._text_26,
            this._el_27,
            this._text_28,
            this._el_29,
            this._text_30,
            this._text_31,
            this._el_32,
            this._text_33,
            this._el_34,
            this._text_35,
            this._text_36,
            this._text_37,
            this._text_38,
            this._text_39,
            this._el_40,
            this._text_41,
            this._anchor_42,
            this._text_43,
            this._el_44,
            this._text_45,
            this._anchor_46,
            this._text_47,
            this._text_48,
            this._el_49,
            this._text_50,
            this._text_51,
            this._text_52
        ]), [
            disposable_0,
            disposable_1,
            disposable_2
        ]);
        return null;
    };
    View_DatepickerComponent1.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import20.TemplateRef) && (42 === requestNodeIndex))) {
            return this._TemplateRef_42_5;
        }
        if (((token === import25.NgFor) && (42 === requestNodeIndex))) {
            return this._NgFor_42_6.context;
        }
        if (((token === import20.TemplateRef) && (46 === requestNodeIndex))) {
            return this._TemplateRef_46_5;
        }
        if (((token === import25.NgFor) && (46 === requestNodeIndex))) {
            return this._NgFor_46_6.context;
        }
        return notFoundResult;
    };
    View_DatepickerComponent1.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_60 = this.parentView.context.animate;
        if (import3.checkBinding(throwOnChange, this._expr_60, currVal_60)) {
            var animationTransition_calendarAnimation = this.componentType.animations['calendarAnimation'](this, this._el_44, ((this._expr_60 == import1.UNINITIALIZED) ? 'void' : this._expr_60), ((currVal_60 == import1.UNINITIALIZED) ? 'void' : currVal_60));
            this._expr_60 = currVal_60;
        }
        var currVal_42_0_0 = this.parentView.context.dayNames;
        this._NgFor_42_6.check_ngForOf(currVal_42_0_0, throwOnChange, false);
        this._NgFor_42_6.ngDoCheck(this, this._anchor_42, throwOnChange);
        var currVal_46_0_0 = this.parentView.context.calendarDays;
        this._NgFor_46_6.check_ngForOf(currVal_46_0_0, throwOnChange, false);
        this._NgFor_46_6.ngDoCheck(this, this._anchor_46, throwOnChange);
        this._vc_42.detectChangesInNestedViews(throwOnChange);
        this._vc_46.detectChangesInNestedViews(throwOnChange);
        var currVal_59 = import3.inlineInterpolate(2, '\n            ', this.parentView.context.currentMonth, ' ', this.parentView.context.currentYear, '\n          ');
        if (import3.checkBinding(throwOnChange, this._expr_59, currVal_59)) {
            this.renderer.setText(this._text_21, currVal_59);
            this._expr_59 = currVal_59;
        }
    };
    View_DatepickerComponent1.prototype.destroyInternal = function () {
        this._vc_42.destroyNestedViews();
        this._vc_46.destroyNestedViews();
    };
    View_DatepickerComponent1.prototype.detachInternal = function () {
        var animationTransition_calendarAnimation = this.componentType.animations['calendarAnimation'](this, this._el_44, this._expr_60, 'void');
    };
    View_DatepickerComponent1.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    View_DatepickerComponent1.prototype.createEmbeddedViewInternal = function (nodeIndex) {
        if ((nodeIndex == 42)) {
            return new View_DatepickerComponent2(this.viewUtils, this, 42, this._anchor_42, this._vc_42);
        }
        if ((nodeIndex == 46)) {
            return new View_DatepickerComponent3(this.viewUtils, this, 46, this._anchor_46, this._vc_46);
        }
        return null;
    };
    View_DatepickerComponent1.prototype.handleEvent_4 = function (eventName, $event) {
        this.markPathToRootAsCheckOnce();
        var result = true;
        if ((eventName == 'click')) {
            var pd_sub_0 = (this.parentView.context.onArrowLeftClick() !== false);
            result = (pd_sub_0 && result);
        }
        return result;
    };
    View_DatepickerComponent1.prototype.handleEvent_23 = function (eventName, $event) {
        this.markPathToRootAsCheckOnce();
        var result = true;
        if ((eventName == 'click')) {
            var pd_sub_0 = (this.parentView.context.onArrowRightClick() !== false);
            result = (pd_sub_0 && result);
        }
        return result;
    };
    View_DatepickerComponent1.prototype.handleEvent_49 = function (eventName, $event) {
        this.markPathToRootAsCheckOnce();
        var result = true;
        if ((eventName == 'click')) {
            var pd_sub_0 = (this.parentView.context.onCancel() !== false);
            result = (pd_sub_0 && result);
        }
        return result;
    };
    return View_DatepickerComponent1;
}(import2.AppView));
var renderType_DatepickerComponent = import3.createRenderComponentType('', 0, import5.ViewEncapsulation.Emulated, styles_DatepickerComponent, { calendarAnimation: DatepickerComponent_calendarAnimation_factory });
export var View_DatepickerComponent0 = (function (_super) {
    __extends(View_DatepickerComponent0, _super);
    function View_DatepickerComponent0(viewUtils, parentView, parentIndex, parentElement) {
        _super.call(this, View_DatepickerComponent0, renderType_DatepickerComponent, import6.ViewType.COMPONENT, viewUtils, parentView, parentIndex, parentElement, import7.ChangeDetectorStatus.CheckAlways);
        this._map_18 = import3.pureProxy1(function (p0) {
            return { 'font-family': p0 };
        });
        this._expr_19 = import1.UNINITIALIZED;
        this._map_20 = import3.pureProxy3(function (p0, p1, p2) {
            return {
                color: p0,
                'background-color': p1,
                border: p2
            };
        });
    }
    View_DatepickerComponent0.prototype.createInternal = function (rootSelector) {
        var parentRenderNode = this.renderer.createViewRoot(this.parentElement);
        this._text_0 = this.renderer.createText(parentRenderNode, '\n    ', null);
        this._el_1 = import3.createRenderElement(this.renderer, parentRenderNode, 'div', new import3.InlineArray2(2, 'class', 'datepicker'), null);
        this._NgStyle_1_3 = new import17.Wrapper_NgStyle(this.parentView.injectorGet(import19.KeyValueDiffers, this.parentIndex), new import9.ElementRef(this._el_1), this.renderer);
        this._text_2 = this.renderer.createText(this._el_1, '\n      ', null);
        this._el_3 = import3.createRenderElement(this.renderer, this._el_1, 'input', new import3.InlineArray4(4, 'class', 'datepicker__input', 'readonly', 'true'), null);
        this._NgStyle_3_3 = new import17.Wrapper_NgStyle(this.parentView.injectorGet(import19.KeyValueDiffers, this.parentIndex), new import9.ElementRef(this._el_3), this.renderer);
        this._DefaultValueAccessor_3_4 = new import26.Wrapper_DefaultValueAccessor(this.renderer, new import9.ElementRef(this._el_3));
        this._NG_VALUE_ACCESSOR_3_5 = [this._DefaultValueAccessor_3_4.context];
        this._NgModel_3_6 = new import27.Wrapper_NgModel(null, null, null, this._NG_VALUE_ACCESSOR_3_5);
        this._NgControl_3_7 = this._NgModel_3_6.context;
        this._NgControlStatus_3_8 = new import28.Wrapper_NgControlStatus(this._NgControl_3_7);
        this._text_4 = this.renderer.createText(this._el_1, '\n      ', null);
        this._anchor_5 = this.renderer.createTemplateAnchor(this._el_1, null);
        this._vc_5 = new import16.ViewContainer(5, 1, this, this._anchor_5);
        this._TemplateRef_5_5 = new import20.TemplateRef_(this, 5, this._anchor_5);
        this._NgIf_5_6 = new import18.Wrapper_NgIf(this._vc_5.vcRef, this._TemplateRef_5_5);
        this._text_6 = this.renderer.createText(this._el_1, '\n    ', null);
        this._text_7 = this.renderer.createText(parentRenderNode, '\n    ', null);
        var disposable_0 = import3.subscribeToRenderElement(this, this._el_3, new import3.InlineArray8(8, 'click', null, 'ngModelChange', null, 'input', null, 'blur', null), this.eventHandler(this.handleEvent_3));
        this._NgModel_3_6.subscribe(this, this.eventHandler(this.handleEvent_3), true);
        this.init(null, (this.renderer.directRenderer ? null : [
            this._text_0,
            this._el_1,
            this._text_2,
            this._el_3,
            this._text_4,
            this._anchor_5,
            this._text_6,
            this._text_7
        ]), [disposable_0]);
        return null;
    };
    View_DatepickerComponent0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import22.NgStyle) && (3 === requestNodeIndex))) {
            return this._NgStyle_3_3.context;
        }
        if (((token === import29.DefaultValueAccessor) && (3 === requestNodeIndex))) {
            return this._DefaultValueAccessor_3_4.context;
        }
        if (((token === import30.NG_VALUE_ACCESSOR) && (3 === requestNodeIndex))) {
            return this._NG_VALUE_ACCESSOR_3_5;
        }
        if (((token === import31.NgModel) && (3 === requestNodeIndex))) {
            return this._NgModel_3_6.context;
        }
        if (((token === import32.NgControl) && (3 === requestNodeIndex))) {
            return this._NgControl_3_7;
        }
        if (((token === import33.NgControlStatus) && (3 === requestNodeIndex))) {
            return this._NgControlStatus_3_8.context;
        }
        if (((token === import20.TemplateRef) && (5 === requestNodeIndex))) {
            return this._TemplateRef_5_5;
        }
        if (((token === import21.NgIf) && (5 === requestNodeIndex))) {
            return this._NgIf_5_6.context;
        }
        if (((token === import22.NgStyle) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 6)))) {
            return this._NgStyle_1_3.context;
        }
        return notFoundResult;
    };
    View_DatepickerComponent0.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_1_0_0 = this._map_18(this.context.fontFamily);
        this._NgStyle_1_3.check_ngStyle(currVal_1_0_0, throwOnChange, false);
        this._NgStyle_1_3.ngDoCheck(this, this._el_1, throwOnChange);
        var currVal_3_0_0 = this._map_20((this.context.altInputStyle ? this.context.colors['white'] : this.context.colors['black']), (this.context.altInputStyle ? this.context.accentColor : this.context.colors['white']), (this.context.altInputStyle ? '' : '1px solid #dadada'));
        this._NgStyle_3_3.check_ngStyle(currVal_3_0_0, throwOnChange, false);
        this._NgStyle_3_3.ngDoCheck(this, this._el_3, throwOnChange);
        this._DefaultValueAccessor_3_4.ngDoCheck(this, this._el_3, throwOnChange);
        var currVal_3_2_0 = this.context.disabled;
        this._NgModel_3_6.check_isDisabled(currVal_3_2_0, throwOnChange, false);
        var currVal_3_2_1 = this.context.inputText;
        this._NgModel_3_6.check_model(currVal_3_2_1, throwOnChange, false);
        this._NgModel_3_6.ngDoCheck(this, this._el_3, throwOnChange);
        this._NgControlStatus_3_8.ngDoCheck(this, this._el_3, throwOnChange);
        var currVal_5_0_0 = this.context.showCalendar;
        this._NgIf_5_6.check_ngIf(currVal_5_0_0, throwOnChange, false);
        this._NgIf_5_6.ngDoCheck(this, this._anchor_5, throwOnChange);
        this._vc_5.detectChangesInNestedViews(throwOnChange);
        var currVal_19 = this.context.placeholder;
        if (import3.checkBinding(throwOnChange, this._expr_19, currVal_19)) {
            this.renderer.setElementProperty(this._el_3, 'placeholder', currVal_19);
            this._expr_19 = currVal_19;
        }
        this._NgControlStatus_3_8.checkHost(this, this, this._el_3, throwOnChange);
    };
    View_DatepickerComponent0.prototype.destroyInternal = function () {
        this._vc_5.destroyNestedViews();
        this._NgModel_3_6.ngOnDestroy();
    };
    View_DatepickerComponent0.prototype.createEmbeddedViewInternal = function (nodeIndex) {
        if ((nodeIndex == 5)) {
            return new View_DatepickerComponent1(this.viewUtils, this, 5, this._anchor_5, this._vc_5);
        }
        return null;
    };
    View_DatepickerComponent0.prototype.handleEvent_3 = function (eventName, $event) {
        this.markPathToRootAsCheckOnce();
        var result = true;
        result = (this._DefaultValueAccessor_3_4.handleEvent(eventName, $event) && result);
        if ((eventName == 'click')) {
            var pd_sub_0 = (this.context.onInputClick() !== false);
            result = (pd_sub_0 && result);
        }
        if ((eventName == 'ngModelChange')) {
            var pd_sub_1 = ((this.context.inputText = $event) !== false);
            result = (pd_sub_1 && result);
        }
        return result;
    };
    return View_DatepickerComponent0;
}(import2.AppView));
