// primeng/components/calendar/calendar.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
import * as import0 from '@angular/core/src/linker/ng_module_factory';
import * as import1 from 'primeng/components/calendar/calendar';
import * as import2 from '@angular/common/src/common_module';
import * as import3 from 'primeng/components/button/button';
import * as import4 from 'primeng/components/inputtext/inputtext';
import * as import5 from '@angular/common/src/localization';
import * as import7 from '@angular/core/src/i18n/tokens';
import * as import8 from '@angular/core/src/change_detection/change_detection_util';
import * as import9 from '@angular/core/src/linker/view';
import * as import10 from '@angular/core/src/linker/view_utils';
import * as import12 from '@angular/core/src/metadata/view';
import * as import13 from 'primeng/components/dom/domhandler';
import * as import14 from '@angular/core/src/linker/view_type';
import * as import15 from '@angular/core/src/change_detection/constants';
import * as import16 from '@angular/core/src/linker/component_factory';
import * as import17 from '@angular/core/src/linker/element_ref';
import * as import18 from '@angular/forms/src/directives/control_value_accessor';
import * as import19 from '@angular/core/src/animation/animation_transition';
import * as import20 from '@angular/core/src/animation/animation_sequence_player';
import * as import21 from '@angular/core/src/animation/animation_styles';
import * as import22 from '@angular/core/src/animation/animation_style_util';
import * as import23 from '@angular/core/src/animation/animation_keyframe';
import * as import24 from '@angular/core/src/animation/animation_player';
import * as import25 from '../../../@angular/common/src/directives/ng_class.ngfactory';
import * as import26 from '../button/button.ngfactory';
import * as import27 from '@angular/core/src/linker/view_container';
import * as import28 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import29 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import30 from '@angular/common/src/directives/ng_class';
import * as import31 from '../../../@angular/common/src/directives/ng_style.ngfactory';
import * as import32 from '../inputtext/inputtext.ngfactory';
import * as import33 from '../../../@angular/common/src/directives/ng_if.ngfactory';
import * as import34 from '@angular/core/src/linker/template_ref';
import * as import35 from '@angular/common/src/directives/ng_style';
import * as import36 from '@angular/common/src/directives/ng_if';
import * as import37 from '../../../@angular/common/src/directives/ng_for.ngfactory';
import * as import38 from '@angular/common/src/directives/ng_for';
import * as import39 from '@angular/core/src/linker/query_list';
var CalendarModuleInjector = (function (_super) {
    __extends(CalendarModuleInjector, _super);
    function CalendarModuleInjector(parent) {
        _super.call(this, parent, [], []);
    }
    Object.defineProperty(CalendarModuleInjector.prototype, "_NgLocalization_4", {
        get: function () {
            if ((this.__NgLocalization_4 == null)) {
                (this.__NgLocalization_4 = new import5.NgLocaleLocalization(this.parent.get(import7.LOCALE_ID)));
            }
            return this.__NgLocalization_4;
        },
        enumerable: true,
        configurable: true
    });
    CalendarModuleInjector.prototype.createInternal = function () {
        this._CommonModule_0 = new import2.CommonModule();
        this._ButtonModule_1 = new import3.ButtonModule();
        this._InputTextModule_2 = new import4.InputTextModule();
        this._CalendarModule_3 = new import1.CalendarModule();
        return this._CalendarModule_3;
    };
    CalendarModuleInjector.prototype.getInternal = function (token, notFoundResult) {
        if ((token === import2.CommonModule)) {
            return this._CommonModule_0;
        }
        if ((token === import3.ButtonModule)) {
            return this._ButtonModule_1;
        }
        if ((token === import4.InputTextModule)) {
            return this._InputTextModule_2;
        }
        if ((token === import1.CalendarModule)) {
            return this._CalendarModule_3;
        }
        if ((token === import5.NgLocalization)) {
            return this._NgLocalization_4;
        }
        return notFoundResult;
    };
    CalendarModuleInjector.prototype.destroyInternal = function () {
    };
    return CalendarModuleInjector;
}(import0.NgModuleInjector));
export var CalendarModuleNgFactory = new import0.NgModuleFactory(CalendarModuleInjector, import1.CalendarModule);
export var Wrapper_Calendar = (function () {
    function Wrapper_Calendar(p0, p1, p2) {
        this._changed = false;
        this.context = new import1.Calendar(p0, p1, p2);
        this._expr_0 = import8.UNINITIALIZED;
        this._expr_1 = import8.UNINITIALIZED;
        this._expr_2 = import8.UNINITIALIZED;
        this._expr_3 = import8.UNINITIALIZED;
        this._expr_4 = import8.UNINITIALIZED;
        this._expr_5 = import8.UNINITIALIZED;
        this._expr_6 = import8.UNINITIALIZED;
        this._expr_7 = import8.UNINITIALIZED;
        this._expr_8 = import8.UNINITIALIZED;
        this._expr_9 = import8.UNINITIALIZED;
        this._expr_10 = import8.UNINITIALIZED;
        this._expr_11 = import8.UNINITIALIZED;
        this._expr_12 = import8.UNINITIALIZED;
        this._expr_13 = import8.UNINITIALIZED;
        this._expr_14 = import8.UNINITIALIZED;
        this._expr_15 = import8.UNINITIALIZED;
        this._expr_16 = import8.UNINITIALIZED;
        this._expr_17 = import8.UNINITIALIZED;
        this._expr_18 = import8.UNINITIALIZED;
        this._expr_19 = import8.UNINITIALIZED;
        this._expr_20 = import8.UNINITIALIZED;
        this._expr_21 = import8.UNINITIALIZED;
        this._expr_22 = import8.UNINITIALIZED;
        this._expr_23 = import8.UNINITIALIZED;
        this._expr_24 = import8.UNINITIALIZED;
        this._expr_25 = import8.UNINITIALIZED;
        this._expr_26 = import8.UNINITIALIZED;
        this._expr_27 = import8.UNINITIALIZED;
        this._expr_28 = import8.UNINITIALIZED;
        this._expr_29 = import8.UNINITIALIZED;
    }
    Wrapper_Calendar.prototype.ngOnDetach = function (view, componentView, el) {
    };
    Wrapper_Calendar.prototype.ngOnDestroy = function () {
        this.context.ngOnDestroy();
        (this.subscription0 && this.subscription0.unsubscribe());
        (this.subscription1 && this.subscription1.unsubscribe());
    };
    Wrapper_Calendar.prototype.check_defaultDate = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import10.checkBinding(throwOnChange, this._expr_0, currValue))) {
            this._changed = true;
            this.context.defaultDate = currValue;
            this._expr_0 = currValue;
        }
    };
    Wrapper_Calendar.prototype.check_style = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import10.checkBinding(throwOnChange, this._expr_1, currValue))) {
            this._changed = true;
            this.context.style = currValue;
            this._expr_1 = currValue;
        }
    };
    Wrapper_Calendar.prototype.check_styleClass = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import10.checkBinding(throwOnChange, this._expr_2, currValue))) {
            this._changed = true;
            this.context.styleClass = currValue;
            this._expr_2 = currValue;
        }
    };
    Wrapper_Calendar.prototype.check_inputStyle = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import10.checkBinding(throwOnChange, this._expr_3, currValue))) {
            this._changed = true;
            this.context.inputStyle = currValue;
            this._expr_3 = currValue;
        }
    };
    Wrapper_Calendar.prototype.check_inputStyleClass = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import10.checkBinding(throwOnChange, this._expr_4, currValue))) {
            this._changed = true;
            this.context.inputStyleClass = currValue;
            this._expr_4 = currValue;
        }
    };
    Wrapper_Calendar.prototype.check_placeholder = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import10.checkBinding(throwOnChange, this._expr_5, currValue))) {
            this._changed = true;
            this.context.placeholder = currValue;
            this._expr_5 = currValue;
        }
    };
    Wrapper_Calendar.prototype.check_disabled = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import10.checkBinding(throwOnChange, this._expr_6, currValue))) {
            this._changed = true;
            this.context.disabled = currValue;
            this._expr_6 = currValue;
        }
    };
    Wrapper_Calendar.prototype.check_dateFormat = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import10.checkBinding(throwOnChange, this._expr_7, currValue))) {
            this._changed = true;
            this.context.dateFormat = currValue;
            this._expr_7 = currValue;
        }
    };
    Wrapper_Calendar.prototype.check_inline = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import10.checkBinding(throwOnChange, this._expr_8, currValue))) {
            this._changed = true;
            this.context.inline = currValue;
            this._expr_8 = currValue;
        }
    };
    Wrapper_Calendar.prototype.check_showOtherMonths = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import10.checkBinding(throwOnChange, this._expr_9, currValue))) {
            this._changed = true;
            this.context.showOtherMonths = currValue;
            this._expr_9 = currValue;
        }
    };
    Wrapper_Calendar.prototype.check_selectOtherMonths = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import10.checkBinding(throwOnChange, this._expr_10, currValue))) {
            this._changed = true;
            this.context.selectOtherMonths = currValue;
            this._expr_10 = currValue;
        }
    };
    Wrapper_Calendar.prototype.check_showIcon = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import10.checkBinding(throwOnChange, this._expr_11, currValue))) {
            this._changed = true;
            this.context.showIcon = currValue;
            this._expr_11 = currValue;
        }
    };
    Wrapper_Calendar.prototype.check_icon = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import10.checkBinding(throwOnChange, this._expr_12, currValue))) {
            this._changed = true;
            this.context.icon = currValue;
            this._expr_12 = currValue;
        }
    };
    Wrapper_Calendar.prototype.check_appendTo = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import10.checkBinding(throwOnChange, this._expr_13, currValue))) {
            this._changed = true;
            this.context.appendTo = currValue;
            this._expr_13 = currValue;
        }
    };
    Wrapper_Calendar.prototype.check_readonlyInput = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import10.checkBinding(throwOnChange, this._expr_14, currValue))) {
            this._changed = true;
            this.context.readonlyInput = currValue;
            this._expr_14 = currValue;
        }
    };
    Wrapper_Calendar.prototype.check_shortYearCutoff = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import10.checkBinding(throwOnChange, this._expr_15, currValue))) {
            this._changed = true;
            this.context.shortYearCutoff = currValue;
            this._expr_15 = currValue;
        }
    };
    Wrapper_Calendar.prototype.check_monthNavigator = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import10.checkBinding(throwOnChange, this._expr_16, currValue))) {
            this._changed = true;
            this.context.monthNavigator = currValue;
            this._expr_16 = currValue;
        }
    };
    Wrapper_Calendar.prototype.check_yearNavigator = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import10.checkBinding(throwOnChange, this._expr_17, currValue))) {
            this._changed = true;
            this.context.yearNavigator = currValue;
            this._expr_17 = currValue;
        }
    };
    Wrapper_Calendar.prototype.check_yearRange = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import10.checkBinding(throwOnChange, this._expr_18, currValue))) {
            this._changed = true;
            this.context.yearRange = currValue;
            this._expr_18 = currValue;
        }
    };
    Wrapper_Calendar.prototype.check_showTime = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import10.checkBinding(throwOnChange, this._expr_19, currValue))) {
            this._changed = true;
            this.context.showTime = currValue;
            this._expr_19 = currValue;
        }
    };
    Wrapper_Calendar.prototype.check_hourFormat = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import10.checkBinding(throwOnChange, this._expr_20, currValue))) {
            this._changed = true;
            this.context.hourFormat = currValue;
            this._expr_20 = currValue;
        }
    };
    Wrapper_Calendar.prototype.check_timeOnly = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import10.checkBinding(throwOnChange, this._expr_21, currValue))) {
            this._changed = true;
            this.context.timeOnly = currValue;
            this._expr_21 = currValue;
        }
    };
    Wrapper_Calendar.prototype.check_required = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import10.checkBinding(throwOnChange, this._expr_22, currValue))) {
            this._changed = true;
            this.context.required = currValue;
            this._expr_22 = currValue;
        }
    };
    Wrapper_Calendar.prototype.check_dataType = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import10.checkBinding(throwOnChange, this._expr_23, currValue))) {
            this._changed = true;
            this.context.dataType = currValue;
            this._expr_23 = currValue;
        }
    };
    Wrapper_Calendar.prototype.check_locale = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import10.checkBinding(throwOnChange, this._expr_24, currValue))) {
            this._changed = true;
            this.context.locale = currValue;
            this._expr_24 = currValue;
        }
    };
    Wrapper_Calendar.prototype.check_tabindex = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import10.checkBinding(throwOnChange, this._expr_25, currValue))) {
            this._changed = true;
            this.context.tabindex = currValue;
            this._expr_25 = currValue;
        }
    };
    Wrapper_Calendar.prototype.check_minDate = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import10.checkBinding(throwOnChange, this._expr_26, currValue))) {
            this._changed = true;
            this.context.minDate = currValue;
            this._expr_26 = currValue;
        }
    };
    Wrapper_Calendar.prototype.check_maxDate = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import10.checkBinding(throwOnChange, this._expr_27, currValue))) {
            this._changed = true;
            this.context.maxDate = currValue;
            this._expr_27 = currValue;
        }
    };
    Wrapper_Calendar.prototype.ngDoCheck = function (view, el, throwOnChange) {
        var changed = this._changed;
        this._changed = false;
        if (!throwOnChange) {
            if ((view.numberOfChecks === 0)) {
                this.context.ngOnInit();
            }
        }
        return changed;
    };
    Wrapper_Calendar.prototype.checkHost = function (view, componentView, el, throwOnChange) {
        var currVal_28 = this.context.filled;
        if (import10.checkBinding(throwOnChange, this._expr_28, currVal_28)) {
            view.renderer.setElementClass(el, 'ui-inputwrapper-filled', currVal_28);
            this._expr_28 = currVal_28;
        }
        var currVal_29 = this.context.focus;
        if (import10.checkBinding(throwOnChange, this._expr_29, currVal_29)) {
            view.renderer.setElementClass(el, 'ui-inputwrapper-focus', currVal_29);
            this._expr_29 = currVal_29;
        }
    };
    Wrapper_Calendar.prototype.handleEvent = function (eventName, $event) {
        var result = true;
        return result;
    };
    Wrapper_Calendar.prototype.subscribe = function (view, _eventHandler, emit0, emit1) {
        this._eventHandler = _eventHandler;
        if (emit0) {
            (this.subscription0 = this.context.onBlur.subscribe(_eventHandler.bind(view, 'onBlur')));
        }
        if (emit1) {
            (this.subscription1 = this.context.onSelect.subscribe(_eventHandler.bind(view, 'onSelect')));
        }
    };
    return Wrapper_Calendar;
}());
var renderType_Calendar_Host = import10.createRenderComponentType('', 0, import12.ViewEncapsulation.None, [], {});
var View_Calendar_Host0 = (function (_super) {
    __extends(View_Calendar_Host0, _super);
    function View_Calendar_Host0(viewUtils, parentView, parentIndex, parentElement) {
        _super.call(this, View_Calendar_Host0, renderType_Calendar_Host, import14.ViewType.HOST, viewUtils, parentView, parentIndex, parentElement, import15.ChangeDetectorStatus.CheckAlways);
    }
    Object.defineProperty(View_Calendar_Host0.prototype, "_NG_VALUE_ACCESSOR_0_5", {
        get: function () {
            if ((this.__NG_VALUE_ACCESSOR_0_5 == null)) {
                (this.__NG_VALUE_ACCESSOR_0_5 = [this._Calendar_0_4.context]);
            }
            return this.__NG_VALUE_ACCESSOR_0_5;
        },
        enumerable: true,
        configurable: true
    });
    View_Calendar_Host0.prototype.createInternal = function (rootSelector) {
        this._el_0 = import10.selectOrCreateRenderHostElement(this.renderer, 'p-calendar', import10.EMPTY_INLINE_ARRAY, rootSelector, null);
        this.compView_0 = new View_Calendar0(this.viewUtils, this, 0, this._el_0);
        this._DomHandler_0_3 = new import13.DomHandler();
        this._Calendar_0_4 = new Wrapper_Calendar(new import17.ElementRef(this._el_0), this._DomHandler_0_3, this.renderer);
        this.compView_0.create(this._Calendar_0_4.context);
        this.init(this._el_0, (this.renderer.directRenderer ? null : [this._el_0]), null);
        return new import16.ComponentRef_(0, this, this._el_0, this._Calendar_0_4.context);
    };
    View_Calendar_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import13.DomHandler) && (0 === requestNodeIndex))) {
            return this._DomHandler_0_3;
        }
        if (((token === import1.Calendar) && (0 === requestNodeIndex))) {
            return this._Calendar_0_4.context;
        }
        if (((token === import18.NG_VALUE_ACCESSOR) && (0 === requestNodeIndex))) {
            return this._NG_VALUE_ACCESSOR_0_5;
        }
        return notFoundResult;
    };
    View_Calendar_Host0.prototype.detectChangesInternal = function (throwOnChange) {
        this._Calendar_0_4.ngDoCheck(this, this._el_0, throwOnChange);
        this._Calendar_0_4.checkHost(this, this.compView_0, this._el_0, throwOnChange);
        this.compView_0.internalDetectChanges(throwOnChange);
        if (!throwOnChange) {
            if ((this.numberOfChecks === 0)) {
                this._Calendar_0_4.context.ngAfterViewInit();
            }
        }
    };
    View_Calendar_Host0.prototype.destroyInternal = function () {
        this.compView_0.destroy();
        this._Calendar_0_4.ngOnDestroy();
    };
    View_Calendar_Host0.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    return View_Calendar_Host0;
}(import9.AppView));
export var CalendarNgFactory = new import16.ComponentFactory('p-calendar', View_Calendar_Host0, import1.Calendar);
var styles_Calendar = [];
var Calendar_overlayState_states = {
    'hidden': { 'opacity': '0' },
    'visible': { 'opacity': '1' },
    '*': {}
};
function Calendar_overlayState_factory(view, element, currentState, nextState) {
    var previousPlayers = view.animationContext.getAnimationPlayers(element, ((nextState == 'void') ? null : 'overlayState'));
    var collectedStyles = {};
    var player = null;
    var totalTime = 0;
    var defaultStateStyles = Calendar_overlayState_states['*'];
    var startStateStyles = Calendar_overlayState_states[currentState];
    if ((startStateStyles == null)) {
        (startStateStyles = defaultStateStyles);
    }
    var endStateStyles = Calendar_overlayState_states[nextState];
    if ((endStateStyles == null)) {
        (endStateStyles = defaultStateStyles);
    }
    if (((player == null) && ((currentState == 'visible') && (nextState == 'hidden')))) {
        player = new import20.AnimationSequencePlayer([view.renderer.animate(element, new import21.AnimationStyles(import22.collectAndResolveStyles(collectedStyles, [startStateStyles])), import22.balanceAnimationKeyframes(collectedStyles, endStateStyles, [
                new import23.AnimationKeyframe(0, new import21.AnimationStyles(import22.collectAndResolveStyles(collectedStyles, [{}]))),
                new import23.AnimationKeyframe(1, new import21.AnimationStyles(import22.collectAndResolveStyles(collectedStyles, [{}])))
            ]), 400, 0, 'ease-in', previousPlayers)]);
        totalTime = 400;
    }
    if (((player == null) && ((currentState == 'hidden') && (nextState == 'visible')))) {
        player = new import20.AnimationSequencePlayer([view.renderer.animate(element, new import21.AnimationStyles(import22.collectAndResolveStyles(collectedStyles, [startStateStyles])), import22.balanceAnimationKeyframes(collectedStyles, endStateStyles, [
                new import23.AnimationKeyframe(0, new import21.AnimationStyles(import22.collectAndResolveStyles(collectedStyles, [{}]))),
                new import23.AnimationKeyframe(1, new import21.AnimationStyles(import22.collectAndResolveStyles(collectedStyles, [{}])))
            ]), 400, 0, 'ease-out', previousPlayers)]);
        totalTime = 400;
    }
    if ((player == null)) {
        (player = new import24.NoOpAnimationPlayer());
    }
    player.onDone(function () {
        player.destroy();
        import22.renderStyles(element, view.renderer, import22.prepareFinalAnimationStyles(startStateStyles, endStateStyles));
    });
    new import20.AnimationSequencePlayer(previousPlayers).destroy();
    import22.renderStyles(element, view.renderer, import22.clearStyles(startStateStyles));
    view.animationContext.queueAnimation(element, 'overlayState', player);
    return new import19.AnimationTransition(player, currentState, nextState, totalTime);
}
var View_Calendar2 = (function (_super) {
    __extends(View_Calendar2, _super);
    function View_Calendar2(viewUtils, parentView, parentIndex, parentElement, declaredViewContainer) {
        _super.call(this, View_Calendar2, renderType_Calendar, import14.ViewType.EMBEDDED, viewUtils, parentView, parentIndex, parentElement, import15.ChangeDetectorStatus.CheckAlways, declaredViewContainer);
        this._expr_4 = import8.UNINITIALIZED;
        this._map_5 = import10.pureProxy2(function (p0, p1) {
            return {
                'ui-datepicker-trigger': p0,
                'ui-state-disabled': p1
            };
        });
    }
    View_Calendar2.prototype.createInternal = function (rootSelector) {
        this._el_0 = import10.createRenderElement(this.renderer, null, 'button', new import10.InlineArray4(4, 'pButton', '', 'type', 'button'), null);
        this._NgClass_0_3 = new import25.Wrapper_NgClass(this.parentView.parentView.parentView.injectorGet(import28.IterableDiffers, this.parentView.parentView.parentIndex), this.parentView.parentView.parentView.injectorGet(import29.KeyValueDiffers, this.parentView.parentView.parentIndex), new import17.ElementRef(this._el_0), this.renderer);
        this._DomHandler_0_4 = new import13.DomHandler();
        this._Button_0_5 = new import26.Wrapper_Button(new import17.ElementRef(this._el_0), this._DomHandler_0_4);
        var disposable_0 = import10.subscribeToRenderElement(this, this._el_0, new import10.InlineArray16(14, 'click', null, 'mouseenter', null, 'mouseleave', null, 'mousedown', null, 'mouseup', null, 'focus', null, 'blur', null), this.eventHandler(this.handleEvent_0));
        this.init(this._el_0, (this.renderer.directRenderer ? null : [this._el_0]), [disposable_0]);
        return null;
    };
    View_Calendar2.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import30.NgClass) && (0 === requestNodeIndex))) {
            return this._NgClass_0_3.context;
        }
        if (((token === import13.DomHandler) && (0 === requestNodeIndex))) {
            return this._DomHandler_0_4;
        }
        if (((token === import3.Button) && (0 === requestNodeIndex))) {
            return this._Button_0_5.context;
        }
        return notFoundResult;
    };
    View_Calendar2.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_0_0_0 = this._map_5(true, this.parentView.parentView.context.disabled);
        this._NgClass_0_3.check_ngClass(currVal_0_0_0, throwOnChange, false);
        this._NgClass_0_3.ngDoCheck(this, this._el_0, throwOnChange);
        var currVal_0_1_0 = this.parentView.parentView.context.icon;
        this._Button_0_5.check_icon(currVal_0_1_0, throwOnChange, false);
        this._Button_0_5.ngDoCheck(this, this._el_0, throwOnChange);
        var currVal_4 = this.parentView.parentView.context.disabled;
        if (import10.checkBinding(throwOnChange, this._expr_4, currVal_4)) {
            this.renderer.setElementProperty(this._el_0, 'disabled', currVal_4);
            this._expr_4 = currVal_4;
        }
        this._Button_0_5.checkHost(this, this, this._el_0, throwOnChange);
        if (!throwOnChange) {
            if ((this.numberOfChecks === 0)) {
                this._Button_0_5.context.ngAfterViewInit();
            }
        }
    };
    View_Calendar2.prototype.destroyInternal = function () {
        this._Button_0_5.ngOnDestroy();
    };
    View_Calendar2.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    View_Calendar2.prototype.handleEvent_0 = function (eventName, $event) {
        this.markPathToRootAsCheckOnce();
        var result = true;
        result = (this._Button_0_5.handleEvent(eventName, $event) && result);
        if ((eventName == 'click')) {
            var pd_sub_0 = (this.parentView.parentView.context.onButtonClick($event, this.parentView._el_1) !== false);
            result = (pd_sub_0 && result);
        }
        return result;
    };
    return View_Calendar2;
}(import9.AppView));
var View_Calendar1 = (function (_super) {
    __extends(View_Calendar1, _super);
    function View_Calendar1(viewUtils, parentView, parentIndex, parentElement, declaredViewContainer) {
        _super.call(this, View_Calendar1, renderType_Calendar, import14.ViewType.EMBEDDED, viewUtils, parentView, parentIndex, parentElement, import15.ChangeDetectorStatus.CheckAlways, declaredViewContainer);
        this._expr_9 = import8.UNINITIALIZED;
        this._expr_10 = import8.UNINITIALIZED;
        this._expr_11 = import8.UNINITIALIZED;
        this._expr_12 = import8.UNINITIALIZED;
        this._expr_13 = import8.UNINITIALIZED;
        this._expr_14 = import8.UNINITIALIZED;
        this._expr_15 = import8.UNINITIALIZED;
    }
    View_Calendar1.prototype.createInternal = function (rootSelector) {
        this._text_0 = this.renderer.createText(null, '\n                ', null);
        this._el_1 = import10.createRenderElement(this.renderer, null, 'input', new import10.InlineArray4(4, 'pInputText', '', 'type', 'text'), null);
        this._NgStyle_1_3 = new import31.Wrapper_NgStyle(this.parentView.parentView.injectorGet(import29.KeyValueDiffers, this.parentView.parentIndex), new import17.ElementRef(this._el_1), this.renderer);
        this._InputText_1_4 = new import32.Wrapper_InputText(new import17.ElementRef(this._el_1));
        this._anchor_2 = this.renderer.createTemplateAnchor(null, null);
        this._vc_2 = new import27.ViewContainer(2, null, this, this._anchor_2);
        this._TemplateRef_2_5 = new import34.TemplateRef_(this, 2, this._anchor_2);
        this._NgIf_2_6 = new import33.Wrapper_NgIf(this._vc_2.vcRef, this._TemplateRef_2_5);
        this._text_3 = this.renderer.createText(null, '\n            ', null);
        var disposable_0 = import10.subscribeToRenderElement(this, this._el_1, new import10.InlineArray16(14, 'focus', null, 'keydown', null, 'click', null, 'blur', null, 'input', null, 'mouseover', null, 'mouseout', null), this.eventHandler(this.handleEvent_1));
        this.init(this._text_3, (this.renderer.directRenderer ? null : [
            this._text_0,
            this._el_1,
            this._anchor_2,
            this._text_3
        ]), [disposable_0]);
        return null;
    };
    View_Calendar1.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import35.NgStyle) && (1 === requestNodeIndex))) {
            return this._NgStyle_1_3.context;
        }
        if (((token === import4.InputText) && (1 === requestNodeIndex))) {
            return this._InputText_1_4.context;
        }
        if (((token === import34.TemplateRef) && (2 === requestNodeIndex))) {
            return this._TemplateRef_2_5;
        }
        if (((token === import36.NgIf) && (2 === requestNodeIndex))) {
            return this._NgIf_2_6.context;
        }
        return notFoundResult;
    };
    View_Calendar1.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_1_0_0 = this.parentView.context.inputStyle;
        this._NgStyle_1_3.check_ngStyle(currVal_1_0_0, throwOnChange, false);
        this._NgStyle_1_3.ngDoCheck(this, this._el_1, throwOnChange);
        this._InputText_1_4.ngDoCheck(this, this._el_1, throwOnChange);
        var currVal_2_0_0 = this.parentView.context.showIcon;
        this._NgIf_2_6.check_ngIf(currVal_2_0_0, throwOnChange, false);
        this._NgIf_2_6.ngDoCheck(this, this._anchor_2, throwOnChange);
        this._vc_2.detectChangesInNestedViews(throwOnChange);
        var currVal_9 = this.parentView.context.required;
        if (import10.checkBinding(throwOnChange, this._expr_9, currVal_9)) {
            this.renderer.setElementAttribute(this._el_1, 'required', ((currVal_9 == null) ? null : currVal_9.toString()));
            this._expr_9 = currVal_9;
        }
        var currVal_10 = this.parentView.context.inputFieldValue;
        if (import10.checkBinding(throwOnChange, this._expr_10, currVal_10)) {
            this.renderer.setElementProperty(this._el_1, 'value', currVal_10);
            this._expr_10 = currVal_10;
        }
        var currVal_11 = this.parentView.context.readonlyInput;
        if (import10.checkBinding(throwOnChange, this._expr_11, currVal_11)) {
            this.renderer.setElementProperty(this._el_1, 'readOnly', currVal_11);
            this._expr_11 = currVal_11;
        }
        var currVal_12 = this.parentView.context.inputStyleClass;
        if (import10.checkBinding(throwOnChange, this._expr_12, currVal_12)) {
            this.renderer.setElementProperty(this._el_1, 'className', currVal_12);
            this._expr_12 = currVal_12;
        }
        var currVal_13 = (this.parentView.context.placeholder || '');
        if (import10.checkBinding(throwOnChange, this._expr_13, currVal_13)) {
            this.renderer.setElementProperty(this._el_1, 'placeholder', currVal_13);
            this._expr_13 = currVal_13;
        }
        var currVal_14 = this.parentView.context.disabled;
        if (import10.checkBinding(throwOnChange, this._expr_14, currVal_14)) {
            this.renderer.setElementProperty(this._el_1, 'disabled', currVal_14);
            this._expr_14 = currVal_14;
        }
        var currVal_15 = this.parentView.context.tabindex;
        if (import10.checkBinding(throwOnChange, this._expr_15, currVal_15)) {
            this.renderer.setElementAttribute(this._el_1, 'tabindex', ((currVal_15 == null) ? null : currVal_15.toString()));
            this._expr_15 = currVal_15;
        }
        this._InputText_1_4.checkHost(this, this, this._el_1, throwOnChange);
    };
    View_Calendar1.prototype.destroyInternal = function () {
        this._vc_2.destroyNestedViews();
    };
    View_Calendar1.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._text_0, ctx);
        cb(this._el_1, ctx);
        cb(this._vc_2.nativeElement, ctx);
        this._vc_2.visitNestedViewRootNodes(cb, ctx);
        cb(this._text_3, ctx);
    };
    View_Calendar1.prototype.createEmbeddedViewInternal = function (nodeIndex) {
        if ((nodeIndex == 2)) {
            return new View_Calendar2(this.viewUtils, this, 2, this._anchor_2, this._vc_2);
        }
        return null;
    };
    View_Calendar1.prototype.handleEvent_1 = function (eventName, $event) {
        this.markPathToRootAsCheckOnce();
        var result = true;
        result = (this._InputText_1_4.handleEvent(eventName, $event) && result);
        if ((eventName == 'focus')) {
            var pd_sub_0 = (this.parentView.context.onInputFocus($event) !== false);
            result = (pd_sub_0 && result);
        }
        if ((eventName == 'keydown')) {
            var pd_sub_1 = (this.parentView.context.onInputKeydown($event) !== false);
            result = (pd_sub_1 && result);
        }
        if ((eventName == 'click')) {
            var pd_sub_2 = ((this.parentView.context.closeOverlay = false) !== false);
            result = (pd_sub_2 && result);
        }
        if ((eventName == 'blur')) {
            var pd_sub_3 = (this.parentView.context.onInputBlur($event) !== false);
            result = (pd_sub_3 && result);
        }
        if ((eventName == 'input')) {
            var pd_sub_4 = (this.parentView.context.onInput($event) !== false);
            result = (pd_sub_4 && result);
        }
        return result;
    };
    return View_Calendar1;
}(import9.AppView));
var View_Calendar4 = (function (_super) {
    __extends(View_Calendar4, _super);
    function View_Calendar4(viewUtils, parentView, parentIndex, parentElement, declaredViewContainer) {
        _super.call(this, View_Calendar4, renderType_Calendar, import14.ViewType.EMBEDDED, viewUtils, parentView, parentIndex, parentElement, import15.ChangeDetectorStatus.CheckAlways, declaredViewContainer);
        this._expr_2 = import8.UNINITIALIZED;
    }
    View_Calendar4.prototype.createInternal = function (rootSelector) {
        this._el_0 = import10.createRenderElement(this.renderer, null, 'span', new import10.InlineArray2(2, 'class', 'ui-datepicker-month'), null);
        this._text_1 = this.renderer.createText(this._el_0, '', null);
        this.init(this._el_0, (this.renderer.directRenderer ? null : [
            this._el_0,
            this._text_1
        ]), null);
        return null;
    };
    View_Calendar4.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_2 = import10.inlineInterpolate(1, '', this.parentView.parentView.context.currentMonthText, '');
        if (import10.checkBinding(throwOnChange, this._expr_2, currVal_2)) {
            this.renderer.setText(this._text_1, currVal_2);
            this._expr_2 = currVal_2;
        }
    };
    View_Calendar4.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    return View_Calendar4;
}(import9.AppView));
var View_Calendar6 = (function (_super) {
    __extends(View_Calendar6, _super);
    function View_Calendar6(viewUtils, parentView, parentIndex, parentElement, declaredViewContainer) {
        _super.call(this, View_Calendar6, renderType_Calendar, import14.ViewType.EMBEDDED, viewUtils, parentView, parentIndex, parentElement, import15.ChangeDetectorStatus.CheckAlways, declaredViewContainer);
        this._expr_2 = import8.UNINITIALIZED;
        this._expr_3 = import8.UNINITIALIZED;
        this._expr_4 = import8.UNINITIALIZED;
    }
    View_Calendar6.prototype.createInternal = function (rootSelector) {
        this._el_0 = import10.createRenderElement(this.renderer, null, 'option', import10.EMPTY_INLINE_ARRAY, null);
        this._text_1 = this.renderer.createText(this._el_0, '', null);
        this.init(this._el_0, (this.renderer.directRenderer ? null : [
            this._el_0,
            this._text_1
        ]), null);
        return null;
    };
    View_Calendar6.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_2 = this.context.index;
        if (import10.checkBinding(throwOnChange, this._expr_2, currVal_2)) {
            this.renderer.setElementProperty(this._el_0, 'value', currVal_2);
            this._expr_2 = currVal_2;
        }
        var currVal_3 = (this.context.index == this.parentView.parentView.parentView.context.currentMonth);
        if (import10.checkBinding(throwOnChange, this._expr_3, currVal_3)) {
            this.renderer.setElementProperty(this._el_0, 'selected', currVal_3);
            this._expr_3 = currVal_3;
        }
        var currVal_4 = import10.inlineInterpolate(1, '', this.context.$implicit, '');
        if (import10.checkBinding(throwOnChange, this._expr_4, currVal_4)) {
            this.renderer.setText(this._text_1, currVal_4);
            this._expr_4 = currVal_4;
        }
    };
    View_Calendar6.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    return View_Calendar6;
}(import9.AppView));
var View_Calendar5 = (function (_super) {
    __extends(View_Calendar5, _super);
    function View_Calendar5(viewUtils, parentView, parentIndex, parentElement, declaredViewContainer) {
        _super.call(this, View_Calendar5, renderType_Calendar, import14.ViewType.EMBEDDED, viewUtils, parentView, parentIndex, parentElement, import15.ChangeDetectorStatus.CheckAlways, declaredViewContainer);
    }
    View_Calendar5.prototype.createInternal = function (rootSelector) {
        this._el_0 = import10.createRenderElement(this.renderer, null, 'select', new import10.InlineArray2(2, 'class', 'ui-datepicker-month'), null);
        this._text_1 = this.renderer.createText(this._el_0, '\n                            ', null);
        this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0, null);
        this._vc_2 = new import27.ViewContainer(2, 0, this, this._anchor_2);
        this._TemplateRef_2_5 = new import34.TemplateRef_(this, 2, this._anchor_2);
        this._NgFor_2_6 = new import37.Wrapper_NgFor(this._vc_2.vcRef, this._TemplateRef_2_5, this.parentView.parentView.parentView.injectorGet(import28.IterableDiffers, this.parentView.parentView.parentIndex), this.parentView.parentView.ref);
        this._text_3 = this.renderer.createText(this._el_0, '\n                        ', null);
        var disposable_0 = import10.subscribeToRenderElement(this, this._el_0, new import10.InlineArray2(2, 'change', null), this.eventHandler(this.handleEvent_0));
        this.init(this._el_0, (this.renderer.directRenderer ? null : [
            this._el_0,
            this._text_1,
            this._anchor_2,
            this._text_3
        ]), [disposable_0]);
        return null;
    };
    View_Calendar5.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import34.TemplateRef) && (2 === requestNodeIndex))) {
            return this._TemplateRef_2_5;
        }
        if (((token === import38.NgFor) && (2 === requestNodeIndex))) {
            return this._NgFor_2_6.context;
        }
        return notFoundResult;
    };
    View_Calendar5.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_2_0_0 = this.parentView.parentView.context.locale.monthNames;
        this._NgFor_2_6.check_ngForOf(currVal_2_0_0, throwOnChange, false);
        this._NgFor_2_6.ngDoCheck(this, this._anchor_2, throwOnChange);
        this._vc_2.detectChangesInNestedViews(throwOnChange);
    };
    View_Calendar5.prototype.destroyInternal = function () {
        this._vc_2.destroyNestedViews();
    };
    View_Calendar5.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    View_Calendar5.prototype.createEmbeddedViewInternal = function (nodeIndex) {
        if ((nodeIndex == 2)) {
            return new View_Calendar6(this.viewUtils, this, 2, this._anchor_2, this._vc_2);
        }
        return null;
    };
    View_Calendar5.prototype.handleEvent_0 = function (eventName, $event) {
        this.markPathToRootAsCheckOnce();
        var result = true;
        if ((eventName == 'change')) {
            var pd_sub_0 = (this.parentView.parentView.context.onMonthDropdownChange($event.target.value) !== false);
            result = (pd_sub_0 && result);
        }
        return result;
    };
    return View_Calendar5;
}(import9.AppView));
var View_Calendar8 = (function (_super) {
    __extends(View_Calendar8, _super);
    function View_Calendar8(viewUtils, parentView, parentIndex, parentElement, declaredViewContainer) {
        _super.call(this, View_Calendar8, renderType_Calendar, import14.ViewType.EMBEDDED, viewUtils, parentView, parentIndex, parentElement, import15.ChangeDetectorStatus.CheckAlways, declaredViewContainer);
        this._expr_2 = import8.UNINITIALIZED;
        this._expr_3 = import8.UNINITIALIZED;
        this._expr_4 = import8.UNINITIALIZED;
    }
    View_Calendar8.prototype.createInternal = function (rootSelector) {
        this._el_0 = import10.createRenderElement(this.renderer, null, 'option', import10.EMPTY_INLINE_ARRAY, null);
        this._text_1 = this.renderer.createText(this._el_0, '', null);
        this.init(this._el_0, (this.renderer.directRenderer ? null : [
            this._el_0,
            this._text_1
        ]), null);
        return null;
    };
    View_Calendar8.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_2 = this.context.$implicit;
        if (import10.checkBinding(throwOnChange, this._expr_2, currVal_2)) {
            this.renderer.setElementProperty(this._el_0, 'value', currVal_2);
            this._expr_2 = currVal_2;
        }
        var currVal_3 = (this.context.$implicit == this.parentView.parentView.parentView.context.currentYear);
        if (import10.checkBinding(throwOnChange, this._expr_3, currVal_3)) {
            this.renderer.setElementProperty(this._el_0, 'selected', currVal_3);
            this._expr_3 = currVal_3;
        }
        var currVal_4 = import10.inlineInterpolate(1, '', this.context.$implicit, '');
        if (import10.checkBinding(throwOnChange, this._expr_4, currVal_4)) {
            this.renderer.setText(this._text_1, currVal_4);
            this._expr_4 = currVal_4;
        }
    };
    View_Calendar8.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    return View_Calendar8;
}(import9.AppView));
var View_Calendar7 = (function (_super) {
    __extends(View_Calendar7, _super);
    function View_Calendar7(viewUtils, parentView, parentIndex, parentElement, declaredViewContainer) {
        _super.call(this, View_Calendar7, renderType_Calendar, import14.ViewType.EMBEDDED, viewUtils, parentView, parentIndex, parentElement, import15.ChangeDetectorStatus.CheckAlways, declaredViewContainer);
    }
    View_Calendar7.prototype.createInternal = function (rootSelector) {
        this._el_0 = import10.createRenderElement(this.renderer, null, 'select', new import10.InlineArray2(2, 'class', 'ui-datepicker-year'), null);
        this._text_1 = this.renderer.createText(this._el_0, '\n                            ', null);
        this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0, null);
        this._vc_2 = new import27.ViewContainer(2, 0, this, this._anchor_2);
        this._TemplateRef_2_5 = new import34.TemplateRef_(this, 2, this._anchor_2);
        this._NgFor_2_6 = new import37.Wrapper_NgFor(this._vc_2.vcRef, this._TemplateRef_2_5, this.parentView.parentView.parentView.injectorGet(import28.IterableDiffers, this.parentView.parentView.parentIndex), this.parentView.parentView.ref);
        this._text_3 = this.renderer.createText(this._el_0, '\n                        ', null);
        var disposable_0 = import10.subscribeToRenderElement(this, this._el_0, new import10.InlineArray2(2, 'change', null), this.eventHandler(this.handleEvent_0));
        this.init(this._el_0, (this.renderer.directRenderer ? null : [
            this._el_0,
            this._text_1,
            this._anchor_2,
            this._text_3
        ]), [disposable_0]);
        return null;
    };
    View_Calendar7.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import34.TemplateRef) && (2 === requestNodeIndex))) {
            return this._TemplateRef_2_5;
        }
        if (((token === import38.NgFor) && (2 === requestNodeIndex))) {
            return this._NgFor_2_6.context;
        }
        return notFoundResult;
    };
    View_Calendar7.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_2_0_0 = this.parentView.parentView.context.yearOptions;
        this._NgFor_2_6.check_ngForOf(currVal_2_0_0, throwOnChange, false);
        this._NgFor_2_6.ngDoCheck(this, this._anchor_2, throwOnChange);
        this._vc_2.detectChangesInNestedViews(throwOnChange);
    };
    View_Calendar7.prototype.destroyInternal = function () {
        this._vc_2.destroyNestedViews();
    };
    View_Calendar7.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    View_Calendar7.prototype.createEmbeddedViewInternal = function (nodeIndex) {
        if ((nodeIndex == 2)) {
            return new View_Calendar8(this.viewUtils, this, 2, this._anchor_2, this._vc_2);
        }
        return null;
    };
    View_Calendar7.prototype.handleEvent_0 = function (eventName, $event) {
        this.markPathToRootAsCheckOnce();
        var result = true;
        if ((eventName == 'change')) {
            var pd_sub_0 = (this.parentView.parentView.context.onYearDropdownChange($event.target.value) !== false);
            result = (pd_sub_0 && result);
        }
        return result;
    };
    return View_Calendar7;
}(import9.AppView));
var View_Calendar9 = (function (_super) {
    __extends(View_Calendar9, _super);
    function View_Calendar9(viewUtils, parentView, parentIndex, parentElement, declaredViewContainer) {
        _super.call(this, View_Calendar9, renderType_Calendar, import14.ViewType.EMBEDDED, viewUtils, parentView, parentIndex, parentElement, import15.ChangeDetectorStatus.CheckAlways, declaredViewContainer);
        this._expr_2 = import8.UNINITIALIZED;
    }
    View_Calendar9.prototype.createInternal = function (rootSelector) {
        this._el_0 = import10.createRenderElement(this.renderer, null, 'span', new import10.InlineArray2(2, 'class', 'ui-datepicker-year'), null);
        this._text_1 = this.renderer.createText(this._el_0, '', null);
        this.init(this._el_0, (this.renderer.directRenderer ? null : [
            this._el_0,
            this._text_1
        ]), null);
        return null;
    };
    View_Calendar9.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_2 = import10.inlineInterpolate(1, '', this.parentView.parentView.context.currentYear, '');
        if (import10.checkBinding(throwOnChange, this._expr_2, currVal_2)) {
            this.renderer.setText(this._text_1, currVal_2);
            this._expr_2 = currVal_2;
        }
    };
    View_Calendar9.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    return View_Calendar9;
}(import9.AppView));
var View_Calendar3 = (function (_super) {
    __extends(View_Calendar3, _super);
    function View_Calendar3(viewUtils, parentView, parentIndex, parentElement, declaredViewContainer) {
        _super.call(this, View_Calendar3, renderType_Calendar, import14.ViewType.EMBEDDED, viewUtils, parentView, parentIndex, parentElement, import15.ChangeDetectorStatus.CheckAlways, declaredViewContainer);
        this._map_37 = import10.pureProxy1(function (p0) {
            return { 'ui-state-hover ui-datepicker-prev-hover': p0 };
        });
        this._map_38 = import10.pureProxy1(function (p0) {
            return { 'ui-state-hover ui-datepicker-next-hover': p0 };
        });
    }
    View_Calendar3.prototype.createInternal = function (rootSelector) {
        this._el_0 = import10.createRenderElement(this.renderer, null, 'div', new import10.InlineArray2(2, 'class', 'ui-datepicker-header ui-widget-header ui-helper-clearfix ui-corner-all'), null);
        this._text_1 = this.renderer.createText(this._el_0, '\n                    ', null);
        this._el_2 = import10.createRenderElement(this.renderer, this._el_0, 'a', new import10.InlineArray4(4, 'class', 'ui-datepicker-prev ui-corner-all', 'href', '#'), null);
        this._NgClass_2_3 = new import25.Wrapper_NgClass(this.parentView.parentView.injectorGet(import28.IterableDiffers, this.parentView.parentIndex), this.parentView.parentView.injectorGet(import29.KeyValueDiffers, this.parentView.parentIndex), new import17.ElementRef(this._el_2), this.renderer);
        this._text_3 = this.renderer.createText(this._el_2, '\n                        ', null);
        this._el_4 = import10.createRenderElement(this.renderer, this._el_2, 'span', new import10.InlineArray2(2, 'class', 'fa fa-angle-left'), null);
        this._text_5 = this.renderer.createText(this._el_2, '\n                    ', null);
        this._text_6 = this.renderer.createText(this._el_0, '\n                    ', null);
        this._el_7 = import10.createRenderElement(this.renderer, this._el_0, 'a', new import10.InlineArray4(4, 'class', 'ui-datepicker-next ui-corner-all', 'href', '#'), null);
        this._NgClass_7_3 = new import25.Wrapper_NgClass(this.parentView.parentView.injectorGet(import28.IterableDiffers, this.parentView.parentIndex), this.parentView.parentView.injectorGet(import29.KeyValueDiffers, this.parentView.parentIndex), new import17.ElementRef(this._el_7), this.renderer);
        this._text_8 = this.renderer.createText(this._el_7, '\n                        ', null);
        this._el_9 = import10.createRenderElement(this.renderer, this._el_7, 'span', new import10.InlineArray2(2, 'class', 'fa fa-angle-right'), null);
        this._text_10 = this.renderer.createText(this._el_7, '\n                    ', null);
        this._text_11 = this.renderer.createText(this._el_0, '\n                    ', null);
        this._el_12 = import10.createRenderElement(this.renderer, this._el_0, 'div', new import10.InlineArray2(2, 'class', 'ui-datepicker-title'), null);
        this._text_13 = this.renderer.createText(this._el_12, '\n                        ', null);
        this._anchor_14 = this.renderer.createTemplateAnchor(this._el_12, null);
        this._vc_14 = new import27.ViewContainer(14, 12, this, this._anchor_14);
        this._TemplateRef_14_5 = new import34.TemplateRef_(this, 14, this._anchor_14);
        this._NgIf_14_6 = new import33.Wrapper_NgIf(this._vc_14.vcRef, this._TemplateRef_14_5);
        this._text_15 = this.renderer.createText(this._el_12, '\n                        ', null);
        this._anchor_16 = this.renderer.createTemplateAnchor(this._el_12, null);
        this._vc_16 = new import27.ViewContainer(16, 12, this, this._anchor_16);
        this._TemplateRef_16_5 = new import34.TemplateRef_(this, 16, this._anchor_16);
        this._NgIf_16_6 = new import33.Wrapper_NgIf(this._vc_16.vcRef, this._TemplateRef_16_5);
        this._text_17 = this.renderer.createText(this._el_12, '\n                        ', null);
        this._anchor_18 = this.renderer.createTemplateAnchor(this._el_12, null);
        this._vc_18 = new import27.ViewContainer(18, 12, this, this._anchor_18);
        this._TemplateRef_18_5 = new import34.TemplateRef_(this, 18, this._anchor_18);
        this._NgIf_18_6 = new import33.Wrapper_NgIf(this._vc_18.vcRef, this._TemplateRef_18_5);
        this._text_19 = this.renderer.createText(this._el_12, '\n                        ', null);
        this._anchor_20 = this.renderer.createTemplateAnchor(this._el_12, null);
        this._vc_20 = new import27.ViewContainer(20, 12, this, this._anchor_20);
        this._TemplateRef_20_5 = new import34.TemplateRef_(this, 20, this._anchor_20);
        this._NgIf_20_6 = new import33.Wrapper_NgIf(this._vc_20.vcRef, this._TemplateRef_20_5);
        this._text_21 = this.renderer.createText(this._el_12, '\n                    ', null);
        this._text_22 = this.renderer.createText(this._el_0, '\n                ', null);
        var disposable_0 = import10.subscribeToRenderElement(this, this._el_2, new import10.InlineArray8(6, 'click', null, 'mouseenter', null, 'mouseleave', null), this.eventHandler(this.handleEvent_2));
        var disposable_1 = import10.subscribeToRenderElement(this, this._el_7, new import10.InlineArray8(6, 'click', null, 'mouseenter', null, 'mouseleave', null), this.eventHandler(this.handleEvent_7));
        this.init(this._el_0, (this.renderer.directRenderer ? null : [
            this._el_0,
            this._text_1,
            this._el_2,
            this._text_3,
            this._el_4,
            this._text_5,
            this._text_6,
            this._el_7,
            this._text_8,
            this._el_9,
            this._text_10,
            this._text_11,
            this._el_12,
            this._text_13,
            this._anchor_14,
            this._text_15,
            this._anchor_16,
            this._text_17,
            this._anchor_18,
            this._text_19,
            this._anchor_20,
            this._text_21,
            this._text_22
        ]), [
            disposable_0,
            disposable_1
        ]);
        return null;
    };
    View_Calendar3.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import30.NgClass) && ((2 <= requestNodeIndex) && (requestNodeIndex <= 5)))) {
            return this._NgClass_2_3.context;
        }
        if (((token === import30.NgClass) && ((7 <= requestNodeIndex) && (requestNodeIndex <= 10)))) {
            return this._NgClass_7_3.context;
        }
        if (((token === import34.TemplateRef) && (14 === requestNodeIndex))) {
            return this._TemplateRef_14_5;
        }
        if (((token === import36.NgIf) && (14 === requestNodeIndex))) {
            return this._NgIf_14_6.context;
        }
        if (((token === import34.TemplateRef) && (16 === requestNodeIndex))) {
            return this._TemplateRef_16_5;
        }
        if (((token === import36.NgIf) && (16 === requestNodeIndex))) {
            return this._NgIf_16_6.context;
        }
        if (((token === import34.TemplateRef) && (18 === requestNodeIndex))) {
            return this._TemplateRef_18_5;
        }
        if (((token === import36.NgIf) && (18 === requestNodeIndex))) {
            return this._NgIf_18_6.context;
        }
        if (((token === import34.TemplateRef) && (20 === requestNodeIndex))) {
            return this._TemplateRef_20_5;
        }
        if (((token === import36.NgIf) && (20 === requestNodeIndex))) {
            return this._NgIf_20_6.context;
        }
        return notFoundResult;
    };
    View_Calendar3.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_2_0_0 = 'ui-datepicker-prev ui-corner-all';
        this._NgClass_2_3.check_klass(currVal_2_0_0, throwOnChange, false);
        var currVal_2_0_1 = this._map_37((this.parentView.context.hoverPrev && !this.parentView.context.disabled));
        this._NgClass_2_3.check_ngClass(currVal_2_0_1, throwOnChange, false);
        this._NgClass_2_3.ngDoCheck(this, this._el_2, throwOnChange);
        var currVal_7_0_0 = 'ui-datepicker-next ui-corner-all';
        this._NgClass_7_3.check_klass(currVal_7_0_0, throwOnChange, false);
        var currVal_7_0_1 = this._map_38((this.parentView.context.hoverNext && !this.parentView.context.disabled));
        this._NgClass_7_3.check_ngClass(currVal_7_0_1, throwOnChange, false);
        this._NgClass_7_3.ngDoCheck(this, this._el_7, throwOnChange);
        var currVal_14_0_0 = !this.parentView.context.monthNavigator;
        this._NgIf_14_6.check_ngIf(currVal_14_0_0, throwOnChange, false);
        this._NgIf_14_6.ngDoCheck(this, this._anchor_14, throwOnChange);
        var currVal_16_0_0 = this.parentView.context.monthNavigator;
        this._NgIf_16_6.check_ngIf(currVal_16_0_0, throwOnChange, false);
        this._NgIf_16_6.ngDoCheck(this, this._anchor_16, throwOnChange);
        var currVal_18_0_0 = this.parentView.context.yearNavigator;
        this._NgIf_18_6.check_ngIf(currVal_18_0_0, throwOnChange, false);
        this._NgIf_18_6.ngDoCheck(this, this._anchor_18, throwOnChange);
        var currVal_20_0_0 = !this.parentView.context.yearNavigator;
        this._NgIf_20_6.check_ngIf(currVal_20_0_0, throwOnChange, false);
        this._NgIf_20_6.ngDoCheck(this, this._anchor_20, throwOnChange);
        this._vc_14.detectChangesInNestedViews(throwOnChange);
        this._vc_16.detectChangesInNestedViews(throwOnChange);
        this._vc_18.detectChangesInNestedViews(throwOnChange);
        this._vc_20.detectChangesInNestedViews(throwOnChange);
    };
    View_Calendar3.prototype.destroyInternal = function () {
        this._vc_14.destroyNestedViews();
        this._vc_16.destroyNestedViews();
        this._vc_18.destroyNestedViews();
        this._vc_20.destroyNestedViews();
    };
    View_Calendar3.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    View_Calendar3.prototype.createEmbeddedViewInternal = function (nodeIndex) {
        if ((nodeIndex == 14)) {
            return new View_Calendar4(this.viewUtils, this, 14, this._anchor_14, this._vc_14);
        }
        if ((nodeIndex == 16)) {
            return new View_Calendar5(this.viewUtils, this, 16, this._anchor_16, this._vc_16);
        }
        if ((nodeIndex == 18)) {
            return new View_Calendar7(this.viewUtils, this, 18, this._anchor_18, this._vc_18);
        }
        if ((nodeIndex == 20)) {
            return new View_Calendar9(this.viewUtils, this, 20, this._anchor_20, this._vc_20);
        }
        return null;
    };
    View_Calendar3.prototype.handleEvent_2 = function (eventName, $event) {
        this.markPathToRootAsCheckOnce();
        var result = true;
        if ((eventName == 'click')) {
            var pd_sub_0 = (this.parentView.context.prevMonth($event) !== false);
            result = (pd_sub_0 && result);
        }
        if ((eventName == 'mouseenter')) {
            var pd_sub_1 = ((this.parentView.context.hoverPrev = true) !== false);
            result = (pd_sub_1 && result);
        }
        if ((eventName == 'mouseleave')) {
            var pd_sub_2 = ((this.parentView.context.hoverPrev = false) !== false);
            result = (pd_sub_2 && result);
        }
        return result;
    };
    View_Calendar3.prototype.handleEvent_7 = function (eventName, $event) {
        this.markPathToRootAsCheckOnce();
        var result = true;
        if ((eventName == 'click')) {
            var pd_sub_0 = (this.parentView.context.nextMonth($event) !== false);
            result = (pd_sub_0 && result);
        }
        if ((eventName == 'mouseenter')) {
            var pd_sub_1 = ((this.parentView.context.hoverNext = true) !== false);
            result = (pd_sub_1 && result);
        }
        if ((eventName == 'mouseleave')) {
            var pd_sub_2 = ((this.parentView.context.hoverNext = false) !== false);
            result = (pd_sub_2 && result);
        }
        return result;
    };
    return View_Calendar3;
}(import9.AppView));
var View_Calendar11 = (function (_super) {
    __extends(View_Calendar11, _super);
    function View_Calendar11(viewUtils, parentView, parentIndex, parentElement, declaredViewContainer) {
        _super.call(this, View_Calendar11, renderType_Calendar, import14.ViewType.EMBEDDED, viewUtils, parentView, parentIndex, parentElement, import15.ChangeDetectorStatus.CheckAlways, declaredViewContainer);
        this._expr_5 = import8.UNINITIALIZED;
    }
    View_Calendar11.prototype.createInternal = function (rootSelector) {
        this._el_0 = import10.createRenderElement(this.renderer, null, 'th', new import10.InlineArray2(2, 'scope', 'col'), null);
        this._text_1 = this.renderer.createText(this._el_0, '\n                                ', null);
        this._el_2 = import10.createRenderElement(this.renderer, this._el_0, 'span', import10.EMPTY_INLINE_ARRAY, null);
        this._text_3 = this.renderer.createText(this._el_2, '', null);
        this._text_4 = this.renderer.createText(this._el_0, '\n                            ', null);
        this.init(this._el_0, (this.renderer.directRenderer ? null : [
            this._el_0,
            this._text_1,
            this._el_2,
            this._text_3,
            this._text_4
        ]), null);
        return null;
    };
    View_Calendar11.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_5 = import10.inlineInterpolate(1, '', this.context.$implicit, '');
        if (import10.checkBinding(throwOnChange, this._expr_5, currVal_5)) {
            this.renderer.setText(this._text_3, currVal_5);
            this._expr_5 = currVal_5;
        }
    };
    View_Calendar11.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    return View_Calendar11;
}(import9.AppView));
var View_Calendar14 = (function (_super) {
    __extends(View_Calendar14, _super);
    function View_Calendar14(viewUtils, parentView, parentIndex, parentElement, declaredViewContainer) {
        _super.call(this, View_Calendar14, renderType_Calendar, import14.ViewType.EMBEDDED, viewUtils, parentView, parentIndex, parentElement, import15.ChangeDetectorStatus.CheckAlways, declaredViewContainer);
        this._map_3 = import10.pureProxy4(function (p0, p1, p2, p3) {
            return {
                'ui-state-active': p0,
                'ui-state-hover': p1,
                'ui-state-highlight': p2,
                'ui-state-disabled': p3
            };
        });
        this._expr_4 = import8.UNINITIALIZED;
    }
    View_Calendar14.prototype.createInternal = function (rootSelector) {
        this._el_0 = import10.createRenderElement(this.renderer, null, 'a', new import10.InlineArray4(4, 'class', 'ui-state-default', 'href', '#'), null);
        this._NgClass_0_3 = new import25.Wrapper_NgClass(this.parentView.parentView.parentView.parentView.parentView.injectorGet(import28.IterableDiffers, this.parentView.parentView.parentView.parentView.parentIndex), this.parentView.parentView.parentView.parentView.parentView.injectorGet(import29.KeyValueDiffers, this.parentView.parentView.parentView.parentView.parentIndex), new import17.ElementRef(this._el_0), this.renderer);
        this._text_1 = this.renderer.createText(this._el_0, '', null);
        var disposable_0 = import10.subscribeToRenderElement(this, this._el_0, new import10.InlineArray8(6, 'click', null, 'mouseenter', null, 'mouseleave', null), this.eventHandler(this.handleEvent_0));
        this.init(this._el_0, (this.renderer.directRenderer ? null : [
            this._el_0,
            this._text_1
        ]), [disposable_0]);
        return null;
    };
    View_Calendar14.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import30.NgClass) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 1)))) {
            return this._NgClass_0_3.context;
        }
        return notFoundResult;
    };
    View_Calendar14.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_0_0_0 = 'ui-state-default';
        this._NgClass_0_3.check_klass(currVal_0_0_0, throwOnChange, false);
        var currVal_0_0_1 = this._map_3(this.parentView.parentView.parentView.parentView.context.isSelected(this.parentView.context.$implicit), (((this.parentView.parentView.parentView.parentView.context.hoverCell == this._el_0) && !this.parentView.parentView.parentView.parentView.context.disabled) && this.parentView.context.$implicit.selectable), this.parentView.parentView.parentView.parentView.context.isToday(this.parentView.context.$implicit), !this.parentView.context.$implicit.selectable);
        this._NgClass_0_3.check_ngClass(currVal_0_0_1, throwOnChange, false);
        this._NgClass_0_3.ngDoCheck(this, this._el_0, throwOnChange);
        var currVal_4 = import10.inlineInterpolate(1, '', this.parentView.context.$implicit.day, '');
        if (import10.checkBinding(throwOnChange, this._expr_4, currVal_4)) {
            this.renderer.setText(this._text_1, currVal_4);
            this._expr_4 = currVal_4;
        }
    };
    View_Calendar14.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    View_Calendar14.prototype.handleEvent_0 = function (eventName, $event) {
        this.markPathToRootAsCheckOnce();
        var result = true;
        if ((eventName == 'click')) {
            var pd_sub_0 = (this.parentView.parentView.parentView.parentView.context.onDateSelect($event, this.parentView.context.$implicit) !== false);
            result = (pd_sub_0 && result);
        }
        if ((eventName == 'mouseenter')) {
            var pd_sub_1 = ((this.parentView.parentView.parentView.parentView.context.hoverCell = this._el_0) !== false);
            result = (pd_sub_1 && result);
        }
        if ((eventName == 'mouseleave')) {
            var pd_sub_2 = ((this.parentView.parentView.parentView.parentView.context.hoverCell = null) !== false);
            result = (pd_sub_2 && result);
        }
        return result;
    };
    return View_Calendar14;
}(import9.AppView));
var View_Calendar13 = (function (_super) {
    __extends(View_Calendar13, _super);
    function View_Calendar13(viewUtils, parentView, parentIndex, parentElement, declaredViewContainer) {
        _super.call(this, View_Calendar13, renderType_Calendar, import14.ViewType.EMBEDDED, viewUtils, parentView, parentIndex, parentElement, import15.ChangeDetectorStatus.CheckAlways, declaredViewContainer);
        this._map_8 = import10.pureProxy3(function (p0, p1, p2) {
            return {
                'ui-datepicker-other-month ui-state-disabled': p0,
                'ui-datepicker-current-day': p1,
                'ui-datepicker-today': p2
            };
        });
    }
    View_Calendar13.prototype.createInternal = function (rootSelector) {
        this._el_0 = import10.createRenderElement(this.renderer, null, 'td', import10.EMPTY_INLINE_ARRAY, null);
        this._NgClass_0_3 = new import25.Wrapper_NgClass(this.parentView.parentView.parentView.parentView.injectorGet(import28.IterableDiffers, this.parentView.parentView.parentView.parentIndex), this.parentView.parentView.parentView.parentView.injectorGet(import29.KeyValueDiffers, this.parentView.parentView.parentView.parentIndex), new import17.ElementRef(this._el_0), this.renderer);
        this._text_1 = this.renderer.createText(this._el_0, '\n                                ', null);
        this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0, null);
        this._vc_2 = new import27.ViewContainer(2, 0, this, this._anchor_2);
        this._TemplateRef_2_5 = new import34.TemplateRef_(this, 2, this._anchor_2);
        this._NgIf_2_6 = new import33.Wrapper_NgIf(this._vc_2.vcRef, this._TemplateRef_2_5);
        this._text_3 = this.renderer.createText(this._el_0, '\n                            ', null);
        this.init(this._el_0, (this.renderer.directRenderer ? null : [
            this._el_0,
            this._text_1,
            this._anchor_2,
            this._text_3
        ]), null);
        return null;
    };
    View_Calendar13.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import34.TemplateRef) && (2 === requestNodeIndex))) {
            return this._TemplateRef_2_5;
        }
        if (((token === import36.NgIf) && (2 === requestNodeIndex))) {
            return this._NgIf_2_6.context;
        }
        if (((token === import30.NgClass) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 3)))) {
            return this._NgClass_0_3.context;
        }
        return notFoundResult;
    };
    View_Calendar13.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_0_0_0 = this._map_8(this.context.$implicit.otherMonth, this.parentView.parentView.parentView.context.isSelected(this.context.$implicit), this.parentView.parentView.parentView.context.isToday(this.context.$implicit));
        this._NgClass_0_3.check_ngClass(currVal_0_0_0, throwOnChange, false);
        this._NgClass_0_3.ngDoCheck(this, this._el_0, throwOnChange);
        var currVal_2_0_0 = (this.context.$implicit.otherMonth ? this.parentView.parentView.parentView.context.showOtherMonths : true);
        this._NgIf_2_6.check_ngIf(currVal_2_0_0, throwOnChange, false);
        this._NgIf_2_6.ngDoCheck(this, this._anchor_2, throwOnChange);
        this._vc_2.detectChangesInNestedViews(throwOnChange);
    };
    View_Calendar13.prototype.destroyInternal = function () {
        this._vc_2.destroyNestedViews();
    };
    View_Calendar13.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    View_Calendar13.prototype.createEmbeddedViewInternal = function (nodeIndex) {
        if ((nodeIndex == 2)) {
            return new View_Calendar14(this.viewUtils, this, 2, this._anchor_2, this._vc_2);
        }
        return null;
    };
    return View_Calendar13;
}(import9.AppView));
var View_Calendar12 = (function (_super) {
    __extends(View_Calendar12, _super);
    function View_Calendar12(viewUtils, parentView, parentIndex, parentElement, declaredViewContainer) {
        _super.call(this, View_Calendar12, renderType_Calendar, import14.ViewType.EMBEDDED, viewUtils, parentView, parentIndex, parentElement, import15.ChangeDetectorStatus.CheckAlways, declaredViewContainer);
    }
    View_Calendar12.prototype.createInternal = function (rootSelector) {
        this._el_0 = import10.createRenderElement(this.renderer, null, 'tr', import10.EMPTY_INLINE_ARRAY, null);
        this._text_1 = this.renderer.createText(this._el_0, '\n                            ', null);
        this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0, null);
        this._vc_2 = new import27.ViewContainer(2, 0, this, this._anchor_2);
        this._TemplateRef_2_5 = new import34.TemplateRef_(this, 2, this._anchor_2);
        this._NgFor_2_6 = new import37.Wrapper_NgFor(this._vc_2.vcRef, this._TemplateRef_2_5, this.parentView.parentView.parentView.injectorGet(import28.IterableDiffers, this.parentView.parentView.parentIndex), this.parentView.parentView.ref);
        this._text_3 = this.renderer.createText(this._el_0, '\n                        ', null);
        this.init(this._el_0, (this.renderer.directRenderer ? null : [
            this._el_0,
            this._text_1,
            this._anchor_2,
            this._text_3
        ]), null);
        return null;
    };
    View_Calendar12.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import34.TemplateRef) && (2 === requestNodeIndex))) {
            return this._TemplateRef_2_5;
        }
        if (((token === import38.NgFor) && (2 === requestNodeIndex))) {
            return this._NgFor_2_6.context;
        }
        return notFoundResult;
    };
    View_Calendar12.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_2_0_0 = this.context.$implicit;
        this._NgFor_2_6.check_ngForOf(currVal_2_0_0, throwOnChange, false);
        this._NgFor_2_6.ngDoCheck(this, this._anchor_2, throwOnChange);
        this._vc_2.detectChangesInNestedViews(throwOnChange);
    };
    View_Calendar12.prototype.destroyInternal = function () {
        this._vc_2.destroyNestedViews();
    };
    View_Calendar12.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    View_Calendar12.prototype.createEmbeddedViewInternal = function (nodeIndex) {
        if ((nodeIndex == 2)) {
            return new View_Calendar13(this.viewUtils, this, 2, this._anchor_2, this._vc_2);
        }
        return null;
    };
    return View_Calendar12;
}(import9.AppView));
var View_Calendar10 = (function (_super) {
    __extends(View_Calendar10, _super);
    function View_Calendar10(viewUtils, parentView, parentIndex, parentElement, declaredViewContainer) {
        _super.call(this, View_Calendar10, renderType_Calendar, import14.ViewType.EMBEDDED, viewUtils, parentView, parentIndex, parentElement, import15.ChangeDetectorStatus.CheckAlways, declaredViewContainer);
    }
    View_Calendar10.prototype.createInternal = function (rootSelector) {
        this._el_0 = import10.createRenderElement(this.renderer, null, 'table', new import10.InlineArray2(2, 'class', 'ui-datepicker-calendar'), null);
        this._text_1 = this.renderer.createText(this._el_0, '\n                    ', null);
        this._el_2 = import10.createRenderElement(this.renderer, this._el_0, 'thead', import10.EMPTY_INLINE_ARRAY, null);
        this._text_3 = this.renderer.createText(this._el_2, '\n                        ', null);
        this._el_4 = import10.createRenderElement(this.renderer, this._el_2, 'tr', import10.EMPTY_INLINE_ARRAY, null);
        this._text_5 = this.renderer.createText(this._el_4, '\n                            ', null);
        this._anchor_6 = this.renderer.createTemplateAnchor(this._el_4, null);
        this._vc_6 = new import27.ViewContainer(6, 4, this, this._anchor_6);
        this._TemplateRef_6_5 = new import34.TemplateRef_(this, 6, this._anchor_6);
        this._NgFor_6_6 = new import37.Wrapper_NgFor(this._vc_6.vcRef, this._TemplateRef_6_5, this.parentView.parentView.injectorGet(import28.IterableDiffers, this.parentView.parentIndex), this.parentView.ref);
        this._text_7 = this.renderer.createText(this._el_4, '\n                        ', null);
        this._text_8 = this.renderer.createText(this._el_2, '\n                    ', null);
        this._text_9 = this.renderer.createText(this._el_0, '\n                    ', null);
        this._el_10 = import10.createRenderElement(this.renderer, this._el_0, 'tbody', import10.EMPTY_INLINE_ARRAY, null);
        this._text_11 = this.renderer.createText(this._el_10, '\n                        ', null);
        this._anchor_12 = this.renderer.createTemplateAnchor(this._el_10, null);
        this._vc_12 = new import27.ViewContainer(12, 10, this, this._anchor_12);
        this._TemplateRef_12_5 = new import34.TemplateRef_(this, 12, this._anchor_12);
        this._NgFor_12_6 = new import37.Wrapper_NgFor(this._vc_12.vcRef, this._TemplateRef_12_5, this.parentView.parentView.injectorGet(import28.IterableDiffers, this.parentView.parentIndex), this.parentView.ref);
        this._text_13 = this.renderer.createText(this._el_10, '\n                    ', null);
        this._text_14 = this.renderer.createText(this._el_0, '\n                ', null);
        this.init(this._el_0, (this.renderer.directRenderer ? null : [
            this._el_0,
            this._text_1,
            this._el_2,
            this._text_3,
            this._el_4,
            this._text_5,
            this._anchor_6,
            this._text_7,
            this._text_8,
            this._text_9,
            this._el_10,
            this._text_11,
            this._anchor_12,
            this._text_13,
            this._text_14
        ]), null);
        return null;
    };
    View_Calendar10.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import34.TemplateRef) && (6 === requestNodeIndex))) {
            return this._TemplateRef_6_5;
        }
        if (((token === import38.NgFor) && (6 === requestNodeIndex))) {
            return this._NgFor_6_6.context;
        }
        if (((token === import34.TemplateRef) && (12 === requestNodeIndex))) {
            return this._TemplateRef_12_5;
        }
        if (((token === import38.NgFor) && (12 === requestNodeIndex))) {
            return this._NgFor_12_6.context;
        }
        return notFoundResult;
    };
    View_Calendar10.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_6_0_0 = this.parentView.context.weekDays;
        this._NgFor_6_6.check_ngForOf(currVal_6_0_0, throwOnChange, false);
        this._NgFor_6_6.ngDoCheck(this, this._anchor_6, throwOnChange);
        var currVal_12_0_0 = this.parentView.context.dates;
        this._NgFor_12_6.check_ngForOf(currVal_12_0_0, throwOnChange, false);
        this._NgFor_12_6.ngDoCheck(this, this._anchor_12, throwOnChange);
        this._vc_6.detectChangesInNestedViews(throwOnChange);
        this._vc_12.detectChangesInNestedViews(throwOnChange);
    };
    View_Calendar10.prototype.destroyInternal = function () {
        this._vc_6.destroyNestedViews();
        this._vc_12.destroyNestedViews();
    };
    View_Calendar10.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    View_Calendar10.prototype.createEmbeddedViewInternal = function (nodeIndex) {
        if ((nodeIndex == 6)) {
            return new View_Calendar11(this.viewUtils, this, 6, this._anchor_6, this._vc_6);
        }
        if ((nodeIndex == 12)) {
            return new View_Calendar12(this.viewUtils, this, 12, this._anchor_12, this._vc_12);
        }
        return null;
    };
    return View_Calendar10;
}(import9.AppView));
var View_Calendar16 = (function (_super) {
    __extends(View_Calendar16, _super);
    function View_Calendar16(viewUtils, parentView, parentIndex, parentElement, declaredViewContainer) {
        _super.call(this, View_Calendar16, renderType_Calendar, import14.ViewType.EMBEDDED, viewUtils, parentView, parentIndex, parentElement, import15.ChangeDetectorStatus.CheckAlways, declaredViewContainer);
        this._expr_15 = import8.UNINITIALIZED;
    }
    View_Calendar16.prototype.createInternal = function (rootSelector) {
        this._el_0 = import10.createRenderElement(this.renderer, null, 'div', new import10.InlineArray2(2, 'class', 'ui-ampm-picker'), null);
        this._text_1 = this.renderer.createText(this._el_0, '\n                        ', null);
        this._el_2 = import10.createRenderElement(this.renderer, this._el_0, 'a', new import10.InlineArray2(2, 'href', '#'), null);
        this._text_3 = this.renderer.createText(this._el_2, '\n                            ', null);
        this._el_4 = import10.createRenderElement(this.renderer, this._el_2, 'span', new import10.InlineArray2(2, 'class', 'fa fa-angle-up'), null);
        this._text_5 = this.renderer.createText(this._el_2, '\n                        ', null);
        this._text_6 = this.renderer.createText(this._el_0, '\n                        ', null);
        this._el_7 = import10.createRenderElement(this.renderer, this._el_0, 'span', import10.EMPTY_INLINE_ARRAY, null);
        this._text_8 = this.renderer.createText(this._el_7, '', null);
        this._text_9 = this.renderer.createText(this._el_0, '\n                        ', null);
        this._el_10 = import10.createRenderElement(this.renderer, this._el_0, 'a', new import10.InlineArray2(2, 'href', '#'), null);
        this._text_11 = this.renderer.createText(this._el_10, '\n                            ', null);
        this._el_12 = import10.createRenderElement(this.renderer, this._el_10, 'span', new import10.InlineArray2(2, 'class', 'fa fa-angle-down'), null);
        this._text_13 = this.renderer.createText(this._el_10, '\n                        ', null);
        this._text_14 = this.renderer.createText(this._el_0, '\n                    ', null);
        var disposable_0 = import10.subscribeToRenderElement(this, this._el_2, new import10.InlineArray2(2, 'click', null), this.eventHandler(this.handleEvent_2));
        var disposable_1 = import10.subscribeToRenderElement(this, this._el_10, new import10.InlineArray2(2, 'click', null), this.eventHandler(this.handleEvent_10));
        this.init(this._el_0, (this.renderer.directRenderer ? null : [
            this._el_0,
            this._text_1,
            this._el_2,
            this._text_3,
            this._el_4,
            this._text_5,
            this._text_6,
            this._el_7,
            this._text_8,
            this._text_9,
            this._el_10,
            this._text_11,
            this._el_12,
            this._text_13,
            this._text_14
        ]), [
            disposable_0,
            disposable_1
        ]);
        return null;
    };
    View_Calendar16.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_15 = import10.inlineInterpolate(1, '', (this.parentView.parentView.context.pm ? 'PM' : 'AM'), '');
        if (import10.checkBinding(throwOnChange, this._expr_15, currVal_15)) {
            this.renderer.setText(this._text_8, currVal_15);
            this._expr_15 = currVal_15;
        }
    };
    View_Calendar16.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    View_Calendar16.prototype.handleEvent_2 = function (eventName, $event) {
        this.markPathToRootAsCheckOnce();
        var result = true;
        if ((eventName == 'click')) {
            var pd_sub_0 = (this.parentView.parentView.context.toggleAMPM($event) !== false);
            result = (pd_sub_0 && result);
        }
        return result;
    };
    View_Calendar16.prototype.handleEvent_10 = function (eventName, $event) {
        this.markPathToRootAsCheckOnce();
        var result = true;
        if ((eventName == 'click')) {
            var pd_sub_0 = (this.parentView.parentView.context.toggleAMPM($event) !== false);
            result = (pd_sub_0 && result);
        }
        return result;
    };
    return View_Calendar16;
}(import9.AppView));
var View_Calendar15 = (function (_super) {
    __extends(View_Calendar15, _super);
    function View_Calendar15(viewUtils, parentView, parentIndex, parentElement, declaredViewContainer) {
        _super.call(this, View_Calendar15, renderType_Calendar, import14.ViewType.EMBEDDED, viewUtils, parentView, parentIndex, parentElement, import15.ChangeDetectorStatus.CheckAlways, declaredViewContainer);
        this._map_61 = import10.pureProxy1(function (p0) {
            return { display: p0 };
        });
        this._expr_62 = import8.UNINITIALIZED;
        this._map_63 = import10.pureProxy1(function (p0) {
            return { display: p0 };
        });
        this._expr_64 = import8.UNINITIALIZED;
    }
    View_Calendar15.prototype.createInternal = function (rootSelector) {
        this._el_0 = import10.createRenderElement(this.renderer, null, 'div', new import10.InlineArray2(2, 'class', 'ui-timepicker ui-widget-header ui-corner-all'), null);
        this._text_1 = this.renderer.createText(this._el_0, '\n                    ', null);
        this._el_2 = import10.createRenderElement(this.renderer, this._el_0, 'div', new import10.InlineArray2(2, 'class', 'ui-hour-picker'), null);
        this._text_3 = this.renderer.createText(this._el_2, '\n                        ', null);
        this._el_4 = import10.createRenderElement(this.renderer, this._el_2, 'a', new import10.InlineArray2(2, 'href', '#'), null);
        this._text_5 = this.renderer.createText(this._el_4, '\n                            ', null);
        this._el_6 = import10.createRenderElement(this.renderer, this._el_4, 'span', new import10.InlineArray2(2, 'class', 'fa fa-angle-up'), null);
        this._text_7 = this.renderer.createText(this._el_4, '\n                        ', null);
        this._text_8 = this.renderer.createText(this._el_2, '\n                        ', null);
        this._el_9 = import10.createRenderElement(this.renderer, this._el_2, 'span', import10.EMPTY_INLINE_ARRAY, null);
        this._NgStyle_9_3 = new import31.Wrapper_NgStyle(this.parentView.parentView.injectorGet(import29.KeyValueDiffers, this.parentView.parentIndex), new import17.ElementRef(this._el_9), this.renderer);
        this._text_10 = this.renderer.createText(this._el_9, '0', null);
        this._el_11 = import10.createRenderElement(this.renderer, this._el_2, 'span', import10.EMPTY_INLINE_ARRAY, null);
        this._text_12 = this.renderer.createText(this._el_11, '', null);
        this._text_13 = this.renderer.createText(this._el_2, '\n                        ', null);
        this._el_14 = import10.createRenderElement(this.renderer, this._el_2, 'a', new import10.InlineArray2(2, 'href', '#'), null);
        this._text_15 = this.renderer.createText(this._el_14, '\n                            ', null);
        this._el_16 = import10.createRenderElement(this.renderer, this._el_14, 'span', new import10.InlineArray2(2, 'class', 'fa fa-angle-down'), null);
        this._text_17 = this.renderer.createText(this._el_14, '\n                        ', null);
        this._text_18 = this.renderer.createText(this._el_2, '\n                    ', null);
        this._text_19 = this.renderer.createText(this._el_0, '\n                    ', null);
        this._el_20 = import10.createRenderElement(this.renderer, this._el_0, 'div', new import10.InlineArray2(2, 'class', 'ui-separator'), null);
        this._text_21 = this.renderer.createText(this._el_20, '\n                        ', null);
        this._el_22 = import10.createRenderElement(this.renderer, this._el_20, 'a', new import10.InlineArray2(2, 'href', '#'), null);
        this._text_23 = this.renderer.createText(this._el_22, '\n                            ', null);
        this._el_24 = import10.createRenderElement(this.renderer, this._el_22, 'span', new import10.InlineArray2(2, 'class', 'fa fa-angle-up'), null);
        this._text_25 = this.renderer.createText(this._el_22, '\n                        ', null);
        this._text_26 = this.renderer.createText(this._el_20, '\n                        ', null);
        this._el_27 = import10.createRenderElement(this.renderer, this._el_20, 'span', import10.EMPTY_INLINE_ARRAY, null);
        this._text_28 = this.renderer.createText(this._el_27, ':', null);
        this._text_29 = this.renderer.createText(this._el_20, '\n                        ', null);
        this._el_30 = import10.createRenderElement(this.renderer, this._el_20, 'a', new import10.InlineArray2(2, 'href', '#'), null);
        this._text_31 = this.renderer.createText(this._el_30, '\n                            ', null);
        this._el_32 = import10.createRenderElement(this.renderer, this._el_30, 'span', new import10.InlineArray2(2, 'class', 'fa fa-angle-down'), null);
        this._text_33 = this.renderer.createText(this._el_30, '\n                        ', null);
        this._text_34 = this.renderer.createText(this._el_20, '\n                    ', null);
        this._text_35 = this.renderer.createText(this._el_0, '\n                    ', null);
        this._el_36 = import10.createRenderElement(this.renderer, this._el_0, 'div', new import10.InlineArray2(2, 'class', 'ui-minute-picker'), null);
        this._text_37 = this.renderer.createText(this._el_36, '\n                        ', null);
        this._el_38 = import10.createRenderElement(this.renderer, this._el_36, 'a', new import10.InlineArray2(2, 'href', '#'), null);
        this._text_39 = this.renderer.createText(this._el_38, '\n                            ', null);
        this._el_40 = import10.createRenderElement(this.renderer, this._el_38, 'span', new import10.InlineArray2(2, 'class', 'fa fa-angle-up'), null);
        this._text_41 = this.renderer.createText(this._el_38, '\n                        ', null);
        this._text_42 = this.renderer.createText(this._el_36, '\n                        ', null);
        this._el_43 = import10.createRenderElement(this.renderer, this._el_36, 'span', import10.EMPTY_INLINE_ARRAY, null);
        this._NgStyle_43_3 = new import31.Wrapper_NgStyle(this.parentView.parentView.injectorGet(import29.KeyValueDiffers, this.parentView.parentIndex), new import17.ElementRef(this._el_43), this.renderer);
        this._text_44 = this.renderer.createText(this._el_43, '0', null);
        this._el_45 = import10.createRenderElement(this.renderer, this._el_36, 'span', import10.EMPTY_INLINE_ARRAY, null);
        this._text_46 = this.renderer.createText(this._el_45, '', null);
        this._text_47 = this.renderer.createText(this._el_36, '\n                        ', null);
        this._el_48 = import10.createRenderElement(this.renderer, this._el_36, 'a', new import10.InlineArray2(2, 'href', '#'), null);
        this._text_49 = this.renderer.createText(this._el_48, '\n                            ', null);
        this._el_50 = import10.createRenderElement(this.renderer, this._el_48, 'span', new import10.InlineArray2(2, 'class', 'fa fa-angle-down'), null);
        this._text_51 = this.renderer.createText(this._el_48, '\n                        ', null);
        this._text_52 = this.renderer.createText(this._el_36, '\n                    ', null);
        this._text_53 = this.renderer.createText(this._el_0, '\n                    ', null);
        this._anchor_54 = this.renderer.createTemplateAnchor(this._el_0, null);
        this._vc_54 = new import27.ViewContainer(54, 0, this, this._anchor_54);
        this._TemplateRef_54_5 = new import34.TemplateRef_(this, 54, this._anchor_54);
        this._NgIf_54_6 = new import33.Wrapper_NgIf(this._vc_54.vcRef, this._TemplateRef_54_5);
        this._text_55 = this.renderer.createText(this._el_0, '\n                ', null);
        var disposable_0 = import10.subscribeToRenderElement(this, this._el_4, new import10.InlineArray2(2, 'click', null), this.eventHandler(this.handleEvent_4));
        var disposable_1 = import10.subscribeToRenderElement(this, this._el_14, new import10.InlineArray2(2, 'click', null), this.eventHandler(this.handleEvent_14));
        var disposable_2 = import10.subscribeToRenderElement(this, this._el_38, new import10.InlineArray2(2, 'click', null), this.eventHandler(this.handleEvent_38));
        var disposable_3 = import10.subscribeToRenderElement(this, this._el_48, new import10.InlineArray2(2, 'click', null), this.eventHandler(this.handleEvent_48));
        this.init(this._el_0, (this.renderer.directRenderer ? null : [
            this._el_0,
            this._text_1,
            this._el_2,
            this._text_3,
            this._el_4,
            this._text_5,
            this._el_6,
            this._text_7,
            this._text_8,
            this._el_9,
            this._text_10,
            this._el_11,
            this._text_12,
            this._text_13,
            this._el_14,
            this._text_15,
            this._el_16,
            this._text_17,
            this._text_18,
            this._text_19,
            this._el_20,
            this._text_21,
            this._el_22,
            this._text_23,
            this._el_24,
            this._text_25,
            this._text_26,
            this._el_27,
            this._text_28,
            this._text_29,
            this._el_30,
            this._text_31,
            this._el_32,
            this._text_33,
            this._text_34,
            this._text_35,
            this._el_36,
            this._text_37,
            this._el_38,
            this._text_39,
            this._el_40,
            this._text_41,
            this._text_42,
            this._el_43,
            this._text_44,
            this._el_45,
            this._text_46,
            this._text_47,
            this._el_48,
            this._text_49,
            this._el_50,
            this._text_51,
            this._text_52,
            this._text_53,
            this._anchor_54,
            this._text_55
        ]), [
            disposable_0,
            disposable_1,
            disposable_2,
            disposable_3
        ]);
        return null;
    };
    View_Calendar15.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import35.NgStyle) && ((9 <= requestNodeIndex) && (requestNodeIndex <= 10)))) {
            return this._NgStyle_9_3.context;
        }
        if (((token === import35.NgStyle) && ((43 <= requestNodeIndex) && (requestNodeIndex <= 44)))) {
            return this._NgStyle_43_3.context;
        }
        if (((token === import34.TemplateRef) && (54 === requestNodeIndex))) {
            return this._TemplateRef_54_5;
        }
        if (((token === import36.NgIf) && (54 === requestNodeIndex))) {
            return this._NgIf_54_6.context;
        }
        return notFoundResult;
    };
    View_Calendar15.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_9_0_0 = this._map_61(((this.parentView.context.currentHour < 10) ? 'inline' : 'none'));
        this._NgStyle_9_3.check_ngStyle(currVal_9_0_0, throwOnChange, false);
        this._NgStyle_9_3.ngDoCheck(this, this._el_9, throwOnChange);
        var currVal_43_0_0 = this._map_63(((this.parentView.context.currentMinute < 10) ? 'inline' : 'none'));
        this._NgStyle_43_3.check_ngStyle(currVal_43_0_0, throwOnChange, false);
        this._NgStyle_43_3.ngDoCheck(this, this._el_43, throwOnChange);
        var currVal_54_0_0 = (this.parentView.context.hourFormat == '12');
        this._NgIf_54_6.check_ngIf(currVal_54_0_0, throwOnChange, false);
        this._NgIf_54_6.ngDoCheck(this, this._anchor_54, throwOnChange);
        this._vc_54.detectChangesInNestedViews(throwOnChange);
        var currVal_62 = import10.inlineInterpolate(1, '', this.parentView.context.currentHour, '');
        if (import10.checkBinding(throwOnChange, this._expr_62, currVal_62)) {
            this.renderer.setText(this._text_12, currVal_62);
            this._expr_62 = currVal_62;
        }
        var currVal_64 = import10.inlineInterpolate(1, '', this.parentView.context.currentMinute, '');
        if (import10.checkBinding(throwOnChange, this._expr_64, currVal_64)) {
            this.renderer.setText(this._text_46, currVal_64);
            this._expr_64 = currVal_64;
        }
    };
    View_Calendar15.prototype.destroyInternal = function () {
        this._vc_54.destroyNestedViews();
    };
    View_Calendar15.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    View_Calendar15.prototype.createEmbeddedViewInternal = function (nodeIndex) {
        if ((nodeIndex == 54)) {
            return new View_Calendar16(this.viewUtils, this, 54, this._anchor_54, this._vc_54);
        }
        return null;
    };
    View_Calendar15.prototype.handleEvent_4 = function (eventName, $event) {
        this.markPathToRootAsCheckOnce();
        var result = true;
        if ((eventName == 'click')) {
            var pd_sub_0 = (this.parentView.context.incrementHour($event) !== false);
            result = (pd_sub_0 && result);
        }
        return result;
    };
    View_Calendar15.prototype.handleEvent_14 = function (eventName, $event) {
        this.markPathToRootAsCheckOnce();
        var result = true;
        if ((eventName == 'click')) {
            var pd_sub_0 = (this.parentView.context.decrementHour($event) !== false);
            result = (pd_sub_0 && result);
        }
        return result;
    };
    View_Calendar15.prototype.handleEvent_38 = function (eventName, $event) {
        this.markPathToRootAsCheckOnce();
        var result = true;
        if ((eventName == 'click')) {
            var pd_sub_0 = (this.parentView.context.incrementMinute($event) !== false);
            result = (pd_sub_0 && result);
        }
        return result;
    };
    View_Calendar15.prototype.handleEvent_48 = function (eventName, $event) {
        this.markPathToRootAsCheckOnce();
        var result = true;
        if ((eventName == 'click')) {
            var pd_sub_0 = (this.parentView.context.decrementMinute($event) !== false);
            result = (pd_sub_0 && result);
        }
        return result;
    };
    return View_Calendar15;
}(import9.AppView));
var renderType_Calendar = import10.createRenderComponentType('', 0, import12.ViewEncapsulation.None, styles_Calendar, { overlayState: Calendar_overlayState_factory });
export var View_Calendar0 = (function (_super) {
    __extends(View_Calendar0, _super);
    function View_Calendar0(viewUtils, parentView, parentIndex, parentElement) {
        _super.call(this, View_Calendar0, renderType_Calendar, import14.ViewType.COMPONENT, viewUtils, parentView, parentIndex, parentElement, import15.ChangeDetectorStatus.CheckAlways);
        this._map_32 = import10.pureProxy2(function (p0, p1) {
            return {
                'ui-calendar': p0,
                'ui-calendar-w-btn': p1
            };
        });
        this._expr_33 = import8.UNINITIALIZED;
        this._map_34 = import10.pureProxy4(function (p0, p1, p2, p3) {
            return {
                'ui-datepicker-inline': p0,
                'ui-shadow': p1,
                'ui-state-disabled': p2,
                'ui-datepicker-timeonly': p3
            };
        });
        this._map_35 = import10.pureProxy1(function (p0) {
            return { display: p0 };
        });
    }
    View_Calendar0.prototype.createInternal = function (rootSelector) {
        var parentRenderNode = this.renderer.createViewRoot(this.parentElement);
        this._viewQuery_datepicker_0 = new import39.QueryList();
        this._text_0 = this.renderer.createText(parentRenderNode, '\n        ', null);
        this._el_1 = import10.createRenderElement(this.renderer, parentRenderNode, 'span', import10.EMPTY_INLINE_ARRAY, null);
        this._NgClass_1_3 = new import25.Wrapper_NgClass(this.parentView.injectorGet(import28.IterableDiffers, this.parentIndex), this.parentView.injectorGet(import29.KeyValueDiffers, this.parentIndex), new import17.ElementRef(this._el_1), this.renderer);
        this._NgStyle_1_4 = new import31.Wrapper_NgStyle(this.parentView.injectorGet(import29.KeyValueDiffers, this.parentIndex), new import17.ElementRef(this._el_1), this.renderer);
        this._text_2 = this.renderer.createText(this._el_1, '\n            ', null);
        this._anchor_3 = this.renderer.createTemplateAnchor(this._el_1, null);
        this._vc_3 = new import27.ViewContainer(3, 1, this, this._anchor_3);
        this._TemplateRef_3_5 = new import34.TemplateRef_(this, 3, this._anchor_3);
        this._NgIf_3_6 = new import33.Wrapper_NgIf(this._vc_3.vcRef, this._TemplateRef_3_5);
        this._text_4 = this.renderer.createText(this._el_1, '\n            ', null);
        this._el_5 = import10.createRenderElement(this.renderer, this._el_1, 'div', new import10.InlineArray2(2, 'class', 'ui-datepicker ui-widget ui-widget-content ui-helper-clearfix ui-corner-all'), null);
        this._NgClass_5_3 = new import25.Wrapper_NgClass(this.parentView.injectorGet(import28.IterableDiffers, this.parentIndex), this.parentView.injectorGet(import29.KeyValueDiffers, this.parentIndex), new import17.ElementRef(this._el_5), this.renderer);
        this._NgStyle_5_4 = new import31.Wrapper_NgStyle(this.parentView.injectorGet(import29.KeyValueDiffers, this.parentIndex), new import17.ElementRef(this._el_5), this.renderer);
        this._text_6 = this.renderer.createText(this._el_5, '\n                ', null);
        this._anchor_7 = this.renderer.createTemplateAnchor(this._el_5, null);
        this._vc_7 = new import27.ViewContainer(7, 5, this, this._anchor_7);
        this._TemplateRef_7_5 = new import34.TemplateRef_(this, 7, this._anchor_7);
        this._NgIf_7_6 = new import33.Wrapper_NgIf(this._vc_7.vcRef, this._TemplateRef_7_5);
        this._text_8 = this.renderer.createText(this._el_5, '\n                ', null);
        this._anchor_9 = this.renderer.createTemplateAnchor(this._el_5, null);
        this._vc_9 = new import27.ViewContainer(9, 5, this, this._anchor_9);
        this._TemplateRef_9_5 = new import34.TemplateRef_(this, 9, this._anchor_9);
        this._NgIf_9_6 = new import33.Wrapper_NgIf(this._vc_9.vcRef, this._TemplateRef_9_5);
        this._text_10 = this.renderer.createText(this._el_5, '\n                ', null);
        this._anchor_11 = this.renderer.createTemplateAnchor(this._el_5, null);
        this._vc_11 = new import27.ViewContainer(11, 5, this, this._anchor_11);
        this._TemplateRef_11_5 = new import34.TemplateRef_(this, 11, this._anchor_11);
        this._NgIf_11_6 = new import33.Wrapper_NgIf(this._vc_11.vcRef, this._TemplateRef_11_5);
        this._text_12 = this.renderer.createText(this._el_5, '\n            ', null);
        this._text_13 = this.renderer.createText(this._el_1, '\n        ', null);
        this._text_14 = this.renderer.createText(parentRenderNode, '\n    ', null);
        var disposable_0 = import10.subscribeToRenderElement(this, this._el_5, new import10.InlineArray2(2, 'click', null), this.eventHandler(this.handleEvent_5));
        this._viewQuery_datepicker_0.reset([new import17.ElementRef(this._el_5)]);
        this.context.overlayViewChild = this._viewQuery_datepicker_0.first;
        this.init(null, (this.renderer.directRenderer ? null : [
            this._text_0,
            this._el_1,
            this._text_2,
            this._anchor_3,
            this._text_4,
            this._el_5,
            this._text_6,
            this._anchor_7,
            this._text_8,
            this._anchor_9,
            this._text_10,
            this._anchor_11,
            this._text_12,
            this._text_13,
            this._text_14
        ]), [disposable_0]);
        return null;
    };
    View_Calendar0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import34.TemplateRef) && (3 === requestNodeIndex))) {
            return this._TemplateRef_3_5;
        }
        if (((token === import36.NgIf) && (3 === requestNodeIndex))) {
            return this._NgIf_3_6.context;
        }
        if (((token === import34.TemplateRef) && (7 === requestNodeIndex))) {
            return this._TemplateRef_7_5;
        }
        if (((token === import36.NgIf) && (7 === requestNodeIndex))) {
            return this._NgIf_7_6.context;
        }
        if (((token === import34.TemplateRef) && (9 === requestNodeIndex))) {
            return this._TemplateRef_9_5;
        }
        if (((token === import36.NgIf) && (9 === requestNodeIndex))) {
            return this._NgIf_9_6.context;
        }
        if (((token === import34.TemplateRef) && (11 === requestNodeIndex))) {
            return this._TemplateRef_11_5;
        }
        if (((token === import36.NgIf) && (11 === requestNodeIndex))) {
            return this._NgIf_11_6.context;
        }
        if (((token === import30.NgClass) && ((5 <= requestNodeIndex) && (requestNodeIndex <= 12)))) {
            return this._NgClass_5_3.context;
        }
        if (((token === import35.NgStyle) && ((5 <= requestNodeIndex) && (requestNodeIndex <= 12)))) {
            return this._NgStyle_5_4.context;
        }
        if (((token === import30.NgClass) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 13)))) {
            return this._NgClass_1_3.context;
        }
        if (((token === import35.NgStyle) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 13)))) {
            return this._NgStyle_1_4.context;
        }
        return notFoundResult;
    };
    View_Calendar0.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_33 = (this.context.inline ? 'visible' : (this.context.overlayVisible ? 'visible' : 'hidden'));
        if (import10.checkBinding(throwOnChange, this._expr_33, currVal_33)) {
            var animationTransition_overlayState = this.componentType.animations['overlayState'](this, this._el_5, ((this._expr_33 == import8.UNINITIALIZED) ? 'void' : this._expr_33), ((currVal_33 == import8.UNINITIALIZED) ? 'void' : currVal_33));
            this._expr_33 = currVal_33;
        }
        var currVal_1_0_0 = this.context.styleClass;
        this._NgClass_1_3.check_klass(currVal_1_0_0, throwOnChange, false);
        var currVal_1_0_1 = this._map_32(true, this.context.showIcon);
        this._NgClass_1_3.check_ngClass(currVal_1_0_1, throwOnChange, false);
        this._NgClass_1_3.ngDoCheck(this, this._el_1, throwOnChange);
        var currVal_1_1_0 = this.context.style;
        this._NgStyle_1_4.check_ngStyle(currVal_1_1_0, throwOnChange, false);
        this._NgStyle_1_4.ngDoCheck(this, this._el_1, throwOnChange);
        var currVal_3_0_0 = !this.context.inline;
        this._NgIf_3_6.check_ngIf(currVal_3_0_0, throwOnChange, false);
        this._NgIf_3_6.ngDoCheck(this, this._anchor_3, throwOnChange);
        var currVal_5_0_0 = 'ui-datepicker ui-widget ui-widget-content ui-helper-clearfix ui-corner-all';
        this._NgClass_5_3.check_klass(currVal_5_0_0, throwOnChange, false);
        var currVal_5_0_1 = this._map_34(this.context.inline, !this.context.inline, this.context.disabled, this.context.timeOnly);
        this._NgClass_5_3.check_ngClass(currVal_5_0_1, throwOnChange, false);
        this._NgClass_5_3.ngDoCheck(this, this._el_5, throwOnChange);
        var currVal_5_1_0 = this._map_35((this.context.inline ? 'inline-block' : (this.context.overlayVisible ? 'block' : 'none')));
        this._NgStyle_5_4.check_ngStyle(currVal_5_1_0, throwOnChange, false);
        this._NgStyle_5_4.ngDoCheck(this, this._el_5, throwOnChange);
        var currVal_7_0_0 = !this.context.timeOnly;
        this._NgIf_7_6.check_ngIf(currVal_7_0_0, throwOnChange, false);
        this._NgIf_7_6.ngDoCheck(this, this._anchor_7, throwOnChange);
        var currVal_9_0_0 = !this.context.timeOnly;
        this._NgIf_9_6.check_ngIf(currVal_9_0_0, throwOnChange, false);
        this._NgIf_9_6.ngDoCheck(this, this._anchor_9, throwOnChange);
        var currVal_11_0_0 = (this.context.showTime || this.context.timeOnly);
        this._NgIf_11_6.check_ngIf(currVal_11_0_0, throwOnChange, false);
        this._NgIf_11_6.ngDoCheck(this, this._anchor_11, throwOnChange);
        this._vc_3.detectChangesInNestedViews(throwOnChange);
        this._vc_7.detectChangesInNestedViews(throwOnChange);
        this._vc_9.detectChangesInNestedViews(throwOnChange);
        this._vc_11.detectChangesInNestedViews(throwOnChange);
    };
    View_Calendar0.prototype.destroyInternal = function () {
        this._vc_3.destroyNestedViews();
        this._vc_7.destroyNestedViews();
        this._vc_9.destroyNestedViews();
        this._vc_11.destroyNestedViews();
    };
    View_Calendar0.prototype.detachInternal = function () {
        var animationTransition_overlayState = this.componentType.animations['overlayState'](this, this._el_5, this._expr_33, 'void');
    };
    View_Calendar0.prototype.createEmbeddedViewInternal = function (nodeIndex) {
        if ((nodeIndex == 3)) {
            return new View_Calendar1(this.viewUtils, this, 3, this._anchor_3, this._vc_3);
        }
        if ((nodeIndex == 7)) {
            return new View_Calendar3(this.viewUtils, this, 7, this._anchor_7, this._vc_7);
        }
        if ((nodeIndex == 9)) {
            return new View_Calendar10(this.viewUtils, this, 9, this._anchor_9, this._vc_9);
        }
        if ((nodeIndex == 11)) {
            return new View_Calendar15(this.viewUtils, this, 11, this._anchor_11, this._vc_11);
        }
        return null;
    };
    View_Calendar0.prototype.handleEvent_5 = function (eventName, $event) {
        this.markPathToRootAsCheckOnce();
        var result = true;
        if ((eventName == 'click')) {
            var pd_sub_0 = (this.context.onDatePickerClick($event) !== false);
            result = (pd_sub_0 && result);
        }
        return result;
    };
    return View_Calendar0;
}(import9.AppView));
