// primeng/components/calendar/calendar.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from '@angular/core/src/linker/ng_module_factory';
import * as import1 from 'primeng/components/calendar/calendar';
import * as import2 from '@angular/common/src/common_module';
import * as import3 from 'primeng/components/button/button';
import * as import4 from 'primeng/components/inputtext/inputtext';
import * as import5 from '@angular/common/src/localization';
import * as import6 from '@angular/core/src/di/injector';
import * as import7 from '@angular/core/src/i18n/tokens';
import * as import8 from '@angular/core/src/change_detection/change_detection_util';
import * as import9 from '@angular/core/src/linker/view';
import * as import10 from '@angular/core/src/linker/view_utils';
import * as import11 from '@angular/core/src/render/api';
import * as import12 from '@angular/core/src/metadata/view';
import * as import13 from 'primeng/components/dom/domhandler';
import * as import14 from '@angular/core/src/linker/view_type';
import * as import15 from '@angular/core/src/change_detection/constants';
import * as import16 from '@angular/core/src/linker/component_factory';
import * as import17 from '@angular/core/src/linker/element_ref';
import * as import18 from '@angular/forms/src/directives/control_value_accessor';
import * as import19 from '@angular/core/src/animation/animation_transition';
import * as import20 from '@angular/core/src/animation/animation_sequence_player';
import * as import21 from '@angular/core/src/animation/animation_styles';
import * as import22 from '@angular/core/src/animation/animation_style_util';
import * as import23 from '@angular/core/src/animation/animation_keyframe';
import * as import24 from '@angular/core/src/animation/animation_player';
import * as import25 from '../../../@angular/common/src/directives/ng_class.ngfactory';
import * as import26 from '../button/button.ngfactory';
import * as import27 from '@angular/core/src/linker/view_container';
import * as import28 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import29 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import30 from '@angular/common/src/directives/ng_class';
import * as import31 from '../../../@angular/common/src/directives/ng_style.ngfactory';
import * as import32 from '../inputtext/inputtext.ngfactory';
import * as import33 from '../../../@angular/common/src/directives/ng_if.ngfactory';
import * as import34 from '@angular/core/src/linker/template_ref';
import * as import35 from '@angular/common/src/directives/ng_style';
import * as import36 from '@angular/common/src/directives/ng_if';
import * as import37 from '../../../@angular/common/src/directives/ng_for.ngfactory';
import * as import38 from '@angular/common/src/directives/ng_for';
import * as import39 from '@angular/core/src/linker/query_list';
class CalendarModuleInjector extends import0.NgModuleInjector<import1.CalendarModule> {
  _CommonModule_0:import2.CommonModule;
  _ButtonModule_1:import3.ButtonModule;
  _InputTextModule_2:import4.InputTextModule;
  _CalendarModule_3:import1.CalendarModule;
  __NgLocalization_4:import5.NgLocaleLocalization;
  constructor(parent:import6.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_4():import5.NgLocaleLocalization {
    if ((this.__NgLocalization_4 == null)) { (this.__NgLocalization_4 = new import5.NgLocaleLocalization(this.parent.get(import7.LOCALE_ID))); }
    return this.__NgLocalization_4;
  }
  createInternal():import1.CalendarModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._ButtonModule_1 = new import3.ButtonModule();
    this._InputTextModule_2 = new import4.InputTextModule();
    this._CalendarModule_3 = new import1.CalendarModule();
    return this._CalendarModule_3;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.ButtonModule)) { return this._ButtonModule_1; }
    if ((token === import4.InputTextModule)) { return this._InputTextModule_2; }
    if ((token === import1.CalendarModule)) { return this._CalendarModule_3; }
    if ((token === import5.NgLocalization)) { return this._NgLocalization_4; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const CalendarModuleNgFactory:import0.NgModuleFactory<import1.CalendarModule> = new import0.NgModuleFactory(CalendarModuleInjector,import1.CalendarModule);
export class Wrapper_Calendar {
  /*private*/ _eventHandler:Function;
  context:import1.Calendar;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  /*private*/ _expr_5:any;
  /*private*/ _expr_6:any;
  /*private*/ _expr_7:any;
  /*private*/ _expr_8:any;
  /*private*/ _expr_9:any;
  /*private*/ _expr_10:any;
  /*private*/ _expr_11:any;
  /*private*/ _expr_12:any;
  /*private*/ _expr_13:any;
  /*private*/ _expr_14:any;
  /*private*/ _expr_15:any;
  /*private*/ _expr_16:any;
  /*private*/ _expr_17:any;
  /*private*/ _expr_18:any;
  /*private*/ _expr_19:any;
  /*private*/ _expr_20:any;
  /*private*/ _expr_21:any;
  /*private*/ _expr_22:any;
  /*private*/ _expr_23:any;
  /*private*/ _expr_24:any;
  /*private*/ _expr_25:any;
  /*private*/ _expr_26:any;
  /*private*/ _expr_27:any;
  /*private*/ _expr_28:any;
  /*private*/ _expr_29:any;
  subscription0:any;
  subscription1:any;
  constructor(p0:any,p1:any,p2:any) {
    this._changed = false;
    this.context = new import1.Calendar(p0,p1,p2);
    this._expr_0 = import8.UNINITIALIZED;
    this._expr_1 = import8.UNINITIALIZED;
    this._expr_2 = import8.UNINITIALIZED;
    this._expr_3 = import8.UNINITIALIZED;
    this._expr_4 = import8.UNINITIALIZED;
    this._expr_5 = import8.UNINITIALIZED;
    this._expr_6 = import8.UNINITIALIZED;
    this._expr_7 = import8.UNINITIALIZED;
    this._expr_8 = import8.UNINITIALIZED;
    this._expr_9 = import8.UNINITIALIZED;
    this._expr_10 = import8.UNINITIALIZED;
    this._expr_11 = import8.UNINITIALIZED;
    this._expr_12 = import8.UNINITIALIZED;
    this._expr_13 = import8.UNINITIALIZED;
    this._expr_14 = import8.UNINITIALIZED;
    this._expr_15 = import8.UNINITIALIZED;
    this._expr_16 = import8.UNINITIALIZED;
    this._expr_17 = import8.UNINITIALIZED;
    this._expr_18 = import8.UNINITIALIZED;
    this._expr_19 = import8.UNINITIALIZED;
    this._expr_20 = import8.UNINITIALIZED;
    this._expr_21 = import8.UNINITIALIZED;
    this._expr_22 = import8.UNINITIALIZED;
    this._expr_23 = import8.UNINITIALIZED;
    this._expr_24 = import8.UNINITIALIZED;
    this._expr_25 = import8.UNINITIALIZED;
    this._expr_26 = import8.UNINITIALIZED;
    this._expr_27 = import8.UNINITIALIZED;
    this._expr_28 = import8.UNINITIALIZED;
    this._expr_29 = import8.UNINITIALIZED;
  }
  ngOnDetach(view:import9.AppView<any>,componentView:import9.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
    this.context.ngOnDestroy();
    (this.subscription0 && this.subscription0.unsubscribe());
    (this.subscription1 && this.subscription1.unsubscribe());
  }
  check_defaultDate(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.defaultDate = currValue;
      this._expr_0 = currValue;
    }
  }
  check_style(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.style = currValue;
      this._expr_1 = currValue;
    }
  }
  check_styleClass(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_2,currValue))) {
      this._changed = true;
      this.context.styleClass = currValue;
      this._expr_2 = currValue;
    }
  }
  check_inputStyle(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_3,currValue))) {
      this._changed = true;
      this.context.inputStyle = currValue;
      this._expr_3 = currValue;
    }
  }
  check_inputStyleClass(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_4,currValue))) {
      this._changed = true;
      this.context.inputStyleClass = currValue;
      this._expr_4 = currValue;
    }
  }
  check_placeholder(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_5,currValue))) {
      this._changed = true;
      this.context.placeholder = currValue;
      this._expr_5 = currValue;
    }
  }
  check_disabled(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_6,currValue))) {
      this._changed = true;
      this.context.disabled = currValue;
      this._expr_6 = currValue;
    }
  }
  check_dateFormat(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_7,currValue))) {
      this._changed = true;
      this.context.dateFormat = currValue;
      this._expr_7 = currValue;
    }
  }
  check_inline(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_8,currValue))) {
      this._changed = true;
      this.context.inline = currValue;
      this._expr_8 = currValue;
    }
  }
  check_showOtherMonths(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_9,currValue))) {
      this._changed = true;
      this.context.showOtherMonths = currValue;
      this._expr_9 = currValue;
    }
  }
  check_selectOtherMonths(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_10,currValue))) {
      this._changed = true;
      this.context.selectOtherMonths = currValue;
      this._expr_10 = currValue;
    }
  }
  check_showIcon(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_11,currValue))) {
      this._changed = true;
      this.context.showIcon = currValue;
      this._expr_11 = currValue;
    }
  }
  check_icon(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_12,currValue))) {
      this._changed = true;
      this.context.icon = currValue;
      this._expr_12 = currValue;
    }
  }
  check_appendTo(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_13,currValue))) {
      this._changed = true;
      this.context.appendTo = currValue;
      this._expr_13 = currValue;
    }
  }
  check_readonlyInput(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_14,currValue))) {
      this._changed = true;
      this.context.readonlyInput = currValue;
      this._expr_14 = currValue;
    }
  }
  check_shortYearCutoff(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_15,currValue))) {
      this._changed = true;
      this.context.shortYearCutoff = currValue;
      this._expr_15 = currValue;
    }
  }
  check_monthNavigator(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_16,currValue))) {
      this._changed = true;
      this.context.monthNavigator = currValue;
      this._expr_16 = currValue;
    }
  }
  check_yearNavigator(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_17,currValue))) {
      this._changed = true;
      this.context.yearNavigator = currValue;
      this._expr_17 = currValue;
    }
  }
  check_yearRange(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_18,currValue))) {
      this._changed = true;
      this.context.yearRange = currValue;
      this._expr_18 = currValue;
    }
  }
  check_showTime(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_19,currValue))) {
      this._changed = true;
      this.context.showTime = currValue;
      this._expr_19 = currValue;
    }
  }
  check_hourFormat(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_20,currValue))) {
      this._changed = true;
      this.context.hourFormat = currValue;
      this._expr_20 = currValue;
    }
  }
  check_timeOnly(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_21,currValue))) {
      this._changed = true;
      this.context.timeOnly = currValue;
      this._expr_21 = currValue;
    }
  }
  check_required(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_22,currValue))) {
      this._changed = true;
      this.context.required = currValue;
      this._expr_22 = currValue;
    }
  }
  check_dataType(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_23,currValue))) {
      this._changed = true;
      this.context.dataType = currValue;
      this._expr_23 = currValue;
    }
  }
  check_locale(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_24,currValue))) {
      this._changed = true;
      this.context.locale = currValue;
      this._expr_24 = currValue;
    }
  }
  check_tabindex(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_25,currValue))) {
      this._changed = true;
      this.context.tabindex = currValue;
      this._expr_25 = currValue;
    }
  }
  check_minDate(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_26,currValue))) {
      this._changed = true;
      this.context.minDate = currValue;
      this._expr_26 = currValue;
    }
  }
  check_maxDate(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import10.checkBinding(throwOnChange,this._expr_27,currValue))) {
      this._changed = true;
      this.context.maxDate = currValue;
      this._expr_27 = currValue;
    }
  }
  ngDoCheck(view:import9.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    if (!throwOnChange) { if ((view.numberOfChecks === 0)) { this.context.ngOnInit(); } }
    return changed;
  }
  checkHost(view:import9.AppView<any>,componentView:import9.AppView<any>,el:any,throwOnChange:boolean):void {
    const currVal_28:any = this.context.filled;
    if (import10.checkBinding(throwOnChange,this._expr_28,currVal_28)) {
      view.renderer.setElementClass(el,'ui-inputwrapper-filled',currVal_28);
      this._expr_28 = currVal_28;
    }
    const currVal_29:any = this.context.focus;
    if (import10.checkBinding(throwOnChange,this._expr_29,currVal_29)) {
      view.renderer.setElementClass(el,'ui-inputwrapper-focus',currVal_29);
      this._expr_29 = currVal_29;
    }
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import9.AppView<any>,_eventHandler:any,emit0:boolean,emit1:boolean):void {
    this._eventHandler = _eventHandler;
    if (emit0) { (this.subscription0 = this.context.onBlur.subscribe(_eventHandler.bind(view,'onBlur'))); }
    if (emit1) { (this.subscription1 = this.context.onSelect.subscribe(_eventHandler.bind(view,'onSelect'))); }
  }
}
var renderType_Calendar_Host:import11.RenderComponentType = import10.createRenderComponentType('',0,import12.ViewEncapsulation.None,([] as any[]),{});
class View_Calendar_Host0 extends import9.AppView<any> {
  _el_0:any;
  compView_0:import9.AppView<import1.Calendar>;
  _DomHandler_0_3:import13.DomHandler;
  _Calendar_0_4:Wrapper_Calendar;
  __NG_VALUE_ACCESSOR_0_5:any[];
  constructor(viewUtils:import10.ViewUtils,parentView:import9.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_Calendar_Host0,renderType_Calendar_Host,import14.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways);
  }
  get _NG_VALUE_ACCESSOR_0_5():any[] {
    if ((this.__NG_VALUE_ACCESSOR_0_5 == null)) { (this.__NG_VALUE_ACCESSOR_0_5 = [this._Calendar_0_4.context]); }
    return this.__NG_VALUE_ACCESSOR_0_5;
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    this._el_0 = import10.selectOrCreateRenderHostElement(this.renderer,'p-calendar',import10.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this.compView_0 = new View_Calendar0(this.viewUtils,this,0,this._el_0);
    this._DomHandler_0_3 = new import13.DomHandler();
    this._Calendar_0_4 = new Wrapper_Calendar(new import17.ElementRef(this._el_0),this._DomHandler_0_3,this.renderer);
    this.compView_0.create(this._Calendar_0_4.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import16.ComponentRef_<any>(0,this,this._el_0,this._Calendar_0_4.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import13.DomHandler) && (0 === requestNodeIndex))) { return this._DomHandler_0_3; }
    if (((token === import1.Calendar) && (0 === requestNodeIndex))) { return this._Calendar_0_4.context; }
    if (((token === import18.NG_VALUE_ACCESSOR) && (0 === requestNodeIndex))) { return this._NG_VALUE_ACCESSOR_0_5; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._Calendar_0_4.ngDoCheck(this,this._el_0,throwOnChange);
    this._Calendar_0_4.checkHost(this,this.compView_0,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
    if (!throwOnChange) { if ((this.numberOfChecks === 0)) { this._Calendar_0_4.context.ngAfterViewInit(); } }
  }
  destroyInternal():void {
    this.compView_0.destroy();
    this._Calendar_0_4.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
export const CalendarNgFactory:import16.ComponentFactory<import1.Calendar> = new import16.ComponentFactory<import1.Calendar>('p-calendar',View_Calendar_Host0,import1.Calendar);
const styles_Calendar:any[] = ([] as any[]);
var Calendar_overlayState_states:any = {
  'hidden': {'opacity': '0'},
  'visible': {'opacity': '1'},
  '*': {}
}
;
function Calendar_overlayState_factory(view:import9.AppView<any>,element:any,currentState:any,nextState:any):import19.AnimationTransition {
  var previousPlayers:any = view.animationContext.getAnimationPlayers(element,((nextState == 'void')? (null as any): 'overlayState'));
  var collectedStyles:any = {};
  var player:any = (null as any);
  var totalTime:any = 0;
  var defaultStateStyles:any = Calendar_overlayState_states['*'];
  var startStateStyles:any = Calendar_overlayState_states[currentState];
  if ((startStateStyles == (null as any))) { (startStateStyles = defaultStateStyles); }
  var endStateStyles:any = Calendar_overlayState_states[nextState];
  if ((endStateStyles == (null as any))) { (endStateStyles = defaultStateStyles); }
  if (((player == (null as any)) && ((currentState == 'visible') && (nextState == 'hidden')))) {
      player = new import20.AnimationSequencePlayer([view.renderer.animate(element,new import21.AnimationStyles(import22.collectAndResolveStyles(collectedStyles,[startStateStyles])),import22.balanceAnimationKeyframes(collectedStyles,endStateStyles,[
        new import23.AnimationKeyframe(0,new import21.AnimationStyles(import22.collectAndResolveStyles(collectedStyles,[{}]))),
        new import23.AnimationKeyframe(1,new import21.AnimationStyles(import22.collectAndResolveStyles(collectedStyles,[{}])))
      ]
    ),400,0,'ease-in',previousPlayers)]);
    totalTime = 400;
  }
  if (((player == (null as any)) && ((currentState == 'hidden') && (nextState == 'visible')))) {
      player = new import20.AnimationSequencePlayer([view.renderer.animate(element,new import21.AnimationStyles(import22.collectAndResolveStyles(collectedStyles,[startStateStyles])),import22.balanceAnimationKeyframes(collectedStyles,endStateStyles,[
        new import23.AnimationKeyframe(0,new import21.AnimationStyles(import22.collectAndResolveStyles(collectedStyles,[{}]))),
        new import23.AnimationKeyframe(1,new import21.AnimationStyles(import22.collectAndResolveStyles(collectedStyles,[{}])))
      ]
    ),400,0,'ease-out',previousPlayers)]);
    totalTime = 400;
  }
  if ((player == (null as any))) { (player = new import24.NoOpAnimationPlayer()); }
  player.onDone(():void => {
    player.destroy();
    import22.renderStyles(element,view.renderer,import22.prepareFinalAnimationStyles(startStateStyles,endStateStyles));
  });
  new import20.AnimationSequencePlayer(previousPlayers).destroy();
  import22.renderStyles(element,view.renderer,import22.clearStyles(startStateStyles));
  view.animationContext.queueAnimation(element,'overlayState',player);
  return new import19.AnimationTransition(player,currentState,nextState,totalTime);
}
class View_Calendar2 extends import9.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import25.Wrapper_NgClass;
  _DomHandler_0_4:import13.DomHandler;
  _Button_0_5:import26.Wrapper_Button;
  /*private*/ _expr_4:any;
  _map_5:any;
  constructor(viewUtils:import10.ViewUtils,parentView:import9.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import27.ViewContainer) {
    super(View_Calendar2,renderType_Calendar,import14.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_4 = import8.UNINITIALIZED;
    this._map_5 = import10.pureProxy2((p0:any,p1:any):{[key: string]:any} => {
      return {
        'ui-datepicker-trigger': p0,
        'ui-state-disabled': p1
      }
      ;
    });
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    this._el_0 = import10.createRenderElement(this.renderer,(null as any),'button',new import10.InlineArray4(4,'pButton','','type','button'),(null as any));
    this._NgClass_0_3 = new import25.Wrapper_NgClass(this.parentView.parentView.parentView.injectorGet(import28.IterableDiffers,this.parentView.parentView.parentIndex),this.parentView.parentView.parentView.injectorGet(import29.KeyValueDiffers,this.parentView.parentView.parentIndex),new import17.ElementRef(this._el_0),this.renderer);
    this._DomHandler_0_4 = new import13.DomHandler();
    this._Button_0_5 = new import26.Wrapper_Button(new import17.ElementRef(this._el_0),this._DomHandler_0_4);
    var disposable_0:Function = import10.subscribeToRenderElement(this,this._el_0,new import10.InlineArray16(14,'click',(null as any),'mouseenter',(null as any),'mouseleave',(null as any),'mousedown',(null as any),'mouseup',(null as any),'focus',(null as any),'blur',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import30.NgClass) && (0 === requestNodeIndex))) { return this._NgClass_0_3.context; }
    if (((token === import13.DomHandler) && (0 === requestNodeIndex))) { return this._DomHandler_0_4; }
    if (((token === import3.Button) && (0 === requestNodeIndex))) { return this._Button_0_5.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this._map_5(true,this.parentView.parentView.context.disabled);
    this._NgClass_0_3.check_ngClass(currVal_0_0_0,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_0_1_0:any = this.parentView.parentView.context.icon;
    this._Button_0_5.check_icon(currVal_0_1_0,throwOnChange,false);
    this._Button_0_5.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_4:any = this.parentView.parentView.context.disabled;
    if (import10.checkBinding(throwOnChange,this._expr_4,currVal_4)) {
      this.renderer.setElementProperty(this._el_0,'disabled',currVal_4);
      this._expr_4 = currVal_4;
    }
    this._Button_0_5.checkHost(this,this,this._el_0,throwOnChange);
    if (!throwOnChange) { if ((this.numberOfChecks === 0)) { this._Button_0_5.context.ngAfterViewInit(); } }
  }
  destroyInternal():void {
    this._Button_0_5.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    result = (this._Button_0_5.handleEvent(eventName,$event) && result);
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.parentView.context.onButtonClick($event,(<View_Calendar1>this.parentView)._el_1)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
class View_Calendar1 extends import9.AppView<any> {
  _text_0:any;
  _el_1:any;
  _NgStyle_1_3:import31.Wrapper_NgStyle;
  _InputText_1_4:import32.Wrapper_InputText;
  _anchor_2:any;
  /*private*/ _vc_2:import27.ViewContainer;
  _TemplateRef_2_5:any;
  _NgIf_2_6:import33.Wrapper_NgIf;
  _text_3:any;
  /*private*/ _expr_9:any;
  /*private*/ _expr_10:any;
  /*private*/ _expr_11:any;
  /*private*/ _expr_12:any;
  /*private*/ _expr_13:any;
  /*private*/ _expr_14:any;
  /*private*/ _expr_15:any;
  constructor(viewUtils:import10.ViewUtils,parentView:import9.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import27.ViewContainer) {
    super(View_Calendar1,renderType_Calendar,import14.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_9 = import8.UNINITIALIZED;
    this._expr_10 = import8.UNINITIALIZED;
    this._expr_11 = import8.UNINITIALIZED;
    this._expr_12 = import8.UNINITIALIZED;
    this._expr_13 = import8.UNINITIALIZED;
    this._expr_14 = import8.UNINITIALIZED;
    this._expr_15 = import8.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    this._text_0 = this.renderer.createText((null as any),'\n                ',(null as any));
    this._el_1 = import10.createRenderElement(this.renderer,(null as any),'input',new import10.InlineArray4(4,'pInputText','','type','text'),(null as any));
    this._NgStyle_1_3 = new import31.Wrapper_NgStyle(this.parentView.parentView.injectorGet(import29.KeyValueDiffers,this.parentView.parentIndex),new import17.ElementRef(this._el_1),this.renderer);
    this._InputText_1_4 = new import32.Wrapper_InputText(new import17.ElementRef(this._el_1));
    this._anchor_2 = this.renderer.createTemplateAnchor((null as any),(null as any));
    this._vc_2 = new import27.ViewContainer(2,(null as any),this,this._anchor_2);
    this._TemplateRef_2_5 = new import34.TemplateRef_(this,2,this._anchor_2);
    this._NgIf_2_6 = new import33.Wrapper_NgIf(this._vc_2.vcRef,this._TemplateRef_2_5);
    this._text_3 = this.renderer.createText((null as any),'\n            ',(null as any));
    var disposable_0:Function = import10.subscribeToRenderElement(this,this._el_1,new import10.InlineArray16(14,'focus',(null as any),'keydown',(null as any),'click',(null as any),'blur',(null as any),'input',(null as any),'mouseover',(null as any),'mouseout',(null as any)),this.eventHandler(this.handleEvent_1));
    this.init(this._text_3,((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._anchor_2,
      this._text_3
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import35.NgStyle) && (1 === requestNodeIndex))) { return this._NgStyle_1_3.context; }
    if (((token === import4.InputText) && (1 === requestNodeIndex))) { return this._InputText_1_4.context; }
    if (((token === import34.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import36.NgIf) && (2 === requestNodeIndex))) { return this._NgIf_2_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1_0_0:any = this.parentView.context.inputStyle;
    this._NgStyle_1_3.check_ngStyle(currVal_1_0_0,throwOnChange,false);
    this._NgStyle_1_3.ngDoCheck(this,this._el_1,throwOnChange);
    this._InputText_1_4.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_2_0_0:any = this.parentView.context.showIcon;
    this._NgIf_2_6.check_ngIf(currVal_2_0_0,throwOnChange,false);
    this._NgIf_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
    const currVal_9:any = this.parentView.context.required;
    if (import10.checkBinding(throwOnChange,this._expr_9,currVal_9)) {
      this.renderer.setElementAttribute(this._el_1,'required',((currVal_9 == null)? (null as any): currVal_9.toString()));
      this._expr_9 = currVal_9;
    }
    const currVal_10:any = this.parentView.context.inputFieldValue;
    if (import10.checkBinding(throwOnChange,this._expr_10,currVal_10)) {
      this.renderer.setElementProperty(this._el_1,'value',currVal_10);
      this._expr_10 = currVal_10;
    }
    const currVal_11:any = this.parentView.context.readonlyInput;
    if (import10.checkBinding(throwOnChange,this._expr_11,currVal_11)) {
      this.renderer.setElementProperty(this._el_1,'readOnly',currVal_11);
      this._expr_11 = currVal_11;
    }
    const currVal_12:any = this.parentView.context.inputStyleClass;
    if (import10.checkBinding(throwOnChange,this._expr_12,currVal_12)) {
      this.renderer.setElementProperty(this._el_1,'className',currVal_12);
      this._expr_12 = currVal_12;
    }
    const currVal_13:any = (this.parentView.context.placeholder || '');
    if (import10.checkBinding(throwOnChange,this._expr_13,currVal_13)) {
      this.renderer.setElementProperty(this._el_1,'placeholder',currVal_13);
      this._expr_13 = currVal_13;
    }
    const currVal_14:any = this.parentView.context.disabled;
    if (import10.checkBinding(throwOnChange,this._expr_14,currVal_14)) {
      this.renderer.setElementProperty(this._el_1,'disabled',currVal_14);
      this._expr_14 = currVal_14;
    }
    const currVal_15:any = this.parentView.context.tabindex;
    if (import10.checkBinding(throwOnChange,this._expr_15,currVal_15)) {
      this.renderer.setElementAttribute(this._el_1,'tabindex',((currVal_15 == null)? (null as any): currVal_15.toString()));
      this._expr_15 = currVal_15;
    }
    this._InputText_1_4.checkHost(this,this,this._el_1,throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._text_0,ctx);
    cb(this._el_1,ctx);
    cb(this._vc_2.nativeElement,ctx);
    this._vc_2.visitNestedViewRootNodes(cb,ctx);
    cb(this._text_3,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import9.AppView<any> {
    if ((nodeIndex == 2)) { return new View_Calendar2(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    return (null as any);
  }
  handleEvent_1(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    result = (this._InputText_1_4.handleEvent(eventName,$event) && result);
    if ((eventName == 'focus')) {
      const pd_sub_0:any = ((<any>this.parentView.context.onInputFocus($event)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'keydown')) {
      const pd_sub_1:any = ((<any>this.parentView.context.onInputKeydown($event)) !== false);
      result = (pd_sub_1 && result);
    }
    if ((eventName == 'click')) {
      const pd_sub_2:any = ((<any>(this.parentView.context.closeOverlay = false)) !== false);
      result = (pd_sub_2 && result);
    }
    if ((eventName == 'blur')) {
      const pd_sub_3:any = ((<any>this.parentView.context.onInputBlur($event)) !== false);
      result = (pd_sub_3 && result);
    }
    if ((eventName == 'input')) {
      const pd_sub_4:any = ((<any>this.parentView.context.onInput($event)) !== false);
      result = (pd_sub_4 && result);
    }
    return result;
  }
}
class View_Calendar4 extends import9.AppView<any> {
  _el_0:any;
  _text_1:any;
  /*private*/ _expr_2:any;
  constructor(viewUtils:import10.ViewUtils,parentView:import9.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import27.ViewContainer) {
    super(View_Calendar4,renderType_Calendar,import14.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_2 = import8.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    this._el_0 = import10.createRenderElement(this.renderer,(null as any),'span',new import10.InlineArray2(2,'class','ui-datepicker-month'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1
    ]
    ),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2:any = import10.inlineInterpolate(1,'',this.parentView.parentView.context.currentMonthText,'');
    if (import10.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setText(this._text_1,currVal_2);
      this._expr_2 = currVal_2;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_Calendar6 extends import9.AppView<any> {
  _el_0:any;
  _text_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  constructor(viewUtils:import10.ViewUtils,parentView:import9.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import27.ViewContainer) {
    super(View_Calendar6,renderType_Calendar,import14.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_2 = import8.UNINITIALIZED;
    this._expr_3 = import8.UNINITIALIZED;
    this._expr_4 = import8.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    this._el_0 = import10.createRenderElement(this.renderer,(null as any),'option',import10.EMPTY_INLINE_ARRAY,(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1
    ]
    ),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2:any = this.context.index;
    if (import10.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setElementProperty(this._el_0,'value',currVal_2);
      this._expr_2 = currVal_2;
    }
    const currVal_3:any = (this.context.index == this.parentView.parentView.parentView.context.currentMonth);
    if (import10.checkBinding(throwOnChange,this._expr_3,currVal_3)) {
      this.renderer.setElementProperty(this._el_0,'selected',currVal_3);
      this._expr_3 = currVal_3;
    }
    const currVal_4:any = import10.inlineInterpolate(1,'',this.context.$implicit,'');
    if (import10.checkBinding(throwOnChange,this._expr_4,currVal_4)) {
      this.renderer.setText(this._text_1,currVal_4);
      this._expr_4 = currVal_4;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_Calendar5 extends import9.AppView<any> {
  _el_0:any;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import27.ViewContainer;
  _TemplateRef_2_5:any;
  _NgFor_2_6:import37.Wrapper_NgFor;
  _text_3:any;
  constructor(viewUtils:import10.ViewUtils,parentView:import9.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import27.ViewContainer) {
    super(View_Calendar5,renderType_Calendar,import14.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    this._el_0 = import10.createRenderElement(this.renderer,(null as any),'select',new import10.InlineArray2(2,'class','ui-datepicker-month'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                            ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import27.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import34.TemplateRef_(this,2,this._anchor_2);
    this._NgFor_2_6 = new import37.Wrapper_NgFor(this._vc_2.vcRef,this._TemplateRef_2_5,this.parentView.parentView.parentView.injectorGet(import28.IterableDiffers,this.parentView.parentView.parentIndex),this.parentView.parentView.ref);
    this._text_3 = this.renderer.createText(this._el_0,'\n                        ',(null as any));
    var disposable_0:Function = import10.subscribeToRenderElement(this,this._el_0,new import10.InlineArray2(2,'change',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import34.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import38.NgFor) && (2 === requestNodeIndex))) { return this._NgFor_2_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2_0_0:any = this.parentView.parentView.context.locale.monthNames;
    this._NgFor_2_6.check_ngForOf(currVal_2_0_0,throwOnChange,false);
    this._NgFor_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import9.AppView<any> {
    if ((nodeIndex == 2)) { return new View_Calendar6(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    return (null as any);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'change')) {
      const pd_sub_0:any = ((<any>this.parentView.parentView.context.onMonthDropdownChange($event.target.value)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
class View_Calendar8 extends import9.AppView<any> {
  _el_0:any;
  _text_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  constructor(viewUtils:import10.ViewUtils,parentView:import9.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import27.ViewContainer) {
    super(View_Calendar8,renderType_Calendar,import14.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_2 = import8.UNINITIALIZED;
    this._expr_3 = import8.UNINITIALIZED;
    this._expr_4 = import8.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    this._el_0 = import10.createRenderElement(this.renderer,(null as any),'option',import10.EMPTY_INLINE_ARRAY,(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1
    ]
    ),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2:any = this.context.$implicit;
    if (import10.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setElementProperty(this._el_0,'value',currVal_2);
      this._expr_2 = currVal_2;
    }
    const currVal_3:any = (this.context.$implicit == this.parentView.parentView.parentView.context.currentYear);
    if (import10.checkBinding(throwOnChange,this._expr_3,currVal_3)) {
      this.renderer.setElementProperty(this._el_0,'selected',currVal_3);
      this._expr_3 = currVal_3;
    }
    const currVal_4:any = import10.inlineInterpolate(1,'',this.context.$implicit,'');
    if (import10.checkBinding(throwOnChange,this._expr_4,currVal_4)) {
      this.renderer.setText(this._text_1,currVal_4);
      this._expr_4 = currVal_4;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_Calendar7 extends import9.AppView<any> {
  _el_0:any;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import27.ViewContainer;
  _TemplateRef_2_5:any;
  _NgFor_2_6:import37.Wrapper_NgFor;
  _text_3:any;
  constructor(viewUtils:import10.ViewUtils,parentView:import9.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import27.ViewContainer) {
    super(View_Calendar7,renderType_Calendar,import14.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    this._el_0 = import10.createRenderElement(this.renderer,(null as any),'select',new import10.InlineArray2(2,'class','ui-datepicker-year'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                            ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import27.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import34.TemplateRef_(this,2,this._anchor_2);
    this._NgFor_2_6 = new import37.Wrapper_NgFor(this._vc_2.vcRef,this._TemplateRef_2_5,this.parentView.parentView.parentView.injectorGet(import28.IterableDiffers,this.parentView.parentView.parentIndex),this.parentView.parentView.ref);
    this._text_3 = this.renderer.createText(this._el_0,'\n                        ',(null as any));
    var disposable_0:Function = import10.subscribeToRenderElement(this,this._el_0,new import10.InlineArray2(2,'change',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import34.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import38.NgFor) && (2 === requestNodeIndex))) { return this._NgFor_2_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2_0_0:any = this.parentView.parentView.context.yearOptions;
    this._NgFor_2_6.check_ngForOf(currVal_2_0_0,throwOnChange,false);
    this._NgFor_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import9.AppView<any> {
    if ((nodeIndex == 2)) { return new View_Calendar8(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    return (null as any);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'change')) {
      const pd_sub_0:any = ((<any>this.parentView.parentView.context.onYearDropdownChange($event.target.value)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
class View_Calendar9 extends import9.AppView<any> {
  _el_0:any;
  _text_1:any;
  /*private*/ _expr_2:any;
  constructor(viewUtils:import10.ViewUtils,parentView:import9.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import27.ViewContainer) {
    super(View_Calendar9,renderType_Calendar,import14.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_2 = import8.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    this._el_0 = import10.createRenderElement(this.renderer,(null as any),'span',new import10.InlineArray2(2,'class','ui-datepicker-year'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1
    ]
    ),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2:any = import10.inlineInterpolate(1,'',this.parentView.parentView.context.currentYear,'');
    if (import10.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setText(this._text_1,currVal_2);
      this._expr_2 = currVal_2;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_Calendar3 extends import9.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _NgClass_2_3:import25.Wrapper_NgClass;
  _text_3:any;
  _el_4:any;
  _text_5:any;
  _text_6:any;
  _el_7:any;
  _NgClass_7_3:import25.Wrapper_NgClass;
  _text_8:any;
  _el_9:any;
  _text_10:any;
  _text_11:any;
  _el_12:any;
  _text_13:any;
  _anchor_14:any;
  /*private*/ _vc_14:import27.ViewContainer;
  _TemplateRef_14_5:any;
  _NgIf_14_6:import33.Wrapper_NgIf;
  _text_15:any;
  _anchor_16:any;
  /*private*/ _vc_16:import27.ViewContainer;
  _TemplateRef_16_5:any;
  _NgIf_16_6:import33.Wrapper_NgIf;
  _text_17:any;
  _anchor_18:any;
  /*private*/ _vc_18:import27.ViewContainer;
  _TemplateRef_18_5:any;
  _NgIf_18_6:import33.Wrapper_NgIf;
  _text_19:any;
  _anchor_20:any;
  /*private*/ _vc_20:import27.ViewContainer;
  _TemplateRef_20_5:any;
  _NgIf_20_6:import33.Wrapper_NgIf;
  _text_21:any;
  _text_22:any;
  _map_37:any;
  _map_38:any;
  constructor(viewUtils:import10.ViewUtils,parentView:import9.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import27.ViewContainer) {
    super(View_Calendar3,renderType_Calendar,import14.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_37 = import10.pureProxy1((p0:any):{[key: string]:any} => {
      return {'ui-state-hover ui-datepicker-prev-hover': p0};
    });
    this._map_38 = import10.pureProxy1((p0:any):{[key: string]:any} => {
      return {'ui-state-hover ui-datepicker-next-hover': p0};
    });
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    this._el_0 = import10.createRenderElement(this.renderer,(null as any),'div',new import10.InlineArray2(2,'class','ui-datepicker-header ui-widget-header ui-helper-clearfix ui-corner-all'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this._el_2 = import10.createRenderElement(this.renderer,this._el_0,'a',new import10.InlineArray4(4,'class','ui-datepicker-prev ui-corner-all','href','#'),(null as any));
    this._NgClass_2_3 = new import25.Wrapper_NgClass(this.parentView.parentView.injectorGet(import28.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import29.KeyValueDiffers,this.parentView.parentIndex),new import17.ElementRef(this._el_2),this.renderer);
    this._text_3 = this.renderer.createText(this._el_2,'\n                        ',(null as any));
    this._el_4 = import10.createRenderElement(this.renderer,this._el_2,'span',new import10.InlineArray2(2,'class','fa fa-angle-left'),(null as any));
    this._text_5 = this.renderer.createText(this._el_2,'\n                    ',(null as any));
    this._text_6 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this._el_7 = import10.createRenderElement(this.renderer,this._el_0,'a',new import10.InlineArray4(4,'class','ui-datepicker-next ui-corner-all','href','#'),(null as any));
    this._NgClass_7_3 = new import25.Wrapper_NgClass(this.parentView.parentView.injectorGet(import28.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import29.KeyValueDiffers,this.parentView.parentIndex),new import17.ElementRef(this._el_7),this.renderer);
    this._text_8 = this.renderer.createText(this._el_7,'\n                        ',(null as any));
    this._el_9 = import10.createRenderElement(this.renderer,this._el_7,'span',new import10.InlineArray2(2,'class','fa fa-angle-right'),(null as any));
    this._text_10 = this.renderer.createText(this._el_7,'\n                    ',(null as any));
    this._text_11 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this._el_12 = import10.createRenderElement(this.renderer,this._el_0,'div',new import10.InlineArray2(2,'class','ui-datepicker-title'),(null as any));
    this._text_13 = this.renderer.createText(this._el_12,'\n                        ',(null as any));
    this._anchor_14 = this.renderer.createTemplateAnchor(this._el_12,(null as any));
    this._vc_14 = new import27.ViewContainer(14,12,this,this._anchor_14);
    this._TemplateRef_14_5 = new import34.TemplateRef_(this,14,this._anchor_14);
    this._NgIf_14_6 = new import33.Wrapper_NgIf(this._vc_14.vcRef,this._TemplateRef_14_5);
    this._text_15 = this.renderer.createText(this._el_12,'\n                        ',(null as any));
    this._anchor_16 = this.renderer.createTemplateAnchor(this._el_12,(null as any));
    this._vc_16 = new import27.ViewContainer(16,12,this,this._anchor_16);
    this._TemplateRef_16_5 = new import34.TemplateRef_(this,16,this._anchor_16);
    this._NgIf_16_6 = new import33.Wrapper_NgIf(this._vc_16.vcRef,this._TemplateRef_16_5);
    this._text_17 = this.renderer.createText(this._el_12,'\n                        ',(null as any));
    this._anchor_18 = this.renderer.createTemplateAnchor(this._el_12,(null as any));
    this._vc_18 = new import27.ViewContainer(18,12,this,this._anchor_18);
    this._TemplateRef_18_5 = new import34.TemplateRef_(this,18,this._anchor_18);
    this._NgIf_18_6 = new import33.Wrapper_NgIf(this._vc_18.vcRef,this._TemplateRef_18_5);
    this._text_19 = this.renderer.createText(this._el_12,'\n                        ',(null as any));
    this._anchor_20 = this.renderer.createTemplateAnchor(this._el_12,(null as any));
    this._vc_20 = new import27.ViewContainer(20,12,this,this._anchor_20);
    this._TemplateRef_20_5 = new import34.TemplateRef_(this,20,this._anchor_20);
    this._NgIf_20_6 = new import33.Wrapper_NgIf(this._vc_20.vcRef,this._TemplateRef_20_5);
    this._text_21 = this.renderer.createText(this._el_12,'\n                    ',(null as any));
    this._text_22 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    var disposable_0:Function = import10.subscribeToRenderElement(this,this._el_2,new import10.InlineArray8(6,'click',(null as any),'mouseenter',(null as any),'mouseleave',(null as any)),this.eventHandler(this.handleEvent_2));
    var disposable_1:Function = import10.subscribeToRenderElement(this,this._el_7,new import10.InlineArray8(6,'click',(null as any),'mouseenter',(null as any),'mouseleave',(null as any)),this.eventHandler(this.handleEvent_7));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._el_4,
      this._text_5,
      this._text_6,
      this._el_7,
      this._text_8,
      this._el_9,
      this._text_10,
      this._text_11,
      this._el_12,
      this._text_13,
      this._anchor_14,
      this._text_15,
      this._anchor_16,
      this._text_17,
      this._anchor_18,
      this._text_19,
      this._anchor_20,
      this._text_21,
      this._text_22
    ]
    ),[
      disposable_0,
      disposable_1
    ]
    );
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import30.NgClass) && ((2 <= requestNodeIndex) && (requestNodeIndex <= 5)))) { return this._NgClass_2_3.context; }
    if (((token === import30.NgClass) && ((7 <= requestNodeIndex) && (requestNodeIndex <= 10)))) { return this._NgClass_7_3.context; }
    if (((token === import34.TemplateRef) && (14 === requestNodeIndex))) { return this._TemplateRef_14_5; }
    if (((token === import36.NgIf) && (14 === requestNodeIndex))) { return this._NgIf_14_6.context; }
    if (((token === import34.TemplateRef) && (16 === requestNodeIndex))) { return this._TemplateRef_16_5; }
    if (((token === import36.NgIf) && (16 === requestNodeIndex))) { return this._NgIf_16_6.context; }
    if (((token === import34.TemplateRef) && (18 === requestNodeIndex))) { return this._TemplateRef_18_5; }
    if (((token === import36.NgIf) && (18 === requestNodeIndex))) { return this._NgIf_18_6.context; }
    if (((token === import34.TemplateRef) && (20 === requestNodeIndex))) { return this._TemplateRef_20_5; }
    if (((token === import36.NgIf) && (20 === requestNodeIndex))) { return this._NgIf_20_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2_0_0:any = 'ui-datepicker-prev ui-corner-all';
    this._NgClass_2_3.check_klass(currVal_2_0_0,throwOnChange,false);
    const currVal_2_0_1:any = this._map_37((this.parentView.context.hoverPrev && !this.parentView.context.disabled));
    this._NgClass_2_3.check_ngClass(currVal_2_0_1,throwOnChange,false);
    this._NgClass_2_3.ngDoCheck(this,this._el_2,throwOnChange);
    const currVal_7_0_0:any = 'ui-datepicker-next ui-corner-all';
    this._NgClass_7_3.check_klass(currVal_7_0_0,throwOnChange,false);
    const currVal_7_0_1:any = this._map_38((this.parentView.context.hoverNext && !this.parentView.context.disabled));
    this._NgClass_7_3.check_ngClass(currVal_7_0_1,throwOnChange,false);
    this._NgClass_7_3.ngDoCheck(this,this._el_7,throwOnChange);
    const currVal_14_0_0:boolean = !this.parentView.context.monthNavigator;
    this._NgIf_14_6.check_ngIf(currVal_14_0_0,throwOnChange,false);
    this._NgIf_14_6.ngDoCheck(this,this._anchor_14,throwOnChange);
    const currVal_16_0_0:any = this.parentView.context.monthNavigator;
    this._NgIf_16_6.check_ngIf(currVal_16_0_0,throwOnChange,false);
    this._NgIf_16_6.ngDoCheck(this,this._anchor_16,throwOnChange);
    const currVal_18_0_0:any = this.parentView.context.yearNavigator;
    this._NgIf_18_6.check_ngIf(currVal_18_0_0,throwOnChange,false);
    this._NgIf_18_6.ngDoCheck(this,this._anchor_18,throwOnChange);
    const currVal_20_0_0:boolean = !this.parentView.context.yearNavigator;
    this._NgIf_20_6.check_ngIf(currVal_20_0_0,throwOnChange,false);
    this._NgIf_20_6.ngDoCheck(this,this._anchor_20,throwOnChange);
    this._vc_14.detectChangesInNestedViews(throwOnChange);
    this._vc_16.detectChangesInNestedViews(throwOnChange);
    this._vc_18.detectChangesInNestedViews(throwOnChange);
    this._vc_20.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_14.destroyNestedViews();
    this._vc_16.destroyNestedViews();
    this._vc_18.destroyNestedViews();
    this._vc_20.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import9.AppView<any> {
    if ((nodeIndex == 14)) { return new View_Calendar4(this.viewUtils,this,14,this._anchor_14,this._vc_14); }
    if ((nodeIndex == 16)) { return new View_Calendar5(this.viewUtils,this,16,this._anchor_16,this._vc_16); }
    if ((nodeIndex == 18)) { return new View_Calendar7(this.viewUtils,this,18,this._anchor_18,this._vc_18); }
    if ((nodeIndex == 20)) { return new View_Calendar9(this.viewUtils,this,20,this._anchor_20,this._vc_20); }
    return (null as any);
  }
  handleEvent_2(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.context.prevMonth($event)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'mouseenter')) {
      const pd_sub_1:any = ((<any>(this.parentView.context.hoverPrev = true)) !== false);
      result = (pd_sub_1 && result);
    }
    if ((eventName == 'mouseleave')) {
      const pd_sub_2:any = ((<any>(this.parentView.context.hoverPrev = false)) !== false);
      result = (pd_sub_2 && result);
    }
    return result;
  }
  handleEvent_7(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.context.nextMonth($event)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'mouseenter')) {
      const pd_sub_1:any = ((<any>(this.parentView.context.hoverNext = true)) !== false);
      result = (pd_sub_1 && result);
    }
    if ((eventName == 'mouseleave')) {
      const pd_sub_2:any = ((<any>(this.parentView.context.hoverNext = false)) !== false);
      result = (pd_sub_2 && result);
    }
    return result;
  }
}
class View_Calendar11 extends import9.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _text_4:any;
  /*private*/ _expr_5:any;
  constructor(viewUtils:import10.ViewUtils,parentView:import9.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import27.ViewContainer) {
    super(View_Calendar11,renderType_Calendar,import14.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_5 = import8.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    this._el_0 = import10.createRenderElement(this.renderer,(null as any),'th',new import10.InlineArray2(2,'scope','col'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                                ',(null as any));
    this._el_2 = import10.createRenderElement(this.renderer,this._el_0,'span',import10.EMPTY_INLINE_ARRAY,(null as any));
    this._text_3 = this.renderer.createText(this._el_2,'',(null as any));
    this._text_4 = this.renderer.createText(this._el_0,'\n                            ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._text_4
    ]
    ),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_5:any = import10.inlineInterpolate(1,'',this.context.$implicit,'');
    if (import10.checkBinding(throwOnChange,this._expr_5,currVal_5)) {
      this.renderer.setText(this._text_3,currVal_5);
      this._expr_5 = currVal_5;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_Calendar14 extends import9.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import25.Wrapper_NgClass;
  _text_1:any;
  _map_3:any;
  /*private*/ _expr_4:any;
  constructor(viewUtils:import10.ViewUtils,parentView:import9.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import27.ViewContainer) {
    super(View_Calendar14,renderType_Calendar,import14.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_3 = import10.pureProxy4((p0:any,p1:any,p2:any,p3:any):{[key: string]:any} => {
      return {
        'ui-state-active': p0,
        'ui-state-hover': p1,
        'ui-state-highlight': p2,
        'ui-state-disabled': p3
      }
      ;
    });
    this._expr_4 = import8.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    this._el_0 = import10.createRenderElement(this.renderer,(null as any),'a',new import10.InlineArray4(4,'class','ui-state-default','href','#'),(null as any));
    this._NgClass_0_3 = new import25.Wrapper_NgClass(this.parentView.parentView.parentView.parentView.parentView.injectorGet(import28.IterableDiffers,this.parentView.parentView.parentView.parentView.parentIndex),this.parentView.parentView.parentView.parentView.parentView.injectorGet(import29.KeyValueDiffers,this.parentView.parentView.parentView.parentView.parentIndex),new import17.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'',(null as any));
    var disposable_0:Function = import10.subscribeToRenderElement(this,this._el_0,new import10.InlineArray8(6,'click',(null as any),'mouseenter',(null as any),'mouseleave',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import30.NgClass) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 1)))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = 'ui-state-default';
    this._NgClass_0_3.check_klass(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = this._map_3(this.parentView.parentView.parentView.parentView.context.isSelected(this.parentView.context.$implicit),(((this.parentView.parentView.parentView.parentView.context.hoverCell == this._el_0) && !this.parentView.parentView.parentView.parentView.context.disabled) && this.parentView.context.$implicit.selectable),this.parentView.parentView.parentView.parentView.context.isToday(this.parentView.context.$implicit),!this.parentView.context.$implicit.selectable);
    this._NgClass_0_3.check_ngClass(currVal_0_0_1,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_4:any = import10.inlineInterpolate(1,'',this.parentView.context.$implicit.day,'');
    if (import10.checkBinding(throwOnChange,this._expr_4,currVal_4)) {
      this.renderer.setText(this._text_1,currVal_4);
      this._expr_4 = currVal_4;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.parentView.parentView.parentView.context.onDateSelect($event,this.parentView.context.$implicit)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'mouseenter')) {
      const pd_sub_1:any = ((<any>(this.parentView.parentView.parentView.parentView.context.hoverCell = this._el_0)) !== false);
      result = (pd_sub_1 && result);
    }
    if ((eventName == 'mouseleave')) {
      const pd_sub_2:any = ((<any>(this.parentView.parentView.parentView.parentView.context.hoverCell = (null as any))) !== false);
      result = (pd_sub_2 && result);
    }
    return result;
  }
}
class View_Calendar13 extends import9.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import25.Wrapper_NgClass;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import27.ViewContainer;
  _TemplateRef_2_5:any;
  _NgIf_2_6:import33.Wrapper_NgIf;
  _text_3:any;
  _map_8:any;
  constructor(viewUtils:import10.ViewUtils,parentView:import9.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import27.ViewContainer) {
    super(View_Calendar13,renderType_Calendar,import14.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_8 = import10.pureProxy3((p0:any,p1:any,p2:any):{[key: string]:any} => {
      return {
        'ui-datepicker-other-month ui-state-disabled': p0,
        'ui-datepicker-current-day': p1,
        'ui-datepicker-today': p2
      }
      ;
    });
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    this._el_0 = import10.createRenderElement(this.renderer,(null as any),'td',import10.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_0_3 = new import25.Wrapper_NgClass(this.parentView.parentView.parentView.parentView.injectorGet(import28.IterableDiffers,this.parentView.parentView.parentView.parentIndex),this.parentView.parentView.parentView.parentView.injectorGet(import29.KeyValueDiffers,this.parentView.parentView.parentView.parentIndex),new import17.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n                                ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import27.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import34.TemplateRef_(this,2,this._anchor_2);
    this._NgIf_2_6 = new import33.Wrapper_NgIf(this._vc_2.vcRef,this._TemplateRef_2_5);
    this._text_3 = this.renderer.createText(this._el_0,'\n                            ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import34.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import36.NgIf) && (2 === requestNodeIndex))) { return this._NgIf_2_6.context; }
    if (((token === import30.NgClass) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 3)))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this._map_8(this.context.$implicit.otherMonth,this.parentView.parentView.parentView.context.isSelected(this.context.$implicit),this.parentView.parentView.parentView.context.isToday(this.context.$implicit));
    this._NgClass_0_3.check_ngClass(currVal_0_0_0,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_2_0_0:any = (this.context.$implicit.otherMonth? this.parentView.parentView.parentView.context.showOtherMonths: true);
    this._NgIf_2_6.check_ngIf(currVal_2_0_0,throwOnChange,false);
    this._NgIf_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import9.AppView<any> {
    if ((nodeIndex == 2)) { return new View_Calendar14(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    return (null as any);
  }
}
class View_Calendar12 extends import9.AppView<any> {
  _el_0:any;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import27.ViewContainer;
  _TemplateRef_2_5:any;
  _NgFor_2_6:import37.Wrapper_NgFor;
  _text_3:any;
  constructor(viewUtils:import10.ViewUtils,parentView:import9.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import27.ViewContainer) {
    super(View_Calendar12,renderType_Calendar,import14.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    this._el_0 = import10.createRenderElement(this.renderer,(null as any),'tr',import10.EMPTY_INLINE_ARRAY,(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                            ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import27.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import34.TemplateRef_(this,2,this._anchor_2);
    this._NgFor_2_6 = new import37.Wrapper_NgFor(this._vc_2.vcRef,this._TemplateRef_2_5,this.parentView.parentView.parentView.injectorGet(import28.IterableDiffers,this.parentView.parentView.parentIndex),this.parentView.parentView.ref);
    this._text_3 = this.renderer.createText(this._el_0,'\n                        ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import34.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import38.NgFor) && (2 === requestNodeIndex))) { return this._NgFor_2_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2_0_0:any = this.context.$implicit;
    this._NgFor_2_6.check_ngForOf(currVal_2_0_0,throwOnChange,false);
    this._NgFor_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import9.AppView<any> {
    if ((nodeIndex == 2)) { return new View_Calendar13(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    return (null as any);
  }
}
class View_Calendar10 extends import9.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _el_4:any;
  _text_5:any;
  _anchor_6:any;
  /*private*/ _vc_6:import27.ViewContainer;
  _TemplateRef_6_5:any;
  _NgFor_6_6:import37.Wrapper_NgFor;
  _text_7:any;
  _text_8:any;
  _text_9:any;
  _el_10:any;
  _text_11:any;
  _anchor_12:any;
  /*private*/ _vc_12:import27.ViewContainer;
  _TemplateRef_12_5:any;
  _NgFor_12_6:import37.Wrapper_NgFor;
  _text_13:any;
  _text_14:any;
  constructor(viewUtils:import10.ViewUtils,parentView:import9.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import27.ViewContainer) {
    super(View_Calendar10,renderType_Calendar,import14.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    this._el_0 = import10.createRenderElement(this.renderer,(null as any),'table',new import10.InlineArray2(2,'class','ui-datepicker-calendar'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this._el_2 = import10.createRenderElement(this.renderer,this._el_0,'thead',import10.EMPTY_INLINE_ARRAY,(null as any));
    this._text_3 = this.renderer.createText(this._el_2,'\n                        ',(null as any));
    this._el_4 = import10.createRenderElement(this.renderer,this._el_2,'tr',import10.EMPTY_INLINE_ARRAY,(null as any));
    this._text_5 = this.renderer.createText(this._el_4,'\n                            ',(null as any));
    this._anchor_6 = this.renderer.createTemplateAnchor(this._el_4,(null as any));
    this._vc_6 = new import27.ViewContainer(6,4,this,this._anchor_6);
    this._TemplateRef_6_5 = new import34.TemplateRef_(this,6,this._anchor_6);
    this._NgFor_6_6 = new import37.Wrapper_NgFor(this._vc_6.vcRef,this._TemplateRef_6_5,this.parentView.parentView.injectorGet(import28.IterableDiffers,this.parentView.parentIndex),this.parentView.ref);
    this._text_7 = this.renderer.createText(this._el_4,'\n                        ',(null as any));
    this._text_8 = this.renderer.createText(this._el_2,'\n                    ',(null as any));
    this._text_9 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this._el_10 = import10.createRenderElement(this.renderer,this._el_0,'tbody',import10.EMPTY_INLINE_ARRAY,(null as any));
    this._text_11 = this.renderer.createText(this._el_10,'\n                        ',(null as any));
    this._anchor_12 = this.renderer.createTemplateAnchor(this._el_10,(null as any));
    this._vc_12 = new import27.ViewContainer(12,10,this,this._anchor_12);
    this._TemplateRef_12_5 = new import34.TemplateRef_(this,12,this._anchor_12);
    this._NgFor_12_6 = new import37.Wrapper_NgFor(this._vc_12.vcRef,this._TemplateRef_12_5,this.parentView.parentView.injectorGet(import28.IterableDiffers,this.parentView.parentIndex),this.parentView.ref);
    this._text_13 = this.renderer.createText(this._el_10,'\n                    ',(null as any));
    this._text_14 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._el_4,
      this._text_5,
      this._anchor_6,
      this._text_7,
      this._text_8,
      this._text_9,
      this._el_10,
      this._text_11,
      this._anchor_12,
      this._text_13,
      this._text_14
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import34.TemplateRef) && (6 === requestNodeIndex))) { return this._TemplateRef_6_5; }
    if (((token === import38.NgFor) && (6 === requestNodeIndex))) { return this._NgFor_6_6.context; }
    if (((token === import34.TemplateRef) && (12 === requestNodeIndex))) { return this._TemplateRef_12_5; }
    if (((token === import38.NgFor) && (12 === requestNodeIndex))) { return this._NgFor_12_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_6_0_0:any = this.parentView.context.weekDays;
    this._NgFor_6_6.check_ngForOf(currVal_6_0_0,throwOnChange,false);
    this._NgFor_6_6.ngDoCheck(this,this._anchor_6,throwOnChange);
    const currVal_12_0_0:any = this.parentView.context.dates;
    this._NgFor_12_6.check_ngForOf(currVal_12_0_0,throwOnChange,false);
    this._NgFor_12_6.ngDoCheck(this,this._anchor_12,throwOnChange);
    this._vc_6.detectChangesInNestedViews(throwOnChange);
    this._vc_12.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_6.destroyNestedViews();
    this._vc_12.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import9.AppView<any> {
    if ((nodeIndex == 6)) { return new View_Calendar11(this.viewUtils,this,6,this._anchor_6,this._vc_6); }
    if ((nodeIndex == 12)) { return new View_Calendar12(this.viewUtils,this,12,this._anchor_12,this._vc_12); }
    return (null as any);
  }
}
class View_Calendar16 extends import9.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _el_4:any;
  _text_5:any;
  _text_6:any;
  _el_7:any;
  _text_8:any;
  _text_9:any;
  _el_10:any;
  _text_11:any;
  _el_12:any;
  _text_13:any;
  _text_14:any;
  /*private*/ _expr_15:any;
  constructor(viewUtils:import10.ViewUtils,parentView:import9.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import27.ViewContainer) {
    super(View_Calendar16,renderType_Calendar,import14.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_15 = import8.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    this._el_0 = import10.createRenderElement(this.renderer,(null as any),'div',new import10.InlineArray2(2,'class','ui-ampm-picker'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                        ',(null as any));
    this._el_2 = import10.createRenderElement(this.renderer,this._el_0,'a',new import10.InlineArray2(2,'href','#'),(null as any));
    this._text_3 = this.renderer.createText(this._el_2,'\n                            ',(null as any));
    this._el_4 = import10.createRenderElement(this.renderer,this._el_2,'span',new import10.InlineArray2(2,'class','fa fa-angle-up'),(null as any));
    this._text_5 = this.renderer.createText(this._el_2,'\n                        ',(null as any));
    this._text_6 = this.renderer.createText(this._el_0,'\n                        ',(null as any));
    this._el_7 = import10.createRenderElement(this.renderer,this._el_0,'span',import10.EMPTY_INLINE_ARRAY,(null as any));
    this._text_8 = this.renderer.createText(this._el_7,'',(null as any));
    this._text_9 = this.renderer.createText(this._el_0,'\n                        ',(null as any));
    this._el_10 = import10.createRenderElement(this.renderer,this._el_0,'a',new import10.InlineArray2(2,'href','#'),(null as any));
    this._text_11 = this.renderer.createText(this._el_10,'\n                            ',(null as any));
    this._el_12 = import10.createRenderElement(this.renderer,this._el_10,'span',new import10.InlineArray2(2,'class','fa fa-angle-down'),(null as any));
    this._text_13 = this.renderer.createText(this._el_10,'\n                        ',(null as any));
    this._text_14 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    var disposable_0:Function = import10.subscribeToRenderElement(this,this._el_2,new import10.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_2));
    var disposable_1:Function = import10.subscribeToRenderElement(this,this._el_10,new import10.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_10));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._el_4,
      this._text_5,
      this._text_6,
      this._el_7,
      this._text_8,
      this._text_9,
      this._el_10,
      this._text_11,
      this._el_12,
      this._text_13,
      this._text_14
    ]
    ),[
      disposable_0,
      disposable_1
    ]
    );
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_15:any = import10.inlineInterpolate(1,'',(this.parentView.parentView.context.pm? 'PM': 'AM'),'');
    if (import10.checkBinding(throwOnChange,this._expr_15,currVal_15)) {
      this.renderer.setText(this._text_8,currVal_15);
      this._expr_15 = currVal_15;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  handleEvent_2(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.parentView.context.toggleAMPM($event)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
  handleEvent_10(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.parentView.context.toggleAMPM($event)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
class View_Calendar15 extends import9.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _el_4:any;
  _text_5:any;
  _el_6:any;
  _text_7:any;
  _text_8:any;
  _el_9:any;
  _NgStyle_9_3:import31.Wrapper_NgStyle;
  _text_10:any;
  _el_11:any;
  _text_12:any;
  _text_13:any;
  _el_14:any;
  _text_15:any;
  _el_16:any;
  _text_17:any;
  _text_18:any;
  _text_19:any;
  _el_20:any;
  _text_21:any;
  _el_22:any;
  _text_23:any;
  _el_24:any;
  _text_25:any;
  _text_26:any;
  _el_27:any;
  _text_28:any;
  _text_29:any;
  _el_30:any;
  _text_31:any;
  _el_32:any;
  _text_33:any;
  _text_34:any;
  _text_35:any;
  _el_36:any;
  _text_37:any;
  _el_38:any;
  _text_39:any;
  _el_40:any;
  _text_41:any;
  _text_42:any;
  _el_43:any;
  _NgStyle_43_3:import31.Wrapper_NgStyle;
  _text_44:any;
  _el_45:any;
  _text_46:any;
  _text_47:any;
  _el_48:any;
  _text_49:any;
  _el_50:any;
  _text_51:any;
  _text_52:any;
  _text_53:any;
  _anchor_54:any;
  /*private*/ _vc_54:import27.ViewContainer;
  _TemplateRef_54_5:any;
  _NgIf_54_6:import33.Wrapper_NgIf;
  _text_55:any;
  _map_61:any;
  /*private*/ _expr_62:any;
  _map_63:any;
  /*private*/ _expr_64:any;
  constructor(viewUtils:import10.ViewUtils,parentView:import9.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import27.ViewContainer) {
    super(View_Calendar15,renderType_Calendar,import14.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_61 = import10.pureProxy1((p0:any):{[key: string]:any} => {
      return {display: p0};
    });
    this._expr_62 = import8.UNINITIALIZED;
    this._map_63 = import10.pureProxy1((p0:any):{[key: string]:any} => {
      return {display: p0};
    });
    this._expr_64 = import8.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    this._el_0 = import10.createRenderElement(this.renderer,(null as any),'div',new import10.InlineArray2(2,'class','ui-timepicker ui-widget-header ui-corner-all'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this._el_2 = import10.createRenderElement(this.renderer,this._el_0,'div',new import10.InlineArray2(2,'class','ui-hour-picker'),(null as any));
    this._text_3 = this.renderer.createText(this._el_2,'\n                        ',(null as any));
    this._el_4 = import10.createRenderElement(this.renderer,this._el_2,'a',new import10.InlineArray2(2,'href','#'),(null as any));
    this._text_5 = this.renderer.createText(this._el_4,'\n                            ',(null as any));
    this._el_6 = import10.createRenderElement(this.renderer,this._el_4,'span',new import10.InlineArray2(2,'class','fa fa-angle-up'),(null as any));
    this._text_7 = this.renderer.createText(this._el_4,'\n                        ',(null as any));
    this._text_8 = this.renderer.createText(this._el_2,'\n                        ',(null as any));
    this._el_9 = import10.createRenderElement(this.renderer,this._el_2,'span',import10.EMPTY_INLINE_ARRAY,(null as any));
    this._NgStyle_9_3 = new import31.Wrapper_NgStyle(this.parentView.parentView.injectorGet(import29.KeyValueDiffers,this.parentView.parentIndex),new import17.ElementRef(this._el_9),this.renderer);
    this._text_10 = this.renderer.createText(this._el_9,'0',(null as any));
    this._el_11 = import10.createRenderElement(this.renderer,this._el_2,'span',import10.EMPTY_INLINE_ARRAY,(null as any));
    this._text_12 = this.renderer.createText(this._el_11,'',(null as any));
    this._text_13 = this.renderer.createText(this._el_2,'\n                        ',(null as any));
    this._el_14 = import10.createRenderElement(this.renderer,this._el_2,'a',new import10.InlineArray2(2,'href','#'),(null as any));
    this._text_15 = this.renderer.createText(this._el_14,'\n                            ',(null as any));
    this._el_16 = import10.createRenderElement(this.renderer,this._el_14,'span',new import10.InlineArray2(2,'class','fa fa-angle-down'),(null as any));
    this._text_17 = this.renderer.createText(this._el_14,'\n                        ',(null as any));
    this._text_18 = this.renderer.createText(this._el_2,'\n                    ',(null as any));
    this._text_19 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this._el_20 = import10.createRenderElement(this.renderer,this._el_0,'div',new import10.InlineArray2(2,'class','ui-separator'),(null as any));
    this._text_21 = this.renderer.createText(this._el_20,'\n                        ',(null as any));
    this._el_22 = import10.createRenderElement(this.renderer,this._el_20,'a',new import10.InlineArray2(2,'href','#'),(null as any));
    this._text_23 = this.renderer.createText(this._el_22,'\n                            ',(null as any));
    this._el_24 = import10.createRenderElement(this.renderer,this._el_22,'span',new import10.InlineArray2(2,'class','fa fa-angle-up'),(null as any));
    this._text_25 = this.renderer.createText(this._el_22,'\n                        ',(null as any));
    this._text_26 = this.renderer.createText(this._el_20,'\n                        ',(null as any));
    this._el_27 = import10.createRenderElement(this.renderer,this._el_20,'span',import10.EMPTY_INLINE_ARRAY,(null as any));
    this._text_28 = this.renderer.createText(this._el_27,':',(null as any));
    this._text_29 = this.renderer.createText(this._el_20,'\n                        ',(null as any));
    this._el_30 = import10.createRenderElement(this.renderer,this._el_20,'a',new import10.InlineArray2(2,'href','#'),(null as any));
    this._text_31 = this.renderer.createText(this._el_30,'\n                            ',(null as any));
    this._el_32 = import10.createRenderElement(this.renderer,this._el_30,'span',new import10.InlineArray2(2,'class','fa fa-angle-down'),(null as any));
    this._text_33 = this.renderer.createText(this._el_30,'\n                        ',(null as any));
    this._text_34 = this.renderer.createText(this._el_20,'\n                    ',(null as any));
    this._text_35 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this._el_36 = import10.createRenderElement(this.renderer,this._el_0,'div',new import10.InlineArray2(2,'class','ui-minute-picker'),(null as any));
    this._text_37 = this.renderer.createText(this._el_36,'\n                        ',(null as any));
    this._el_38 = import10.createRenderElement(this.renderer,this._el_36,'a',new import10.InlineArray2(2,'href','#'),(null as any));
    this._text_39 = this.renderer.createText(this._el_38,'\n                            ',(null as any));
    this._el_40 = import10.createRenderElement(this.renderer,this._el_38,'span',new import10.InlineArray2(2,'class','fa fa-angle-up'),(null as any));
    this._text_41 = this.renderer.createText(this._el_38,'\n                        ',(null as any));
    this._text_42 = this.renderer.createText(this._el_36,'\n                        ',(null as any));
    this._el_43 = import10.createRenderElement(this.renderer,this._el_36,'span',import10.EMPTY_INLINE_ARRAY,(null as any));
    this._NgStyle_43_3 = new import31.Wrapper_NgStyle(this.parentView.parentView.injectorGet(import29.KeyValueDiffers,this.parentView.parentIndex),new import17.ElementRef(this._el_43),this.renderer);
    this._text_44 = this.renderer.createText(this._el_43,'0',(null as any));
    this._el_45 = import10.createRenderElement(this.renderer,this._el_36,'span',import10.EMPTY_INLINE_ARRAY,(null as any));
    this._text_46 = this.renderer.createText(this._el_45,'',(null as any));
    this._text_47 = this.renderer.createText(this._el_36,'\n                        ',(null as any));
    this._el_48 = import10.createRenderElement(this.renderer,this._el_36,'a',new import10.InlineArray2(2,'href','#'),(null as any));
    this._text_49 = this.renderer.createText(this._el_48,'\n                            ',(null as any));
    this._el_50 = import10.createRenderElement(this.renderer,this._el_48,'span',new import10.InlineArray2(2,'class','fa fa-angle-down'),(null as any));
    this._text_51 = this.renderer.createText(this._el_48,'\n                        ',(null as any));
    this._text_52 = this.renderer.createText(this._el_36,'\n                    ',(null as any));
    this._text_53 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this._anchor_54 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_54 = new import27.ViewContainer(54,0,this,this._anchor_54);
    this._TemplateRef_54_5 = new import34.TemplateRef_(this,54,this._anchor_54);
    this._NgIf_54_6 = new import33.Wrapper_NgIf(this._vc_54.vcRef,this._TemplateRef_54_5);
    this._text_55 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    var disposable_0:Function = import10.subscribeToRenderElement(this,this._el_4,new import10.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_4));
    var disposable_1:Function = import10.subscribeToRenderElement(this,this._el_14,new import10.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_14));
    var disposable_2:Function = import10.subscribeToRenderElement(this,this._el_38,new import10.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_38));
    var disposable_3:Function = import10.subscribeToRenderElement(this,this._el_48,new import10.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_48));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._el_4,
      this._text_5,
      this._el_6,
      this._text_7,
      this._text_8,
      this._el_9,
      this._text_10,
      this._el_11,
      this._text_12,
      this._text_13,
      this._el_14,
      this._text_15,
      this._el_16,
      this._text_17,
      this._text_18,
      this._text_19,
      this._el_20,
      this._text_21,
      this._el_22,
      this._text_23,
      this._el_24,
      this._text_25,
      this._text_26,
      this._el_27,
      this._text_28,
      this._text_29,
      this._el_30,
      this._text_31,
      this._el_32,
      this._text_33,
      this._text_34,
      this._text_35,
      this._el_36,
      this._text_37,
      this._el_38,
      this._text_39,
      this._el_40,
      this._text_41,
      this._text_42,
      this._el_43,
      this._text_44,
      this._el_45,
      this._text_46,
      this._text_47,
      this._el_48,
      this._text_49,
      this._el_50,
      this._text_51,
      this._text_52,
      this._text_53,
      this._anchor_54,
      this._text_55
    ]
    ),[
      disposable_0,
      disposable_1,
      disposable_2,
      disposable_3
    ]
    );
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import35.NgStyle) && ((9 <= requestNodeIndex) && (requestNodeIndex <= 10)))) { return this._NgStyle_9_3.context; }
    if (((token === import35.NgStyle) && ((43 <= requestNodeIndex) && (requestNodeIndex <= 44)))) { return this._NgStyle_43_3.context; }
    if (((token === import34.TemplateRef) && (54 === requestNodeIndex))) { return this._TemplateRef_54_5; }
    if (((token === import36.NgIf) && (54 === requestNodeIndex))) { return this._NgIf_54_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_9_0_0:any = this._map_61(((this.parentView.context.currentHour < 10)? 'inline': 'none'));
    this._NgStyle_9_3.check_ngStyle(currVal_9_0_0,throwOnChange,false);
    this._NgStyle_9_3.ngDoCheck(this,this._el_9,throwOnChange);
    const currVal_43_0_0:any = this._map_63(((this.parentView.context.currentMinute < 10)? 'inline': 'none'));
    this._NgStyle_43_3.check_ngStyle(currVal_43_0_0,throwOnChange,false);
    this._NgStyle_43_3.ngDoCheck(this,this._el_43,throwOnChange);
    const currVal_54_0_0:any = (this.parentView.context.hourFormat == '12');
    this._NgIf_54_6.check_ngIf(currVal_54_0_0,throwOnChange,false);
    this._NgIf_54_6.ngDoCheck(this,this._anchor_54,throwOnChange);
    this._vc_54.detectChangesInNestedViews(throwOnChange);
    const currVal_62:any = import10.inlineInterpolate(1,'',this.parentView.context.currentHour,'');
    if (import10.checkBinding(throwOnChange,this._expr_62,currVal_62)) {
      this.renderer.setText(this._text_12,currVal_62);
      this._expr_62 = currVal_62;
    }
    const currVal_64:any = import10.inlineInterpolate(1,'',this.parentView.context.currentMinute,'');
    if (import10.checkBinding(throwOnChange,this._expr_64,currVal_64)) {
      this.renderer.setText(this._text_46,currVal_64);
      this._expr_64 = currVal_64;
    }
  }
  destroyInternal():void {
    this._vc_54.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import9.AppView<any> {
    if ((nodeIndex == 54)) { return new View_Calendar16(this.viewUtils,this,54,this._anchor_54,this._vc_54); }
    return (null as any);
  }
  handleEvent_4(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.context.incrementHour($event)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
  handleEvent_14(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.context.decrementHour($event)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
  handleEvent_38(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.context.incrementMinute($event)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
  handleEvent_48(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.context.decrementMinute($event)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
var renderType_Calendar:import11.RenderComponentType = import10.createRenderComponentType('',0,import12.ViewEncapsulation.None,styles_Calendar,{overlayState: Calendar_overlayState_factory});
export class View_Calendar0 extends import9.AppView<import1.Calendar> {
  _viewQuery_datepicker_0:import39.QueryList<any>;
  _text_0:any;
  _el_1:any;
  _NgClass_1_3:import25.Wrapper_NgClass;
  _NgStyle_1_4:import31.Wrapper_NgStyle;
  _text_2:any;
  _anchor_3:any;
  /*private*/ _vc_3:import27.ViewContainer;
  _TemplateRef_3_5:any;
  _NgIf_3_6:import33.Wrapper_NgIf;
  _text_4:any;
  _el_5:any;
  _NgClass_5_3:import25.Wrapper_NgClass;
  _NgStyle_5_4:import31.Wrapper_NgStyle;
  _text_6:any;
  _anchor_7:any;
  /*private*/ _vc_7:import27.ViewContainer;
  _TemplateRef_7_5:any;
  _NgIf_7_6:import33.Wrapper_NgIf;
  _text_8:any;
  _anchor_9:any;
  /*private*/ _vc_9:import27.ViewContainer;
  _TemplateRef_9_5:any;
  _NgIf_9_6:import33.Wrapper_NgIf;
  _text_10:any;
  _anchor_11:any;
  /*private*/ _vc_11:import27.ViewContainer;
  _TemplateRef_11_5:any;
  _NgIf_11_6:import33.Wrapper_NgIf;
  _text_12:any;
  _text_13:any;
  _text_14:any;
  _map_32:any;
  /*private*/ _expr_33:any;
  _map_34:any;
  _map_35:any;
  constructor(viewUtils:import10.ViewUtils,parentView:import9.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_Calendar0,renderType_Calendar,import14.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import15.ChangeDetectorStatus.CheckAlways);
    this._map_32 = import10.pureProxy2((p0:any,p1:any):{[key: string]:any} => {
      return {
        'ui-calendar': p0,
        'ui-calendar-w-btn': p1
      }
      ;
    });
    this._expr_33 = import8.UNINITIALIZED;
    this._map_34 = import10.pureProxy4((p0:any,p1:any,p2:any,p3:any):{[key: string]:any} => {
      return {
        'ui-datepicker-inline': p0,
        'ui-shadow': p1,
        'ui-state-disabled': p2,
        'ui-datepicker-timeonly': p3
      }
      ;
    });
    this._map_35 = import10.pureProxy1((p0:any):{[key: string]:any} => {
      return {display: p0};
    });
  }
  createInternal(rootSelector:string):import16.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._viewQuery_datepicker_0 = new import39.QueryList<any>();
    this._text_0 = this.renderer.createText(parentRenderNode,'\n        ',(null as any));
    this._el_1 = import10.createRenderElement(this.renderer,parentRenderNode,'span',import10.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_1_3 = new import25.Wrapper_NgClass(this.parentView.injectorGet(import28.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import29.KeyValueDiffers,this.parentIndex),new import17.ElementRef(this._el_1),this.renderer);
    this._NgStyle_1_4 = new import31.Wrapper_NgStyle(this.parentView.injectorGet(import29.KeyValueDiffers,this.parentIndex),new import17.ElementRef(this._el_1),this.renderer);
    this._text_2 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._anchor_3 = this.renderer.createTemplateAnchor(this._el_1,(null as any));
    this._vc_3 = new import27.ViewContainer(3,1,this,this._anchor_3);
    this._TemplateRef_3_5 = new import34.TemplateRef_(this,3,this._anchor_3);
    this._NgIf_3_6 = new import33.Wrapper_NgIf(this._vc_3.vcRef,this._TemplateRef_3_5);
    this._text_4 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_5 = import10.createRenderElement(this.renderer,this._el_1,'div',new import10.InlineArray2(2,'class','ui-datepicker ui-widget ui-widget-content ui-helper-clearfix ui-corner-all'),(null as any));
    this._NgClass_5_3 = new import25.Wrapper_NgClass(this.parentView.injectorGet(import28.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import29.KeyValueDiffers,this.parentIndex),new import17.ElementRef(this._el_5),this.renderer);
    this._NgStyle_5_4 = new import31.Wrapper_NgStyle(this.parentView.injectorGet(import29.KeyValueDiffers,this.parentIndex),new import17.ElementRef(this._el_5),this.renderer);
    this._text_6 = this.renderer.createText(this._el_5,'\n                ',(null as any));
    this._anchor_7 = this.renderer.createTemplateAnchor(this._el_5,(null as any));
    this._vc_7 = new import27.ViewContainer(7,5,this,this._anchor_7);
    this._TemplateRef_7_5 = new import34.TemplateRef_(this,7,this._anchor_7);
    this._NgIf_7_6 = new import33.Wrapper_NgIf(this._vc_7.vcRef,this._TemplateRef_7_5);
    this._text_8 = this.renderer.createText(this._el_5,'\n                ',(null as any));
    this._anchor_9 = this.renderer.createTemplateAnchor(this._el_5,(null as any));
    this._vc_9 = new import27.ViewContainer(9,5,this,this._anchor_9);
    this._TemplateRef_9_5 = new import34.TemplateRef_(this,9,this._anchor_9);
    this._NgIf_9_6 = new import33.Wrapper_NgIf(this._vc_9.vcRef,this._TemplateRef_9_5);
    this._text_10 = this.renderer.createText(this._el_5,'\n                ',(null as any));
    this._anchor_11 = this.renderer.createTemplateAnchor(this._el_5,(null as any));
    this._vc_11 = new import27.ViewContainer(11,5,this,this._anchor_11);
    this._TemplateRef_11_5 = new import34.TemplateRef_(this,11,this._anchor_11);
    this._NgIf_11_6 = new import33.Wrapper_NgIf(this._vc_11.vcRef,this._TemplateRef_11_5);
    this._text_12 = this.renderer.createText(this._el_5,'\n            ',(null as any));
    this._text_13 = this.renderer.createText(this._el_1,'\n        ',(null as any));
    this._text_14 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    var disposable_0:Function = import10.subscribeToRenderElement(this,this._el_5,new import10.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_5));
    this._viewQuery_datepicker_0.reset([new import17.ElementRef(this._el_5)]);
    this.context.overlayViewChild = this._viewQuery_datepicker_0.first;
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2,
      this._anchor_3,
      this._text_4,
      this._el_5,
      this._text_6,
      this._anchor_7,
      this._text_8,
      this._anchor_9,
      this._text_10,
      this._anchor_11,
      this._text_12,
      this._text_13,
      this._text_14
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import34.TemplateRef) && (3 === requestNodeIndex))) { return this._TemplateRef_3_5; }
    if (((token === import36.NgIf) && (3 === requestNodeIndex))) { return this._NgIf_3_6.context; }
    if (((token === import34.TemplateRef) && (7 === requestNodeIndex))) { return this._TemplateRef_7_5; }
    if (((token === import36.NgIf) && (7 === requestNodeIndex))) { return this._NgIf_7_6.context; }
    if (((token === import34.TemplateRef) && (9 === requestNodeIndex))) { return this._TemplateRef_9_5; }
    if (((token === import36.NgIf) && (9 === requestNodeIndex))) { return this._NgIf_9_6.context; }
    if (((token === import34.TemplateRef) && (11 === requestNodeIndex))) { return this._TemplateRef_11_5; }
    if (((token === import36.NgIf) && (11 === requestNodeIndex))) { return this._NgIf_11_6.context; }
    if (((token === import30.NgClass) && ((5 <= requestNodeIndex) && (requestNodeIndex <= 12)))) { return this._NgClass_5_3.context; }
    if (((token === import35.NgStyle) && ((5 <= requestNodeIndex) && (requestNodeIndex <= 12)))) { return this._NgStyle_5_4.context; }
    if (((token === import30.NgClass) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 13)))) { return this._NgClass_1_3.context; }
    if (((token === import35.NgStyle) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 13)))) { return this._NgStyle_1_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_33:any = (this.context.inline? 'visible': (this.context.overlayVisible? 'visible': 'hidden'));
    if (import10.checkBinding(throwOnChange,this._expr_33,currVal_33)) {
      var animationTransition_overlayState:any = this.componentType.animations['overlayState'](this,this._el_5,((this._expr_33 == import8.UNINITIALIZED)? 'void': this._expr_33),((currVal_33 == import8.UNINITIALIZED)? 'void': currVal_33));
      this._expr_33 = currVal_33;
    }
    const currVal_1_0_0:any = this.context.styleClass;
    this._NgClass_1_3.check_klass(currVal_1_0_0,throwOnChange,false);
    const currVal_1_0_1:any = this._map_32(true,this.context.showIcon);
    this._NgClass_1_3.check_ngClass(currVal_1_0_1,throwOnChange,false);
    this._NgClass_1_3.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_1_1_0:any = this.context.style;
    this._NgStyle_1_4.check_ngStyle(currVal_1_1_0,throwOnChange,false);
    this._NgStyle_1_4.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_3_0_0:boolean = !this.context.inline;
    this._NgIf_3_6.check_ngIf(currVal_3_0_0,throwOnChange,false);
    this._NgIf_3_6.ngDoCheck(this,this._anchor_3,throwOnChange);
    const currVal_5_0_0:any = 'ui-datepicker ui-widget ui-widget-content ui-helper-clearfix ui-corner-all';
    this._NgClass_5_3.check_klass(currVal_5_0_0,throwOnChange,false);
    const currVal_5_0_1:any = this._map_34(this.context.inline,!this.context.inline,this.context.disabled,this.context.timeOnly);
    this._NgClass_5_3.check_ngClass(currVal_5_0_1,throwOnChange,false);
    this._NgClass_5_3.ngDoCheck(this,this._el_5,throwOnChange);
    const currVal_5_1_0:any = this._map_35((this.context.inline? 'inline-block': (this.context.overlayVisible? 'block': 'none')));
    this._NgStyle_5_4.check_ngStyle(currVal_5_1_0,throwOnChange,false);
    this._NgStyle_5_4.ngDoCheck(this,this._el_5,throwOnChange);
    const currVal_7_0_0:boolean = !this.context.timeOnly;
    this._NgIf_7_6.check_ngIf(currVal_7_0_0,throwOnChange,false);
    this._NgIf_7_6.ngDoCheck(this,this._anchor_7,throwOnChange);
    const currVal_9_0_0:boolean = !this.context.timeOnly;
    this._NgIf_9_6.check_ngIf(currVal_9_0_0,throwOnChange,false);
    this._NgIf_9_6.ngDoCheck(this,this._anchor_9,throwOnChange);
    const currVal_11_0_0:any = (this.context.showTime || this.context.timeOnly);
    this._NgIf_11_6.check_ngIf(currVal_11_0_0,throwOnChange,false);
    this._NgIf_11_6.ngDoCheck(this,this._anchor_11,throwOnChange);
    this._vc_3.detectChangesInNestedViews(throwOnChange);
    this._vc_7.detectChangesInNestedViews(throwOnChange);
    this._vc_9.detectChangesInNestedViews(throwOnChange);
    this._vc_11.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_3.destroyNestedViews();
    this._vc_7.destroyNestedViews();
    this._vc_9.destroyNestedViews();
    this._vc_11.destroyNestedViews();
  }
  detachInternal():void {
    var animationTransition_overlayState:any = this.componentType.animations['overlayState'](this,this._el_5,this._expr_33,'void');
  }
  createEmbeddedViewInternal(nodeIndex:number):import9.AppView<any> {
    if ((nodeIndex == 3)) { return new View_Calendar1(this.viewUtils,this,3,this._anchor_3,this._vc_3); }
    if ((nodeIndex == 7)) { return new View_Calendar3(this.viewUtils,this,7,this._anchor_7,this._vc_7); }
    if ((nodeIndex == 9)) { return new View_Calendar10(this.viewUtils,this,9,this._anchor_9,this._vc_9); }
    if ((nodeIndex == 11)) { return new View_Calendar15(this.viewUtils,this,11,this._anchor_11,this._vc_11); }
    return (null as any);
  }
  handleEvent_5(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.context.onDatePickerClick($event)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
