// primeng/components/breadcrumb/breadcrumb.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from '@angular/core/src/linker/ng_module_factory';
import * as import1 from 'primeng/components/breadcrumb/breadcrumb';
import * as import2 from '@angular/common/src/common_module';
import * as import3 from '@angular/common/src/localization';
import * as import4 from '@angular/core/src/di/injector';
import * as import5 from '@angular/core/src/i18n/tokens';
import * as import6 from '@angular/core/src/change_detection/change_detection_util';
import * as import7 from '@angular/core/src/linker/view';
import * as import8 from '@angular/core/src/linker/view_utils';
import * as import9 from '@angular/core/src/render/api';
import * as import10 from '@angular/core/src/metadata/view';
import * as import11 from '@angular/core/src/linker/view_type';
import * as import12 from '@angular/core/src/change_detection/constants';
import * as import13 from '@angular/core/src/linker/component_factory';
import * as import14 from '@angular/router/src/router';
import * as import15 from '@angular/core/src/linker/view_container';
import * as import16 from '../../../@angular/common/src/directives/ng_class.ngfactory';
import * as import17 from '../../../@angular/common/src/directives/ng_if.ngfactory';
import * as import18 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import19 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import20 from '@angular/core/src/linker/element_ref';
import * as import21 from '@angular/core/src/linker/template_ref';
import * as import22 from '@angular/common/src/directives/ng_class';
import * as import23 from '@angular/common/src/directives/ng_if';
import * as import24 from '@angular/core/src/security';
import * as import25 from '../../../@angular/common/src/directives/ng_style.ngfactory';
import * as import26 from '../../../@angular/common/src/directives/ng_for.ngfactory';
import * as import27 from '@angular/common/src/directives/ng_for';
import * as import28 from '@angular/common/src/directives/ng_style';
class BreadcrumbModuleInjector extends import0.NgModuleInjector<import1.BreadcrumbModule> {
  _CommonModule_0:import2.CommonModule;
  _BreadcrumbModule_1:import1.BreadcrumbModule;
  __NgLocalization_2:import3.NgLocaleLocalization;
  constructor(parent:import4.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_2():import3.NgLocaleLocalization {
    if ((this.__NgLocalization_2 == null)) { (this.__NgLocalization_2 = new import3.NgLocaleLocalization(this.parent.get(import5.LOCALE_ID))); }
    return this.__NgLocalization_2;
  }
  createInternal():import1.BreadcrumbModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._BreadcrumbModule_1 = new import1.BreadcrumbModule();
    return this._BreadcrumbModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import1.BreadcrumbModule)) { return this._BreadcrumbModule_1; }
    if ((token === import3.NgLocalization)) { return this._NgLocalization_2; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const BreadcrumbModuleNgFactory:import0.NgModuleFactory<import1.BreadcrumbModule> = new import0.NgModuleFactory(BreadcrumbModuleInjector,import1.BreadcrumbModule);
export class Wrapper_Breadcrumb {
  /*private*/ _eventHandler:Function;
  context:import1.Breadcrumb;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  constructor(p0:any) {
    this._changed = false;
    this.context = new import1.Breadcrumb(p0);
    this._expr_0 = import6.UNINITIALIZED;
    this._expr_1 = import6.UNINITIALIZED;
    this._expr_2 = import6.UNINITIALIZED;
    this._expr_3 = import6.UNINITIALIZED;
  }
  ngOnDetach(view:import7.AppView<any>,componentView:import7.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
    this.context.ngOnDestroy();
  }
  check_model(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.model = currValue;
      this._expr_0 = currValue;
    }
  }
  check_style(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.style = currValue;
      this._expr_1 = currValue;
    }
  }
  check_styleClass(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_2,currValue))) {
      this._changed = true;
      this.context.styleClass = currValue;
      this._expr_2 = currValue;
    }
  }
  check_home(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_3,currValue))) {
      this._changed = true;
      this.context.home = currValue;
      this._expr_3 = currValue;
    }
  }
  ngDoCheck(view:import7.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    return changed;
  }
  checkHost(view:import7.AppView<any>,componentView:import7.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import7.AppView<any>,_eventHandler:any):void {
    this._eventHandler = _eventHandler;
  }
}
var renderType_Breadcrumb_Host:import9.RenderComponentType = import8.createRenderComponentType('',0,import10.ViewEncapsulation.None,([] as any[]),{});
class View_Breadcrumb_Host0 extends import7.AppView<any> {
  _el_0:any;
  compView_0:import7.AppView<import1.Breadcrumb>;
  _Breadcrumb_0_3:Wrapper_Breadcrumb;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_Breadcrumb_Host0,renderType_Breadcrumb_Host,import11.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import12.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import13.ComponentRef<any> {
    this._el_0 = import8.selectOrCreateRenderHostElement(this.renderer,'p-breadcrumb',import8.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this.compView_0 = new View_Breadcrumb0(this.viewUtils,this,0,this._el_0);
    this._Breadcrumb_0_3 = new Wrapper_Breadcrumb(this.injectorGet(import14.Router,this.parentIndex));
    this.compView_0.create(this._Breadcrumb_0_3.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import13.ComponentRef_<any>(0,this,this._el_0,this._Breadcrumb_0_3.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import1.Breadcrumb) && (0 === requestNodeIndex))) { return this._Breadcrumb_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._Breadcrumb_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
    this._Breadcrumb_0_3.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
export const BreadcrumbNgFactory:import13.ComponentFactory<import1.Breadcrumb> = new import13.ComponentFactory<import1.Breadcrumb>('p-breadcrumb',View_Breadcrumb_Host0,import1.Breadcrumb);
const styles_Breadcrumb:any[] = ([] as any[]);
class View_Breadcrumb1 extends import7.AppView<any> {
  _el_0:any;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import15.ViewContainer) {
    super(View_Breadcrumb1,renderType_Breadcrumb,import11.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import12.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import13.ComponentRef<any> {
    this._el_0 = import8.createRenderElement(this.renderer,(null as any),'li',new import8.InlineArray2(2,'class','ui-breadcrumb-chevron fa fa-chevron-right'),(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return (null as any);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_Breadcrumb3 extends import7.AppView<any> {
  _el_0:any;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import15.ViewContainer) {
    super(View_Breadcrumb3,renderType_Breadcrumb,import11.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import12.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import13.ComponentRef<any> {
    this._el_0 = import8.createRenderElement(this.renderer,(null as any),'li',new import8.InlineArray2(2,'class','ui-breadcrumb-chevron fa fa-chevron-right'),(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return (null as any);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_Breadcrumb2 extends import7.AppView<any> {
  _text_0:any;
  _el_1:any;
  _text_2:any;
  _el_3:any;
  _NgClass_3_3:import16.Wrapper_NgClass;
  _text_4:any;
  _el_5:any;
  _text_6:any;
  _text_7:any;
  _text_8:any;
  _text_9:any;
  _anchor_10:any;
  /*private*/ _vc_10:import15.ViewContainer;
  _TemplateRef_10_5:any;
  _NgIf_10_6:import17.Wrapper_NgIf;
  _text_11:any;
  /*private*/ _expr_16:any;
  _map_17:any;
  /*private*/ _expr_18:any;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import15.ViewContainer) {
    super(View_Breadcrumb2,renderType_Breadcrumb,import11.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import12.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_16 = import6.UNINITIALIZED;
    this._map_17 = import8.pureProxy1((p0:any):{[key: string]:any} => {
      return {'ui-state-disabled': p0};
    });
    this._expr_18 = import6.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import13.ComponentRef<any> {
    this._text_0 = this.renderer.createText((null as any),'\n                    ',(null as any));
    this._el_1 = import8.createRenderElement(this.renderer,(null as any),'li',new import8.InlineArray2(2,'role','menuitem'),(null as any));
    this._text_2 = this.renderer.createText(this._el_1,'\n                        ',(null as any));
    this._el_3 = import8.createRenderElement(this.renderer,this._el_1,'a',new import8.InlineArray2(2,'class','ui-menuitem-link'),(null as any));
    this._NgClass_3_3 = new import16.Wrapper_NgClass(this.parentView.parentView.injectorGet(import18.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import19.KeyValueDiffers,this.parentView.parentIndex),new import20.ElementRef(this._el_3),this.renderer);
    this._text_4 = this.renderer.createText(this._el_3,'\n                            ',(null as any));
    this._el_5 = import8.createRenderElement(this.renderer,this._el_3,'span',new import8.InlineArray2(2,'class','ui-menuitem-text'),(null as any));
    this._text_6 = this.renderer.createText(this._el_5,'',(null as any));
    this._text_7 = this.renderer.createText(this._el_3,'\n                        ',(null as any));
    this._text_8 = this.renderer.createText(this._el_1,'\n                    ',(null as any));
    this._text_9 = this.renderer.createText((null as any),'\n                    ',(null as any));
    this._anchor_10 = this.renderer.createTemplateAnchor((null as any),(null as any));
    this._vc_10 = new import15.ViewContainer(10,(null as any),this,this._anchor_10);
    this._TemplateRef_10_5 = new import21.TemplateRef_(this,10,this._anchor_10);
    this._NgIf_10_6 = new import17.Wrapper_NgIf(this._vc_10.vcRef,this._TemplateRef_10_5);
    this._text_11 = this.renderer.createText((null as any),'\n                ',(null as any));
    var disposable_0:Function = import8.subscribeToRenderElement(this,this._el_3,new import8.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_3));
    this.init(this._text_11,((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2,
      this._el_3,
      this._text_4,
      this._el_5,
      this._text_6,
      this._text_7,
      this._text_8,
      this._text_9,
      this._anchor_10,
      this._text_11
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import22.NgClass) && ((3 <= requestNodeIndex) && (requestNodeIndex <= 7)))) { return this._NgClass_3_3.context; }
    if (((token === import21.TemplateRef) && (10 === requestNodeIndex))) { return this._TemplateRef_10_5; }
    if (((token === import23.NgIf) && (10 === requestNodeIndex))) { return this._NgIf_10_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_3_0_0:any = 'ui-menuitem-link';
    this._NgClass_3_3.check_klass(currVal_3_0_0,throwOnChange,false);
    const currVal_3_0_1:any = this._map_17(this.context.$implicit.disabled);
    this._NgClass_3_3.check_ngClass(currVal_3_0_1,throwOnChange,false);
    this._NgClass_3_3.ngDoCheck(this,this._el_3,throwOnChange);
    const currVal_10_0_0:boolean = !this.context.last;
    this._NgIf_10_6.check_ngIf(currVal_10_0_0,throwOnChange,false);
    this._NgIf_10_6.ngDoCheck(this,this._anchor_10,throwOnChange);
    this._vc_10.detectChangesInNestedViews(throwOnChange);
    const currVal_16:any = (this.context.$implicit.url || '#');
    if (import8.checkBinding(throwOnChange,this._expr_16,currVal_16)) {
      this.renderer.setElementProperty(this._el_3,'href',this.viewUtils.sanitizer.sanitize(import24.SecurityContext.URL,currVal_16));
      this._expr_16 = currVal_16;
    }
    const currVal_18:any = import8.inlineInterpolate(1,'',this.context.$implicit.label,'');
    if (import8.checkBinding(throwOnChange,this._expr_18,currVal_18)) {
      this.renderer.setText(this._text_6,currVal_18);
      this._expr_18 = currVal_18;
    }
  }
  destroyInternal():void {
    this._vc_10.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._text_0,ctx);
    cb(this._el_1,ctx);
    cb(this._text_9,ctx);
    cb(this._vc_10.nativeElement,ctx);
    this._vc_10.visitNestedViewRootNodes(cb,ctx);
    cb(this._text_11,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import7.AppView<any> {
    if ((nodeIndex == 10)) { return new View_Breadcrumb3(this.viewUtils,this,10,this._anchor_10,this._vc_10); }
    return (null as any);
  }
  handleEvent_3(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.context.itemClick($event,this.context.$implicit)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
var renderType_Breadcrumb:import9.RenderComponentType = import8.createRenderComponentType('',0,import10.ViewEncapsulation.None,styles_Breadcrumb,{});
export class View_Breadcrumb0 extends import7.AppView<import1.Breadcrumb> {
  _text_0:any;
  _el_1:any;
  _NgClass_1_3:import16.Wrapper_NgClass;
  _NgStyle_1_4:import25.Wrapper_NgStyle;
  _text_2:any;
  _el_3:any;
  _text_4:any;
  _el_5:any;
  _text_6:any;
  _anchor_7:any;
  /*private*/ _vc_7:import15.ViewContainer;
  _TemplateRef_7_5:any;
  _NgIf_7_6:import17.Wrapper_NgIf;
  _text_8:any;
  _anchor_9:any;
  /*private*/ _vc_9:import15.ViewContainer;
  _TemplateRef_9_5:any;
  _NgFor_9_6:import26.Wrapper_NgFor;
  _text_10:any;
  _text_11:any;
  _text_12:any;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_Breadcrumb0,renderType_Breadcrumb,import11.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import12.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import13.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n        ',(null as any));
    this._el_1 = import8.createRenderElement(this.renderer,parentRenderNode,'div',import8.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_1_3 = new import16.Wrapper_NgClass(this.parentView.injectorGet(import18.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import19.KeyValueDiffers,this.parentIndex),new import20.ElementRef(this._el_1),this.renderer);
    this._NgStyle_1_4 = new import25.Wrapper_NgStyle(this.parentView.injectorGet(import19.KeyValueDiffers,this.parentIndex),new import20.ElementRef(this._el_1),this.renderer);
    this._text_2 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_3 = import8.createRenderElement(this.renderer,this._el_1,'ul',import8.EMPTY_INLINE_ARRAY,(null as any));
    this._text_4 = this.renderer.createText(this._el_3,'\n                ',(null as any));
    this._el_5 = import8.createRenderElement(this.renderer,this._el_3,'li',new import8.InlineArray2(2,'class','fa fa-home'),(null as any));
    this._text_6 = this.renderer.createText(this._el_3,'\n                ',(null as any));
    this._anchor_7 = this.renderer.createTemplateAnchor(this._el_3,(null as any));
    this._vc_7 = new import15.ViewContainer(7,3,this,this._anchor_7);
    this._TemplateRef_7_5 = new import21.TemplateRef_(this,7,this._anchor_7);
    this._NgIf_7_6 = new import17.Wrapper_NgIf(this._vc_7.vcRef,this._TemplateRef_7_5);
    this._text_8 = this.renderer.createText(this._el_3,'\n                ',(null as any));
    this._anchor_9 = this.renderer.createTemplateAnchor(this._el_3,(null as any));
    this._vc_9 = new import15.ViewContainer(9,3,this,this._anchor_9);
    this._TemplateRef_9_5 = new import21.TemplateRef_(this,9,this._anchor_9);
    this._NgFor_9_6 = new import26.Wrapper_NgFor(this._vc_9.vcRef,this._TemplateRef_9_5,this.parentView.injectorGet(import18.IterableDiffers,this.parentIndex),this.ref);
    this._text_10 = this.renderer.createText(this._el_3,'\n            ',(null as any));
    this._text_11 = this.renderer.createText(this._el_1,'\n        ',(null as any));
    this._text_12 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    var disposable_0:Function = import8.subscribeToRenderElement(this,this._el_5,new import8.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_5));
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2,
      this._el_3,
      this._text_4,
      this._el_5,
      this._text_6,
      this._anchor_7,
      this._text_8,
      this._anchor_9,
      this._text_10,
      this._text_11,
      this._text_12
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import21.TemplateRef) && (7 === requestNodeIndex))) { return this._TemplateRef_7_5; }
    if (((token === import23.NgIf) && (7 === requestNodeIndex))) { return this._NgIf_7_6.context; }
    if (((token === import21.TemplateRef) && (9 === requestNodeIndex))) { return this._TemplateRef_9_5; }
    if (((token === import27.NgFor) && (9 === requestNodeIndex))) { return this._NgFor_9_6.context; }
    if (((token === import22.NgClass) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 11)))) { return this._NgClass_1_3.context; }
    if (((token === import28.NgStyle) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 11)))) { return this._NgStyle_1_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1_0_0:any = this.context.styleClass;
    this._NgClass_1_3.check_klass(currVal_1_0_0,throwOnChange,false);
    const currVal_1_0_1:any = 'ui-breadcrumb ui-widget ui-widget-header ui-helper-clearfix ui-corner-all';
    this._NgClass_1_3.check_ngClass(currVal_1_0_1,throwOnChange,false);
    this._NgClass_1_3.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_1_1_0:any = this.context.style;
    this._NgStyle_1_4.check_ngStyle(currVal_1_1_0,throwOnChange,false);
    this._NgStyle_1_4.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_7_0_0:any = this.context.model;
    this._NgIf_7_6.check_ngIf(currVal_7_0_0,throwOnChange,false);
    this._NgIf_7_6.ngDoCheck(this,this._anchor_7,throwOnChange);
    const currVal_9_0_0:any = this.context.model;
    this._NgFor_9_6.check_ngForOf(currVal_9_0_0,throwOnChange,false);
    this._NgFor_9_6.ngDoCheck(this,this._anchor_9,throwOnChange);
    this._vc_7.detectChangesInNestedViews(throwOnChange);
    this._vc_9.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_7.destroyNestedViews();
    this._vc_9.destroyNestedViews();
  }
  createEmbeddedViewInternal(nodeIndex:number):import7.AppView<any> {
    if ((nodeIndex == 7)) { return new View_Breadcrumb1(this.viewUtils,this,7,this._anchor_7,this._vc_7); }
    if ((nodeIndex == 9)) { return new View_Breadcrumb2(this.viewUtils,this,9,this._anchor_9,this._vc_9); }
    return (null as any);
  }
  handleEvent_5(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.context.onHomeClick($event)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
