// primeng/components/inputswitch/inputswitch.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from '@angular/core/src/linker/ng_module_factory';
import * as import1 from 'primeng/components/inputswitch/inputswitch';
import * as import2 from '@angular/common/src/common_module';
import * as import3 from '@angular/common/src/localization';
import * as import4 from '@angular/core/src/di/injector';
import * as import5 from '@angular/core/src/i18n/tokens';
import * as import6 from '@angular/core/src/change_detection/change_detection_util';
import * as import7 from '@angular/core/src/linker/view';
import * as import8 from '@angular/core/src/linker/view_utils';
import * as import9 from '@angular/core/src/render/api';
import * as import10 from '@angular/core/src/metadata/view';
import * as import11 from 'primeng/components/dom/domhandler';
import * as import12 from '@angular/core/src/linker/view_type';
import * as import13 from '@angular/core/src/change_detection/constants';
import * as import14 from '@angular/core/src/linker/component_factory';
import * as import15 from '@angular/core/src/linker/element_ref';
import * as import16 from '@angular/forms/src/directives/control_value_accessor';
import * as import17 from '../../../@angular/common/src/directives/ng_class.ngfactory';
import * as import18 from '../../../@angular/common/src/directives/ng_style.ngfactory';
import * as import19 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import20 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import21 from '@angular/common/src/directives/ng_class';
import * as import22 from '@angular/common/src/directives/ng_style';
class InputSwitchModuleInjector extends import0.NgModuleInjector<import1.InputSwitchModule> {
  _CommonModule_0:import2.CommonModule;
  _InputSwitchModule_1:import1.InputSwitchModule;
  __NgLocalization_2:import3.NgLocaleLocalization;
  constructor(parent:import4.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_2():import3.NgLocaleLocalization {
    if ((this.__NgLocalization_2 == null)) { (this.__NgLocalization_2 = new import3.NgLocaleLocalization(this.parent.get(import5.LOCALE_ID))); }
    return this.__NgLocalization_2;
  }
  createInternal():import1.InputSwitchModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._InputSwitchModule_1 = new import1.InputSwitchModule();
    return this._InputSwitchModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import1.InputSwitchModule)) { return this._InputSwitchModule_1; }
    if ((token === import3.NgLocalization)) { return this._NgLocalization_2; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const InputSwitchModuleNgFactory:import0.NgModuleFactory<import1.InputSwitchModule> = new import0.NgModuleFactory(InputSwitchModuleInjector,import1.InputSwitchModule);
export class Wrapper_InputSwitch {
  /*private*/ _eventHandler:Function;
  context:import1.InputSwitch;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  subscription0:any;
  constructor(p0:any,p1:any) {
    this._changed = false;
    this.context = new import1.InputSwitch(p0,p1);
    this._expr_0 = import6.UNINITIALIZED;
    this._expr_1 = import6.UNINITIALIZED;
    this._expr_2 = import6.UNINITIALIZED;
    this._expr_3 = import6.UNINITIALIZED;
    this._expr_4 = import6.UNINITIALIZED;
  }
  ngOnDetach(view:import7.AppView<any>,componentView:import7.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
    (this.subscription0 && this.subscription0.unsubscribe());
  }
  check_onLabel(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.onLabel = currValue;
      this._expr_0 = currValue;
    }
  }
  check_offLabel(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.offLabel = currValue;
      this._expr_1 = currValue;
    }
  }
  check_disabled(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_2,currValue))) {
      this._changed = true;
      this.context.disabled = currValue;
      this._expr_2 = currValue;
    }
  }
  check_style(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_3,currValue))) {
      this._changed = true;
      this.context.style = currValue;
      this._expr_3 = currValue;
    }
  }
  check_styleClass(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import8.checkBinding(throwOnChange,this._expr_4,currValue))) {
      this._changed = true;
      this.context.styleClass = currValue;
      this._expr_4 = currValue;
    }
  }
  ngDoCheck(view:import7.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    return changed;
  }
  checkHost(view:import7.AppView<any>,componentView:import7.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import7.AppView<any>,_eventHandler:any,emit0:boolean):void {
    this._eventHandler = _eventHandler;
    if (emit0) { (this.subscription0 = this.context.onChange.subscribe(_eventHandler.bind(view,'onChange'))); }
  }
}
var renderType_InputSwitch_Host:import9.RenderComponentType = import8.createRenderComponentType('',0,import10.ViewEncapsulation.None,([] as any[]),{});
class View_InputSwitch_Host0 extends import7.AppView<any> {
  _el_0:any;
  compView_0:import7.AppView<import1.InputSwitch>;
  _DomHandler_0_3:import11.DomHandler;
  _InputSwitch_0_4:Wrapper_InputSwitch;
  __NG_VALUE_ACCESSOR_0_5:any[];
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_InputSwitch_Host0,renderType_InputSwitch_Host,import12.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import13.ChangeDetectorStatus.CheckAlways);
  }
  get _NG_VALUE_ACCESSOR_0_5():any[] {
    if ((this.__NG_VALUE_ACCESSOR_0_5 == null)) { (this.__NG_VALUE_ACCESSOR_0_5 = [this._InputSwitch_0_4.context]); }
    return this.__NG_VALUE_ACCESSOR_0_5;
  }
  createInternal(rootSelector:string):import14.ComponentRef<any> {
    this._el_0 = import8.selectOrCreateRenderHostElement(this.renderer,'p-inputSwitch',import8.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this.compView_0 = new View_InputSwitch0(this.viewUtils,this,0,this._el_0);
    this._DomHandler_0_3 = new import11.DomHandler();
    this._InputSwitch_0_4 = new Wrapper_InputSwitch(new import15.ElementRef(this._el_0),this._DomHandler_0_3);
    this.compView_0.create(this._InputSwitch_0_4.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import14.ComponentRef_<any>(0,this,this._el_0,this._InputSwitch_0_4.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import11.DomHandler) && (0 === requestNodeIndex))) { return this._DomHandler_0_3; }
    if (((token === import1.InputSwitch) && (0 === requestNodeIndex))) { return this._InputSwitch_0_4.context; }
    if (((token === import16.NG_VALUE_ACCESSOR) && (0 === requestNodeIndex))) { return this._NG_VALUE_ACCESSOR_0_5; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._InputSwitch_0_4.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
    if (!throwOnChange) {
      if ((this.numberOfChecks === 0)) { this._InputSwitch_0_4.context.ngAfterViewInit(); }
      this._InputSwitch_0_4.context.ngAfterViewChecked();
    }
  }
  destroyInternal():void {
    this.compView_0.destroy();
    this._InputSwitch_0_4.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
export const InputSwitchNgFactory:import14.ComponentFactory<import1.InputSwitch> = new import14.ComponentFactory<import1.InputSwitch>('p-inputSwitch',View_InputSwitch_Host0,import1.InputSwitch);
const styles_InputSwitch:any[] = ([] as any[]);
var renderType_InputSwitch:import9.RenderComponentType = import8.createRenderComponentType('',0,import10.ViewEncapsulation.None,styles_InputSwitch,{});
export class View_InputSwitch0 extends import7.AppView<import1.InputSwitch> {
  _text_0:any;
  _el_1:any;
  _NgClass_1_3:import17.Wrapper_NgClass;
  _NgStyle_1_4:import18.Wrapper_NgStyle;
  _text_2:any;
  _el_3:any;
  _text_4:any;
  _el_5:any;
  _text_6:any;
  _text_7:any;
  _text_8:any;
  _el_9:any;
  _text_10:any;
  _el_11:any;
  _text_12:any;
  _text_13:any;
  _text_14:any;
  _el_15:any;
  _NgClass_15_3:import17.Wrapper_NgClass;
  _text_16:any;
  _el_17:any;
  _text_18:any;
  _el_19:any;
  _text_20:any;
  _text_21:any;
  _text_22:any;
  _map_26:any;
  /*private*/ _expr_27:any;
  /*private*/ _expr_28:any;
  _map_29:any;
  constructor(viewUtils:import8.ViewUtils,parentView:import7.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_InputSwitch0,renderType_InputSwitch,import12.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import13.ChangeDetectorStatus.CheckAlways);
    this._map_26 = import8.pureProxy3((p0:any,p1:any,p2:any):{[key: string]:any} => {
      return {
        'ui-inputswitch ui-widget ui-widget-content ui-corner-all': p0,
        'ui-state-disabled': p1,
        'ui-inputswitch-checked': p2
      }
      ;
    });
    this._expr_27 = import6.UNINITIALIZED;
    this._expr_28 = import6.UNINITIALIZED;
    this._map_29 = import8.pureProxy2((p0:any,p1:any):{[key: string]:any} => {
      return {
        'ui-inputswitch-handle ui-state-default': p0,
        'ui-state-focus': p1
      }
      ;
    });
  }
  createInternal(rootSelector:string):import14.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n        ',(null as any));
    this._el_1 = import8.createRenderElement(this.renderer,parentRenderNode,'div',import8.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_1_3 = new import17.Wrapper_NgClass(this.parentView.injectorGet(import19.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import20.KeyValueDiffers,this.parentIndex),new import15.ElementRef(this._el_1),this.renderer);
    this._NgStyle_1_4 = new import18.Wrapper_NgStyle(this.parentView.injectorGet(import20.KeyValueDiffers,this.parentIndex),new import15.ElementRef(this._el_1),this.renderer);
    this._text_2 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_3 = import8.createRenderElement(this.renderer,this._el_1,'div',new import8.InlineArray2(2,'class','ui-inputswitch-off'),(null as any));
    this._text_4 = this.renderer.createText(this._el_3,'\n                ',(null as any));
    this._el_5 = import8.createRenderElement(this.renderer,this._el_3,'span',new import8.InlineArray2(2,'class','ui-inputswitch-offlabel'),(null as any));
    this._text_6 = this.renderer.createText(this._el_5,'',(null as any));
    this._text_7 = this.renderer.createText(this._el_3,'\n            ',(null as any));
    this._text_8 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_9 = import8.createRenderElement(this.renderer,this._el_1,'div',new import8.InlineArray2(2,'class','ui-inputswitch-on'),(null as any));
    this._text_10 = this.renderer.createText(this._el_9,'\n                ',(null as any));
    this._el_11 = import8.createRenderElement(this.renderer,this._el_9,'span',new import8.InlineArray2(2,'class','ui-inputswitch-onlabel'),(null as any));
    this._text_12 = this.renderer.createText(this._el_11,'',(null as any));
    this._text_13 = this.renderer.createText(this._el_9,'\n            ',(null as any));
    this._text_14 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_15 = import8.createRenderElement(this.renderer,this._el_1,'div',import8.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_15_3 = new import17.Wrapper_NgClass(this.parentView.injectorGet(import19.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import20.KeyValueDiffers,this.parentIndex),new import15.ElementRef(this._el_15),this.renderer);
    this._text_16 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_17 = import8.createRenderElement(this.renderer,this._el_1,'div',new import8.InlineArray2(2,'class','ui-helper-hidden-accessible'),(null as any));
    this._text_18 = this.renderer.createText(this._el_17,'\n                ',(null as any));
    this._el_19 = import8.createRenderElement(this.renderer,this._el_17,'input',new import8.InlineArray4(4,'readonly','readonly','type','checkbox'),(null as any));
    this._text_20 = this.renderer.createText(this._el_17,'\n            ',(null as any));
    this._text_21 = this.renderer.createText(this._el_1,'\n        ',(null as any));
    this._text_22 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    var disposable_0:Function = import8.subscribeToRenderElement(this,this._el_1,new import8.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_1));
    var disposable_1:Function = import8.subscribeToRenderElement(this,this._el_19,new import8.InlineArray4(4,'focus',(null as any),'blur',(null as any)),this.eventHandler(this.handleEvent_19));
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2,
      this._el_3,
      this._text_4,
      this._el_5,
      this._text_6,
      this._text_7,
      this._text_8,
      this._el_9,
      this._text_10,
      this._el_11,
      this._text_12,
      this._text_13,
      this._text_14,
      this._el_15,
      this._text_16,
      this._el_17,
      this._text_18,
      this._el_19,
      this._text_20,
      this._text_21,
      this._text_22
    ]
    ),[
      disposable_0,
      disposable_1
    ]
    );
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import21.NgClass) && (15 === requestNodeIndex))) { return this._NgClass_15_3.context; }
    if (((token === import21.NgClass) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 21)))) { return this._NgClass_1_3.context; }
    if (((token === import22.NgStyle) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 21)))) { return this._NgStyle_1_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1_0_0:any = this.context.styleClass;
    this._NgClass_1_3.check_klass(currVal_1_0_0,throwOnChange,false);
    const currVal_1_0_1:any = this._map_26(true,this.context.disabled,this.context.checked);
    this._NgClass_1_3.check_ngClass(currVal_1_0_1,throwOnChange,false);
    this._NgClass_1_3.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_1_1_0:any = this.context.style;
    this._NgStyle_1_4.check_ngStyle(currVal_1_1_0,throwOnChange,false);
    this._NgStyle_1_4.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_15_0_0:any = this._map_29(true,this.context.focused);
    this._NgClass_15_3.check_ngClass(currVal_15_0_0,throwOnChange,false);
    this._NgClass_15_3.ngDoCheck(this,this._el_15,throwOnChange);
    const currVal_27:any = import8.inlineInterpolate(1,'',this.context.offLabel,'');
    if (import8.checkBinding(throwOnChange,this._expr_27,currVal_27)) {
      this.renderer.setText(this._text_6,currVal_27);
      this._expr_27 = currVal_27;
    }
    const currVal_28:any = import8.inlineInterpolate(1,'',this.context.onLabel,'');
    if (import8.checkBinding(throwOnChange,this._expr_28,currVal_28)) {
      this.renderer.setText(this._text_12,currVal_28);
      this._expr_28 = currVal_28;
    }
  }
  handleEvent_1(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.context.toggle($event,this._el_19)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
  handleEvent_19(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'focus')) {
      const pd_sub_0:any = ((<any>this.context.onFocus($event)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'blur')) {
      const pd_sub_1:any = ((<any>this.context.onBlur($event)) !== false);
      result = (pd_sub_1 && result);
    }
    return result;
  }
}
